---
title: The genetic basis of fitness in \textit{Drosophila}&#58; A genome-wide association study
authors:
  - name: Heidi Wong
    department: Department of Computer Science
    affiliation: Cranberry-Lemon University
    location: Pittsburgh, PA 15213
    email: heidi@email.com
  - name: Luke Holman
    thanks: Previous address&#58; Melbourne. 
    department: School of Applied Sciences
    affiliation: Edinburgh Napier U
    location: Edinburgh, UK. 
    email: l.holman@napier.ac.uk
abstract: |
  Enter the text of your abstract here.
keywords:
  - blah
  - blee
  - bloo
  - these are optional and can be removed
  
bibliography: references.bib

biblio-style: unsrt
output: 
  rticles::arxiv_article:
    keep_tex: true
---

\section*{Introduction}
\lipsum[2]

 the \textit{Drosophila} Genetic Reference Panel (DGRP), a collection of almost entirely homozygous lines that represent a snapshot of natural genetic variation from a population in North Carolina (REF: Mackay et al., 2012)

\section*{Methods}

\subsection*{Fly stocks and husbandry}
Our study focused on 125 lines randomly selected from the DGRP. All flies were reared in 25mm vials with Hoff food medium (REF?), lightly sprinkled with dried yeast, at a temperature of 25^o^C. We verified the genotype of each DGRP line using the restriction-based assay PCR described in (Mackay et al., 2012) for the eight most diagnostic markers, and verifed the genotypes of the lines prior to data collection. In addition to the DGRP, we used two stocks carrying the visible markers, \textit{brown$^1$} (\textit{bw$^1$}) and \textit{P{FRT(w$^{hs}$)}G13 P{Ubi-GFP.nls}2R1 P{Ubi-GFP.nls}2R2} as mates and competitors for the DGRP flies; these stocks are hereafter termed _bw_ and _GFP_ respectively (_GFP_: green fluorescent protein). 


\subsection*{Measuring line means for male and female early- and later-life fitness}

_Overview_: We aimed to produce a holistic measure of the fitness of males and females from each DGRP line, both for young flies (i.e. those that had just reached reproductive maturity) and older individuals. We regard a 'fit female' _Drosophila_ as one that can survive in mixed-sex conditions, compete for food, and produce large numbers of eggs that hatch. We regard a fit male as one that that can similarly survive and feed, and which can outcompete other males in pre- and post-mating sexual selection. We therefore measured four phenotypic traits on each DGRP line using standardised assays, which for brevity we term early-life and late-life male and female fitness (while recognising that these are phenotypes that potentially correlate with fitness, and not fitness itself). 

We first highlight some features of the design of our fitness assays that influence how one should interpret the four measured phenotypes. Firstly, both assays were designed to foster social interactions and competition between adults, such that our fitness measures potentially depend on phenotypes with a social or competitive component (e.g. female resistance to male courtship, or competition over limited yeast food). Secondly, the focal flies and their same-sex competitors were not replaced if they died, such that our fitness measures incorporate variation in mortality as well as progeny production. Thirdly, we quantified fitness by counting the number of offspring hatched, which avoids confounding variation in adult fitness components with variation in larva-to-adult survival. Finally, the focal flies in our fitness assays were reared from first-instar larvae at a standardised density of 100 per vial, minimising non-genetic differences between lines. We also reared the _bw_ and _GFP_ individuals in a standardised fashion by placing 15 1- to 4-day-old mated females into yeasted vials, then collecting virgin offspring 36h later.

The fitness assays were run across nine blocks, and DGRP line 352 was included in every block, providing a reference point to help statistically estimate block effects on fitness. In each block, we measured the four phenotypes in 8-17 lines, plus the reference line, 352. 



_Female fitness assay_: 5 females from the focal DGRP line were placed in a food vial (termed the 'interaction vial') with 15 _GFP_ males and 10 _bw_ females (all flies were 2- to 3-day-old virgins). After allowing the flies to interact for 3 days, the 5 DGRP females were moved to an 'egg collection' vial filled with grape-agar medium to oviposit for 24h, before being returned to the original interaction vial with the _GFP_ and _bw_ flies. The eggs were allowed to develop into first instar larvae, where were then counted, giving the data used to estimate measure of early-life female fitness. To measure female late-life fitness, we waited 8 days (tipping once into a fresh vial), then replaced the old _GFP_ males with 15 new 2- to 3 days-old virgin _GFP_ males, and waited a further 3 days. The DGRP females were then moved to egg collection vials for 24h, and we again counted the total number of 1st instar larvae that emerged. 

To estimate the line mean for each female early- and late-life fitness, we fit a Bayesian multivariate generalised linear mixed model (GLMM, implemented in the R package `brms`), with the early- and late-life progeny counts as response variables and line, block, and group ID as crossed random effects (the random effects were assumed to have correlated effects on the two response variables, with the correlations being estimated from the data). The response variables followed a Poisson distribution with log link. We then used the fitted model to derive the posterior predicted values for the line mean for each trait. These predicted values, which correct the fitness estimates for block effects and account for pseudoreplication, were used in all downstream analyses of the female fitness line means. We derived predictions on the scale of linear predictor and scaled them to have a mean of zero and unit variance. We also used the GLMM to find the posterior intraclass correlation coefficient for the line random effect, i.e. the proportion of variation in female fitness that is explained by DGRP line. 

_Male fitness assay_: To measure male fitness, we placed 5 males from the focal DGRP line in an interaction vial with 10 _GFP_ males and 15 _bw_ females (all flies were 2- to 3-day-old virgins). After allowing the flies to interact and mate for 3 days, the females were moved to an egg collection vial for 24h. The females were then transferred back to their original interaction vial with the DGRP males and GFP males. After allowing the collected eggs to develop for 24h, and a random sample of up to 200 first instar larvae was collected and scored for GFP presence/absence, giving the data used to estimate male early-life fitness. The flies were then left for 8 days, and were tipped into a fresh interaction vial once during this time. Then, when the DGRP and GFP males were approximately 14 days old, the _bw_ females were replaced with 15 new 2- to 3-day-old virgin _bw_ females, and the flies were left to interact for 3 days. The females were then placed in a new egg collection vial to oviposit for 24h, and we scored the presence/absence of GFP in up to 200 larvae from these vials 24h later to yield the data for the late-life male fitness estimate.

We similarly used a multivariate GLMM to estimate the line mean male fitness. The procedure was similar to the model used for females, except that the response variable was the proportion of offspring sired (modelled using the binomial family and logit link), and we additionally corrected for the number of live competitor GFP males that were present at the time the females were removed for egg collection (by including the number of competitors as a covariate). Thus, we assume that the GFP males died randomly with respect to the genotype of the DGRP males, and adjusted the fitness of each line accordingly. We again derived the posterior predictions of the line means, and estimated the proportion of variance in siring success that was explained by line.


\subsection*{Quantifying genetic (co)variance}

Write?

\subsection*{Univariate GWAS and multivariate adaptive shrinkage}
In short, we performed a genome-wide association study (GWAS) using the software GEMMA (ref), and used multivariate adaptive shrinkage implemented in the R package `mashr` (REF) to process and interpret the GWAS results. 

We included in the GWAS all loci (SNPs and indels) that had been genotyped in at least 90% of the DGRP lines in our study and which had a minor allele frequency of $\ge5\%$. There were XX such loci, although only XX of these were in <100% linkage disequilibrium with all other loci; we therefore analysed these XX variants or variant groups (the latter were identified using PLINK). Missing genotypes were imputed using Beagle (REF). Using GEMMA's `lmm` function, we ran one one univariate linear mixed model per variant per phenotype, to estimate the variant effect sizes, standard errors, and $p$-values while adjusting for population structure via eigen decomposition of the genomic relatedness matrix. p-values were Benjamini-Hochberg corrected to control the false discovery rate.

Variant effect sizes are usually measured with considerable error in GWAS, and univariate GWAS ignores the correlations that likely exist between effects on the four phenotypes we measured. Furthermore, a key aim of our study is to estimate the frequency of sexuall-antagonistic and age-antagonisitic variants. The analysis using `mashr` can... 
and it can be valuable to run additional analyses on the effect sizes to estimate the true effect sizes, and to identify which effects are robust and mitigate 'winner's curse' effects (REF). 



We estimated selection on each variant using linear regression (REFS). The statistical model used to test each variant was a simple linear model with the formula Y ~ Genotype, where Y is relative fitness (e.g. the predicted line mean for male early-life fitness, divided by the average of all the predicted line means), and Genotype is the genotype of the focal line. Genotype was coded as a 0 for lines homozygous for the reference allele or 1 for lines homozygous the alternate allele for the focal variant - heterozygous loci are rare in the DGRP, and were excluded from analysis (because the genotype is unknown for these loci in our study). We defined the reference allele as the one that was most common across the entire panel of DGRP lines (n = 205), such that a positive slope means that the minor (i.e. rarer) allele confers higher fitness, and a negative slope means that the major (commoner) allele confers higher fitness. For each variant, we recorded the effect size, the associated standard error, and the t, df and p values for the test.

Our approach is equivalent to performing a GWAS with relative fitness as the response variable. The reason we did not use the GWAS pipeline generously provided online by the creators of the DGRP is that we wished to obtain effect size for every variant, and the pipeline only provides effect size for statistically significant variants. However, we did compare the results of our analysis with the results obtained by the Mackay lab’s pipeline, and obtained essentially identical results (e.g. our analysis identified the same statistically significant variant), suggesting that our approach were very similar. One difference is that the Mackay lab’s pipeline estimates the effects of each variant after correcting for the presence/absence of Wolbachia and chromosomal inversions in each line. We conducted pilot analyses which showed that including these variables yielded very similar results (because Wolbachia and chromosomal inversions were both unassociated with fitness in our study; p > 0.05), and so we elected to leave them out of our models for simplicity.

\subsection*{Measuring the specificity of selection across sexes and age classes}

We calculated a selection index, termed I, for each variant, by adapting the formula from Innocenti and Morrow (XX):

FORMULA

Innocenti and Morrow referred to I as an “index of sex-specific selection”, but I is equally useful for any study that measured selection in two categories of individuals; in our case, we calculated I to compare the effects of each variant on A) male and female fitness, and B) early life and late life fitness. 

When I is positive, selection is “concordant”, meaning that there is selection in both sexes or age classes, in the same direction (i.e. the same variant is associated with elevated fitness in both cases). When I is close to zero, selection is absent in one or both cases. When I is negative, selection is “antagonistic” meaning that there is selection in both sexes or age classes, but the variant that is associated with higher fitness is different in each case. We calculated I to compare four kinds of selection: A) males vs females, in early life; B) males vs females, in late life; C) young vs old males; D) young vs old females.

To numerically estimate the uncertainty associated with each estimate of I, we generated 1000 independent samples of b_i and b_j by drawing random numbers from a normal distribution with a mean and standard deviation obtained from the models used to estimate b_i and b_j. This yielded 1000 estimates of I, from which we recorded the median and 95% quantiles, which approximate the 95% confidence limits on I given the uncertainty associated with b_i and b_j.

\subsection*{Annotations for each variant}
We relied on the annotations generated by the creators of the DGRP, who used the software SnpEff to classify each variant by site class (e.g. whether the variant is in an intron, or a non-synonymous codon position, etc). Additionally, we assigned a list of KEGG and GO terms to each variant, which matched those associated with the gene (or genes) in which that variant resides (obtained from NCBI).

\section*{Results}

\subsection*{Variance and covariance in fitness across lines}

There was substantial variation in line mean relative fitness, for both sexes and both age classes (Figure 1).  All correlations in Figure 1 are positive and significant (p < 0.001), indicating that lines with high male fitness tended to have high female fitness, and lines with high early-life fitness tended to have high late-life fitness. 

In spite of the overall positive correlations between sexes and age classes, some lines ranked highly for female fitness had low male fitness, and some that ranked highly for early life fitness had low late-life fitness (and vice versa; Figure 2). In sum, the data suggest that the majority of genetic variance in fitness is concordant across sexes and age classes, but alleles with antagonistic fitness effects may nevertheless exist.

\subsection*{Genetic variance and covariance in fitness}

The estimated G matrix for our four traits is shown in Table 1. Fitness was highly heritable…

\subsection*{Distribution of fitness effects across variants}

Figure 3 plots the estimated effects of each of the XX variants on each of the four fitness traits: positive numbers mean that the commonest variant was associated with elevated fitness, and negative numbers indicate the reverse. 

Variants had smaller average effects on relative fitness in males relative to females, and in young individuals relative to older ones. There was positive covariance for all combinations, such that variants that positively affected one fitness component tended to positively affect another (Figure 3; Table XX). Additionally, we detected XX variants that affected fitness in two fitness components, but all of these affected fitness in a concordant rather than antagonistic fashion; for example, we did not find any alleles that elevated male fitness but reduced female fitness. 

The mean effect on fitness across all variants was slightly negative for all traits, which means that the minor alleles were, on average, associated with lower fitness than the major alleles (Table XX). Additionally, the distribution of fitness effects was significantly positively skewed for all four fitness measures, such that there were was an excess of loci with highly positive fitness effects relative to the number with highly negative effects (Figure 3; Table XX). This means that among the subset of loci with extreme effects on fitness, the beneficial allele tended to be the minor rather than major allele.


\subsection*{Almost all loci affect fitness, or are close to a locus that does}
Inspired by Boyle et al. (xx), we sorted all of the variants by their fitness effects, placed them in bins of 1000, and then calculated the average fitness effect for each bin. Figure 4 shows that there was a very tight correlation (XXX) between the average effects of the variants in each bin on male and female fitness. 
As well as reaffirming our earlier results that there is a positive genetic correlation between male and female fitness, these results reveal that fitness is an extremely polygenic trait. The male and female fitness measurements were collected independently, and so Figure 4 allows us to distinguish small but genuine effects from statistical noise, despite the low power of our study (and most GWAS) to detect variants with weak effects. To see why, consider an alternative hypothesis, in which the great majority of variants have no effect on fitness, and the genetic (co)variance in fitness reported above resulted from a small subset (dozens or hundreds) of variants with comparatively large average effects. The plot in Figure 4 would then be flat in the centre with steep inflections at each end. The straight line that we see instead suggests that there a large number of variants that each affect fitness - typically in both sexes, in the same direction - whose effect sizes range from tiny to moderate (see Boyle et al. XXX). Put another way, the effect size of each 1000-variant bin on female fitness was replicated for male fitness, suggesting that the effects are real, as opposed to being deviations from a true effect of zero caused by statistical uncertainty.

\subsection*{Selection estimates and allele frequencies are correlated}
At loci for which the minor allele was associated with higher fitness, the minor allele tended to be more common across the DGRP. This confirms the intuitive prediction that...


\subsection*{Distribution of fitness effects across chromosomes}
To do...

\section*{Discussion}



\newpage
\section*{Tables}

**Table 1**: List of variables, and their corresponding parameter(s) in the model, which were varied in order to study their effects on extinction.
# ```{r xtable, results="asis"}
# print_table1()
# ```

\newpage
\section*{Figures}

<!-- \begin{figure}[h] -->
<!-- \centering -->
<!-- \includegraphics[width=1.0\textwidth]{../figures/figure1_moths.pdf} -->
<!-- \caption{\footnotesize{Some hypothetical \textit{Z}-linked suppression drives considered in this study. Panel A illustrates normal inheritance of sex chromosomes in a wild type \textit{ZW} female (assumed to be mated to a wild type \textit{ZZ} male; not shown): the offspring sex ratio is even. In panel B, the female carries a \textit{W}-shredder allele ($Z^*$) that kills gametes or offspring early enough that missing daughters are replaced with more $Z^*$-bearing sons. In panel C, the lost daughters are not replaced, though their absence increases the survival probability of the sons somewhat (shown by their larger size), causing super-Mendelian inheritance of the $Z^*$ allele. Lastly, panel D shows a \textit{Z}-linked female-sterilising allele (e.g. an allele that cleaves the \textit{W} chromosome or a female-essential gene in somatic cells); since it is strongly disadvantageous in females, such an allele would go extinct unless it benefits from gene drive in heterozygous males.}} -->
<!-- \end{figure} -->
<!-- \newpage -->



