---
title: "Analysis of eQTLs"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r libraries, message=FALSE, warning=FALSE, results='hide'}
#library(plyr)
library(tidyverse)
library(glue)
library(future.apply)
library(lme4)
options(stringsAsFactors = FALSE)

# Load the wolbachia infection status data from the DGRP website
# wolbachia <- read_csv("data/input/wolbachia.csv") %>% 
#   mutate(line = paste("line", line, sep = ""))

db <- DBI::dbConnect(RSQLite::SQLite(), "data/derived/annotations.sqlite3")

# Load the chomosomal inversion data from the DGRP website
# these are the 5 inversions that Huang et al. PNAS corrected for
# inversions <- read_csv("data/input/inversion genotypes.csv") %>%
#     mutate(line = paste("line", line, sep = "")) %>%
#     select(line, `In(2L)t`, `In(2R)NS`, `In(3R)P`, `In(3R)K`, `In(3R)Mo`) 
```

## Define helper functions
```{r}
# Helper to run shell commands
run_command <- function(shell_command, wd = getwd(), path = ""){
  calling_env <- parent.frame()
  cat(system(glue("cd ", wd, path, "\n",shell_command, .envir = calling_env), intern = TRUE), sep = '\n')
}

# For loading the Huang et al microarray data
load_expression_data <- function(sex){
  
  if(sex != "both"){
    expression <- glue("data/input/huang_transcriptome/dgrp.array.exp.{sex}.txt") %>% read_delim(delim = " ")
    sample_names <- names(expression)[names(expression) != "gene"] %>% str_remove(":[12]") 
    gene_names <- expression$gene
    expression <- expression %>% select(-gene) %>% as.matrix() %>% t() 
    rownames(expression) <- sample_names # rows are samples, columns are genes
    colnames(expression) <- gene_names 
    return(expression %>% as.data.frame() %>% tibble::rownames_to_column("line") %>% as_tibble() %>% mutate(line = str_remove_all(line, "[.]1")))
  }
  
  females <- read_delim("data/input/huang_transcriptome/dgrp.array.exp.female.txt", delim = " ")
  names(females)[-1] <- paste("F_", names(females)[-1],sep="") #%>% str_remove(":[12]") 
  females <- females %>% left_join(read_delim("data/input/huang_transcriptome/dgrp.array.exp.male.txt", delim = " "), by = "gene")
  sample_names <- names(females)[names(females) != "gene"] %>% str_remove(":[12]") 
  gene_names <- females$gene
  sex <- ifelse(str_detect(sample_names, "F_"), "female", "male")
  line <- str_remove_all(sample_names, "F_")
  females <- females %>% select(-gene) %>% t()
  colnames(females) <- gene_names
  list(
    sampleIDs = tibble(sex, line),
    expression = females
  )
}

# Find the line means, and standardise the line mean expression levels to mean = 0, var = 1, separately for each sex
# Quicker than the built-in scale() function? No as.numeric needed, anyway.
quick_scale <- function(x) (x - .Internal(mean(x))) / sd(x)

# Get transcript data for both sexes for specific transcript(s)
get_transcript_data <- function(FBID){
  if(FBID == "all") return(predicted_line_mean_expression)
  predicted_line_mean_expression %>% select(line, starts_with(!!FBID))
}

# Edit the DGRP (all lines) .fam file to add the 18,000 transcripts, for both sexes (so, add 36,000 extra columns) 
replace_pheno_data <-function(FBID) {
  focal <- read_delim("data/derived/dgrp2_QC_all_lines.fam", delim = " ", col_names = FALSE)[,1:5] %>%
    left_join(get_transcript_data(FBID), by = c("X1" = "line")) 
  
  for(i in 6:ncol(focal)) focal[is.na(focal[,i]), i] <- -9 # Replace NA with -9, for PLINK
  
  write.table(focal, file = "data/derived/dgrp2_QC_all_lines.fam", 
              col.names = FALSE, row.names = FALSE, 
              quote = FALSE, sep = " ")
}
```

## Make the expression trait phenotype file

This file is set up for PLINK. The first two columns have the line name, and the next 36,000 have all the female and male line mean expression values (i.e. the average of the two samples per sex per line, for each of the 18,000 transcripts).
```{r}
make_phenotype_file <- function(){
  expression_data <- load_expression_data("both")
  expression_data <- as_tibble(cbind(expression_data[[1]], expression_data[[2]]))
  expression_data$line <- gsub("_", "", expression_data$line) 
  
  expression_data <- expression_data %>%
    group_by(line, sex) %>%
    summarise_all(mean)
  
  expression_data <- left_join(
    expression_data %>% filter(sex == "female") %>% select(-sex) %>% rename_all(function(x) paste(x, "female", sep = "_")) %>% rename(line = line_female),
    expression_data %>% filter(sex == "male") %>% select(-sex) %>% rename_all(function(x) paste(x, "male", sep = "_")) %>% rename(line = line_male),
    by = "line")
  for(i in 2:ncol(expression_data)) expression_data[is.na(expression_data[,i]), i] <- -9
  namess <- names(expression_data)[-1]
  
  expression_data <- expression_data %>%
    mutate(line2 = line) %>%
    select(line, line2, !! sort(namess)) 
  write_delim(data.frame(n = 1:length(namess), trait = namess), "data/derived/eQTL_results/eQTL_traits.txt", col_names = FALSE, delim = " ")
  write_delim(expression_data, "data/derived/eQTL_results/eQTL_phenotypes.txt", col_names = FALSE, delim = " ")
}

if(!file.exists("data/derived/eQTL_results/eQTL_traits.txt")) make_phenotype_file()
```

## Make the covariates file

```{r}
make_covariates_files <- function(){
  bed <- "dgrp2_QC_all_lines -maf 0.05"
  
  # Calculate the centered GRM 
  run_command("gemma -bfile {bed} -gk 1 -o GRM_transcriptome", path = "/data/derived")
  
  # Perform decomposition of the GRM, and save its eigenvalues and eigenvectors
  run_command("gemma -bfile {bed} -k ./output/GRM_transcriptome.cXX.txt -eigen -o eigen_decomp_transcriptome", path = "/data/derived")
  
  # Load the top 10 eigenvectors of the GRM
  eigenvectors <- read_tsv("data/derived/output/eigen_decomp_transcriptome.eigenU.txt", col_names = FALSE) 
  lines_in_fam <- read_tsv("data/derived/dgrp2_QC_all_lines.fam", col_names = FALSE) %>% # 185 lines with transcriptome data 
    filter(!str_detect(X1, "0 0 2 -9 -9")) %>%
    mutate(line = str_extract(X1, "line[:digit:]+")) %>% select(line)
  eigenvectors <- eigenvectors[, (ncol(eigenvectors) - 9):ncol(eigenvectors)]
  names(eigenvectors) <- paste("PC", 10:1, sep = "")
  eigenvectors <- bind_cols(lines_in_fam, eigenvectors)
  
  # Set up the chromosomal inversion and Wolbachia covariate data 
  wolbachia <- wolbachia %>%
    mutate(wolbachia = as.numeric(ifelse(wolbachia == "y", 1, 0))) 
  
  
  # Bind the expression data to the 10 PCs, inversion, and wolbachia data
  covariates <- eigenvectors %>%
    left_join(wolbachia, by = "line") %>%
    mutate(FID = line, IID = line) %>%
    select(FID, IID, wolbachia, starts_with("PC"))
  
  covariates %>% write_delim("data/derived/eQTL_results/covariates.txt", delim = " ")
}

if(!file.exists("data/derived/eQTL_results/covariates.txt")) make_covariates_files()
```


<!-- ## Load up the expression data  -->

<!-- This microarray data from Huang et al. _PNAS_ was downloaded from the [DGRP website](http://dgrp2.gnets.ncsu.edu/data.html).  -->


<!-- <!-- imilarly to that study, we first estimated the line mean expression level for each transcript in each sex using a linear mixed model that corrects for the _Wolbachia_ infection and the first 10 principal components of the Genomic Relatedness Matrix. A notable difference is that we use a single model for both sexes, with line as a random intercept and sex as a random slope -- this allows the model to use the information contained in the inter-sex correlation when estimating the BLUPs for each line. --> -->

<!-- <!-- The following code chunk produces the object `predicted_line_mean_expression`, which contains the estimated expression value for each combination of sex and transcript, on a normalised scale (i.e. with mean 0 and SD 1). For some transcripts, the predicted expression values are identical for both sexes in each line, but there is variation among lines. For others, the expression values are equal in magnitude but opposite in sign within lines between sexes. The former occurs when the 'sex' random slope explains little variation; the latter occurs when the 'line' random intercept explains little variation. Note that neither situation means mean there is no sex difference in expression: just that there is no detectable between-line variation that affects expression in a sex-specific manner (former), or that the only detectable genetic variance is that which changes the sexual dimorphism (latter). --> -->

<!-- ```{r} -->

<!-- if(!file.exists("data/derived/predicted_line_mean_expression.rds")){ -->
<!--   expression_data <- load_expression_data("both") -->
<!--   expression_data <- as_tibble(cbind(expression_data[[1]], expression_data[[2]])) -->
<!--   expression_data$line <- gsub("_", "", expression_data$line)  -->

<!--   # Calculate the centered GRM for the 185 lines with transcriptome data -->
<!--   run_command("gemma -bfile {bed} -gk 1 -o GRM_transcriptome", path = "/data/derived") -->

<!--   # Perform decomposition of the GRM, and save its eigenvalues and eigenvectors -->
<!--   run_command("gemma -bfile {bed} -k ./output/GRM_transcriptome.cXX.txt -eigen -o eigen_decomp_transcriptome", path = "/data/derived") -->

<!--   # Load the top 10 eigenvectors of the GRM -->
<!--   eigenvectors <- read_tsv("data/derived/output/eigen_decomp_transcriptome.eigenU.txt", col_names = FALSE)  -->
<!--   lines_in_fam <- read_tsv("data/derived/dgrp2_QC_all_lines.fam", col_names = FALSE) %>% # 185 lines with transcriptome data  -->
<!--     filter(!str_detect(X1, "0 0 2 -9 -9")) %>% -->
<!--     mutate(line = str_extract(X1, "line[:digit:]+")) %>% select(line) -->
<!--   eigenvectors <- eigenvectors[, (ncol(eigenvectors) - 9):ncol(eigenvectors)] -->
<!--   names(eigenvectors) <- paste("PC", 10:1, sep = "") -->
<!--   eigenvectors <- bind_cols(lines_in_fam, eigenvectors) -->

<!--   # Set up the chromosomal inversion and Wolbachia covariate data  -->
<!--   wolbachia <- wolbachia %>% -->
<!--     mutate(wolbachia = as.numeric(ifelse(wolbachia == "y", 1, 0)))  -->

<!--   # inversions <- inversions %>% -->
<!--   #   rename(Inv_1 = `In(2L)t`, -->
<!--   #          Inv_2 = `In(2R)NS`, -->
<!--   #          Inv_3 = `In(3R)P`, -->
<!--   #          Inv_4 = `In(3R)K`, -->
<!--   #          Inv_5 = `In(3R)Mo`) -->


<!--   # Bind the expression data to the 10 PCs, inversion, and wolbachia data -->
<!--   expression_data <- expression_data %>% -->
<!--     left_join(eigenvectors, by = "line") %>% -->
<!--     left_join(inversions, by = "line") %>% -->
<!--     left_join(wolbachia, by = "line") -->

<!--   # Using parallelisation, fit a linear mixed model with lme4 package, and use it to estimate  -->
<!--   # line mean expression level for each sex, controlling for the PCs and wolbachia infection status -->
<!--   # This calculates BLUPs for the expression data, as in Huang et al., prior to GWAS to find the eQTLs -->
<!--   focal_cols <- which(substr(names(expression_data), 1, 4) %in% c("FBgn", "XLOC")) -->
<!--   jobs <- split(focal_cols, ceiling(seq_along(focal_cols) / 2400)) -->

<!--   predict_expression <- function(j){  -->
<!--     transcript <- names(expression_data)[j] -->
<!--     form <- as.formula(paste(transcript, "~ sex + wolbachia + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + Inv_1 + Inv_2 + Inv_3 + Inv_4 + Inv_5 + (sex | line)")) -->
<!--     model <- lmer(form, data = expression_data) -->
<!--     new <- expand.grid(sex = c("female", "male"), # Model and predict expression for uninfected line with 'average' multilocus genotype -->
<!--                        line = unique(expression_data$line), -->
<!--                        wolbachia = 0,  -->
<!--                        PC1 = 0, PC2 = 0, PC3 = 0, PC4 = 0, PC5 = 0,  -->
<!--                        PC6 = 0, PC7 = 0, PC8 = 0, PC9 = 0, PC10 = 0,  -->
<!--                        Inv_1 = "ST", Inv_2 = "ST", Inv_3 = "ST", Inv_4 = "ST", Inv_5 = "ST", -->
<!--                        stringsAsFactors = FALSE) -->
<!--     predicted_expression <- data.frame(new, expr = predict(model, newdata = new)) %>%  -->
<!--       select(line, sex, expr) %>% -->
<!--       spread(sex, expr) -->
<!--     names(predicted_expression)[2:3] <- c(paste(transcript, "female", sep = "_"), -->
<!--                                           paste(transcript, "male",   sep = "_")) -->
<!--     return(predicted_expression[, 2:3]) -->
<!--   } -->

<!--   plan("multicore") -->
<!--   predicted_line_mean_expression <- future_lapply(jobs, function(job_chunk) { -->
<!--     results <- vector(length(job_chunk), mode = "list") -->
<!--     for(i in 1:length(job_chunk)) { -->
<!--       results[[i]] <- predict_expression(job_chunk[i]) -->
<!--     } -->
<!--     do.call("cbind", results) -->
<!--   }) %>% do.call("cbind", .)  -->

<!--   # Fix the names (screwed up by future_lapply?) -->
<!--   names(predicted_line_mean_expression) <- substr(names(predicted_line_mean_expression), 3, nchar(names(predicted_line_mean_expression))) -->

<!--   # Scale all the transcript line means have mean 0, variance 1, and add the line names -->
<!--   predicted_line_mean_expression <- bind_cols(line = unique(expression_data$line),  -->
<!--                                               predicted_line_mean_expression %>% mutate_all(quick_scale)) -->

<!--   # Save it -->
<!--   write_rds(predicted_line_mean_expression, "data/derived/predicted_line_mean_expression.rds") -->
<!-- } else { # if already done, just load the file -->
<!--   predicted_line_mean_expression <- read_rds("data/derived/predicted_line_mean_expression.rds") -->
<!-- } -->
<!-- ``` -->



## Set up for running the eQTL linear model GWAS

### Identify clumps of SNPs in perfect LD
Here, we identify SNPs that are in <100% linkage disequilibrium with each other, since it's a waste of time to test these separately, as the estimates and p-values would be identical. This is necessary because it takes a long time to test every SNP on all 36k male and female transcripts.
```{r}
# Open a PLINK .assoc file as a tibble
open_plink_results <- function(file){
  read.table(file, sep = "" , header = TRUE, na.strings = "", stringsAsFactors = FALSE) %>% as_tibble()
}


if(!file.exists("data/derived/SNP_clumps_eQTL_analysis.txt")){
  
  # Analyse the first expression trait, with no covariates etc
  run_command("plink --bfile dgrp2_QC_all_lines --pheno eQTL_results/eQTL_phenotypes.txt --mpheno 1 --linear --allow-no-sex --out assoc_results", path = "/data/derived")

  eQTL_SNP_clumps <- open_plink_results("data/derived/assoc_results.assoc.linear") %>%
    mutate(result = paste(BETA, STAT,   P, sep = " ")) %>%
    select(SNP, result) %>%
    group_by(result) %>% 
    summarise(index_SNP = SNP[sample(n(), 1)], # index SNP is randomly selected from all the SNPs in perfect LD
              SNPs = paste0(SNP, collapse = "; ")) %>% 
    ungroup() %>% select(index_SNP, SNPs) %>% arrange(index_SNP)
  
  # There are 988322 total SNPs that pass quality control
  # This is reduced to 693,928 SNPs if we only run one test for each clump of SNPs in 100% LD (so, we can ignore 294394)
  eQTL_SNP_clumps %>% 
    write_tsv("data/derived/eQTL_results/SNP_clumps_eQTL_analysis.txt")
  
  rm(eQTL_SNP_clumps) # tidy up
  unlink(c("data/derived/output/assoc_results.log",
            "data/derived/output/assoc_results.assoc.linear"))
}
```

### Use PLINK to prepare stripped-down BIM/BED/FAM files

Here, we remove all the SNPs other than the 'index SNPs' of each of the SNP slumps that are in 100% LD. This reduced the number of SNPS substantially, greatly saving the computation time.
```{r}
plink  <- bigsnpr::download_plink()

read_tsv("data/derived/eQTL_results/SNP_clumps_eQTL_analysis.txt") %>% select(index_SNP) %>%
  write_tsv("data/derived/SNPs_to_keep.txt", col_names = FALSE)

run_command(glue("plink --bfile dgrp2_QC_all_lines",
                 " --extract SNPs_to_keep.txt", 
                 " --geno 0.1 --maf 0.05 --allow-no-sex",
                 " --make-bed --out dgrp2_QC_all_lines"), path = "/data/derived/")

unlink(c("data/derived/dgrp2_QC_all_lines.bim~", 
         "data/derived/dgrp2_QC_all_lines.bed~", 
         "data/derived/dgrp2_QC_all_lines.fam~", 
         "data/derived/dgrp2_QC_all_lines.log", 
         "data/derived/SNPs_to_keep.txt"))
```

## Run the eQTL analysis using PLINK

```{r}
# Get the 36k sex-specific transcripts that were measured in Huang et al's microarrays
FBID_by_sex <- read_delim("data/derived/eQTL_results/eQTL_traits.txt", delim = " ", col_names = c("n", "FBID"))

# Get the 18,000 transcripts
FBIDs <- gsub("_male", "", gsub("_female", "", FBID_by_sex$FBID)) %>% unique() %>% sort()

# Cull to the 5752 transcripts with heritability >= 0.2
heritable_FBIDs <- read_csv("data/input/huang_2015_tableS5_transcript_heritability.csv") %>%
  filter(female_narrow_heritability >= 0.2 | male_narrow_heritability >= 0.2) %>% pull(FBID)
FBIDs <- FBIDs[FBIDs %in% heritable_FBIDs]

# To prevent re-running:
FBIDs <- FBIDs[!(FBIDs %in% gsub("[.]rds", "", list.files("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/real_data")))]

do_one_eQTL <- function(focal_FBID, mode = "original_data"){
  
  if(mode == "original_data") {
    bfile <- "dgrp2_QC_all_lines"
    output_file <- glue("data/derived/eQTL_results/real_data/{focal_FBID}.rds")
  }
  if(mode == "shuffled_data") {
    bfile <- "dgrp2_QC_all_lines_shuffled"
    output_file <- glue("data/derived/eQTL_results/shuffled_data/{focal_FBID}.rds")
  }
  
  # Use the PCs and Wolbachia as covariates, and standardise the trait (so beta is in standard deviations)
  options <- glue("--bfile {bfile} --pheno eQTL_results/eQTL_phenotypes.txt --linear --covar eQTL_results/covariates.txt --allow-no-sex --standard-beta --noweb")
  
  columns <- FBID_by_sex %>% filter(grepl(focal_FBID, FBID)) 
  female_col <- columns %>% filter(grepl("female", FBID)) %>% pull(n)
  male_col <- columns %>% filter(!grepl("female", FBID)) %>% pull(n)
  outfile_female <- paste(focal_FBID, "_female", sep = "")
  outfile_male <- paste(focal_FBID, "_male", sep = "")

  run_command("plink {options} --mpheno {female_col} --out {outfile_female}", path = "/data/derived")
  run_command("plink {options} --mpheno {male_col} --out {outfile_male}", path = "/data/derived")
  
  results_females <- open_plink_results(glue("data/derived/{outfile_female}.assoc.linear")) %>% 
    filter(TEST == "ADD") %>% select(SNP, A1, BETA, STAT, P)
  results_males <- open_plink_results(glue("data/derived/{outfile_male}.assoc.linear")) %>% 
    filter(TEST == "ADD") %>% select(SNP, A1, BETA, STAT, P)
  
  hits_either_sex <- unique(c(
    results_females %>% filter(P < 0.000001) %>% pull(SNP), 
    results_males   %>% filter(P < 0.000001) %>% pull(SNP)))
  
  if(length(hits_either_sex) > 0){ # save the results for SNPs (putative eQTLs) that are sig in one or both sexes
    bind_rows(results_females %>% filter(SNP %in% hits_either_sex) %>% mutate(sex = "f"), 
              results_males %>% filter(SNP %in% hits_either_sex) %>% mutate(sex = "m")) %>% 
      mutate(FBID = focal_FBID) %>% 
      select(FBID, sex, everything()) %>%
      saveRDS(file = output_file)
  }
  
  try(unlink(glue("data/derived/{focal_FBID}_female.log"))) # delete the PLINK files
  try(unlink(glue("data/derived/{focal_FBID}_female.assoc.linear")))
  try(unlink(glue("data/derived/{focal_FBID}_male.log"))) 
  try(unlink(glue("data/derived/{focal_FBID}_male.assoc.linear")))
  return(NULL)
  
}
# system.time(do_one_eQTL(FBIDs[1]))
# readRDS("data/derived/eQTL_results/real_data/FBgn0000024.rds")

plan("multicore")
future_lapply(FBIDs, do_one_eQTL)

list.files("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/real_data", full.names = T) %>% 
  lapply(readRDS) %>% do.call("rbind", .) %>% 
  write_delim("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/all_eQTL_results.txt", delim = " ")
```


```{r}
do_snp_clumping_eQTLs <- function(){
  
  dat <- read_delim("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/all_eQTL_results.txt", delim = " ") 
  female <- dat %>% filter(sex == "f")
  male   <- dat %>% filter(sex == "m")
  rm(dat)
  
  # Clump eQTLs using the p-values for the effects on females
  write_delim(female, "~/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt", delim = " ")
  run_command(glue("plink --bfile dgrp2_QC_all_lines", 
                   " --clump ~/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt",
                   " --clump-field P --clump-p1 0.000001 --clump-p2 0.01"), # Each clump must contain one snp with p < 10 ^ -5
              path = "/data/derived")
  file.rename("data/derived/plink.clumped", "data/derived/eQTL_SNP_clumps_female.txt")
  unlink("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt")
  
  # Clump eQTLs using the p-values for the effects on males
  write_delim(male, "~/Rprojects/fitnessGWAS/data/derived/eQTL_results/male_eQTL_results.txt", delim = " ")
  run_command(glue("plink --bfile dgrp2_QC_all_lines", 
                   " --clump ~/Rprojects/fitnessGWAS/data/derived/eQTL_results/male_eQTL_results.txt",
                   " --clump-field P --clump-p1 0.000001 --clump-p2 0.01"), # Each clump must contain one snp with p < 10 ^ -5
              path = "/data/derived")
  file.rename("data/derived/plink.clumped", "data/derived/eQTL_SNP_clumps_male.txt")
  unlink("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/male_eQTL_results.txt")
  
  parse_clump_file <- function(file){
    clumps <- read.table(file, header = TRUE, stringsAsFactors = FALSE) %>% as_tibble() %>% 
      rename(index_SNP = SNP,
             index_SNP_p_value = P,
             num_SNPs_in_clump = TOTAL,
             not_sig = NSIG, # Number of clumped SNPs that are not significant ( p > 0.05 )
             sig_p05 = S05, # Number of clumped SNPs 0.01 < p < 0.05
             sig_p01 = S01, # Number of clumped SNPs 0.001 < p < 0.01
             sig_p001 = S001, # Number of clumped SNPs 0.0001 < p < 0.001 
             sig_p0001 = S0001, # Number of clumped SNPs p < 0.0001
             other_snps = SP2) %>%
      mutate(other_snps = gsub("\\(1\\),", ", ", other_snps),
             other_snps = gsub("\\(1\\)", "", other_snps),
             other_snps = replace(other_snps, other_snps == "NONE", NA)) %>%
      select(-CHR, -F, -BP) 
    
    SNPs <- apply(clumps, 1, function(x) {
      res <- c(x[1], strsplit(x[9], split = ", ")[[1]])
      res[!is.na(res)]
    })
    
    clumps$FBIDs <- lapply(SNPs, function(x) {
      focal <- tbl(db, "variants") %>% filter(SNP %in% x) %>% collect()
      if(nrow(focal) == 0) return(NA)
      focal %>% pull(FBID) %>% unique() %>% paste0(collapse = ", ")
    }) %>% unlist()
    
    clumps$genes <- lapply(clumps$FBIDs, function(x) {
      if(is.na(x)) return(NA)
      FB <- strsplit(x, split = ", ")[[1]]
      return(tbl(db, "genes") %>% filter(FBID %in% FB) %>% pull(gene_name) %>% paste0(collapse = ", "))
    }) %>% unlist()
    
    clumps
  }
  
  parse_clump_file("data/derived/eQTL_SNP_clumps_female.txt") %>% write_tsv("data/derived/eQTL_SNP_clumps_female.txt")
  parse_clump_file("data/derived/eQTL_SNP_clumps_male.txt") %>% write_tsv("data/derived/eQTL_SNP_clumps_male.txt")
  unlink("data/derived/plink.log")
}

do_snp_clumping_eQTLs()
```





```{r}
xx <- read_csv("~/Rprojects/fitnessGWAS/data/derived/eQTL_results/all_eQTL_results.csv") %>% arrange(P) 
ggplot(xx %>% distinct(FBID, sex, .keep_all = T), aes(sex, BETA, group = FBID)) + geom_line(alpha = 0.3)
xx %>% group_by(FBID, SNP) %>%
  summarise(both_sig = ifelse(all(P < 1e-5), "Yes", "No"),
            opposite = ifelse(sign(BETA[1] != BETA[2]), "Yes", "No")) %>%
  filter(both_sig == "Yes" & opposite == "Yes")
xx %>% distinct(FBID, sex, .keep_all = T) %>% select(FBID, sex, BETA) %>% tidyr::spread(sex, BETA) %>% ggplot(aes(f,m)) + geom_point(alpha=0.4)
#list.files("data/derived/eQTL_results/real_data",full.names = T) %>% lapply(readRDS) %>% do.call("rbind", .) %>% write_csv("data/derived/eQTL_results/all_eQTL_results.csv") 
```


@----------------------------------------------------------@
|        PLINK!       |     v1.07      |   10/Aug/2009     |
|----------------------------------------------------------|
|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |
|----------------------------------------------------------|
|  For documentation, citation & bug-report instructions:  |
|        http://pngu.mgh.harvard.edu/purcell/plink/        |
@----------------------------------------------------------@

Web-based version check ( --noweb to skip )
Connecting to web...  failed connection

Problem connecting to web

Writing this text to log file [ plink.log ]
Analysis started: Mon Feb  4 13:59:19 2019

Options in effect:
	--bfile dgrp2_QC_all_lines
	--clump /Users/lholman/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt
	--clump-field P
	--clump-p1 1

Reading map (extended format) from [ dgrp2_QC_all_lines.bim ] 
620428 markers to be included from [ dgrp2_QC_all_lines.bim ]
Reading pedigree information from [ dgrp2_QC_all_lines.fam ] 
205 individuals read from [ dgrp2_QC_all_lines.fam ] 
205 individuals with nonmissing phenotypes
Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)
Missing phenotype value is also -9
0 cases, 205 controls and 0 missing
0 males, 205 females, and 0 of unspecified sex
Reading genotype bitfile from [ dgrp2_QC_all_lines.bed ] 
Detected that binary PED file is v1.00 SNP-major mode
Before frequency and genotyping pruning, there are 620428 SNPs
205 founders and 0 non-founders found
Total genotyping rate in remaining individuals is 1
0 SNPs failed missingness test ( GENO > 1 )
0 SNPs failed frequency test ( MAF < 0 )
After frequency and genotyping pruning, there are 620428 SNPs
After filtering, 0 cases, 205 controls and 0 missing
After filtering, 0 males, 205 females, and 0 of unspecified sex

Parameters for --clump:
          p-value threshold for index SNPs = 1
      Physical (kb) threshold for clumping = 250
     LD (r-squared) threshold for clumping = 0.5
        p-value threshold for clumped SNPs = 0.01

Reading results for clumping from [ /Users/lholman/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt ]
Extracting fields SNP and P
Indexing on all files
Writing clumped results file to [ plink.clumped ]

>   run_command(glue("plink --bfile dgrp2_QC_all_lines", 
+                    " --clump ~/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt",
+                    " --clump-field P --clump-p1 0.000001 --clump-p2 0.01"), # Each clump must contain one snp with p < 10 ^ -5
+               path = "/data/derived")

@----------------------------------------------------------@
|        PLINK!       |     v1.07      |   10/Aug/2009     |
|----------------------------------------------------------|
|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |
|----------------------------------------------------------|
|  For documentation, citation & bug-report instructions:  |
|        http://pngu.mgh.harvard.edu/purcell/plink/        |
@----------------------------------------------------------@

Web-based version check ( --noweb to skip )
Recent cached web-check found...Problem connecting to web

Writing this text to log file [ plink.log ]
Analysis started: Mon Feb  4 14:00:07 2019

Options in effect:
	--bfile dgrp2_QC_all_lines
	--clump /Users/lholman/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt
	--clump-field P
	--clump-p1 0.000001
	--clump-p2 0.01

Reading map (extended format) from [ dgrp2_QC_all_lines.bim ] 
620428 markers to be included from [ dgrp2_QC_all_lines.bim ]
Reading pedigree information from [ dgrp2_QC_all_lines.fam ] 
205 individuals read from [ dgrp2_QC_all_lines.fam ] 
205 individuals with nonmissing phenotypes
Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)
Missing phenotype value is also -9
0 cases, 205 controls and 0 missing
0 males, 205 females, and 0 of unspecified sex
Reading genotype bitfile from [ dgrp2_QC_all_lines.bed ] 
Detected that binary PED file is v1.00 SNP-major mode
Before frequency and genotyping pruning, there are 620428 SNPs
205 founders and 0 non-founders found
Total genotyping rate in remaining individuals is 1
0 SNPs failed missingness test ( GENO > 1 )
0 SNPs failed frequency test ( MAF < 0 )
After frequency and genotyping pruning, there are 620428 SNPs
After filtering, 0 cases, 205 controls and 0 missing
After filtering, 0 males, 205 females, and 0 of unspecified sex

Parameters for --clump:
          p-value threshold for index SNPs = 1e-06
      Physical (kb) threshold for clumping = 250
     LD (r-squared) threshold for clumping = 0.5
        p-value threshold for clumped SNPs = 0.01

Reading results for clumping from [ /Users/lholman/Rprojects/fitnessGWAS/data/derived/eQTL_results/female_eQTL_results.txt ]
Extracting fields SNP and P
Indexing on all files
Writing clumped results file to [ plink.clumped ]
Formed 10373 clumps of the top 15143 SNPs, so far            

Analysis finished: Mon Feb  4 16:46:09 2019

<!-- ```{r} -->
<!-- make_shuffled_plink_data <- function(seed){ -->
<!--   set.seed(seed) -->

<!--   fam <- read_delim("data/derived/dgrp2_QC_all_lines.fam", delim = " ", col_names = FALSE) -->
<!--   non_missing_rows <- which(fam$X6 != -9) -->
<!--   shuffle_map <- data.frame(old = non_missing_rows, new = non_missing_rows[sample(length(non_missing_rows))]) -->
<!--   fam[shuffle_map$new, ] <- fam[shuffle_map$old, ] -->

<!--   file.copy("data/derived/dgrp2_QC_all_lines.bim", "data/derived/dgrp2_QC_all_lines_shuffled.bim") -->
<!--   file.copy("data/derived/dgrp2_QC_all_lines.bed", "data/derived/dgrp2_QC_all_lines_shuffled.bed") -->
<!--   write_delim(fam, "data/derived/dgrp2_QC_all_lines_shuffled.fam", delim = " ", col_names = FALSE) -->
<!-- } -->

<!-- make_shuffled_plink_data(1) -->
<!-- ``` -->



<!-- ## Run the eQTL analysis -->
<!-- This took a few days on my 8-core Mac.  -->
<!-- ```{r} -->
<!-- # Get vector of all the 36k sex-specific transcripts that were measured in Huang et al's microarrays -->
<!-- FBID_by_sex <- read_delim("data/derived/eQTL_phenotypes.txt", delim = " ", n_max = 1) -->
<!-- FBIDs <- gsub("_female", "", FBID_by_sex) -->
<!-- FBIDs <- gsub("_male", "", FBIDs) -->
<!-- FBIDs <- sort(unique(FBIDs)) -->

<!-- # Cull to just the FBIDs that were heritable > 0.2 in one or both sexes -->
<!-- heritable_FBIDs <- read_csv("data/input/huang_2015_tableS5_transcript_heritability.csv") %>% -->
<!--   filter(female_narrow_heritability >= 0.2 | male_narrow_heritability >= 0.2) %>% pull(FBID) -->
<!-- FBIDs <- FBIDs[FBIDs %in% heritable_FBIDs] -->


<!-- # Function to look for eQTLs using a simple linear model in GEMMA -->
<!-- # The LMM version takes too long, and initial trials showed the results were very similar. -->
<!-- # So, we decided to adjsut the expression data for genome-wide genetic variation and Wolbachia, and then analyse that with a simple LM -->
<!-- do_one_eQTL <- function(FBID, mode = "original_data"){ -->

<!--   cov <- "eQTL_results/covariates.txt" -->
<!--   options <- glue("--covar {cov} --allow-no-sex --standard-beta") -->
<!--   options <- glue("--allow-no-sex") -->

<!--   run_command("plink --bfile dgrp2_QC_all_lines --pheno eQTL_phenotypes.txt --mpheno 4 --linear {options} --out assoc_results", path = "/data/derived") -->

<!--   # Find index of the focal sex-specific transcript column, e.g. "FBgn10005_female" -->
<!--   FBID_female <- paste(FBID, "_female", sep = "") -->
<!--   FBID_male <- paste(FBID, "_male", sep = "") -->
<!--   female_index <- which(FBID_by_sex == FBID_female)  -->
<!--   male_index   <- which(FBID_by_sex == FBID_male)  -->

<!--   if(mode == "original_data") { -->
<!--     bfile <- "dgrp2_QC_all_lines" -->
<!--     output_file <- glue("data/derived/eQTL_results/real_data/{FBID}.rds") -->
<!--   } -->
<!--   if(mode == "shuffled_data") { -->
<!--     bfile <- "dgrp2_QC_all_lines_shuffled" -->
<!--     output_file <- glue("data/derived/eQTL_results/shuffled_data/{FBID}.rds") -->
<!--   } -->

<!--   # Run the male and female models in GEMMA. It's a set of simple linear models, on the corrected expression data -->
<!--   shh <- capture.output(run_command("gemma -bfile {bfile} -lm 1 -n {female_index} -o eQTL_{FBID}_females", path = "/data/derived")) -->
<!--   shh <- capture.output(run_command("gemma -bfile {bfile} -lm 1 -n {male_index}   -o eQTL_{FBID}_males",   path = "/data/derived")) -->

<!--   # There are some transcripts where all the data is missing for one sex, or the test otherwise fails. We just remove these transcripts -->
<!--   results_females <- glue("data/derived/output/eQTL_{FBID}_females.assoc.txt") -->
<!--   results_males <- glue("data/derived/output/eQTL_{FBID}_males.assoc.txt") -->
<!--   if(!all(file.exists(c(results_females, results_males)))) return(NULL) -->

<!--   results_females <- read_tsv(results_females, col_types = cols(), progress = FALSE) %>%  -->
<!--     mutate(FBID = FBID_female)  -->
<!--   results_males <- read_tsv(results_males, col_types = cols(), progress = FALSE) %>%  -->
<!--     mutate(FBID = FBID_male)   -->

<!--   # Get a vector of eQTLs that are significant in one or both sexes with p < 0.000001 -->
<!--   hits_either_sex <- unique(c( -->
<!--     results_females %>% filter(p_wald < 0.000001) %>% pull(rs),  -->
<!--     results_males   %>% filter(p_wald < 0.000001) %>% pull(rs))) -->

<!--   if(length(hits_either_sex) > 0){ # get the results for these eQTLs, including the result for the non-significant sex -->
<!--     hits <- bind_rows(results_females, results_males) %>%  -->
<!--       select(rs, FBID, beta, se, p_wald) %>% -->
<!--       filter(rs %in% hits_either_sex)  -->

<!--     saveRDS(hits, file = output_file) -->
<!--   } -->

<!--   try(unlink(glue("data/derived/output/eQTL_{FBID}_females.log.txt"))) # delete the GEMMA files -->
<!--   try(unlink(glue("data/derived/output/eQTL_{FBID}_females.assoc.txt"))) -->
<!--   try(unlink(glue("data/derived/output/eQTL_{FBID}_males.log.txt"))) -->
<!--   try(unlink(glue("data/derived/output/eQTL_{FBID}_males.assoc.txt"))) -->
<!--   return(NULL) -->
<!-- } -->

<!-- # do_one_eQTL("FBgn0014417") -->
<!-- # do_one_eQTL("FBgn0014417", mode = "shuffled_data") -->

<!-- # Run the eQTL GWAS, one transcript at a time, across multiple cores, and save the results -->
<!-- # The result file has about XXXX hits at p < 0.000001; presumably many are false positives, though some have p < 1e-50 etc.  -->
<!-- plan("multicore") -->
<!-- future_lapply(FBIDs[!(FBIDs %in% gsub("[.]rds", "", list.files("data/derived/eQTL_results/real_data")))], do_one_eQTL)  -->
<!-- future_lapply(FBIDs[!(FBIDs %in% gsub("[.]rds", "", list.files("data/derived/eQTL_results/shuffled_data")))], do_one_eQTL, mode = "shuffled_data")  -->

<!-- list.files("data/derived/eQTL_results/real_data", full.names = TRUE) %>%  -->
<!--   lapply(readRDS) %>% do.call("rbind", .) %>% -->
<!--   rename(eQTL = rs) %>% -->
<!--   write_csv("data/derived/eQTL_results/all_eQTL_results.csv") -->

<!-- list.files("data/derived/eQTL_results/shuffled_data", full.names = TRUE) %>%  -->
<!--   lapply(readRDS) %>% do.call("rbind", .) %>% -->
<!--   rename(eQTL = rs) %>% -->
<!--   write_csv("data/derived/eQTL_results/shuffled_data_eQTL_results.csv") -->

<!-- # unlink(list.files("data/derived/eQTL_results", pattern = "rds")) # use to clean up -->

<!-- rm(list = c("FBIDs", "FBID_by_sex")) -->
<!-- ``` -->


<!-- # Snipped code for running LMM version. Takes too long to do all transcripts this way! -->


<!-- # Remove eigen vectors + values for which the eigenvalue is close to zero, and save the modified file -->
<!-- # values <- read_tsv("data/derived/output/eigen_decomp_transcriptome.eigenD.txt", col_names = "eigenvalue")$eigenvalue -->
<!-- # vectors <- read_tsv("data/derived/output/eigen_decomp_transcriptome.eigenU.txt", col_names = FALSE) -->
<!-- #  -->
<!-- # n_eigenvectors <- 10 -->
<!-- # vectors <- vectors[, (ncol(vectors)-n_eigenvectors+1):ncol(vectors)] -->
<!-- # values <- values %>% tail(10) -->
<!-- # write_tsv(vectors, path = "data/derived/output/eigen_decomp_transcriptome.eigenU.txt", col_names = FALSE) -->
<!-- # write_tsv(data.frame(value = values), path = "data/derived/output/eigen_decomp_transcriptome.eigenD.txt", col_names = FALSE) -->

<!-- # Define the paths to the eigenvalues and eigenvectors -->
<!-- # GRM <- "-d ./output/eigen_decomp_transcriptome.eigenD.txt -u ./output/eigen_decomp_transcriptome.eigenU.txt" -->
