---
title: "GWAS for fitness in Drosophila"
output:
  html_document: default
  html_notebook: default
---

```{r load_data message=FALSE, warning=FALSE, results="hide"}
library(dplyr)
library(ggplot2)
library(readr)
library(tidyr)
library(purrr)
library(gridExtra)
library(bigsnpr)

# Connect to the database of annotations
db <- DBI::dbConnect(RSQLite::SQLite(), "data/derived/annotations.sqlite3")

# Results of univariate mixed models, corrected by ashr
lmm_results <- read_csv("data/derived/lmm_results_ashr.csv") 
```





## Make some plots about the SNP data

### Chromosomal coverage

The SNPs that survived pruning have good coverage of the major chromosomes, with the exception of regions near the centromere, and on chromosome 4 (which does not recombine, so presumably it has much higher LD than the others).

```{r echo=FALSE}
if(file.exists('data/derived/dgrp2_QC_focal_lines.bk')) unlink('data/derived/dgrp2_QC_focal_lines.bk')
```

```{r fig.showtext=TRUE}
dgrp_bed <- snp_readBed("data/derived/dgrp2_QC_focal_lines.bed") %>% snp_attach()

chrs <- paste("Chromosome", c("2L", "2R", "3L", "3R", "X", "4"))

data.frame(pos = dgrp_bed$map$physical.pos,
           chr = chrs[dgrp_bed$map$chromosome]) %>%
  ggplot(aes(pos/ 10^6, fill = chr)) +
  geom_density(adjust = 0.08) + 
  facet_wrap(~chr, scales = "free", ncol = 2) +
  xlab("Chromosomal position (Megabases)") + ylab("Marker density") +
  theme_minimal() + 
  theme(text = element_text(family = "Raleway")) +
  theme(legend.position = "none")
```

### Relatedness among the lines in our sample

The denodrogram shows that a few pairs of lines are quite strongly genetically similar, while the majority show no relatedness. This supports our decision to use linear mixed models for our association tests. 
```{r fig.height=15}
get_SNP_genotype_matrix <- function(bigsnp_object){
  mat <- bigsnp_object$genotypes[1:nrow(bigsnp_object$genotypes), 1:ncol(bigsnp_object$genotypes)]
  rownames(mat) <- gsub("line", "", bigsnp_object$fam$family.ID)
  colnames(mat) <- bigsnp_object$map$marker.ID
  mat
}

geno_matrix <- get_SNP_geno_matrix(dgrp_bed)
n_snps <- ncol(geno_matrix)
n_lines <- nrow(geno_matrix)

dendr <- dendro_data(geno_matrix %>% dist %>% hclust, type="rectangle") 

ggplot() + 
  geom_segment(data = segment(dendr), aes(x, y, xend = xend, yend = yend)) + 
  geom_text(data = label(dendr), aes(x, y, label = paste(" ", label), hjust = 0), size = 3) +
  coord_flip() + scale_y_reverse(expand=c(0.2, 0)) + 
  theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        axis.title.y=element_blank(),
        panel.background=element_rect(fill="white"),
        panel.grid=element_blank()) + 
  ylab("Distance")
```
<br></br> 
**Figure SX**: Cluster dendrogram showing the relatedness structure between the DGRP lines that we measured. The lines were clustered based on 

```{r echo=FALSE}
rm(list = c("geno_matrix", "dgrp_bed"))
unlink('data/derived/dgrp2_QC_focal_lines.bk')
```



```{r}
SNP_clumps %>%
  select(SNP_clump, contains("LFSR")) %>% select(-contains("canon")) %>%
  filter_at(vars(contains("LFSR")), any_vars(. < 10e-5)) %>% nrow()
  

# Calculate antagonism....
antagonism_evidence_ratios <- SNP_clumps %>%
  filter_at(vars(contains("LFSR")), any_vars(. < 10e-2))  %>%
  mutate(pp_female_early = ifelse(beta_female_early_mashr_ED > 0, LFSR_female_early_mashr_ED, 1 - LFSR_female_early_mashr_ED),
         pp_female_late  = ifelse(beta_female_late_mashr_ED > 0, LFSR_female_late_mashr_ED, 1 - LFSR_female_late_mashr_ED),
         pp_male_early   = ifelse(beta_male_early_mashr_ED > 0, LFSR_male_early_mashr_ED, 1 - LFSR_male_early_mashr_ED),
         pp_male_late    = ifelse(beta_male_late_mashr_ED > 0, LFSR_male_late_mashr_ED, 1 - LFSR_male_late_mashr_ED)) %>%
  mutate(p_sex_concord_early = pp_female_early * pp_male_early + (1 - pp_female_early) * (1 - pp_male_early),
         p_sex_antag_early = pp_female_early * (1 - pp_male_early) + (1 - pp_female_early) * pp_male_early,
         p_sex_concord_late  = pp_female_late * pp_male_late+ (1 - pp_female_late) * (1 - pp_male_late),
         p_sex_antag_late = pp_female_late * (1 - pp_male_late) + (1 - pp_female_late) * pp_male_late,
         p_age_concord_females = pp_female_early * pp_female_late + (1 - pp_female_early) * (1 - pp_female_late),
         p_age_antag_females = pp_female_early * (1 - pp_female_late) + (1 - pp_female_early) * pp_female_late,
         p_age_concord_males = pp_male_early * pp_male_late + (1 - pp_male_early) * (1 - pp_male_late),
         p_age_antag_males = pp_male_early * (1 - pp_male_late) + (1 - pp_male_early) * pp_male_late) %>%
  mutate(sex_antagonism_early = p_sex_concord_early / p_sex_antag_early,
         sex_antagonism_late = p_sex_concord_late / p_sex_antag_late,
         age_antagonism_females = p_age_concord_females / p_age_antag_females,
         age_antagonism_males = p_age_concord_males / p_age_antag_males) %>%
  select(SNP_clump,  starts_with("sex_"), starts_with("age_")) %>%
  gather(trait, evidence_ratio, -SNP_clump) 



antagonism_evidence_ratios %>%
  ggplot(aes(log2(evidence_ratio))) + 
  geom_histogram(data=subset(antagonism_evidence_ratios, log2(evidence_ratio) < 0), bins = 500, fill = "#FF635C") + 
  geom_histogram(data=subset(antagonism_evidence_ratios, log2(evidence_ratio) > 0), bins = 500, fill = "#5B8AFD") +
  coord_cartesian(xlim = c(-10, 10), ylim = c(0, 7000)) + 
  scale_x_continuous(breaks = seq(-10, 10, by = 2), labels = c(paste("1/",(rev(2 ^ seq(2, 10, by = 2))), sep = ""), 2 ^ seq(0, 10, by = 2))) + 
  facet_wrap(~trait) + xlab("Evidence ratio\n(log scale, blue: concordant, red: antagonistic)") + ylab("Number of variants")

antagonism_evidence_ratios %>% filter(trait == "sex_antagonism_early") %>% arrange(log(evidence_ratio)) %>% left_join(tbl(db, "variants") %>% collect(), by = c("SNP_clump" = "SNP")) %>% left_join(tbl(db, "genes") %>% collect(), by = "FBID") %>% print(n = 100)
```







```{r}
lmm_results %>% 
  select(rs, beta, posterior.estimate, parameter) %>%
  left_join(db %>% tbl("variants") %>% filter(id %in% lmm_results$rs) %>% collect(), by = c("rs" = "id")) %>%
  mutate(site.class = replace(site.class, is.na(site.class), "INTERGENIC")) %>%
  ggplot(aes(y = posterior.estimate, x = site.class)) + 
  facet_wrap(~parameter) + 
  geom_boxplot() + coord_flip() 

read_tsv("data/derived/output/all_four_traits.assoc.txt") %>% 
  select(rs, p_wald) %>%
  left_join(db %>% tbl("variants") %>% filter(id %in% lmm_results$rs) %>% collect(), by = c("rs" = "id")) %>%
  mutate(site.class = replace(site.class, is.na(site.class), "INTERGENIC")) %>%
  ggplot(aes(y = -log10(p_wald), x = site.class)) +
  geom_boxplot() + coord_flip() 
```


### Distributions of SNP estimates
```{r}
lmm_results %>%
  ggplot(aes(posterior.estimate, weight = se)) +
  geom_density() + 
  facet_wrap(~parameter, scales = "free")
```


```{r}
left_join(lm_results, lmm_results, by = c("rs", "parameter")) %>%
  ggplot(aes(x = posterior.estimate.x, y = posterior.estimate.y)) + 
  geom_point() + 
  facet_wrap(~parameter, scales = "free") + 
  xlab("Posterior effect size estimate from linear model") + 
  ylab("Posterior effect size estimate from mixed model")
```



```{r}
make_snp_figure <- function(x, y, results_file, xlab, ylab){
  
  dat <- read_tsv(results_file) %>%
    mutate(sig = "No",
           sig = replace(sig, p_wald <= 0.0001, "Yes"))
  print( dat %>% filter(sig == "Yes"))
  
  ggplot(dat, aes_string(x, y)) + 
    geom_point(alpha = 0.1) + 
    geom_hline(yintercept = 0, linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 2) + 
    stat_smooth(colour = "steelblue",  method = "gam") + 
    geom_point(data = dat %>% filter(sig == "Yes"), aes_string(x, y), fill = "tomato", pch=21, colour = "black", size=2) + 
    xlab(xlab) + ylab(ylab)
}
file <- "data/derived/output/all_four_traits.assoc.txt"
grid.arrange(
  make_snp_figure("beta_1", "beta_3", file, "Effect of SNP on female early-life fitness", "Effect of SNP on male early-life fitness"),
  make_snp_figure("beta_2", "beta_4", file, "Effect of SNP on female late-life fitness", "Effect of SNP on male late-life fitness"),
  make_snp_figure("beta_1", "beta_2", file, "Effect of SNP on early female early-life fitness", "Effect of SNP on female late-life fitness"),
  make_snp_figure("beta_3", "beta_4", file, "Effect of SNP on early male early-life fitness", "Effect of SNP on male late-life fitness"), 
  ncol = 2)
```

```{r}
lmm_ashr_results <- read_csv("data/derived/lmm_results_ashr.csv") %>% 
  select(rs, posterior.estimate, parameter, pp, np) %>%
  spread(parameter, posterior.estimate) %>%
  mutate(p_value = map2_dbl(pp, np, ~ max(.x, .y))) %>%
  group_by(rs) %>%
  summarise(female_early = female_early[!is.na(female_early)], 
            female_late = female_late[!is.na(female_late)], 
            male_early = male_early[!is.na(male_early)], 
            male_late = male_late[!is.na(male_late)],
            p_value_bayesian = min(1 - p_value)) %>%
  ungroup() %>% arrange(p_value_bayesian) %>%
  mutate(sig = "No",
         sig = replace(sig, p_value_bayesian <= 0.05, "Yes")) 

make_snp_figure2 <- function(par1, par2, xlab, ylab){
  dat <- lmm_ashr_results
  ggplot(dat, aes_string(par1, par2)) + 
    geom_point(alpha = 0.1) + 
    geom_hline(yintercept = 0, linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 2) + 
    stat_smooth(colour = "steelblue") + 
    geom_point(data = dat %>% filter(sig == "Yes"), aes_string(par1, par2), fill = "tomato", pch=21, colour = "black", size=2) + 
    xlab(xlab) + ylab(ylab)
}
grid.arrange(
  make_snp_figure2("male_early", "female_early", "Effect of SNP on female early-life fitness", "Effect of SNP on male fitness"),
  make_snp_figure2("male_late", "female_late", "Effect of SNP on female fitness", "Effect of SNP on male fitness"),
  make_snp_figure2("female_early", "female_late", "Effect of SNP on early female fitness", "Effect of SNP on late female fitness"),
  make_snp_figure2("male_early", "male_late", "Effect of SNP on early male fitness", "Effect of SNP on late male fitness"), 
  ncol = 2)
```




```{r}
lm_results %>% filter(parameter %in% c("male_late", "female_late"),
                      pp >= 0.5 | np >= 0.5) %>%
  select(rs) %>% left_join(read.delim("data/derived/output/female_late_male_late.assoc.txt")) %>%
  ggplot(aes(beta_1, beta_2)) + geom_point() + geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) + xlab("Effect of SNP on female fitness") + ylab("Effect of SNP on male fitness")

x <- read.delim("data/derived/output/female_early_lm.assoc.txt")
y <- read.delim("data/derived/output/female_early_lmm.assoc.txt")
left_join(x, y, by = "rs") %>% ggplot(aes(p_wald.x, p_wald.y)) + geom_point()

read.delim("data/derived/output/female_early_male_early.assoc.txt") %>% arrange(p_wald) %>% head()
x <- read.delim("data/derived/output/female_late_male_late.assoc.txt")
ggplot(x %>% filter(p_wald < 0.01), aes(beta_1, beta_2)) + geom_point() 

x <- read.delim("data/derived/output/female_early_lm.assoc.txt")
x <- read.delim("data/derived/output/female_early_lmm.assoc.txt")
y <- read.delim("data/derived/output/female_early_bslmm.param.txt") %>% mutate(beta = alpha + beta*gamma)
left_join(x, y, by = "rs") %>% ggplot(aes(beta.x, beta.y)) + geom_point() + geom_abline()

x <- read.delim("data/derived/output/female_early_bslmm.hyp.txt", sep = "\t", row.names = NULL)
names(x) <- c(names(x)[-1], "trim") 
x <- x %>% select(-trim) %>% mutate(h = as.numeric(h))
plot6 <-  bayesplot::mcmc_areas(x$pge / (x$p))
do.call("grid.arrange", lapply(1:5, function(i) bayesplot::mcmc_areas(x[, i, drop = FALSE])))
```



## eQTL analysis results
This section uses the data generated by the script `analysis/eQTL_analysis.Rmd`, which in turn uses the raw data from Huang et al. _PNAS_. 

```{r}
# Load the eQTL data
eQTL_results <- read_csv("data/derived/eQTL_results/all_eQTL_results.csv") %>% 
  arrange(eQTL) %>% # filter(-log10(p_wald) > 8) %>%
  mutate(sex = ifelse(str_detect(FBID, "female"), "Female", "Male"),
         transcript = str_extract(FBID, "FBgn[:digit:]+"),
         transcript = ifelse(is.na(transcript), str_extract(FBID, "XLOC_[:digit:]+"), transcript)) %>% 
  select(eQTL, transcript, sex, beta, se, p_wald)


# Load the eQTL data
eQTL_results_shuffled <- read_csv("data/derived/eQTL_results/shuffled_data_eQTL_results.csv") %>% 
  arrange(eQTL) %>% # filter(-log10(p_wald) > 8) %>%
  mutate(sex = ifelse(str_detect(FBID, "female"), "Female", "Male"),
         transcript = str_extract(FBID, "FBgn[:digit:]+"),
         transcript = ifelse(is.na(transcript), str_extract(FBID, "XLOC_[:digit:]+"), transcript)) %>% 
  select(eQTL, transcript, sex, beta, se, p_wald)



# Find the strongest p-value for each combination of eQTL and transcript, and apply a cutoff of p < 1e-08
# Thus all the eQTLs in the following plots/tables affect expression in one (or both) sexes with at least this much significance
max_log_p <- eQTL_results %>%
  group_by(eQTL, transcript) %>%
  summarise(max_log_p = max(-log10(p_wald))) %>%
  filter(max_log_p > 8) 

max_log_p_shuffled <- eQTL_results_shuffled %>%
  group_by(eQTL, transcript) %>%
  summarise(max_log_p = max(-log10(p_wald))) %>%
  filter(max_log_p > 8) 

eQTL_results <- eQTL_results %>% filter(eQTL %in% max_log_p$eQTL & transcript %in% max_log_p$transcript)
eQTL_results_shuffled <- eQTL_results_shuffled %>% filter(eQTL %in% max_log_p_shuffled$eQTL & transcript %in% max_log_p_shuffled$transcript)

# culling this way reduces the number of eQTLs to plot from 166,000 down to 10,800, giving more stringent results and faster plots

# table(floor(max_log_p$max_log_p))
#    8    9   10   11   12   13   14   15 
# 2962  778  248   82   26    7    6    1 

table(floor(max_log_p_shuffled$max_log_p))

```

```{r}
# eQTLs that boost expression tend to be the major allele, while those that depress expression tend to be the minor allele... could be opposite way (but this fits prediction)
# Alleles that depress expression in one sex, and boost expression in the other, are quite common!
# Alleles that affect one sex but not the other are very rare
grid.arrange(
  eQTL_results %>% ggplot(aes(sex, beta, group = transcript)) + geom_line(alpha = 0.5),
  eQTL_results %>% ggplot(aes(beta)) + geom_histogram(bins = 100) + coord_flip(), 
ncol = 2)
```

```{r}
# List of transcripts for which we found eQTLs with a marked sex-specific effect
# This will equate to a list of transcripts where the Line x Sex interaction was considerable in the model used to get predicted line mean expression

wide_results_table <- left_join(
  eQTL_results %>%
    select(-se, -p_wald) %>%
    spread(sex, beta) %>%
    left_join(tbl(db, "genes") %>% select(FBID, gene_name) %>% collect(), by = c("transcript" = "FBID")) %>%
    select(eQTL, transcript, gene_name, Female, Male) %>%
    rename(Female_beta = Female, Male_beta = Male),
  eQTL_results %>%
    select(-beta, -p_wald) %>%
    spread(sex, se) %>%
    rename(Female_SE = Female, Male_SE = Male),
  by = c("eQTL", "transcript")) %>%
  left_join(
    eQTL_results %>%
      select(-se, -beta) %>%
      mutate(p_wald = sprintf("%.1f", round(-log10(p_wald), 1))) %>%
      spread(sex, p_wald) %>%
      rename(Female_p = Female, Male_p = Male), 
    by = c("eQTL", "transcript")
)  

wide_results_table %>% arrange(-(as.numeric(Female_p))) %>% print(n = 200)

wide_results_table %>%
  filter(Female_p > 8 & Male_p > 8 & sign(Male_beta != Female_beta)) %>%
  arrange(abs(Male_beta - Female_beta)) %>% print(n=Inf)
```


```{r}
P_cutoff <- 6 # We require this cutoff to be sure there is a non-zero effect in the below figure

types_of_eQTL_table <- wide_results_table %>%
  mutate(effect_male = "none",
         effect_male = replace(effect_male, Male_beta < 0 & Male_p > P_cutoff, "negative"),
         effect_male = replace(effect_male, Male_beta > 0 & Male_p > P_cutoff, "positive"),
         effect_female = "none",
         effect_female = replace(effect_female, Female_beta < 0 & Female_p > P_cutoff, "negative"),
         effect_female = replace(effect_female, Female_beta > 0 & Female_p > P_cutoff, "positive"),
         x = "none",
         x = replace(x, effect_male == "positive" & effect_female == "positive", "Positive in both sexes"),
         x = replace(x, effect_male == "negative" & effect_female == "negative", "Negative in both sexes"),
         x = replace(x, effect_male == "positive" & effect_female == "negative", "Positive in males, negative in females"),
         x = replace(x, effect_male == "negative" & effect_female == "positive", "Positive in females, negative in males"),
         x = replace(x, effect_male == "none" & effect_female == "positive", "Positive, but only in females"),
         x = replace(x, effect_male == "none" & effect_female == "negative", "Negative, but only in females"),
         x = replace(x, effect_female == "none" & effect_male == "positive", "Positive, but only in males"),
         x = replace(x, effect_female == "none" & effect_male == "negative", "Negative, but only in males")) %>%
  left_join(tbl(db, "variants") %>% select(SNP, chr, position, site.class, MAF, FBID) %>% rename(nearby_gene = FBID) %>%
              collect(), by = c("eQTL" = "SNP")) 

types_of_eQTL_table %>%
  pull(x) %>% table(x = .) %>% sort() %>% as_tibble() %>% filter(x != "none") %>% mutate(x = factor(x,x), percentage = 100 * n/sum(n)) %>%
  ggplot(aes(x, n)) + geom_bar(stat = "identity") + coord_flip() + xlab("Effect of the major allele on gene expression") + ylab("% eQTLs")

with(types_of_eQTL_table, table(x, chr)) %>% as_tibble() %>% filter(x != "none") %>% arrange(chr, n) %>% mutate(x = factor(x,unique(x)), percentage = 100 * n/sum(n)) %>%
  ggplot(aes(x, n)) + geom_bar(stat = "identity") + coord_flip() + xlab("Effect of the major allele on gene expression") + ylab("% eQTLs") + facet_wrap(~chr)
```

```{r}
xx <- types_of_eQTL_table %>% select(eQTL, transcript) %>% rename(affected_gene = transcript) %>% distinct()
xx <- xx %>% left_join(tbl(db, "variants") %>% select(SNP, FBID, chr, position) %>% collect(), by = c("affected_gene" = "FBID")) %>%
  group_by(eQTL, affected_gene) %>%
  summarise(chr = chr[1], 
            approx_position = round(mean(as.numeric(position)))) %>%
  mutate(split = strsplit(eQTL, split = "_"),
         eQTL_chr = map_chr(split, ~ .x[1]),
         eQTL_pos = map_chr(split, ~ .x[2])) %>% select(-split)

sum(xx$chr == xx$eQTL_chr, na.rm=T)
sum(is.na(xx$chr == xx$eQTL_chr))

xx %>% filter(chr == eQTL_chr) %>% arrange(abs(approx_position - as.numeric(eQTL_pos))) 
```


```{r}
# Experiments looking at site class and chr
xx <- wide_results_table %>%
  group_by(eQTL, transcript) %>%
  summarise(mean_beta_male = mean(Male_beta),
            mean_beta_female = mean(Female_beta)) %>%
  left_join(tbl(db, "variants") %>% select(SNP, chr, position, site.class, MAF) %>%
              collect(), by = c("eQTL" = "SNP")) 

# funky MAF results
ggplot(xx, aes(mean_beta_male, MAF)) + geom_point()
ggplot(xx, aes(mean_beta_female, MAF)) + geom_point()
ggplot(xx, aes(abs(mean_beta_female - mean_beta_male), MAF)) + geom_point()



xx %>% ggplot(aes(sex, mean_beta, group = eQTL)) + 
  geom_line(alpha = 0.05) + 
  facet_wrap(~site.class)
```
