<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>GWAS for fitness in Drosophila</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">GWAS for fitness in Drosophila</h1>

</div>


<p><strong>Last updated:</strong> 2018-10-22</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180914)</code> </summary></p>
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> 509a9d7 </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/derived/.DS_Store
    Ignored:    data/derived/output/.DS_Store
    Ignored:    data/input/.DS_Store
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  .pversion
    Untracked:  ExDeconDemo_c.log
    Untracked:  ExDeconDemo_loglike.log
    Untracked:  TEST_Venn.tiff
    Untracked:  analysis/checking_mashr_results.Rmd
    Untracked:  analysis/eQTL_analysis.Rmd
    Untracked:  code/Drosophila_GWAS.Rmd
    Untracked:  data/derived/.!25440!dgrp2_focal_lines.bed
    Untracked:  data/derived/.pversion
    Untracked:  data/derived/GRM.txt
    Untracked:  data/derived/SNP_clumps_eQTL_analysis.txt
    Untracked:  data/derived/SNP_list_for_GEMMA_transcriptome.txt
    Untracked:  data/derived/TWAS_result_females.csv
    Untracked:  data/derived/TWAS_result_males.csv
    Untracked:  data/derived/all_four_traits_SNP_clumps.txt
    Untracked:  data/derived/all_univariate_GEMMA_results.csv
    Untracked:  data/derived/annotations.sqlite3
    Untracked:  data/derived/covariate_file_for_GEMMA.txt
    Untracked:  data/derived/covariate_file_for_GEMMA_transcriptome.txt
    Untracked:  data/derived/dgrp2_QC_all_lines.bed
    Untracked:  data/derived/dgrp2_QC_all_lines.bim
    Untracked:  data/derived/dgrp2_QC_all_lines.fam
    Untracked:  data/derived/dgrp2_QC_all_lines.log
    Untracked:  data/derived/dgrp2_QC_focal_lines.bed
    Untracked:  data/derived/dgrp2_QC_focal_lines.bim
    Untracked:  data/derived/dgrp2_QC_focal_lines.fam
    Untracked:  data/derived/dgrp2_QC_focal_lines.log
    Untracked:  data/derived/dgrp2_QC_focal_lines.rds
    Untracked:  data/derived/female_early_SNP_clumps.txt
    Untracked:  data/derived/female_late_SNP_clumps.txt
    Untracked:  data/derived/lmm_ED_mashr_results.rds
    Untracked:  data/derived/lmm_canonical_mashr_results.rds
    Untracked:  data/derived/luke_eQTL_results.csv
    Untracked:  data/derived/male_early_SNP_clumps.txt
    Untracked:  data/derived/male_late_SNP_clumps.txt
    Untracked:  data/derived/mashr_results_ED.rds
    Untracked:  data/derived/mashr_results_canonical.rds
    Untracked:  data/derived/output/.pversion
    Untracked:  data/derived/output/GRM_transcriptome.cXX.txt
    Untracked:  data/derived/output/GRM_transcriptome.log.txt
    Untracked:  data/derived/output/all_four_traits.assoc.txt
    Untracked:  data/derived/output/all_four_traits.log.txt
    Untracked:  data/derived/output/allele_freq_count.assoc.txt
    Untracked:  data/derived/output/allele_freq_count.log.txt
    Untracked:  data/derived/output/eQTL_test.assoc.txt
    Untracked:  data/derived/output/eQTL_test.log.txt
    Untracked:  data/derived/output/eigen_decomp.log.txt
    Untracked:  data/derived/output/eigen_decomp_transcriptome.eigenD.txt
    Untracked:  data/derived/output/eigen_decomp_transcriptome.eigenU.txt
    Untracked:  data/derived/output/eigen_decomp_transcriptome.log.txt
    Untracked:  data/derived/plink.clumped
    Untracked:  data/derived/plink.frq
    Untracked:  data/input/.pversion
    Untracked:  data/input/clough_2014_dsx_targets.csv
    Untracked:  data/input/dgrp2.fam.bak
    Untracked:  data/input/huang_2015_tableS11_eQTL_females.csv
    Untracked:  data/input/huang_2015_tableS12_eQTL_males.csv
    Untracked:  data/input/huang_2015_tableS2_gene_expression.csv
    Untracked:  data/input/huang_2015_tableS5_transcript_heritability.csv
    Untracked:  dgrp2_clean.log
    Untracked:  docs/figure/checking_mashr_results.Rmd/
    Untracked:  gwas_adaptive_shrinkage.R

Unstaged changes:
    Modified:   analysis/GWAS_tables.Rmd
    Modified:   analysis/TWAS.Rmd
    Deleted:    data/derived/DGRP_SNP_genos.sqlite3
    Deleted:    data/derived/gwas_db.sqlite3
    Deleted:    data/derived/lm_results_ashr.csv
    Modified:   data/derived/lmm_results_ashr.csv
    Deleted:    data/derived/output/DGRP_GRM.cXX.txt
    Deleted:    data/derived/output/DGRP_GRM.log.txt
    Deleted:    data/derived/output/female_early_bslmm.bv.txt
    Deleted:    data/derived/output/female_early_bslmm.gamma.txt
    Deleted:    data/derived/output/female_early_bslmm.hyp.txt
    Deleted:    data/derived/output/female_early_bslmm.log.txt
    Deleted:    data/derived/output/female_early_bslmm.param.txt
    Deleted:    data/derived/output/female_early_bslmm_preds.log.txt
    Deleted:    data/derived/output/female_early_bslmm_preds.prdt.txt
    Deleted:    data/derived/output/female_early_female_late.assoc.txt
    Deleted:    data/derived/output/female_early_female_late.log.txt
    Deleted:    data/derived/output/female_early_lm.assoc.txt
    Deleted:    data/derived/output/female_early_lm.log.txt
    Modified:   data/derived/output/female_early_lmm.assoc.txt
    Modified:   data/derived/output/female_early_lmm.log.txt
    Deleted:    data/derived/output/female_early_male_early.assoc.txt
    Deleted:    data/derived/output/female_early_male_early.log.txt
    Deleted:    data/derived/output/female_late_lm.assoc.txt
    Deleted:    data/derived/output/female_late_lm.log.txt
    Modified:   data/derived/output/female_late_lmm.assoc.txt
    Modified:   data/derived/output/female_late_lmm.log.txt
    Deleted:    data/derived/output/female_late_male_late.assoc.txt
    Deleted:    data/derived/output/female_late_male_late.log.txt
    Deleted:    data/derived/output/male_early_lm.assoc.txt
    Deleted:    data/derived/output/male_early_lm.log.txt
    Modified:   data/derived/output/male_early_lmm.assoc.txt
    Modified:   data/derived/output/male_early_lmm.log.txt
    Deleted:    data/derived/output/male_early_male_late.assoc.txt
    Deleted:    data/derived/output/male_early_male_late.log.txt
    Deleted:    data/derived/output/male_late_lm.assoc.txt
    Deleted:    data/derived/output/male_late_lm.log.txt
    Modified:   data/derived/output/male_late_lmm.assoc.txt
    Modified:   data/derived/output/male_late_lmm.log.txt
    Deleted:    data/derived/output/result.log.txt
    Deleted:    data/derived/output/result.prdt.txt
    Deleted:    data/derived/summed_interSNP_correlations.csv
    Deleted:    data/derived/trimmed_DGRP.bed
    Deleted:    data/derived/trimmed_DGRP.bim
    Deleted:    data/derived/trimmed_DGRP.fam
    Deleted:    data/derived/trimmed_DGRP.rds
    Modified:   data/input/dgrp2.fam

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<hr />
<pre class="r"><code>library(dplyr)
library(tibble)
library(readr)
library(glue)
library(stringr)
library(tidyr)
library(future.apply)
options(stringsAsFactors = FALSE)

# Load the wolbachia infection status data from the DGRP website
wolbachia &lt;- read_csv(&quot;data/input/wolbachia.csv&quot;) %&gt;% 
  mutate(line = paste(&quot;line&quot;, line, sep = &quot;&quot;))</code></pre>
<div id="define-helper-functions" class="section level2">
<h2>Define helper functions</h2>
<pre class="r"><code># Helper to run shell commands
run_command &lt;- function(shell_command, wd = getwd(), path = &quot;&quot;){
  calling_env &lt;- parent.frame()
  cat(system(glue(&quot;cd &quot;, wd, path, &quot;\n&quot;,shell_command, .envir = calling_env), intern = TRUE), sep = &#39;\n&#39;)
}

# For loading the Huang et al microarray data
load_expression_data &lt;- function(sex){
  
  if(sex != &quot;both&quot;){
    expression &lt;- glue(&quot;data/input/dgrp.array.exp.{sex}.txt&quot;) %&gt;% read_delim(delim = &quot; &quot;)
    sample_names &lt;- names(expression)[names(expression) != &quot;gene&quot;] %&gt;% str_remove(&quot;:[12]&quot;) 
    gene_names &lt;- expression$gene
    expression &lt;- expression %&gt;% select(-gene) %&gt;% as.matrix() %&gt;% t() 
    rownames(expression) &lt;- sample_names # rows are samples, columns are genes
    colnames(expression) &lt;- gene_names 
    return(expression %&gt;% as.data.frame() %&gt;% tibble::rownames_to_column(&quot;line&quot;) %&gt;% as_tibble() %&gt;% mutate(line = str_remove_all(line, &quot;[.]1&quot;)))
  }
  
  females &lt;- read_delim(&quot;data/input/dgrp.array.exp.female.txt&quot;, delim = &quot; &quot;)
  names(females)[-1] &lt;- paste(&quot;F_&quot;, names(females)[-1],sep=&quot;&quot;) #%&gt;% str_remove(&quot;:[12]&quot;) 
  females &lt;- females %&gt;% left_join(read_delim(&quot;data/input/dgrp.array.exp.male.txt&quot;, delim = &quot; &quot;), by = &quot;gene&quot;)
  sample_names &lt;- names(females)[names(females) != &quot;gene&quot;] %&gt;% str_remove(&quot;:[12]&quot;) 
  gene_names &lt;- females$gene
  sex &lt;- ifelse(str_detect(sample_names, &quot;F_&quot;), &quot;female&quot;, &quot;male&quot;)
  line &lt;- str_remove_all(sample_names, &quot;F_&quot;)
  females &lt;- females %&gt;% select(-gene) %&gt;% t()
  colnames(females) &lt;- gene_names
  list(
    sampleIDs = tibble(sex, line),
    expression = females
  )
}

# Find the line means, and standardise the line mean expression levels to mean = 0, var = 1, separately for each sex
# Less hassle than the built-in scale() function
quick_scale &lt;- function(x) (x - mean(x)) / sd(x)

# Get transcript data for both sexes for specific transcript(s)
get_transcript_data &lt;- function(FBID){
  if(FBID == &quot;all&quot;) return(line_mean_expression)
  line_mean_expression %&gt;% select(line, starts_with(!!FBID))
}

# Edit the DGRP (all lines) fam file to add the 18,000 transcripts, for both sexes (so, add 36,000 extra columns) 
replace_pheno_data &lt;-function(FBID) {
  focal &lt;- read_delim(&quot;data/derived/dgrp2_QC_all_lines.fam&quot;, delim = &quot; &quot;, col_names = FALSE)[,1:5] %&gt;%
    left_join(get_transcript_data(FBID), by = c(&quot;X1&quot; = &quot;line&quot;)) 
  
  for(i in 6:ncol(focal)) focal[is.na(focal[,i]), i] &lt;- -9 # Replace NA with -9, for PLINK
  
  write.table(focal, file = &quot;data/derived/dgrp2_QC_all_lines.fam&quot;, 
              col.names = FALSE, row.names = FALSE, 
              quote = FALSE, sep = &quot; &quot;)
}</code></pre>
</div>
<div id="load-up-the-expression-data" class="section level2">
<h2>Load up the expression data</h2>
<p>This microarray data from Huang et al. was downloaded from the <a href="http://dgrp2.gnets.ncsu.edu/data.html">DGRP website</a>.</p>
<pre class="r"><code># Load the data, and find the line means for each transcript
line_mean_expression &lt;- load_expression_data(&quot;female&quot;) %&gt;%
  group_by(line) %&gt;%
  summarise_all(funs(mean)) %&gt;%
  rename_at(vars(-line), ~str_c(., &quot;_female&quot;)) %&gt;%
  left_join(load_expression_data(&quot;male&quot;) %&gt;%
              group_by(line) %&gt;%
              summarise_all(funs(mean)) %&gt;%
              rename_at(vars(-line), ~str_c(., &quot;_male&quot;)),
            by = &quot;line&quot;) %&gt;% as.data.frame()

# Scale the line mean expression values to mean 0, variance 1
jobs &lt;- split(2:ncol(line_mean_expression), ceiling(seq_along(2:ncol(line_mean_expression)) / 5000))
plan(&quot;multicore&quot;)
line_mean_expression &lt;- future_lapply(jobs, function(job_chunk) {
  for(i in 1:length(job_chunk)) {
    line_mean_expression[, job_chunk[i]] &lt;- quick_scale(line_mean_expression[, job_chunk[i]])
  }
  line_mean_expression[, c(1, job_chunk)]
})

line_mean_expression &lt;- line_mean_expression %&gt;% 
  join_all(by = &quot;line&quot;, type = &quot;left&quot;, match = &quot;all&quot;) %&gt;% 
  mutate(line = str_remove_all(line, &quot;_&quot;)) %&gt;% as_tibble()
rm(jobs)</code></pre>
</div>
<div id="set-up-for-running-the-eqtl-linear-model-gwas" class="section level2">
<h2>Set up for running the eQTL linear model GWAS</h2>
<p>Here, we set up the files needed by GEMMA, and identify SNPs that are in &lt;100% linkage disequilibrium with each other (since it’s a waste of time to test these separately, as the estimates and p-values would be identical).</p>
<pre class="r"><code># Set up the Wolbachia covariate data
lines_in_fam &lt;- read_delim(&quot;data/derived/dgrp2_QC_all_lines.fam&quot;, 
                           col_names = FALSE, delim = &quot; &quot;) %&gt;% 
  select(X1) %&gt;% rename(line = X1)

# Edit the wolbahia data to match the right lines, and code infection as 0 or 1 instead of &quot;n&quot; and &quot;y&quot;
wolbachia_edited &lt;- left_join(lines_in_fam, wolbachia, by = &quot;line&quot;) %&gt;%
  mutate(intercept = 1,
         wolbachia = as.numeric(ifelse(wolbachia == &quot;y&quot;, 1, 0))) %&gt;% 
  select(intercept, wolbachia)
wolbachia_edited %&gt;% 
  write_delim(path = &quot;data/derived/covariate_file_for_GEMMA_transcriptome.txt&quot;, col_names = FALSE)

# Do a quick test run. This is used to find clumps of SNPs that are in perfect LD with each other without using 
# Slower methods that look at the actual genotype data
bed &lt;- &quot;dgrp2_QC_all_lines -maf 0.05&quot;
replace_pheno_data(&quot;FBgn0000014&quot;)
run_command(&quot;gemma -bfile {bed} {GRM} -lm 1 -r2 0.99 -c covariate_file_for_GEMMA_transcriptome.txt -n 1 -o eQTL_test&quot;, path = &quot;/data/derived&quot;)

# Sort by p, and rename the SNP column
read_tsv(&quot;data/derived/output/eQTL_test.assoc.txt&quot;) %&gt;% 
  arrange(p_wald) %&gt;% rename(SNP = rs) %&gt;% write_tsv(&quot;data/derived/output/eQTL_test.assoc.txt&quot;)

# Make a list of index SNPs that are in &lt;100% LD with each other, 
# and a list mapping index SNPs to others from their 100% LD clumps
eQTL_SNP_clumps &lt;- read_tsv(&quot;data/derived/output/eQTL_test.assoc.txt&quot;) %&gt;% 
  mutate(result = paste(af, beta, se, p_wald, sep = &quot; &quot;)) %&gt;%
  select(SNP, result) %&gt;%
  group_by(result) %&gt;% 
  summarise(index_SNP = SNP[1],
            SNPs = paste0(SNP, collapse = &quot;; &quot;)) %&gt;% 
  ungroup() %&gt;% select(index_SNP, SNPs)

eQTL_SNP_clumps %&gt;% 
  select(index_SNP) %&gt;% 
  write_tsv(&quot;data/derived/SNP_list_for_GEMMA_transcriptome.txt&quot;, col_names = FALSE)

eQTL_SNP_clumps %&gt;% 
  write_csv(&quot;data/derived/SNP_clumps_eQTL_analysis.txt&quot;)</code></pre>
</div>
<div id="run-the-eqtl-analysis" class="section level2">
<h2>Run the eQTL analysis</h2>
<p>This took about 4.5 days on my 8-core Mac.</p>
<pre class="r"><code># GEMMA options. Linear model, Wald tests, Wolbachia data, exclude SNPS &gt;99% correlated with Wolbachia, only use the SNPs in the list
opts &lt;- &quot;-lm 1 -r2 0.99 -c covariate_file_for_GEMMA_transcriptome.txt -snps SNP_list_for_GEMMA_transcriptome.txt&quot;

# Make a fam file with all the 36,000+ gene expression traits (that&#39;s 18k per sex)
replace_pheno_data(&quot;all&quot;)

# Get vector of all the transcripts that were measured in Huang et al&#39;s microarrays
FBIDs &lt;- names(line_mean_expression)[-1]

# Function to look for eQTLs using a simple linear model in GEMMA
# The LMM version takes too long, and initial trials showed the results were very similar.
do_one_eQTL &lt;- function(FBID){
  
  # Find index of the focal sex-specific transcript column, e.g. &quot;FBgn10005_females&quot;
  index &lt;- which(FBIDs == FBID) %&gt;% as.character() 
  
  # Run the linear model in GEMMA
  run_command(&quot;gemma -bfile {bed} {opts} -n {index} -o eQTL_{FBID}&quot;, path = &quot;/data/derived&quot;)
  
  # Find all the &#39;significant&#39; eQTLs using threshold p &lt; 0.00001, and save the relevant columns for the significant ones
  hits &lt;- glue(&quot;data/derived/output/eQTL_{FBID}.assoc.txt&quot;) %&gt;%
    read_tsv() %&gt;% 
    select(rs, beta, se, p_wald) %&gt;%
    filter(p_wald &lt; 0.00001) %&gt;% 
    arrange(p_wald) %&gt;% rename(eQTL = rs) %&gt;%
    mutate(FBID = FBID) %&gt;%
    select(eQTL, FBID, everything())
  
  unlink(glue(&quot;data/derived/output/eQTL_{FBID}.log.txt&quot;)) # Tidy up the GEMMA files
  unlink(glue(&quot;data/derived/output/eQTL_{FBID}.assoc.txt&quot;))
  return(hits)
}

# Run the eQTL GWAS, one transcript at a time, across multiple cores, and save the results
# The result file has about 7 million hits at p &lt; 0.00001; presumably many are false positives, though some have p &lt; 1e-50 etc. 
plan(&quot;multicore&quot;)
future_lapply(FBIDs, do_one_eQTL) %&gt;%
  do.call(&quot;rbind&quot;, .) %&gt;%
  write_csv(&quot;data/derived/luke_eQTL_results.csv&quot;)</code></pre>
<!-- # Snipped code for running LMM version. Takes too long to do all transcripts this way! -->
<!-- # # Calculate the centered GRM for the 185 lines with transcriptome data -->
<!-- # run_command("gemma -bfile {bed} -r2 0.99 -c covariate_file_for_GEMMA_transcriptome.txt -gk 1 -o GRM_transcriptome", path = "/data/derived")  -->
<!-- #  -->
<!-- # # Perform decomposition of the GRM, and save its eigenvalues and eigenvectors -->
<!-- # run_command("gemma -bfile {bed} -r2 0.99 -c covariate_file_for_GEMMA_transcriptome.txt -k ./output/GRM_transcriptome.cXX.txt -eigen -o eigen_decomp_transcriptome", path = "/data/derived")  -->
<!-- # Remove eigen vectors + values for which the eigenvalue is close to zero, and save the modified file -->
<!-- # values <- read_tsv("data/derived/output/eigen_decomp_transcriptome.eigenD.txt", col_names = "eigenvalue")$eigenvalue -->
<!-- # vectors <- read_tsv("data/derived/output/eigen_decomp_transcriptome.eigenU.txt", col_names = FALSE) -->
<!-- #  -->
<!-- # n_eigenvectors <- 10 -->
<!-- # vectors <- vectors[, (ncol(vectors)-n_eigenvectors+1):ncol(vectors)] -->
<!-- # values <- values %>% tail(10) -->
<!-- # write_tsv(vectors, path = "data/derived/output/eigen_decomp_transcriptome.eigenU.txt", col_names = FALSE) -->
<!-- # write_tsv(data.frame(value = values), path = "data/derived/output/eigen_decomp_transcriptome.eigenD.txt", col_names = FALSE) -->
<!-- # Define the paths to the eigenvalues and eigenvectors -->
<!-- # GRM <- "-d ./output/eigen_decomp_transcriptome.eigenD.txt -u ./output/eigen_decomp_transcriptome.eigenU.txt" -->
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2.2     future.apply_1.0.1 future_1.9.0      
[4] tidyr_0.8.1        stringr_1.3.1      glue_1.3.0        
[7] readr_1.1.1        tibble_1.4.2       dplyr_0.7.6       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.19      knitr_1.20        bindr_0.1.1      
 [4] whisker_0.3-2     magrittr_1.5      workflowr_1.1.1  
 [7] hms_0.4.2         tidyselect_0.2.4  R6_2.2.2         
[10] rlang_0.2.2       globals_0.12.2    tools_3.5.1      
[13] parallel_3.5.1    R.oo_1.22.0       git2r_0.23.0     
[16] htmltools_0.3.6   yaml_2.2.0        rprojroot_1.3-2  
[19] digest_0.6.15     assertthat_0.2.0  crayon_1.3.4     
[22] purrr_0.2.5       codetools_0.2-15  R.utils_2.7.0    
[25] evaluate_0.11     rmarkdown_1.10    stringi_1.2.4    
[28] pillar_1.3.0      compiler_3.5.1    backports_1.1.2  
[31] R.methodsS3_1.7.1 listenv_0.7.0     pkgconfig_2.0.1  </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
