<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Estimating the mean phenotype for each DGRP line</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Estimating the mean phenotype for each DGRP line</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Rprojects/fitnessGWAS/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreec606d3d8748d48cc6dd479febb9ce71effc8bcfdtargetblankc606d3da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/c606d3d8748d48cc6dd479febb9ce71effc8bcfd" target="_blank">c606d3d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreec606d3d8748d48cc6dd479febb9ce71effc8bcfdtargetblankc606d3da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/c606d3d8748d48cc6dd479febb9ce71effc8bcfd" target="_blank">c606d3d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    analysis/plot_models_variant_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/GO_KEGG_enrichment.Rmd
    Untracked:  analysis/TWAS_tables.Rmd
    Untracked:  code/GO_and_KEGG_gsea.R
    Untracked:  code/eQTL_analysis.Rmd
    Untracked:  code/make_sites_files_for_ARGweaver.R
    Untracked:  code/plotting_results.Rmd
    Untracked:  code/run_argweaver.R
    Untracked:  code/run_mashr.R
    Untracked:  data/argweaver/
    Untracked:  data/input/TF_binding_sites.csv
    Untracked:  figures/GWAS_mixture_proportions.rds
    Untracked:  figures/GWAS_stats_figure.pdf
    Untracked:  figures/TWAS_mixture_proportions.rds
    Untracked:  figures/TWAS_stats_figure.pdf
    Untracked:  figures/antagonism_ratios.pdf
    Untracked:  figures/composite_mixture_figure.pdf
    Untracked:  figures/eff_size_histos.pdf
    Untracked:  figures/stats_figure.pdf
    Untracked:  manuscript/

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Modified:   analysis/GWAS_tables.Rmd
    Modified:   analysis/TWAS.Rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/eQTL_analysis.Rmd
    Modified:   analysis/gwas_adaptive_shrinkage.Rmd
    Modified:   analysis/license.Rmd
    Modified:   analysis/make_annotation_database.Rmd
    Modified:   analysis/plot_line_means.Rmd
    Modified:   analysis/plotting_results.Rmd
    Deleted:    code/gwas_adaptive_shrinkage.R
    Modified:   data/input/female_fitness.csv
    Modified:   data/input/female_fitness_CLEANED.csv
    Modified:   data/input/male_fitness.csv
    Modified:   data/input/male_fitness_CLEANED.csv
    Modified:   figures/figure1.eps
    Deleted:    figures/figure1.png
    Modified:   figures/figure2.eps

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/get_predicted_line_means.Rmd</code>) and HTML (<code>docs/get_predicted_line_means.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/836a780685e8ac20355b0a92672276f7ca763481/docs/get_predicted_line_means.html" target="_blank">836a780</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/359ff37c4f4021451e81122393307a28013165a9/docs/get_predicted_line_means.html" target="_blank">359ff37</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/b1836d76a4d03f10415ebbd0698c4bf262bbf2f9/analysis/get_predicted_line_means.Rmd" target="_blank">b1836d7</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/get_predicted_line_means.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/get_predicted_line_means.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(brms)
library(mice)
library(ICC)
library(future)
library(future.apply)
library(kableExtra)</code></pre>
<div id="estimate-the-line-means-for-relative-fitness" class="section level2">
<h2>Estimate the line means for relative fitness</h2>
<p>Most studies of the DGRP calculate “line means” simply by averaging the phenotypes of several individuals per line. Instead, we used Bayesian generalised linear mixed model (GLMMs) to estimate the true line means for each of the 4 fitness traits, implemented in the <code>brms</code> package.</p>
<p>One notable advantage of using a model instead of a simple average is that a model allows one to statistically remove confounding factors such as block effects, which would otherwise bias the line means. Note that we intentionally measured line 352 in every block in our experiments, in order to improve our precision to estimate block effects.</p>
<p>Additionally, we must properly account for our experimental design in order to avoid pseudoreplication. Our design involved repeatedly measuring the same individuals to determine early- and late-life fitness.</p>
<p>For the progeny count data used to measure early and late female fitness, we used multivariate model with two components, one for early- and one for late-life fitness, which had the following formulae (in <code>brms</code>-like pseudocode):</p>
<p><code>Progeny count (early) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)</code></p>
<p><code>Progeny count (late) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)</code></p>
<p>To explain, the terms with the form <code>(1 | x | y)</code> represent random intercepts for line, block, and vial ID, where the letter between the pipe operator indicates which random intercepts should be assumed to be (potentially) correlated with which other random intercepts. We therefore assume that the effect of line, block, and vial ID on each observation are potentially correlated for both of the response variables, and estiamte that correlation while fitting the multivariate model.</p>
<p>The progeny count data were assumed to follow a Poisson distribution (with the standard log link function).</p>
<p>Similarly, for our measure of male fitness (proportion of offspring sired), we used a multivariate model consisting of a pair of binomial GLMMs, with the following formula:</p>
<p><code>Proportion of progeny sired (early) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)</code></p>
<p><code>Proportion of progeny sired (late) = Intercept + num.GFP.males + (1 | p | line) + (1 | q | block) + (1 | r | vial)</code></p>
<p>Where the response variable was a Bernoulli outcome describing the paternity of each offspring observed. Note that in the late life fitness model, we included the number of surviving rival (GFP) males as an additional predictor. We reasoned that the rival males might die at random with respect to the genotype of the focal males, which lowers competition on the focal males, and so we can get an improved estimate of the focal males’ siring success by statistically correcting for their deaths via the model.</p>
<div id="define-functions-for-prediction" class="section level3">
<h3>Define functions for prediction</h3>
<pre class="r"><code>load_and_tidy_fitness_data &lt;- function(){
  female_fitness &lt;- read_csv(&quot;data/input/female_fitness.csv&quot;) %&gt;% 
    arrange(line) %&gt;%
    split(paste(.$block, .$line)) %&gt;%
    map(~ .x %&gt;% mutate(replicate = 1:n())) %&gt;%
    bind_rows() %&gt;%
    mutate(vial = 1:n(),
           vial = paste(&quot;F&quot;, line, block, replicate, sep = &quot;_&quot;),
           line = paste(&quot;line_&quot;, line, sep = &quot;&quot;)) %&gt;% 
    dplyr::select(-num.F.late1, -num.F.late2, -num.laying.F.early, -replicate) %&gt;%
    mutate(female.fitness.early = replace(female.fitness.early, is.na(female.fitness.early), 0),
           female.fitness.late = replace(female.fitness.late, is.na(female.fitness.late), 0)) 
  
  male_fitness &lt;- read_csv(&quot;data/input/male_fitness.csv&quot;) %&gt;% 
    arrange(line) %&gt;%
    split(paste(.$block, .$line)) %&gt;%
    map(~ .x %&gt;% mutate(replicate = 1:n())) %&gt;%
    bind_rows() %&gt;%
    mutate(vial = paste(&quot;M&quot;, line, block, replicate, sep = &quot;_&quot;),
           line = paste(&quot;line_&quot;, line, sep = &quot;&quot;)) %&gt;%
    dplyr::select(-num.DGRP.males, -replicate)
  
  # Remove the small number of observations where the females had no offspring, so we cannot measure the
  # rival vs focal male competitive abilities
  # NOTE: Not needed any more, because now I impute this as missing data
  # exclude_early &lt;- male_fitness$early.male.rival + male_fitness$early.male.focal == 0
  # exclude_late &lt;- male_fitness$late.male.rival + male_fitness$late.male.focal == 0
  # male_fitness$early.male.rival[exclude_early] &lt;- NA
  # male_fitness$early.male.focal[exclude_early] &lt;- NA
  # male_fitness$late.male.rival[exclude_late] &lt;- NA
  # male_fitness$late.male.focal[exclude_late] &lt;- NA
  
  # Save these tidy files, for data archiving
  write.csv(female_fitness, file = &quot;data/input/female_fitness_CLEANED.csv&quot;, row.names = FALSE)
  write.csv(male_fitness, file = &quot;data/input/male_fitness_CLEANED.csv&quot;, row.names = FALSE)
  
  list(female_fitness = female_fitness, male_fitness = male_fitness)
}


# This function calculates the traditional line means (i.e. simple averages) for each line
get_line_means &lt;- function(){
  left_join(
    load_and_tidy_fitness_data()[[1]] %&gt;% 
      select(line, female.fitness.early, female.fitness.late) %&gt;% 
      group_by(line) %&gt;%
      summarise(female.fitness.early = mean(female.fitness.early), 
                female.fitness.late = mean(female.fitness.late)),
    load_and_tidy_fitness_data()[[2]] %&gt;% 
      group_by(line) %&gt;%
      summarise(early.male.rival = sum(early.male.rival),
                early.male.focal = sum(early.male.focal),
                late.male.rival = sum(late.male.rival),
                late.male.focal = sum(late.male.focal),
                male.fitness.early = early.male.focal / (early.male.focal + early.male.rival),
                male.fitness.late = late.male.focal / (late.male.focal + late.male.rival)) %&gt;%
      select(line, male.fitness.early, male.fitness.late),
    by = &quot;line&quot;) %&gt;% as.data.frame()
}

# A function to calculate each line&#39;s expected fitness, using brms models to adjust for block + vial random effects
# For the male late life assay, we also adjust for the number of competitor males that died 
# (this assumes that their deaths happened randomly with respect to which line we were measuring)
get_predicted_line_means &lt;- function(overwrite = FALSE){
  
  # If it&#39;s already calculated, just load up the file. Otherwise, make the file
  out.file &lt;- &quot;data/derived/predicted_line_means.csv&quot;
  if(file.exists(out.file) &amp; !overwrite) return(read_csv(out.file))
  
  ####### Load the individual-level data for females
  female_fitness &lt;- load_and_tidy_fitness_data()[[1]] 
  
  ####### Load the individual-level data for males, and impute missing values
  
  # For one vial, there is no early-life measurement because the females did not produce offspring. 
  # Therefore we impute that datapoint and make 5 imputed datasets using the package &#39;mice&#39;
  # See this link for explanation: https://cran.r-project.org/web/packages/brms/vignettes/brms_missings.html
  male_fitness &lt;- load_and_tidy_fitness_data()[[2]]
  male_fitness &lt;- male_fitness %&gt;% 
    mutate(early.male.rival = replace(early.male.rival, early.male.rival + early.male.focal == 0, NA),
           early.male.focal = replace(early.male.focal, is.na(early.male.rival), NA)) 
  male_fitness_imputed &lt;- mice(male_fitness, m = 5, print = FALSE, seed = 1)
  male_fitness_imputed &lt;- lapply(1:5, function(i) { # for each imputed dataset, 
    complete(male_fitness_imputed, i) %&gt;% 
      mutate(total1 = early.male.rival + early.male.focal, # tally up the focal+rival offspring for binomial model
             total2 = late.male.rival + late.male.focal,
             num.GFP.males = as.numeric(scale(num.GFP.males))) %&gt;% # scale this covariate
      # For late-life assays where the 5 focal males all died before reaching age 14, 
      # we need to assign them zero fitness. We did this by assuming that the rivals sired 100/100 offspring.
      # This scenario happened in 47 vials (out of 810)
      mutate(late.male.focal = replace(late.male.focal, total2 == 0, 100),
             total2 = replace(total2, total2 == 0, 100))
  })

  ####### Run model on the female data 
  
  prior_females &lt;- 
    c(set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;line&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;block&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;vial&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;line&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;block&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;vial&quot;))
  
  prior_males &lt;- 
    c(set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;earlymalefocal&#39;, group = &quot;line&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;earlymalefocal&#39;, group = &quot;block&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;earlymalefocal&#39;, group = &quot;vial&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;latemalefocal&#39;, group = &quot;line&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;latemalefocal&#39;, group = &quot;block&quot;),
      set_prior(&quot;cauchy(0, 0.1)&quot;, class = &quot;sd&quot;, resp = &#39;latemalefocal&#39;, group = &quot;vial&quot;))
      
   
  
  
  female1 &lt;- bf(female.fitness.early ~ (1 | p | line) + (1 | q | block) + (1 | r | vial))
  female2 &lt;- bf(female.fitness.late ~  (1 | p | line) + (1 | q | block) + (1 | r | vial))
  female_model &lt;- brm(female1 + female2, 
                      family = &quot;poisson&quot;, data = female_fitness,
                      prior = prior_females,
                      iter = 5000, chains = 4, cores = 4,
                      seed = 1, control = list(adapt_delta = 0.999, max_treedepth = 15)) 
  writeLines(capture.output(summary(female_model)), con = &quot;data/derived/model_summary_females.txt&quot;)
  
  saveRDS(female_model, &quot;data/derived/female_fitness_prediction_model.rds&quot;) # maybe delete later - probably not needed?
  
  
  ### Run model on male data 
  # Fit num.GFP.males, but only for the late life data
  male1 &lt;- bf(early.male.focal | trials(total1) ~ (1 | p | line) + (1 | q | block) + (1 | r | vial))
  male2 &lt;- bf(late.male.focal  | trials(total2) ~ num.GFP.males + (1 | p | line) + (1 | q | block) + (1 | r | vial))
  
  # Run 5 models, on each of the 5 imputed datasets, and combine the results
  male_model &lt;- brm_multiple(male1 + male2,
                             family = &quot;binomial&quot;, 
                             prior = prior_males,
                             data = male_fitness_imputed,
                             iter = 5000, chains = 4, cores = 4,
                             seed = 1, control = list(adapt_delta = 0.999, max_treedepth = 15)) 
  writeLines(capture.output(summary(male_model)), con = &quot;data/derived/model_summary_males.txt&quot;)
  saveRDS(male_model, &quot;data/derived/male_fitness_prediction_model.rds&quot;) # maybe delete later - probably not needed?
  male_model &lt;- readRDS(&quot;data/derived/male_fitness_prediction_model.rds&quot;) 
  
  ####### Define new data for prediction
  new.data.female &lt;- female_fitness %&gt;% 
    select(line) %&gt;% distinct() %&gt;% arrange(line)
  
  new.data.male &lt;- male_fitness %&gt;% 
    select(line) %&gt;% distinct() %&gt;% arrange(line) %&gt;% 
    mutate(total1 = 100, # I checked, and changing this number does not affect the line means, as expected
           total2 = 100, 
           num.GFP.males = male_fitness %&gt;% .$num.GFP.males %&gt;% mean(na.rm=TRUE))
  
  # Get predicted lines means for early and late life *female* fitness on linear predictor scale (i.e. fairly normal)
  female_line_preds &lt;- data.frame(new.data.female, 
                                  fitted(female_model, 
                                         newdata = new.data.female, 
                                         re_formula = &quot;~ ( 1 | line)&quot;, 
                                         scale = &quot;linear&quot;)) %&gt;%
    select(line, Estimate.femalefitnessearly, Estimate.femalefitnesslate) %&gt;% 
    rename(female.fitness.early = Estimate.femalefitnessearly,
           female.fitness.late = Estimate.femalefitnesslate) %&gt;%
    mutate(female.fitness.early = as.numeric(scale(female.fitness.early)), # scale the line means
           female.fitness.late = as.numeric(scale(female.fitness.late)))
  
  # Get predicted lines means for early and late life *male* fitness on linear predictor scale (i.e. fairly normal)
  male_line_preds &lt;- data.frame(new.data.male, 
                                fitted(male_model, 
                                       newdata = new.data.male, 
                                       re_formula = &quot;~ (1 | line)&quot;, 
                                       scale = &quot;linear&quot;)) %&gt;%
    select(line, Estimate.earlymalefocal, Estimate.latemalefocal) %&gt;% 
    rename(male.fitness.early = Estimate.earlymalefocal,
           male.fitness.late = Estimate.latemalefocal) %&gt;%
    mutate(male.fitness.early = as.numeric(scale(male.fitness.early)), # scale the line means
           male.fitness.late = as.numeric(scale(male.fitness.late))) 

  ## Define a function to get the ICC for line, for each posterior sample of predicted line means
  get_ICC_line &lt;- function(predictions){
    cols &lt;- str_detect(names(predictions), &quot;X&quot;)
    first_posterior_column &lt;- which(cols)[1] 
    n_draws &lt;- sum(cols)
    plan(&quot;multiprocess&quot;)
    future_lapply(0:(n_draws - 1), function(i){
       ICCbare(predictions$line, predictions[, i + first_posterior_column])
    }) %&gt;% unlist()
  }
  
  # Calculate ICC for each sample, separately for old and young females
  # For females, this is done on the estimated line mean progeny produced (i.e. response scale)
  predicted &lt;- fitted(female_model, summary = FALSE)
  Line_ICC_female_early &lt;- get_ICC_line(data.frame(female_fitness, t(predicted[, , 1])))
  Line_ICC_female_late &lt;- get_ICC_line(data.frame(female_fitness, t(predicted[, , 2])))
  
  # Do the same for males
  # For males, this is done on the estimated proportion progeny sired (i.e. response scale)
  new_data &lt;- male_fitness %&gt;% 
    select(block, line, vial.ID) %&gt;% 
    mutate(total1 = 100, 
           total2 = 100, 
           num.GFP.males = male_fitness %&gt;% .$num.GFP.males %&gt;% mean(na.rm=TRUE))
  predicted &lt;- fitted(male_model, summary = FALSE, newdata = new_data)
  Line_ICC_male_early &lt;- get_ICC_line(data.frame(new_data, t(predicted[, , 1])))
  Line_ICC_male_late &lt;- get_ICC_line(data.frame(new_data, t(predicted[, , 2])))
  
  ICC &lt;- list(Line_ICC_female_early, 
              Line_ICC_female_late,
              Line_ICC_male_early,
              Line_ICC_male_late) %&gt;% lapply(posterior_summary) %&gt;% 
    do.call(&quot;rbind&quot;, .) %&gt;% as_tibble() %&gt;%
    mutate(Fitness_trait = c(&quot;Female early-life fitness&quot;, &quot;Female late-life fitness&quot;,
                             &quot;Male early-life fitness&quot;, &quot;Male late-life fitness&quot;))
  write.csv(ICC, file = &quot;data/derived/ICC.csv&quot;, row.names = TRUE)
  

  # Record which block(s) each line was measured in. Generally one block, except for ref line (all blocks)
  line_blocks &lt;- rbind(female_fitness %&gt;% select(block, line), 
                       male_fitness   %&gt;% select(block, line)) %&gt;% 
    mutate(block = replace(block, line == &quot;352&quot;, &quot;reference_line&quot;)) %&gt;% 
    distinct() %&gt;% 
    group_by(line) %&gt;% summarise(block = paste0(block, collapse = &quot;, &quot;)) 
  
  predicted_line_means &lt;- female_line_preds %&gt;%
    left_join(male_line_preds, by = &quot;line&quot;) %&gt;% 
    left_join(line_blocks, by = &quot;line&quot;) %&gt;%
    arrange(line) 
  
  write.csv(predicted_line_means, out.file, row.names = FALSE)
}</code></pre>
</div>
<div id="predict-line-means-and-save-the-results" class="section level3">
<h3>Predict line means and save the results</h3>
<p>The object <code>line_means</code> is the raw line means: unscaled, arithmetic means and in the original units.</p>
<p>The object <code>predicted_line_means</code> is the estimated line means from the Bayesian models. These means are on the scale of the linear predictor (i.e. Poisson for females, binomial for males) rather than on the scale of the original response. They have been scaled to have a mean of 0 and a variance of 1.</p>
<pre class="r"><code>line_means &lt;- get_line_means()
predicted_line_means &lt;- get_predicted_line_means(overwrite = FALSE) </code></pre>
</div>
<div id="summary-of-model-used-to-predict-female-line-means" class="section level3">
<h3>Summary of model used to predict <strong>female</strong> line means:</h3>
<pre class="r"><code>cat(readLines(&#39;data/derived/model_summary_females.txt&#39;), sep = &#39;\n&#39;)</code></pre>
<pre style = "max-height:300px; overflow-y: auto;"> Family: MV(poisson, poisson) 
  Links: mu = log
         mu = log 
Formula: female.fitness.early ~ (1 | p | line) + (1 | q | block) + (1 | r | vial) 
         female.fitness.late ~ (1 | p | line) + (1 | q | block) + (1 | r | vial) 
   Data: female_fitness (Number of observations: 816) 
Samples: 4 chains, each with iter = 5000; warmup = 2500; thin = 1;
         total post-warmup samples = 10000

Group-Level Effects: 
~block (Number of levels: 10) 
                                                              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(femalefitnessearly_Intercept)                                  0.29      0.09     0.16     0.50 1.00     6955     7704
sd(femalefitnesslate_Intercept)                                   1.21      0.35     0.70     2.05 1.00     7505     8048
cor(femalefitnessearly_Intercept,femalefitnesslate_Intercept)    -0.42      0.29    -0.88     0.22 1.00     8010     7082

~line (Number of levels: 125) 
                                                              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(femalefitnessearly_Intercept)                                  0.58      0.05     0.50     0.68 1.00     5014     7236
sd(femalefitnesslate_Intercept)                                   1.82      0.17     1.51     2.18 1.00     5993     7883
cor(femalefitnessearly_Intercept,femalefitnesslate_Intercept)     0.68      0.07     0.53     0.79 1.00     7663     7679

~vial (Number of levels: 816) 
                                                              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(femalefitnessearly_Intercept)                                  0.52      0.02     0.49     0.55 1.00     4219     6935
sd(femalefitnesslate_Intercept)                                   1.54      0.07     1.42     1.68 1.00     4543     6625
cor(femalefitnessearly_Intercept,femalefitnesslate_Intercept)     0.21      0.05     0.12     0.30 1.00     7886     7637

Population-Level Effects: 
                             Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
femalefitnessearly_Intercept     4.40      0.11     4.19     4.62 1.00    11285     6886
femalefitnesslate_Intercept      0.70      0.44    -0.16     1.57 1.00    13404     7848

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).
</pre>
</div>
<div id="summary-of-model-used-to-predict-male-line-means" class="section level3">
<h3>Summary of model used to predict <strong>male</strong> line means:</h3>
<pre class="r"><code>cat(readLines(&#39;data/derived/model_summary_males.txt&#39;), sep = &#39;\n&#39;)</code></pre>
<pre style = "max-height:300px; overflow-y: auto;"> Family: MV(binomial, binomial) 
  Links: mu = logit
         mu = logit 
Formula: early.male.focal | trials(total1) ~ (1 | p | line) + (1 | q | block) + (1 | r | vial) 
         late.male.focal | trials(total2) ~ num.GFP.males + (1 | p | line) + (1 | q | block) + (1 | r | vial) 
   Data: male_fitness_imputed (Number of observations: 810) 
Samples: 20 chains, each with iter = 5000; warmup = 2500; thin = 1;
         total post-warmup samples = 50000

Group-Level Effects: 
~block (Number of levels: 10) 
                                                      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(earlymalefocal_Intercept)                              0.74      0.20     0.46     1.21 1.00    25462    31760
sd(latemalefocal_Intercept)                               0.34      0.22     0.01     0.83 1.00     6387    19201
cor(earlymalefocal_Intercept,latemalefocal_Intercept)     0.36      0.40    -0.62     0.94 1.00    30792    28410

~line (Number of levels: 124) 
                                                      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(earlymalefocal_Intercept)                              0.54      0.05     0.45     0.65 1.00    17106    27248
sd(latemalefocal_Intercept)                               0.85      0.20     0.45     1.22 1.01     2221     2337
cor(earlymalefocal_Intercept,latemalefocal_Intercept)     0.68      0.15     0.36     0.95 1.01     2784     3159

~vial (Number of levels: 810) 
                                                      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(earlymalefocal_Intercept)                              0.81      0.03     0.76     0.86 1.01     1459    30181
sd(latemalefocal_Intercept)                               2.79      0.11     2.58     3.01 1.00     9096    21491
cor(earlymalefocal_Intercept,latemalefocal_Intercept)     0.18      0.04     0.10     0.26 1.01     2437    14178

Population-Level Effects: 
                            Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
earlymalefocal_Intercept        0.15      0.25    -0.34     0.66 1.00    23930    27025
latemalefocal_Intercept        -0.81      0.19    -1.17    -0.42 1.00    25469    27873
latemalefocal_num.GFP.males    -0.13      0.12    -0.37     0.10 1.00    13718    23568

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).
</pre>
</div>
<div id="percent-of-variance-in-fitness-explained-by-dgrp-line" class="section level3">
<h3>Percent of variance in fitness explained by ‘DGRP line’:</h3>
<p>The table shows the posterior estimate of the intra-class correlation coefficient (ICC, which measures the proportion of variance within and between lines), using the above models to adjust for block effects (and one covariate in the male late-life fitness analysis). ICC was calculated using <code>ICC::ICCbare</code>.</p>
<pre class="r"><code>read.csv(&#39;data/derived/ICC.csv&#39;) %&gt;% 
  select(Fitness_trait, Estimate, Est.Error, Q2.5, Q97.5) %&gt;%
  rename(`Fitness trait` = Fitness_trait, `Est. Error` = Est.Error, `Lower 95% CI` = Q2.5, `Upper 95% CI` = Q97.5) %&gt;% 
  mutate(across(where(is.numeric), ~ .x * 100)) %&gt;%
  kable(digits = 2) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Fitness trait
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Est. Error
</th>
<th style="text-align:right;">
Lower 95% CI
</th>
<th style="text-align:right;">
Upper 95% CI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female early-life fitness
</td>
<td style="text-align:right;">
62.53
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
61.23
</td>
<td style="text-align:right;">
63.85
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late-life fitness
</td>
<td style="text-align:right;">
55.99
</td>
<td style="text-align:right;">
0.99
</td>
<td style="text-align:right;">
54.09
</td>
<td style="text-align:right;">
57.96
</td>
</tr>
<tr>
<td style="text-align:left;">
Male early-life fitness
</td>
<td style="text-align:right;">
44.38
</td>
<td style="text-align:right;">
0.97
</td>
<td style="text-align:right;">
42.47
</td>
<td style="text-align:right;">
46.26
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late-life fitness
</td>
<td style="text-align:right;">
9.26
</td>
<td style="text-align:right;">
6.74
</td>
<td style="text-align:right;">
-1.53
</td>
<td style="text-align:right;">
19.89
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="estimate-genetic-covariance-from-the-genomic-relatedness-matrix" class="section level2">
<h2>Estimate genetic (co)variance from the genomic relatedness matrix</h2>
<p>The following fits a multivariate mixed model using <code>brms</code>, to estimate the amount of variance explained by the DGRP line, accounting for the genomic relatedness matrix. The response variables are the predicted line means for the four phenotypes, as estimated in the previous section. We fit line as a random effect, specifying a covariance structure among the line effects that is equal to the genomic relatedness matrix (GRM), which we estimate using the SNP genotypes of the full sample of 205 DGRP lines. We use a somewhat informative prior to help the model converge; the prior is that the line and residual variances are not large (which must be true, given that the response variables have overall variance of 1), and that only 10% of the phenotype variance is explained by line (the posterior estimates suggest much higher line variance, indicating strong signal in the data).</p>
<p>The table below indicates that much of the variance in fitness is due to additive genetic effects, and that there is a positive geneti</p>
<pre class="r"><code># Run quant gen model using brms
if(!file.exists(&quot;data/derived/brms_quant_gen.rds&quot;)){
  
  library(bigsnpr)
  library(MCMCglmm)
  library(sommer)
  
  # Get the data
  mcmc_dat &lt;- predicted_line_means %&gt;% 
    mutate(line = factor(str_remove_all(line, &quot;_&quot;))) %&gt;%
    filter(!is.na(male.fitness.early)) %&gt;%
    rename(femalefitnessearly = female.fitness.early,
           femalefitnesslate = female.fitness.late,
           malefitnessearly = male.fitness.early,
           malefitnesslate = male.fitness.late)
  
  # Make inverse genomic relatedness matrix
  bigsnp &lt;- snp_readBed(&quot;data/derived/dgrp2_QC_all_lines.bed&quot;, backingfile = tempfile())
  bigsnp_attached &lt;- snp_attach(bigsnp)
  G &lt;- A.mat(bigsnp_attached$genotypes[1:nrow(bigsnp_attached$genotypes), 
                                       1:ncol(bigsnp_attached$genotypes)] - 1) 
  colnames(G) &lt;- bigsnp_attached$fam[,1]
  rownames(G) &lt;- bigsnp_attached$fam[,1]
  focal_lines &lt;- mcmc_dat %&gt;% pull(line)
  G &lt;- G[rownames(G) %in% focal_lines, 
         colnames(G) %in% focal_lines]
  G &lt;- cov2cor(G)
  
  brms_quant_gen &lt;- brm(
    bf(femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) +
   #   bf(femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) +
      bf(malefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) +
   #   bf(malefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) +
      set_rescor(FALSE),
    data = mcmc_dat, data2 = list(G = G), 
    iter = 30000, chains = 4, cores = 1, warmup = 20000,
    control = list(adapt_delta = 0.99999, max_treedepth = 20),
    
    prior = c(
      prior(normal(0.5, 0.3), &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;line&quot;),
      # prior(normal(0.5, 0.12), &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;line&quot;),
      prior(normal(0.5, 0.3), &quot;sd&quot;, resp = &#39;malefitnessearly&#39;, group = &quot;line&quot;),
      # prior(normal(0.5, 0.12), &quot;sd&quot;, resp = &#39;malefitnesslate&#39;, group = &quot;line&quot;),
      prior(normal(0.5, 0.3), &quot;sigma&quot;, resp = &#39;femalefitnessearly&#39;),
      # prior(normal(0.5, 0.12), &quot;sigma&quot;, resp = &#39;femalefitnesslate&#39;),
      prior(normal(0.5, 0.3), &quot;sigma&quot;, resp = &#39;malefitnessearly&#39;),
      prior(lkj(0.5), class = &quot;cor&quot;, group = &quot;line&quot;)#,
      # prior(normal(0.5, 0.12), &quot;sigma&quot;, resp = &#39;malefitnesslate&#39;)
    )
    
    # prior = c( # worth a try too...
    #   prior(Gamma(4,7), &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;line&quot;),
    #   prior(Gamma(4,7), &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;line&quot;),
    #   prior(Gamma(4,7), &quot;sd&quot;, resp = &#39;malefitnessearly&#39;, group = &quot;line&quot;),
    #   prior(Gamma(4,7), &quot;sd&quot;, resp = &#39;malefitnesslate&#39;, group = &quot;line&quot;),
    #   prior(Gamma(4,7), &quot;sigma&quot;, resp = &#39;femalefitnessearly&#39;),
    #   prior(Gamma(4,7), &quot;sigma&quot;, resp = &#39;femalefitnesslate&#39;),
    #   prior(Gamma(4,7), &quot;sigma&quot;, resp = &#39;malefitnessearly&#39;),
    #   prior(Gamma(4,7), &quot;sigma&quot;, resp = &#39;malefitnesslate&#39;)
    # )
    
    
    # prior = c(
    #   # prior(cauchy(0.5, 1), class = &quot;sigma&quot;, resp = &#39;femalefitnessearly&#39;),
    #   # prior(cauchy(0.5, 1), class = &quot;sigma&quot;, resp = &#39;femalefitnesslate&#39;),
    #   # prior(cauchy(0.5, 1), class = &quot;sigma&quot;, resp = &#39;malefitnessearly&#39;),
    #   # prior(cauchy(0.5, 1), class = &quot;sigma&quot;, resp = &#39;malefitnesslate&#39;),
    #   prior(cauchy(0.5, 1), class = &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;line&quot;),
    #   prior(cauchy(0.5, 1), class = &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;line&quot;),
    #   prior(cauchy(0.5, 1), class = &quot;sd&quot;, resp = &#39;malefitnessearly&#39;, group = &quot;line&quot;),
    #   prior(cauchy(0.5, 1), class = &quot;sd&quot;, resp = &#39;malefitnesslate&#39;, group = &quot;line&quot;),
    #   prior(lkj(0.5), class = &quot;cor&quot;, group = &quot;line&quot;)
    )
  )
  saveRDS(brms_quant_gen, &quot;data/derived/brms_quant_gen.rds&quot;)
} else {
  brms_quant_gen &lt;- readRDS(&quot;data/derived/brms_quant_gen.rds&quot;)
}

p &lt;- posterior_samples(brms_quant_gen)
p &lt;- tibble(
  VA_FE = sqrt(p[, &quot;sd_line__femalefitnessearly_Intercept&quot;]),
  VA_FL = sqrt(p[, &quot;sd_line__femalefitnesslate_Intercept&quot;]),
  VA_ME = sqrt(p[, &quot;sd_line__malefitnessearly_Intercept&quot;]),
  VA_ML = sqrt(p[, &quot;sd_line__malefitnesslate_Intercept&quot;]),
  h2_FE = VA_FE / (VA_FE + sqrt(p[, &quot;sigma_femalefitnessearly&quot;])),
  h2_FL = VA_FL / (VA_FL + sqrt(p[, &quot;sigma_femalefitnesslate&quot;])),
  h2_ME = VA_ME / (VA_ME + sqrt(p[, &quot;sigma_malefitnessearly&quot;])),
  h2_ML = VA_ML / (VA_ML + sqrt(p[, &quot;sigma_malefitnesslate&quot;])),
  corr_FE_FL = p[,&quot;cor_line__femalefitnessearly_Intercept__femalefitnesslate_Intercept&quot;],
  corr_FE_ME = p[,&quot;cor_line__femalefitnessearly_Intercept__malefitnessearly_Intercept&quot;],
  corr_FL_ME = p[,&quot;cor_line__femalefitnesslate_Intercept__malefitnessearly_Intercept&quot;],
  corr_FE_ML = p[,&quot;cor_line__femalefitnessearly_Intercept__malefitnesslate_Intercept&quot;],
  corr_FL_ML = p[,&quot;cor_line__femalefitnesslate_Intercept__malefitnesslate_Intercept&quot;],
  corr_ME_ML = p[,&quot;cor_line__malefitnessearly_Intercept__malefitnesslate_Intercept&quot;],
  cov_FE_FL = corr_FE_FL * VA_FE^2 * VA_FL^2,
  cov_FE_ME = corr_FE_ME * VA_FE^2 * VA_ME^2,
  cov_FL_ME = corr_FL_ME * VA_FL^2 * VA_ME^2,
  cov_FE_ML = corr_FE_ML * VA_FE^2 * VA_ML^2,
  cov_FL_ML = corr_FL_ML * VA_FL^2 * VA_ML^2,
  cov_ME_ML = corr_ME_ML * VA_ME^2 * VA_ML^2,
  corr_FE_ME_minus_corr_FL_ML = corr_FE_ME - corr_FL_ML,
  corr_FE_FL_minus_corr_ME_ML = corr_FE_FL - corr_ME_ML
) %&gt;% gather(Parameter, value)

p %&gt;%
  group_by(Parameter) %&gt;%
  summarise(summ = list(posterior_summary(value)),
            Estimate = map_dbl(summ, ~ .x[[1]]),
            `Est. Error` = map_dbl(summ, ~ .x[[2]]),
            `Lower 95% CI` = map_dbl(summ, ~ .x[[3]]),
            `Upper 95% CI` = map_dbl(summ, ~ .x[[4]]),
            .groups = &quot;drop&quot;) %&gt;%
  dplyr::select(-summ) %&gt;%
  mutate(sp = str_split(Parameter, &quot;_&quot;),
         Type = factor(map_chr(sp, ~ .x[[1]]), c(&quot;h2&quot;, &quot;corr&quot;, &quot;VA&quot;, &quot;cov&quot;)),
         Parameter = map_chr(sp, ~ paste0(.x[-1], collapse = &quot; - &quot;))) %&gt;%
  arrange(Type) %&gt;%
  dplyr::select(-sp, -Type) %&gt;%
  kable(digits = 3) %&gt;%
  kable_styling(full_width = FALSE) %&gt;%
  pack_rows(&quot;Heritabilities&quot;, 1, 4) %&gt;%
  pack_rows(&quot;Genetic correlations&quot;, 5, 10) %&gt;%
  pack_rows(&quot;Additive genetic variances&quot;, 11, 14) %&gt;%
  pack_rows(&quot;Additive genetic covariances&quot;, 15, 20)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre style = "max-height:300px; overflow-y: auto;">R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.1.0   future.apply_1.5.0 future_1.17.0      ICC_2.3.0         
 [5] mice_3.9.0         brms_2.14.4        Rcpp_1.0.4.6       forcats_0.5.0     
 [9] stringr_1.4.0      dplyr_1.0.0        purrr_0.3.4        readr_1.3.1       
[13] tidyr_1.1.0        tibble_3.0.1       ggplot2_3.3.2      tidyverse_1.3.0   
[17] knitrhooks_0.0.4   knitr_1.30         workflowr_1.6.2   

loaded via a namespace (and not attached):
  [1] readxl_1.3.1         backports_1.1.7      plyr_1.8.6          
  [4] igraph_1.2.5         splines_4.0.3        listenv_0.8.0       
  [7] crosstalk_1.1.0.1    TH.data_1.0-10       rstantools_2.1.1    
 [10] inline_0.3.15        digest_0.6.25        htmltools_0.5.0     
 [13] rsconnect_0.8.16     fansi_0.4.1          magrittr_2.0.1      
 [16] globals_0.12.5       modelr_0.1.8         RcppParallel_5.0.1  
 [19] matrixStats_0.56.0   xts_0.12-0           sandwich_2.5-1      
 [22] prettyunits_1.1.1    colorspace_1.4-1     blob_1.2.1          
 [25] rvest_0.3.5          haven_2.3.1          xfun_0.19           
 [28] callr_3.4.3          crayon_1.3.4         jsonlite_1.7.0      
 [31] lme4_1.1-23          survival_3.2-7       zoo_1.8-8           
 [34] glue_1.4.2           gtable_0.3.0         emmeans_1.4.7       
 [37] webshot_0.5.2        V8_3.4.0             pkgbuild_1.0.8      
 [40] rstan_2.21.2         abind_1.4-5          scales_1.1.1        
 [43] mvtnorm_1.1-0        DBI_1.1.0            miniUI_0.1.1.1      
 [46] viridisLite_0.3.0    xtable_1.8-4         stats4_4.0.3        
 [49] StanHeaders_2.21.0-3 DT_0.13              htmlwidgets_1.5.1   
 [52] httr_1.4.1           threejs_0.3.3        ellipsis_0.3.1      
 [55] pkgconfig_2.0.3      loo_2.3.1            dbplyr_1.4.4        
 [58] tidyselect_1.1.0     rlang_0.4.6          reshape2_1.4.4      
 [61] later_1.0.0          munsell_0.5.0        cellranger_1.1.0    
 [64] tools_4.0.3          cli_2.0.2            generics_0.0.2      
 [67] broom_0.5.6          ggridges_0.5.2       evaluate_0.14       
 [70] fastmap_1.0.1        yaml_2.2.1           processx_3.4.2      
 [73] fs_1.4.1             nlme_3.1-149         whisker_0.4         
 [76] mime_0.9             projpred_2.0.2       xml2_1.3.2          
 [79] compiler_4.0.3       bayesplot_1.7.2      shinythemes_1.1.2   
 [82] rstudioapi_0.11      gamm4_0.2-6          curl_4.3            
 [85] reprex_0.3.0         statmod_1.4.34       stringi_1.5.3       
 [88] highr_0.8            ps_1.3.3             Brobdingnag_1.2-6   
 [91] lattice_0.20-41      Matrix_1.2-18        nloptr_1.2.2.1      
 [94] markdown_1.1         shinyjs_1.1          vctrs_0.3.0         
 [97] pillar_1.4.4         lifecycle_0.2.0      bridgesampling_1.0-0
[100] estimability_1.3     httpuv_1.5.3.1       R6_2.4.1            
[103] promises_1.1.0       gridExtra_2.3        codetools_0.2-16    
[106] boot_1.3-25          colourpicker_1.0     MASS_7.3-53         
[109] gtools_3.8.2         assertthat_0.2.1     rprojroot_1.3-2     
[112] withr_2.2.0          shinystan_2.5.0      multcomp_1.4-13     
[115] mgcv_1.8-33          parallel_4.0.3       hms_0.5.3           
[118] grid_4.0.3           coda_0.19-3          minqa_1.2.4         
[121] rmarkdown_2.5        git2r_0.27.1         shiny_1.4.0.2       
[124] lubridate_1.7.8      base64enc_0.1-3      dygraphs_1.1.1.6    
</pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
