<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Correcting the GWAS results using multivariate adaptive shrinkage</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Correcting the GWAS results using
multivariate adaptive shrinkage</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree6b63c187d70dfe7d9a5c1a06528d2691cecf7d00targetblank6b63c18a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/fitnessGWAS/tree/6b63c187d70dfe7d9a5c1a06528d2691cecf7d00" target="_blank">6b63c18</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree6b63c187d70dfe7d9a5c1a06528d2691cecf7d00targetblank6b63c18a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/fitnessGWAS/tree/6b63c187d70dfe7d9a5c1a06528d2691cecf7d00" target="_blank">6b63c18</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  JAGS-4.3.0/
    Untracked:  biv_mod.jags
    Untracked:  old_analyses/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/gwas_adaptive_shrinkage.Rmd</code>) and HTML
(<code>docs/gwas_adaptive_shrinkage.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/6b63c187d70dfe7d9a5c1a06528d2691cecf7d00/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">6b63c18</a>
</td>
<td>
lukeholman
</td>
<td>
2023-08-10
</td>
<td>
wflow_publish("analysis/gwas_adaptive_shrinkage.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/16ba9478b80e22a3b4c704d7710fe36e267d129e/docs/gwas_adaptive_shrinkage.html" target="_blank">16ba947</a>
</td>
<td>
lukeholman
</td>
<td>
2023-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/926d76f46d3853c29d793048d1c9ec4b23f9bec4/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">926d76f</a>
</td>
<td>
lukeholman
</td>
<td>
2023-07-31
</td>
<td>
wflow_publish("analysis/gwas_adaptive_shrinkage.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3a5750d27bd2a271c583d40eae71f0ac91f48159/docs/gwas_adaptive_shrinkage.html" target="_blank">3a5750d</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/a5026e756507f24bd0cf7c395914164748e68f77/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">a5026e7</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
wflow_publish("analysis/gwas_adaptive_shrinkage.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/5078e01b5db118785e4b6b5e0966e9374fc951f9/docs/gwas_adaptive_shrinkage.html" target="_blank">5078e01</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/0ab0260cbb5c427f3b2001bd7c7ce6e92cce3697/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">0ab0260</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
wflow_publish("analysis/gwas_adaptive_shrinkage.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/gwas_adaptive_shrinkage.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/gwas_adaptive_shrinkage.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3855d3359649c74f955042725d07276742cb649a/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3855d3359649c74f955042725d07276742cb649a/docs/gwas_adaptive_shrinkage.html" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/gwas_adaptive_shrinkage.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/gwas_adaptive_shrinkage.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<!-- Note: `mashr` requires a large amount of RAM, and so to get this script to run on a very large SNP dataset, you might need to launch R from the Terminal with extra memory allocation. I used `knitr::purl(input = "analysis/gwas_adaptive_shrinkage.Rmd", output = "code/run_mashr.R")` to generate an R script from this R Markdown document, then `env R_MAX_VSIZE=700Gb Rscript code/run_mashr.R`, and then run on a high-RAM computer. -->
<pre class="r"><code>library(tidyverse)
library(ashr)
library(mashr) 
library(glue)
library(kableExtra)
library(gridExtra)

# Get the list of SNPs (or SNP clumps in 100% LD) which are in approx. LD with one another
get_mashr_data &lt;- function(){
  
  SNPs_in_LD &lt;- read.table(&quot;data/derived/SNPs_in_LD&quot;, header = FALSE)
  
  data_for_mashr &lt;- read_csv(&quot;data/derived/all_univariate_GEMMA_results.csv&quot;) %&gt;% 
    filter(str_detect(SNPs, &quot;,&quot;) | SNPs %in% SNPs_in_LD$V1) %&gt;% 
    select(SNPs, starts_with(&quot;beta&quot;), starts_with(&quot;SE&quot;)) 
  
  single_snps &lt;- data_for_mashr %&gt;% filter(!str_detect(SNPs, &quot;,&quot;))
  multiple_snps &lt;- data_for_mashr %&gt;% filter(str_detect(SNPs, &quot;,&quot;))
  multiple_snps &lt;- multiple_snps[enframe(strsplit(multiple_snps$SNPs, split = &quot;, &quot;)) %&gt;% 
                                   unnest(value) %&gt;% 
                                   filter(value %in% SNPs_in_LD$V1) %&gt;% 
                                   pull(name), ]
  bind_rows(single_snps, multiple_snps) %&gt;% arrange(SNPs)
}</code></pre>
<div id="run-multivariate-adaptive-shrinkage-using-mashr"
class="section level2">
<h2>Run multivariate adaptive shrinkage using <code>mashr</code></h2>
<p>The following code runs the package <code>mashr</code>, which
attempts to infer more accurate estimates of the true SNP effect sizes
by applying shrinkage based on the multivariate structure of the data.
<code>mashr</code> accepts a matrix of four effect sizes for each locus
(one fore each fitness trait) as well as the four associated standard
errors. <code>mashr</code> then uses Bayesian multivariate mixture
models to attempt to improve the estimates of the true effect sizes by
applying shrinkage. It can also calculate the local false sign rate and
other useful things. For more information on <code>mashr</code>, see the
<a href="https://stephenslab.github.io/mashr/index.html">package
website</a>, and the associated paper by Urbut, Wang, and Stephens (<a
href="doi:10.1101/096552" class="uri">doi:10.1101/096552</a>).</p>
<p>There are two ways to run <code>mashr</code>: fitting a mixture model
where the covariance matrices describing the true relationships between
the effects sizes are estimated from the data by a deconvolution
algorithm (termed the ‘data-driven’ approach), and an alternative
approach where these covariance matrices are defined <em>a priori</em>
by the user (the ‘canonical’ approach). The following code runs both
approaches, though only the analysis using the “data-driven” approach is
presented in the paper (we felt that the canonical approach did not add
any new insights, and the likelihood of the <code>mashr</code> model was
far better when using the data-driven approach).</p>
<!-- In the 'canonical' analysis, we were _a priori_ interested in determining the relative abundances of variants that affect fitness in the following list of possible ways: -->
<!--   - null (no effect on fitness) -->
<!--   - uniform effect (the fitness effect of the allele is identical on all sexes and age classes; termed `equal_effects` in `mashr`) -->
<!--   - same sign, variable magnitude (the fitness effect of the allele has the same sign for all sexes and age classes, though its magnitude is variable; termed `simple_het` in `mashr`) -->
<!--   - sexually antagonistic (one allele is good for one sex and bad for the other, though this effect is the same in young and old individuals) -->
<!--   - age antagonistic (one allele is good for one age class and bad for the other, though this effect is the same in males and females) -->
<!--   - female-specific (there is an effect in females but no effect in males; same sign of effect across age classes) -->
<!--   - male-specific (there is an effect in males but no effect in females; same sign of effect across age classes) -->
<!--   - early-life-specific (the locus affects the fitness of young flies of both sexes, but has no effect on old flies) -->
<!--   - late-life-specific (the locus affects the fitness of old flies of both sexes, but has no effect on young flies) -->
<!--   - female-specific effect, with a variable effect across age classes -->
<!--   - male-specific effect, with an effect magnitude and/or sign that changes with age -->
<!--   - female-specific effect, with an effect magnitude and/or sign that changes with age -->
<!--   - early-life-specific effect, with an effect magnitude and/or sign that differs between sexes -->
<!--   - late-life-specific effect, with an effect magnitude and/or sign that differs between sexes -->
<!-- These categories were chosen because our _a priori_ hypothesis is that different loci conceivably affect fitness in a manner that depends on age, sex, and the age-sex interaction. There were 46 covariance matrices, including the null, though most of these are inferred to be absent or very rare and thus have essentially no influence on the results of the mixture model. -->
<p>The <code>mashr</code> analysis use the default ‘null-biased’ prior,
meaning that our prior is that loci with no effect on any of the fitness
components are 10-fold more common than any of the other possibilities
(reflecting our expectation that genetic polymorphisms with strong
effects on fitness are probably rare).</p>
<p>Because <code>mashr</code> models are computationally intensive, and
to avoid issues of pseudoreplication in downstream analyses, we ran
<code>mashr</code> on a subset of the 1,207,357 loci that were analysed
using GEMMA. We pruned this list to a subset of 209,053 loci that were
in approximate linkage disequilibrium with one another.</p>
<pre class="r"><code>run_mashr &lt;- function(beta_and_se, mashr_mode, ED_p_cutoff = NULL){
  
  print(str(beta_and_se))
  
  mashr_setup &lt;- function(beta_and_se){
    betas &lt;- beta_and_se %&gt;% select(starts_with(&quot;beta&quot;)) %&gt;% as.matrix()
    SEs &lt;- beta_and_se %&gt;% select(starts_with(&quot;SE&quot;)) %&gt;% as.matrix()
    rownames(betas) &lt;- beta_and_se$SNP
    rownames(SEs) &lt;- beta_and_se$SNP
    mash_set_data(betas, SEs)
  }
  
  mash_data &lt;- mashr_setup(beta_and_se)
  
  # Setting mashr_mode == &quot;ED&quot; makes mashr select the covariance matrices for us, using the
  # software Extreme Deconvolution. This software &quot;reconstructs the error-deconvolved or &#39;underlying&#39; 
  # distribution function common to all samples, even when the individual data points are samples from different distributions&quot;
  # Following the mashr vignette, we initialise the algorithm in ED using the principal components of the strongest effects in the dataset
  # Reference for ED: https://arxiv.org/abs/0905.2979
  if(mashr_mode == &quot;ED&quot;){
    # Find the strongest effects in the data
    m.1by1 &lt;- mash_1by1(mash_data) 
    strong &lt;- get_significant_results(m.1by1, thresh = ED_p_cutoff)   
    # Obtain data-driven covariance matrices by running Extreme Deconvolution
    U.pca &lt;- cov_pca(mash_data, npc = 4, subset = strong)
    U &lt;- cov_ed(mash_data, U.pca, subset = strong)
  }
  
  # Otherwise, we define the covariance matrices ourselves (a long list of a priori interesting matrices are checked)
  if(mashr_mode == &quot;canonical&quot;){
    make_SA_matrix &lt;- function(r) matrix(c(1,1,r,r,1,1,r,r,r,r,1,1,r,r,1,1), ncol=4)
    make_age_antag_matrix &lt;- function(r) matrix(c(1,r,1,r,r,1,r,1,1,r,1,r,r,1,r,1), ncol=4)
    make_sex_specific &lt;- function(mat, sex){
      if(sex == &quot;F&quot;) {mat[, 3:4] &lt;- 0;  mat[3:4, ] &lt;- 0}
      if(sex == &quot;M&quot;) {mat[, 1:2] &lt;- 0; mat[1:2, ] &lt;- 0}
      mat
    }
    make_age_specific &lt;- function(mat, age){
      if(age == &quot;early&quot;) {mat[, c(2,4)] &lt;- 0; mat[c(2,4), ] &lt;- 0}
      if(age == &quot;late&quot;)  {mat[, c(1,3)] &lt;- 0; mat[c(1,3), ] &lt;- 0}
      mat
    }

    add_matrix &lt;- function(mat, mat_list, name){
      mat_list[[length(mat_list) + 1]] &lt;- mat
      names(mat_list)[length(mat_list)] &lt;- name
      mat_list
    }
    id_matrix &lt;- matrix(1, ncol=4, nrow=4)

    # Get the mashr default canonical covariance matrices: this includes the ones
    # called &quot;null&quot;, &quot;uniform&quot;, and &quot;same sign&quot; in the list that precedes this code chunk
    U &lt;- cov_canonical(mash_data)

    # And now our custom covariance matrices:

    # Identical across ages, but sex-antagonistic
    U &lt;- make_SA_matrix(-0.25) %&gt;% add_matrix(U, &quot;Sex_antag_0.25&quot;)
    U &lt;- make_SA_matrix(-0.5) %&gt;% add_matrix(U, &quot;Sex_antag_0.5&quot;)
    U &lt;- make_SA_matrix(-0.75) %&gt;% add_matrix(U, &quot;Sex_antag_0.75&quot;)
    U &lt;- make_SA_matrix(-1) %&gt;% add_matrix(U, &quot;Sex_antag_1.0&quot;)

    # Identical across sexes, but age-antagonistic
    U &lt;- make_age_antag_matrix(-0.25) %&gt;% add_matrix(U, &quot;Age_antag_0.25&quot;)
    U &lt;- make_age_antag_matrix(-0.5) %&gt;% add_matrix(U, &quot;Age_antag_0.5&quot;)
    U &lt;- make_age_antag_matrix(-0.75) %&gt;% add_matrix(U, &quot;Age_antag_0.75&quot;)
    U &lt;- make_age_antag_matrix(-1) %&gt;% add_matrix(U, &quot;Age_antag_1.0&quot;)

    # Sex-specific, identical effect in young and old
    U &lt;- id_matrix %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_1&quot;)
    U &lt;- id_matrix %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_1&quot;)

    # Age-specific, identical effect in males and females
    U &lt;- id_matrix %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_specific_1&quot;)
    U &lt;- id_matrix %&gt;% make_age_specific(&quot;late&quot;)  %&gt;% add_matrix(U, &quot;Late_life_specific_1&quot;)

    # Positively correlated but variable effect across ages, and also sex-specific
    U &lt;- make_age_antag_matrix(0.25) %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_0.25&quot;)
    U &lt;- make_age_antag_matrix(0.5) %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_0.5&quot;)
    U &lt;- make_age_antag_matrix(0.75) %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_0.75&quot;)
    U &lt;- make_age_antag_matrix(0.25) %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_0.25&quot;)
    U &lt;- make_age_antag_matrix(0.5) %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_0.5&quot;)
    U &lt;- make_age_antag_matrix(0.75) %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_0.75&quot;)

    # Negatively correlated across ages, and also sex-specific
    U &lt;- make_age_antag_matrix(-0.25) %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_age_antag_0.25&quot;)
    U &lt;- make_age_antag_matrix(-0.5) %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_age_antag_0.5&quot;)
    U &lt;- make_age_antag_matrix(-0.75) %&gt;% make_sex_specific(&quot;F&quot;) %&gt;% add_matrix(U, &quot;Female_specific_age_antag_0.75&quot;)
    U &lt;- make_age_antag_matrix(-0.25) %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_age_antag_0.25&quot;)
    U &lt;- make_age_antag_matrix(-0.5) %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_age_antag_0.5&quot;)
    U &lt;- make_age_antag_matrix(-0.75) %&gt;% make_sex_specific(&quot;M&quot;) %&gt;% add_matrix(U, &quot;Male_specific_age_antag_0.75&quot;)

    # Positively correlated but variable effect across sexes, and also age-specific
    U &lt;- make_SA_matrix(0.25) %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_specific_0.25&quot;)
    U &lt;- make_SA_matrix(0.5) %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_specific_0.5&quot;)
    U &lt;- make_SA_matrix(0.75) %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_specific_0.75&quot;)
    U &lt;- make_SA_matrix(0.25) %&gt;% make_age_specific(&quot;late&quot;) %&gt;% add_matrix(U, &quot;Late_life_specific_0.25&quot;)
    U &lt;- make_SA_matrix(0.5) %&gt;% make_age_specific(&quot;late&quot;) %&gt;% add_matrix(U, &quot;Late_life_specific_0.5&quot;)
    U &lt;- make_SA_matrix(0.75) %&gt;% make_age_specific(&quot;late&quot;) %&gt;% add_matrix(U, &quot;Late_life_specific_0.75&quot;)

    # Negatively correlated but variable effect across sexes, and also age-specific
    U &lt;- make_SA_matrix(-0.25) %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_antag_0.25&quot;)
    U &lt;- make_SA_matrix(-0.5) %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_antag_0.5&quot;)
    U &lt;- make_SA_matrix(-0.75) %&gt;% make_age_specific(&quot;early&quot;) %&gt;% add_matrix(U, &quot;Early_life_antag_0.75&quot;)
    U &lt;- make_SA_matrix(-0.25) %&gt;% make_age_specific(&quot;late&quot;) %&gt;% add_matrix(U, &quot;Late_life_antag_0.25&quot;)
    U &lt;- make_SA_matrix(-0.5) %&gt;% make_age_specific(&quot;late&quot;) %&gt;% add_matrix(U, &quot;Late_life_antag_0.5&quot;)
    U &lt;- make_SA_matrix(-0.75) %&gt;% make_age_specific(&quot;late&quot;) %&gt;% add_matrix(U, &quot;Late_life_antag_0.75&quot;)
  }
  
  return(mash(data = mash_data, Ulist = U)) # Run mashr
}


if(!file.exists(&quot;data/derived/mashr_results_canonical.rds&quot;)){
  
  data_for_mashr &lt;- get_mashr_data()
  
  print(&quot;Starting the data-driven analysis&quot;)
  run_mashr(data_for_mashr, mashr_mode = &quot;ED&quot;, ED_p_cutoff = 0.2) %&gt;%
    saveRDS(file = &quot;data/derived/mashr_results_ED.rds&quot;)
  
  print(&quot;Starting the canonical analysis&quot;)
  run_mashr(data_for_mashr, mashr_mode = &quot;canonical&quot;) %&gt;%
    saveRDS(file = &quot;data/derived/mashr_results_canonical.rds&quot;)
  
  mashr_results_ED &lt;- read_rds(&quot;data/derived/mashr_results_ED.rds&quot;)
  mashr_results_canonical &lt;- read_rds(&quot;data/derived/mashr_results_canonical.rds&quot;)
  
} else {
  data_for_mashr &lt;- get_mashr_data()
  mashr_results_ED &lt;- read_rds(&quot;data/derived/mashr_results_ED.rds&quot;)
  mashr_results_canonical &lt;- read_rds(&quot;data/derived/mashr_results_canonical.rds&quot;)
}</code></pre>
<!-- ### Chromosome-specific analyses -->
<!-- This section runs a single canonical analysis for each of the major chromosomes (2L, 2R, 3L, 3R, and X; chromosome 4 was ommited because it carries only a few polymorphisms not in strong LD). The purpose of this analysis is to copare the inferred frequencies of different types of variants on each chromosome, e.g. to test whether the X chromosome is enriched for sexually antagonistic loci as predicted by some theoretical models. -->
<!-- ```{r mashr_by_chromosome} -->
<!-- mashr_one_chromosome <- function(chr){ -->
<!--   data_for_mashr <- get_mashr_data() -->
<!--   focal_data <- data_for_mashr %>%  -->
<!--     filter(grepl(glue("{chr}_"), SNPs)) %>% -->
<!--     select(starts_with("beta"), starts_with("SE")) -->
<!--   run_mashr(focal_data, mashr_mode = "canonical") %>% -->
<!--     saveRDS(file = glue("data/derived/mashr_results_canonical_chr{chr}.rds")) -->
<!-- } -->
<!-- if(!file.exists("data/derived/mashr_results_canonical_chrX.rds")){ -->
<!--   print("Starting the chromosome-specific analysis") -->
<!--   lapply(c("2L", "2R", "3L", "3R", "X"), mashr_one_chromosome) -->
<!-- }  -->
<!-- ``` -->
<!-- ## Find mixture probabilities for each SNP -->
<!-- This uses the canonical analysis' classifications. Each SNP gets a posterior probability that it belongs to the $i$'th mixture component -- only the mixture components that are not very rare are included. These are: `equal_effects` (i.e. the SNP is predicted to affect all 4 traits equally), female-specific and male-specific (i.e. an effect on females/males only, which is concordant across age categories), sexually antagonistic (again, regardless of age), and null. The null category is the rarest one, despite the prior assuming null SNPs are $10\times$ more common than any other type. The analysis therefore suggests that most SNPs either affect some/all of our 4 phenotypes, or (more likely) are in linkage disequilibrium with a SNP which does.  -->
<!-- ```{r get_mixture_assignments, message=FALSE} -->
<!-- # Get the mixture weights, as advised by mash authors here: https://github.com/stephenslab/mashr/issues/68 -->
<!-- posterior_weights_cov <- mashr_results_canonical$posterior_weights  -->
<!-- colnames(posterior_weights_cov) <- sapply( -->
<!--   str_split(colnames(posterior_weights_cov), '\\.'),  -->
<!--   function(x) { -->
<!--     if(length(x) == 1) return(x) -->
<!--     else if(length(x) == 2) return(x[1]) -->
<!--     else if(length(x) == 3) return(paste(x[1], x[2], sep = ".")) -->
<!--   }) -->
<!-- posterior_weights_cov <- t(rowsum(t(posterior_weights_cov),  -->
<!--                                   colnames(posterior_weights_cov))) -->
<!-- data_for_mashr <- get_mashr_data() -->
<!-- # Make a neat dataframe -->
<!-- mixture_assignment_probabilities <- data.frame( -->
<!--   SNP_clump = data_for_mashr$SNPs, -->
<!--   posterior_weights_cov, -->
<!--   stringsAsFactors = FALSE -->
<!-- ) %>% as_tibble() %>% -->
<!--   rename(P_equal_effects = equal_effects, -->
<!--          P_female_specific = Female_specific_1, -->
<!--          P_male_specific = Male_specific_1, -->
<!--          P_null = null, -->
<!--          P_sex_antag = Sex_antag_0.25) -->
<!-- ``` -->
</div>
<div id="add-the-mashr-results-to-the-database" class="section level2">
<h2>Add the <code>mashr</code> results to the database</h2>
<p>Here, we make a single large dataframe holding all of the ‘raw’
results from the univariate GEMMA analysis, and the corresponding
“shrinked” results from <code>mashr</code> (for both the canonical and
data-driven <code>mashr</code> analyses). Because it is so large, we add
this sheet of results to the database, allowing memory-efficient
searching, joins, etc.</p>
<pre class="r"><code>all_univariate_lmm_results &lt;- read_csv(&quot;data/derived/all_univariate_GEMMA_results.csv&quot;) %&gt;% 
  rename_at(vars(-SNPs), ~ str_c(., &quot;_raw&quot;))

mashr_snps &lt;- data_for_mashr$SNPs

# canonical_estimates &lt;- get_pm(mashr_results_canonical) %&gt;% 
#   as_tibble() %&gt;% 
#   rename_all(~str_c(., &quot;_mashr_canonical&quot;))

ED_estimates &lt;- get_pm(mashr_results_ED) %&gt;% 
  as_tibble() %&gt;% 
  rename_all(~str_c(., &quot;_mashr_ED&quot;))

# lfsr_canonical &lt;- get_lfsr(mashr_results_canonical) %&gt;% 
#   as_tibble() %&gt;% 
#   rename_all(~str_replace_all(., &quot;beta&quot;, &quot;LFSR&quot;)) %&gt;% 
#   rename_all(~str_c(., &quot;_mashr_canonical&quot;))

lfsr_ED &lt;- get_lfsr(mashr_results_ED) %&gt;% 
  as_tibble() %&gt;% 
  rename_all(~str_replace_all(., &quot;beta&quot;, &quot;LFSR&quot;)) %&gt;% 
  rename_all(~str_c(., &quot;_mashr_ED&quot;))


all_mashr_results &lt;- bind_cols(
  tibble(SNPs = mashr_snps), 
  # canonical_estimates, 
  ED_estimates, 
  # lfsr_canonical, 
  lfsr_ED)

all_univariate_lmm_results &lt;- left_join(
  all_univariate_lmm_results, 
  all_mashr_results, by = &quot;SNPs&quot;)

nested &lt;- all_univariate_lmm_results %&gt;% 
  filter(str_detect(SNPs, &quot;, &quot;))
split_snps &lt;- strsplit(nested$SNPs, split = &quot;, &quot;)
nested &lt;- lapply(1:nrow(nested), 
                 function(i) {
                   data.frame(SNP = split_snps[[i]],    
                              SNP_clump = nested$SNPs[i],
                              nested[i,] %&gt;% select(-SNPs), stringsAsFactors = FALSE)
                   }) %&gt;%
  do.call(&quot;rbind&quot;, .) %&gt;% as_tibble()
rm(split_snps)

all_univariate_lmm_results &lt;- all_univariate_lmm_results %&gt;% 
  filter(!str_detect(SNPs, &quot;, &quot;)) %&gt;%
  rename(SNP_clump = SNPs) %&gt;% mutate(SNP = SNP_clump) %&gt;%
  select(SNP, SNP_clump, everything()) %&gt;%
  bind_rows(nested) %&gt;%
  arrange(SNP)

# Merge in the mixture proportions
# all_univariate_lmm_results &lt;- 
#   all_univariate_lmm_results %&gt;%
#   left_join(mixture_assignment_probabilities, by = &quot;SNP_clump&quot;)

db &lt;- DBI::dbConnect(RSQLite::SQLite(), 
                     &quot;data/derived/annotations.sqlite3&quot;, create = FALSE)

DBI::dbRemoveTable(db, &quot;univariate_lmm_results&quot;)
db %&gt;% copy_to(all_univariate_lmm_results,
               &quot;univariate_lmm_results&quot;, temporary = FALSE)</code></pre>
<!-- ### Log likelihood -->
<!-- The data-driven covariance matrices have a likelihood that is 99.2% as high as for the canonical matrices, even though the canonical analysis has far, far more parameters (46 matrices vs 6). This indicates that the data-driven covariance matrices provide a better fit to the data, as expected (see: https://stephenslab.github.io/mashr/articles/simulate_noncanon.html). Thus, we use the data-driven covariance matrices when we wish to derive 'adjusted' effect sizes for each SNP (i.e. adjusted for winner's/loser's curse effects, and for the statistically inferred covariance structure for the variant effects on the four phenotypes). The canonical covariance matrices are instead used for classifying SNPs into easy-to-see categories (e.g. sex-specific, sexually antagonistic, concordant, etc), and estimating the % SNPs that belong to each category -->
<!-- ```{r load_mashr_results, echo=FALSE} -->
<!-- mashr_results_canonical <- read_rds("data/derived/mashr_results_canonical.rds") -->
<!-- mashr_results_ED <- read_rds("data/derived/mashr_results_ED.rds") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- tibble(`Mashr version` = c("A. Data-driven covariance matrices", -->
<!--                            "B. Cononical covariance matrices", -->
<!--                            "Likelihood ratio (A / B)"), -->
<!--        `Log likelihood` = c(get_loglik(mashr_results_ED), -->
<!--                             get_loglik(mashr_results_canonical), -->
<!--                             get_loglik(mashr_results_ED) / get_loglik(mashr_results_canonical))) %>% -->
<!--   kable() %>% -->
<!--   kable_styling() -->
<!-- ``` -->
<!-- ## Proportion of 'significant' effect sizes which share the same sign for each pair of phenotypes -->
<!-- To find the proportion of sex or age-antagonistic SNPs among the 209,053 that were tested using `mashr`, we simply take 1 minus this matrix, e.g. $100\times(1 - 0.976) = 2.4\%$ of the loci that significantly (i.e. LFSR < 0.05) affected both male and female early-life fitness had an _opposite_ effect on each sex (as opposed to a same-sign effect).  -->
<!-- Note that this approach is a very conservative way of measuring the % sexually antagonistic (or age antagonistic) loci in the genome, because the LFSR needs to be < 0.05 for _both_ traits (meaning there are 2 chances to make a 'false negative', which matters because the power is low for any individual locus). Also note that this matrix was calculated using the data-driven (not canonical) covariance matrices, which are expected to more accurately estimate effect size of each SNP. -->
<!-- The matrix highlights that there are zero loci where one allele significantly improves early-life fitness and lower late-life fitness with a sex (they might exist, but none were detected using this conservative test and our data). However, some sexually antagonistic loci were detected: about 2.4% of loci were sexually antagonistic in the early-life assay, and 1.2% in the late-life assay.  -->
<!-- ```{r pairwise_sharing} -->
<!-- library(kableExtra) -->
<!-- get_pairwise_sharing(mashr_results_ED,  -->
<!--                      factor = 0,  -->
<!--                      lfsr_thresh = 0.1) %>%  -->
<!--   round(3) %>% kable() %>% kable_styling() -->
<!-- ``` -->
<!-- ## Inspecting the effect of adaptive shrinkage on the SNP effect size estimates -->
<!-- The plots below reveal that `mashr` did indeed shrink the effect size estimates towards zero. The amount of shrinkage applied was very similar when applying `mashr` using either the data-driven or canonical covariance matrices. -->
<!-- ```{r mashr_check_plot, fig.height=3.1, fig.width=7.3} -->
<!-- db <- DBI::dbConnect(RSQLite::SQLite(),  -->
<!--                      "data/derived/annotations.sqlite3") -->
<!-- # Results for all 1,613,615 SNPs, even those that are in 100% LD with others (these are grouped up by the SNP_clump column) -->
<!-- all_snps <- tbl(db, "univariate_lmm_results") -->
<!-- # All SNPs and SNP groups that are in <100% LD with one another (n = 1,207,357) -->
<!-- SNP_clumps <- all_snps %>% select(-SNP) %>% distinct() %>% collect(n=Inf) -->
<!-- # Subsetting variable to get the approx-LD subset of SNPs -->
<!-- LD_subset <- !is.na(SNP_clumps$LFSR_female_early_mashr_ED) -->
<!-- hex_plot <- function(x, y, xlab, ylab){ -->
<!--   filter(SNP_clumps,LD_subset) %>%  -->
<!--     ggplot(aes_string(x, y)) +  -->
<!--     geom_abline(linetype = 2) +  -->
<!--     geom_vline(xintercept = 0, linetype = 3) + -->
<!--     geom_hline(yintercept = 0, linetype = 3) + -->
<!--     stat_binhex(bins = 200, colour = "#FFFFFF00") +  -->
<!--     scale_fill_distiller(palette = "Purples") +  -->
<!--     coord_cartesian(xlim = c(-1,1), ylim = c(-0.55, 0.3)) +  -->
<!--     theme_minimal() + xlab(xlab) + ylab(ylab) + -->
<!--     theme(legend.position = "none") -->
<!-- } -->
<!-- grid.arrange( -->
<!--   hex_plot("beta_female_early_raw",  -->
<!--            "beta_female_early_mashr_canonical",  -->
<!--            "Raw estimate of SNP\neffect size from LMM",  -->
<!--            "Corrected estimate from\nmashr (canonical)"), -->
<!--   hex_plot("beta_female_early_raw",  -->
<!--            "beta_female_early_mashr_ED",  -->
<!--            "Raw estimate of SNP\neffect size from LMM",  -->
<!--            "Corrected estimate from\nmashr (data-driven)"), -->
<!--   hex_plot("beta_female_early_mashr_canonical",  -->
<!--          "beta_female_early_mashr_ED",  -->
<!--          "Corrected estimate from\nmashr (canonical)",  -->
<!--          "Corrected estimate from\nmashr (data-driven)"), -->
<!--   ncol = 3) -->
<!-- ``` -->
<!-- **Figure SX**: Plots comparing the raw effect sizes for each locus (calculated by linear mixed models implemented in GEMMA, LMM) with the effect sizes obtained using adaptive shrinkage implemented in mashr (either in 'data-driven' or 'canonical' methods). The dashed line shows $y = x$, such that the first two plots illustrate that both forms of shrinkage moved both negative and positive effects towards zero. The third plot illustrates that very similar effect sizes were obtained whether we used the data-driven or canonical method. -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3     kableExtra_1.3.4  glue_1.6.2        mashr_0.2.69     
 [5] ashr_2.2-54       lubridate_1.9.2   forcats_1.0.0     stringr_1.5.0    
 [9] dplyr_1.1.0       purrr_1.0.1       readr_2.1.4       tidyr_1.3.0      
[13] tibble_3.1.8      ggplot2_3.4.1     tidyverse_2.0.0   workflowr_1.7.0.4

loaded via a namespace (and not attached):
 [1] httr_1.4.5        sass_0.4.5        bit64_4.0.5       vroom_1.6.1      
 [5] jsonlite_1.8.4    viridisLite_0.4.1 bslib_0.4.2       rmeta_3.0        
 [9] assertthat_0.2.1  getPass_0.2-2     mixsqp_0.3-48     yaml_2.3.7       
[13] pillar_1.8.1      lattice_0.20-45   digest_0.6.31     promises_1.2.0.1 
[17] rvest_1.0.3       colorspace_2.1-0  htmltools_0.5.4   httpuv_1.6.9     
[21] Matrix_1.5-1      plyr_1.8.8        pkgconfig_2.0.3   invgamma_1.1     
[25] mvtnorm_1.1-3     scales_1.2.1      webshot_0.5.4     processx_3.8.0   
[29] svglite_2.1.1     whisker_0.4.1     later_1.3.0       tzdb_0.3.0       
[33] timechange_0.2.0  git2r_0.31.0      generics_0.1.3    ellipsis_0.3.2   
[37] cachem_1.0.7      withr_2.5.0       cli_3.6.0         crayon_1.5.2     
[41] magrittr_2.0.3    evaluate_0.20     ps_1.7.2          fs_1.6.1         
[45] fansi_1.0.4       xml2_1.3.3        truncnorm_1.0-8   tools_4.2.2      
[49] hms_1.1.2         lifecycle_1.0.3   munsell_0.5.0     irlba_2.3.5.1    
[53] callr_3.7.3       compiler_4.2.2    jquerylib_0.1.4   systemfonts_1.0.4
[57] rlang_1.0.6       grid_4.2.2        rstudioapi_0.14   rmarkdown_2.20   
[61] gtable_0.3.1      abind_1.4-5       R6_2.5.1          knitr_1.42       
[65] bit_4.0.5         fastmap_1.1.1     utf8_1.2.2        rprojroot_2.0.3  
[69] stringi_1.7.12    parallel_4.2.2    SQUAREM_2021.1    Rcpp_1.0.10      
[73] vctrs_0.5.2       tidyselect_1.2.0  xfun_0.37        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
