<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tables summarising the GWAS results</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tables summarising the GWAS results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreec386787073d6bc80f7be30ea346adfbae7746636targetblankc386787a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/c386787073d6bc80f7be30ea346adfbae7746636" target="_blank">c386787</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreec386787073d6bc80f7be30ea346adfbae7746636targetblankc386787a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/c386787073d6bc80f7be30ea346adfbae7746636" target="_blank">c386787</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    analysis/plot_models_variant_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/GO_KEGG_enrichment.Rmd
    Untracked:  analysis/TWAS_tables.Rmd
    Untracked:  analysis/plot_models_variant_effects.Rmd
    Untracked:  code/GO_and_KEGG_gsea.R
    Untracked:  code/make_sites_files_for_ARGweaver.R
    Untracked:  code/run_argweaver.R
    Untracked:  code/run_mashr.R
    Untracked:  data/argweaver/
    Untracked:  data/input/TF_binding_sites.csv
    Untracked:  figures/GWAS_mixture_proportions.rds
    Untracked:  figures/GWAS_stats_figure.pdf
    Untracked:  figures/TWAS_mixture_proportions.rds
    Untracked:  figures/TWAS_stats_figure.pdf
    Untracked:  figures/antagonism_ratios.pdf
    Untracked:  figures/composite_mixture_figure.pdf
    Untracked:  figures/eff_size_histos.pdf
    Untracked:  figures/stats_figure.pdf
    Untracked:  manuscript/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/GWAS_tables.Rmd
    Modified:   analysis/TWAS.Rmd
    Modified:   analysis/checking_mashr_results.Rmd
    Modified:   analysis/eQTL_analysis.Rmd
    Modified:   analysis/get_predicted_line_means.Rmd
    Modified:   analysis/gwas_adaptive_shrinkage.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/make_annotation_database.Rmd
    Modified:   analysis/perform_gwas.Rmd
    Modified:   analysis/plot_line_means.Rmd
    Modified:   analysis/plotting_results.Rmd
    Deleted:    code/gwas_adaptive_shrinkage.R
    Modified:   data/input/female_fitness.csv
    Modified:   data/input/female_fitness_CLEANED.csv
    Modified:   data/input/male_fitness.csv
    Modified:   data/input/male_fitness_CLEANED.csv
    Deleted:    figures/figure1.eps
    Deleted:    figures/figure1.png
    Deleted:    figures/figure2.eps

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/GWAS_tables.Rmd</code>) and HTML (<code>docs/GWAS_tables.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/a7f514637424eb2e3326986a7ebee00b287e671f/analysis/GWAS_tables.Rmd" target="_blank">a7f5146</a>
</td>
<td>
Luke Holman
</td>
<td>
2019-01-01
</td>
<td>
Lots of work on eQTLs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/GWAS_tables.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/GWAS_tables.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-and-clean-the-gwas-results" class="section level2">
<h2>Load and clean the GWAS results</h2>
<pre class="r"><code>library(tidyverse)
library(kableExtra)
library(DT)

kable_table &lt;- function(df) {
  kable(df, &quot;html&quot;) %&gt;%
  kable_styling() %&gt;%
  scroll_box(height = &quot;300px&quot;)
}

my_data_table &lt;- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;GWAS_sig_loci&#39;)),
      pageLength = 50
    )
  )
}

db &lt;- DBI::dbConnect(RSQLite::SQLite(), &quot;data/derived/annotations.sqlite3&quot;)


# Load the SNP-clumped GWAS results, created with GEMMA and clumped with PLINK
xx &lt;- bind_rows(
  read_tsv(&quot;data/derived/female_early_SNP_clumps.txt&quot;),
  read_tsv(&quot;data/derived/female_late_SNP_clumps.txt&quot;),
  read_tsv(&quot;data/derived/male_early_SNP_clumps.txt&quot;),
  read_tsv(&quot;data/derived/male_late_SNP_clumps.txt&quot;)) %&gt;%
  distinct(index_SNP, .keep_all = TRUE) %&gt;%
  mutate(FBIDs = str_remove_all(FBIDs, &quot;, NA&quot;),
         FBIDs = replace(FBIDs, is.na(FBIDs), &quot;&quot;)) %&gt;%
  mutate(genes = str_remove_all(genes, &quot;, NA&quot;),
         genes = replace(genes, is.na(genes), &quot;&quot;)) %&gt;%
  mutate(`Index variant log10 p` = round(-1 * log10(index_SNP_p_value), 2)) %&gt;%
  select(index_SNP, other_snps, FBIDs, genes, `Index variant log10 p`) 

# Because we clumped separately for the 4 traits, some of the &quot;index SNPs&quot; appear inside the ranges
# tagged by other index SNPs, so there is some redundancy. The following ugly code sorts this out 
# by putting index SNPs that appear inside the range of another index SNP into the &#39;other SNPs&#39; 
# column for that index SNP
ranges &lt;- xx$other_snps %&gt;% 
  str_extract_all(&quot;[_][:digit:]+&quot;) %&gt;% 
  map(~ as.numeric(str_remove_all(.x, &quot;_&quot;))) %&gt;% 
    map(~ c(min(.x), max(.x)))

ranges &lt;- map_df(1:nrow(xx), 
                 ~ data.frame(index = xx$index_SNP[.x], 
                              min =ranges[[.x]][1], 
                              max = ranges[[.x]][2]))%&gt;%
  filter(!is.na(min)) %&gt;% 
  mutate(chr = substr(index, 1,2))

to_modify &lt;- lapply(1:nrow(xx), function(i){
  chr &lt;- substr(xx$index_SNP[i],1,2) # get chr and pos
  pos &lt;- as.numeric(str_remove_all(str_extract_all(xx$index_SNP[i], &quot;_[:digit:]+&quot;), &quot;_&quot;))
  foc_ranges &lt;- ranges[ranges$chr == chr &amp; 
                         ranges$min &lt;= pos &amp; 
                         ranges$max &gt;= pos &amp;
                         ranges$index != xx$index_SNP[i], ]
  if(nrow(foc_ranges) &gt; 0) {
    return(data.frame(index_SNP = foc_ranges$index, 
                      to_add = xx$index_SNP[i]))
  }
  return(NULL)
}) %&gt;% bind_rows()

xx &lt;- xx %&gt;% 
  filter(!(index_SNP %in% to_modify$to_add)) %&gt;% 
  left_join(to_modify) %&gt;% 
  mutate(to_add = replace(to_add, is.na(to_add), &quot;NA&quot;))

xx &lt;- xx %&gt;% 
  mutate(other_snps = str_split(str_c(xx$other_snps, to_add, sep = &quot;, &quot;), &quot;, &quot;),
              other_snps = map_chr(other_snps, 
                                   ~ paste0(.x[.x != &quot;NA&quot;], collapse = &quot;, &quot;))) %&gt;%
  select(-to_add) %&gt;%
  mutate(other_snps = str_replace_all(other_snps, &quot;NA&quot;, &quot;&quot;))

# Now get the mashr-ED adjusted effect sizes for the index SNPs, 
# and their mixture assignment probs.
univariate_lmm_results &lt;- 
  tbl(db, &quot;univariate_lmm_results&quot;) %&gt;%
  filter(SNP %in% !! xx$index_SNP) %&gt;%
  select(-contains(&quot;canonical&quot;), 
         -contains(&quot;raw&quot;)) %&gt;% 
  inner_join(tbl(db, &quot;variants&quot;) %&gt;% 
               select(-chr, -position), 
             by = &quot;SNP&quot;) %&gt;%
  left_join(
    tbl(db, &quot;genes&quot;) %&gt;% 
      select(FBID, gene_name), by = &quot;FBID&quot;) %&gt;%
  collect(n=Inf) %&gt;%
  arrange(LFSR_female_early_mashr_ED) %&gt;%
  mutate_at(vars(contains(&quot;LFSR&quot;)), ~ -log10(.x)) %&gt;%
  mutate_if(is.numeric, ~ round(.x, 2)) %&gt;%
  rename_all(~ gsub(&quot;beta_&quot;, &quot;&quot;, .x)) %&gt;%
  rename_all(~ gsub(&quot;_mashr_ED&quot;, &quot;&quot;, .x)) %&gt;%
  arrange(-P_equal_effects) %&gt;%
  select(SNP, starts_with(&quot;female&quot;), 
         starts_with(&quot;male&quot;), starts_with(&quot;P_&quot;)) %&gt;%
  distinct()

GWAS_table &lt;- xx %&gt;%
  left_join(univariate_lmm_results, by = c(&quot;index_SNP&quot; = &quot;SNP&quot;)) %&gt;%
  arrange(index_SNP) %&gt;%
  rename(`Index variant` = index_SNP,
         `Other variants` = other_snps,
         `Genes` = genes, 
         `Female early` = female_early,
         `Female late` = female_late,
         `Male early` = male_early,
         `Male late` = male_late) %&gt;%
  rename_all(~ str_replace_all(.x, &quot;_&quot;, &quot; &quot;))</code></pre>
</div>
<div id="table-of-significant-groups-of-linked-variants" class="section level2">
<h2>Table of significant groups of linked variants</h2>
<p>This table shows loci (i.e. groups of linked variants) that passed the arbitrary statistical significance threshold of <span class="math inline">\(p &lt; 10^{-5}\)</span> for one or more of the four phenotypes (in a linear mixed model GWAS implemented in GEMMA). The loci were identified using the SNP clumping method of PLINK, and each group of variants contains one or more variants that met the significance threshold. The third and fourth columns identify the gene(s) whose exons and introns overlap the focal locus. Columns 6-9 show the estimated effect size of the index variant on the four phenotypes (corrected by <code>mashr</code> using the data-driven method), while the final columns show the mixture proportions estimated by <code>mashr</code> using the canonical method.</p>
<pre class="r"><code>GWAS_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-80fbd31f46e71aef4d0e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-80fbd31f46e71aef4d0e">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["2L_10373585_SNP","2L_11155919_SNP","2L_1139431_SNP","2L_11398129_SNP","2L_12202323_SNP","2L_12549398_SNP","2L_12648406_SNP","2L_1278985_SNP","2L_13556986_SNP","2L_14031137_SNP","2L_16187978_SNP","2L_16337997_SNP","2L_17726477_SNP","2L_20590123_SNP","2L_2505499_DEL","2L_3554753_SNP","2L_3600161_SNP","2L_3701702_SNP","2L_3958319_DEL","2L_5368255_SNP","2L_7122507_SNP","2L_7331146_SNP","2L_7331381_SNP","2L_9468807_SNP","2R_12633014_SNP","2R_13736790_SNP","2R_14993758_SNP","2R_15508894_SNP","2R_15509740_SNP","2R_15751542_SNP","2R_16248031_SNP","2R_17087079_SNP","2R_18873460_SNP","2R_3743872_SNP","2R_6047260_SNP","2R_6737444_SNP","2R_7156618_SNP","3L_10151651_SNP","3L_10301535_SNP","3L_1031366_DEL","3L_11211699_SNP","3L_11939104_SNP","3L_13517615_SNP","3L_15903275_SNP","3L_18121280_SNP","3L_18730211_DEL","3L_20120237_SNP","3L_2209187_DEL","3L_2323352_SNP","3L_2328261_SNP","3L_3137834_SNP","3L_3594782_SNP","3L_4717489_SNP","3L_5377054_SNP","3L_5974927_SNP","3L_7526058_SNP","3L_8174557_SNP","3L_8175769_SNP","3L_9968267_SNP","3R_10360251_SNP","3R_12151940_SNP","3R_18114557_SNP","3R_19041727_SNP","3R_20857921_SNP","3R_23561644_SNP","3R_23796542_SNP","3R_27005488_SNP","3R_27075585_SNP","3R_6838975_SNP","3R_6856604_SNP","3R_9824987_SNP","X_1556505_SNP","X_19894144_SNP","X_20284816_SNP","X_2319952_SNP","X_4606984_SNP","X_4607565_SNP","X_5252097_SNP","X_5435203_SNP","X_5536216_SNP","X_5536506_SNP","X_769353_SNP","X_8641795_SNP"],["","2L_11154177_SNP, 2L_11155583_SNP, 2L_11155917_SNP, 2L_11157800_SNP, 2L_11168167_SNP, 2L_11168194_SNP, 2L_11168516_SNP, 2L_11168644_DEL, 2L_11168667_SNP, 2L_11168668_SNP, 2L_11169157_SNP, 2L_11173821_INS","2L_1139426_SNP","2L_11397742_DEL, 2L_11397787_SNP, 2L_11397788_SNP, 2L_11397799_SNP, 2L_11397803_SNP","2L_12202268_SNP, 2L_12202298_SNP, 2L_12202313_SNP, 2L_12202314_SNP, 2L_12202331_SNP","2L_12549394_SNP","2L_12648991_SNP, 2L_12649033_SNP","2L_1279083_SNP","","2L_14030994_SNP, 2L_14031181_SNP","","2L_16337862_SNP, 2L_16337900_SNP, 2L_16337977_SNP, 2L_16338008_SNP, 2L_16338031_SNP, 2L_16338039_SNP","2L_17726291_SNP, 2L_17726352_SNP","","2L_2505514_SNP","","2L_3600014_SNP, 2L_3600373_SNP","2L_3701391_SNP","2L_3958224_SNP, 2L_3958275_SNP","2L_5368097_SNP, 2L_5368108_SNP, 2L_5368159_SNP, 2L_5368173_SNP, 2L_5368218_SNP, 2L_5368246_SNP, 2L_5368915_SNP","2L_7122285_SNP, 2L_7122653_SNP, 2L_7122731_SNP, 2L_7122847_SNP, 2L_7122848_SNP, 2L_7122930_SNP","2L_7331966_SNP, 2L_7331990_INS, 2L_7335159_SNP, 2L_7332711_SNP","2L_7326191_SNP, 2L_7327873_SNP","2L_9468802_SNP","2R_12632996_SNP, 2R_12633101_SNP, 2R_12633217_SNP","","2R_14993454_SNP, 2R_14993748_SNP, 2R_14993888_SNP","","2R_15509746_SNP","","2R_16248026_SNP","2R_17087007_SNP, 2R_17087041_SNP, 2R_17087142_SNP","","2R_3743460_SNP, 2R_3743503_SNP, 2R_3743708_SNP, 2R_3743755_SNP, 2R_3744011_SNP","2R_6043364_SNP, 2R_6044249_SNP, 2R_6046521_SNP, 2R_6047236_SNP, 2R_6047237_SNP, 2R_6047990_SNP","","2R_7156596_SNP","","3L_10300828_SNP, 3L_10300973_SNP, 3L_10301056_SNP, 3L_10301601_SNP","","3L_11211457_SNP, 3L_11211561_SNP","3L_11939100_SNP","","3L_15904207_SNP, 3L_15905526_SNP, 3L_15905532_SNP, 3L_15909524_SNP, 3L_15909547_SNP, 3L_15909552_SNP, 3L_15909743_SNP, 3L_15909758_SNP, 3L_15909764_SNP, 3L_15911766_SNP, 3L_15912334_SNP","3L_18121271_SNP","3L_18730160_DEL, 3L_18730191_DEL","3L_20088717_SNP, 3L_20096742_SNP, 3L_20097074_SNP, 3L_20097715_SNP, 3L_20097802_SNP, 3L_20098999_INS, 3L_20100738_SNP, 3L_20100749_SNP, 3L_20101143_INS, 3L_20101312_SNP, 3L_20102907_SNP, 3L_20102966_SNP, 3L_20117284_SNP, 3L_20119670_SNP, 3L_20119874_SNP, 3L_20122496_SNP, 3L_20123198_SNP, 3L_20123996_SNP, 3L_20132136_SNP","3L_2209152_SNP, 3L_2209212_DEL, 3L_2209237_SNP, 3L_2209241_SNP","3L_2323066_SNP, 3L_2323119_SNP, 3L_2323367_SNP, 3L_2323463_SNP, 3L_2323476_SNP, 3L_2323481_INS, 3L_2323619_SNP, 3L_2325020_SNP, 3L_2325447_SNP, 3L_2325499_SNP","3L_2325051_SNP, 3L_2325090_SNP, 3L_2325567_SNP, 3L_2325582_SNP, 3L_2326150_SNP, 3L_2326159_SNP, 3L_2326280_SNP, 3L_2327474_SNP, 3L_2327591_SNP, 3L_2327896_DEL, 3L_2327987_DEL, 3L_2328026_SNP, 3L_2328047_SNP, 3L_2328051_INS, 3L_2328061_SNP, 3L_2328072_SNP, 3L_2328091_SNP, 3L_2328203_SNP, 3L_2328302_SNP, 3L_2328425_SNP, 3L_2328431_SNP, 3L_2328472_SNP, 3L_2328480_SNP, 3L_2328527_SNP, 3L_2328638_SNP, 3L_2328678_SNP, 3L_2327475_SNP","","","","3L_5377046_SNP","","3L_7526057_SNP","","3L_8174824_SNP, 3L_8175469_SNP","3L_9961541_SNP, 3L_9964424_SNP, 3L_9965231_SNP, 3L_9965237_SNP, 3L_9965457_SNP, 3L_9968962_SNP, 3L_9968963_SNP, 3L_9968979_SNP, 3L_9969002_SNP, 3L_9969555_DEL, 3L_9970356_SNP, 3L_9970627_SNP, 3L_9971036_SNP, 3L_9971571_SNP, 3L_9971660_SNP, 3L_9974259_SNP, 3L_9974554_SNP, 3L_9977398_SNP, 3L_9977471_SNP","3R_10359904_SNP","3R_12151769_SNP, 3R_12151901_SNP, 3R_12151979_SNP","","3R_19041730_SNP, 3R_19041741_SNP, 3R_19041789_SNP, 3R_19041795_SNP, 3R_19041908_SNP","3R_20857835_SNP, 3R_20857894_SNP, 3R_20857895_SNP, 3R_20858414_SNP, 3R_20858458_SNP","3R_23561601_SNP","","3R_27006630_SNP, 3R_27011788_SNP","3R_27075537_SNP, 3R_27075746_SNP","3R_6831961_SNP, 3R_6832020_SNP, 3R_6832050_SNP, 3R_6839274_SNP, 3R_6840122_SNP, 3R_6840418_SNP, 3R_6840579_SNP, 3R_6840867_SNP","3R_6851940_SNP, 3R_6851947_SNP, 3R_6859823_SNP, 3R_6859985_SNP, 3R_6860039_SNP, 3R_6860052_SNP, 3R_6860060_SNP, 3R_6875921_SNP","","X_1557360_SNP","X_19894112_SNP, X_19894141_DEL","X_20281971_SNP, X_20284048_SNP, X_20284556_SNP, X_20284783_SNP, X_20285498_SNP, X_20285500_SNP, X_20285715_SNP, X_20286007_SNP, X_20286073_SNP, X_20286104_SNP, X_20286667_SNP, X_20287032_SNP","X_2318891_SNP, X_2318929_SNP, X_2318971_SNP, X_2319024_SNP, X_2319101_SNP, X_2319540_SNP, X_2319993_SNP, X_2320169_SNP, X_2320420_SNP, X_2320422_SNP, X_2320445_SNP, X_2320478_SNP, X_2320505_DEL, X_2320508_DEL","","","","X_5435183_SNP, X_5435294_SNP","X_5536116_SNP, X_5536276_SNP, X_5536283_SNP","X_5536448_SNP","X_768879_SNP, X_769361_SNP, X_769621_SNP, X_769623_SNP, X_770229_SNP, X_770778_SNP","X_8641777_SNP"],["FBgn0041723, FBgn0053303","FBgn0032350, FBgn0259822","FBgn0031306","","FBgn0000114","","","FBgn0041097","FBgn0259984","FBgn0028872, FBgn0266822","FBgn0001991","FBgn0028899, FBgn0259151","FBgn0015609","FBgn0266214","","","","FBgn0259957, FBgn0259956, FBgn0262621, FBgn0259955, FBgn0265370","FBgn0051774","","FBgn0085407","FBgn0031902","","","FBgn0034151, FBgn0259967","FBgn0041585, FBgn0015522","FBgn0004168","FBgn0003435","FBgn0003435","FBgn0043532","","FBgn0034602, FBgn0265180, FBgn0265191","FBgn0003900, FBgn0034807, FBgn0261706, FBgn0052833, FBgn0263551","FBgn0085390","FBgn0033494, FBgn0265512","FBgn0050015","FBgn0033603","FBgn0052057","","FBgn0024277","FBgn0261553","FBgn0262401","FBgn0264001","FBgn0036544","FBgn0036778","","FBgn0036935, FBgn0265890, FBgn0036936, FBgn0036937, FBgn0036939, FBgn0005198, FBgn0036940, FBgn0036942","","FBgn0035331","FBgn0035331","FBgn0035400","FBgn0005640, FBgn0053504","FBgn0035574","","","FBgn0028582","FBgn0035848, FBgn0035847","FBgn0035848, FBgn0035849","FBgn0026404, FBgn0036062, FBgn0036063, FBgn0259932, FBgn0263251, FBgn0040823","FBgn0038237","FBgn0038415, FBgn0040271, FBgn0038414","FBgn0051163","FBgn0262975","FBgn0039260","FBgn0005659","FBgn0003890","FBgn0051004","FBgn0051005","FBgn0083950","FBgn0083950","FBgn0264493","FBgn0000210, FBgn0025390","FBgn0031082","FBgn0031107, FBgn0031108, FBgn0040651, FBgn0031109, FBgn0031110","FBgn0052797, FBgn0001150","FBgn0029723","FBgn0029723","FBgn0029761","FBgn0263512","FBgn0259994","FBgn0259994","FBgn0264449, FBgn0040341",""],["uncharacterized protein, rhomboid-5","uncharacterized protein, Ca2+-channel-protein-beta-subunit","uncharacterized protein","","bruno 1","","","roundabout 3","kuzbanian","uncharacterized protein, long non-coding RNA:CR45284","Ca[2+]-channel protein alpha[[1]] subunit D","uncharacterized protein, uncharacterized protein","Cadherin-N","long non-coding RNA:CR44909","","","","uncharacterized protein, uncharacterized protein, Seminal fluid protein 24C1, uncharacterized protein, long non-coding RNA:CR44311","friend of echinoid","","PDGF- and VEGF-related factor 3","Wnt oncogene analog 6","","","uncharacterized protein, Seminal fluid protein 53D","olf186-M, olf186-F","5-hydroxytryptamine (serotonin) receptor 1A","smooth","smooth","Odorant-binding protein 56i","","Leucine-rich repeat activity-regulated protein at synapses, Glycogenin, uncharacterized protein","uncharacterized protein, uncharacterized protein, twist, antisense RNA:CR42742, mir-4939 stem loop","Diacyl glycerol kinase","KCNQ potassium channel, mulet","uncharacterized protein","Cuticular protein 47Ef","defective proboscis extension response 10","","trio","uncharacterized protein","mir-285 stem loop","bruno 3","sugar-free frosting","Cyp312a1","","uncharacterized protein, uncharacterized protein, Ionotropic receptor 76b, uncharacterized protein, gigas, obstructor-J, uncharacterized protein, long non-coding RNA:CR44679","","Myosuppressin receptor 1","Myosuppressin receptor 1","uncharacterized protein","Ecdysone-induced protein 63E, small nuclear RNA U7","Rho guanine nucleotide exchange factor at 64C","","","liquid facets","methuselah-like 7, UDP-xylose synthase","UDP-xylose synthase, estrogen-related receptor","uncharacterized protein, Death regulator Nedd2-like caspase, uncharacterized protein, variable nurse cells, defective proboscis extension response 6, uncharacterized protein","Phosphodiesterase 6","uncharacterized protein, uncharacterized protein, Sulfated","Shal K[+] channel interacting protein","cap-n-collar","Smg6","Ets at 98B","beta-Tubulin at 97EF","mesh","qless","sidestep VI","sidestep VI","roadkill","Mucin related 2B","uncharacterized protein","HECT and RLD domain containing protein 2, uncharacterized protein, Odorant-binding protein 19a, Odorant-binding protein 19b, uncharacterized protein","giant, uncharacterized protein","Proctolin receptor","Proctolin receptor","small conductance calcium-activated potassium channel","Visual system homeobox 2","Otopetrin-like a","Otopetrin-like a","uncharacterized protein, uncharacterized protein",""],[6.44,5.04,5.14,5.62,5.91,5.77,5.12,7.04,5.32,5.11,5.13,5.18,6.04,5.17,6.46,5.06,5.15,5.19,5.4,5.07,5.83,6.7,7.4,5.05,5.1,5.47,7.26,6.03,5.84,5.04,6.78,6.13,5.47,5.02,5.02,5.06,5.49,5.48,5.09,5.58,5.49,5.89,5.01,5.07,5.2,5.75,5.25,6.85,5.15,6.27,5.17,5.08,5.93,5.02,5.32,5.3,5.09,5.18,5.08,5.17,5.92,5.55,5.46,5.33,5.23,5.18,6.11,5.08,5.71,5.16,5.74,5.05,5.28,5.76,6.33,5.13,5.24,5.54,5.4,5.46,5.39,6.58,5.3],[0.2,0.16,0.05,-0.24,-0.23,-0.21,-0.04,-0.41,-0.09,0.22,-0.14,-0.13,-0.07,-0.11,-0.25,-0.11,-0.2,-0.36,-0.11,0.28,-0.09,-0.18,-0.3,0.18,-0.14,-0.13,-0.22,-0.1,-0.08,-0.05,-0.18,-0.16,-0.31,-0.23,0.15,-0.18,-0.21,-0.18,-0.17,-0.04,0.05,-0.24,-0.17,-0.14,-0.19,-0.08,-0.13,-0.25,-0.23,-0.36,-0.05,-0.1,-0.06,-0.19,0.06,-0.17,0,-0,-0.04,-0.02,0.04,0.21,0.08,-0.09,-0.3,-0.32,null,-0.22,0.18,-0.19,null,0.03,-0.36,-0.29,-0.3,-0.16,-0.33,-0.23,-0.18,-0.29,-0.17,-0.23,-0.14],[0.27,0.22,0.06,-0.33,-0.33,-0.29,-0.06,-0.55,-0.13,0.32,-0.19,-0.17,-0.1,-0.15,-0.35,-0.15,-0.28,-0.5,-0.16,0.39,-0.13,-0.25,-0.42,0.26,-0.19,-0.19,-0.31,-0.14,-0.1,-0.07,-0.25,-0.22,-0.42,-0.31,0.21,-0.25,-0.29,-0.25,-0.23,-0.05,0.06,-0.34,-0.23,-0.19,-0.27,-0.1,-0.18,-0.36,-0.32,-0.49,-0.08,-0.14,-0.09,-0.26,0.08,-0.24,0,-0,-0.06,-0.02,0.05,0.29,0.12,-0.13,-0.42,-0.44,null,-0.3,0.25,-0.26,null,0.04,-0.49,-0.4,-0.41,-0.22,-0.47,-0.33,-0.26,-0.4,-0.24,-0.32,-0.19],[0.39,0.29,0.17,-0.03,-0.03,-0.12,-0.14,-0.16,-0.03,0.19,-0.03,-0.09,-0.17,-0.02,-0.08,-0.23,-0.02,-0.18,-0.27,0.15,-0.24,-0.05,0.02,-0.02,-0.05,-0.23,-0.05,-0.27,-0.23,-0.2,-0.04,-0.03,-0.15,-0.08,0.1,-0.09,-0.17,-0.35,-0.06,-0.15,0.25,0.02,-0.09,-0.05,-0.04,-0.04,-0.21,-0.04,-0.06,-0.16,-0.19,-0.15,-0.18,-0.12,0.21,-0.07,0.18,0.19,0.17,-0.13,0.16,0.11,0.19,-0.2,-0.23,-0.13,null,-0.14,0.06,-0.02,null,0.21,-0.13,-0.14,-0.12,-0.08,-0.24,-0.09,-0.01,-0.05,0.01,-0.07,-0.04],[0.23,0.17,0.09,-0.03,-0.04,-0.08,-0.09,-0.12,-0.02,0.14,-0.03,-0.06,-0.1,-0.02,-0.07,-0.14,-0.03,-0.13,-0.16,0.1,-0.15,-0.04,-0.01,0,-0.04,-0.15,-0.04,-0.18,-0.13,-0.12,-0.04,-0.03,-0.11,-0.06,0.06,-0.07,-0.11,-0.21,-0.05,-0.09,0.14,0,-0.06,-0.03,-0.03,-0.03,-0.14,-0.04,-0.05,-0.11,-0.12,-0.09,-0.12,-0.08,0.11,-0.05,0.1,0.11,0.09,-0.07,0.09,0.08,0.11,-0.13,-0.15,-0.11,null,-0.09,0.05,-0.02,null,0.12,-0.1,-0.09,-0.09,-0.06,-0.17,-0.07,-0.02,-0.04,-0.01,-0.06,-0.03],[0.99,0.99,0.69,0.02,0.04,0.77,0.62,0.04,0.56,1,0.12,0.91,0.88,0.13,0.18,0.98,0.01,0.5,0.94,0.64,0.9,0.1,0,0.01,0.2,0.96,0.03,0.84,0.76,0.6,0.07,0.12,0.63,0.19,0.88,0.74,1,0.99,0.36,0.57,0.53,0.01,0.67,0.28,0.03,0.69,0.98,0.03,0.06,0.14,0.68,0.93,0.7,0.91,0.76,0.43,0.24,0.24,0.13,0.4,0.52,0.82,0.94,0.88,0.99,0.54,null,0.88,0.09,0.01,null,0.29,0.05,0.39,0.27,0.74,0.98,0.36,0.01,0,0.01,0.17,0.26],[0,0,0.01,0.96,0.95,0.22,0,0.96,0.34,0,0.81,0.07,0,0.77,0.81,0,0.96,0.5,0,0.35,0,0.87,1,0.9,0.75,0,0.96,0,0,0,0.91,0.83,0.36,0.79,0.11,0.24,0,0,0.61,0.01,0,0.98,0.31,0.66,0.94,0.22,0,0.96,0.91,0.86,0.01,0.01,0,0.08,0,0.55,0.01,0.02,0.03,0.01,0,0.17,0,0,0.01,0.46,null,0.12,0.89,0.96,null,0,0.95,0.6,0.72,0.23,0.02,0.61,0.95,1,0.9,0.81,0.68],[0.01,0.01,0.2,0,0,0,0.3,0,0.01,0,0,0,0.09,0,0,0.02,0,0,0.05,0,0.09,0,0,0,0,0.02,0,0.15,0.2,0.35,0,0,0,0,0,0,0,0.01,0,0.32,0.38,0,0,0,0,0.02,0.01,0,0,0,0.25,0.04,0.25,0,0.19,0,0.53,0.49,0.44,0.37,0.36,0,0.04,0.1,0,0,null,0,0,0,null,0.59,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,null,0,0,0,null,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0.09,0.02,0.02,0.01,0.08,0,0.09,0,0.07,0.01,0.03,0.1,0.01,0.01,0.02,0,0.01,0,0.01,0.02,0,0.09,0.04,0.01,0.01,0.01,0.04,0.05,0.02,0.05,0,0.02,0.01,0.02,0,0,0.03,0.11,0.09,0.02,0.02,0.06,0.03,0.07,0,0.01,0.02,0,0.06,0.02,0.05,0.01,0.05,0.02,0.22,0.25,0.41,0.22,0.12,0.01,0.02,0.02,0,0,null,0,0.02,0.03,null,0.12,0,0.01,0.01,0.03,0,0.03,0.04,0,0.09,0.01,0.06]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Index variant<\/th>\n      <th>Other variants<\/th>\n      <th>FBIDs<\/th>\n      <th>Genes<\/th>\n      <th>Index variant log10 p<\/th>\n      <th>Female early<\/th>\n      <th>Female late<\/th>\n      <th>Male early<\/th>\n      <th>Male late<\/th>\n      <th>P equal effects<\/th>\n      <th>P female specific<\/th>\n      <th>P male specific<\/th>\n      <th>P null<\/th>\n      <th>P sex antag<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"GWAS_sig_loci"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8,9,10,11,12,13]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<!-- # Results of one multivariate linear mixed model run in GEMMA: -->
<!-- # multivariate_lmm_results <-  -->
<!-- #   read_tsv("data/derived/output/all_four_traits.assoc.txt") %>% -->
<!-- #   filter(p_fdr < 0.05) %>%  -->
<!-- #   inner_join(tbl(db, "variants") %>%  -->
<!-- #                select(-chr, -position) %>% -->
<!-- #                collect(), by = "SNP") %>%  -->
<!-- #   arrange(p_wald) %>%  -->
<!-- #   select(-contains("Vbeta")) %>% -->
<!-- #   left_join( -->
<!-- #     tbl(db, "genes") %>%  -->
<!-- #       select(FBID, gene_name) %>%  -->
<!-- #       collect(n=Inf), by = "FBID") %>% -->
<!-- #   mutate(gene_name = replace(gene_name, is.na(FBID), NA))  -->
<!-- #  -->
<!-- # multivariate_lmm_results <- multivariate_lmm_results[,c(1:8, 14, 9:13)] -->
<!-- #  -->
<!-- # multivariate_lmm_results <- multivariate_lmm_results %>% -->
<!-- #   mutate_at(vars(starts_with("p_")), ~ -log10(.x)) %>% -->
<!-- #   mutate_if(is.numeric, ~ format(round(.x, 2), nsmall = 2)) %>% -->
<!-- #   left_join(tbl(db, "univariate_lmm_results") %>%  -->
<!-- #               select(SNP, SNP_clump) %>%  -->
<!-- #               collect(), -->
<!-- #             by = "SNP") %>% -->
<!-- #   distinct(SNP_clump, .keep_all = TRUE) %>% -->
<!-- #   select(SNP, SNP_clump, everything()) -->
<!-- ## Results of multivariate association tests -->
<!-- ```{r message=FALSE} -->
<!-- multivariate_lmm_results  %>% -->
<!--   add_colour() %>% -->
<!--   kable(format = "html", escape = FALSE) %>% -->
<!--   kable_styling("striped") %>% -->
<!--   scroll_box(height = "300px") -->
<!-- ``` -->
<!-- ### All four traits -->
<!-- ```{r message=FALSE} -->
<!-- read_tsv("data/derived/all_four_traits_SNP_clumps.txt") %>% -->
<!--   kable_table() -->
<!-- ``` -->
<!-- ```{r eval=F} -->
<!-- # http://bergmanlab.genetics.uga.edu/flyreg/v2.0/Target.html -->
<!-- TFBS <- read_csv("data/input/TF_binding_sites.csv") -->
<!-- ranges <- xx$other_snps %>%  -->
<!--   str_extract_all("[_][:digit:]+") %>%  -->
<!--   map(~ as.numeric(str_remove_all(.x, "_"))) %>%  -->
<!--   map(~ c(min(.x), max(.x))) -->
<!-- ranges <- map_df(1:nrow(xx), ~ data.frame(index = xx$index_SNP[.x], min =ranges[[.x]][1], max = ranges[[.x]][2]))%>% -->
<!--   filter(!is.na(min)) %>% mutate(chr = substr(index, 1,2)) %>% -->
<!--   filter(!is.infinite(min)) -->
<!-- ranges_index <- xx$index_SNP %>%  -->
<!--   str_extract_all("[_][:digit:]+") %>%  -->
<!--   map(~ as.numeric(str_remove_all(.x, "_"))) %>%  -->
<!--   map(~ c(min(.x), max(.x))) -->
<!-- ranges_index <- map_df(1:nrow(xx), ~ data.frame(index = xx$index_SNP[.x],  -->
<!--                                                 min =ranges_index[[.x]][1],  -->
<!--                                                 max = ranges_index[[.x]][2])) %>% -->
<!--   filter(!is.na(min)) %>% mutate(chr = substr(index, 1,2)) %>% -->
<!--   filter(!(index %in% ranges$index)) -->
<!-- sig_ranges <- bind_rows(ranges, ranges_index) %>%  -->
<!--   arrange(index) %>% -->
<!--   mutate(chr = replace(chr, chr == "X_", "X")) -->
<!-- for(i in 1:nrow(sig_ranges)){ -->
<!--   hits <- TFBS %>% -->
<!--     filter(chr == sig_ranges$chr[i]) %>% -->
<!--     filter(start > sig_ranges$min[i]  & start < sig_ranges$max[i] | -->
<!--              stop > sig_ranges$min[i] & stop < sig_ranges$max[i]) -->
<!--   if(nrow(hits) > 0) print(hits) -->
<!-- } -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] DT_0.13          kableExtra_1.1.0 forcats_0.5.0    stringr_1.4.0   
 [5] dplyr_1.0.0      purrr_0.3.4      readr_1.3.1      tidyr_1.1.0     
 [9] tibble_3.0.1     ggplot2_3.3.2    tidyverse_1.3.0 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      lubridate_1.7.8   lattice_0.20-41   assertthat_0.2.1 
 [5] rprojroot_1.3-2   digest_0.6.25     R6_2.4.1          cellranger_1.1.0 
 [9] backports_1.1.7   reprex_0.3.0      RSQLite_2.2.0     evaluate_0.14    
[13] httr_1.4.1        pillar_1.4.4      rlang_0.4.6       readxl_1.3.1     
[17] rstudioapi_0.11   whisker_0.4       blob_1.2.1        rmarkdown_2.5    
[21] webshot_0.5.2     htmlwidgets_1.5.1 bit_1.1-15.2      munsell_0.5.0    
[25] broom_0.5.6       compiler_4.0.3    httpuv_1.5.3.1    modelr_0.1.8     
[29] xfun_0.19         pkgconfig_2.0.3   htmltools_0.5.0   tidyselect_1.1.0 
[33] workflowr_1.6.2   fansi_0.4.1       viridisLite_0.3.0 crayon_1.3.4     
[37] dbplyr_1.4.4      withr_2.2.0       later_1.0.0       grid_4.0.3       
[41] nlme_3.1-149      jsonlite_1.7.0    gtable_0.3.0      lifecycle_0.2.0  
[45] DBI_1.1.0         git2r_0.27.1      magrittr_2.0.1    scales_1.1.1     
[49] cli_2.0.2         stringi_1.5.3     fs_1.4.1          promises_1.1.0   
[53] xml2_1.3.2        ellipsis_0.3.1    generics_0.0.2    vctrs_0.3.0      
[57] tools_4.0.3       bit64_0.9-7       glue_1.4.2        crosstalk_1.1.0.1
[61] hms_0.5.3         yaml_2.2.1        colorspace_1.4-1  rvest_0.3.5      
[65] memoise_1.1.0     knitr_1.30        haven_2.3.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
