<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tables summarising the GWAS results</title>

<script src="site_libs/jquery-3.5.0/jquery-3.5.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tables summarising the GWAS results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreef04d7d31d8cf33da6cb948d0b26804b433d35fabtargetblankf04d7d3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/f04d7d31d8cf33da6cb948d0b26804b433d35fab" target="_blank">f04d7d3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreef04d7d31d8cf33da6cb948d0b26804b433d35fabtargetblankf04d7d3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/f04d7d31d8cf33da6cb948d0b26804b433d35fab" target="_blank">f04d7d3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    figures/fig1_inkscape.svg
    Ignored:    figures/figure1a.pdf
    Ignored:    figures/figure1b.pdf

Untracked files:
    Untracked:  big_model.rds
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds
    Untracked:  old_analyses/

Unstaged changes:
    Modified:   .gitignore
    Modified:   figures/GWAS_stats_figure.pdf
    Modified:   figures/SNP_effect_ED.pdf
    Modified:   figures/TWAS_stats_figure.pdf
    Modified:   figures/antagonism_ratios.pdf
    Modified:   figures/boyle_plot.pdf
    Modified:   figures/composite_mixture_figure.pdf
    Modified:   figures/eff_size_histos.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/GWAS_tables.Rmd</code>) and HTML (<code>docs/GWAS_tables.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/GWAS_tables.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/01226ab272891d92a8444e19cac034c1b079d263/analysis/GWAS_tables.Rmd" target="_blank">01226ab</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/GWAS_tables.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/GWAS_tables.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3855d3359649c74f955042725d07276742cb649a/analysis/GWAS_tables.Rmd" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3855d3359649c74f955042725d07276742cb649a/docs/GWAS_tables.html" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/a7f514637424eb2e3326986a7ebee00b287e671f/analysis/GWAS_tables.Rmd" target="_blank">a7f5146</a>
</td>
<td>
Luke Holman
</td>
<td>
2019-01-01
</td>
<td>
Lots of work on eQTLs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/GWAS_tables.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/GWAS_tables.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(kableExtra)
library(DT)

sci_notation &lt;- function(x) formatC(x, format = &quot;e&quot;, digits = 1)

kable_table &lt;- function(df) {
  kable(df, &quot;html&quot;) %&gt;%
  kable_styling() %&gt;%
  scroll_box(height = &quot;300px&quot;)
}

my_data_table &lt;- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;GWAS_sig_loci&#39;)),
      pageLength = 50
    )
  ) %&gt;%
    formatStyle(
        columns = c(&quot;Female early effect&quot;, &quot;Male early effect&quot;, &quot;Female late effect&quot;, &quot;Male late effect&quot;), 
        color = styleInterval(cuts = 0, values = c(&quot;tomato&quot;, &quot;steelblue&quot;)),
        fontWeight = &quot;bold&quot;)
}

db &lt;- DBI::dbConnect(RSQLite::SQLite(), &quot;data/derived/annotations.sqlite3&quot;)

p_cutoff &lt;- 1e-05

univariate_lmm_results &lt;- 
  tbl(db, &quot;univariate_lmm_results&quot;) %&gt;%
  select(SNP, SNP_clump, contains(&quot;raw&quot;), -contains(&quot;SE&quot;)) %&gt;% 
  filter(pvalue_female_early_raw &lt; p_cutoff | pvalue_female_late_raw &lt; p_cutoff | 
           pvalue_male_early_raw &lt; p_cutoff | pvalue_male_late_raw &lt; p_cutoff) %&gt;% 
  inner_join(tbl(db, &quot;variants&quot;), by = &quot;SNP&quot;) %&gt;%
  left_join(
    tbl(db, &quot;genes&quot;) %&gt;% 
      select(FBID, gene_name), by = &quot;FBID&quot;) %&gt;%
  collect(n=Inf) %&gt;%
  mutate_at(vars(contains(&quot;beta&quot;)), ~ format(round(.x, 2), nsmall = 2)) %&gt;% 
  rename_all(~ gsub(&quot;beta_&quot;, &quot;&quot;, .x)) %&gt;%
  rename_all(~ gsub(&quot;_raw&quot;, &quot;&quot;, .x)) %&gt;%
  select(SNP, SNP_clump, MAF, FBID, gene_name, site.class, starts_with(&quot;female&quot;), 
          starts_with(&quot;male&quot;), starts_with(&quot;P_&quot;), starts_with(&quot;pvalue&quot;)) %&gt;%
  distinct()

GWAS_table &lt;- univariate_lmm_results %&gt;% 
  split(.$SNP_clump) %&gt;% # Only keep one SNP from each clump
  map_df(~ mutate(.x, sum_p = pvalue_female_early + pvalue_female_late + pvalue_male_early + pvalue_male_late) %&gt;% 
        arrange(sum_p) %&gt;% head(1)) %&gt;% 
  select(-sum_p) %&gt;% 
  mutate_at(vars(contains(&quot;pval&quot;)), sci_notation) %&gt;% 
  mutate(SNP_clump = replace(SNP_clump, SNP == SNP_clump, &quot;&quot;)) %&gt;% 
  rename(`Variant` = SNP,
         `Variant clump` = SNP_clump,
         `Site class` = site.class,
         `Gene` = gene_name, 
         `Female early effect` = female_early,
         `Female late effect` = female_late,
         `Male early effect` = male_early,
         `Male late effect` = male_late,
         `Female early pval` = pvalue_female_early,
         `Female late pval` = pvalue_female_late,
         `Male early pval` = pvalue_male_early,
         `Male late pval` = pvalue_male_late) 

write_csv(GWAS_table, &quot;data/derived/GWAS_significant_snps.csv&quot;)</code></pre>
<div id="table-showing-significant-loci-from-the-gwas" class="section level2">
<h2>Table showing significant loci from the GWAS</h2>
<p><em>Table SX</em>: This table shows loci (either individual SNPs and indels, or variant clumps in complete linkage disequilibrium) that passed the statistical significance threshold of <span class="math inline">\(p &lt; 10^{-5}\)</span> for at least one of the four phenotypes, in a linear mixed model GWAS implemented in GEMMA. Column 3 gives the minor allele frequency, columns 4-5 identify genes that overlap the focal locus, and column 6 shows the site class. Columns 7-10 show the estimated effect size of the index variant on the four phenotypes (where positive values mean that the minor allele is associated with higher fitness, and negative values that the minor allele is associated with lower fitness), and the final columns show the corresponding <span class="math inline">\(p\)</span>-values.</p>
<pre class="r"><code>GWAS_table %&gt;% my_data_table()</code></pre>
<div id="htmlwidget-80fbd31f46e71aef4d0e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-80fbd31f46e71aef4d0e">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["2L_10373585_SNP","2L_10666944_SNP","2L_11155919_SNP","2L_1139426_SNP","2L_11398129_SNP","2L_11900442_SNP","2L_12362698_SNP","2L_12600320_SNP","2L_1278985_SNP","2L_1279083_SNP","2L_13736785_SNP","2L_14031137_SNP","2L_15469662_SNP","2L_15469663_SNP","2L_16337977_SNP","2L_1992654_SNP","2L_20590123_SNP","2L_2505499_DEL","2L_3585735_SNP","2L_3958319_DEL","2L_7327873_SNP","2L_7331146_SNP","2L_7331381_SNP","2L_7331966_SNP","2L_7332711_SNP","2L_9468802_SNP","2R_11451981_SNP","2R_11587427_SNP","2R_13736790_SNP","2R_14993454_SNP","2R_14993748_SNP","2R_14993888_SNP","2R_15508894_SNP","2R_15509740_SNP","2R_16248031_SNP","2R_16716337_SNP","2R_18582247_SNP","3L_10151651_SNP","3L_10301535_SNP","3L_10619878_SNP","3L_11211699_SNP","3L_11350860_INS","3L_11939100_SNP","3L_13517615_SNP","3L_1396396_SNP","3L_18121280_SNP","3L_20120237_SNP","3L_2209187_DEL","3L_2327474_SNP","3L_2327987_DEL","3L_2328091_SNP","3L_2579211_SNP","3L_3852986_SNP","3L_4337825_SNP","3L_4717489_SNP","3L_5974927_SNP","3L_7709650_SNP","3R_15069383_SNP","3R_20857835_SNP","3R_23465476_SNP","3R_23561601_SNP","3R_23561644_SNP","3R_26029886_SNP","3R_6838975_SNP","3R_6840122_SNP","3R_7772970_SNP","X_13282830_SNP","X_1556505_SNP","X_19894141_DEL","X_19894144_SNP","X_20284816_SNP","X_2319952_SNP","X_5435183_SNP","X_5435203_SNP","X_5536216_SNP","X_5536276_SNP","X_769353_SNP","X_8641777_SNP","X_889209_SNP"],["","","","2L_1139426_SNP, 2L_1139431_SNP","","","","","","","","2L_14031137_SNP, 2L_14031181_SNP","2L_15469662_SNP, 2L_15469667_SNP, 2L_15469687_SNP","","2L_16337977_SNP, 2L_16338008_SNP","","","2L_2505499_DEL, 2L_2505514_SNP","","","","","","2L_7331966_SNP, 2L_7331990_INS, 2L_7336684_SNP","","2L_9468802_SNP, 2L_9468807_SNP","","","","","2R_14993748_SNP, 2R_14993758_SNP","","","","","","","","","","","","3L_11939100_SNP, 3L_11939104_SNP","","","","","","3L_2327474_SNP, 3L_2327591_SNP, 3L_2327896_DEL, 3L_2328261_SNP, 3L_2328302_SNP","3L_2327987_DEL, 3L_2328026_SNP","","","3L_3852986_SNP, 3L_3853115_SNP","","","","","","3R_20857835_SNP, 3R_20857894_SNP, 3R_20857895_SNP, 3R_20857921_SNP","","","","","","","","","","","","","X_2319952_SNP, X_2319993_SNP","","","","","X_769353_SNP, X_769361_SNP","X_8641777_SNP, X_8641795_SNP",""],[0.161,0.2049,0.1512,0.1171,0.1512,0.3512,0.4683,0.361,0.1073,0.1073,0.3951,0.3756,0.1854,0.2,0.1659,0.1073,0.1463,0.2098,0.1122,0.161,0.0878,0.05854,0.1268,0.06829,0.122,0.2341,0.1171,0.2537,0.08293,0.122,0.1122,0.122,0.4341,0.361,0.122,0.2537,0.1415,0.1073,0.1463,0.2,0.1512,0.4341,0.07805,0.1268,0.1902,0.2829,0.1512,0.2098,0.08293,0.1366,0.09268,0.2341,0.08293,0.3512,0.1024,0.3122,0.1268,0.0878,0.1073,0.09756,0.2585,0.1951,0.07317,0.4634,0.4976,0.08293,0.122,0.3756,0.0878,0.0878,0.1122,0.1512,0.4634,0.439,0.1854,0.2488,0.2585,0.05366,0.06341],["FBgn0041723","FBgn0051871","FBgn0032350","FBgn0031306",null,"FBgn0264815","FBgn0262475","FBgn0085424","FBgn0041097","FBgn0041097","FBgn0051813","FBgn0028872","FBgn0266893","FBgn0266893","FBgn0259151","FBgn0043539","FBgn0266214",null,null,"FBgn0051774",null,null,null,null,null,null,"FBgn0034032","FBgn0050080","FBgn0041585","FBgn0004168","FBgn0004168","FBgn0004168","FBgn0003435","FBgn0003435",null,"FBgn0016984","FBgn0005631","FBgn0052057",null,"FBgn0085267","FBgn0261553",null,"FBgn0262401","FBgn0264001","FBgn0003138","FBgn0036778","FBgn0036939",null,"FBgn0035331","FBgn0035331","FBgn0035331","FBgn0010909","FBgn0013751","FBgn0035533","FBgn0035574",null,"FBgn0261788","FBgn0038693","FBgn0039260","FBgn0266579","FBgn0005659","FBgn0005659",null,"FBgn0083950","FBgn0083950","FBgn0037963","FBgn0030481","FBgn0000210","FBgn0031082","FBgn0031082","FBgn0040651","FBgn0052797","FBgn0263512","FBgn0263512","FBgn0259994","FBgn0259994","FBgn0264449",null,null],["rhomboid-5","uncharacterized protein","uncharacterized protein","uncharacterized protein",null,"Phosphodiesterase 1c","bruno 2","nubbin","roundabout 3","roundabout 3","uncharacterized protein","uncharacterized protein","long non-coding RNA:CR45353","long non-coding RNA:CR45353","uncharacterized protein","Odorant-binding protein 22a","long non-coding RNA:CR44909",null,null,"friend of echinoid",null,null,null,null,null,null,"uncharacterized protein","uncharacterized protein","olf186-F","5-hydroxytryptamine (serotonin) receptor 1A","5-hydroxytryptamine (serotonin) receptor 1A","5-hydroxytryptamine (serotonin) receptor 1A","smooth","smooth",null,"skittles","roundabout 1","defective proboscis extension response 10",null,"uncharacterized protein","uncharacterized protein",null,"mir-285 stem loop","bruno 3",null,"Cyp312a1","uncharacterized protein",null,"Myosuppressin receptor 1","Myosuppressin receptor 1","Myosuppressin receptor 1","misshapen","Arrowhead","Cdc42-interacting protein 4","Rho guanine nucleotide exchange factor at 64C",null,"Ankyrin 2","uncoordinated 79","Smg6","tau","Ets at 98B","Ets at 98B",null,"sidestep VI","sidestep VI","Cadherin 87A","uncharacterized protein",null,"uncharacterized protein","uncharacterized protein","uncharacterized protein","uncharacterized protein","Visual system homeobox 2","Visual system homeobox 2","Otopetrin-like a","Otopetrin-like a","uncharacterized protein",null,null],["INTRON","INTRON","UPSTREAM","INTRON","INTERGENIC","INTRON","INTRON","INTRON","INTRON","INTRON","UTR_3_PRIME","NON_SYNONYMOUS_CODING","UPSTREAM","UPSTREAM","DOWNSTREAM","UPSTREAM","INTRON","INTERGENIC","INTERGENIC","INTRON","INTERGENIC","INTERGENIC","INTERGENIC","INTERGENIC","INTERGENIC","INTERGENIC","INTRON","UTR_3_PRIME","INTRON","INTRON","INTRON","INTRON","INTRON","INTRON","INTERGENIC","SYNONYMOUS_CODING","SYNONYMOUS_CODING","INTRON","INTERGENIC","SYNONYMOUS_CODING","INTRON","INTERGENIC","DOWNSTREAM","UTR_3_PRIME","INTRON","DOWNSTREAM","INTRON","INTERGENIC","INTRON","INTRON","INTRON","INTRON","INTRON","INTRON","INTRON","INTERGENIC","INTRON","SYNONYMOUS_CODING","SYNONYMOUS_CODING","DOWNSTREAM","UPSTREAM","UPSTREAM","INTERGENIC","INTRON","INTRON","INTRON","UTR_3_PRIME","UTR_3_PRIME","DOWNSTREAM","DOWNSTREAM","UTR_3_PRIME","UPSTREAM","INTRON","INTRON","INTRON","INTRON","INTRON","INTERGENIC","INTERGENIC"],[" 0.21"," 0.02"," 0.32"," 0.04","-0.40","-0.39"," 0.11"," 0.13","-0.68","-0.64","-0.04"," 0.22","-0.54","-0.53","-0.54","-0.70","-0.53","-0.29","-0.57","-0.18","-0.88","-0.92","-0.81","-0.87","-0.65"," 0.27","-0.34","-0.27","-0.39","-0.74","-0.91","-0.82","-0.10","-0.03","-0.75","-0.02","-0.63","-0.36","-0.58"," 0.10"," 0.02","-0.09","-0.79","-0.52","-0.04","-0.29","-0.25","-0.34","-0.69","-0.56","-0.68"," 0.14","-0.78"," 0.00","-0.11"," 0.12","-0.09","-0.47","-0.16","-0.79","-0.20","-0.32","-0.81"," 0.27","-0.20","-0.66","-0.60"," 0.03","-0.70","-0.78","-0.66","-0.34","-0.22","-0.25","-0.47","-0.38","-0.21","-0.81","-0.75"],[" 0.33","-0.04"," 0.22"," 0.04","-0.54","-0.19"," 0.08"," 0.19","-0.59","-0.58","-0.14"," 0.42","-0.12","-0.17","-0.27","-0.15","-0.12","-0.51","-0.22","-0.17","-0.43","-0.36","-0.46","-0.30","-0.32"," 0.46","-0.62","-0.44","-0.24","-0.24","-0.34","-0.25","-0.14","-0.14","-0.27"," 0.01","-0.23","-0.20","-0.33","-0.17"," 0.03","-0.08","-0.54","-0.27","-0.12","-0.45","-0.21","-0.56","-0.54","-0.33","-0.48","-0.09","-0.12","-0.10","-0.04"," 0.03"," 0.04","-0.70","-0.14","-0.27","-0.46","-0.55","-0.32"," 0.39","-0.39","-0.27","-0.43"," 0.01","-0.58","-0.63","-0.49","-0.68","-0.39","-0.41","-0.49","-0.43","-0.52","-0.36","-0.32"],[" 0.62","-0.49"," 0.59"," 0.65"," 0.07","-0.10"," 0.41"," 0.19","-0.15","-0.12","-0.40"," 0.22"," 0.08"," 0.08","-0.33","-0.05"," 0.02","-0.02"," 0.08","-0.57"," 0.23"," 0.12"," 0.28"," 0.25"," 0.05","-0.16","-0.24","-0.17","-0.60"," 0.05"," 0.15"," 0.19","-0.33","-0.42"," 0.11"," 0.09"," 0.11","-0.61"," 0.04","-0.26"," 0.62"," 0.13"," 0.18","-0.19","-0.53","-0.03","-0.33"," 0.04","-0.23","-0.09","-0.17","-0.22","-0.09"," 0.15","-0.40"," 0.43"," 0.61"," 0.13","-0.43"," 0.36","-0.14","-0.30"," 0.02"," 0.08","-0.08"," 0.05"," 0.13"," 0.39","-0.07","-0.07","-0.27","-0.14"," 0.06"," 0.07","-0.03"," 0.00","-0.06","-0.04","-0.01"],[" 0.36","-0.31"," 0.28"," 0.21"," 0.01","-0.12"," 0.26"," 0.39","-0.20","-0.14","-0.14"," 0.28"," 0.08"," 0.13","-0.11","-0.04"," 0.08","-0.15"," 0.03","-0.34"," 0.05","-0.08"," 0.00"," 0.03","-0.12","-0.06","-0.22","-0.06","-0.84","-0.04","-0.11","-0.08","-0.41","-0.19","-0.08"," 0.50"," 0.11","-0.42","-0.19","-0.50"," 0.27"," 0.39"," 0.33","-0.06","-0.23"," 0.07","-0.54","-0.03","-0.10","-0.14","-0.11","-0.47"," 0.15"," 0.43","-0.75"," 0.17","-0.02"," 0.09","-0.74"," 0.16","-0.09","-0.25"," 0.08"," 0.00"," 0.10","-0.02"," 0.12"," 0.22","-0.16","-0.16","-0.06","-0.09","-0.04","-0.01"," 0.06"," 0.04","-0.07"," 0.08","-0.14"],["1.2e-01","8.5e-01","2.4e-02","7.7e-01","1.2e-03","8.4e-06","2.2e-01","1.6e-01","5.0e-07","1.3e-06","6.7e-01","1.8e-02","5.9e-06","5.5e-06","5.6e-06","7.4e-06","1.9e-06","3.8e-03","8.8e-06","1.8e-01","4.2e-07","1.4e-07","2.8e-08","3.4e-06","1.1e-06","9.8e-03","1.8e-02","6.7e-03","4.5e-02","6.9e-06","1.9e-08","1.5e-07","2.9e-01","7.8e-01","6.7e-08","8.6e-01","4.6e-06","1.0e-02","9.3e-06","4.0e-01","8.9e-01","3.2e-01","1.9e-06","9.2e-06","7.6e-01","3.7e-03","6.2e-02","3.7e-03","1.1e-06","7.5e-06","1.6e-06","2.3e-01","8.7e-06","1.0e+00","4.9e-01","2.3e-01","5.1e-01","2.6e-03","3.0e-01","3.0e-06","5.2e-02","5.5e-03","4.0e-06","2.5e-03","2.4e-02","7.0e-06","5.2e-06","7.8e-01","9.2e-06","2.6e-06","5.9e-06","1.4e-02","1.3e-02","5.2e-03","1.1e-05","9.8e-05","3.6e-02","4.2e-06","5.3e-06"],["1.3e-02","7.5e-01","1.2e-01","7.9e-01","6.1e-06","3.9e-02","3.7e-01","5.0e-02","1.8e-05","1.5e-05","1.3e-01","2.1e-06","3.4e-01","1.7e-01","2.5e-02","3.6e-01","2.9e-01","2.0e-07","1.0e-01","2.0e-01","2.0e-02","5.2e-02","2.9e-03","1.3e-01","2.1e-02","9.1e-06","8.0e-06","5.4e-06","2.2e-01","1.6e-01","4.4e-02","1.2e-01","1.4e-01","1.3e-01","6.3e-02","9.4e-01","1.0e-01","1.5e-01","1.3e-02","1.4e-01","8.1e-01","4.0e-01","1.5e-03","2.6e-02","3.4e-01","5.3e-06","1.2e-01","5.1e-07","2.4e-04","1.0e-02","1.1e-03","4.3e-01","5.2e-01","3.1e-01","8.0e-01","7.6e-01","8.0e-01","4.7e-06","3.7e-01","1.3e-01","4.6e-06","2.8e-07","8.5e-02","5.1e-06","8.6e-06","7.6e-02","1.4e-03","9.2e-01","2.9e-04","2.1e-04","1.2e-03","2.4e-07","6.0e-06","2.4e-06","4.6e-06","7.1e-06","3.5e-08","4.9e-02","6.1e-02"],["4.5e-07","5.2e-06","5.1e-06","4.3e-06","5.8e-01","3.0e-01","3.7e-06","4.2e-02","3.0e-01","3.7e-01","5.2e-06","1.6e-02","5.4e-01","5.3e-01","6.4e-03","7.6e-01","8.7e-01","8.3e-01","5.4e-01","7.4e-06","2.0e-01","5.1e-01","6.2e-02","1.9e-01","7.4e-01","1.5e-01","1.2e-01","9.5e-02","1.8e-03","7.9e-01","3.7e-01","2.4e-01","1.7e-04","1.3e-06","4.5e-01","3.9e-01","4.5e-01","6.7e-06","7.6e-01","2.7e-02","1.0e-06","1.5e-01","2.9e-01","1.2e-01","6.6e-06","8.0e-01","9.3e-03","7.6e-01","1.1e-01","4.8e-01","2.3e-01","5.3e-02","6.4e-01","1.2e-01","1.0e-02","3.6e-06","6.1e-06","4.2e-01","6.7e-03","3.0e-02","1.7e-01","7.9e-03","9.1e-01","3.9e-01","3.7e-01","7.4e-01","3.4e-01","7.0e-06","6.8e-01","6.8e-01","7.5e-02","2.9e-01","4.7e-01","4.6e-01","8.0e-01","1.0e+00","5.6e-01","8.4e-01","9.3e-01"],["3.9e-03","6.0e-03","3.4e-02","1.5e-01","9.2e-01","2.0e-01","3.1e-03","7.7e-06","1.5e-01","2.9e-01","1.2e-01","1.9e-03","5.0e-01","2.8e-01","3.6e-01","7.7e-01","4.7e-01","1.5e-01","8.4e-01","8.7e-03","7.7e-01","6.4e-01","9.8e-01","8.8e-01","3.9e-01","6.0e-01","1.6e-01","5.5e-01","6.6e-06","8.4e-01","5.4e-01","6.4e-01","2.0e-06","4.0e-02","5.7e-01","5.0e-07","4.6e-01","2.0e-03","1.5e-01","9.7e-06","4.2e-02","8.5e-06","5.5e-02","6.1e-01","5.2e-02","4.9e-01","7.5e-06","8.2e-01","4.9e-01","2.5e-01","4.3e-01","9.5e-06","4.1e-01","2.2e-06","3.6e-07","6.8e-02","8.9e-01","6.0e-01","1.0e-06","3.4e-01","3.7e-01","2.8e-02","6.7e-01","9.7e-01","2.6e-01","8.9e-01","3.7e-01","1.5e-02","3.5e-01","3.5e-01","6.8e-01","5.1e-01","6.2e-01","9.1e-01","6.2e-01","6.7e-01","4.7e-01","6.5e-01","4.1e-01"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Variant<\/th>\n      <th>Variant clump<\/th>\n      <th>MAF<\/th>\n      <th>FBID<\/th>\n      <th>Gene<\/th>\n      <th>Site class<\/th>\n      <th>Female early effect<\/th>\n      <th>Female late effect<\/th>\n      <th>Male early effect<\/th>\n      <th>Male late effect<\/th>\n      <th>Female early pval<\/th>\n      <th>Female late pval<\/th>\n      <th>Male early pval<\/th>\n      <th>Male late pval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"GWAS_sig_loci"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-weight':'bold','color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"tomato\" : \"steelblue\"});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-weight':'bold','color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"tomato\" : \"steelblue\"});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-weight':'bold','color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"tomato\" : \"steelblue\"});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-weight':'bold','color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"tomato\" : \"steelblue\"});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<div id="table-showing-the-numbers-of-loci-related-to-fitness" class="section level3">
<h3>Table showing the numbers of loci related to fitness</h3>
<p>The table tallies up the numbers of clumps of perfectly linked loci showing a particular significant relationship with fitness (rows) for various different <span class="math inline">\(p\)</span>-value thresholds (columns). For example, ‘Female early only’ counts the number of loci whose expression significantly correlated with mean female early life fitness across lines. ‘Age concordant, males’ counts loci whose expression correlated with early- <em>and</em> late-life fitness in males, in the same direction, while ‘Age antagonistic, males’ counts loci showing a significant, opposite relationships with male early- and male late-life fitness. Similarly, ‘Sex concordant, early’ counts loci showing a concordant effect on early life fitness in both males and females, and ‘Sex antagonistic, late’ counts those showing opposing effects on late-life fitness in males and females. The categories are all mutually exclusive, such that loci are only counted towards the most specific category that applies to them. Note that this method has low power to detect loci that correlate with two or more fitness metrics, because there are two or more opportunities to make a ‘false negative’ error, and the power is low for any given locus (this is why we use a range of p-values, including some low ones).</p>
<pre class="r"><code>make_tally &lt;- function(p_cutoff){
  GWAS_results &lt;- tbl(db, &quot;univariate_lmm_results&quot;) %&gt;%
    select(SNP_clump, contains(&quot;raw&quot;), -contains(&quot;SE&quot;)) %&gt;% 
    mutate(sum_p = pvalue_female_early_raw + pvalue_female_late_raw + pvalue_male_early_raw + pvalue_male_late_raw) %&gt;% 
    arrange(sum_p) %&gt;% select(-sum_p) %&gt;% 
    rename(beta_FE = beta_female_early_raw, beta_FL = beta_female_late_raw,
           beta_ME = beta_male_early_raw, beta_ML = beta_male_late_raw,
           pval_FE = pvalue_female_early_raw, pval_FL = pvalue_female_late_raw,
           pval_ME = pvalue_male_early_raw, pval_ML = pvalue_male_late_raw) %&gt;% 
    distinct() %&gt;% # Analyse clumps of SNPs in &lt;100% LD with one another
    collect(n = Inf) %&gt;% 
    mutate(
      type = 
        factor(case_when(
          
          (sign(beta_FE) != sign(beta_ME) &amp; pval_FE &lt; p_cutoff &amp; pval_ME &lt; p_cutoff) &amp; 
            (sign(beta_FL) != sign(beta_ML) &amp; pval_FL &lt; p_cutoff &amp; pval_ML &lt; p_cutoff) ~ &quot;Sex antagonistic, both ages&quot;,
          
          (sign(beta_FE) == sign(beta_ME) &amp; pval_FE &lt; p_cutoff &amp; pval_ME &lt; p_cutoff) &amp; 
            (sign(beta_FL) == sign(beta_ML) &amp; pval_FL &lt; p_cutoff &amp; pval_ML &lt; p_cutoff) ~ &quot;Sex concordant, both ages&quot;,
          
          (sign(beta_FE) != sign(beta_FL) &amp; pval_FE &lt; p_cutoff &amp; pval_FL &lt; p_cutoff) &amp; 
            (sign(beta_ME) != sign(beta_ML) &amp; pval_ME &lt; p_cutoff &amp; pval_ML &lt; p_cutoff) ~ &quot;Age antagonistic, both sexes&quot;,
          
          (sign(beta_FE) == sign(beta_FL) &amp; pval_FE &lt; p_cutoff &amp; pval_FL &lt; p_cutoff) &amp;
            (sign(beta_ME) == sign(beta_ML) &amp; pval_ME &lt; p_cutoff &amp; pval_ML &lt; p_cutoff) ~ &quot;Age concordant, both sexes&quot;,
          
          sign(beta_FE) != sign(beta_ME) &amp; pval_FE &lt; p_cutoff &amp; pval_ME &lt; p_cutoff ~ &quot;Sex antagonistic, early&quot;,
          sign(beta_FL) != sign(beta_ML) &amp; pval_FL &lt; p_cutoff &amp; pval_ML &lt; p_cutoff ~ &quot;Sex antagonistic, late&quot;,
          
          sign(beta_FE) == sign(beta_ME) &amp; pval_FE &lt; p_cutoff &amp; pval_ME &lt; p_cutoff ~ &quot;Sex concordant, early&quot;,
          sign(beta_FL) == sign(beta_ML) &amp; pval_FL &lt; p_cutoff &amp; pval_ML &lt; p_cutoff ~ &quot;Sex concordant, late&quot;,
          
          sign(beta_FE) != sign(beta_FL) &amp; pval_FE &lt; p_cutoff &amp; pval_FL &lt; p_cutoff ~ &quot;Age antagonistic, females&quot;,
          sign(beta_ME) != sign(beta_ML) &amp; pval_ME &lt; p_cutoff &amp; pval_ML &lt; p_cutoff ~ &quot;Age antagonistic, males&quot;,
          
          sign(beta_FE) == sign(beta_FL) &amp; pval_FE &lt; p_cutoff &amp; pval_FL &lt; p_cutoff ~ &quot;Age concordant, females&quot;,
          sign(beta_ME) == sign(beta_ML) &amp; pval_ME &lt; p_cutoff &amp; pval_ML &lt; p_cutoff ~ &quot;Age concordant, males&quot;,
          
          pval_FE &lt; p_cutoff ~ &quot;Female early only&quot;,
          pval_FL &lt; p_cutoff ~ &quot;Female late only&quot;,
          pval_ME &lt; p_cutoff ~ &quot;Male early only&quot;,
          pval_ML &lt; p_cutoff ~ &quot;Male late only&quot;,
          TRUE ~ &quot;Uncorrelated with fitness&quot;
        ), levels = c(&quot;Sex antagonistic, both ages&quot;, &quot;Sex concordant, both ages&quot;, 
                      &quot;Age antagonistic, both sexes&quot;, &quot;Age concordant, both sexes&quot;,
                      &quot;Sex antagonistic, early&quot;, &quot;Sex concordant, early&quot;, &quot;Sex antagonistic, late&quot;, &quot;Sex concordant, late&quot;,
                      &quot;Age antagonistic, females&quot;, &quot;Age antagonistic, males&quot;, &quot;Age concordant, females&quot;, &quot;Age concordant, males&quot;,
                      &quot;Female early only&quot;, &quot;Male early only&quot;, &quot;Female late only&quot;, &quot;Male late only&quot;, &quot;Uncorrelated with fitness&quot;)))
  
  tally &lt;- table(GWAS_results$type) %&gt;% sort(decreasing = T) %&gt;% enframe()
  names(tally)[2] &lt;- paste(&quot;p &lt;&quot;, p_cutoff)
  tally
}

tally_table &lt;- make_tally(0.01) %&gt;% 
  left_join(make_tally(0.001), by = &quot;name&quot;) %&gt;% 
  left_join(make_tally(0.0001), by = &quot;name&quot;) %&gt;% 
  left_join(make_tally(0.00001), by = &quot;name&quot;) %&gt;% 
  left_join(make_tally(0.000001), by = &quot;name&quot;) %&gt;% 
  left_join(make_tally(0.0000001), by = &quot;name&quot;) %&gt;% 
  rename(`Relationship to fitness` = name)

tally_table %&gt;% kable_table()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Relationship to fitness
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p &lt; 0.01
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p &lt; 0.001
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p &lt; 1e-04
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p &lt; 1e-05
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p &lt; 1e-06
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p &lt; 1e-07
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Uncorrelated with fitness
</td>
<td style="text-align:right;">
1154817
</td>
<td style="text-align:right;">
1201180
</td>
<td style="text-align:right;">
1206595
</td>
<td style="text-align:right;">
1207278
</td>
<td style="text-align:right;">
1207342
</td>
<td style="text-align:right;">
1207353
</td>
</tr>
<tr>
<td style="text-align:left;">
Male early only
</td>
<td style="text-align:right;">
12827
</td>
<td style="text-align:right;">
1457
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Female early only
</td>
<td style="text-align:right;">
12686
</td>
<td style="text-align:right;">
1753
</td>
<td style="text-align:right;">
251
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late only
</td>
<td style="text-align:right;">
12315
</td>
<td style="text-align:right;">
1418
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late only
</td>
<td style="text-align:right;">
11512
</td>
<td style="text-align:right;">
1421
</td>
<td style="text-align:right;">
185
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
Age concordant, females
</td>
<td style="text-align:right;">
1743
</td>
<td style="text-align:right;">
95
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age concordant, males
</td>
<td style="text-align:right;">
875
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex concordant, early
</td>
<td style="text-align:right;">
318
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex concordant, late
</td>
<td style="text-align:right;">
229
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex antagonistic, early
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex antagonistic, late
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex concordant, both ages
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex antagonistic, both ages
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age antagonistic, both sexes
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age concordant, both sexes
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age antagonistic, females
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age antagonistic, males
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] DT_0.13          kableExtra_1.3.4 forcats_0.5.0    stringr_1.4.0   
 [5] dplyr_1.0.0      purrr_0.3.4      readr_2.0.0      tidyr_1.1.0     
 [9] tibble_3.0.1     ggplot2_3.3.2    tidyverse_1.3.0  workflowr_1.6.2 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      svglite_1.2.3     lubridate_1.7.10  lattice_0.20-41  
 [5] assertthat_0.2.1  rprojroot_1.3-2   digest_0.6.25     R6_2.4.1         
 [9] cellranger_1.1.0  backports_1.1.7   reprex_0.3.0      RSQLite_2.2.0    
[13] evaluate_0.14     highr_0.8         httr_1.4.1        pillar_1.4.4     
[17] gdtools_0.2.2     rlang_0.4.6       readxl_1.3.1      rstudioapi_0.11  
[21] whisker_0.4       jquerylib_0.1.1   blob_1.2.1        rmarkdown_2.11   
[25] webshot_0.5.2     htmlwidgets_1.5.1 bit_1.1-15.2      munsell_0.5.0    
[29] broom_0.5.6       compiler_4.0.3    httpuv_1.5.3.1    modelr_0.1.8     
[33] xfun_0.26         pkgconfig_2.0.3   systemfonts_0.2.2 htmltools_0.5.0  
[37] tidyselect_1.1.0  fansi_0.4.1       viridisLite_0.3.0 crayon_1.3.4     
[41] tzdb_0.1.2        dbplyr_1.4.4      withr_2.2.0       later_1.0.0      
[45] grid_4.0.3        nlme_3.1-149      jsonlite_1.7.0    gtable_0.3.0     
[49] lifecycle_0.2.0   DBI_1.1.0         git2r_0.27.1      magrittr_2.0.1   
[53] scales_1.1.1      vroom_1.5.3       cli_2.0.2         stringi_1.5.3    
[57] fs_1.4.1          promises_1.1.0    xml2_1.3.2        ellipsis_0.3.1   
[61] generics_0.0.2    vctrs_0.3.0       tools_4.0.3       bit64_0.9-7      
[65] glue_1.4.2        crosstalk_1.1.0.1 hms_0.5.3         parallel_4.0.3   
[69] yaml_2.2.1        colorspace_1.4-1  rvest_0.3.5       memoise_1.1.0    
[73] knitr_1.32        haven_2.3.1      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
