<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>The genetic basis of fitness in Drosophila; A genome-wide association study</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">The genetic basis of fitness in
<em>Drosophila</em>; A genome-wide association study</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree67235d7163d7f79332d9725b4e8d864bf2056aeftargetblank67235d7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/fitnessGWAS/tree/67235d7163d7f79332d9725b4e8d864bf2056aef" target="_blank">67235d7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree67235d7163d7f79332d9725b4e8d864bf2056aeftargetblank67235d7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/fitnessGWAS/tree/67235d7163d7f79332d9725b4e8d864bf2056aef" target="_blank">67235d7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    old_analyses/.DS_Store

Untracked files:
    Untracked:  code/~$in_paper_figures.docx
    Untracked:  old_analyses/Data for old analyses/
    Untracked:  old_analyses/eQTL_analysis.Rmd
    Untracked:  old_analyses/fitness_data.csv
    Untracked:  old_analyses/gcta_quant_genetics_OLD.Rmd
    Untracked:  old_analyses/quantitative_genetics_OLD_brms.Rmd

Unstaged changes:
    Modified:   code/pdf_supp_material.Rmd
    Modified:   code/pdf_supp_material.pdf
    Modified:   figures/fig2_SNPs_manhattan_plot.png
    Modified:   figures/fig3_boyle_plot.pdf
    Modified:   figures/fig4_mutation_load.pdf
    Modified:   figures/fig5_quartiles_plot.pdf
    Modified:   figures/fig6_antagonism_ratios.pdf
    Modified:   figures/fig7_models.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/67235d7163d7f79332d9725b4e8d864bf2056aef/analysis/index.Rmd" target="_blank">67235d7</a>
</td>
<td>
lukeholman
</td>
<td>
2023-04-16
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/b41137de3488bc91036af6849a190b77584fd248/docs/index.html" target="_blank">b41137d</a>
</td>
<td>
lukeholman
</td>
<td>
2023-04-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/41d5725c2eeb662a0225375a896824757322d6b3/analysis/index.Rmd" target="_blank">41d5725</a>
</td>
<td>
lukeholman
</td>
<td>
2023-04-16
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/71237e864af5ae11914f301bbe78067d64203a20/docs/index.html" target="_blank">71237e8</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/be63a3f316b86e523a0e48df113e6844dc62d1da/analysis/index.Rmd" target="_blank">be63a3f</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/79db4a4015c1239f4c88b9d4791f80b6348e37ed/docs/index.html" target="_blank">79db4a4</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/24aedf38df994c859b25663f40763f5ab97150cd/analysis/index.Rmd" target="_blank">24aedf3</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3c8a5db1b17038cd9408f8e758d0a6ab7c726b99/docs/index.html" target="_blank">3c8a5db</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8027f80941b8d986182edf6e0d01aeba6da0322d/analysis/index.Rmd" target="_blank">8027f80</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3174ec1287379447c7dda2c0173d51070da4a81f/docs/index.html" target="_blank">3174ec1</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/d94e22dc2fbcacfe3430866ac5f7116866a93ec3/analysis/index.Rmd" target="_blank">d94e22d</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-14
</td>
<td>
New quant gen and other changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/ea38a0ea1612f4c31eb0645bf3006ac90c900791/docs/index.html" target="_blank">ea38a0e</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/5795d892c30c87d78cc5190e704adba92ec17425/analysis/index.Rmd" target="_blank">5795d89</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/f80d48ee2eeef8e7119108e97d9515be30071ab4/docs/index.html" target="_blank">f80d48e</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/b483da1c29a2640a1c297cdaee3a27a63ed59fc3/analysis/index.Rmd" target="_blank">b483da1</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/434c1d70ac81830bb36c83807088295f894a656a/docs/index.html" target="_blank">434c1d7</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/13fc6288283ab3b0ba31c9a656167d7fd1ac8f9b/analysis/index.Rmd" target="_blank">13fc628</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/1d527114eb68cf48e3f0915e1323a01ae2823bfc/docs/index.html" target="_blank">1d52711</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8b7b14d24937d59714c14a69a412464a6f01be62/analysis/index.Rmd" target="_blank">8b7b14d</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/c10e53e344cff0eac98f9cb0796c72c333138591/docs/index.html" target="_blank">c10e53e</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/990336e0baf3283078cc939b762a0321aad84546/analysis/index.Rmd" target="_blank">990336e</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/f04d7d31d8cf33da6cb948d0b26804b433d35fab/docs/index.html" target="_blank">f04d7d3</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/index.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/01226ab272891d92a8444e19cac034c1b079d263/analysis/index.Rmd" target="_blank">01226ab</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/649495630a2841c831131d6331c1a1ced7cfeeac/docs/index.html" target="_blank">6494956</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3cbeccba287d5cfeb9c0653d2df1aebca3aaa364/analysis/index.Rmd" target="_blank">3cbeccb</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/a7065f6797c85b97a31d722e358f71b329664051/docs/index.html" target="_blank">a7065f6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/44c520dbee148852f5dbc68babf78135ecf6c9b4/analysis/index.Rmd" target="_blank">44c520d</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/index.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/index.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/a50524c5ff72847ab816a8055b4be7aed4eff6db/docs/index.html" target="_blank">a50524c</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3f847f8473476aa7e4d1ddd96e945d842bd9a814/analysis/index.Rmd" target="_blank">3f847f8</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/f5c3861f5121897ed479152763cf9dda9a2bbf18/docs/index.html" target="_blank">f5c3861</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/937e1ee1f46f3f64d5a047cd582e65bff476da1c/analysis/index.Rmd" target="_blank">937e1ee</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/871ae81fa56175675b53d22ca29bf9323c3ed50a/docs/index.html" target="_blank">871ae81</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/index.html" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/836a780685e8ac20355b0a92672276f7ca763481/docs/index.html" target="_blank">836a780</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/359ff37c4f4021451e81122393307a28013165a9/docs/index.html" target="_blank">359ff37</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/5506c4b50ea385dbe5174320ad41d797c64d0676/docs/index.html" target="_blank">5506c4b</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/390a393c1d62df397c0d74de89fcb6e4b2b6ec20/analysis/index.Rmd" target="_blank">390a393</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/index.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/index.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This website relates to the forthcoming paper ‘Pleiotropic fitness
effects across sexes and age classes in the <em>Drosophila</em> genome
and transcriptome’.</p>
<!-- by Wong and Holman (for contact info, see [lukeholman.org](https://www.lukeholman.org/)). -->
<p>Click the headings below to see code, results, plots, tables and
figures from this study.</p>
<div
id="setting-up-a-database-to-hold-variantgene-annotations-and-gwas-results"
class="section level3">
<h3><a href="make_annotation_database.html">1. Setting up a database to
hold variant/gene annotations and GWAS results</a></h3>
<p>This script creates a SQLite3 database holding two tables: one with
annotations for each SNP/indel variant (annotations created by the
Mackay lab), and one with annotations for each gene (from annotation
hub). In step 4 below, we also add the GWAS results to this database,
allowing memory-efficient handling of the results.</p>
</div>
<div id="estimating-line-mean-fitness-using-bayesian-models"
class="section level3">
<h3><a href="get_predicted_line_means.html">2. Estimating line mean
fitness using Bayesian models</a></h3>
<p>This script uses Bayesian mixed models implemented in the package
<code>brms</code> to estimate the line means for our four fitness
traits, while imputing missing values and adjusting for block
effects.</p>
</div>
<div id="tables-of-the-raw-data-and-estimated-line-means"
class="section level3">
<h3><a href="raw_data_and_line_means.html">3. Tables of the raw data and
estimated line means</a></h3>
<p>To facilitate data re-use, we here provide tables showing the raw
data (i.e. the measurements of male and female fitness that were
collected on each individual replicate vial), as well as the estimated
line means that were calculated in <a
href="get_predicted_line_means.html">2. Estimating line mean fitness
using Bayesian models</a>.</p>
</div>
<div id="calculating-quantitative-genetic-parameters"
class="section level3">
<h3><a href="quant_genetics.html">4. Calculating quantitative genetic
parameters</a></h3>
<p>We first present a table showing the proportion of variance in
fitness explained by ‘DGRP line’, which approximates heritability. We
then estimate the correlations among line means in the 4 fitness traits,
which approximates genetic correlations.</p>
</div>
<div id="running-the-gwas" class="section level3">
<h3><a href="perfom_gwas.html">5. Running the GWAS</a></h3>
<p>This script first performs quality control and imputation on the
dataset of SNPs and indels for the DGRP (e.g filtering by MAF). Second,
it runs mixed model association tests on our four phenotypes using the
software <a
href="http://www.xzlab.org/software/GEMMAmanual.pdf">GEMMA</a>. Third,
it uses PLINK to identify groups of loci that are in complete linkage
disequilibrium (‘SNP clumps’), and to identify a subset of SNPs that are
in approximate LD for downstream analyses.</p>
</div>
<div id="applying-adaptive-shrinkage-to-the-gwas-results"
class="section level3">
<h3><a href="gwas_adaptive_shrinkage.html">6. Applying adaptive
shrinkage to the GWAS results</a></h3>
<p>This script uses the R package <code>mashr</code> to perform
multivariate adaptive shrinkage on the results of the GWAS, for an
LD-pruned subset of loci. This produces corrected estimates of each
SNP’s effect size, and allows estimation of the frequencies of different
types of loci (e.g. sexually- or age-antagonistic loci).</p>
</div>
<div id="running-the-twas-and-applying-mashr-to-the-twas-results"
class="section level3">
<h3><a href="TWAS.html">7. Running the TWAS and applying mashr to the
TWAS results</a></h3>
<p>This script uses the transcriptomic data on the DGRP from Huang et
al. 2015 <em>PNAS</em> to run a ‘transcriptome-wide association study’
(TWAS). In this script, we</p>
<ul>
<li>Calculate the average expression and sex bias in expression for each
transcript in Huang et al.</li>
<li>Perform ‘TWAS’, i.e. searching for transcripts whose expression
correlates with our fitness measures across DGRP lines</li>
<li>Adjust the results of the TWAS linear models using
<code>mashr</code>.</li>
</ul>
</div>
<div id="plots-tables-and-analyses-of-the-results"
class="section level2">
<h2>Plots, tables, and analyses of the results</h2>
<div id="plots-showing-line-mean-fitness" class="section level3">
<h3><a href="plot_line_means.html">8. Plots showing line mean
fitness</a></h3>
<p>This script plots the estimated line means for each of the four
fitness metrics, i.e. Figure 1 in the paper.</p>
</div>
<div id="tables-of-gwas-results" class="section level3">
<h3><a href="GWAS_tables.html">9. Tables of GWAS results</a></h3>
<p>This script presents a searchable HTML table showing a list of
significant SNPs and indels from the GWAS, with annotations, effect
sizes, and <span class="math inline">\(p\)</span>-values for each.</p>
</div>
<div id="tables-of-twas-results" class="section level3">
<h3><a href="TWAS_tables.html">10. Tables of TWAS results</a></h3>
<p>This script presents a searchable HTML table showing a list of
significant transcripts from the TWAS, with annotations, effect sizes,
and <span class="math inline">\(p\)</span>-values for each.</p>
</div>
<div id="plots-and-statistical-analyses" class="section level3">
<h3><a href="plot_models_variant_effects.html">11. Plots and statistical
analyses</a></h3>
<p>Here, we present various plots and statistical analyses of the GWAS
and TWAS results, specifically:</p>
<!-- - Q-Q plots of the GWAS results -->
<ul>
<li>Hex bin plots showing the correlations in effect sizes from GWAS
across the 4 fitness traits</li>
<li>A statistical test showing that minor alleles tend to be associated
with lower fitness</li>
<li>A plot inspired by Boyle et al. 2017 (“An expanded view of complex
traits: from polygenic to omnigenic”, <em>Cell</em>) illustrating that
fitness is highly polygenic (‘omnigenic’) and pleiotropy between male
and female fitness is common.</li>
<li>An analysis of mutation load, testing whether the number of
putatively harmful alleles present in each DGRP line correlates with
fitness.</li>
<li>Bar plots showing an estimate of the frequencies of sexually
antagonistic and sexually concordant SNPs/indels from GWAS, and
transcripts from TWAS (plus the same for age concordant vs age
antagonistic variants/transcripts)</li>
<li>Statistics indicating that the frequency of sexually antagonistic
loci declines with age</li>
<li>Plots of the evidence ratio for each locus and transcript, where the
ER compares evidence for the hypotheses that 1) the locus is concordant
between sexes or ages, and B) the locus is antagonistic between sexes or
ages. This plot avoids having to create a binary distinction between
antagonistic and concordant loci, when in reality there is a continuum
of evidence.</li>
<li>Statistical models of the evidence ratios, showing <em>inter
alia</em> that candidate sexually antagonistic alleles tend to be more
common and to be enriched on the X chromosome.</li>
<li>GO enrichment of the top 1% most sexually antagonistic transcripts,
implicating GO terms relating to mitochondria among other things.</li>
</ul>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
