<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Searching for loci underlying sex- and age-specific fitness</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Searching for loci underlying sex- and age-specific fitness</h1>

</div>


<p><strong>Last updated:</strong> 2018-10-04</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> 450cb72 </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/derived/.DS_Store
    Ignored:    data/derived/output/.DS_Store
    Ignored:    data/input/.DS_Store
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  .pversion
    Untracked:  ExDeconDemo_c.log
    Untracked:  ExDeconDemo_loglike.log
    Untracked:  TEST_Venn.tiff
    Untracked:  analysis/checking_mashr_results.Rmd
    Untracked:  code/Drosophila_GWAS.Rmd
    Untracked:  data/derived/.!25440!dgrp2_focal_lines.bed
    Untracked:  data/derived/.pversion
    Untracked:  data/derived/GRM.txt
    Untracked:  data/derived/TWAS_result_females.csv
    Untracked:  data/derived/TWAS_result_males.csv
    Untracked:  data/derived/all_four_traits_SNP_clumps.txt
    Untracked:  data/derived/all_univariate_GEMMA_results.csv
    Untracked:  data/derived/annotations.sqlite3
    Untracked:  data/derived/dgrp2_QC_all_lines.bed
    Untracked:  data/derived/dgrp2_QC_all_lines.bim
    Untracked:  data/derived/dgrp2_QC_all_lines.fam
    Untracked:  data/derived/dgrp2_QC_all_lines.log
    Untracked:  data/derived/dgrp2_QC_focal_lines.bed
    Untracked:  data/derived/dgrp2_QC_focal_lines.bim
    Untracked:  data/derived/dgrp2_QC_focal_lines.bk
    Untracked:  data/derived/dgrp2_QC_focal_lines.fam
    Untracked:  data/derived/dgrp2_QC_focal_lines.log
    Untracked:  data/derived/dgrp2_QC_focal_lines.rds
    Untracked:  data/derived/dgrp2_transcriptome_subset.bed
    Untracked:  data/derived/dgrp2_transcriptome_subset.bim
    Untracked:  data/derived/dgrp2_transcriptome_subset.fam
    Untracked:  data/derived/dgrp2_transcriptome_subset.log
    Untracked:  data/derived/female_early_SNP_clumps.txt
    Untracked:  data/derived/female_late_SNP_clumps.txt
    Untracked:  data/derived/lines_to_keep.txt
    Untracked:  data/derived/lmm_ED_mashr_results.rds
    Untracked:  data/derived/lmm_canonical_mashr_results.rds
    Untracked:  data/derived/male_early_SNP_clumps.txt
    Untracked:  data/derived/male_late_SNP_clumps.txt
    Untracked:  data/derived/mashr_results_ED.rds
    Untracked:  data/derived/mashr_results_canonical.rds
    Untracked:  data/derived/output/.pversion
    Untracked:  data/derived/output/GRM.cXX.txt
    Untracked:  data/derived/output/GRM.log.txt
    Untracked:  data/derived/output/GRM_transcriptome.cXX.txt
    Untracked:  data/derived/output/GRM_transcriptome.log.txt
    Untracked:  data/derived/output/all_four_traits.assoc.txt
    Untracked:  data/derived/output/all_four_traits.log.txt
    Untracked:  data/derived/output/allele_freq_count.assoc.txt
    Untracked:  data/derived/output/allele_freq_count.log.txt
    Untracked:  data/derived/output/eigen_decomp.eigenD.txt
    Untracked:  data/derived/output/eigen_decomp.eigenU.txt
    Untracked:  data/derived/output/eigen_decomp.log.txt
    Untracked:  data/derived/output/eigen_decomp_transcriptome.eigenD.txt
    Untracked:  data/derived/output/eigen_decomp_transcriptome.eigenU.txt
    Untracked:  data/derived/output/eigen_decomp_transcriptome.log.txt
    Untracked:  data/derived/plink.frq
    Untracked:  data/input/.pversion
    Untracked:  data/input/dgrp2.fam copy
    Untracked:  data/input/dgrp2.fam.bak
    Untracked:  data/input/huang_2015_tableS11_eQTL_females.csv
    Untracked:  data/input/huang_2015_tableS12_eQTL_males.csv
    Untracked:  data/input/huang_2015_tableS2_gene_expression.csv
    Untracked:  data/input/huang_2015_tableS5_transcript_heritability.csv
    Untracked:  dgrp2_clean.log
    Untracked:  docs/figure/checking_mashr_results.Rmd/
    Untracked:  gwas_adaptive_shrinkage.R

Unstaged changes:
    Modified:   analysis/GWAS_tables.Rmd
    Modified:   analysis/TWAS.Rmd
    Modified:   analysis/gwas_adaptive_shrinkage.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/make_annotation_database.Rmd
    Modified:   analysis/perform_gwas.Rmd
    Modified:   analysis/plotting_results.Rmd
    Deleted:    data/derived/DGRP_SNP_genos.sqlite3
    Deleted:    data/derived/gwas_db.sqlite3
    Deleted:    data/derived/lm_results_ashr.csv
    Modified:   data/derived/lmm_results_ashr.csv
    Deleted:    data/derived/output/DGRP_GRM.cXX.txt
    Deleted:    data/derived/output/DGRP_GRM.log.txt
    Deleted:    data/derived/output/female_early_bslmm.bv.txt
    Deleted:    data/derived/output/female_early_bslmm.gamma.txt
    Deleted:    data/derived/output/female_early_bslmm.hyp.txt
    Deleted:    data/derived/output/female_early_bslmm.log.txt
    Deleted:    data/derived/output/female_early_bslmm.param.txt
    Deleted:    data/derived/output/female_early_bslmm_preds.log.txt
    Deleted:    data/derived/output/female_early_bslmm_preds.prdt.txt
    Deleted:    data/derived/output/female_early_female_late.assoc.txt
    Deleted:    data/derived/output/female_early_female_late.log.txt
    Deleted:    data/derived/output/female_early_lm.assoc.txt
    Deleted:    data/derived/output/female_early_lm.log.txt
    Modified:   data/derived/output/female_early_lmm.assoc.txt
    Modified:   data/derived/output/female_early_lmm.log.txt
    Deleted:    data/derived/output/female_early_male_early.assoc.txt
    Deleted:    data/derived/output/female_early_male_early.log.txt
    Deleted:    data/derived/output/female_late_lm.assoc.txt
    Deleted:    data/derived/output/female_late_lm.log.txt
    Modified:   data/derived/output/female_late_lmm.assoc.txt
    Modified:   data/derived/output/female_late_lmm.log.txt
    Deleted:    data/derived/output/female_late_male_late.assoc.txt
    Deleted:    data/derived/output/female_late_male_late.log.txt
    Deleted:    data/derived/output/male_early_lm.assoc.txt
    Deleted:    data/derived/output/male_early_lm.log.txt
    Modified:   data/derived/output/male_early_lmm.assoc.txt
    Modified:   data/derived/output/male_early_lmm.log.txt
    Deleted:    data/derived/output/male_early_male_late.assoc.txt
    Deleted:    data/derived/output/male_early_male_late.log.txt
    Deleted:    data/derived/output/male_late_lm.assoc.txt
    Deleted:    data/derived/output/male_late_lm.log.txt
    Modified:   data/derived/output/male_late_lmm.assoc.txt
    Modified:   data/derived/output/male_late_lmm.log.txt
    Deleted:    data/derived/output/result.log.txt
    Deleted:    data/derived/output/result.prdt.txt
    Deleted:    data/derived/summed_interSNP_correlations.csv
    Deleted:    data/derived/trimmed_DGRP.bed
    Deleted:    data/derived/trimmed_DGRP.bim
    Deleted:    data/derived/trimmed_DGRP.fam
    Deleted:    data/derived/trimmed_DGRP.rds
    Modified:   data/input/dgrp2.fam

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
3db7587
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-10-01
</td>
<td style="text-align:left;">
1st October - lots more work
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
b6c43ed
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
<td style="text-align:left;">
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="steps-of-the-analysis" class="section level2">
<h2>Steps of the analysis</h2>
<div id="setting-up-a-database-to-hold-variantgene-annotations-and-gwas-results" class="section level3">
<h3><a href="get_predicted_line_means.html">1. Setting up a database to hold variant/gene annotations and GWAS results</a></h3>
<p>This script makes SQLite3 database holding two tables: one with annotations for each variant (provided by the Mackay lab), and one with annotations for each gene (from Bioconductor). Later on, we add the GWAS results to the database as well, allowing for memory-efficient processing of the results.</p>
</div>
<div id="estimating-line-mean-fitness-using-bayesian-models" class="section level3">
<h3><a href="get_predicted_line_means.html">1. Estimating line mean fitness using Bayesian models</a></h3>
<p>Using Bayesian mixed models implemented in the package <code>brms</code> to estimate the line means for our four fitness traits.</p>
</div>
<div id="running-the-gwas" class="section level3">
<h3><a href="perfom_gwas.html">2. Running the GWAS</a></h3>
<p>The script first performs quality control and imputation on the dataset of SNPs and indels for the DGRP. Second, it runs univariate and multivariate association tests on our four fitness traits using the software GEMMA.</p>
</div>
<div id="a.-applying-adaptive-shrinkage-to-the-gwas-results" class="section level3">
<h3><a href="gwas_adaptive_shrinkage.html">3a. Applying adaptive shrinkage to the GWAS results</a></h3>
<p>Here, we use the package <code>mashr</code> to perform multivariate adaptive shrinkage on the results of the univariate association tests. We ran <code>mashr</code> using two modes: canonical, and data-driven.</p>
</div>
<div id="b.-plots-illustrating-how-adaptive-shrinkage-affected-the-results" class="section level3">
<h3><a href="checking_mashr_results.html">3b. Plots illustrating how adaptive shrinkage affected the results</a></h3>
<p>This script generates some plots to check that <code>mashr</code> is performing as expected. The plots also led us to select the data-driven <code>mashr</code> results as the most conservative.</p>
</div>
<div id="analyses-using-the-transcriptomics-data-from-huang-et-al.-2015-pnas" class="section level3">
<h3><a href="TWAS.html">4. Analyses using the transcriptomics data from Huang et al. 2015 <em>PNAS</em></a></h3>
<hr />
</div>
</div>
<div id="results-of-the-analysis" class="section level2">
<h2>Results of the analysis</h2>
<div id="plots-showing-line-mean-fitness" class="section level3">
<h3><a href="plot_line_means.html">Plots showing line mean fitness</a></h3>
</div>
<div id="tables-of-gwas-results" class="section level3">
<h3><a href="GWAS_tables.html">Tables of GWAS results</a></h3>
</div>
<div id="twas" class="section level3">
<h3><a href="XXXX.html">TWAS</a></h3>
<p>Heritability - Correlated with selection experienced by the gene? Angatonism? - Correlated with sex-specificity?</p>
<p>eQTLs - Are there eQTLs with matching, independent, and opposite-sex effects? NEED TO RE-DO eQTL ANALYSIS A BIT - Are there more overlaps between the eQTLs and fitness QTLs than expected by chance? Correlation in p-vals?</p>
<p>Network - Modules are not correlated with fitness, so no evidence of selection on them? - Find the PCAs of the transcriptome, male vs female axis? Young vs old axis? - Are the modules heritable, and are there SNPs for eigengenes?</p>
<p>Predictions - eQTLs that affect transciption the same way in both sexes will have SA fitness effects for transcripts under SA selection (or concordant for concordant) We need: - Effect size for the eQTL on the transcript - Effect size for the eQTL on fitness - Effect size for the transcript on fitness - “Allele 1 made the transcript increase in both sexes, good for males and bad for females” - eQTLs that only affect transciption in one sex will have a more sex-specific fitness effect as well (not concordant, not antag) - eQTLs where the + and - alleles are swapped between sexes will be antagonistic or concordant, depending on sex-specific selection on that transcript - Transcripts will show the twin peaks relationship between sex bias in expression and SA selection - Genes targetted by dsx will be extra antagonistic</p>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
