<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Making a database of variant and gene annotations</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Making a database of variant and gene annotations</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree01226ab272891d92a8444e19cac034c1b079d263targetblank01226aba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/01226ab272891d92a8444e19cac034c1b079d263" target="_blank">01226ab</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree01226ab272891d92a8444e19cac034c1b079d263targetblank01226aba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/01226ab272891d92a8444e19cac034c1b079d263" target="_blank">01226ab</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    figures/fig1_inkscape.svg
    Ignored:    figures/figure1a.pdf
    Ignored:    figures/figure1b.pdf

Untracked files:
    Untracked:  big_model.rds
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds
    Untracked:  old_analyses/

Unstaged changes:
    Modified:   .gitignore
    Modified:   figures/GWAS_stats_figure.pdf
    Modified:   figures/SNP_effect_ED.pdf
    Modified:   figures/TWAS_stats_figure.pdf
    Modified:   figures/antagonism_ratios.pdf
    Modified:   figures/boyle_plot.pdf
    Modified:   figures/composite_mixture_figure.pdf
    Modified:   figures/eff_size_histos.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/make_annotation_database.Rmd</code>) and HTML (<code>docs/make_annotation_database.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/make_annotation_database.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/make_annotation_database.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3855d3359649c74f955042725d07276742cb649a/analysis/make_annotation_database.Rmd" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3855d3359649c74f955042725d07276742cb649a/docs/make_annotation_database.html" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/make_annotation_database.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/make_annotation_database.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(dplyr)
library(stringr)
library(future.apply)
library(org.Dm.eg.db) # install via source(&quot;https://bioconductor.org/biocLite.R&quot;); biocLite(&quot;org.Dm.eg.db&quot;)
library(GO.db)
options(future.globals.maxSize = 2000 * 1024 ^ 2, 
        stringsAsFactors = FALSE)

# Helper function to split a vector into chunks 
chunker &lt;- function(x, max_chunk_size) split(x, ceiling(seq_along(x) / max_chunk_size))

# database of D. mel annotations from bioconductor
con &lt;- dbconn(org.Dm.eg.db)</code></pre>
<div id="get-the-annotations-for-each-dgrp-variant" class="section level2">
<h2>Get the annotations for each DGRP variant</h2>
<p>The following function temporarily loads the &gt;1GB annotation file provided on the DGRP website at <a href="http://dgrp2.gnets.ncsu.edu/data/website/dgrp.fb557.annot.txt">http://dgrp2.gnets.ncsu.edu/data/website/dgrp.fb557.annot.txt</a>. We then extract the following variables for each variant, and save them in a SQLite database for memory-efficient searching inside R:</p>
<ul>
<li>The Flybase ID(s), if the variant is within or close to one or more genes</li>
<li>The site class of the variant (e.g. intron, 5’-UTR…)</li>
<li>The distance-to-gene in nucleotides (for UPSTREAM and DOWNSTREAM variants only)</li>
</ul>
<pre class="r"><code>get_variant_annotations &lt;- function(){
  
  # Load up the big annotation file, get pertinent info. It&#39;s stored in some sort of text string format
  annot &lt;- read.table(&quot;data/input/dgrp.fb557.annot.txt&quot;, header = FALSE, stringsAsFactors = FALSE)
  
  get.info &lt;- function(rows){
    lapply(rows, function(row){
      site.class.field &lt;- strsplit(annot$V3[row], split = &quot;]&quot;)[[1]][1]
      num.genes &lt;- str_count(site.class.field, &quot;;&quot;) + 1
      output &lt;- cbind(rep(annot$V1[row], num.genes), 
                      do.call(&quot;rbind&quot;, lapply(strsplit(site.class.field, split = &quot;;&quot;)[[1]], 
                                              function(x) strsplit(x, split = &quot;[|]&quot;)[[1]])))
      if(ncol(output) == 5) return(output[,c(1,2,4,5)]) # only return SNPs that have some annotation. Don&#39;t get the gene symbol
      else return(NULL)
    }) %&gt;% do.call(&quot;rbind&quot;, .)
  }
  
  plan(&quot;multisession&quot;)
  variant.details &lt;- future_lapply(chunker(1:nrow(annot), max_chunk_size = 10000), get.info) %&gt;% 
    do.call(&quot;rbind&quot;, .) %&gt;% as.data.frame()
  
  names(variant.details) &lt;- c(&quot;SNP&quot;, &quot;FBID&quot;, &quot;site.class&quot;, &quot;distance.to.gene&quot;)
  variant.details$FBID &lt;- unlist(str_extract_all(variant.details$FBID, &quot;FBgn[:digit:]+&quot;)) # clean up text strings for Flybase ID
  variant.details %&gt;%
    dplyr::filter(site.class != &quot;FBgn0003638&quot;) %&gt;% # NB this is a bug in the DGRP&#39;s annotation file
    mutate(chr = str_remove_all(substr(SNP, 1, 2), &quot;_&quot;)) # get chromosome now for faster sorting later
}</code></pre>
</div>
<div id="get-the-annotations-for-each-drosophila-gene" class="section level2">
<h2>Get the annotations for each <em>Drosophila</em> gene</h2>
<p>The following function gets the annotations for the all the genes covered by DGRP variants, from the <code>org.Dm.eg.db</code> database object from Bioconductor. I don’t like the <code>select</code> interface to those objects (it messes with any R code that uses <code>dplyr</code>), so here I save the info into the SQLite database for later access.</p>
<div id="table-to-convert-among-gene-ids-and-names" class="section level3">
<h3>Table to convert among gene IDs and names</h3>
<pre class="r"><code>get_gene_annotations &lt;- function(){
  tbl(con, &quot;genes&quot;) %&gt;%
    left_join(tbl(con, &quot;flybase&quot;), by = &quot;_id&quot;) %&gt;%
    left_join(tbl(con, &quot;gene_info&quot;), by = &quot;_id&quot;) %&gt;% 
    left_join(tbl(con, &quot;chromosomes&quot;), by = &quot;_id&quot;) %&gt;%
    dplyr::select(flybase_id, gene_name, symbol, gene_id, chromosome) %&gt;%
    dplyr::rename(FBID = flybase_id, gene_symbol = symbol, entrez_id = gene_id) %&gt;%
    collect(n = Inf)
}</code></pre>
</div>
<div id="kegg-annotations" class="section level3">
<h3>KEGG annotations</h3>
<pre class="r"><code>get_KEGG &lt;- function(){
  tbl(dbconn(org.Dm.eg.db), &quot;kegg&quot;) %&gt;%
    left_join(tbl(con, &quot;flybase&quot;), by = &quot;_id&quot;) %&gt;% 
    dplyr::select(flybase_id, path_id) %&gt;%
    dplyr::rename(FBID = flybase_id, kegg_id = path_id) %&gt;%
    collect(n = Inf)
}</code></pre>
</div>
<div id="go-annotations" class="section level3">
<h3>GO annotations</h3>
<pre class="r"><code>get_GO &lt;- function(){
  tbl(dbconn(org.Dm.eg.db), &quot;go_all&quot;) %&gt;%
    left_join(tbl(con, &quot;flybase&quot;), by = &quot;_id&quot;) %&gt;% 
    dplyr::select(flybase_id, go_id, ontology) %&gt;%
    dplyr::rename(FBID = flybase_id) %&gt;%
    collect(n = Inf)
}
GO &lt;- get_GO()</code></pre>
</div>
<div id="go-term-meanings" class="section level3">
<h3>GO term meanings</h3>
<pre class="r"><code>go_meanings &lt;- suppressMessages(
  AnnotationDbi::select(GO.db, 
                        GO$go_id, c(&quot;GOID&quot;, &quot;ONTOLOGY&quot;, &quot;TERM&quot;)))
names(go_meanings) &lt;- c(&quot;GO&quot;, &quot;ontology&quot;, &quot;term&quot;)
go_meanings &lt;- distinct(go_meanings)</code></pre>
</div>
</div>
<div id="create-the-sqlite-database-and-add-various-tables-of-annotations" class="section level2">
<h2>Create the SQLite database and add various tables of annotations</h2>
<pre class="r"><code>if(file.exists(&quot;data/derived/annotations.sqlite3&quot;)) unlink(&quot;data/derived/annotations.sqlite3&quot;)

db &lt;- DBI::dbConnect(RSQLite::SQLite(), &quot;data/derived/annotations.sqlite3&quot;, create = TRUE)

db %&gt;% copy_to(get_variant_annotations(), 
               &quot;variants&quot;, temporary = FALSE, 
               indexes = list(&quot;SNP&quot;, &quot;FBID&quot;, &quot;chr&quot;, &quot;site.class&quot;)) 

db %&gt;% copy_to(get_gene_annotations(), 
               &quot;genes&quot;, temporary = FALSE)

db %&gt;% copy_to(GO, &quot;GO&quot;, temporary = FALSE)

db %&gt;% copy_to(get_KEGG(), 
               &quot;KEGG&quot;, temporary = FALSE)

db %&gt;% copy_to(go_meanings, 
               &quot;go_meanings&quot;, temporary = FALSE)</code></pre>
<div id="view-the-table-of-variant-annotations" class="section level3">
<h3>View the table of variant annotations</h3>
<p>The <code>variants</code> table is expanded upon in the script <code>perform_gwas.Rmd</code>, which also adds the minor allele frequencies, the alleles that were treated as the reference and alternate, etc.</p>
<pre class="r"><code>db &lt;- DBI::dbConnect(RSQLite::SQLite(), &quot;data/derived/annotations.sqlite3&quot;)
db %&gt;% tbl(&quot;variants&quot;)</code></pre>
<pre><code># Source:   table&lt;variants&gt; [?? x 9]
# Database: sqlite 3.30.1
#   [/Users/lholman/Rprojects/fitnessGWAS/data/derived/annotations.sqlite3]
   SNP   FBID  site.class distance.to.gene chr   position   MAF minor_allele
   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;            &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;       
 1 2L_1… FBgn… NON_SYNON… 0                2L    10000016 0.463 C           
 2 2L_1… FBgn… INTRON     0                2L    10000016 0.463 C           
 3 2L_1… FBgn… SYNONYMOU… 0                2L    10000033 0.483 G           
 4 2L_1… FBgn… INTRON     0                2L    10000033 0.483 G           
 5 2L_1… FBgn… INTRON     0                2L    10000089 0.429 C           
 6 2L_1… FBgn… NON_SYNON… 0                2L    10000089 0.429 C           
 7 2L_1… FBgn… INTRON     0                2L    10000135 0.478 A           
 8 2L_1… FBgn… NON_SYNON… 0                2L    10000135 0.478 A           
 9 2L_1… FBgn… NON_SYNON… 0                2L    10000234 0.4   C           
10 2L_1… FBgn… INTRON     0                2L    10000234 0.4   C           
# … with more rows, and 1 more variable: major_allele &lt;chr&gt;</code></pre>
</div>
<div id="view-the-table-of-gene-annotations" class="section level3">
<h3>View the table of gene annotations</h3>
<pre class="r"><code>db %&gt;% tbl(&quot;genes&quot;)</code></pre>
<pre><code># Source:   table&lt;genes&gt; [?? x 5]
# Database: sqlite 3.30.1
#   [/Users/lholman/Rprojects/fitnessGWAS/data/derived/annotations.sqlite3]
   FBID        gene_name               gene_symbol entrez_id chromosome
   &lt;chr&gt;       &lt;chr&gt;                   &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;     
 1 FBgn0040373 uncharacterized protein CG3038      30970     X         
 2 FBgn0040372 G9a                     G9a         30971     X         
 3 FBgn0261446 uncharacterized protein CG13377     30972     X         
 4 FBgn0000316 cinnamon                cin         30973     X         
 5 FBgn0005427 erect wing              ewg         30975     X         
 6 FBgn0040370 uncharacterized protein CG13375     30976     X         
 7 FBgn0040371 uncharacterized protein CG12470     30977     X         
 8 FBgn0029521 Odorant receptor 1a     Or1a        30978     X         
 9 FBgn0024989 uncharacterized protein CG3777      30979     X         
10 FBgn0004034 yellow                  y           30980     X         
# … with more rows</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] GO.db_3.11.4         org.Dm.eg.db_3.11.4  AnnotationDbi_1.50.0
 [4] IRanges_2.22.2       S4Vectors_0.26.1     Biobase_2.48.0      
 [7] BiocGenerics_0.34.0  future.apply_1.5.0   future_1.17.0       
[10] stringr_1.4.0        dplyr_1.0.0          workflowr_1.6.2     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6     dbplyr_1.4.4     compiler_4.0.3   pillar_1.4.4    
 [5] later_1.0.0      git2r_0.27.1     tools_4.0.3      bit_1.1-15.2    
 [9] digest_0.6.25    memoise_1.1.0    RSQLite_2.2.0    evaluate_0.14   
[13] lifecycle_0.2.0  tibble_3.0.1     pkgconfig_2.0.3  rlang_0.4.6     
[17] cli_2.0.2        DBI_1.1.0        yaml_2.2.1       xfun_0.22       
[21] knitr_1.32       generics_0.0.2   fs_1.4.1         vctrs_0.3.0     
[25] globals_0.12.5   bit64_0.9-7      rprojroot_1.3-2  tidyselect_1.1.0
[29] glue_1.4.2       listenv_0.8.0    R6_2.4.1         fansi_0.4.1     
[33] rmarkdown_2.5    blob_1.2.1       purrr_0.3.4      magrittr_2.0.1  
[37] whisker_0.4      backports_1.1.7  promises_1.1.0   codetools_0.2-16
[41] htmltools_0.5.0  ellipsis_0.3.1   assertthat_0.2.1 httpuv_1.5.3.1  
[45] utf8_1.1.4       stringi_1.5.3    crayon_1.3.4    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
