<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GWAS for fitness in Drosophila</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">GWAS for fitness in Drosophila</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree2080e0c3655d7a874de57c46fa63b3828250aaa2targetblank2080e0ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/fitnessGWAS/tree/2080e0c3655d7a874de57c46fa63b3828250aaa2" target="_blank">2080e0c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree2080e0c3655d7a874de57c46fa63b3828250aaa2targetblank2080e0ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/fitnessGWAS/tree/2080e0c3655d7a874de57c46fa63b3828250aaa2" target="_blank">2080e0c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  big_model.rds
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds

Unstaged changes:
    Modified:   .gitignore
    Modified:   code/pdf_supp_material.Rmd
    Modified:   code/pdf_supp_material.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/perform_gwas.Rmd</code>) and HTML
(<code>docs/perform_gwas.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/2080e0c3655d7a874de57c46fa63b3828250aaa2/analysis/perform_gwas.Rmd" target="_blank">2080e0c</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
<td>
wflow_publish("analysis/perform_gwas.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/d94e22dc2fbcacfe3430866ac5f7116866a93ec3/analysis/perform_gwas.Rmd" target="_blank">d94e22d</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-14
</td>
<td>
New quant gen and other changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/0a415e319a073797e6b6ea9a30327b0b035408c7/docs/perform_gwas.html" target="_blank">0a415e3</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/perform_gwas.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/9f7618929a446215b9da90c5a8c3fc748f674e37/docs/perform_gwas.html" target="_blank">9f76189</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1f52aabd1d1260296dd0394f862024b9bbfa11d2/analysis/perform_gwas.Rmd" target="_blank">1f52aab</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-27
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/perform_gwas.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/perform_gwas.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/410fb733e98fda42c7872a829607b7ff926ff6ac/docs/perform_gwas.html" target="_blank">410fb73</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/871ae81fa56175675b53d22ca29bf9323c3ed50a/docs/perform_gwas.html" target="_blank">871ae81</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/perform_gwas.html" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/836a780685e8ac20355b0a92672276f7ca763481/docs/perform_gwas.html" target="_blank">836a780</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/359ff37c4f4021451e81122393307a28013165a9/docs/perform_gwas.html" target="_blank">359ff37</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/5506c4b50ea385dbe5174320ad41d797c64d0676/docs/perform_gwas.html" target="_blank">5506c4b</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/f64a3e97eddb218609bfe9dd0b9c3b4d282b51b2/docs/perform_gwas.html" target="_blank">f64a3e9</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/fa1b9b0ab0805beb3f89b75330f07950953fc892/analysis/perform_gwas.Rmd" target="_blank">fa1b9b0</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/perform_gwas.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/perform_gwas.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(dplyr)
library(ggplot2)
# library(ggdendro)
library(glue)
library(bigsnpr) # to install:   devtools::install_github(&quot;privefl/bigsnpr&quot;)
library(readr)
library(pander)
library(purrr)
library(future.apply) # for parallel code

# Load the predicted line means, as calculated by get_predicted_line_means
predicted_line_means &lt;- read_csv(&quot;data/derived/predicted_line_means.csv&quot;)
plink &lt;- paste(getwd(), &quot;code/plink&quot;, sep = &quot;/&quot;)

# # Load the wolbachia infection status data from the DGRP website
# wolbachia &lt;- read_csv(&quot;data/input/wolbachia.csv&quot;) %&gt;% 
#   mutate(line = paste(&quot;line&quot;, line, sep = &quot;&quot;))
# 
# # Load the chomosomal inversion data from the DGRP website
# # these are the 5 inversions that Huang et al. PNAS corrected for
# inversions &lt;- read_csv(&quot;data/input/inversion genotypes.csv&quot;) %&gt;%
#     mutate(line = paste(&quot;line&quot;, line, sep = &quot;&quot;)) %&gt;%
#     select(line, `In(2L)t`, `In(2R)NS`, `In(3R)P`, `In(3R)K`, `In(3R)Mo`) 

# helper function to pass commands to the terminal
# Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`, 
# ensuring that the Terminal output will be printed in this knitr report. 
run_command &lt;- function(shell_command, wd = getwd(), path = &quot;&quot;){
  cat(system(glue(&quot;cd &quot;, wd, path, &quot;\n&quot;,shell_command), intern = TRUE), sep = &#39;\n&#39;)
}</code></pre>
<div id="perform-snp-quality-control-and-imputation"
class="section level2">
<h2>Perform SNP quality control and imputation</h2>
<p>We cleaned up the DGRP’s .bed/.bim/.fam files (available from the <a
href="http://dgrp2.gnets.ncsu.edu/">Mackay lab website</a>) as
follows:</p>
<ol style="list-style-type: decimal">
<li>Remove any SNPs for which genotypes are missing for &gt;10% of the
DGRP lines. We then use the software <a
href="https://faculty.washington.edu/browning/beagle/beagle.html">Beagle</a>
to impute the remaining missing genotypes.</li>
<li>Remove SNPs with a minor allele frequency of less than 5%</li>
</ol>
<p>Note that in the PLINK-formatted genotype files, lines fixed for the
major allele are coded as 2, and lines fixed for the minor allele as 0.
This means that in the association tests we calculate, negative effect
sizes mean that the minor allele is associated with lower fitness, while
positive effect sizes means that the minor allele is associated with
higher fitness.</p>
<pre class="r"><code>beagle &lt;- bigsnpr::download_beagle()

perform_SNP_QC_and_imputation &lt;- function(phenotypes){
  
  if(&quot;block&quot; %in% names(phenotypes)) phenotypes &lt;- phenotypes %&gt;% select(-block)
  
  # Make a list of the lines in our sample and save as a text file for passing to PLINK
  lines_to_keep &lt;- gsub(&quot;_&quot;, &quot;&quot;, phenotypes$line) %&gt;% cbind(.,.)
  write.table(lines_to_keep, row.names = FALSE, col.names = FALSE, file = &quot;data/derived/lines_to_keep.txt&quot;, quote = FALSE)
  
  # Define a function to add our phenotype data to a .fam file, which is needed for GWAS analysis and to make sure PLINK includes these samples
  # The &#39;phenotypes&#39; data frame needs to have a column called &#39;line&#39;
  add_phenotypes_to_fam &lt;- function(filepath, phenotypes){
    read_delim(filepath, col_names = FALSE, delim = &quot; &quot;) %&gt;% 
      select(X1, X2, X3, X4, X5) %&gt;% # Get all the non-phenotype columns
      left_join(phenotypes %&gt;% 
                  mutate(line = gsub(&quot;_&quot;, &quot;&quot;, line)), 
                by = c(&quot;X1&quot; = &quot;line&quot;)) %&gt;%
      write.table(file = filepath, 
                  col.names = FALSE, row.names = FALSE, 
                  quote = FALSE, sep = &quot; &quot;)
  }
  
  # Use Plink to clean and subset the DGRP&#39;s SNP data as follows:
  # Only keep SNPs for which at least 90% of DGRP lines were successfully genotyped (--geno 0.1)
  # Only keep SNPs with a minor allele frequency of 0.05 or higher (--maf 0.05)
  # Finally, write the processed BIM/BED/FAM files to the data/derived directory
  run_command(glue(&quot;{plink} --bfile dgrp2&quot;,
                   &quot; --geno 0.1 --maf 0.05 --allow-no-sex&quot;, 
                   &quot; --make-bed --out ../derived/dgrp2_QC_all_lines&quot;), path = &quot;/data/input/&quot;)
  
  # Use the shell command &#39;sed&#39; to remove underscores from the DGRP line names in the .fam file (e.g. &#39;line_120&#39; becomes &#39;line120&#39;)
  # Otherwise, these underscores cause trouble when we need to convert from PLINK to vcf format (vcf format uses underscore as a separator)
  for(i in 1:2) run_command(&quot;sed -i &#39;&#39; &#39;s/_//&#39; dgrp2_QC_all_lines.fam&quot;, path = &quot;/data/derived/&quot;)
  
  # Now impute the missing genotypes using Beagle
  # This part uses the data for the full DGRP panel of &gt;200 lines, to infer missing genotypes as accurately as possible. 
  # This step uses a lot of memory (I set to 28MB max, and it used 26.5GB), but maybe it can also run on a less powerful computer?
  # The bigsnpr package provides a helpful wrapper for Beagle called snp_beagleImpute(): it translates to a VCF file and back again using PLINK
  snp_beagleImpute(beagle, plink, 
                   bedfile.in = &quot;data/derived/dgrp2_QC_all_lines.bed&quot;, 
                   bedfile.out = &quot;data/derived/dgrp2_QC_all_lines_imputed.bed&quot;,
                   ncores = 7, 
                   memory.max = 20)
  
  # assign a sex of &#39;female&#39; to all the DGRP lines (Beagle removes the sex, and it seems PLINK needs individuals to have a sex, 
  # despite PLINK having a command called --allow-no-sex)
  run_command(&quot;sed -i &#39;&#39; &#39;s/    0   0   0/  0   0   2/&#39; dgrp2_QC_all_lines_imputed.fam&quot;, path = &quot;/data/derived/&quot;)
  
  # Re-write the .bed file, to make sure the MAF threshold and minor/major allele designations are correctly assigned post-Beagle
  run_command(glue(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed&quot;,
                   &quot; --geno 0.1 --maf 0.05 --allow-no-sex&quot;, 
                   &quot; --make-bed --out dgrp2_QC_all_lines_imputed_correct&quot;), path = &quot;/data/derived/&quot;)
  #unlink(list.files(&quot;data/derived&quot;, pattern = &quot;~&quot;, full.names = TRUE)) # delete the original files, which were given a ~ file name by PLINK
  
  # Use PLINK to get the allele IDs and calculate the MAFs across the whole DGRP, for all SNPs that survived QC
  # The file created is called data/derived/plink.frq
  run_command(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct --freq&quot;, path = &quot;/data/derived&quot;)

  # Now cull the PLINK files to just the lines that we measured, and re-apply the 
  # MAF cut-off of 0.05 for the new smaller sample of DGRP lines
  run_command(glue(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct&quot;,
                   &quot; --keep-allele-order&quot;, # force PLINK to retain the major/minor allele designations that apply to the DGRP as a whole
                   &quot; --keep lines_to_keep.txt --geno 0.1 --maf 0.05&quot;, 
                   &quot; --make-bed --out dgrp2_QC_focal_lines&quot;), path = &quot;/data/derived/&quot;)
  
  # edit the new FAM file to add the phenotype data from &#39;phenotypes&#39;
  add_phenotypes_to_fam(&quot;data/derived/dgrp2_QC_focal_lines.fam&quot;, phenotypes)
  
  # Clean up:
  unlink(c(&quot;data/derived/lines_to_keep.txt&quot;, 
           &quot;data/derived/plink.log&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed.bed&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed.bim&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed.fam&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed.log&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed_correct.bed&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed_correct.bim&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed_correct.fam&quot;,
           &quot;data/derived/dgrp2_QC_all_lines_imputed_correct.log&quot;))
}

perform_SNP_QC_and_imputation(phenotypes = predicted_line_means)</code></pre>
</div>
<div id="run-gwas-using-gemma" class="section level2">
<h2>Run GWAS using GEMMA</h2>
<!-- ### Create a file of _Wolbachia_ infection status -->
<!-- GEMMA requires that the "covariate" file be in a specific format. Each row needs to match an individual (i.e. a DGRP line) in the PLINK file, and the order should be the same. Each column is one of the covariates. To fit an intercept in addition to the covariates, there should be a column of 1s. Thus, here we make a two-column tsv file wiht the intercept in column 1, and the _Wolbachia_ infection status (a y or a n) in column 2. We elected not to fit the presence/absence of the major inversions (as in sometimes done in studies of the DGRP), since that seems like a trait that will already be captured by the decomposed eigenvectors of the SNP data (since inversions create a block of linked SNPs). -->
<!-- ```{r eval=FALSE} -->
<!-- # Get the lines for the coming GWAS from the FAM file -->
<!-- lines_in_fam <- read_delim("data/derived/dgrp2_QC_focal_lines.fam",  -->
<!--                            col_names = FALSE, delim = " ") %>%  -->
<!--   select(X1) %>% rename(line = X1) -->
<!-- # merge in the covariates: the intercept, the Wolbachia, and the Inversion genotypes -->
<!-- covariates <- left_join(lines_in_fam, wolbachia, by = "line") %>% -->
<!--   left_join(inversions %>% -->
<!--               rename(Inv_1 = `In(2L)t`, -->
<!--                      Inv_2 = `In(2R)NS`, -->
<!--                      Inv_3 = `In(3R)P`, -->
<!--                      Inv_4 = `In(3R)K`, -->
<!--                      Inv_5 = `In(3R)Mo`), by = "line") %>% -->
<!--   mutate(intercept = 1, -->
<!--          wolbachia = as.numeric(ifelse(wolbachia == "y", 1, 0))) %>%  -->
<!--   select(intercept, wolbachia, Inv_1, Inv_2, Inv_3, Inv_4, Inv_5) -->
<!-- write_delim(covariates, path = "data/derived/covariate_file_for_GEMMA.txt", col_names = FALSE) -->
<!-- ``` -->
<div id="decompose-the-genomic-relatedness-matrix"
class="section level3">
<h3>Decompose the genomic relatedness matrix</h3>
<p>This step is needed as a set up for the mixed model association
tests, which use the eigenvectors and values of the GRM to adjust for
‘population structure’ among our samples.</p>
<pre class="r"><code># Any command with {bed} uses this particular PLINK file, double-checking that all the SNPs have MAF &gt; 0.05
bed &lt;- &quot;dgrp2_QC_focal_lines -maf 0.05&quot; 
#cov &lt;- &quot;-c covariate_file_for_GEMMA.txt&quot; # not used at present

# Calculate the *centered* GRM for the focal lines (option -gk 1)
run_command(&quot;gemma -bfile {bed} -r2 0.99 -gk 1 -o GRM&quot;, path = &quot;/data/derived&quot;) 

# Perform decomposition of the GRM, and save its eigenvalues and eigenvectors
run_command(&quot;gemma -bfile {bed} -r2 0.99 -k ./output/GRM.cXX.txt -eigen -o eigen_decomp&quot;, path = &quot;/data/derived&quot;) 

# Remove eigen vectors + values for which the eigenvalue is close to zero, and save the modified file
values &lt;- read_tsv(&quot;data/derived/output/eigen_decomp.eigenD.txt&quot;, col_names = &quot;eigenvalue&quot;)$eigenvalue
vectors &lt;- read_tsv(&quot;data/derived/output/eigen_decomp.eigenU.txt&quot;, col_names = FALSE)
missing_row &lt;- which(is.na(read_delim(&quot;data/derived/dgrp2_QC_focal_lines.fam&quot;, delim = &quot; &quot;, col_names = FALSE)$X8))

write_tsv(vectors, path = &quot;data/derived/output/eigen_decomp.eigenU_female.txt&quot;, col_names = FALSE)
write_tsv(data.frame(values), path = &quot;data/derived/output/eigen_decomp.eigenD_female.txt&quot;, col_names = FALSE)

write_tsv(vectors[-missing_row, -missing_row], path = &quot;data/derived/output/eigen_decomp.eigenU_male.txt&quot;, col_names = FALSE)
write_tsv(data.frame(values[-missing_row]), path = &quot;data/derived/output/eigen_decomp.eigenD_male.txt&quot;, col_names = FALSE)
unlink(&quot;data/derived/output/eigen_decomp.eigenD.txt&quot;)
unlink(&quot;data/derived/output/eigen_decomp.eigenU.txt&quot;)
# vectors &lt;- vectors[,-(which(values &lt; 0.2))]
# values &lt;- values[-(which(values &lt; 0.2))]
# write_tsv(vectors, path = &quot;data/derived/output/eigen_decomp.eigenU.txt&quot;, col_names = FALSE)
# write_tsv(data.frame(value = values), path = &quot;data/derived/output/eigen_decomp.eigenD.txt&quot;, col_names = FALSE)</code></pre>
</div>
<div id="run-univariate-and-multivariate-association-tests"
class="section level3">
<h3>Run univariate and multivariate association tests</h3>
<p>The following code chunk runs 5 linear mixed models, implemented in
<a href="http://www.xzlab.org/software/GEMMAmanual.pdf">GEMMA</a>. Each
of the univariate linear mixed models uses the decomposed genomic
relatedness matrix from above, and has one of our four fitness traits as
the response variable. The multivariate model contains all four traits,
and also uses the decomposed genomic relatedness matrix.</p>
<pre class="r"><code># The option &quot;-lmm 1&quot; runs a linear mixed model using Wald test to get the p-values
# The option &quot;-r2 0.95&quot; only keeps SNPs that are &lt;95% correlated with Wolbachia infection status
# The option &quot;-c covariate_file_for_GEMMA.txt&quot; adds in the Wolbahia infection status as a covariate, and also tells the model to fit an intercept
opts &lt;- &quot;-lmm 1&quot;
GRM_female &lt;- &quot;-d ./output/eigen_decomp.eigenD_female.txt -u ./output/eigen_decomp.eigenU_female.txt&quot;
GRM_male &lt;- &quot;-d ./output/eigen_decomp.eigenD_male.txt -u ./output/eigen_decomp.eigenU_male.txt&quot;
#cov &lt;- &quot;-c covariate_file_for_GEMMA.txt -r2 0.99&quot; # not used at present

c(&quot;gemma -bfile {bed} {GRM_female} {opts} -n 1 -o female_early_lmm&quot;,
  &quot;gemma -bfile {bed} {GRM_female} {opts} -n 2 -o female_late_lmm&quot;,
  &quot;gemma -bfile {bed} {GRM_male} {opts} -n 3 -o male_early_lmm&quot;,
  &quot;gemma -bfile {bed} {GRM_male} {opts} -n 4 -o male_late_lmm&quot;) %&gt;%
  #&quot;gemma -bfile {bed} {GRM_male} {opts} -n 1 2 3 4 -o all_four_traits&quot;) %&gt;% 
  lapply(run_command, path = &quot;/data/derived&quot;)</code></pre>
<!-- # Alternative PVE estimates from GEMMA -->
<!-- ### Inspect the GEMMA log files from the association tests {.tabset} -->
<!-- According to the manual, GEMMA calculates the relatedness matrix from the SNP data, then "extracts the matrix elements corresponding to the analyzed individuals (which may be smaller than the number of total individuals), centers the matrix, and then performs an eigendecomposition". The aim is to better estimate the effects of each SNP independently of the rest of the genome.  -->
<!-- The metric `pve` in the outputs below is "the proportion of variance in the phenotype explained by typed genotypes", also called SNP heritability. Next, `vg` is $V_g$, the variance in phenotype due to variance in genotype, while `ve` is $V_e$, the variance in phenotype due to variance in environment. Note that we present alternative estimates from gcta64 for the heritabilities in the manuscript, though these estimates from GEMMA are qualitatively very similar.  -->
<!-- #### Female early-life fitness -->
<!-- ```{r lmm1} -->
<!-- cat(readLines("data/derived/output/female_early_lmm.log.txt"), sep = "\n") -->
<!-- ``` -->
<!-- #### Male early-life fitness -->
<!-- ```{r lmm2} -->
<!-- cat(readLines("data/derived/output/male_early_lmm.log.txt"), sep = "\n") -->
<!-- ``` -->
<!-- #### Female late-life fitness -->
<!-- ```{r lmm3} -->
<!-- cat(readLines("data/derived/output/female_late_lmm.log.txt"), sep = "\n") -->
<!-- ``` -->
<!-- #### Male late-life fitness -->
<!-- ```{r lmm4} -->
<!-- cat(readLines("data/derived/output/male_late_lmm.log.txt"), sep = "\n") -->
<!-- ``` -->
</div>
<div
id="add-the-allele-ids-and-minor-allele-frequencies-mafs-to-variant-annotation-database"
class="section level3">
<h3>Add the allele IDs and minor allele frequencies (MAFs) to variant
annotation database</h3>
<p>This part uses the full DGRP panel, because our aim is to determine
how common these alleles are in the original wild population from which
the DGRP was captured. These MAFs are the ones used in our statistical
analyses and plots.</p>
<pre class="r"><code># Extract and save the MAFs, SNP positions, and major/minor alleles
MAFs &lt;- read.table(&quot;data/derived/plink.frq&quot;, header = TRUE, stringsAsFactors = FALSE) %&gt;% 
  mutate(position = map_chr(strsplit(SNP, split = &quot;_&quot;), function(x) x[2])) %&gt;%
  select(SNP, position, MAF, A1, A2) %&gt;% 
  rename(minor_allele = A1,
         major_allele = A2) 

db &lt;- DBI::dbConnect(RSQLite::SQLite(), &quot;data/derived/annotations.sqlite3&quot;, create = FALSE)

MAFs &lt;- tbl(db, &quot;variants&quot;) %&gt;% 
  select(SNP, FBID, site.class, distance.to.gene, chr) %&gt;% collect() %&gt;%
  full_join(MAFs, by = &quot;SNP&quot;) %&gt;%
  filter(!is.na(MAF)) %&gt;%
  mutate(site.class = replace(site.class, is.na(site.class), &quot;INTERGENIC&quot;))

# Delete the original variant annotation table from the db, and add the new one back in
db %&gt;% db_drop_table(table = &quot;variants&quot;) 
db %&gt;% copy_to(MAFs, 
               &quot;variants&quot;, temporary = FALSE, 
               indexes = list(&quot;SNP&quot;, &quot;FBID&quot;, &quot;chr&quot;, &quot;site.class&quot;)) 
rm(MAFs)</code></pre>
</div>
<div id="clean-up-the-gemma-results" class="section level3">
<h3>Clean up the GEMMA results</h3>
<p>Discard columns we don’t need, and rename the SNP column from the
name assigned by GEMMA (<code>rs</code>) to something more sensible
(<code>SNP</code>).</p>
<pre class="r"><code># The 4 univariate files:
c(&quot;data/derived/output/female_early_lmm.assoc.txt&quot;,
  &quot;data/derived/output/female_late_lmm.assoc.txt&quot;,
  &quot;data/derived/output/male_early_lmm.assoc.txt&quot;,
  &quot;data/derived/output/male_late_lmm.assoc.txt&quot;) %&gt;%
  lapply(function(filename){
    df &lt;- read_tsv(filename)
    names(df)[names(df) == &quot;rs&quot;] &lt;- &quot;SNP&quot;
    df %&gt;% select(SNP, beta, se, p_wald) %&gt;% 
      write_tsv(filename)
  })</code></pre>
</div>
<div id="perform-snp-clumping-using-plink" class="section level3">
<h3>Perform SNP clumping using PLINK</h3>
<pre class="r"><code># Use the un-manipulated DGRP bed/bim/fam files, since they don&#39;t throw an error from plink
bed &lt;- &quot;../input/dgrp2&quot;

# Remind plink to only use the SNPs that passed QC
# read_tsv(&quot;data/derived/output/female_early_lmm.assoc.txt&quot;) %&gt;% select(SNP) %&gt;%
#   write_tsv(col_names = FALSE, path = &quot;data/derived/snp_list.txt&quot;)


do_snp_clumping &lt;- function(trait){

  run_command(glue(&quot;plink --bfile {bed}&quot;, 
                   &quot; --clump output/{trait}.assoc.txt&quot;,
                   &quot; --clump-field p_wald --clump-p1 0.00001&quot;), # Each clump must contain one snp with p &lt; 10 ^ -5
              path = &quot;/data/derived&quot;)
  trait &lt;- gsub(&quot;_lmm&quot;, &quot;&quot;, trait)
  file_name &lt;- glue(&quot;data/derived/{trait}_SNP_clumps.txt&quot;)
  file.rename(&quot;data/derived/plink.clumped&quot;, file_name)
  
  clumps &lt;- read.table(file_name, header = TRUE) %&gt;% 
    as_tibble() %&gt;%
    rename(index_SNP = SNP,
           index_SNP_p_value = P,
           num_SNPs_in_clump = TOTAL,
           not_sig = NSIG, # Number of clumped SNPs that are not significant ( p &gt; 0.05 )
           sig_p05 = S05, # Number of clumped SNPs 0.01 &lt; p &lt; 0.05
           sig_p01 = S01, # Number of clumped SNPs 0.001 &lt; p &lt; 0.01
           sig_p001 = S001, # Number of clumped SNPs 0.0001 &lt; p &lt; 0.001 
           sig_p0001 = S0001, # Number of clumped SNPs p &lt; 0.0001
           other_snps = SP2) %&gt;%
    mutate(other_snps = gsub(&quot;\\(1\\),&quot;, &quot;, &quot;, other_snps),
           other_snps = gsub(&quot;\\(1\\)&quot;, &quot;&quot;, other_snps),
           other_snps = replace(other_snps, other_snps == &quot;NONE&quot;, NA)) %&gt;%
    select(-CHR, -F, -BP) 
  
  SNPs &lt;- apply(clumps, 1, function(x) {
    res &lt;- c(x[1], strsplit(x[9], split = &quot;, &quot;)[[1]])
    res[!is.na(res)]
  })
  
  db_local &lt;- tbl(db, &quot;variants&quot;) %&gt;% select(SNP, FBID) %&gt;% collect() 
  clumps$FBIDs &lt;- lapply(SNPs, function(x) {
    focal &lt;- db_local %&gt;% filter(SNP %in% x) 
    if(nrow(focal) == 0) return(NA)
    focal %&gt;% pull(FBID) %&gt;% unique() %&gt;% paste0(collapse = &quot;, &quot;)
    }) %&gt;% unlist()
  
  db_local &lt;- tbl(db, &quot;genes&quot;) %&gt;% collect()
  clumps$genes &lt;- lapply(clumps$FBIDs, function(x) {
    if(is.na(x)) return(NA)
    FB &lt;- strsplit(x, split = &quot;, &quot;)[[1]]
    return(db_local %&gt;% filter(FBID %in% FB) %&gt;% pull(gene_name) %&gt;% paste0(collapse = &quot;, &quot;))
  }) %&gt;% unlist()
  
  write_tsv(clumps, file_name)
  unlink(&quot;data/derived/plink.log&quot;)
}

lapply(c(&quot;female_early_lmm&quot;, &quot;female_late_lmm&quot;, &quot;male_early_lmm&quot;, &quot;male_late_lmm&quot;), do_snp_clumping)</code></pre>
</div>
</div>
<div id="concatenate-the-results-of-the-four-univariate-gwas"
class="section level2">
<h2>Concatenate the results of the four univariate GWAS</h2>
<p>Load up the GWAS results, which were produced by using linear mixed
models implemented in GEMMA, and arrange the data in a ‘wide’ format,
where each row is one SNP, and the columns hold the estimates of <span
class="math inline">\(\beta\)</span>, the standard error of <span
class="math inline">\(\beta\)</span>, and the p-value for the GEMMA
association tests on each of the four fitness traits.</p>
<pre class="r"><code>load_gwas_results &lt;- function(fitness.component){
  new_names &lt;- c(x = &quot;beta&quot;, y = &quot;se&quot;, z = &quot;p_wald&quot;)
  names(new_names) &lt;- c(paste(&quot;beta_&quot;, fitness.component, sep = &quot;&quot;), 
                        paste(&quot;SE_&quot;, fitness.component, sep = &quot;&quot;), 
                        paste(&quot;pvalue_&quot;, fitness.component, sep = &quot;&quot;))
  
  paste(&quot;data/derived/output/&quot;, fitness.component, &quot;_lmm.assoc.txt&quot;, sep = &quot;&quot;) %&gt;%
    read_tsv() %&gt;%
    select(SNP, beta, se, p_wald) %&gt;%
    rename(!!new_names)
}
all_univariate_lmm_results &lt;- load_gwas_results(&quot;female_early&quot;) %&gt;%
  full_join(load_gwas_results(&quot;female_late&quot;), by = &quot;SNP&quot;) %&gt;%
  full_join(load_gwas_results(&quot;male_early&quot;), by = &quot;SNP&quot;) %&gt;%
  full_join(load_gwas_results(&quot;male_late&quot;), by = &quot;SNP&quot;) </code></pre>
</div>
<div id="prune-snps-that-are-in-100-linkage-disequilibrium"
class="section level2">
<h2>Prune SNPs that are in 100% linkage disequilibrium</h2>
<p>Because GEMMA is quick and easy to run, we did not bother working out
how to edit PLINK files to identify SNPs that are in complete linkage
disequilibrium with one another (and there is lots of short- and
long-range LD in the dataset). However, it is easy to perform LD pruning
at this stage, because we can take advantage of the fact that SNPs in
perfect LD with each other will have identical estimates of <span
class="math inline">\(|\beta|\)</span> and its standard error, for all
four fitness traits. The following code groups together SNPs that have
identical association test results, producing a data frame in which the
number of rows is equal to the number of unique datasets
(i.e. combinations of genotypes and phenotypes) that were statistically
analysed by GEMMA.</p>
<pre class="r"><code># Group loci with identical GWAS results (these are in 100% LD with each other)
all_univariate_lmm_results &lt;- all_univariate_lmm_results %&gt;%
  mutate(pasted = paste(beta_female_early, beta_female_late, beta_male_early, beta_male_late, 
                        SE_female_early, SE_female_late, SE_male_early, SE_male_late)) %&gt;%
  group_by(pasted) %&gt;%
  summarise(SNPs = paste0(SNP, collapse = &quot;, &quot;),    # If there are multiple SNPs, concatenate their names
            beta_female_early = beta_female_early[1], 
            beta_female_late = beta_female_late[1], 
            beta_male_early = beta_male_early[1], 
            beta_male_late = beta_male_late[1],
            SE_female_early = SE_female_early[1], 
            SE_female_late = SE_female_late[1], 
            SE_male_early = SE_male_early[1], 
            SE_male_late = SE_male_late[1],
            pvalue_female_early = pvalue_female_early[1], 
            pvalue_female_late = pvalue_female_late[1], 
            pvalue_male_early = pvalue_male_early[1], 
            pvalue_male_late = pvalue_male_late[1]) %&gt;%
  ungroup() %&gt;% select(-pasted) %&gt;% 
  arrange(SNPs)


all_univariate_lmm_results &lt;- read_csv(&quot;data/derived/all_univariate_GEMMA_results.csv&quot;)
# a handful of SNPs with low MAF could not be estimated and they have beta and SE == NA, so remove them now
# I think these are SNPs where the only lines with the minor allele were the lines where we measured female but not male fitness
SNPs_to_remove &lt;- all_univariate_lmm_results$SNPs[!complete.cases(all_univariate_lmm_results)] 

all_univariate_lmm_results &lt;- all_univariate_lmm_results %&gt;% 
  filter(!(SNPs %in% SNPs_to_remove))

write_csv(all_univariate_lmm_results, file = &quot;data/derived/all_univariate_GEMMA_results.csv&quot;)
unlink(&quot;data/derived/output/female_early_lmm.assoc.txt&quot;)
unlink(&quot;data/derived/output/female_late_lmm.assoc.txt&quot;)
unlink(&quot;data/derived/output/male_early_lmm.assoc.txt&quot;)
unlink(&quot;data/derived/output/male_late_lmm.assoc.txt&quot;)</code></pre>
</div>
<div id="create-a-reduced-list-of-ld-pruned-snps-with-plink"
class="section level2">
<h2>Create a reduced list of LD-pruned SNPs with PLINK</h2>
<p>To keep the computation time and memory usage for <code>mashr</code>
manageable, we did not analyse every SNP analysed in the GWAS
(i.e. 1207357 SNPs), but rather a subset of them that were approximately
in linkage disequilibrium. We identified this LD-pruned set of SNPs
using the PLINK arguments <code>--indep-pairwise 100 10 0.2</code>,
i.e. pruning within 100kB sliding windows, sliding 10 variants along
with each step, and allowing a maximum pairwise <span
class="math inline">\(r^2\)</span> threshold of 0.2 between loci. With
these parameters, 1420071 SNPs were removed, leaving 2.26581^{5} for
downstream analysis.</p>
<pre class="r"><code># indep-pairwise arguments are: 
# 100kB window size, 
# variant count to shift the window by 10 variants at the end of each step, 
# pairwise r^2 threshold of 0.2
run_command(glue(&quot;{plink} --bfile dgrp2_QC_focal_lines&quot;,
                 &quot; --indep-pairwise 100 10 0.2&quot;), path = &quot;/data/derived/&quot;)</code></pre>
<pre><code>PLINK v1.90b6.24 64-bit (6 Jun 2021)           www.cog-genomics.org/plink/1.9/
(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to plink.log.
Options in effect:
  --bfile dgrp2_QC_focal_lines
  --indep-pairwise 100 10 0.2

32768 MB RAM detected; reserving 16384 MB for main workspace.
1609590 variants loaded from .bim file.
125 people (0 males, 125 females) loaded from .fam.
125 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 125 founders and 0 nonfounders present.
Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
Total genotyping rate is exactly 1.
1609590 variants and 125 people pass filters and QC.
Phenotype data is quantitative.

1%
2%
3%
4%
5%
6%
7%
8%
9%
10%
11%
12%
13%
14%
15%
16%
17%
18%
19%
20%
21%
22%
23%
24%
25%
26%
27%
28%
29%
30%
31%
32%
33%
34%
35%
36%
37%
38%
39%
40%
41%
42%
43%
44%
45%
46%
47%
48%
49%
50%
51%
52%
53%
54%
55%
56%
57%
58%
59%
60%
61%
62%
63%
64%
65%
66%
67%
68%
69%
70%
71%
72%
73%
74%
75%
76%
77%
78%
79%
80%
81%
82%
83%
84%
85%
86%
87%
88%
89%
90%
91%
92%
93%
94%
95%
96%
97%
98%
99%
Pruned 304959 variants from chromosome 1, leaving 45707.

1%
2%
2%
3%
4%
5%
5%
6%
7%
8%
8%
9%
10%
11%
11%
12%
13%
14%
14%
15%
16%
17%
17%
18%
19%
20%
20%
21%
22%
23%
23%
24%
25%
26%
26%
27%
28%
29%
30%
31%
32%
33%
34%
35%
36%
37%
38%
39%
40%
41%
42%
43%
44%
45%
46%
47%
48%
49%
50%
51%
52%
53%
54%
55%
56%
57%
58%
59%
60%
61%
62%
63%
64%
65%
66%
67%
68%
69%
70%
71%
72%
73%
74%
75%
76%
77%
78%
79%
80%
81%
82%
83%
84%
85%
86%
87%
88%
89%
90%
91%
92%
93%
94%
95%
96%
97%
98%
99%
Pruned 275365 variants from chromosome 2, leaving 42305.

1%
2%
3%
4%
5%
6%
7%
8%
9%
10%
11%
12%
13%
14%
15%
16%
17%
18%
19%
20%
21%
22%
23%
24%
25%
26%
27%
28%
29%
30%
31%
32%
33%
34%
35%
36%
37%
38%
39%
40%
41%
42%
43%
44%
45%
46%
47%
48%
49%
50%
51%
52%
53%
54%
55%
56%
56%
57%
57%
58%
58%
59%
59%
60%
60%
61%
61%
62%
63%
64%
65%
66%
67%
68%
69%
70%
71%
72%
73%
74%
75%
76%
77%
78%
79%
80%
81%
82%
83%
84%
85%
86%
87%
88%
89%
90%
91%
92%
93%
94%
95%
96%
97%
98%
99%
Pruned 335169 variants from chromosome 3, leaving 47815.

1%
2%
2%
3%
4%
5%
6%
7%
8%
9%
10%
10%
11%
12%
13%
13%
14%
15%
16%
17%
18%
19%
20%
21%
21%
22%
23%
24%
25%
26%
27%
28%
29%
30%
31%
32%
32%
33%
34%
35%
36%
37%
38%
39%
40%
41%
42%
43%
43%
44%
45%
46%
47%
48%
49%
50%
51%
52%
53%
54%
54%
55%
56%
57%
58%
59%
60%
61%
62%
63%
64%
65%
65%
66%
67%
68%
69%
70%
71%
72%
73%
74%
75%
76%
76%
77%
78%
79%
80%
81%
82%
83%
84%
85%
86%
87%
87%
88%
89%
90%
91%
92%
93%
94%
95%
96%
97%
98%
98%
99%
Pruned 280320 variants from chromosome 4, leaving 40044.

1%
2%
3%
4%
5%
6%
6%
7%
8%
9%
10%
11%
12%
13%
13%
14%
15%
16%
17%
18%
19%
20%
21%
22%
22%
23%
24%
25%
26%
27%
28%
29%
29%
30%
31%
32%
33%
34%
35%
36%
36%
37%
38%
39%
40%
41%
42%
43%
44%
45%
45%
46%
47%
48%
49%
50%
51%
52%
52%
53%
54%
55%
56%
57%
58%
59%
60%
61%
62%
63%
64%
65%
66%
67%
68%
68%
69%
70%
71%
72%
73%
74%
75%
75%
76%
77%
78%
79%
80%
81%
82%
83%
84%
85%
86%
87%
88%
89%
90%
91%
91%
92%
93%
94%
95%
96%
97%
98%
98%
99%
Pruned 200786 variants from chromosome 5, leaving 34649.

1%
2%
3%
4%
5%
6%
7%
8%
9%
10%
11%
12%
13%
14%
14%
15%
16%
16%
17%
18%
19%
20%
21%
22%
23%
24%
25%
26%
27%
28%
29%
30%
31%
31%
32%
33%
33%
34%
35%
36%
37%
38%
39%
40%
41%
42%
43%
44%
45%
46%
47%
48%
48%
49%
50%
50%
51%
52%
53%
54%
55%
56%
57%
58%
59%
60%
61%
62%
63%
64%
65%
65%
66%
67%
67%
68%
69%
70%
71%
72%
73%
74%
75%
76%
77%
78%
79%
80%
81%
82%
82%
83%
84%
84%
85%
86%
87%
88%
89%
90%
91%
92%
93%
94%
95%
96%
97%
98%
99%
Pruned 2384 variants from chromosome 6, leaving 87.
Pruning complete.  1398983 of 1609590 variants removed.
Writing...
Marker lists written to plink.prune.in and plink.prune.out .</code></pre>
<pre class="r"><code>file.rename(&quot;data/derived/plink.prune.in&quot;, &quot;data/derived/SNPs_in_LD&quot;)</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>unlink(&quot;gwas_data/derived/plink.prune.out&quot;)</code></pre>
<!-- ## Run Bayesian Sparse Linear Mixed Models -->
<!-- Here is a screenshot from the relevant part of the GEMMA manual, explaining how the BSLMMs work, and the key parameters that are estimated: -->
<!-- !["Screenshot from the GEMMA manual"](data/input/GEMMA_manual_screenshot.png) -->
<!-- I run these with a burn-in period of 50,000 iterations, then 10m sampling iterations, recording every 1000^{th} iteration. This may seem extreme, but I picked it because there was massive auto-correlation in the MCMC chain with the default settings. -->
<!-- Glossary of abbreviations and parameters used by GEMMA: -->
<!--  - PVE: "Proportion of (phenotypic) variance explained", by the SNPs and the random effect terms together -->
<!--  - PGE: "Proportion of _genetic_ variance explained by the sparse effects terms".  -->
<!--  Useful quote from Zhou: I think you can treat both h and rho as parameters and only rely on only PVE and PGE. The result essentially says that a total 3414 SNPs explain approximately 33% of phenotypic variance (i.e. PVE=0.33). Among these 3414 SNPs, a small proportion of them (\approx 158) have relatively large effects and explain a total of 48% PVE (i.e. PGE=0.48). Although these 158 SNPs have relatively larger effects than the rest of the SNPs, their absolute effect sizes are still small. All these quantities provide a description of the genetic architecture of the phenotype, and give much more information than simply categorizing a trait into polygenic vs oligogenic.  -->
<!-- ### Female early-life fitness -->
<!-- ```{r bslmm1} -->
<!-- run_command("gemma -bfile trimmed_DGRP -maf 0.05 -n 1 -bslmm 1 -w 50000 -s 10000000 -rpace 1000 -seed 1 -o female_early_bslmm") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- make_trimmed_BED <- function(phenotypes){ -->
<!--   # Define a function to add phenotype data to a .fam file, for later GWAS analysis -->
<!--   # 'phenotypes' needs to have a column called 'line' -->
<!--   add_phenotypes_to_fam <- function(filepath, phenotypes){ -->
<!--     read.delim(filepath, header = FALSE, stringsAsFactors = FALSE) %>%  -->
<!--       select(V1, V2, V3, V4, V5) %>% # Get all the non-phenotype columns -->
<!--       left_join(phenotypes,  -->
<!--                 by = c("V1" = "line")) %>% -->
<!--       write.table(file = filepath,  -->
<!--                   col.names = FALSE, row.names = FALSE,  -->
<!--                   quote = FALSE, sep = "\t") -->
<!--   } -->
<!-- #  if(file.exists("data/derived/trimmed_DGRP.bed")) return(NULL) -->
<!--   # Load the BED/BIM/FAM files as a bigsnp object -->
<!--   bigsnp <- snp_readBed("data/input/dgrp2.bed", backingfile = tempfile()) -->
<!--   bigsnp_attached <- snp_attach(bigsnp) -->
<!--   ###################################################### -->
<!--   # 1. Perform simple SNP quality control using PLINK -->
<!--   ###################################################### -->
<!--   bedfile <- "data/input/dgrp2.bed" -->
<!--   plink <- download_plink() -->
<!--   bigsnp <- snp_plinkQC(plink.path = plink, -->
<!--                         prefix.in = "data/input/dgrp2", -->
<!--                         prefix.out = tempfile(), -->
<!--                         maf = 0.05, # MAF >= 0.05 -->
<!--                         geno = 0.05,   # acceptable number missing genotypes per locus: none -->
<!--                         mind = 0.1,   # acceptable proportion of missing genotypes per individual (i.e. DGRP line): none -->
<!--                         hwe = 0,    # do not filter by HWE test p-value (p = 0 is kept) -->
<!--                         autosome.only = FALSE) # Keep X chromosome -->
<!--   bigsnp <- snp_readBed(bigsnp) -->
<!--   bigsnp_attached <- snp_attach(bigsnp) -->
<!--   ###### Important messages from this QC: ######## -->
<!--   # Total genotyping rate is 0.956183. -->
<!--   # 1050867 variants removed due to missing genotype data (--geno). -->
<!--   # 2266579 variants removed due to minor allele threshold(s) -->
<!--   # (--maf/--max-maf/--mac/--max-mac). -->
<!--   # 1120981 variants and 193 people pass filters and QC. -->
<!--   # cull the PLINK files to just the DGRP lines that we measured -->
<!--   focal.lines <- phenotypes$line %>% as.character() -->
<!--   bigsnp_attached <- subset(bigsnp_attached,  -->
<!--                             ind.row = bigsnp_attached$fam$family.ID %in% focal.lines) %>% -->
<!--     snp_attach() -->
<!--   beagle <- download_beagle() -->
<!--   # Delete these files, because snp_writeBed() throws an error rather than over-writing -->
<!--   unlink(c("data/derived/dgrp2_after_QC.bed",  -->
<!--            "data/derived/dgrp2_after_QC.bim", "data/derived/dgrp2_after_QC.fam"))  -->
<!--   # Write the QC'd BED files for all 205 DGRP lines to disk -->
<!--   # This is is needed to use PLINK on them, and also because we now have a SNP set that is not pruned for LD -->
<!--   snp_writeBed(bigsnp_attached, "data/derived/dgrp2_after_QC.bed") -->
<!--   # Add our phenotype data to this new .fam file -->
<!--   add_phenotypes_to_fam("data/derived/dgrp2_after_QC.fam",  -->
<!--                         phenotypes %>% select(-block)) -->
<!--   ################################################################## -->
<!--   # 2. Remove SNPs that do not have MAF < 0.05 within our 115 lines -->
<!--   ################################################################## -->
<!--   bigsnp <- snp_plinkQC(plink.path = plink, -->
<!--                         prefix.in = "data/derived/dgrp2_after_QC", -->
<!--                         prefix.out = tempfile(), -->
<!--                         maf = 0.05, -->
<!--                         geno = 0, -->
<!--                         mind = 1, -->
<!--                         hwe = 0, -->
<!--                         autosome.only = FALSE) -->
<!--   bigsnp_attached <- snp_readBed(bigsnp) %>% snp_attach() -->
<!--   # unlink(temp.file) -->
<!--   ##### PLINK messages: ###### -->
<!--   # 1637 variants removed due to minor allele threshold(s) -->
<!--   # (--maf/--max-maf/--mac/--max-mac). -->
<!--   # 12954 variants and 115 people pass filters and QC. -->
<!--   ################################################################## -->
<!--   # 3. Perform short-range SNP clumping, using snp_clumping function -->
<!--   ################################################################## -->
<!--   # Sum the significant correlations of each SNP to each other SNP -->
<!--   summed_correlations <- apply(snp_cor(bigsnp_attached$genotypes), 1, sum) %>% -->
<!--     round(2) -->
<!--   # Identify 1 SNP from each 'clump' of highly linked SNPs -->
<!--   # Nearby SNPs that are correlated with a squared-corr of at least .1, i.e. r = +/- 0.316, are counted as clumped -->
<!--   snp_clumps <- snp_clumping(G = bigsnp_attached$genotypes,  -->
<!--                              infos.chr = bigsnp_attached$map$chromosome,  -->
<!--                              thr.r2 = 0.2, -->
<!--                              size = 1000, # 1kB windows -->
<!--                              is.size.in.bp = TRUE, -->
<!--                              infos.pos = bigsnp_attached$map$physical.pos)  -->
<!--   # After this there are 7523 SNPs left -->
<!--   clumped_subset <- subset(bigsnp_attached, ind.col = snp_clumps) %>% snp_attach() -->
<!--   ###################################################### -->
<!--   # 4. Perform long-range LD pruning using snp_autoSVD -->
<!--   ###################################################### -->
<!--   SVD_output <- snp_autoSVD(G = clumped_subset$genotypes, -->
<!--                             infos.chr = clumped_subset$map$chromosome, -->
<!--                             infos.pos = clumped_subset$map$physical.pos, -->
<!--                             thr.r2 = 0.2) -->
<!--   # Vector of SNP IDs to keep -->
<!--   SVD_output <- attributes(SVD_output)$subset -->
<!--   # After this there are 6897 SNPs left -->
<!--   # Retain all the SNPs that passed both types of clumping -->
<!--   to.keep <- snp_clumps[SVD_output] -->
<!--   clumped_subset2 <- subset(bigsnp_attached, ind.col = to.keep) %>% snp_attach() -->
<!--   summed_correlations <- data.frame(id = clumped_subset2$map$marker.ID, -->
<!--                                     summed_corr = summed_correlations[to.keep]) -->
<!--   # Delete these files, because snp_writeBed() throws an error rather than over-writing -->
<!--   unlink(c("data/derived/trimmed_DGRP.bed",  -->
<!--            "data/derived/trimmed_DGRP.bim", "data/derived/trimmed_DGRP.fam"))  -->
<!--   # Save the fully-pruned SNP data, and the SNP "tagging" data -->
<!--   snp_writeBed(clumped_subset2, "data/derived/trimmed_DGRP.bed") -->
<!--   write.csv(summed_correlations, "data/derived/summed_interSNP_correlations.csv", row.names = FALSE) -->
<!--   # Lastly, add our phenotype data to the new .fam file -->
<!--   add_phenotypes_to_fam("data/derived/trimmed_DGRP.fam",  -->
<!--                         phenotypes %>% select(-block)) -->
<!-- } -->
<!-- make_trimmed_BED(phenotypes = predicted_line_means) -->
<!-- if(file.exists('data/derived/trimmed_DGRP.bk')) unlink('data/derived/trimmed_DGRP.bk') -->
<!-- dgrp_bed <- snp_readBed("data/derived/trimmed_DGRP.bed") %>% snp_attach() -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] future.apply_1.10.0 future_1.30.0       purrr_1.0.1        
 [4] pander_0.6.5        readr_2.1.4         bigsnpr_1.11.6     
 [7] bigstatsr_1.5.12    glue_1.6.2          ggplot2_3.4.1      
[10] dplyr_1.1.0         workflowr_1.7.0.4  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10        lattice_0.20-45    listenv_0.9.0      getPass_0.2-2     
 [5] ps_1.7.2           rprojroot_2.0.3    digest_0.6.31      foreach_1.5.2     
 [9] utf8_1.2.2         parallelly_1.34.0  R6_2.5.1           bigsparser_0.6.1  
[13] evaluate_0.20      httr_1.4.5         pillar_1.8.1       flock_0.7         
[17] rlang_1.0.6        rstudioapi_0.14    data.table_1.14.6  whisker_0.4.1     
[21] callr_3.7.3        jquerylib_0.1.4    Matrix_1.5-1       rmarkdown_2.20    
[25] bigparallelr_0.3.2 stringr_1.5.0      bit_4.0.5          munsell_0.5.0     
[29] compiler_4.2.2     httpuv_1.6.9       xfun_0.37          pkgconfig_2.0.3   
[33] globals_0.16.2     htmltools_0.5.4    tidyselect_1.2.0   tibble_3.1.8      
[37] codetools_0.2-18   fansi_1.0.4        crayon_1.5.2       tzdb_0.3.0        
[41] withr_2.5.0        later_1.3.0        grid_4.2.2         jsonlite_1.8.4    
[45] gtable_0.3.1       lifecycle_1.0.3    git2r_0.31.0       magrittr_2.0.3    
[49] scales_1.2.1       vroom_1.6.1        cli_3.6.0          stringi_1.7.12    
[53] cachem_1.0.7       fs_1.6.1           promises_1.2.0.1   doRNG_1.8.6       
[57] doParallel_1.0.17  bslib_0.4.2        ellipsis_0.3.2     generics_0.1.3    
[61] vctrs_0.5.2        cowplot_1.1.1      iterators_1.0.14   tools_4.2.2       
[65] bit64_4.0.5        hms_1.1.2          rngtools_1.5.2     processx_3.8.0    
[69] parallel_4.2.2     fastmap_1.1.1      yaml_2.3.7         colorspace_2.1-0  
[73] bigassertr_0.1.6   knitr_1.42         sass_0.4.5        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
