<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>GWAS for fitness in Drosophila</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">GWAS for fitness in Drosophila</h1>

</div>


<p><strong>Last updated:</strong> 2018-09-14</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180914)</code> </summary></p>
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> 449a929 </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/derived/.DS_Store
    Ignored:    data/derived/output/.DS_Store
    Ignored:    data/input/.DS_Store
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  code/Drosophila_GWAS.Rmd
    Untracked:  data/derived/output/male_late_lm,.assoc.txt
    Untracked:  data/derived/output/male_late_lm,.log.txt
    Untracked:  data/derived/trimmed_DGRP.bk

Unstaged changes:
    Modified:   analysis/get_predicted_line_means.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/perform_gwas.Rmd
    Modified:   analysis/plot_line_means.Rmd
    Modified:   data/derived/output/DGRP_GRM.log.txt
    Modified:   data/derived/output/female_early_bslmm.bv.txt
    Modified:   data/derived/output/female_early_bslmm.gamma.txt
    Modified:   data/derived/output/female_early_bslmm.hyp.txt
    Modified:   data/derived/output/female_early_bslmm.log.txt
    Modified:   data/derived/output/female_early_bslmm.param.txt
    Modified:   data/derived/output/female_early_female_late.log.txt
    Modified:   data/derived/output/female_early_lm.log.txt
    Modified:   data/derived/output/female_early_lmm.log.txt
    Modified:   data/derived/output/female_early_male_early.log.txt
    Modified:   data/derived/output/female_late_lm.log.txt
    Modified:   data/derived/output/female_late_lmm.log.txt
    Modified:   data/derived/output/female_late_male_late.log.txt
    Modified:   data/derived/output/male_early_lm.log.txt
    Modified:   data/derived/output/male_early_lmm.log.txt
    Modified:   data/derived/output/male_early_male_late.log.txt
    Modified:   data/derived/output/male_late_lmm.log.txt
    Modified:   figures/figure1.eps
    Modified:   figures/figure2.eps

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
449a929
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
e981b22
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
<td style="text-align:left;">
Made GWAS page
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
90a30e9
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
<td style="text-align:left;">
Made figures Rmd
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
90a30e9
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
<td style="text-align:left;">
Made figures Rmd
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<pre class="r"><code>library(ggplot2)
library(ggExtra)
library(dplyr)
library(tidyr)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(showtext) # For fancy Google font in figures
font_add_google(name = &quot;Raleway&quot;, family = &quot;Raleway&quot;, regular.wt = 400, bold.wt = 700) # Install font from Google Fonts

# Load the predicted line means, as calculated by get_predicted_line_means
predicted_line_means &lt;- read.csv(&quot;data/derived/predicted_line_means.csv&quot;, stringsAsFactors = FALSE)</code></pre>
<div id="variance-and-covariance-in-line-mean-phenotypes" class="section level2">
<h2>Variance and covariance in line mean phenotypes</h2>
<p>Generally there is positive covariance between line means for different traits, and all 4 measures of fitness exhibit considerable phenotypic variance across lines.</p>
<pre class="r"><code>lims &lt;- c(1.1*min(apply(predicted_line_means[,2:5], 2, min)), 
          1.1*max(apply(predicted_line_means[,2:5], 2, max)))

fix.title &lt;- function(x){
  x[x == &quot;female.fitness.early&quot; | x == &quot;femalefitnessearly&quot;] &lt;- &quot;Female early-life fitness&quot;
  x[x == &quot;male.fitness.early&quot; | x == &quot;malefitnessearly&quot;] &lt;- &quot;Male early-life fitness&quot;
  x[x == &quot;female.fitness.late&quot; | x == &quot;femalefitnesslate&quot;] &lt;- &quot;Female late-life fitness&quot;
  x[x == &quot;male.fitness.late&quot; | x == &quot;malefitnesslate&quot;] &lt;- &quot;Male late-life fitness&quot;
  x
}

make_figure_1 &lt;- function(){

  nice.plot &lt;- function(df, v1, v2){
    
    formula &lt;- as.formula(paste(v2, &quot;~&quot;, v1))
    model &lt;- summary(lm(formula, data = df))
    r2 &lt;- format(model$r.squared %&gt;% round(2), nsmall = 2)
    slope &lt;- format(model$coefficients[2,1] %&gt;% round(2), nsmall = 2)
    se &lt;- format(model$coefficients[2,2] %&gt;% round(2), nsmall = 2)
    # print(model$coefficients[2,4]) # print p-value of the correlation
    text1 &lt;- paste(&quot;R^2 == &quot;, r2, sep = &quot;&quot;)
    text2 &lt;- paste(&quot;β = &quot;, slope, &quot; \u00B1 &quot;, se, sep = &quot;&quot;)
    
    pp &lt;- df %&gt;% 
      ggplot(aes_string(x = v1, y = v2)) + 
      annotate(&quot;text&quot;, x=min(lims)+0.1, y=max(lims), label = text1, parse = TRUE, hjust= 0) +
      annotate(&quot;text&quot;, x=min(lims)+0.1, y=max(lims)-0.4, label = text2, hjust= 0) +
      geom_point(alpha = 0.7) + 
      stat_smooth(method = &quot;lm&quot;, level = 0, colour = &quot;tomato&quot;) + 
      xlab(fix.title(v1)) + ylab(fix.title(v2)) + 
      theme_classic() + 
      theme(text = element_text(family = &quot;Raleway&quot;)) +
      scale_x_continuous(limits = lims) + 
      scale_y_continuous(limits = lims)
    
    if(v1 == &quot;male.fitness.early&quot; &amp; v2 == &quot;female.fitness.early&quot;) cols &lt;- c(&quot;lightblue&quot;, &quot;pink&quot;)
    if(v1 == &quot;male.fitness.late&quot; &amp; v2 == &quot;female.fitness.late&quot;)  cols &lt;- c(&quot;steelblue&quot;, &quot;deeppink2&quot;)
    if(v1 == &quot;male.fitness.early&quot; &amp; v2 == &quot;male.fitness.late&quot;) cols &lt;- c(&quot;lightblue&quot;, &quot;steelblue&quot;)
    if(v1 == &quot;female.fitness.early&quot; &amp; v2 == &quot;female.fitness.late&quot;) cols &lt;- c(&quot;pink&quot;, &quot;deeppink2&quot;)
    
    ggExtra::ggMarginal(pp, type = &quot;histogram&quot;, bins = 15, xparams = list(fill = cols[1]), yparams = list(fill = cols[2]))
  }
  
  p1 &lt;- nice.plot(predicted_line_means, &quot;male.fitness.early&quot;, &quot;female.fitness.early&quot;)
  p2 &lt;- nice.plot(predicted_line_means, &quot;male.fitness.late&quot;, &quot;female.fitness.late&quot;)
  p3 &lt;- nice.plot(predicted_line_means, &quot;male.fitness.early&quot;, &quot;male.fitness.late&quot;)
  p4 &lt;- nice.plot(predicted_line_means, &quot;female.fitness.early&quot;, &quot;female.fitness.late&quot;)
  full_plot &lt;- grid.arrange(p1, p2, p3, p4)
  
  cairo_ps(&quot;figures/figure1.eps&quot;, height = 9, width = 9)
  full_plot
  invisible(dev.off())
}
make_figure_1()</code></pre>
<p><img src="figure/plot_line_means.Rmd/unnamed-chunk-2-1.png" width="864" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-2-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
90a30e9
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p><br></br> <strong>Figure 1</strong>: Correlations among estimated line means for the four fitness components. The line means were estimated from Bayesian mixed models that account for block effects, and the non-independence of our early- and late-life fitness measurements.</p>
</div>
<div id="interaction-plot-showing-trait-covariance-across-lines" class="section level2">
<h2>Interaction plot showing trait covariance across lines</h2>
<p>All possible types of lines were observed: some lines are uniformly bad, or uniformly good, or good in one sex or age class but bad in the other.</p>
<pre class="r"><code>make_figure_2 &lt;- function(){
  out.file &lt;- &quot;figures/figure2.eps&quot;
  
  interaction.plot &lt;- function(predicted_line_means, x1, x2, title, sex.or.age){
    if(sex.or.age == &quot;sex&quot;){
      x.labs &lt;-  c(&quot;Female&quot;, &quot;Male&quot;) 
      cols &lt;- c(&quot;red&quot;, &quot;darkgrey&quot;, &quot;blue&quot;)
    } else {
      x.labs &lt;- c(&quot;Young&quot;, &quot;Old&quot;)
      cols &lt;- c(&quot;green&quot;, &quot;darkgrey&quot;, &quot;purple&quot;)
    }
    
    df &lt;- predicted_line_means %&gt;% select_(x1, x2) 
    df$rank.x1 &lt;- rank(df[,1]) / max(abs(rank(df[,1])))
    df$rank.x2 &lt;- rank(df[,2]) / max(abs(rank(df[,2])))
    df %&gt;% mutate(slope = rank.x1 - rank.x2,
                  line = 1:length(rank.x1)) %&gt;%
      gather(key = sex_or_age, value = fitness, rank.x1, rank.x2) %&gt;% 
      mutate(fitness = fitness / max(fitness),
             title = title) %&gt;%  
      ggplot(aes(x = sex_or_age, y = fitness, group = line, colour = slope)) + 
      geom_line(size=0.8, alpha=.7) + 
      scale_color_gradient2(low = cols[1], mid = cols[2], high = cols[3]) + 
      scale_x_discrete(expand = c(0.1,0.1), labels = x.labs) + 
      theme_classic(15) + 
      theme(strip.background = element_blank(),
            strip.text = element_text(hjust =0.1, face = &quot;bold&quot;)) +
      xlab(NULL) + ylab(NULL) + 
      facet_wrap(~title) + 
      theme(legend.position = &quot;none&quot;)
  }
  
  cairo_ps(&quot;figures/figure2.eps&quot;, height = 9, width = 9)
  grid.arrange(
    interaction.plot(predicted_line_means, &quot;female.fitness.early&quot;, &quot;male.fitness.early&quot;, &quot;Early-life fitness&quot;, &quot;sex&quot;),
    interaction.plot(predicted_line_means, &quot;female.fitness.late&quot;, &quot;male.fitness.late&quot;, &quot;Late-life fitness&quot;, &quot;sex&quot;),
    interaction.plot(predicted_line_means, &quot;female.fitness.early&quot;, &quot;female.fitness.late&quot;, &quot;Females&quot;, &quot;age&quot;),
    interaction.plot(predicted_line_means, &quot;male.fitness.early&quot;, &quot;male.fitness.late&quot;, &quot;Males&quot;, &quot;age&quot;),
    ncol = 2, left = &quot;Fitness rank among the 115 lines (0 is worst, 1 is best)&quot;, bottom = &quot;Sex or age category&quot;
  )
  invisible(dev.off())
}
make_figure_2()</code></pre>
<p><br></br> <strong>Figure 2</strong>: The relative fitness ranks for each line, for four pairs of fitness traits. The y-axis was calculated by taking the adjusted line mean fitnesses, ranking them, and then dividing by the number of lines. The intensity and hue of the colour helps highlight genotypes that rank highly for one fitness component but not the other.</p>
</div>
<div id="relating-fitness-data-to-genotype" class="section level1">
<h1>Relating fitness data to genotype</h1>
<div id="check-for-effects-of-wolbachia-and-inversions" class="section level2">
<h2>Check for effects of wolbachia and inversions</h2>
<p>The Mackay lab have collected data on the Wolbachia infection status of each line, and recorded which chromosomal inversions each line carries. It is possible to statistically account for these variables when doing GWAS, but it makes the analysis more complex. Our fitness data are not correlated with the <em>Wolbachia</em> or inversion data, and in light of this we elected not to include them as predictors when writing our GWAS functions below. Additional analyses (not shown) suggest that the results are essentially identical if we run the models with these predictors included. However, including these parameters as random effects slows down the Bayesian GWAS a lot (and it already takes months of CPU time), so we elected to leave them out.</p>
<pre class="r"><code>inv.data &lt;- predicted_line_means %&gt;% 
  left_join(read.csv(&quot;data/input/inversion genotypes.csv&quot;, stringsAsFactors = FALSE) %&gt;% 
              mutate(line = paste(&quot;line_&quot;, line, sep = &quot;&quot;)), by = &quot;line&quot;) %&gt;% 
  left_join(read.csv(&quot;data/input/wolbachia.csv&quot;, stringsAsFactors = FALSE) %&gt;% 
              mutate(line = paste(&quot;line_&quot;, line, sep = &quot;&quot;)), by = &quot;line&quot;) %&gt;% select(-line, -block)
inv.data[inv.data == &quot;homozygous&quot;] &lt;- NA
inv.data[inv.data == &quot;ST&quot;] &lt;- 0
inv.data[inv.data == &quot;INV&quot;] &lt;- 1
inv.data[inv.data == &quot;n&quot;] &lt;- 0
inv.data[inv.data == &quot;y&quot;] &lt;- 1
for(i in 1:ncol(inv.data)) inv.data[,i] &lt;- inv.data[,i] %&gt;% as.numeric()
inversion.cors &lt;- cor(inv.data, use = &quot;pairwise.complete.obs&quot;)
heatmap(inversion.cors[!is.na(inversion.cors[,1]), !is.na(inversion.cors[,1])])</code></pre>
<p><img src="figure/plot_line_means.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
e981b22
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
</tr>
<tr>
<td style="text-align:left;">
90a30e9
</td>
<td style="text-align:left;">
Luke Holman
</td>
<td style="text-align:left;">
2018-09-14
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2.2     showtext_0.5-1     showtextdb_2.0    
 [4] sysfonts_0.7.2     RColorBrewer_1.1-2 gridExtra_2.3     
 [7] tidyr_0.8.1        dplyr_0.7.6        ggExtra_0.8       
[10] ggplot2_3.0.0     

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.18      later_0.7.3       compiler_3.5.1   
 [4] pillar_1.3.0      git2r_0.23.0      plyr_1.8.4       
 [7] workflowr_1.1.1   bindr_0.1.1       R.methodsS3_1.7.1
[10] R.utils_2.7.0     tools_3.5.1       digest_0.6.15    
[13] jsonlite_1.5      evaluate_0.11     tibble_1.4.2     
[16] gtable_0.2.0      pkgconfig_2.0.1   rlang_0.2.1      
[19] shiny_1.1.0       curl_3.2          yaml_2.2.0       
[22] withr_2.1.2       stringr_1.3.1     knitr_1.20       
[25] rprojroot_1.3-2   tidyselect_0.2.4  glue_1.3.0       
[28] R6_2.2.2          rmarkdown_1.10    purrr_0.2.5      
[31] magrittr_1.5      whisker_0.3-2     promises_1.0.1   
[34] backports_1.1.2   scales_1.0.0      htmltools_0.3.6  
[37] assertthat_0.2.0  xtable_1.8-2      mime_0.5         
[40] colorspace_1.3-2  httpuv_1.4.5      labeling_0.3     
[43] miniUI_0.1.1.1    stringi_1.2.4     lazyeval_0.2.1   
[46] munsell_0.5.0     crayon_1.3.4      R.oo_1.22.0      </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
