<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Plotting the estimated mean phenotype for each DGRP line</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Plotting the estimated mean phenotype for each DGRP line</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreeaf15dd63a476b2499f2be7f3002df3ec72c28522targetblankaf15dd6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/af15dd63a476b2499f2be7f3002df3ec72c28522" target="_blank">af15dd6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreeaf15dd63a476b2499f2be7f3002df3ec72c28522targetblankaf15dd6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/af15dd63a476b2499f2be7f3002df3ec72c28522" target="_blank">af15dd6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    figures/fig1_inkscape.svg
    Ignored:    figures/figure1a.pdf
    Ignored:    figures/figure1b.pdf

Untracked files:
    Untracked:  big_model.rds
    Untracked:  code/gcta64
    Untracked:  code/plink
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds
    Untracked:  figures/SNP_effect_ED.pdf
    Untracked:  figures/boyle_plot.pdf
    Untracked:  figures/fig1.pdf
    Untracked:  manuscript/GWAS_manuscript.log
    Untracked:  old_analyses/

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplots.pdf
    Deleted:    analysis/GO_KEGG_enrichment.Rmd
    Deleted:    analysis/plotting_results.Rmd
    Deleted:    code/GO_and_KEGG_gsea.R
    Deleted:    code/make_sites_files_for_ARGweaver.R
    Deleted:    code/plotting_results.Rmd
    Deleted:    code/run_argweaver.R
    Modified:   code/run_mashr.R
    Deleted:    data/input/clough_2014_dsx_targets.csv
    Modified:   figures/GWAS_stats_figure.pdf
    Modified:   figures/TWAS_stats_figure.pdf
    Modified:   figures/antagonism_ratios.pdf
    Modified:   figures/composite_mixture_figure.pdf
    Modified:   figures/eff_size_histos.pdf
    Deleted:    figures/figure1.eps
    Deleted:    figures/figure2.eps
    Modified:   manuscript/GWAS_manuscript.Rmd
    Modified:   manuscript/GWAS_manuscript.pdf
    Modified:   manuscript/GWAS_manuscript.tex

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plot_line_means.Rmd</code>) and HTML (<code>docs/plot_line_means.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/plot_line_means.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/871ae81fa56175675b53d22ca29bf9323c3ed50a/docs/plot_line_means.html" target="_blank">871ae81</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/a16e7516fb3553521a62f87d0a3e6228801d0017/analysis/plot_line_means.Rmd" target="_blank">a16e751</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/plot_line_means.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/plot_line_means.html" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(ggExtra)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(showtext) # For fancy Google font in figures
font_add_google(name = &quot;Raleway&quot;, family = &quot;Raleway&quot;, regular.wt = 400, bold.wt = 700) # Install font from Google Fonts
showtext::showtext.auto()

# Load the predicted line means, as calculated by get_predicted_line_means
predicted_line_means &lt;- read_csv(&quot;data/derived/predicted_line_means.csv&quot;)</code></pre>
<div id="variance-and-covariance-in-line-mean-phenotypes" class="section level2">
<h2>Variance and covariance in line mean phenotypes</h2>
<p>Generally there is positive covariance between line means for different traits, and all 4 measures of fitness exhibit considerable phenotypic variance across lines.</p>
<pre class="r"><code>lims &lt;- c(1.1*min(apply(predicted_line_means[,2:5], 2, min)), 
          1.1*max(apply(predicted_line_means[,2:5], 2, max)))

fix.title &lt;- function(x){
  x[x == &quot;female.fitness.early&quot; | x == &quot;femalefitnessearly&quot;] &lt;- &quot;Female early-life fitness&quot;
  x[x == &quot;male.fitness.early&quot; | x == &quot;malefitnessearly&quot;] &lt;- &quot;Male early-life fitness&quot;
  x[x == &quot;female.fitness.late&quot; | x == &quot;femalefitnesslate&quot;] &lt;- &quot;Female late-life fitness&quot;
  x[x == &quot;male.fitness.late&quot; | x == &quot;malefitnesslate&quot;] &lt;- &quot;Male late-life fitness&quot;
  x
}

make_figure_1 &lt;- function(){

  nice.plot &lt;- function(df, v1, v2){
    
    formula &lt;- as.formula(paste(v2, &quot;~&quot;, v1))
    model &lt;- summary(lm(formula, data = df))
    r2 &lt;- format(model$r.squared %&gt;% round(2), nsmall = 2)
    slope &lt;- format(model$coefficients[2,1] %&gt;% round(2), nsmall = 2)
    se &lt;- format(model$coefficients[2,2] %&gt;% round(2), nsmall = 2)
    # print(model$coefficients[2,4]) # print p-value of the correlation
    text1 &lt;- paste(&quot;R^2 == &quot;, r2, sep = &quot;&quot;)
    text2 &lt;- paste(&quot;β = &quot;, slope, &quot; \u00B1 &quot;, se, sep = &quot;&quot;)
    
    pp &lt;- df %&gt;% 
      ggplot(aes_string(x = v1, y = v2)) + 
      annotate(&quot;text&quot;, x=min(lims) + 0.1, y=max(lims), label = text1, parse = TRUE, hjust= 0) +
      annotate(&quot;text&quot;, x=min(lims) + 0.1, y=max(lims)-0.4, label = text2, hjust= 0) +
      geom_point(alpha = 0.7) + 
      stat_smooth(method = &quot;lm&quot;, level = 0, colour = &quot;tomato&quot;) + 
      xlab(fix.title(v1)) + ylab(fix.title(v2)) + 
      theme_classic() + 
      theme(text = element_text(family = &quot;Raleway&quot;)) +
      scale_x_continuous(limits = lims) + 
      scale_y_continuous(limits = lims)
    
    if(v1 == &quot;male.fitness.early&quot; &amp; v2 == &quot;female.fitness.early&quot;) cols &lt;- c(&quot;lightblue&quot;, &quot;pink&quot;)
    if(v1 == &quot;male.fitness.late&quot; &amp; v2 == &quot;female.fitness.late&quot;)  cols &lt;- c(&quot;steelblue&quot;, &quot;deeppink2&quot;)
    if(v1 == &quot;male.fitness.early&quot; &amp; v2 == &quot;male.fitness.late&quot;) cols &lt;- c(&quot;lightblue&quot;, &quot;steelblue&quot;)
    if(v1 == &quot;female.fitness.early&quot; &amp; v2 == &quot;female.fitness.late&quot;) cols &lt;- c(&quot;pink&quot;, &quot;deeppink2&quot;)
    
    ggExtra::ggMarginal(pp, type = &quot;histogram&quot;, bins = 15, xparams = list(fill = cols[1]), yparams = list(fill = cols[2]))
  }
  
  p1 &lt;- nice.plot(predicted_line_means, &quot;male.fitness.early&quot;, &quot;female.fitness.early&quot;)
  p2 &lt;- nice.plot(predicted_line_means, &quot;male.fitness.late&quot;, &quot;female.fitness.late&quot;)
  p3 &lt;- nice.plot(predicted_line_means, &quot;male.fitness.early&quot;, &quot;male.fitness.late&quot;)
  p4 &lt;- nice.plot(predicted_line_means, &quot;female.fitness.early&quot;, &quot;female.fitness.late&quot;)
  full_plot &lt;- grid.arrange(p1, p2, p3, p4)
  
  # Saving does not work (ggmarginal + showtext issue?), so get it from the HTML output
  # svg(&quot;figures/figure1.svg&quot;, height = 9, width = 9)
  # full_plot
  # invisible(dev.off())
}
make_figure_1()</code></pre>
<p><img src="figure/plot_line_means.Rmd/unnamed-chunk-2-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/871ae81fa56175675b53d22ca29bf9323c3ed50a/docs/figure/plot_line_means.Rmd/unnamed-chunk-2-1.png" target="_blank">871ae81</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/docs/figure/plot_line_means.Rmd/unnamed-chunk-2-1.png" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><br></br> <strong>Figure 1</strong>: Correlations among estimated line means for the four fitness components. The line means were estimated from Bayesian mixed models that account for block effects, and the non-independence of our early- and late-life fitness measurements.</p>
</div>
<div id="interaction-plot-showing-trait-covariance-across-lines" class="section level2">
<h2>Interaction plot showing trait covariance across lines</h2>
<p>All possible types of lines were observed: some lines are uniformly bad, or uniformly good, or good in one sex or age class but bad in the other.</p>
<pre class="r"><code>make_figure_2 &lt;- function(){

  interaction.plot &lt;- function(predicted_line_means, x1, x2, title, sex.or.age){
    if(sex.or.age == &quot;sex&quot;){
      x.labs &lt;-  c(&quot;Female&quot;, &quot;Male&quot;) 
      cols &lt;- c(&quot;steelblue&quot;, &quot;darkgrey&quot;, &quot;deeppink2&quot;)
    } else {
      x.labs &lt;- c(&quot;Early life&quot;, &quot;Late life&quot;)
      cols &lt;- c(&quot;green&quot;, &quot;darkgrey&quot;, &quot;purple&quot;)
    }
    
    df &lt;- predicted_line_means %&gt;% select(!!x1, !!x2) 
    df$rank.x1 &lt;- rank(df[,1]) / max(abs(rank(df[,1])))
    df$rank.x2 &lt;- rank(df[,2]) / max(abs(rank(df[,2])))
    df %&gt;% mutate(slope = rank.x1 - rank.x2,
                  line = 1:length(rank.x1)) %&gt;%
      gather(key = sex_or_age, value = fitness, rank.x1, rank.x2) %&gt;% 
      mutate(fitness = fitness / max(fitness),
             title = title) %&gt;%  
      ggplot(aes(x = sex_or_age, y = fitness, group = line, colour = slope)) + 
      geom_line(size = 0.4, alpha = 0.7) + 
      scale_color_gradient2(low = cols[1], mid = cols[2], high = cols[3]) + 
      scale_x_discrete(expand = c(0.1,0.1), labels = x.labs) + 
      theme_classic(15) + 
      theme(strip.background = element_blank(),
            text = element_text(family = &quot;Raleway&quot;)) +
      xlab(NULL) + ylab(NULL) + 
      facet_wrap(~title) + 
      theme(legend.position = &quot;none&quot;)
  }
  
  full_plot &lt;- grid.arrange(
    interaction.plot(predicted_line_means, &quot;female.fitness.early&quot;, &quot;male.fitness.early&quot;, &quot;Early-life fitness&quot;, &quot;sex&quot;),
    interaction.plot(predicted_line_means, &quot;female.fitness.late&quot;, &quot;male.fitness.late&quot;, &quot;Late-life fitness&quot;, &quot;sex&quot;),
    interaction.plot(predicted_line_means, &quot;female.fitness.early&quot;, &quot;female.fitness.late&quot;, &quot;Females&quot;, &quot;age&quot;),
    interaction.plot(predicted_line_means, &quot;male.fitness.early&quot;, &quot;male.fitness.late&quot;, &quot;Males&quot;, &quot;age&quot;),
    ncol = 2, left = &quot;Fitness rank among the 125 lines&quot;, bottom = &quot;Sex or age category&quot;
  )
}
make_figure_2()</code></pre>
<pre><code>Warning: attributes are not identical across measure variables;
they will be dropped

Warning: attributes are not identical across measure variables;
they will be dropped

Warning: attributes are not identical across measure variables;
they will be dropped

Warning: attributes are not identical across measure variables;
they will be dropped</code></pre>
<p><img src="figure/plot_line_means.Rmd/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;" /> <br></br> <strong>Figure 2</strong>: The relative fitness ranks for each line, for four pairs of fitness traits. The y-axis was calculated by taking the adjusted line mean fitnesses, ranking them, and then dividing by the number of lines. The intensity and hue of the colour helps highlight genotypes that rank highly for one fitness component but not the other.</p>
<!-- # Relating fitness data to genotype -->
<!-- ## Check for effects of wolbachia and inversions -->
<!-- The Mackay lab have collected data on the Wolbachia infection status of each line, and recorded which chromosomal inversions each line carries. It is possible to statistically account for these variables when doing GWAS, but it makes the analysis more complex. Our fitness data are not correlated with the _Wolbachia_ or inversion data, and in light of this we elected not to include them as predictors when writing our GWAS functions below. Additional analyses (not shown) suggest that the results are essentially identical if we run the models with these predictors included. However, including these parameters as random effects slows down the Bayesian GWAS a lot (and it already takes months of CPU time), so we elected to leave them out. -->
<!-- ```{r message=FALSE, warning=FALSE} -->
<!-- inv.data <- predicted_line_means %>%  -->
<!--   left_join(read.csv("data/input/inversion genotypes.csv", stringsAsFactors = FALSE) %>%  -->
<!--               mutate(line = paste("line_", line, sep = "")), by = "line") %>%  -->
<!--   left_join(read.csv("data/input/wolbachia.csv", stringsAsFactors = FALSE) %>%  -->
<!--               mutate(line = paste("line_", line, sep = "")), by = "line") %>% select(-line, -block) -->
<!-- inv.data[inv.data == "homozygous"] <- NA -->
<!-- inv.data[inv.data == "ST"] <- 0 -->
<!-- inv.data[inv.data == "INV"] <- 1 -->
<!-- inv.data[inv.data == "n"] <- 0 -->
<!-- inv.data[inv.data == "y"] <- 1 -->
<!-- for(i in 1:ncol(inv.data)) inv.data[,i] <- inv.data[,i] %>% as.numeric() -->
<!-- inversion.cors <- cor(inv.data, use = "pairwise.complete.obs") -->
<!-- heatmap(inversion.cors[!is.na(inversion.cors[,1]), !is.na(inversion.cors[,1])]) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] showtext_0.9-2     showtextdb_3.0     sysfonts_0.8.3     RColorBrewer_1.1-2
 [5] gridExtra_2.3      ggExtra_0.9        forcats_0.5.0      stringr_1.4.0     
 [9] dplyr_1.0.0        purrr_0.3.4        readr_2.0.0        tidyr_1.1.0       
[13] tibble_3.0.1       ggplot2_3.3.2      tidyverse_1.3.0    workflowr_1.6.2   

loaded via a namespace (and not attached):
 [1] nlme_3.1-149     fs_1.4.1         lubridate_1.7.10 bit64_0.9-7     
 [5] httr_1.4.1       rprojroot_1.3-2  tools_4.0.3      backports_1.1.7 
 [9] R6_2.4.1         DBI_1.1.0        mgcv_1.8-33      colorspace_1.4-1
[13] withr_2.2.0      tidyselect_1.1.0 bit_1.1-15.2     curl_4.3        
[17] compiler_4.0.3   git2r_0.27.1     cli_2.0.2        rvest_0.3.5     
[21] xml2_1.3.2       labeling_0.3     scales_1.1.1     digest_0.6.25   
[25] rmarkdown_2.5    pkgconfig_2.0.3  htmltools_0.5.0  dbplyr_1.4.4    
[29] fastmap_1.0.1    highr_0.8        rlang_0.4.6      readxl_1.3.1    
[33] rstudioapi_0.11  shiny_1.4.0.2    generics_0.0.2   farver_2.0.3    
[37] jsonlite_1.7.0   vroom_1.5.3      magrittr_2.0.1   Matrix_1.2-18   
[41] Rcpp_1.0.4.6     munsell_0.5.0    fansi_0.4.1      lifecycle_0.2.0 
[45] stringi_1.5.3    whisker_0.4      yaml_2.2.1       blob_1.2.1      
[49] parallel_4.0.3   promises_1.1.0   crayon_1.3.4     miniUI_0.1.1.1  
[53] lattice_0.20-41  haven_2.3.1      splines_4.0.3    hms_0.5.3       
[57] knitr_1.32       pillar_1.4.4     reprex_0.3.0     glue_1.4.2      
[61] evaluate_0.14    modelr_0.1.8     vctrs_0.3.0      tzdb_0.1.2      
[65] httpuv_1.5.3.1   cellranger_1.1.0 gtable_0.3.0     assertthat_0.2.1
[69] xfun_0.22        mime_0.9         xtable_1.8-4     broom_0.5.6     
[73] later_1.0.0      ellipsis_0.3.1  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
