<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Plots and models of variant effect sizes</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Plots and models of variant effect
sizes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreede68e73b42e2d94da04fa2ca7252370c35b4872ctargetblankde68e73a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/fitnessGWAS/tree/de68e73b42e2d94da04fa2ca7252370c35b4872c" target="_blank">de68e73</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreede68e73b42e2d94da04fa2ca7252370c35b4872ctargetblankde68e73a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/fitnessGWAS/tree/de68e73b42e2d94da04fa2ca7252370c35b4872c" target="_blank">de68e73</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    old_analyses/.DS_Store

Untracked files:
    Untracked:  old_analyses/Data for old analyses/
    Untracked:  old_analyses/eQTL_analysis.Rmd
    Untracked:  old_analyses/fitness_data.csv
    Untracked:  old_analyses/gcta_quant_genetics_OLD.Rmd
    Untracked:  old_analyses/quantitative_genetics_OLD_brms.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/plot_models_variant_effects.Rmd</code>) and HTML
(<code>docs/plot_models_variant_effects.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/358afb47ecdf8a5f5f2f16cb2d94fc30a90d5aff/analysis/plot_models_variant_effects.Rmd" target="_blank">358afb4</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Tidying up and small edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/plot_models_variant_effects.html" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/9f76cf2487968f4212262d3de838f7bfc71ffde7/analysis/plot_models_variant_effects.Rmd" target="_blank">9f76cf2</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/plot_models_variant_effects.html" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c2119179fd068f579259ae22a2f4024a2595ec3b/analysis/plot_models_variant_effects.Rmd" target="_blank">c211917</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/d94e22dc2fbcacfe3430866ac5f7116866a93ec3/analysis/plot_models_variant_effects.Rmd" target="_blank">d94e22d</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-14
</td>
<td>
New quant gen and other changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/23341aac6b1256e8cda4b82bb7141797ef9576f1/docs/plot_models_variant_effects.html" target="_blank">23341aa</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/0f059040133feab4e23c5fc96ebf8e2f554aa4a8/analysis/plot_models_variant_effects.Rmd" target="_blank">0f05904</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/559337114060bc72179743800c02f969ff177660/docs/plot_models_variant_effects.html" target="_blank">5593371</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/70671b40e78088467dc1578d0514a00ee6f4fc48/analysis/plot_models_variant_effects.Rmd" target="_blank">70671b4</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/45e139cecb4ad369f4d31ff28f1ec19a3fb391f0/docs/plot_models_variant_effects.html" target="_blank">45e139c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/5cf7b5e0c3c867464a52080b33321eee78a82be8/analysis/plot_models_variant_effects.Rmd" target="_blank">5cf7b5e</a>
</td>
<td>
lukeholman
</td>
<td>
2022-07-29
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/490266c958359428cc4b979291695f7e55e15b71/docs/plot_models_variant_effects.html" target="_blank">490266c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/ffbefdd67333798c444ca43781dcd651a1318842/analysis/plot_models_variant_effects.Rmd" target="_blank">ffbefdd</a>
</td>
<td>
lukeholman
</td>
<td>
2022-03-10
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/plot_models_variant_effects.html" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/17a62a70f2650bc75ac9272083bbd22fb6acb667/analysis/plot_models_variant_effects.Rmd" target="_blank">17a62a7</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/plot_models_variant_effects.html" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/86444b2dbf34d4aaae69120242409639b78bef20/analysis/plot_models_variant_effects.Rmd" target="_blank">86444b2</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/58de93a3e57d893dde7350fa504c5b1b35061ac8/docs/plot_models_variant_effects.html" target="_blank">58de93a</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/d573064c5223b276f7abcd63aa97f01e6f19643a/analysis/plot_models_variant_effects.Rmd" target="_blank">d573064</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/652106368c95ac4a04b2383d560d3c8073990f18/docs/plot_models_variant_effects.html" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/9deec2d2a0b6fabe844d63c9d77c62841b5cc81e/analysis/plot_models_variant_effects.Rmd" target="_blank">9deec2d</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/5a59d9f982db9a933d0cea448822677a000087c9/docs/plot_models_variant_effects.html" target="_blank">5a59d9f</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/de681496ef6dbd7c9cde9f258aba20fedaecf37f/analysis/plot_models_variant_effects.Rmd" target="_blank">de68149</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
<td>
wflow_publish("analysis/plot_models_variant_effects.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/plot_models_variant_effects.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/4953d582430e0be9ac5f56212ce2e1ee0478d936/docs/plot_models_variant_effects.html" target="_blank">4953d58</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/2bf87505a56e12debcab79eb42fd1313bc1b2f31/analysis/plot_models_variant_effects.Rmd" target="_blank">2bf8750</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/plot_models_variant_effects.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/plot_models_variant_effects.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/871ae81fa56175675b53d22ca29bf9323c3ed50a/docs/plot_models_variant_effects.html" target="_blank">871ae81</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/plot_models_variant_effects.html" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c606d3d8748d48cc6dd479febb9ce71effc8bcfd/analysis/plot_models_variant_effects.Rmd" target="_blank">c606d3d</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(gridExtra)
library(qqman)
library(ggbeeswarm)
library(Hmisc)
library(showtext) # For fancy Google font in figures
library(mashr)
library(kableExtra)
library(cowplot)
library(grid)
library(RColorBrewer)
library(car)
library(brms)
library(tidybayes)
library(bigsnpr)
library(glue)
library(patchwork)
library(ggpubr)
library(staplr) 

# Note: you may need to re-download plink to get this to run on non-Mac systems
# I used bigsnpr::download_plink()
plink &lt;- paste(getwd(), &quot;code/plink&quot;, sep = &quot;/&quot;)

font_add_google(name = &quot;Raleway&quot;, family = &quot;Raleway&quot;, regular.wt = 400, bold.wt = 700) # Install font from Google Fonts
showtext_auto()

db &lt;- DBI::dbConnect(RSQLite::SQLite(), 
                     &quot;data/derived/annotations.sqlite3&quot;)

# Results for all 1,613,615 SNPs, even those that are in 100% LD with others (these are grouped up by the SNP_clump column)
all_snps &lt;- tbl(db, &quot;univariate_lmm_results&quot;)

# All SNPs and SNP groups that are in &lt;100% LD with one another (n = 1,207,357)
SNP_clumps &lt;- all_snps %&gt;% select(-SNP) %&gt;% distinct() %&gt;% collect(n = Inf)

# Subsetting variable to get the approx-LD subset of SNPs
LD_subset &lt;- !is.na(SNP_clumps$LFSR_female_early_mashr_ED)

# Load the predicted line means, as calculated by get_predicted_line_means
predicted_line_means &lt;- read_csv(&quot;data/derived/predicted_line_means.csv&quot;)

# Load and clean the effect sizes from GWAS and join with SNP annotations
univariate_lmm_results &lt;- tbl(db, &quot;univariate_lmm_results&quot;) %&gt;% 
  select(-contains(&quot;canonical&quot;), 
         -contains(&quot;raw&quot;)) %&gt;% 
  inner_join(tbl(db, &quot;variants&quot;) %&gt;% 
               select(SNP, FBID, site.class, distance.to.gene, MAF), 
             by = &quot;SNP&quot;) %&gt;%
  left_join(
    tbl(db, &quot;genes&quot;) %&gt;% 
      select(FBID, gene_name), by = &quot;FBID&quot;) %&gt;%
  collect(n = Inf) %&gt;%
  rename_all(~ gsub(&quot;beta_&quot;, &quot;&quot;, .x)) %&gt;%
  rename_all(~ gsub(&quot;_mashr_ED&quot;, &quot;&quot;, .x))

univariate_lmm_results &lt;- univariate_lmm_results %&gt;%
  mutate(site.class = gsub(&quot;5_&quot;, &quot;5-&quot;, site.class),
         site.class = gsub(&quot;3_&quot;, &quot;3-&quot;, site.class),
         site.class = gsub(&quot;NON_&quot;, &quot;NON-&quot;, site.class),
         site.class = gsub(&quot;_&quot;, &quot; &quot;, site.class),
         site.class = capitalize(tolower(site.class)),
         site.class = gsub(&quot;Utr&quot;, &quot;UTR&quot;, site.class)
  )</code></pre>
<!-- ```{r} -->
<!-- # testing ruzicka comparison -->
<!-- xx <- read_tsv("~/Downloads/Assoc_Data.txt") -->
<!-- xx %>% arrange(-MAF_DGRP) %>% head() %>% as.data.frame() -->
<!-- ruz_data <- xx %>% select(ruzicka_SNP_name = Predictor, chr = Chromosome, position = Basepair, ruzicka_MAF = MAF, ruzicka_effect_size = Effect, ruzicka_wald_stat = Wald_Stat, ruzicka_wald_p = Wald_P) -->
<!-- merged <- all_snps %>% select(SNP, SNP_clump, contains("_mashr_ED")) %>% -->
<!--   filter(!is.na(beta_female_early_mashr_ED)) %>% -->
<!--   # collect(n=1000) %>% -->
<!--   collect(n=Inf) %>% -->
<!--   distinct() %>% -->
<!--   mutate(position = str_split(SNP, "_"), -->
<!--          chr = map_chr(position, ~ .x[1]), -->
<!--          position = as.numeric(map_chr(position, ~ .x[2]))) %>% -->
<!--   left_join(ruz_data, by = c("position", "chr")) -->
<!-- merged <- merged[complete.cases(merged), ] -->
<!-- merged <- left_join(merged, -->
<!--           antagonism_evidence_ratios_gwas %>% arrange(inter_sex_early) %>% mutate(inter_sex_early = log2(inter_sex_early))) -->
<!-- ggplot(merged, aes(x = ruzicka_wald_p, y = inter_sex_early)) + -->
<!--   geom_point() -->
<!-- ruzicka_SA <- xx %>% arrange(-MAF_DGRP) %>% filter(fdr<0.3) %>% select(chr = Chromosome, position = Basepair) -->
<!-- ruz_hits <- merged %>% filter(paste(chr, position) %in% paste(ruzicka_SA$chr, ruzicka_SA$position)) %>% pull(inter_sex_early) -->
<!-- nonruz_hits <- merged %>% filter(!(paste(chr, position) %in% paste(ruzicka_SA$chr, ruzicka_SA$position))) %>% pull(inter_sex_early) -->
<!-- wilcox.test(nonruz_hits, ruz_hits) -->
<!-- median(nonruz_hits) -->
<!-- median(ruz_hits) -->
<!-- hist(-1 * (merged %>% filter(inter_sex_early <= quantile(inter_sex_early, 0.99)) %>% pull(ruzicka_effect_size) %>% log10)) -->
<!-- bind_rows(merged %>% filter(inter_sex_early <= quantile(inter_sex_early, 0.001)) %>% mutate(type ="holmanSA"), -->
<!--           merged %>% filter(inter_sex_early >= quantile(inter_sex_early, 0.999)) %>% mutate(type ="holmanSC")) %>% -->
<!--   ggplot(aes(type, ruzicka_effect_size)) + -->
<!--   geom_boxplot(outlier.shape = NA) + geom_beeswarm(alpha=0.3, aes(size = -1*log10(ruzicka_wald_p))) -->
<!-- merged %>% filter(inter_sex_early <= quantile(inter_sex_early, 0.01) & -->
<!--                     ruzicka_wald_stat < 0 & -->
<!--                     ruzicka_wald_p < quantile(ruzicka_wald_p, 0.01)) %>% -->
<!--   pull(SNP_clump) -->
<!-- tbl(db, "variants") %>% filter(SNP=="3L_4306295_SNP") -->
<!-- tbl(db, "variants") %>% filter(SNP=="3L_4346362_SNP") -->
<!-- merged %>% filter(SNP=="3L_4346362_SNP") %>% as.data.frame() -->
<!-- xx<- select(merged, SNP, ruzicka_MAF) %>% left_join(tbl(db, "variants") %>% select(SNP, MAF) %>%  filter(SNP %in% !!merged$SNP) %>% collect(n=Inf)) -->
<!-- cor(xx$ruzicka_MAF, xx$MAF, use = "pairwise.complete.obs") -->
<!-- ``` -->
<div id="q-q-plots" class="section level2 tabset">
<h2 class="tabset">Q-Q plots</h2>
<p>Here are some quantile-quantile plots, which are commonly used to
check GWAS results, and to test the hypothesis that there are more SNPs
than expected showing large effects on the trait of interest. There is a
modest excess of loci with effects on female fitness, but not much of a
visible excess for males. These plots use the raw <span
class="math inline">\(p\)</span>-values results from GEMMA.</p>
<div id="female-early-life-fitness" class="section level3">
<h3>Female early-life fitness</h3>
<pre class="r"><code>univariate_lmm_pvals &lt;- SNP_clumps %&gt;%
  select(contains(&quot;pvalue&quot;)) %&gt;% filter(LD_subset) %&gt;% as.data.frame()
qqman::qq(univariate_lmm_pvals$pvalue_female_early_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot1-1">
Past versions of qqplot1-1.png
</button>
</p>
<div id="fig-qqplot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/qqplot1-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot1-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="female-late-life-fitness" class="section level3">
<h3>Female late-life fitness</h3>
<pre class="r"><code>qqman::qq(univariate_lmm_pvals$pvalue_female_late_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot2-1">
Past versions of qqplot2-1.png
</button>
</p>
<div id="fig-qqplot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/qqplot2-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot2-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="male-early-life-fitness" class="section level3">
<h3>Male early-life fitness</h3>
<pre class="r"><code>qqman::qq(univariate_lmm_pvals$pvalue_male_early_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot3-1">
Past versions of qqplot3-1.png
</button>
</p>
<div id="fig-qqplot3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/qqplot3-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot3-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="male-late-life-fitness" class="section level3">
<h3>Male late-life fitness</h3>
<pre class="r"><code>qqman::qq(univariate_lmm_pvals$pvalue_male_late_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot4-1">
Past versions of qqplot4-1.png
</button>
</p>
<div id="fig-qqplot4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/qqplot4-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot4-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="hex-bin-plots-and-correlations-in-variant-effect-sizes"
class="section level2">
<h2>Hex bin plots and correlations in variant effect sizes</h2>
<div
id="effect-sizes-adjusted-using-data-driven-adaptive-shrinkage-in-mashr"
class="section level3 tabset">
<h3 class="tabset">Effect sizes adjusted using ‘data-driven’ adaptive
shrinkage in <code>mashr</code></h3>
<div id="plot" class="section level4">
<h4>Plot</h4>
<pre class="r"><code>snp_effect_fig &lt;- bind_rows(
  SNP_clumps %&gt;%
    filter(LD_subset) %&gt;%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_early_mashr_ED, male = beta_male_early_mashr_ED) %&gt;% 
    mutate(age_class = &quot;A. Early-life fitness&quot;),
  SNP_clumps %&gt;%
    filter(LD_subset) %&gt;%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_late_mashr_ED, male = beta_male_late_mashr_ED) %&gt;% 
    mutate(age_class = &quot;B. Late-life fitness&quot;)) %&gt;% 
  ggplot(aes(female, male)) + 
  geom_vline(xintercept = 0, linetype = 3) +
  geom_hline(yintercept = 0, linetype = 3) +
  stat_binhex(bins = 50)  +
  geom_density_2d(colour = &quot;white&quot;, alpha = 0.6) + 
  scale_fill_viridis_c() + 
  facet_wrap(~ age_class) +
  theme_bw() + xlab(&quot;Effect on female fitness&quot;) + ylab(&quot;Effect on male fitness&quot;) +
  theme(legend.position = &quot;none&quot;,
        strip.background = element_blank(),
        strip.text = element_text(hjust=0)) + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12))

fig2_top &lt;- snp_effect_fig

snp_effect_fig</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/hex1-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hex1-1">
Past versions of hex1-1.png
</button>
</p>
<div id="fig-hex1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/490266c958359428cc4b979291695f7e55e15b71/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">490266c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Top part of Figure 2</strong>: <code>mashr</code>-adjusted
effect sizes of 1207357 loci (i.e. groups of one or more polymorphic
sites in complete linkage disequilibrium) on male and female early- and
late-life fitness. The data have been binned into hexagons, with the
colour and contour lines indicating the number of loci. The diagonal
line represents <span class="math inline">\(y=x\)</span>. Positive
effect sizes indicate that the minor allele is associated with higher
fitness, while negative effects indicate that the major allele is
associated with higher fitness.</p>
</div>
<div id="pearson-correlation-matrix" class="section level4">
<h4>Pearson correlation matrix</h4>
<pre class="r"><code>SNP_clumps %&gt;%
  select(contains(&quot;mashr_ED&quot;)) %&gt;%
  select(contains(&quot;beta&quot;)) %&gt;%
  rename_all(~ paste(ifelse(str_detect(.x, &quot;female&quot;), &quot;Female&quot;, &quot;Male&quot;), 
                     ifelse(str_detect(.x, &quot;early&quot;), &quot;early&quot;, &quot;late&quot;))) %&gt;%
  cor(use = &quot;pairwise.complete.obs&quot;) %&gt;% 
  kable(digits = 3) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female early
</th>
<th style="text-align:right;">
Female late
</th>
<th style="text-align:right;">
Male early
</th>
<th style="text-align:right;">
Male late
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female early
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.998
</td>
<td style="text-align:right;">
0.911
</td>
<td style="text-align:right;">
0.950
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late
</td>
<td style="text-align:right;">
0.998
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.880
</td>
<td style="text-align:right;">
0.927
</td>
</tr>
<tr>
<td style="text-align:left;">
Male early
</td>
<td style="text-align:right;">
0.911
</td>
<td style="text-align:right;">
0.880
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.994
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late
</td>
<td style="text-align:right;">
0.950
</td>
<td style="text-align:right;">
0.927
</td>
<td style="text-align:right;">
0.994
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="unadjusted-effect-sizes" class="section level3 tabset">
<h3 class="tabset">Unadjusted effect sizes</h3>
<div id="plot-1" class="section level4">
<h4>Plot</h4>
<p>Uses the variant effect sizes output by GEMMA, without applying any
shrinkage (i.e. this is the input data that was adjusted using
<code>mashr</code>).</p>
<pre class="r"><code>bind_rows(
  SNP_clumps %&gt;%
    filter(LD_subset) %&gt;%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_early_raw, male = beta_male_early_raw) %&gt;% 
    mutate(age_class = &quot;A. Early-life fitness&quot;),
  SNP_clumps %&gt;%
    filter(LD_subset) %&gt;%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_late_raw, male = beta_male_late_raw) %&gt;% 
    mutate(age_class = &quot;B. Late-life fitness&quot;)) %&gt;% 
  ggplot(aes(female, male)) + 
  geom_vline(xintercept = 0, linetype = 3) +
  geom_hline(yintercept = 0, linetype = 3) +
  stat_binhex(bins = 50)  +
  geom_density_2d(colour = &quot;white&quot;, alpha = 0.6) + 
  scale_fill_viridis_c() + 
  facet_wrap(~ age_class) +
  theme_bw() + xlab(&quot;Effect on female fitness&quot;) + ylab(&quot;Effect on male fitness&quot;) +
  theme(legend.position = &quot;none&quot;,
        strip.background = element_blank(),
        strip.text = element_text(hjust=0)) + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/hex3-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hex3-1">
Past versions of hex3-1.png
</button>
</p>
<div id="fig-hex3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/490266c958359428cc4b979291695f7e55e15b71/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">490266c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pearson-correlation-matrix-1" class="section level4">
<h4>Pearson correlation matrix</h4>
<pre class="r"><code>SNP_clumps %&gt;%
  select(contains(&quot;raw&quot;)) %&gt;%
  select(contains(&quot;beta&quot;)) %&gt;%
  rename_all(~ paste(ifelse(str_detect(.x, &quot;female&quot;), &quot;Female&quot;, &quot;Male&quot;), 
                     ifelse(str_detect(.x, &quot;early&quot;), &quot;early&quot;, &quot;late&quot;))) %&gt;%
  cor(use = &quot;pairwise.complete.obs&quot;) %&gt;% 
  kable(digits = 3) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female early
</th>
<th style="text-align:right;">
Female late
</th>
<th style="text-align:right;">
Male early
</th>
<th style="text-align:right;">
Male late
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female early
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.567
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:right;">
0.118
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late
</td>
<td style="text-align:right;">
0.567
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
0.167
</td>
</tr>
<tr>
<td style="text-align:left;">
Male early
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.434
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:right;">
0.167
</td>
<td style="text-align:right;">
0.434
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="average-effect-sizes-are-negative"
class="section level2 tabset">
<h2 class="tabset">Average effect sizes are negative</h2>
<p>Each of the following four tests is an intercept-only linear model,
weighted by the inverse of the standard error for the focal variant’s
effect size (so, loci where the effect effect size was measured with
more precision are weighted more heavily). The tests are run on the
LD-pruned subset of SNPs, minimising pseudoreplication caused by
non-independence in the data. An intercept term less than zero indicates
that on average, the minor allele tends to be associated with lower
values of the focal fitness trait (compared to the major allele).</p>
<p>These results indicate that the minor allele tends to <em>reduce</em>
fitness, relative to the major allele. It’s a weak effect (note the
small value in the Estimate column), this may reflect the large
uncertainty with which the effect sizes are measured, in addition to a
true biological result that most loci have little or no relationship
with the fitness traits we measured.</p>
<div id="female-early-life" class="section level3">
<h3>Female early-life</h3>
<pre class="r"><code>mod1 &lt;- summary(lm(beta_female_early_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_female_early_raw))
mod1</code></pre>
<pre><code>
Call:
lm(formula = beta_female_early_raw ~ 1, data = SNP_clumps %&gt;% 
    filter(LD_subset), weights = 1/SE_female_early_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.98458 -0.22630  0.00919  0.23696  1.48781 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.0016820  0.0002555  -6.584 4.58e-11 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3497 on 208986 degrees of freedom</code></pre>
</div>
<div id="female-late-life" class="section level3">
<h3>Female late-life</h3>
<pre class="r"><code>mod2 &lt;- summary(lm(beta_female_late_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_female_late_raw))
mod2</code></pre>
<pre><code>
Call:
lm(formula = beta_female_late_raw ~ 1, data = SNP_clumps %&gt;% 
    filter(LD_subset), weights = 1/SE_female_late_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.94128 -0.23405  0.00409  0.23909  1.47335 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.0025016  0.0002598   -9.63   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3549 on 208986 degrees of freedom</code></pre>
</div>
<div id="male-early-life" class="section level3">
<h3>Male early-life</h3>
<pre class="r"><code>mod3 &lt;- summary(lm(beta_male_early_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_male_early_raw))
mod3</code></pre>
<pre><code>
Call:
lm(formula = beta_male_early_raw ~ 1, data = SNP_clumps %&gt;% filter(LD_subset), 
    weights = 1/SE_male_early_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.58868 -0.23295 -0.00219  0.23114  1.76956 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.0016140  0.0002542   -6.35 2.15e-10 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3503 on 208986 degrees of freedom</code></pre>
</div>
<div id="male-late-life" class="section level3">
<h3>Male late-life</h3>
<pre class="r"><code>mod4 &lt;- summary(lm(beta_male_late_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_male_late_raw))
mod4</code></pre>
<pre><code>
Call:
lm(formula = beta_male_late_raw ~ 1, data = SNP_clumps %&gt;% filter(LD_subset), 
    weights = 1/SE_male_late_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.89125 -0.22874  0.00114  0.23182  1.59205 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.001943   0.000251  -7.739    1e-14 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3479 on 208986 degrees of freedom</code></pre>
<pre class="r"><code>medians &lt;- SNP_clumps %&gt;% filter(LD_subset) %&gt;% 
  summarise(beta_female_early_raw = mean(beta_female_early_raw),
            beta_female_late_raw = mean(beta_female_late_raw),
            beta_male_early_raw = mean(beta_male_early_raw),
            beta_male_late_raw = mean(beta_male_late_raw)) %&gt;% 
  unlist() %&gt;% unname()

rbind(mod1$coefficients,
      mod2$coefficients,
      mod3$coefficients,
      mod4$coefficients) %&gt;% as_tibble() %&gt;% 
  mutate(`Fitness component` = 
           c(&quot;Female fitness early&quot;, &quot;Female fitness late&quot;, 
             &quot;Male fitness early&quot;, &quot;Male fitness late&quot;), 
         .before = names(.)[1]) %&gt;% 
  mutate(Estimate = paste(format(round(Estimate, 5), nsmall = 5), 
                          &quot; (&quot;, format(round(`Std. Error`, 5), nsmall = 5), &quot;)&quot;, sep = &quot;&quot;)) %&gt;% 
  mutate(`Median variant effect` = format(round(medians, 5), nsmall = 5), .after = &quot;Estimate&quot;) %&gt;% 
  rename(p = `Pr(&gt;|t|)`, `Mean (SE) variant effect` = Estimate) %&gt;% 
  mutate(p = formatC(p, format = &quot;e&quot;, digits = 2)) %&gt;% 
  select(-`Std. Error`) %&gt;% 
  saveRDS(&quot;data/derived/supp_table_of_variant_effect_means.rds&quot;)</code></pre>
</div>
</div>
<div id="manhattan-plot" class="section level2">
<h2>Manhattan plot</h2>
<p><strong>Figure legend</strong>: Manhattan plot showing the p-values
(-log_{10} transformed) for each variant, from mixed model GWAS of
female (top) and male (bottom) early-life fitness testing the null
hypothesis that the two alleles are associated with equal fitness
values.</p>
<pre class="r"><code>manhattan_data &lt;- tbl(db, &quot;univariate_lmm_results&quot;) %&gt;% 
  select(SNP, starts_with(&quot;P_&quot;),
         pvalue_female_early_raw, pvalue_male_early_raw) %&gt;%
  distinct() %&gt;%
  collect(n=Inf) %&gt;%
  mutate(position = str_split(SNP, &quot;_&quot;),
         chr = map_chr(position, ~ .x[1]),
         position = as.numeric(map_chr(position, ~ .x[2]))) %&gt;% 
  filter(chr != &quot;4&quot;) 

max_pos &lt;- manhattan_data %&gt;%
  group_by(chr) %&gt;%
  summarise(max_pos = max(position), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()
max_pos$max_pos &lt;- c(0, cumsum(max_pos$max_pos[1:4]))

manhattan_data &lt;- manhattan_data %&gt;%
    left_join(max_pos, by = &quot;chr&quot;) %&gt;%
    mutate(position = position + max_pos) 

# left_axis_label &lt;- expression(paste(&quot;Effect on female early-life fitness (-&quot;, Log[10], &quot; p)&quot;))


p1 &lt;- manhattan_data %&gt;%
  ggplot(aes(position, -1 * log10(pvalue_female_early_raw), 
             group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  scale_colour_brewer(palette = &quot;Paired&quot;, name = &quot;Chromosome&quot;) + 
  scale_y_continuous(limits = c(0,8)) +
  ylab(&quot;&quot;) +  xlab(&quot;&quot;) +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        text = element_text(family = &quot;Raleway&quot;, size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

p2 &lt;- manhattan_data %&gt;%
  ggplot(aes(position, -1 * log10(pvalue_male_early_raw), 
             group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  scale_colour_brewer(palette = &quot;Paired&quot;, name = &quot;Chromosome&quot;) + 
  xlab(&quot;Position&quot;) + ylab(&quot;&quot;) +
  scale_y_reverse(limits = c(8,0)) +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        text = element_text(family = &quot;Raleway&quot;, size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

grid_arrange_shared_legend &lt;- function(..., ncol = length(list(...)), nrow = 1) {

  plots &lt;- list(...)
  g &lt;- ggplotGrob(plots[[1]] + theme(legend.position = &quot;right&quot;))$grobs
  legend &lt;- g[[which(sapply(g, function(x) x$name) == &quot;guide-box&quot;)]]
  lheight &lt;- sum(legend$height)
  lwidth &lt;- sum(legend$width)
  gl &lt;- lapply(plots, function(x) x + theme(legend.position = &quot;none&quot;))
  gl &lt;- c(gl, ncol = ncol, nrow = nrow)
  
  left_label &lt;- text_grob(expression(Effect~on~early-life~fitness~(-Log[10]~p)), 
                          size = 12, family = &quot;Raleway&quot;, rot = 90)

  combined &lt;- arrangeGrob(
    arrangeGrob(gl[[1]], gl[[2]], nrow = 2, 
                left = left_label),
    legend,
    ncol = 2,
    widths = unit.c(unit(1, &quot;npc&quot;) - lwidth, lwidth))
  return(combined)
}
grid_arrange_shared_legend(p1, p2)</code></pre>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name              grob
1 1 (1-1,1-1) arrange   gtable[arrange]
2 2 (1-1,2-2) arrange gtable[guide-box]</code></pre>
</div>
<div id="investigating-pleiotropy-and-polygenicity-of-fitness"
class="section level2">
<h2>Investigating pleiotropy and polygenicity of fitness</h2>
<p>Inspired by Boyle et al. 2017 (<em>Cell</em>, specifically the
analysis in their Figure 1C), we sorted all the variants in order of
effect size on female fitness, placed the variants in bins of 1000, and
then calculated the average effect size for each bin for both male and
female fitness. This analysis was performed on the raw SNP effect sizes
from GEMMA, pruned to a set of 208987 SNPs in approximate LD with one
another.</p>
<p>If there is pleiotropy between male and female fitness, we predict a
correlation between the effect sizes for male and female fitness; on the
contrary if there were no pleiotropy, we would see no correlation.
Moreover, if fitness is highly polygenic (‘omnigenic’; Boyle et
al. 2017), we predict a tight, straight line relationship, because each
bin would contain some variants with small but genuine associations with
fitness, and these effects would replicate in the other fitness trait.
On the contrary, if genetic variance in fitness stems from just a few
genes with larger effects (with most loci having no true effect on
fitness), the relationship between male and female fitness would be flat
in the center and sloped at the extremes.</p>
<p>Figure 3 shows that there was a very tight correlation between the
average effects of the variants in each bin on male and female fitness.
The data therefore suggest that variants associated with male fitness
tend to also affect female fitness (in the same direction), and that a
very large number of loci have small, concordant effects on fitness in
both sexes.</p>
<p>Interestingly there appears to be a curve in Figure 3A to the left of
the x-axis, indicating that variants where the minor allele is strongly,
negatively associated with female fitness are (on average) less
negatively associated with male fitness than expected based on
predictions from variants with weaker effects on female fitness. One
possible explanation is that alleles that are highly highly detrimental
to both sexes are usually purged by selection (or at least kept below
the 5% MAF threshold that we used), whereas female-harming alleles that
are neutral or beneficial in males are purged less often, resulting a
greater proportion of female-limited or sexually antagonistic alleles
towards the left of the x-axis.</p>
<pre class="r"><code>p1_data &lt;- SNP_clumps %&gt;%
  filter(LD_subset) %&gt;% # The figure looks the same whether or not the data are thinned to the LD set of SNPs
  arrange(beta_female_early_raw) %&gt;%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         facet = &quot;A. Early-life fitness&quot;) %&gt;%
  group_by(bin, facet) %&gt;%
  summarise(females = mean(beta_female_early_raw), males = mean(beta_male_early_raw)) 

p2_data &lt;- SNP_clumps %&gt;%
  filter(LD_subset) %&gt;% 
  arrange(beta_female_late_raw) %&gt;%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         facet = &quot;B. Late-life fitness&quot;) %&gt;%
  group_by(bin, facet) %&gt;%
  summarise(females = mean(beta_female_late_raw), males = mean(beta_male_late_raw))

boyle_plot &lt;- bind_rows(p1_data, p2_data) %&gt;% 
  ggplot(aes(females, males)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  geom_point(alpha = 0.8) +
  stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 0.6) +
  facet_wrap(~ facet) +
  xlab(&quot;Mean effect size on female fitness&quot;) + ylab(&quot;Mean effect size on male fitness&quot;) + 
  theme_bw() +
  theme(strip.background = element_blank(),
        strip.text = element_text(hjust=0)) + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12))

boyle_plot</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-9-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-9-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="correlations-between-mutation-load-and-fitness"
class="section level2">
<h2>Correlations between mutation load and fitness</h2>
<div id="count-all-the-mutations-in-each-dgrp-line"
class="section level3">
<h3>Count all the mutations in each DGRP line</h3>
<p>The following code counts the number of mutations per DGRP line which
A) have a minor allele frequency of 0.05 or lower, and B) were
classified by SnpEff as having an effect that we designated to be a
“major effect”, defined as an insertion, deletion, or nonsynonymous
substitution inside a coding sequence (as well as gains and losses of
start codons). We reasoned that such alleles are probably mostly
deleterious. The following mutation types were classified as major
effect mutations (see <code>SnpEff</code>’s classification scheme <a
href="https://pcingola.github.io/SnpEff/se_outputsummary/">here</a>):</p>
<ul>
<li>EXON_DELETED</li>
<li>NON_SYNONYMOUS_CODING</li>
<li>FRAME_SHIFT</li>
<li>CODON_CHANGE</li>
<li>CODON_INSERTION</li>
<li>CODON_CHANGE_PLUS_CODON_INSERTION</li>
<li>CODON_DELETION</li>
<li>CODON_CHANGE_PLUS_CODON_DELETION</li>
<li>STOP_GAINED</li>
<li>STOP_LOST</li>
<li>RARE_AMINO_ACID</li>
<li>START_LOST</li>
<li>START_GAINED</li>
</ul>
<pre class="r"><code># Load up the DGRP genotype data using the bigsnpr package
if(!file.exists(&quot;snp_backing_file.bk&quot;)) {
  rds &lt;- snp_readBed(&quot;data/input/dgrp2.bed&quot;, backingfile = &quot;snp_backing_file&quot;)
}
bed_file &lt;- snp_attach(&quot;snp_backing_file.rds&quot;)
annot &lt;- read.table(&quot;data/input/dgrp.fb557.annot.txt&quot;, 
                    header = FALSE, stringsAsFactors = FALSE)

# Use PLINK to count allele freqs in the whole DGRP
plink &lt;- paste(getwd(), &quot;code/plink&quot;, sep = &quot;/&quot;)
run_command &lt;- function(shell_command, wd = getwd(), path = &quot;&quot;){
  cat(system(glue(&quot;cd &quot;, wd, path, &quot;\n&quot;,shell_command), intern = TRUE), sep = &#39;\n&#39;)
}
run_command(&quot;{plink} --bfile dgrp2 --freq&quot;, path = &quot;/data/input&quot;)</code></pre>
<pre><code>PLINK v1.90b6.24 64-bit (6 Jun 2021)           www.cog-genomics.org/plink/1.9/
(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to plink.log.
Options in effect:
  --bfile dgrp2
  --freq

32768 MB RAM detected; reserving 16384 MB for main workspace.
4438427 variants loaded from .bim file.
205 people (0 males, 205 females) loaded from .fam.
205 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 205 founders and 0 nonfounders present.
Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
Total genotyping rate is 0.956183.
--freq: Allele frequencies (founders only) written to plink.frq .</code></pre>
<pre class="r"><code># Get a list of variants that constitute a &#39;major mutation&#39; (types as shown in code below)

major_effect_types &lt;- c(
  &quot;EXON_DELETED&quot;, &quot;NON_SYNONYMOUS_CODING&quot;,
  &quot;FRAME_SHIFT&quot;, &quot;CODON_CHANGE&quot;,
  &quot;CODON_INSERTION&quot;,
  &quot;CODON_CHANGE_PLUS_CODON_INSERTION&quot;,
  &quot;CODON_DELETION&quot;,
  &quot;CODON_CHANGE_PLUS_CODON_DELETION&quot;,
  &quot;STOP_GAINED&quot;,
  &quot;STOP_LOST&quot;,
  &quot;RARE_AMINO_ACID&quot;,
  &quot;START_LOST&quot;,
  &quot;START_GAINED&quot;
)

all_major_mutations &lt;- annot$V1[unique(c(
  which(str_detect(annot$V3, major_effect_types[1])),
  which(str_detect(annot$V3, major_effect_types[2])),
  which(str_detect(annot$V3, major_effect_types[3])),
  which(str_detect(annot$V3, major_effect_types[4])),
  which(str_detect(annot$V3, major_effect_types[5])),
  which(str_detect(annot$V3, major_effect_types[6])),
  which(str_detect(annot$V3, major_effect_types[7])),
  which(str_detect(annot$V3, major_effect_types[8])),
  which(str_detect(annot$V3, major_effect_types[9])),
  which(str_detect(annot$V3, major_effect_types[10])),
  which(str_detect(annot$V3, major_effect_types[11])),
  which(str_detect(annot$V3, major_effect_types[12])),
  which(str_detect(annot$V3, major_effect_types[13]))
))]

# Get a list of variants with 0 &gt; MAF &lt; 0.05
rare_alleles &lt;- read.table(&quot;data/input/plink.frq&quot;, header = T) %&gt;% 
  filter(MAF &lt; 0.05 &amp; MAF &gt; 0) %&gt;% pull(SNP)

# Get the indexes of variants that are major mutations and also have MAF &lt; 0.05
indexes &lt;- intersect(
  which(bed_file$map$marker.ID %in% all_major_mutations), 
  which(bed_file$map$marker.ID %in% rare_alleles)
)

# Function to count the number of mutations in all 205 DGRP lines
count_mutations &lt;- function(indexes){
  tibble(
    line = bed_file$fam$family.ID,
    mutation_load = sapply(
      1:205, function(i) sum(bed_file$genotypes[i, ][indexes] == 2, na.rm = T))
  )
}

mutation_load &lt;- count_mutations(indexes)

# Join the mutation counts with our line mean fitness data
joined &lt;- left_join(
  predicted_line_means %&gt;% 
    select(-block) %&gt;% 
    gather(fitness_trait, fitness_value, -line),
  mutation_load, by = join_by(line)) %&gt;% 
  mutate(fitness_trait = str_to_sentence(str_replace_all(fitness_trait, &quot;[.]&quot;, &quot; &quot;)))</code></pre>
</div>
<div id="run-bayesian-multivariate-models" class="section level3">
<h3>Run Bayesian multivariate models</h3>
<p>The following Bayesian multivariate model estiamtes the relationship
between mutation count and each of the four fitness traits, across DGRP
lines. The model has the multivariate formula
<code>(FE, FL, ME, ML) ~ mutation_count</code>, assumes Gaussian errors,
and estimates the residual correlation between the four fitness traits
(thereby accounting for the correlation between fitness traits when
estimating the relationship with mutation count). The model uses
<code>brms</code> default priors.</p>
<pre class="r"><code>brms_data_muts &lt;- joined %&gt;% 
  mutate(fitness_trait = str_replace_all(fitness_trait, &quot; &quot;, &quot;_&quot;)) %&gt;% 
  spread(fitness_trait, fitness_value) %&gt;% 
  mutate(mutation_count = mutation_load / 100) %&gt;% select(-mutation_load)

mut_model &lt;- brm(bf(
  mvbind(Female_fitness_early, 
         Female_fitness_late, 
         Male_fitness_early, 
         Male_fitness_late) ~ 
    mutation_count, sigma ~ 0) + set_rescor(TRUE), 
  data = brms_data_muts,
  warmup = 2000, iter = 8000, chains = 4, cores = 4) </code></pre>
<pre><code>Running /Library/Frameworks/R.framework/Resources/bin/R CMD SHLIB foo.c
clang -mmacosx-version-min=10.13 -I&quot;/Library/Frameworks/R.framework/Resources/include&quot; -DNDEBUG   -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/Rcpp/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/unsupported&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/BH/include&quot; -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/src/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppParallel/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/rstan/include&quot; -DEIGEN_NO_DEBUG  -DBOOST_DISABLE_ASSERTS  -DBOOST_PENDING_INTEGER_LOG2_HPP  -DSTAN_THREADS  -DBOOST_NO_AUTO_PTR  -include &#39;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp&#39;  -D_REENTRANT -DRCPP_PARALLEL_USE_TBB=1   -I/usr/local/include   -fPIC  -Wall -g -O2  -c foo.c -o foo.o
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Dense:1:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Core:88:
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:1: error: unknown type name &#39;namespace&#39;
namespace Eigen {
^
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:16: error: expected &#39;;&#39; after top level declarator
namespace Eigen {
               ^
               ;
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Dense:1:
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Core:96:10: fatal error: &#39;complex&#39; file not found
#include &lt;complex&gt;
         ^~~~~~~~~
3 errors generated.
make: *** [foo.o] Error 1</code></pre>
<div id="model-checks-posterior-predictive-plots"
class="section level4">
<h4>Model checks (posterior predictive plots)</h4>
<p>Here, we plot the distribution of the model’s posterior predictions,
for each of the four response variables, for 50 samples from the
posterior. If the model’s assumptions and prior are suitable, there
should be a good match between the posterior predictions (shown in blue;
one line per sample from the posterior) and the original data (shown in
black). See <code>?pp_check</code>.</p>
<pre class="r"><code>do_ppcheck &lt;- function(rr, title){
    pp_check(mut_model, resp = rr, ndraws = 50) + 
    scale_y_continuous(limits = c(0, 0.55)) +
    labs(subtitle=title) + 
    theme_bw() + 
    theme(legend.position = &quot;none&quot;) 
}

grid.arrange(
  do_ppcheck(&quot;Femalefitnessearly&quot;, &quot;Female fitness early&quot;),
  do_ppcheck(&quot;Femalefitnesslate&quot;, &quot;Female fitness late&quot;),
  do_ppcheck(&quot;Malefitnessearly&quot;, &quot;Male fitness early&quot;),
  do_ppcheck(&quot;Malefitnesslate&quot;, &quot;Male fitness late&quot;))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/5a59d9f982db9a933d0cea448822677a000087c9/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">5a59d9f</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="plot-mutation-load-against-line-mean-fitness"
class="section level3">
<h3>Plot mutation load against line mean fitness</h3>
<p>The plot shows the correlation, across DGRP lines of fitness with the
mutation load. The top row shows the effect of ‘major effect’ mutations
on each of the four fitness traits, while the bottom row shows the
effect of mutations with a minor allele frequency (MAF) below 5%. The
number of mutations of the specified type (in 100s) is shown on the
x-axis, and the regression line comes from the previously mentioned
Bayesian multivariate model (which accounts for covariance between the
four fitness traits when estimating the effects of mutation load).</p>
<p>The figure illustrates that there is large variation in the number of
putatively deleterious mutations between lines, and there is a negative
correlation with fitness (especially in females).</p>
<pre class="r"><code>mut_plot_data &lt;- conditional_effects(mut_model, plot = F)

get_regression_line_data &lt;- function(plot_data){
  regression_line_rare_muts &lt;- bind_rows(
    plot_data[[1]] %&gt;% 
      select(mutation_count, estimate__, lower__, upper__) %&gt;% 
      mutate(trait = &quot;A. Female early-life fitness&quot;),
    plot_data[[2]] %&gt;% 
      select(mutation_count, estimate__, lower__, upper__) %&gt;% 
      mutate(trait = &quot;B. Female late-life fitness&quot;),
    plot_data[[3]] %&gt;% 
      select(mutation_count, estimate__, lower__, upper__) %&gt;% 
      mutate(trait = &quot;C. Male early-life fitness&quot;),
    plot_data[[4]] %&gt;% 
      select(mutation_count, estimate__, lower__, upper__) %&gt;% 
      mutate(trait = &quot;D. Male late-life fitness&quot;)) %&gt;% 
    mutate(mut_type = &quot;Load of mutations with MAF &lt; 0.05&quot;) %&gt;% 
    as_tibble()
}

get_points_data &lt;- function(brms_data){
  brms_data %&gt;% select(-line) %&gt;% 
  rename(`A. Female early-life fitness` = Female_fitness_early,
         `B. Female late-life fitness` = Female_fitness_late,
         `C. Male early-life fitness` = Male_fitness_early,
         `D. Male late-life fitness` = Male_fitness_late) %&gt;% 
  gather(trait, estimate__, -mutation_count)
}


mut_load_correlation_plot &lt;- get_regression_line_data(mut_plot_data) %&gt;% 
  ggplot(aes(x = mutation_count, y = estimate__)) + 
  geom_ribbon(aes(ymin = lower__, ymax = upper__, linetype = NA, fill = trait)) +
  geom_line(lwd = 0.5) +
  geom_point(data = get_points_data(brms_data_muts), alpha = 0.7, size = 0.7) +
  labs(x = &quot;Mutation load (100s)&quot;, 
       y = &quot;Line mean fitness&quot;) + 
  facet_wrap(~trait, nrow = 2) +
  scale_fill_manual(values = c(&quot;pink&quot;, &quot;deeppink2&quot;, &quot;lightblue&quot;, &quot;steelblue&quot;)) + 
  theme_bw() + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 11)) +
  theme(plot.title = element_text(hjust = 0), 
        strip.text = element_text(hjust = 0),
        strip.background = element_blank(),
        legend.position = &quot;none&quot;)

mut_load_correlation_plot</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="estimated-effect-size-of-mutation-load-on-fitness"
class="section level3 tabset">
<h3 class="tabset">Estimated effect size of mutation load on
fitness</h3>
<div id="plot-2" class="section level4">
<h4>Plot</h4>
<p>The left plot shows the posterior estimates of the effect size of
mutation load on fitness (i.e. the ‘regression slope’ from the previous
figure). The coloured area shows the posterior distribution, and the
horizontal bar summarises its median and 66% and 95% credible intervals.
The right plot shows the difference in this effect size between fitness
traits, e.g. FE - FL shows the difference in regression slope between
female early-life and late-life fitness. A positive sign indicates that
mutation load has more negative effect on the second trait
(e.g. mutations appear more costly to older females than younger).</p>
<pre class="r"><code>posterior_muts &lt;- as_draws_df(mut_model, variable = &quot;^b_&quot;, regex = T) %&gt;% 
  as_tibble() %&gt;% select(contains(&quot;mutation&quot;))

make_plot1 &lt;- function(posterior, title){
  tibble(
    `Female\nearly-life\n(FE)` = unlist(posterior[,1]), 
    `Female\nlate-life\n(FL)` = unlist(posterior[,2]),  
    `Male\nearly-life\n(ME)` = unlist(posterior[,3]),  
    `Male\nlate-life\n(ML)` = unlist(posterior[,4])) %&gt;% 
    gather() %&gt;% 
    mutate(key = factor(key, rev(unique(key))),
           x = title) %&gt;% 
    ggplot(aes(value, key)) + 
    geom_vline(xintercept = 0, linetype = 2) +
    stat_halfeye(aes(fill = key)) + 
    facet_wrap(~ x) +
    labs(x = &quot;Change per 100 mutations&quot;, y = NULL) +
    scale_fill_manual(values = rev(c(&quot;pink&quot;, &quot;deeppink2&quot;, &quot;lightblue&quot;, &quot;steelblue&quot;))) + 
    theme_bw() + 
    theme(text = element_text(family = &quot;Raleway&quot;, size = 11)) +
    theme(plot.title = element_text(hjust = 0), 
          strip.text = element_text(hjust = 0),
          strip.background = element_blank(),
          panel.background = element_blank(),
          legend.position = &quot;none&quot;)
}

make_plot2 &lt;- function(posterior, title, col){
  tibble(
    `ME − FE` = unlist(posterior[,3] - posterior[,1]), 
    `ML − FE` = unlist(posterior[,4] - posterior[,1]), 
    `ME − FL` = unlist(posterior[,3] - posterior[,2]), 
    `ML − FL` = unlist(posterior[,4] - posterior[,2]), 
    `FE − FL` = unlist(posterior[,1] - posterior[,2]), 
    `ME − ML` = unlist(posterior[,3] - posterior[,4]) ) %&gt;% 
    gather() %&gt;% 
    mutate(key = factor(key, rev(c(&quot;FE − FL&quot;, &quot;ME − ML&quot;,
                                   &quot;ME − FE&quot;, &quot;ML − FE&quot;,
                                   &quot;ME − FL&quot;, &quot;ML − FL&quot;))),
           x = title) %&gt;% 
    ggplot(aes(value, key)) + 
    geom_vline(xintercept = 0, linetype = 2) +
    stat_halfeye(fill = col) + labs(x = &quot;Effect size difference&quot;, y = NULL) +
    facet_wrap(~ x) +
    
    theme_bw() + 
    theme(text = element_text(family = &quot;Raleway&quot;, size = 11)) +
    theme(plot.title = element_text(hjust = 0), 
          strip.text = element_text(hjust = 0),
          panel.background = element_blank(),
          strip.background = element_blank())
}

pp1 &lt;- make_plot1(posterior_muts, &quot;E. Effect sizes (posterior)&quot;)
pp2 &lt;- make_plot2(posterior_muts, &quot;F. Differences in effect size&quot;, col = &quot;#F5CA7B&quot;)

plot_grid(mut_load_correlation_plot,
          arrangeGrob(pp1, pp2, ncol=2), rel_widths = c(1,1.1))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-13-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-13-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-13-1.png" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mut_load_correlation_plot + pp1 + pp2 + 
  plot_layout(ncol = 3, widths = c(1, 0.5,0.5))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(&quot;figures/fig4_mutation_load.pdf&quot;, height = 4.7, width = 9.55)
mut_load_correlation_plot + pp1 + pp2 + 
  plot_layout(ncol = 3, widths = c(1, 0.5,0.5))
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code># Clean up files:
unlink(list(&quot;snp_backing_file.rds&quot;, 
            &quot;snp_backing_file.bk&quot;, 
            &quot;data/input/plink.log&quot;,
            &quot;data/input/plink.frq&quot;))</code></pre>
</div>
<div id="table" class="section level4">
<h4>Table</h4>
<pre class="r"><code>make_posterior_table &lt;- function(posterior){
  as_tibble(rbind(
    posterior_summary(posterior[,1]),
    posterior_summary(posterior[,2]),
    posterior_summary(posterior[,3]),
    posterior_summary(posterior[,4]),
    posterior_summary(posterior[,1] - posterior[,2]), 
    posterior_summary(posterior[,3] - posterior[,4]), 
    posterior_summary(posterior[,3] - posterior[,1]), 
    posterior_summary(posterior[,4] - posterior[,1]), 
    posterior_summary(posterior[,3] - posterior[,2]), 
    posterior_summary(posterior[,4] - posterior[,2]))) %&gt;% 
    mutate(`Effect size or effect size difference` = c(
      &quot;Female early-life (FE)&quot;, &quot;Female late-life (FL)&quot;,
      &quot;Male late-life (ME)&quot;, &quot;Male late-life (ML)&quot;,
      &quot;FE - FL&quot;, &quot;ME - ML&quot;,
      &quot;ME - FE&quot;, &quot;ML - FE&quot;, 
      &quot;ME - FL&quot;, &quot;ML - FL&quot;), .before = 1) %&gt;% 
    rename(&quot;Error&quot; = `Est.Error`,
           `Lower 95% CI` = Q2.5,
           `Upper 95% CI` = Q97.5,)
  
}

saveRDS(make_posterior_table(posterior_muts), 
        &quot;data/derived/supp_table_mutation_load_effects.rds&quot;)

make_posterior_table(posterior_muts) %&gt;% 
  kable(digits = 3) %&gt;% 
  kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Effect size or effect size difference
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Error
</th>
<th style="text-align:right;">
Lower 95% CI
</th>
<th style="text-align:right;">
Upper 95% CI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female early-life (FE)
</td>
<td style="text-align:right;">
-0.032
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
-0.075
</td>
<td style="text-align:right;">
0.011
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late-life (FL)
</td>
<td style="text-align:right;">
-0.054
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
-0.097
</td>
<td style="text-align:right;">
-0.011
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late-life (ME)
</td>
<td style="text-align:right;">
-0.024
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
-0.066
</td>
<td style="text-align:right;">
0.019
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late-life (ML)
</td>
<td style="text-align:right;">
-0.040
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
-0.083
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;">
FE - FL
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
-0.009
</td>
<td style="text-align:right;">
0.052
</td>
</tr>
<tr>
<td style="text-align:left;">
ME - ML
</td>
<td style="text-align:right;">
0.017
</td>
<td style="text-align:right;">
0.012
</td>
<td style="text-align:right;">
-0.006
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;">
ME - FE
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.028
</td>
<td style="text-align:right;">
-0.046
</td>
<td style="text-align:right;">
0.063
</td>
</tr>
<tr>
<td style="text-align:left;">
ML - FE
</td>
<td style="text-align:right;">
-0.008
</td>
<td style="text-align:right;">
0.029
</td>
<td style="text-align:right;">
-0.065
</td>
<td style="text-align:right;">
0.049
</td>
</tr>
<tr>
<td style="text-align:left;">
ME - FL
</td>
<td style="text-align:right;">
0.030
</td>
<td style="text-align:right;">
0.026
</td>
<td style="text-align:right;">
-0.021
</td>
<td style="text-align:right;">
0.082
</td>
</tr>
<tr>
<td style="text-align:left;">
ML - FL
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
0.027
</td>
<td style="text-align:right;">
-0.039
</td>
<td style="text-align:right;">
0.066
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="frequencies-of-antagonistic-loci-and-transcripts"
class="section level2">
<h2>Frequencies of antagonistic loci and transcripts</h2>
<p>The following figures show the proportions of loci/transcripts whose
effect on female fitness (or early-life fitness) is similar or different
to their effect on male fitness (or late-life fitness). The figures were
made by first placing variant or transcript effect sizes (for female or
early-life fitness) into quartiles, which we label as negative effects,
weak negative effects, weak positive effects, and positive effects
(taking advantage of the fact that the median effect size is very close
to zero). We then do the same for male or late-life fitness, and plot
the number of transcripts showing each combination of quartiles, giving
an indication of how many loci/transcripts have aligned or opposing
effects on the different fitness metrics.</p>
<p>The following plots were made after processing the effect sizes with
<code>mashr</code>, which should reduce the number of false signals.
Because there is a positive correlation between the sexes and age
classes, the shrinkage applied by <code>mashr</code> reduces the number
of loci/transcripts that appear to show antagonism between ages and
sexes (this should control the number of ‘false positive’ antagonistic
loci/transcripts).</p>
<div id="frequencies-of-antagonistic-loci" class="section level3">
<h3>Frequencies of antagonistic loci</h3>
<div id="comparing-effects-on-the-sexes" class="section level4 tabset">
<h4 class="tabset">Comparing effects on the sexes</h4>
<p>Rather few loci are in the 1st quartile in their effect on male
fitness and the 4th quartile for their effect on female fitness, or
<em>vice versa</em>, suggesting that variants with strongly sex-opposite
effects on fitness are rare</p>
<div id="figure" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>SNP_effects &lt;- SNP_clumps %&gt;%
  filter(LD_subset) %&gt;% 
  select(contains(&quot;beta&quot;)) %&gt;% 
  select(contains(&quot;ED&quot;)) # this can be changed to &quot;raw&quot; to see the non-mashr effect sizes (i.e. statistical noise)
names(SNP_effects)[grepl(&quot;female_early&quot;, names(SNP_effects))] &lt;- &quot;FE&quot;
names(SNP_effects)[grepl(&quot;female_late&quot;, names(SNP_effects))] &lt;- &quot;FL&quot;
names(SNP_effects)[grepl(&quot;male_early&quot;, names(SNP_effects))] &lt;- &quot;ME&quot;
names(SNP_effects)[grepl(&quot;male_late&quot;, names(SNP_effects))] &lt;- &quot;ML&quot;

temp &lt;- SNP_effects %&gt;% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %&gt;% 
  mutate_at(vars(starts_with(&quot;quartile&quot;)), ~ {
    SNP_effects &lt;- .x
    SNP_effects[SNP_effects == 1] &lt;- &quot;Negative&quot;
    SNP_effects[SNP_effects == 2] &lt;- &quot;Weakly\nnegative&quot;
    SNP_effects[SNP_effects == 3] &lt;- &quot;Weakly\npositive&quot;
    SNP_effects[SNP_effects == 4] &lt;- &quot;Positive&quot;
    factor(SNP_effects, c(&quot;Negative&quot;, &quot;Weakly\nnegative&quot;, &quot;Weakly\npositive&quot;, &quot;Positive&quot;))})

mat_list &lt;- list(table(temp$quartile_FE, temp$quartile_ME),
                 table(temp$quartile_FL, temp$quartile_ML),
                 table(temp$quartile_FE, temp$quartile_FL),
                 table(temp$quartile_ME, temp$quartile_ML))  %&gt;% 
   map(~ as.data.frame(.x)) 
names(mat_list) &lt;- c(&quot;FE, ME&quot;, &quot;FL, ML&quot;, &quot;FE, FL&quot;, &quot;ME, ML&quot;)
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

cols &lt;- c(brewer.pal(9, &quot;Reds&quot;)[c(6,3)], brewer.pal(9, &quot;Blues&quot;)[c(3,6)])

focal_dat &lt;- rbind(
    as.data.frame(mat_list[[1]]) %&gt;% 
      mutate(pp = &quot;A. Early-life fitness&quot;), 
    as.data.frame(mat_list[[2]]) %&gt;% 
      mutate(pp = &quot;B. Late-life fitness&quot;)) 

labs &lt;- c(&quot;Negative&quot;, &quot;Weakly negative&quot;, &quot;Weakly positive&quot;, &quot;Positive&quot;)

fig_5_top_row &lt;- focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nmale fitness&quot;,
                    labels = labs) + 
  xlab(&quot;Association with female fitness&quot;) + 
  ylab(&quot;Number of variants&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        text = element_text(family = &quot;Raleway&quot;, size = 12),
        strip.text = element_text(hjust = 0))

# To count the total number of transcripts tested:
# focal_dat$Freq[focal_dat$pp == &quot;C. Early-life fitness&quot;] %&gt;% sum()

fig_5_top_row</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-15-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-15-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-15-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-15-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-1" class="section level5">
<h5>Table</h5>
<pre class="r"><code>supp_tabl_gwas_intersex &lt;- focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given association with female fitness)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of variants` = Freq) 

saveRDS(supp_tabl_gwas_intersex, &quot;data/derived/supp_tabl_gwas_intersex.rds&quot;)

supp_tabl_gwas_intersex %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Age class
</th>
<th style="text-align:left;">
Association with female fitness
</th>
<th style="text-align:left;">
Association with male fitness
</th>
<th style="text-align:right;">
Number of variants
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given association with female fitness)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
41947
</td>
<td style="text-align:right;">
20.07
</td>
<td style="text-align:right;">
80.29
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
9075
</td>
<td style="text-align:right;">
4.34
</td>
<td style="text-align:right;">
17.37
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1096
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
2.10
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
9459
</td>
<td style="text-align:right;">
4.53
</td>
<td style="text-align:right;">
18.10
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
31336
</td>
<td style="text-align:right;">
14.99
</td>
<td style="text-align:right;">
59.98
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
10546
</td>
<td style="text-align:right;">
5.05
</td>
<td style="text-align:right;">
20.18
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
906
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.73
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
738
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
1.41
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
10950
</td>
<td style="text-align:right;">
5.24
</td>
<td style="text-align:right;">
20.96
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
31731
</td>
<td style="text-align:right;">
15.18
</td>
<td style="text-align:right;">
60.73
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
8828
</td>
<td style="text-align:right;">
4.22
</td>
<td style="text-align:right;">
16.90
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.20
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
886
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
1.70
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
8874
</td>
<td style="text-align:right;">
4.25
</td>
<td style="text-align:right;">
16.99
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
42383
</td>
<td style="text-align:right;">
20.28
</td>
<td style="text-align:right;">
81.12
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
42917
</td>
<td style="text-align:right;">
20.54
</td>
<td style="text-align:right;">
82.14
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
8492
</td>
<td style="text-align:right;">
4.06
</td>
<td style="text-align:right;">
16.25
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
745
</td>
<td style="text-align:right;">
0.36
</td>
<td style="text-align:right;">
1.43
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
93
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.18
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
8787
</td>
<td style="text-align:right;">
4.20
</td>
<td style="text-align:right;">
16.82
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
32734
</td>
<td style="text-align:right;">
15.66
</td>
<td style="text-align:right;">
62.65
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
10158
</td>
<td style="text-align:right;">
4.86
</td>
<td style="text-align:right;">
19.44
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
568
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
1.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
495
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
0.95
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
10381
</td>
<td style="text-align:right;">
4.97
</td>
<td style="text-align:right;">
19.87
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
33013
</td>
<td style="text-align:right;">
15.80
</td>
<td style="text-align:right;">
63.19
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
8358
</td>
<td style="text-align:right;">
4.00
</td>
<td style="text-align:right;">
16.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
640
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
1.22
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
8331
</td>
<td style="text-align:right;">
3.99
</td>
<td style="text-align:right;">
15.95
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
43227
</td>
<td style="text-align:right;">
20.68
</td>
<td style="text-align:right;">
82.74
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="comparing-effects-on-the-age-classes"
class="section level4 tabset">
<h4 class="tabset">Comparing effects on the age classes</h4>
<div id="figure-1" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>focal_dat &lt;- rbind(
  as.data.frame(mat_list[[3]]) %&gt;% 
    mutate(pp = &quot;A. Female fitness&quot;), 
  as.data.frame(mat_list[[4]]) %&gt;% 
    mutate(pp = &quot;B. Male fitness&quot;)) 

focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nlate-life fitness&quot;,
                    labels = labs) + 
  xlab(&quot;Association with early-life fitness&quot;) + 
  ylab(&quot;Number of variants&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = &quot;Raleway&quot;, size = 12))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-17-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-17-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-17-1.png" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-17-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-2" class="section level5">
<h5>Table</h5>
<pre class="r"><code>supp_tabl_gwas_interage &lt;- focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given association with early-life fitness)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of variants` = Freq)

saveRDS(supp_tabl_gwas_interage, &quot;data/derived/supp_tabl_gwas_interage.rds&quot;)

supp_tabl_gwas_interage %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:left;">
Association with early-life fitness
</th>
<th style="text-align:left;">
Association with late-life fitness
</th>
<th style="text-align:right;">
Number of variants
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given association with early-life fitness)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
50560
</td>
<td style="text-align:right;">
24.19
</td>
<td style="text-align:right;">
96.77
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1687
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
3.23
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
1687
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
3.23
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
48453
</td>
<td style="text-align:right;">
23.18
</td>
<td style="text-align:right;">
92.74
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
2107
</td>
<td style="text-align:right;">
1.01
</td>
<td style="text-align:right;">
4.03
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
2107
</td>
<td style="text-align:right;">
1.01
</td>
<td style="text-align:right;">
4.03
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
48489
</td>
<td style="text-align:right;">
23.20
</td>
<td style="text-align:right;">
92.81
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1651
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
3.16
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1651
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
3.16
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
50595
</td>
<td style="text-align:right;">
24.21
</td>
<td style="text-align:right;">
96.84
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
49584
</td>
<td style="text-align:right;">
23.73
</td>
<td style="text-align:right;">
94.90
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
2661
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
5.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
2657
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
5.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
46371
</td>
<td style="text-align:right;">
22.19
</td>
<td style="text-align:right;">
88.75
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
3218
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
6.16
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.01
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
3214
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
6.15
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
46519
</td>
<td style="text-align:right;">
22.26
</td>
<td style="text-align:right;">
89.04
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
2508
</td>
<td style="text-align:right;">
1.20
</td>
<td style="text-align:right;">
4.80
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
2509
</td>
<td style="text-align:right;">
1.20
</td>
<td style="text-align:right;">
4.80
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
49736
</td>
<td style="text-align:right;">
23.80
</td>
<td style="text-align:right;">
95.20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="run-a-chi2-test" class="section level4">
<h4>Run a <span class="math inline">\(\chi^2\)</span> test</h4>
<p>The following <span class="math inline">\(\chi^2\)</span> test
examines the null hypothesis that the proportion of candidate sexually
antagonistic loci (i.e. those with a quartile 1 effect on fitness of sex
<span class="math inline">\(i\)</span> and a quartile 4 effect on sex
<span class="math inline">\(j\)</span>) is equal when fitness
associations are calculated using the early- and late-life fitness data.
This null hypothesis is rejected: the % of SA loci is 0.111% at early
life and 0.067% at late life, which is a 1.6-fold difference.</p>
<pre class="r"><code>chisq_table &lt;- as.table(rbind(c(129+103, 208987 - (129+103)), c(93+48, 208987 - (93+48))))
dimnames(chisq_table) &lt;- list(life_stage = c(&quot;Early_life&quot;, &quot;Late_life&quot;),
                              antagonism_status = c(&quot;Sex_antag&quot;,&quot;Non_sex_antag&quot;))
chisq_table</code></pre>
<pre><code>            antagonism_status
life_stage   Sex_antag Non_sex_antag
  Early_life       232        208755
  Late_life        141        208846</code></pre>
<pre class="r"><code>chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 21.735, df = 1, p-value = 3.13e-06</code></pre>
</div>
</div>
<div id="frequencies-of-antagonistic-transcripts"
class="section level3">
<h3>Frequencies of antagonistic transcripts</h3>
<div id="comparing-effects-on-the-sexes-1"
class="section level4 tabset">
<h4 class="tabset">Comparing effects on the sexes</h4>
<div id="figure-2" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>TWAS_ED &lt;- readRDS(&quot;data/derived/TWAS/TWAS_ED.rds&quot;)

binned_transcripts &lt;- data.frame(
  FBID = read_csv(&quot;data/derived/TWAS/TWAS_results.csv&quot;)$FBID,
  as.data.frame(get_pm(TWAS_ED))) %&gt;% 
  as_tibble() %&gt;% rename_all(~ str_remove_all(.x, &quot;beta_&quot;)) %&gt;% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %&gt;% 
  mutate_at(vars(starts_with(&quot;quartile&quot;)), ~ {
    SNP_effects &lt;- .x
    SNP_effects[SNP_effects == 1] &lt;- &quot;Negative&quot;
    SNP_effects[SNP_effects == 2] &lt;- &quot;Weakly\nnegative&quot;
    SNP_effects[SNP_effects == 3] &lt;- &quot;Weakly\npositive&quot;
    SNP_effects[SNP_effects == 4] &lt;- &quot;Positive&quot;
    factor(SNP_effects, c(&quot;Negative&quot;, &quot;Weakly\nnegative&quot;, &quot;Weakly\npositive&quot;, &quot;Positive&quot;))})


mat_list &lt;- list(table(binned_transcripts$quartile_FE, binned_transcripts$quartile_ME),
                 table(binned_transcripts$quartile_FL, binned_transcripts$quartile_ML),
                 table(binned_transcripts$quartile_FE, binned_transcripts$quartile_FL),
                 table(binned_transcripts$quartile_ME, binned_transcripts$quartile_ML))  %&gt;% 
   map(~ as.data.frame(.x)) 
names(mat_list) &lt;- c(&quot;FE, ME&quot;, &quot;FL, ML&quot;, &quot;FE, FL&quot;, &quot;ME, ML&quot;)
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

focal_dat &lt;- rbind(
    as.data.frame(mat_list[[1]]) %&gt;% 
      mutate(pp = &quot;C. Early-life fitness&quot;), 
    as.data.frame(mat_list[[2]]) %&gt;% 
      mutate(pp = &quot;D. Late-life fitness&quot;)) 

# To count the total number of transcripts tested, 14286:
# focal_dat$Freq[focal_dat$pp == &quot;C. Early-life fitness&quot;] %&gt;% sum()

fig_5_bottom_row &lt;- focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nmale fitness&quot;,
                    labels = labs) + 
  xlab(&quot;Association with female fitness&quot;) + 
  ylab(&quot;Number of transcripts&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        text = element_text(family = &quot;Raleway&quot;, size = 12),
        strip.text = element_text(hjust = 0))
        # strip.text = element_text(hjust = 0, size = 12))

fig_5_bottom_row</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-20-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-20-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e4f2f4f17ad066aea2499537ec7bd625b587cbdf/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-20-1.png" target="_blank">e4f2f4f</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-20-1.png" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-3" class="section level5">
<h5>Table</h5>
<pre class="r"><code>supp_tabl_twas_intersex &lt;- focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given association with female fitness)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of transcripts` = Freq)

saveRDS(supp_tabl_twas_intersex, &quot;data/derived/supp_tabl_twas_intersex.rds&quot;)

supp_tabl_twas_intersex %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Age class
</th>
<th style="text-align:left;">
Association with female fitness
</th>
<th style="text-align:left;">
Association with male fitness
</th>
<th style="text-align:right;">
Number of transcripts
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given association with female fitness)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
1578
</td>
<td style="text-align:right;">
11.05
</td>
<td style="text-align:right;">
44.18
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
752
</td>
<td style="text-align:right;">
5.26
</td>
<td style="text-align:right;">
21.05
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
551
</td>
<td style="text-align:right;">
3.86
</td>
<td style="text-align:right;">
15.43
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
691
</td>
<td style="text-align:right;">
4.84
</td>
<td style="text-align:right;">
19.34
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
773
</td>
<td style="text-align:right;">
5.41
</td>
<td style="text-align:right;">
21.64
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1168
</td>
<td style="text-align:right;">
8.18
</td>
<td style="text-align:right;">
32.70
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1043
</td>
<td style="text-align:right;">
7.30
</td>
<td style="text-align:right;">
29.20
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
588
</td>
<td style="text-align:right;">
4.12
</td>
<td style="text-align:right;">
16.46
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
532
</td>
<td style="text-align:right;">
3.72
</td>
<td style="text-align:right;">
14.90
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1058
</td>
<td style="text-align:right;">
7.41
</td>
<td style="text-align:right;">
29.63
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1201
</td>
<td style="text-align:right;">
8.41
</td>
<td style="text-align:right;">
33.63
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
780
</td>
<td style="text-align:right;">
5.46
</td>
<td style="text-align:right;">
21.84
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
689
</td>
<td style="text-align:right;">
4.82
</td>
<td style="text-align:right;">
19.29
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
594
</td>
<td style="text-align:right;">
4.16
</td>
<td style="text-align:right;">
16.63
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
776
</td>
<td style="text-align:right;">
5.43
</td>
<td style="text-align:right;">
21.73
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1512
</td>
<td style="text-align:right;">
10.58
</td>
<td style="text-align:right;">
42.34
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
1603
</td>
<td style="text-align:right;">
11.22
</td>
<td style="text-align:right;">
44.88
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
758
</td>
<td style="text-align:right;">
5.31
</td>
<td style="text-align:right;">
21.22
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
534
</td>
<td style="text-align:right;">
3.74
</td>
<td style="text-align:right;">
14.95
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
677
</td>
<td style="text-align:right;">
4.74
</td>
<td style="text-align:right;">
18.95
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
765
</td>
<td style="text-align:right;">
5.35
</td>
<td style="text-align:right;">
21.42
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1175
</td>
<td style="text-align:right;">
8.22
</td>
<td style="text-align:right;">
32.89
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1053
</td>
<td style="text-align:right;">
7.37
</td>
<td style="text-align:right;">
29.48
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
579
</td>
<td style="text-align:right;">
4.05
</td>
<td style="text-align:right;">
16.21
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
540
</td>
<td style="text-align:right;">
3.78
</td>
<td style="text-align:right;">
15.12
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1061
</td>
<td style="text-align:right;">
7.43
</td>
<td style="text-align:right;">
29.71
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1206
</td>
<td style="text-align:right;">
8.44
</td>
<td style="text-align:right;">
33.77
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
764
</td>
<td style="text-align:right;">
5.35
</td>
<td style="text-align:right;">
21.39
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
664
</td>
<td style="text-align:right;">
4.65
</td>
<td style="text-align:right;">
18.59
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
578
</td>
<td style="text-align:right;">
4.05
</td>
<td style="text-align:right;">
16.19
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
778
</td>
<td style="text-align:right;">
5.45
</td>
<td style="text-align:right;">
21.79
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1551
</td>
<td style="text-align:right;">
10.86
</td>
<td style="text-align:right;">
43.43
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="run-a-chi2-test-1" class="section level4">
<h4>Run a <span class="math inline">\(\chi^2\)</span> test</h4>
<p>The following <span class="math inline">\(\chi^2\)</span> test
examines the null hypothesis that the proportion of candidate sexually
antagonistic transcripts (i.e. those with a quartile 1 effect on fitness
of sex <span class="math inline">\(i\)</span> and a quartile 4 effect on
sex <span class="math inline">\(j\)</span>) is equal when fitness
associations are calculated using the early- and late-life fitness data.
This null hypothesis is not rejected: the % of SA transcripts is 9.66%
at early life and 9.39% at late life, which is a 1.03-fold
difference.</p>
<pre class="r"><code>chisq_table &lt;- as.table(rbind(c(691+689, 14286 - (691+689)), c(677 + 664, 14286 - (677 + 664))))
dimnames(chisq_table) &lt;- list(life_stage = c(&quot;Early_life&quot;, &quot;Late_life&quot;),
                              antagonism_status = c(&quot;Sex_antag&quot;,&quot;Non_sex_antag&quot;))
chisq_table</code></pre>
<pre><code>            antagonism_status
life_stage   Sex_antag Non_sex_antag
  Early_life      1380         12906
  Late_life       1341         12945</code></pre>
<pre class="r"><code>chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 0.58655, df = 1, p-value = 0.4438</code></pre>
</div>
<div id="comparing-effects-on-the-age-classes-1"
class="section level4 tabset">
<h4 class="tabset">Comparing effects on the age classes</h4>
<div id="figure-3" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>focal_dat &lt;- rbind(
  as.data.frame(mat_list[[3]]) %&gt;% 
    mutate(pp = &quot;A. Female fitness&quot;), 
  as.data.frame(mat_list[[4]]) %&gt;% 
    mutate(pp = &quot;B. Male fitness&quot;))

focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nlate-life fitness&quot;,
                    labels = labs) + 
  xlab(&quot;Association with early-life fitness&quot;) + 
  ylab(&quot;Number of transcripts&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = &quot;Raleway&quot;, size = 12))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-24-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-24-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-4" class="section level5">
<h5>Table</h5>
<pre class="r"><code>supp_tabl_twas_interage &lt;- focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given association with early-life fitness)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of transcripts` = Freq)

saveRDS(supp_tabl_twas_interage, &quot;data/derived/supp_tabl_twas_interage.rds&quot;)

supp_tabl_twas_interage %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:left;">
Association with early-life fitness
</th>
<th style="text-align:left;">
Association with late-life fitness
</th>
<th style="text-align:right;">
Number of transcripts
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given association with early-life fitness)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
3523
</td>
<td style="text-align:right;">
24.66
</td>
<td style="text-align:right;">
98.63
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
1.37
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
1.37
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
3464
</td>
<td style="text-align:right;">
24.25
</td>
<td style="text-align:right;">
96.98
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
1.65
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
1.65
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
3450
</td>
<td style="text-align:right;">
24.15
</td>
<td style="text-align:right;">
96.61
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.74
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.74
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
3509
</td>
<td style="text-align:right;">
24.56
</td>
<td style="text-align:right;">
98.26
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
3550
</td>
<td style="text-align:right;">
24.85
</td>
<td style="text-align:right;">
99.38
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
3521
</td>
<td style="text-align:right;">
24.65
</td>
<td style="text-align:right;">
98.57
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.81
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.81
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
3520
</td>
<td style="text-align:right;">
24.64
</td>
<td style="text-align:right;">
98.57
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
3549
</td>
<td style="text-align:right;">
24.84
</td>
<td style="text-align:right;">
99.38
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="plotting-and-modelling-the-evidence-for-antagonism"
class="section level2">
<h2>Plotting and modelling the evidence for antagonism</h2>
<div id="calculate-the-evidence-ratios" class="section level3">
<h3>Calculate the evidence ratios</h3>
<div id="for-the-gwas-results" class="section level4">
<h4>For the GWAS results</h4>
<pre class="r"><code>get_antagonism_ratios &lt;- function(dat){
  dat %&gt;%
    
    # Convert the LFSR to the probability that the effect size is positive
    mutate(pp_female_early = ifelse(pheno1 &gt; 0, LFSR1, 1 - LFSR1),
           pp_female_late  = ifelse(pheno2 &gt; 0, LFSR2, 1 - LFSR2),
           pp_male_early   = ifelse(pheno3 &gt; 0, LFSR3, 1 - LFSR3),
           pp_male_late    = ifelse(pheno4 &gt; 0, LFSR4, 1 - LFSR4)) %&gt;%
    
    # Calculate the probabilities that beta_i and beta_j have the same/opposite signs
    mutate(p_sex_concord_early = pp_female_early * pp_male_early + 
             (1 - pp_female_early) * (1 - pp_male_early),
           p_sex_antag_early = pp_female_early * (1 - pp_male_early) + 
             (1 - pp_female_early) * pp_male_early,
           p_sex_concord_late  = pp_female_late * pp_male_late + 
             (1 - pp_female_late) * (1 - pp_male_late),
           p_sex_antag_late = pp_female_late * (1 - pp_male_late) + 
             (1 - pp_female_late) * pp_male_late,
           p_age_concord_females = pp_female_early * pp_female_late + 
             (1 - pp_female_early) * (1 - pp_female_late),
           p_age_antag_females = pp_female_early * (1 - pp_female_late) + 
             (1 - pp_female_early) * pp_female_late,
           p_age_concord_males = pp_male_early * pp_male_late + (1 - pp_male_early) * (1 - pp_male_late),
           p_age_antag_males = pp_male_early * (1 - pp_male_late) + (1 - pp_male_early) * pp_male_late) %&gt;%
    
    # Find the ratios of some of these two probabilities (i.e. the &quot;evidence ratios&quot;)
    mutate(inter_sex_early = p_sex_concord_early / p_sex_antag_early,
           inter_sex_late = p_sex_concord_late / p_sex_antag_late,
           inter_age_females = p_age_concord_females / p_age_antag_females,
           inter_age_males = p_age_concord_males / p_age_antag_males) 
}

antagonism_evidence_ratios_gwas &lt;- all_snps %&gt;%
  rename(pheno1 = beta_female_early_mashr_ED, 
         pheno2 = beta_female_late_mashr_ED, 
         pheno3 = beta_male_early_mashr_ED, 
         pheno4 = beta_male_late_mashr_ED,
         LFSR1 = LFSR_female_early_mashr_ED, 
         LFSR2 = LFSR_female_late_mashr_ED, 
         LFSR3 = LFSR_male_early_mashr_ED, 
         LFSR4 = LFSR_male_late_mashr_ED) %&gt;% 
  select(SNP, SNP_clump, starts_with(&quot;pheno&quot;), LFSR1, LFSR2, LFSR3, LFSR4) %&gt;% 
  filter(!is.na(pheno1)) %&gt;% 
  collect(n=Inf) %&gt;% 
  distinct() %&gt;% 
  get_antagonism_ratios() %&gt;%
  select(SNP, SNP_clump, starts_with(&quot;inter&quot;)) </code></pre>
</div>
<div id="for-the-twas-results" class="section level4">
<h4>For the TWAS results</h4>
<pre class="r"><code>TWAS_ED &lt;- readRDS(&quot;data/derived/TWAS/TWAS_ED.rds&quot;)

twas &lt;- data.frame(
  FBID = read_csv(&quot;data/derived/TWAS/TWAS_results.csv&quot;)$FBID,
      as.data.frame(get_pm(TWAS_ED)),
      as.data.frame(get_lfsr(TWAS_ED)) %&gt;% 
        rename_all(~str_replace_all(., &quot;beta&quot;, &quot;LFSR&quot;))) %&gt;% 
      as_tibble() %&gt;%
  left_join(tbl(db, &quot;genes&quot;) %&gt;% select(FBID, chromosome) %&gt;% collect(), by = &quot;FBID&quot;) %&gt;%
  left_join(read_csv(&quot;data/derived/gene_expression_by_sex.csv&quot;), by = &quot;FBID&quot;) %&gt;%
  filter(chromosome %in% c(&quot;2L&quot;, &quot;2R&quot;, &quot;3L&quot;, &quot;3R&quot;, &quot;X&quot;)) %&gt;%
  rename(pheno1 = beta_FE, 
         pheno2 = beta_FL, 
         pheno3 = beta_ME, 
         pheno4 = beta_ML,
         LFSR1 = LFSR_FE, 
         LFSR2 = LFSR_FL, 
         LFSR3 = LFSR_ME, 
         LFSR4 = LFSR_ML)

antagonism_evidence_ratios_twas &lt;- twas %&gt;% 
  get_antagonism_ratios() %&gt;%
  select(FBID, chromosome, male_bias_in_expression, AveExpr, starts_with(&quot;inter&quot;)) %&gt;% 
  mutate(chromosome = relevel(factor(chromosome), ref = &quot;X&quot;))</code></pre>
</div>
</div>
<div id="plot-the-evidence-ratios" class="section level3">
<h3>Plot the evidence ratios</h3>
<pre class="r"><code>make_evidence_ratio_plot &lt;- function(ERs, ymax, ylab){
  # Argument needs to be a dataframe of loci or transcripts, each identified by a column called &quot;identifier&quot;
  # There should be a col called &quot;evidence_ratio&quot;, calculated from the LFSR from ED mashr. The &quot;trait&quot; col says which type of ER it is.
  
  ERs$trait[ERs$trait == &quot;inter_sex_early&quot;] &lt;- &quot;Inter-sex (early life)&quot;
  ERs$trait[ERs$trait == &quot;inter_sex_late&quot;] &lt;- &quot;Inter-sex (late life)&quot;
  ERs$trait[ERs$trait == &quot;inter_age_females&quot;] &lt;- &quot;Inter-age (females)&quot;
  ERs$trait[ERs$trait == &quot;inter_age_males&quot;] &lt;- &quot;Inter-age (males)&quot;
  
  antagonism_evidence_ratios &lt;- ERs %&gt;%
    mutate(trait = factor(trait, c(&quot;Inter-sex (early life)&quot;,
                                   &quot;Inter-sex (late life)&quot;,
                                   &quot;Inter-age (females)&quot;,
                                   &quot;Inter-age (males)&quot;)))
  
  antagonism_evidence_ratios %&gt;%
    ggplot(aes(log2(evidence_ratio))) + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio &lt; 1), 
                   bins = 500, fill = &quot;#FF635C&quot;) + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio &gt; 1), 
                   bins = 500, fill = &quot;#5B8AFD&quot;) +
    coord_cartesian(xlim = c(-10, 10), ylim = c(0, ymax)) + 
    scale_x_continuous(breaks = c(-10,  -6,  -2,   2,   6,  10), 
                       labels = c(paste(&quot;1/&quot;,2 ^ c(10,  6,  2), sep = &quot;&quot;), 2 ^ c(2,6,10))) + 
    facet_wrap(~ trait) + 
    xlab(&quot;Evidence ratio (log2 scale)&quot;) + ylab(ylab) + 
    theme_bw() + 
    theme(panel.border = element_rect(size = 0.8),
          text = element_text(family = &quot;Raleway&quot;, size = 12),
          axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
          strip.background = element_blank())
}

GWAS_ratios_plot &lt;- antagonism_evidence_ratios_gwas %&gt;% 
  select(SNP_clump,  starts_with(&quot;inter&quot;)) %&gt;%
  distinct() %&gt;% # Include SNPs in 100% LD a single time, such that these clumps are the unit of replication
  gather(trait, evidence_ratio, -SNP_clump) %&gt;% 
  rename(identifier = SNP_clump) %&gt;% 
  make_evidence_ratio_plot(ymax = 5000, ylab = &quot;Number of loci&quot;)

# For counting the number of transcripts with 50x more evidence for antagonism 
# antagonism_evidence_ratios_twas %&gt;% 
#   select(FBID,  starts_with(&quot;inter&quot;)) %&gt;%
#   gather(trait, evidence_ratio, -FBID) %&gt;% 
#   rename(identifier = FBID) %&gt;% filter(evidence_ratio &lt; 1/50) %&gt;% summarise(n())

TWAS_ratios_plot &lt;- antagonism_evidence_ratios_twas %&gt;% 
  select(FBID,  starts_with(&quot;inter&quot;)) %&gt;%
  gather(trait, evidence_ratio, -FBID) %&gt;% 
  rename(identifier = FBID) %&gt;% 
  make_evidence_ratio_plot(ymax = 500, ylab = &quot;Number of transcripts&quot;)

pp &lt;- plot_grid(GWAS_ratios_plot, TWAS_ratios_plot, 
                   labels = c(&#39;A&#39;, &#39;B&#39;), label_size = 12)

ggsave(&quot;figures/fig6_antagonism_ratios.pdf&quot;, pp, width = 8.5, height = 4.9)

pp</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-evidence_ratio_plot-1">
Past versions of evidence_ratio_plot-1.png
</button>
</p>
<div id="fig-evidence_ratio_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c63ff6283bcda0ea9a7db6dc201d1d3bf3bdb4b0/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">c63ff62</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/1bb603c18144f3ad95844a63f1f0dedcab9e2d0b/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">1bb603c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="model-the-evidence-ratios" class="section level3">
<h3>Model the evidence ratios</h3>
<div id="inter-sex-early-life-model-gwas" class="section level4">
<h4>Inter-sex, early-life model (GWAS)</h4>
<p>The initial full model contains the predictors <code>MAF</code>,
<code>chromosome</code>, and <code>major_effect</code> (which records
whether or not the focal variant causes an insertion, deletion, or
nonsynonymous change in a coding sequence). <code>major_effect</code>
was not significant, so the model was refitted without it.</p>
<pre class="r"><code># Annotate data with MAF, site, class, etc. 
dat &lt;- left_join(antagonism_evidence_ratios_gwas,
          collect(tbl(db, &quot;variants&quot;)), by = &quot;SNP&quot;, multiple = &quot;all&quot;) %&gt;% 
  distinct() %&gt;% rename(chromosome = chr)

# Remove chromosome 4 (too few sites)
dat &lt;- dat %&gt;% filter(chromosome != &quot;4&quot;) 
dat$chromosome &lt;- relevel(factor(dat$chromosome), ref = &quot;X&quot;)

# Classify mutations as being major effect or not
dat &lt;- dat %&gt;% 
  mutate(major_effect = &quot;No&quot;,
         major_effect = replace(major_effect, site.class %in% major_effect_types, &quot;Yes&quot;))

# For mutations with multiple site classes, record whether any site class is major or not,
# ensuring that there is one row in &#39;dat&#39; for each SNP
dat &lt;- dat %&gt;% 
  group_by(SNP_clump) %&gt;% 
  summarise(inter_sex_early = inter_sex_early[1],
            inter_sex_late = inter_sex_late[1],
            MAF = MAF[1],
            chromosome = chromosome[1],
            major_effect = ifelse(any(major_effect == &quot;Yes&quot;), &quot;Yes&quot;, &quot;No&quot;))

n_loci &lt;- prettyNum(nrow(dat), big.mark = &quot;,&quot;, scientific = FALSE)

intersex_early_model_gwas &lt;- glm(antagonistic ~ chromosome + MAF + major_effect, 
             data = dat %&gt;% 
               mutate(antagonistic = as.numeric(inter_sex_early &lt;= 
                        quantile(dat$inter_sex_early, probs = 0.01))), 
    family = &quot;binomial&quot;)
car::Anova(intersex_early_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
             LR Chisq Df Pr(&gt;Chisq)    
chromosome      11.28  4    0.02356 *  
MAF            626.64  1    &lt; 2e-16 ***
major_effect     0.00  1    0.96629    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_early_model_gwas &lt;- update(intersex_early_model_gwas,  ~ . -major_effect)
car::Anova(intersex_early_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
           LR Chisq Df Pr(&gt;Chisq)    
chromosome    11.28  4    0.02357 *  
MAF          626.69  1    &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_early_model_gwas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ chromosome + MAF, family = &quot;binomial&quot;, 
    data = dat %&gt;% mutate(antagonistic = as.numeric(inter_sex_early &lt;= 
        quantile(dat$inter_sex_early, probs = 0.01))))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.2251  -0.1762  -0.1208  -0.0943   3.4180  

Coefficients:
             Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  -5.83036    0.08601 -67.787  &lt; 2e-16 ***
chromosome2L -0.26312    0.08101  -3.248  0.00116 ** 
chromosome2R -0.14154    0.07893  -1.793  0.07295 .  
chromosome3L -0.13342    0.07776  -1.716  0.08622 .  
chromosome3R -0.08420    0.08011  -1.051  0.29321    
MAF           4.35588    0.18431  23.634  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 18501  on 165100  degrees of freedom
Residual deviance: 17865  on 165095  degrees of freedom
AIC: 17877

Number of Fisher Scoring iterations: 8</code></pre>
</div>
<div id="inter-sex-late-life-model-gwas" class="section level4">
<h4>Inter-sex, late-life model (GWAS)</h4>
<p>The initial full model contains the predictors <code>MAF</code>,
<code>chromosome</code>, and <code>major_effect</code> (which records
whether or not the focal variant causes an insertion, deletion, or
nonsynonymous change in a coding sequence). <code>major_effect</code>
was not significant, so the model was refitted without it.</p>
<pre class="r"><code>intersex_late_model_gwas &lt;- glm(antagonistic ~ chromosome + MAF + major_effect, 
             data = dat %&gt;% 
               mutate(antagonistic = inter_sex_late &lt;= 
                        quantile(dat$inter_sex_late, probs = 0.01)), 
    family = &quot;binomial&quot;)
car::Anova(intersex_late_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
             LR Chisq Df Pr(&gt;Chisq)    
chromosome       8.06  4    0.08948 .  
MAF            536.20  1    &lt; 2e-16 ***
major_effect     0.14  1    0.71216    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_late_model_gwas &lt;- update(intersex_late_model_gwas,  ~ . -major_effect)
car::Anova(intersex_late_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
           LR Chisq Df Pr(&gt;Chisq)    
chromosome     8.07  4    0.08891 .  
MAF          536.46  1    &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_late_model_gwas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ chromosome + MAF, family = &quot;binomial&quot;, 
    data = dat %&gt;% mutate(antagonistic = inter_sex_late &lt;= quantile(dat$inter_sex_late, 
        probs = 0.01)))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.2201  -0.1746  -0.1232  -0.0982   3.3606  

Coefficients:
             Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  -5.69639    0.08403 -67.791  &lt; 2e-16 ***
chromosome2L -0.21978    0.07980  -2.754  0.00588 ** 
chromosome2R -0.16418    0.07904  -2.077  0.03779 *  
chromosome3L -0.14343    0.07764  -1.847  0.06468 .  
chromosome3R -0.12751    0.08057  -1.583  0.11351    
MAF           3.99585    0.18092  22.087  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 18501  on 165100  degrees of freedom
Residual deviance: 17959  on 165095  degrees of freedom
AIC: 17971

Number of Fisher Scoring iterations: 8</code></pre>
</div>
<div id="inter-sex-early-life-model-twas" class="section level4">
<h4>Inter-sex, early-life model (TWAS)</h4>
<p>The full model fitted here contains the predictors
<code>chromosome</code>, <code>absolute_sex_bias_in_expression</code>,
and <code>mean_expression_level</code>. Only the latter is significant,
with highly-expressed transcripts being less likely to be sexually
antagonistic.</p>
<pre class="r"><code>n_transcripts &lt;- prettyNum(nrow(antagonism_evidence_ratios_twas), 
                           big.mark = &quot;,&quot;, scientific = FALSE)

twas_model_data &lt;- antagonism_evidence_ratios_twas %&gt;% 
  mutate(antagonistic = inter_sex_early &lt;= 
           quantile(antagonism_evidence_ratios_twas$inter_sex_early, probs=0.01),
         absolute_sex_bias_in_expression = abs(male_bias_in_expression),
         mean_expression_level = AveExpr)


intersex_early_twas &lt;- glm(
  antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level, 
  data = twas_model_data, 
  family = &quot;binomial&quot;)
car::Anova(intersex_early_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                                LR Chisq Df Pr(&gt;Chisq)   
chromosome                        3.5741  4   0.466695   
absolute_sex_bias_in_expression   0.0446  1   0.832701   
mean_expression_level             8.7170  1   0.003153 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_early_twas &lt;- update(
  intersex_early_twas,  
  ~ . -absolute_sex_bias_in_expression -chromosome)
car::Anova(intersex_early_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                      LR Chisq Df Pr(&gt;Chisq)   
mean_expression_level   9.2385  1    0.00237 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_early_twas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ mean_expression_level, family = &quot;binomial&quot;, 
    data = twas_model_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.1848  -0.1543  -0.1391  -0.1256   3.2353  

Coefficients:
                      Estimate Std. Error z value Pr(&gt;|z|)   
(Intercept)            -1.3052     1.0838  -1.204  0.22848   
mean_expression_level  -0.5401     0.1792  -3.014  0.00258 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1590.3  on 14190  degrees of freedom
Residual deviance: 1581.0  on 14189  degrees of freedom
AIC: 1585

Number of Fisher Scoring iterations: 7</code></pre>
</div>
<div id="inter-sex-late-life-model-twas" class="section level4">
<h4>Inter-sex, late-life model (TWAS)</h4>
<p>The full model fitted here contains the predictors
<code>chromosome</code>, <code>absolute_sex_bias_in_expression</code>,
and <code>mean_expression_level</code>. Only the latter is significant,
with highly-expressed transcripts being less likely to be sexually
antagonistic.</p>
<pre class="r"><code>intersex_late_model_twas &lt;- glm(
  antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level,
  data = antagonism_evidence_ratios_twas %&gt;% 
    mutate(antagonistic = inter_sex_late &lt;= 
             quantile(antagonism_evidence_ratios_twas$inter_sex_late, probs=0.01),
           absolute_sex_bias_in_expression = abs(male_bias_in_expression),
           mean_expression_level = AveExpr), 
  family = &quot;binomial&quot;)

car::Anova(intersex_late_model_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                                LR Chisq Df Pr(&gt;Chisq)   
chromosome                        3.7808  4   0.436479   
absolute_sex_bias_in_expression   0.0003  1   0.986291   
mean_expression_level             8.3362  1   0.003886 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_late_model_twas &lt;- update(
  intersex_late_model_twas, 
  ~ . -absolute_sex_bias_in_expression -chromosome)
car::Anova(intersex_late_model_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                      LR Chisq Df Pr(&gt;Chisq)   
mean_expression_level   8.7545  1   0.003088 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_late_model_twas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ mean_expression_level, family = &quot;binomial&quot;, 
    data = antagonism_evidence_ratios_twas %&gt;% mutate(antagonistic = inter_sex_late &lt;= 
        quantile(antagonism_evidence_ratios_twas$inter_sex_late, 
            probs = 0.01), absolute_sex_bias_in_expression = abs(male_bias_in_expression), 
        mean_expression_level = AveExpr))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.1836  -0.1540  -0.1392  -0.1261   3.2298  

Coefficients:
                      Estimate Std. Error z value Pr(&gt;|z|)   
(Intercept)            -1.3931     1.0835  -1.286  0.19852   
mean_expression_level  -0.5255     0.1790  -2.935  0.00333 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1590.3  on 14190  degrees of freedom
Residual deviance: 1581.5  on 14189  degrees of freedom
AIC: 1585.5

Number of Fisher Scoring iterations: 7</code></pre>
</div>
<div id="plotting-key-results-of-these-models" class="section level4">
<h4>Plotting key results of these models</h4>
<p>Panels A and B show the predicted effects of minor allele frequency
and chromosome, respectively, on the probability that a locus is among
the top 1% candidate sexually antagonistic loci, as ranked by their
evidence ratio. The model predictions come from a binomial GLM with
minor allele frequency and chromosome as predictor variables. Panel C
shows the predicted effect of average gene expression level on the
probability that a transcript is among the top 1% candidate sexually
antagonistic transcript, from a binomial GLM with expression level as a
predictor. The dashed lines and error bars show 95% confidence
intervals, estimated as 1.96SE.</p>
<pre class="r"><code># plotting models
new &lt;- data.frame(MAF = seq(from = 0.05, to = 0.5, by = 0.01), chromosome = &quot;X&quot;)
inv_logit &lt;- function(x) exp(x)/(1+exp(x))
preds &lt;- data.frame(new, 
                    prop = predict(intersex_early_model_gwas, 
                                   newdata = new, 
                                   se.fit=T, type = &quot;link&quot;)) %&gt;% 
  mutate(y = prop.fit, 
         lwr = prop.fit - 1.96*prop.se.fit, 
         upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), 
         lwr = inv_logit(lwr), 
         upr = inv_logit(upr))

p1 &lt;- preds %&gt;% 
  mutate(panel = &quot;A.&quot;) %&gt;% 
  ggplot(aes(MAF,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  scale_x_continuous(limits = c(0, 0.5)) + 
  scale_y_continuous(limits = c(0, max(preds$y))) + 
  facet_wrap( ~ panel) + 
  theme_bw() + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12)) +
  theme(strip.background = element_blank(), 
        strip.text = element_text(hjust = 0)) +
  ylab(&quot;Probability of being in the top 1%\nsexually antagonistic loci (\u00B1 95% CIs)&quot;) +
  xlab(&quot;Minor allele frequency&quot;)

new2 &lt;- data.frame(chromosome = levels(dat$chromosome), MAF = 0.3)
inv_logit &lt;- function(x) exp(x)/(1+exp(x))
preds2 &lt;- data.frame(new2, prop = predict(intersex_early_model_gwas, newdata = new2, se.fit=T, type = &quot;link&quot;)) %&gt;% 
  mutate(y = prop.fit, 
         lwr = prop.fit - 1.96*prop.se.fit, 
         upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), 
         lwr = inv_logit(lwr), 
         upr = inv_logit(upr))

p2 &lt;- preds2 %&gt;% 
  mutate(panel = &quot;B.&quot;) %&gt;%
  ggplot(aes(chromosome,  y)) + 
  geom_point(size = 3) + 
  geom_errorbar(aes(ymin = lwr, ymax = upr), width = 0) + 
  facet_wrap( ~ panel) + 
  theme_bw() +
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12)) +
  theme(strip.background = element_blank(), strip.text = element_text(hjust = 0)) +
  ylab(&quot; \n &quot;) +
  xlab(&quot;Chromosome arm&quot;)

new &lt;- data.frame(
  mean_expression_level = seq(
    from = min(twas_model_data$mean_expression_level), 
    to = max(twas_model_data$mean_expression_level), by = 0.01), 
  chromosome = &quot;X&quot;, 
  absolute_sex_bias_in_expression = 0,
  mean_expression_level = mean(twas_model_data$absolute_sex_bias_in_expression))

preds &lt;- data.frame(new, 
                    prop = predict(intersex_early_twas, 
                                   newdata = new, 
                                   se.fit=T, type = &quot;link&quot;)) %&gt;% 
  mutate(y = prop.fit, 
         lwr = prop.fit - 1.96*prop.se.fit, 
         upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), 
         lwr = inv_logit(lwr), 
         upr = inv_logit(upr))

p3 &lt;- preds %&gt;% 
  mutate(panel = &quot;C.&quot;) %&gt;%
  ggplot(aes(mean_expression_level,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  ylab(&quot;Probability of being in the top 1%\nsexually antagonistic transcripts (\u00B1 95% CIs)&quot;) +
  xlab(&quot;Average expression level&quot;) + 
  facet_wrap( ~ panel) + 
  theme_bw() + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12)) +
  theme(strip.background = element_blank(), strip.text = element_text(hjust = 0)) 


blank &lt;- ggplot() + theme_void()

grid.arrange(arrangeGrob(p1, p2, ncol = 2), 
             arrangeGrob(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/ER_figure-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ER_figure-1">
Past versions of ER_figure-1.png
</button>
</p>
<div id="fig-ER_figure-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/14a2316cebcb883f62cef9e6efbbee5b65810110/docs/figure/plot_models_variant_effects.Rmd/ER_figure-1.png" target="_blank">14a2316</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pdf(&quot;figures/fig7_models.pdf&quot;, height = 7, width = 7)
grid.arrange(arrangeGrob(p1, p2, ncol = 2), 
             arrangeGrob(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div id="go-enrichment-of-transcripts-with-extreme-evidence-ratios"
class="section level3 tabset">
<h3 class="tabset">GO enrichment of transcripts with extreme evidence
ratios</h3>
<p>The following code performs a GO enrichment test using the Wilcoxon
method, which tests for enrichment of GO terms at the top and the bottom
of a gene list that has been ordered by some variable. In this case, the
ordering variable is the evidence ratio for early-life sexual
antagonism/concordance that is plotted in the top left of panel B in the
previous figure. Transcripts with a more negative evidence ratio are
more likely to be sexually antagonistic, while those with a more
positive evidence ratio are more likely to be sexually concordant. The
two tables below (accessible by clicking the tabs) illustrate the GO:
Biological Process terms that are most associated with the most sexually
antagonistic and sexually concordant transcripts, according to the
Wilcoxon method. The <code>GOfuncR::go_enrich()</code> function uses
permutation to calculate the ‘family wise error rate’ (FWER) to correct
for multiple testing, and we define significantly enriched GO terms as
those with FWER &lt; 0.05. See the <code>GOfuncR</code> <a
href="https://bioconductor.org/packages/devel/bioc/vignettes/GOfuncR/inst/doc/GOfuncR.html">vignette</a>
for more information on this method.</p>
<pre class="r"><code>library(GOfuncR) # BiocManager::install(&quot;GOfuncR&quot;)
library(org.Dm.eg.db) 

# Get a list of Drosophila genes that have GO terms (can&#39;t analyse genes that do not)
genes_with_GO &lt;- select(org.Dm.eg.db, 
                        keys = keys(org.Dm.eg.db), 
                        columns = c(&quot;SYMBOL&quot;,&quot;GO&quot;)) %&gt;% 
  filter(GO != &quot;&lt;NA&gt;&quot;) %&gt;% dplyr::pull(SYMBOL)

# This code gets the gene symbols for each transcript and sets up the data in the format required by GOfuncR
# for the Wilcoxon test, which works on a gene list that is ordered by a variable 
# (here, the variable is the evidence ratio for early-life sexual antagonism/concordance)
wilcoxon_input &lt;- antagonism_evidence_ratios_twas %&gt;% 
  dplyr::select(FLYBASE = FBID, gene_score = inter_sex_early) %&gt;% 
  left_join(select(org.Dm.eg.db, 
                   keys = keys(org.Dm.eg.db), 
                   columns = c(&quot;SYMBOL&quot;,&quot;FLYBASE&quot;)), by = &quot;FLYBASE&quot;) %&gt;% 
  mutate(gene_score = log2(gene_score)) %&gt;% 
  dplyr::select(SYMBOL, gene_score) %&gt;% 
  dplyr::filter(SYMBOL %in% genes_with_GO) %&gt;% 
  dplyr::arrange(gene_score) %&gt;% as.data.frame()

# Run the GO enrichment test
GO_results_wilcoxon &lt;- go_enrich(wilcoxon_input, orgDb=&#39;org.Dm.eg.db&#39;, test=&#39;wilcoxon&#39;, 
                                 silent = T, domains = &quot;biological_process&quot;,
                                 n_randsets = 1000)


most_antagonistic_genes_GO &lt;- GO_results_wilcoxon$results %&gt;% # most antagonistic
  arrange(FWER_low_rank) %&gt;% 
  filter(raw_p_low_rank &lt; 0.001) %&gt;% 
  dplyr::select(node_id, node_name, raw_p_low_rank, FWER_low_rank) %&gt;% 
  dplyr::rename(GO = node_id, Meaning = node_name, `log10 p` = raw_p_low_rank, FWER = FWER_low_rank) %&gt;% 
  mutate(`log10 p` = format(round(-1 * log10(`log10 p`), 1), nsmall = 1))


most_concordant_genes_GO &lt;- GO_results_wilcoxon$results %&gt;% # most concordant
  filter(FWER_high_rank &lt; 0.05) %&gt;% 
  dplyr::select(node_id, node_name, raw_p_high_rank, FWER_high_rank)  %&gt;% 
  dplyr::rename(GO = node_id, Meaning = node_name, `log10 p` = raw_p_high_rank, FWER = FWER_high_rank) %&gt;% 
  mutate(`log10 p` = format(round(-1 * log10(`log10 p`), 1), nsmall = 1))

saveRDS(most_concordant_genes_GO, &quot;data/derived/supp_table_GO_terms.rds&quot;)</code></pre>
<div id="go-enrichment-among-most-concordant-transcripts"
class="section level4">
<h4>GO enrichment among most concordant transcripts</h4>
<p><strong><em>Table</em></strong>: The table shows all the GO terms
with a family wise error rate (FWER) threshold of FWER &lt; 0.05, when
testing for GO enrichment among genes whose transcripts had a high
evidence ratio (indicating greater likelihood of being sexually
concordant, as measured in the early-life fitness assay). GO terms
related to gene expression and translation were enriched, which suggests
that there is strong, sexually concordant selection on the expression
levels of genes involved in gene expression and translation.</p>
<pre class="r"><code>most_concordant_genes_GO %&gt;% 
  kable() %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
GO
</th>
<th style="text-align:left;">
Meaning
</th>
<th style="text-align:left;">
log10 p
</th>
<th style="text-align:right;">
FWER
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0002181" class="uri">GO:0002181</a>
</td>
<td style="text-align:left;">
cytoplasmic translation
</td>
<td style="text-align:left;">
13.4
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006518" class="uri">GO:0006518</a>
</td>
<td style="text-align:left;">
peptide metabolic process
</td>
<td style="text-align:left;">
11.5
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043603" class="uri">GO:0043603</a>
</td>
<td style="text-align:left;">
cellular amide metabolic process
</td>
<td style="text-align:left;">
10.6
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043043" class="uri">GO:0043043</a>
</td>
<td style="text-align:left;">
peptide biosynthetic process
</td>
<td style="text-align:left;">
9.5
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006412" class="uri">GO:0006412</a>
</td>
<td style="text-align:left;">
translation
</td>
<td style="text-align:left;">
9.3
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043604" class="uri">GO:0043604</a>
</td>
<td style="text-align:left;">
amide biosynthetic process
</td>
<td style="text-align:left;">
7.8
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034641" class="uri">GO:0034641</a>
</td>
<td style="text-align:left;">
cellular nitrogen compound metabolic process
</td>
<td style="text-align:left;">
7.4
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0022613" class="uri">GO:0022613</a>
</td>
<td style="text-align:left;">
ribonucleoprotein complex biogenesis
</td>
<td style="text-align:left;">
5.8
</td>
<td style="text-align:right;">
0.003
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010467" class="uri">GO:0010467</a>
</td>
<td style="text-align:left;">
gene expression
</td>
<td style="text-align:left;">
5.1
</td>
<td style="text-align:right;">
0.011
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006364" class="uri">GO:0006364</a>
</td>
<td style="text-align:left;">
rRNA processing
</td>
<td style="text-align:left;">
5.1
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042254" class="uri">GO:0042254</a>
</td>
<td style="text-align:left;">
ribosome biogenesis
</td>
<td style="text-align:left;">
4.9
</td>
<td style="text-align:right;">
0.018
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042273" class="uri">GO:0042273</a>
</td>
<td style="text-align:left;">
ribosomal large subunit biogenesis
</td>
<td style="text-align:left;">
4.9
</td>
<td style="text-align:right;">
0.018
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016072" class="uri">GO:0016072</a>
</td>
<td style="text-align:left;">
rRNA metabolic process
</td>
<td style="text-align:left;">
4.8
</td>
<td style="text-align:right;">
0.022
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000463" class="uri">GO:0000463</a>
</td>
<td style="text-align:left;">
maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S
rRNA, LSU-rRNA)
</td>
<td style="text-align:left;">
4.7
</td>
<td style="text-align:right;">
0.025
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042274" class="uri">GO:0042274</a>
</td>
<td style="text-align:left;">
ribosomal small subunit biogenesis
</td>
<td style="text-align:left;">
4.5
</td>
<td style="text-align:right;">
0.043
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044271" class="uri">GO:0044271</a>
</td>
<td style="text-align:left;">
cellular nitrogen compound biosynthetic process
</td>
<td style="text-align:left;">
4.5
</td>
<td style="text-align:right;">
0.045
</td>
</tr>
</tbody>
</table>
</div>
<div id="go-enrichment-among-most-anatagonistic-transcripts"
class="section level4">
<h4>GO enrichment among most anatagonistic transcripts</h4>
<p><strong><em>Table</em></strong>: The table shows all the GO terms
with a Wilcoxon test p-value below 0.001, when testing for GO enrichment
among genes whose transcripts had a low evidence ratio (indicating
greater likelihood of being sexually antagonistic, as measured in the
early-life fitness assay). However, no GO terms were significantly
enriched among the most sexually antagonistic genes using family wise
error rate (FWER) threshold of FWER &lt; 0.05, and so these enrichment
results may represent false positives due to multiple testing.</p>
<pre class="r"><code>most_antagonistic_genes_GO %&gt;% 
  kable() %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
GO
</th>
<th style="text-align:left;">
Meaning
</th>
<th style="text-align:left;">
log10 p
</th>
<th style="text-align:right;">
FWER
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0019932" class="uri">GO:0019932</a>
</td>
<td style="text-align:left;">
second-messenger-mediated signaling
</td>
<td style="text-align:left;">
4.5
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030431" class="uri">GO:0030431</a>
</td>
<td style="text-align:left;">
sleep
</td>
<td style="text-align:left;">
4.2
</td>
<td style="text-align:right;">
0.080
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0023052" class="uri">GO:0023052</a>
</td>
<td style="text-align:left;">
signaling
</td>
<td style="text-align:left;">
4.2
</td>
<td style="text-align:right;">
0.091
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007154" class="uri">GO:0007154</a>
</td>
<td style="text-align:left;">
cell communication
</td>
<td style="text-align:left;">
4.2
</td>
<td style="text-align:right;">
0.091
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007165" class="uri">GO:0007165</a>
</td>
<td style="text-align:left;">
signal transduction
</td>
<td style="text-align:left;">
3.8
</td>
<td style="text-align:right;">
0.253
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051239" class="uri">GO:0051239</a>
</td>
<td style="text-align:left;">
regulation of multicellular organismal process
</td>
<td style="text-align:left;">
3.6
</td>
<td style="text-align:right;">
0.358
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034227" class="uri">GO:0034227</a>
</td>
<td style="text-align:left;">
tRNA thio-modification
</td>
<td style="text-align:left;">
3.3
</td>
<td style="text-align:right;">
0.534
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007423" class="uri">GO:0007423</a>
</td>
<td style="text-align:left;">
sensory organ development
</td>
<td style="text-align:left;">
3.3
</td>
<td style="text-align:right;">
0.544
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903047" class="uri">GO:1903047</a>
</td>
<td style="text-align:left;">
mitotic cell cycle process
</td>
<td style="text-align:left;">
3.3
</td>
<td style="text-align:right;">
0.615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034329" class="uri">GO:0034329</a>
</td>
<td style="text-align:left;">
cell junction assembly
</td>
<td style="text-align:left;">
3.2
</td>
<td style="text-align:right;">
0.639
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035556" class="uri">GO:0035556</a>
</td>
<td style="text-align:left;">
intracellular signal transduction
</td>
<td style="text-align:left;">
3.0
</td>
<td style="text-align:right;">
0.803
</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] org.Dm.eg.db_3.16.0  AnnotationDbi_1.60.2 IRanges_2.32.0      
 [4] S4Vectors_0.36.2     Biobase_2.58.0       BiocGenerics_0.44.0 
 [7] GOfuncR_1.18.0       vioplot_0.4.0        zoo_1.8-11          
[10] sm_2.2-5.7.1         staplr_3.1.1         ggpubr_0.6.0        
[13] patchwork_1.1.2      glue_1.6.2           bigsnpr_1.11.6      
[16] bigstatsr_1.5.12     tidybayes_3.0.3      brms_2.18.0         
[19] Rcpp_1.0.10          car_3.1-1            carData_3.0-5       
[22] RColorBrewer_1.1-3   cowplot_1.1.1        kableExtra_1.3.4    
[25] mashr_0.2.69         ashr_2.2-54          showtext_0.9-5      
[28] showtextdb_3.0       sysfonts_0.8.8       Hmisc_4.7-2         
[31] Formula_1.2-4        survival_3.4-0       lattice_0.20-45     
[34] ggbeeswarm_0.7.1     qqman_0.1.8          gridExtra_2.3       
[37] lubridate_1.9.2      forcats_1.0.0        stringr_1.5.0       
[40] dplyr_1.1.0          purrr_1.0.1          readr_2.1.4         
[43] tidyr_1.3.0          tibble_3.1.8         ggplot2_3.4.1       
[46] tidyverse_2.0.0      workflowr_1.7.0.4   

loaded via a namespace (and not attached):
  [1] bigreadr_0.2.5         coda_0.19-4            ragg_1.2.5            
  [4] bit64_4.0.5            knitr_1.42             irlba_2.3.5.1         
  [7] dygraphs_1.1.1.6       data.table_1.14.6      rpart_4.1.19          
 [10] inline_0.3.19          RCurl_1.98-1.9         KEGGREST_1.38.0       
 [13] doParallel_1.0.17      generics_0.1.3         callr_3.7.3           
 [16] RSQLite_2.3.0          bit_4.0.5              tzdb_0.3.0            
 [19] webshot_0.5.4          xml2_1.3.3             httpuv_1.6.9          
 [22] StanHeaders_2.21.0-7   assertthat_0.2.1       isoband_0.2.7         
 [25] xfun_0.37              hms_1.1.2              ggdist_3.2.1          
 [28] jquerylib_0.1.4        bayesplot_1.10.0       rJava_1.0-6           
 [31] evaluate_0.20          promises_1.2.0.1       fansi_1.0.4           
 [34] dbplyr_2.3.0           igraph_1.3.5           DBI_1.1.3             
 [37] htmlwidgets_1.6.1      tensorA_0.36.2         ellipsis_0.3.2        
 [40] crosstalk_1.2.0        backports_1.4.1        calibrate_1.7.7       
 [43] markdown_1.4           RcppParallel_5.1.6     deldir_1.0-6          
 [46] vctrs_0.5.2            abind_1.4-5            cachem_1.0.7          
 [49] withr_2.5.0            checkmate_2.1.0        vroom_1.6.1           
 [52] xts_0.12.2             prettyunits_1.1.1      svglite_2.1.1         
 [55] cluster_2.1.4          crayon_1.5.2           pkgconfig_2.0.3       
 [58] labeling_0.4.2         GenomeInfoDb_1.34.9    nlme_3.1-160          
 [61] vipor_0.4.5            nnet_7.3-18            rlang_1.0.6           
 [64] lifecycle_1.0.3        miniUI_0.1.1.1         colourpicker_1.2.0    
 [67] invgamma_1.1           distributional_0.3.1   tcltk_4.2.2           
 [70] rprojroot_2.0.3        matrixStats_0.63.0     rngtools_1.5.2        
 [73] Matrix_1.5-1           loo_2.5.1              base64enc_0.1-3       
 [76] beeswarm_0.4.0         whisker_0.4.1          processx_3.8.0        
 [79] png_0.1-8              viridisLite_0.4.1      bitops_1.0-7          
 [82] getPass_0.2-2          Biostrings_2.66.0      blob_1.2.3            
 [85] doRNG_1.8.6            mixsqp_0.3-48          SQUAREM_2021.1        
 [88] parallelly_1.34.0      mapplots_1.5.1         jpeg_0.1-10           
 [91] shinystan_2.6.0        rstatix_0.7.2          ggsignif_0.6.4        
 [94] rmeta_3.0              scales_1.2.1           memoise_2.0.1         
 [97] magrittr_2.0.3         plyr_1.8.8             hexbin_1.28.2         
[100] zlibbioc_1.44.0        threejs_0.3.3          compiler_4.2.2        
[103] rstantools_2.2.0       flock_0.7              cli_3.6.0             
[106] XVector_0.38.0         ps_1.7.2               bigsparser_0.6.1      
[109] Brobdingnag_1.2-9      htmlTable_2.4.1        MASS_7.3-58.1         
[112] mgcv_1.8-41            tidyselect_1.2.0       stringi_1.7.12        
[115] textshaping_0.3.6      highr_0.10             yaml_2.3.7            
[118] svUnit_1.0.6           latticeExtra_0.6-30    bridgesampling_1.1-2  
[121] sass_0.4.5             tools_4.2.2            bigassertr_0.1.6      
[124] timechange_0.2.0       parallel_4.2.2         rstudioapi_0.14       
[127] foreach_1.5.2          foreign_0.8-83         git2r_0.31.0          
[130] rmio_0.4.0             posterior_1.3.1        farver_2.1.1          
[133] digest_0.6.31          shiny_1.7.4            ff_4.0.9              
[136] GenomicRanges_1.50.2   broom_1.0.3            later_1.3.0           
[139] bigparallelr_0.3.2     httr_1.4.5             colorspace_2.1-0      
[142] rvest_1.0.3            fs_1.6.1               truncnorm_1.0-8       
[145] splines_4.2.2          shinythemes_1.2.0      systemfonts_1.0.4     
[148] xtable_1.8-4           jsonlite_1.8.4         rstan_2.21.8          
[151] R6_2.5.1               pillar_1.8.1           htmltools_0.5.4       
[154] mime_0.12              fastmap_1.1.1          DT_0.27               
[157] codetools_0.2-18       pkgbuild_1.4.0         mvtnorm_1.1-3         
[160] utf8_1.2.2             bslib_0.4.2            arrayhelpers_1.1-0    
[163] curl_5.0.0             gtools_3.9.4           shinyjs_2.1.0         
[166] interp_1.1-3           rmarkdown_2.20         munsell_0.5.0         
[169] GenomeInfoDbData_1.2.9 iterators_1.0.14       reshape2_1.4.4        
[172] gtable_0.3.1          </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
