<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Plots and models of variant effect sizes</title>

<script src="site_libs/jquery-3.5.0/jquery-3.5.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Plots and models of variant effect sizes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreed573064c5223b276f7abcd63aa97f01e6f19643atargetblankd573064a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/d573064c5223b276f7abcd63aa97f01e6f19643a" target="_blank">d573064</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreed573064c5223b276f7abcd63aa97f01e6f19643atargetblankd573064a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/d573064c5223b276f7abcd63aa97f01e6f19643a" target="_blank">d573064</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store

Untracked files:
    Untracked:  big_model.rds
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds

Unstaged changes:
    Modified:   .gitignore
    Modified:   figures/fig2_SNPs_manhattan_plot.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plot_models_variant_effects.Rmd</code>) and HTML (<code>docs/plot_models_variant_effects.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/d573064c5223b276f7abcd63aa97f01e6f19643a/analysis/plot_models_variant_effects.Rmd" target="_blank">d573064</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-23
</td>
<td>
wflow_publish(“analysis/plot_models_variant_effects.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/652106368c95ac4a04b2383d560d3c8073990f18/docs/plot_models_variant_effects.html" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/9deec2d2a0b6fabe844d63c9d77c62841b5cc81e/analysis/plot_models_variant_effects.Rmd" target="_blank">9deec2d</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
<td>
wflow_publish(“analysis/plot_models_variant_effects.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/5a59d9f982db9a933d0cea448822677a000087c9/docs/plot_models_variant_effects.html" target="_blank">5a59d9f</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/de681496ef6dbd7c9cde9f258aba20fedaecf37f/analysis/plot_models_variant_effects.Rmd" target="_blank">de68149</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
<td>
wflow_publish(“analysis/plot_models_variant_effects.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/plot_models_variant_effects.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/4953d582430e0be9ac5f56212ce2e1ee0478d936/docs/plot_models_variant_effects.html" target="_blank">4953d58</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/2bf87505a56e12debcab79eb42fd1313bc1b2f31/analysis/plot_models_variant_effects.Rmd" target="_blank">2bf8750</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/plot_models_variant_effects.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/plot_models_variant_effects.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/871ae81fa56175675b53d22ca29bf9323c3ed50a/docs/plot_models_variant_effects.html" target="_blank">871ae81</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/plot_models_variant_effects.html" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/c606d3d8748d48cc6dd479febb9ce71effc8bcfd/analysis/plot_models_variant_effects.Rmd" target="_blank">c606d3d</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big first commit 2021
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(gridExtra)
library(qqman)
library(ggbeeswarm)
library(Hmisc)
library(showtext) # For fancy Google font in figures
library(mashr)
library(kableExtra)
library(cowplot)
library(grid)
library(RColorBrewer)
font_add_google(name = &quot;Raleway&quot;, family = &quot;Raleway&quot;, regular.wt = 400, bold.wt = 700) # Install font from Google Fonts
showtext_auto()

db &lt;- DBI::dbConnect(RSQLite::SQLite(), 
                     &quot;data/derived/annotations.sqlite3&quot;)

# Results for all 1,613,615 SNPs, even those that are in 100% LD with others (these are grouped up by the SNP_clump column)
all_snps &lt;- tbl(db, &quot;univariate_lmm_results&quot;)

# All SNPs and SNP groups that are in &lt;100% LD with one another (n = 1,207,357)
SNP_clumps &lt;- all_snps %&gt;% select(-SNP) %&gt;% distinct() %&gt;% collect(n = Inf)

# Subsetting variable to get the approx-LD subset of SNPs
LD_subset &lt;- !is.na(SNP_clumps$LFSR_female_early_mashr_ED)</code></pre>
<div id="load-and-clean-the-variant-effect-sizes" class="section level2">
<h2>Load and clean the variant effect sizes</h2>
<p>Modify column names, reorder columns, filter based on the p-value, etc.</p>
<pre class="r"><code># Univariate analysis, mashr-adjusted, data-driven approach
univariate_lmm_results &lt;- tbl(db, &quot;univariate_lmm_results&quot;) %&gt;% 
  select(-contains(&quot;canonical&quot;), 
         -contains(&quot;raw&quot;)) %&gt;% 
  inner_join(tbl(db, &quot;variants&quot;) %&gt;% 
               select(SNP, FBID, site.class, distance.to.gene, MAF), 
             by = &quot;SNP&quot;) %&gt;%
  left_join(
    tbl(db, &quot;genes&quot;) %&gt;% 
      select(FBID, gene_name), by = &quot;FBID&quot;) %&gt;%
  collect(n = Inf) %&gt;%
  rename_all(~ gsub(&quot;beta_&quot;, &quot;&quot;, .x)) %&gt;%
  rename_all(~ gsub(&quot;_mashr_ED&quot;, &quot;&quot;, .x))

univariate_lmm_results &lt;- univariate_lmm_results %&gt;%
  mutate(site.class = gsub(&quot;5_&quot;, &quot;5-&quot;, site.class),
         site.class = gsub(&quot;3_&quot;, &quot;3-&quot;, site.class),
         site.class = gsub(&quot;NON_&quot;, &quot;NON-&quot;, site.class),
         site.class = gsub(&quot;_&quot;, &quot; &quot;, site.class),
         site.class = capitalize(tolower(site.class)),
         site.class = gsub(&quot;Utr&quot;, &quot;UTR&quot;, site.class)
  )</code></pre>
</div>
<div id="q-q-plots" class="section level2 tabset">
<h2>Q-Q plots</h2>
<p>Here are some quantile-quantile plots, which are commonly used to check GWAS results, and to test the hypothesis that there are more SNPs than expected showing large effects on the trait of interest. There is an excess of loci with effects on female fitness, but not much of a visible excess for males. These plots use the raw <span class="math inline">\(p\)</span>-values results from GEMMA.</p>
<div id="female-early-life-fitness" class="section level3">
<h3>Female early-life fitness</h3>
<pre class="r"><code>univariate_lmm_pvals &lt;- SNP_clumps %&gt;%
  select(contains(&quot;pvalue&quot;)) %&gt;% filter(LD_subset) %&gt;% as.data.frame()
qqman::qq(univariate_lmm_pvals$pvalue_female_early_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot1-1">
Past versions of qqplot1-1.png
</button>
</p>
<div id="fig-qqplot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot1-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="female-late-life-fitness" class="section level3">
<h3>Female late-life fitness</h3>
<pre class="r"><code>qqman::qq(univariate_lmm_pvals$pvalue_female_late_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot2-1">
Past versions of qqplot2-1.png
</button>
</p>
<div id="fig-qqplot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot2-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="male-early-life-fitness" class="section level3">
<h3>Male early-life fitness</h3>
<pre class="r"><code>qqman::qq(univariate_lmm_pvals$pvalue_male_early_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot3-1">
Past versions of qqplot3-1.png
</button>
</p>
<div id="fig-qqplot3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot3-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="male-late-life-fitness" class="section level3">
<h3>Male late-life fitness</h3>
<pre class="r"><code>qqman::qq(univariate_lmm_pvals$pvalue_male_late_raw)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/qqplot4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qqplot4-1">
Past versions of qqplot4-1.png
</button>
</p>
<div id="fig-qqplot4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/qqplot4-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="hex-bin-plots-and-correlations-in-variant-effect-sizes" class="section level2">
<h2>Hex bin plots and correlations in variant effect sizes</h2>
<div id="effect-sizes-adjusted-using-data-driven-adaptive-shrinkage-in-mashr" class="section level3 tabset">
<h3>Effect sizes adjusted using ‘data-driven’ adaptive shrinkage in <code>mashr</code></h3>
<div id="plot" class="section level4">
<h4>Plot</h4>
<pre class="r"><code>hex_plot &lt;- function(x, y, xlab, ylab, title,
                     xlim = c(-0.5, 0.5),
                     ylim = c(-0.3, 0.3)
                     ){
  dat &lt;- SNP_clumps %&gt;%
    mutate(facet = title) %&gt;% 
    filter(LD_subset) # ensure only the LD SNPs from mashr are plotted
  ggplot(dat, aes_string(x, y)) + 
    geom_abline(linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 3) +
    geom_hline(yintercept = 0, linetype = 3) +
    stat_binhex(bins = 50)  +
    geom_density_2d(colour = &quot;white&quot;, alpha = 0.6) + 
    scale_fill_viridis_c() + 
    coord_cartesian(xlim = xlim, ylim = ylim) + 
    facet_wrap(~ facet) +
    theme_bw() + xlab(xlab) + ylab(ylab) +
    theme(legend.position = &quot;none&quot;,
          strip.background = element_blank(),
          strip.text = element_text(hjust=0))   + 
    theme(text = element_text(family = &quot;Raleway&quot;, size = 12))

}

p1 &lt;- hex_plot(&quot;beta_female_early_mashr_ED&quot;, 
               &quot;beta_male_early_mashr_ED&quot;, 
               &quot;Effect on female fitness&quot;, 
               NULL,
               &quot;A. Early life fitness&quot;)

p2 &lt;- hex_plot(&quot;beta_female_late_mashr_ED&quot;, 
               &quot;beta_male_late_mashr_ED&quot;, 
               &quot;Effect on female fitness&quot;, 
               NULL,
               &quot;B. Late life fitness&quot;)

grid.arrange(p1, p2, ncol = 2, left = &quot;Effect on male fitness&quot;)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/hex1-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hex1-1">
Past versions of hex1-1.png
</button>
</p>
<div id="fig-hex1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/hex1-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 2</strong>: Effect sizes of 1207357 loci (i.e. groups of one or more polymorphic sites in complete linkage disequilibrium) on male and female fitness, plotted separately for the early-life and late-life estimates. The effect sizes estimated using GEMMA have been corrected using <code>mashr</code>, using the data-driven method to apply shrinkage (Figure SX shows the raw estimates). The data have been binned into hexagons, with the colour and contour lines indicating the number of loci. The diagonal line represents <span class="math inline">\(y=x\)</span>. Positive effect sizes indicate that the minor allele is associated with higher fitness.</p>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/5a59d9f982db9a933d0cea448822677a000087c9/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-3-1.png" target="_blank">5a59d9f</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pearson-correlation" class="section level4">
<h4>Pearson correlation</h4>
<pre class="r"><code>SNP_clumps %&gt;%
  select(contains(&quot;mashr_ED&quot;)) %&gt;%
  select(contains(&quot;beta&quot;)) %&gt;%
  rename_all(~ paste(ifelse(str_detect(.x, &quot;female&quot;), &quot;Female&quot;, &quot;Male&quot;), 
                     ifelse(str_detect(.x, &quot;early&quot;), &quot;early&quot;, &quot;late&quot;))) %&gt;%
  cor(use = &quot;pairwise.complete.obs&quot;) %&gt;% 
  kable(digits = 3) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female early
</th>
<th style="text-align:right;">
Female late
</th>
<th style="text-align:right;">
Male early
</th>
<th style="text-align:right;">
Male late
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female early
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.998
</td>
<td style="text-align:right;">
0.911
</td>
<td style="text-align:right;">
0.950
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late
</td>
<td style="text-align:right;">
0.998
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.880
</td>
<td style="text-align:right;">
0.927
</td>
</tr>
<tr>
<td style="text-align:left;">
Male early
</td>
<td style="text-align:right;">
0.911
</td>
<td style="text-align:right;">
0.880
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.994
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late
</td>
<td style="text-align:right;">
0.950
</td>
<td style="text-align:right;">
0.927
</td>
<td style="text-align:right;">
0.994
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
</tbody>
</table>
<!-- ### Effect sizes adjusted using 'canonical' adaptive shrinkage {.tabset} -->
<!-- #### Plot -->
<!-- The unusual distribution here reflects the fact that this analysis imposed constraints on the effects of each locus. For example, loci assigned to the 'Female-specific' category with high probability necessarily have an effect size close to zero in males. The figure should be interpreted cautiously - the main purpose of the canonical analysis is to inference the mixture proportions, and to assign mixture probabilities to each locus, not to accurately estimate the true effect size of each variant.  -->
<!-- ```{r hex2, message=FALSE, warning=FALSE, fig.showtext=TRUE, fig.height = 5, fig.width = 9} -->
<!-- p1 <- hex_plot("beta_female_early_mashr_canonical",  -->
<!--            "beta_male_early_mashr_canonical",  -->
<!--            "Effect on female fitness",  -->
<!--            NULL, -->
<!--            "A. Early life fitness") -->
<!-- p2 <- hex_plot("beta_female_late_mashr_canonical",  -->
<!--            "beta_male_late_mashr_canonical",  -->
<!--            "Effect on female fitness",  -->
<!--            NULL, -->
<!--            "B. Late life fitness") -->
<!-- grid.arrange(p1, p2, ncol = 2, left = "Effect on male fitness") -->
<!-- ``` -->
<!-- #### Pearson correlation -->
<!-- ```{r} -->
<!-- SNP_clumps %>% -->
<!--   select(contains("mashr_canonical")) %>% -->
<!--   select(contains("beta")) %>% -->
<!--   rename_all(~ paste(ifelse(str_detect(.x, "female"), "Female", "Male"),  -->
<!--                      ifelse(str_detect(.x, "early"), "early", "late"))) %>% -->
<!--   cor(use = "pairwise.complete.obs") %>%  -->
<!--   kable(digits = 3) %>% kable_styling(full_width = FALSE) -->
<!-- ``` -->
</div>
</div>
<div id="unadjusted-effect-sizes" class="section level3 tabset">
<h3>Unadjusted effect sizes</h3>
<div id="plot-1" class="section level4">
<h4>Plot</h4>
<p>Uses the variant effect sizes output by GEMMA, without applying any shrinkage (i.e. this is the data that was adjusted using <code>mashr</code>).</p>
<pre class="r"><code>p1 &lt;- hex_plot(&quot;beta_female_early_raw&quot;, 
           &quot;beta_male_early_raw&quot;, 
           &quot;Effect on female fitness&quot;, 
           NULL,
           &quot;A. Early life fitness&quot;,
           xlim = c(-1, 1),
           ylim = c(-1, 1))

p2 &lt;- hex_plot(&quot;beta_female_late_raw&quot;, 
           &quot;beta_male_late_raw&quot;, 
           &quot;Effect on female fitness&quot;, 
           NULL,
           &quot;B. Late life fitness&quot;,
           xlim = c(-1, 1),
           ylim = c(-1, 1))

grid.arrange(p1, p2, ncol = 2, left = &quot;Effect on male fitness&quot;)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/hex3-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hex3-1">
Past versions of hex3-1.png
</button>
</p>
<div id="fig-hex3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/hex3-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pearson-correlation-1" class="section level4">
<h4>Pearson correlation</h4>
<pre class="r"><code>SNP_clumps %&gt;%
  select(contains(&quot;raw&quot;)) %&gt;%
  select(contains(&quot;beta&quot;)) %&gt;%
  rename_all(~ paste(ifelse(str_detect(.x, &quot;female&quot;), &quot;Female&quot;, &quot;Male&quot;), 
                     ifelse(str_detect(.x, &quot;early&quot;), &quot;early&quot;, &quot;late&quot;))) %&gt;%
  cor(use = &quot;pairwise.complete.obs&quot;) %&gt;% 
  kable(digits = 3) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female early
</th>
<th style="text-align:right;">
Female late
</th>
<th style="text-align:right;">
Male early
</th>
<th style="text-align:right;">
Male late
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female early
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.567
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:right;">
0.118
</td>
</tr>
<tr>
<td style="text-align:left;">
Female late
</td>
<td style="text-align:right;">
0.567
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
0.167
</td>
</tr>
<tr>
<td style="text-align:left;">
Male early
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.434
</td>
</tr>
<tr>
<td style="text-align:left;">
Male late
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:right;">
0.167
</td>
<td style="text-align:right;">
0.434
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="average-effect-sizes-are-negative" class="section level2 tabset">
<h2>Average effect sizes are negative</h2>
<p>Each of the following four tests is an intercept-only linear model, weighted by the inverse of the standard error for the focal variant’s effect size (so, loci where the effect effect size was measured with more precision are weighted more heavily). The tests are run on the LD-pruned subset of SNPs, minimising pseudoreplication. A non-zero intercept term indicates that major (or minor) alleles tend to have consistently positive or negative associations with the focal fitness trait.</p>
<p>These results indicate that the minor allele tends to <em>reduce</em> fitness, relative to the major allele. It’s a weak effect (note the small value in the Estimate column), this may reflect the large uncertainty with which the effect sizes are measured, in addition to a true biological result that most loci have little or no relationship with the fitness traits we measured.</p>
<div id="female-early-life" class="section level3">
<h3>Female early-life</h3>
<pre class="r"><code>summary(lm(beta_female_early_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_female_early_raw))</code></pre>
<pre><code>
Call:
lm(formula = beta_female_early_raw ~ 1, data = SNP_clumps %&gt;% 
    filter(LD_subset), weights = 1/SE_female_early_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.98458 -0.22630  0.00919  0.23696  1.48781 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.0016820  0.0002555  -6.584 4.58e-11 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3497 on 208986 degrees of freedom</code></pre>
</div>
<div id="female-late-life" class="section level3">
<h3>Female late-life</h3>
<pre class="r"><code>summary(lm(beta_female_late_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_female_late_raw))</code></pre>
<pre><code>
Call:
lm(formula = beta_female_late_raw ~ 1, data = SNP_clumps %&gt;% 
    filter(LD_subset), weights = 1/SE_female_late_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.94128 -0.23405  0.00409  0.23909  1.47335 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.0025016  0.0002598   -9.63   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3549 on 208986 degrees of freedom</code></pre>
</div>
<div id="male-early-life" class="section level3">
<h3>Male early-life</h3>
<pre class="r"><code>summary(lm(beta_male_early_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_male_early_raw))</code></pre>
<pre><code>
Call:
lm(formula = beta_male_early_raw ~ 1, data = SNP_clumps %&gt;% filter(LD_subset), 
    weights = 1/SE_male_early_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.58868 -0.23295 -0.00219  0.23114  1.76956 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.0016140  0.0002542   -6.35 2.15e-10 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3503 on 208986 degrees of freedom</code></pre>
</div>
<div id="male-late-life" class="section level3">
<h3>Male late-life</h3>
<pre class="r"><code>summary(lm(beta_male_late_raw ~ 1, 
           data = SNP_clumps %&gt;% filter(LD_subset), 
           weights = 1 / SE_male_late_raw))</code></pre>
<pre><code>
Call:
lm(formula = beta_male_late_raw ~ 1, data = SNP_clumps %&gt;% filter(LD_subset), 
    weights = 1/SE_male_late_raw)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-1.89125 -0.22874  0.00114  0.23182  1.59205 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.001943   0.000251  -7.739    1e-14 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3479 on 208986 degrees of freedom</code></pre>
</div>
</div>
<div id="manhattan-plot" class="section level2">
<h2>Manhattan plot</h2>
<p><strong>Figure legend</strong>: Manhattan plot showing the p-values (-log_{10} transformed) for each variant, from mixed model GWAS of female (top) and male (bottom) early-life fitness testing the null hypothesis that the two alleles are associated with equal fitness values.</p>
<pre class="r"><code>manhattan_data &lt;- tbl(db, &quot;univariate_lmm_results&quot;) %&gt;% 
  select(SNP, starts_with(&quot;P_&quot;),
         pvalue_female_early_raw, pvalue_male_early_raw) %&gt;%
  distinct() %&gt;%
  collect(n=Inf) %&gt;%
  mutate(position = str_split(SNP, &quot;_&quot;),
         chr = map_chr(position, ~ .x[1]),
         position = as.numeric(map_chr(position, ~ .x[2]))) %&gt;% 
         # P_pleiotropy = 1 - P_null - P_female_specific - P_male_specific,
         # SA_not_concord = P_sex_antag &gt; P_equal_effects,
         # top_SA = P_sex_antag &gt;= quantile(P_sex_antag, probs=0.9999)) %&gt;% 
  filter(chr != &quot;4&quot;) 

max_pos &lt;- manhattan_data %&gt;%
  group_by(chr) %&gt;%
  summarise(max_pos = max(position), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()
max_pos$max_pos &lt;- c(0, cumsum(max_pos$max_pos[1:4]))

manhattan_data &lt;- manhattan_data %&gt;%
    left_join(max_pos, by = &quot;chr&quot;) %&gt;%
    mutate(position = position + max_pos) 


p1 &lt;- manhattan_data %&gt;%
  ggplot(aes(position, -1 * log10(pvalue_female_early_raw), group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  # geom_point(data = manhattan_data %&gt;% filter(top_SA), 
  #            aes(y = 8), size = 3, pch = 6) +
  scale_colour_brewer(palette = &quot;Paired&quot;, name = &quot;Chromosome&quot;) + 
  scale_y_continuous(limits = c(0,8)) +
  ylab(expression(paste(&quot;Effect on female early-life fitness (-&quot;, Log[10], &quot; p)&quot;))) + 
  xlab(&quot;&quot;) +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        # text = element_text(family = &quot;Raleway&quot;, size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

p2 &lt;- manhattan_data %&gt;%
  ggplot(aes(position, -1 * log10(pvalue_male_early_raw), group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  scale_colour_brewer(palette = &quot;Paired&quot;, name = &quot;Chromosome&quot;) + 
  ylab(expression(paste(&quot;Effect on male early-life fitness (-&quot;, Log[10], &quot; p)&quot;))) + 
  xlab(&quot;Position&quot;) +
  scale_y_reverse(limits = c(8,0)) +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        # text = element_text(family = &quot;Raleway&quot;, size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

grid_arrange_shared_legend &lt;- function(..., ncol = length(list(...)), nrow = 1) {

  plots &lt;- list(...)
  # position &lt;- match.arg(position)
  g &lt;- ggplotGrob(plots[[1]] + theme(legend.position = &quot;right&quot;))$grobs
  legend &lt;- g[[which(sapply(g, function(x) x$name) == &quot;guide-box&quot;)]]
  lheight &lt;- sum(legend$height)
  lwidth &lt;- sum(legend$width)
  gl &lt;- lapply(plots, function(x) x + theme(legend.position=&quot;none&quot;))
  gl &lt;- c(gl, ncol = ncol, nrow = nrow)
  
  combined &lt;- arrangeGrob(arrangeGrob(gl[[1]], gl[[2]], nrow = 2),
                          legend,
                          ncol = 2,
                          widths = unit.c(unit(1, &quot;npc&quot;) - lwidth, lwidth))
  
  grid.newpage()
  grid.draw(combined)
  invisible(combined) # return gtable invisibly
}

grid_arrange_shared_legend(p1, p2)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-10-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="investigating-pleiotropy-and-polygenicity-of-fitness" class="section level2">
<h2>Investigating pleiotropy and polygenicity of fitness</h2>
<p>Inspired by Boyle et al. 2017 (<em>Cell</em>, specifically the analysis in their Figure 1C), we sorted all the variants in order of effect size on female fitness, placed the variants in bins of 1000, and then calculated the average effect size for each bin for both male and female fitness. This analysis was performed on the raw SNP effect sizes from GEMMA, pruned to a set of SNPs in approximate LD with one another.</p>
<p>If there is pleiotropy between male and female fitness, we would predict a correlation between the effect sizes for male and female fitness; on the contrary if there were no pleiotropy, we would see no correlation.</p>
<p>Moreover, if fitness is highly polygenic (‘omnigenic’; Boyle et al. 2017), we predict a tight, straight line relationship, because each bin would contain some variants with small but genuine associations with fitness, and these effects would replicate in the other fitness trait. On the contrary if genetic variance in fitness stems from just a few genes with larger effects, the relationship between male and female fitness would be flat in the center and sloped at the extremes.</p>
<p>Figure 4 shows that there was a very tight correlation between the average effects of the variants in each bin on male and female fitness. The data therefore suggest that variants associated with male fitness tend to also affect female fitness (in the same direction), and that a very large number of loci have small, concordant effects on fitness in both sexes.</p>
<p>Interestingly there appears to be a curve in Figure 4A to the left of the x-axis, indicating that variants where the minor allele is strongly, negatively associated with female fitness are (on average) less negatively associated with male fitness than expected based on predictions from variants with weaker effects on female fitness. One possible explanation is that alleles that are highly highly detrimental to both sexes are usually purged by selection (or at least kept below the 5% MAF threshold that we used), whereas female-harming alleles that are neutral or beneficial in males are purged less often, resulting a greater proportion of female-limited or sexually antagonistic alleles towards the left of the x-axis.</p>
<pre class="r"><code>p1 &lt;- SNP_clumps %&gt;%
  filter(LD_subset) %&gt;% # The figure looks the same whether or not the data are thinned to the LD set of SNPs
  arrange(beta_female_early_raw) %&gt;%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         age = &quot;A. Early life&quot;) %&gt;%
  group_by(bin, age) %&gt;%
  summarise(females = mean(beta_female_early_raw), males = mean(beta_male_early_raw)) %&gt;%
  ggplot(aes(females, males)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  geom_point() +
  stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 0.6) +
  facet_wrap(~ age) +
  scale_y_continuous(limits = c(-0.1, 0.2)) +
  xlab(NULL) + ylab(NULL) + 
  theme_bw() +
  theme(strip.background = element_blank(),
        strip.text = element_text(hjust=0))   + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12))

p2 &lt;- SNP_clumps %&gt;%
  filter(LD_subset) %&gt;% 
  arrange(beta_female_late_raw) %&gt;%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         age = &quot;B. Late life&quot;) %&gt;%
  group_by(bin, age) %&gt;%
  summarise(females = mean(beta_female_late_raw), males = mean(beta_male_late_raw)) %&gt;%
  ggplot(aes(females, males)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  geom_point() +
  stat_smooth(method = &quot;lm&quot;, formula = y ~ x + I(x^2), size = 0.6) +
  facet_wrap(~ age) +
  scale_y_continuous(limits = c(-0.1, 0.2)) +
  xlab(NULL) + ylab(NULL) + 
  theme_bw() +
  theme(strip.background = element_blank(),
        strip.text = element_text(hjust=0))   + 
  theme(text = element_text(family = &quot;Raleway&quot;, size = 12))

grid.arrange(p1, p2, nrow = 1, bottom = &quot;Mean effect size on female fitness&quot;, 
  left = &quot;Mean effect size on male fitness&quot;)</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/5a59d9f982db9a933d0cea448822677a000087c9/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">5a59d9f</a>
</td>
<td>
lukeholman
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-11-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>quartz_off_screen 
                2 </code></pre>
<!-- ## Effect of the minor allele -->
<!-- ```{r echo=FALSE} -->
<!-- n_loci <- tbl(db, "univariate_lmm_results") %>% pull(P_sex_antag) %>% length() -->
<!-- fraction_n <- floor(n_loci*0.001) -->
<!-- n_loci <- prettyNum(n_loci, big.mark = ",", scientific = FALSE) -->
<!-- fraction_n <- prettyNum(fraction_n, big.mark = ",", scientific = FALSE) -->
<!-- ``` -->
<!-- Evolutionary theory makes testable predictions about the effects of the major and minor alleles on fitness-related phenotypes. For example, the minor allele should generally be associated with the lower-fitness phenotype due to selection. Furthermore, it is interesting to ask whether the female-beneficial allele or the male-beneficial allele tends to be the major allele at sexually-antagonistic loci, as this reveals something about which sex is under stronger selection in the wild (provided that our lab-based fitness assays are somewhat correlated with natural selection). Finally, we might expect there to be a difference in effect sizes between loci where the minor allele is quite rare, compared to loci where the minor allele is quite common, for a few reasons. Selection should keep MAF low at loci with a strong (sex-averaged) fitness effect rare.  -->
<!-- To address these questions graphically, we here plot the effects on male and female early-life fitness of the minor allele, either for A) all loci in the LD-pruned set, B) the top 1% most sexually-antagonistic loci in this set, and C) the top 1% most sexually concordant loci in the set. For the latter two cases, we used the evidence ratios calculated above to rank loci from most to least sexually antagonistic (or concordant). The data are further subdivided into loci with lower MAF (between 0.05 and 0.1) and higher MAF (above 0.1). The plot shows the `mashr`-adjusted effect sizes, and considers the effect on early-life fitness in males and females (the corresponding plot for late-life fitness is very similar). -->
<!-- Considering all loci, most alleles have an estimated effect size close to zero, and there is higher variance for male fitness -->
<!-- ```{r} -->
<!-- histo_data <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_early < quantile(inter_sex_early, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_early > quantile(inter_sex_early, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_early_mashr_ED, beta_male_early_mashr_ED, class, MAF) -->
<!-- all_loci <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_early_mashr_ED, beta_male_early_mashr_ED, class, MAF) -->
<!-- bind_rows(histo_data, all_loci) %>%  -->
<!--   filter(!is.na(MAF)) %>%  -->
<!--   arrange(MAF, class) %>%  -->
<!--   mutate(facet = paste(class, MAF, sep = "\n"), -->
<!--          facet = factor(facet, unique(facet))) %>%  -->
<!-- ggplot(aes(beta_female_early_mashr_ED)) +  -->
<!--   geom_vline(xintercept = 0, linetype = 2) + -->
<!--   geom_histogram(bins=30, fill="tomato", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(aes(beta_male_early_mashr_ED), bins=30, fill="steelblue", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(bins=30, fill=NA, colour="black") +  -->
<!--   geom_histogram(aes(beta_male_early_mashr_ED), bins=30,fill=NA,colour="black") +  -->
<!--   facet_wrap(~ facet, scales = "free_y") + -->
<!--   xlab("Effect of the minor allele on early-life fitness\n(females:red, males:blue)") +  -->
<!--   theme_bw() + -->
<!--   theme(legend.position = "none", -->
<!--         strip.background = element_blank(),  -->
<!--         strip.text = element_text(hjust = 0), -->
<!--         text = element_text(family = "Raleway", size = 12)) + -->
<!--   ylab("Number of loci") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- histo_data <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_late < quantile(inter_sex_late, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_late > quantile(inter_sex_late, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_late_mashr_ED, beta_male_late_mashr_ED, class, MAF) -->
<!-- all_loci <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_late_mashr_ED, beta_male_late_mashr_ED, class, MAF) -->
<!-- bind_rows(histo_data, all_loci) %>%  -->
<!--   filter(!is.na(MAF)) %>%  -->
<!--   arrange(MAF, class) %>%  -->
<!--   mutate(facet = paste(class, MAF, sep = "\n"), -->
<!--          facet = factor(facet, unique(facet))) %>%  -->
<!--   ggplot(aes(beta_female_late_mashr_ED)) +  -->
<!--   geom_vline(xintercept = 0, linetype = 2) + -->
<!--   geom_histogram(bins=30, fill="tomato", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(aes(beta_male_late_mashr_ED), bins=30, fill="steelblue", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(bins=30, fill=NA, colour="black") +  -->
<!--   geom_histogram(aes(beta_male_late_mashr_ED), bins=30,fill=NA,colour="black") +  -->
<!--   facet_wrap(~ facet, scales = "free_y") + -->
<!--   xlab("Effect of the minor allele on late-life fitness\n(females:red, males:blue)") +  -->
<!--   theme_bw() + -->
<!--   theme(legend.position = "none", -->
<!--         strip.background = element_blank(),  -->
<!--         strip.text = element_text(hjust = 0), -->
<!--         text = element_text(family = "Raleway", size = 12)) + -->
<!--   ylab("Number of loci") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- histo_data_early <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_early < quantile(inter_sex_early, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_early > quantile(inter_sex_early, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_early_mashr_ED, beta_male = beta_male_early_mashr_ED, class, MAF) -->
<!-- all_loci_early <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_early_mashr_ED, beta_male = beta_male_early_mashr_ED, class, MAF) -->
<!-- histo_data_late <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_late < quantile(inter_sex_late, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_late > quantile(inter_sex_late, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_late_mashr_ED, beta_male = beta_male_late_mashr_ED, class, MAF) -->
<!-- all_loci_late <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_late_mashr_ED, beta_male = beta_male_late_mashr_ED, class, MAF) -->
<!-- bind_rows( -->
<!--   bind_rows(histo_data_early, all_loci_early) %>%  -->
<!--     mutate(stage = "early") %>%  -->
<!--     mutate(facet = paste(class, "early", sep = "\n")), -->
<!--   bind_rows(histo_data_late, all_loci_late) %>%  -->
<!--     mutate(stage = "late") %>%  -->
<!--     mutate(facet = paste(class, "late", sep = "\n")))%>%  -->
<!--   arrange(stage, class) %>%  -->
<!--   mutate(facet = factor(facet, unique(facet))) %>%  -->
<!--   # filter(MAF == "0.05 < MAF < 0.1") %>%  -->
<!--   ggplot(aes(beta_female)) +  -->
<!--   geom_vline(xintercept = 0, linetype = 2) + -->
<!--   geom_histogram(bins=30, fill="tomato", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(aes(beta_male), bins=30, fill="steelblue", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(bins=30, fill=NA, colour="black") +  -->
<!--   geom_histogram(aes(beta_male), bins=30,fill=NA,colour="black") +  -->
<!--   facet_wrap(~ facet, scales = "free_y") + -->
<!--   xlab("Effect of the minor allele on fitness\n(females:red, males:blue)") +  -->
<!--   theme_bw() + -->
<!--   theme(legend.position = "none", -->
<!--         strip.background = element_blank(),  -->
<!--         strip.text = element_text(hjust = 0), -->
<!--         text = element_text(family = "Raleway", size = 12)) + -->
<!--   ylab("Number of loci") -->
<!-- ``` -->
</div>
<div id="frequencies-of-antagonistic-loci-and-transcripts" class="section level2">
<h2>Frequencies of antagonistic loci and transcripts</h2>
<p>The following figures show the proportions of loci/transcripts whose effect on female fitness (or early-life fitness) is similar or different to their effect on male fitness (or late-life fitness). The figures were made by first placing variant or transcript effect sizes (for female or early-life fitness) into quartiles, which we label as negative effects, weak negative effects, weak positive effects, and positive effects (taking advantage of the fact that the median effect size is very close to zero). We then do the same for male or late-life fitness, and plot the number of transcripts showing each combination of quartiles, giving an indication of how many loci/transcripts have aligned or opposing effects on the different fitness metrics.</p>
<p>The following plots were made after processing the effect sizes with <code>mashr</code>, which should reduce the number of false signals. Because there is a positive correlation between the sexes and age classes, the shrinkage applied by <code>mashr</code> reduces the number of loci/transcripts that appear to show antagonism between ages and sexes (this should control the number of ‘false positive’ antagonistic loci/transcripts).</p>
<div id="frequencies-of-antagonistic-loci" class="section level3">
<h3>Frequencies of antagonistic loci</h3>
<div id="comparing-effects-on-the-sexes" class="section level4 tabset">
<h4>Comparing effects on the sexes</h4>
<p>Rather few loci are in the 1st quartile in their effect on male fitness and the 4th quartile for their effect on female fitness, or <em>vice versa</em>, suggesting that variants with strongly sex-opposite effects on fitness are rare</p>
<div id="figure" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>SNP_effects &lt;- SNP_clumps %&gt;%
  filter(LD_subset) %&gt;% 
  select(contains(&quot;beta&quot;)) %&gt;% 
  select(contains(&quot;ED&quot;)) # this can be changed to &quot;raw&quot; to see the non-mashr effect sizes (i.e. statistical noise)
names(SNP_effects)[grepl(&quot;female_early&quot;, names(SNP_effects))] &lt;- &quot;FE&quot;
names(SNP_effects)[grepl(&quot;female_late&quot;, names(SNP_effects))] &lt;- &quot;FL&quot;
names(SNP_effects)[grepl(&quot;male_early&quot;, names(SNP_effects))] &lt;- &quot;ME&quot;
names(SNP_effects)[grepl(&quot;male_late&quot;, names(SNP_effects))] &lt;- &quot;ML&quot;

temp &lt;- SNP_effects %&gt;% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %&gt;% 
  mutate_at(vars(starts_with(&quot;quartile&quot;)), ~ {
    SNP_effects &lt;- .x
    SNP_effects[SNP_effects == 1] &lt;- &quot;Negative&quot;
    SNP_effects[SNP_effects == 2] &lt;- &quot;Weakly\nnegative&quot;
    SNP_effects[SNP_effects == 3] &lt;- &quot;Weakly\npositive&quot;
    SNP_effects[SNP_effects == 4] &lt;- &quot;Positive&quot;
    factor(SNP_effects, c(&quot;Negative&quot;, &quot;Weakly\nnegative&quot;, &quot;Weakly\npositive&quot;, &quot;Positive&quot;))})

mat_list &lt;- list(table(temp$quartile_FE, temp$quartile_ME),
                 table(temp$quartile_FL, temp$quartile_ML),
                 table(temp$quartile_FE, temp$quartile_FL),
                 table(temp$quartile_ME, temp$quartile_ML))  %&gt;% 
   map(~ as.data.frame(.x)) 
names(mat_list) &lt;- c(&quot;FE, ME&quot;, &quot;FL, ML&quot;, &quot;FE, FL&quot;, &quot;ME, ML&quot;)
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

cols &lt;- c(brewer.pal(9, &quot;Reds&quot;)[c(6,3)], brewer.pal(9, &quot;Blues&quot;)[c(3,6)])

focal_dat &lt;- rbind(
    as.data.frame(mat_list[[1]]) %&gt;% 
      mutate(pp = &quot;A. Early-life fitness&quot;), 
    as.data.frame(mat_list[[2]]) %&gt;% 
      mutate(pp = &quot;B. Late-life fitness&quot;)) 

fig_4_top_row &lt;- focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nmale fitness&quot;) + 
  xlab(&quot;Association with female fitness&quot;) + 
  ylab(&quot;Number of variants&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        #strip.text = element_text(hjust = 0, family = &quot;Raleway&quot;, size = 12))
        strip.text = element_text(hjust = 0, size = 12))

# To count the total number of transcripts tested:
# focal_dat$Freq[focal_dat$pp == &quot;C. Early-life fitness&quot;] %&gt;% sum()

fig_4_top_row</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-12-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table" class="section level5">
<h5>Table</h5>
<pre class="r"><code>focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of variants` = Freq) %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Age class
</th>
<th style="text-align:left;">
Association with female fitness
</th>
<th style="text-align:left;">
Association with male fitness
</th>
<th style="text-align:right;">
Number of variants
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given effect on females)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
41947
</td>
<td style="text-align:right;">
20.07
</td>
<td style="text-align:right;">
80.29
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
9075
</td>
<td style="text-align:right;">
4.34
</td>
<td style="text-align:right;">
17.37
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1096
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
2.10
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
9459
</td>
<td style="text-align:right;">
4.53
</td>
<td style="text-align:right;">
18.10
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
31336
</td>
<td style="text-align:right;">
14.99
</td>
<td style="text-align:right;">
59.98
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
10546
</td>
<td style="text-align:right;">
5.05
</td>
<td style="text-align:right;">
20.18
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
906
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.73
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
738
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
1.41
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
10950
</td>
<td style="text-align:right;">
5.24
</td>
<td style="text-align:right;">
20.96
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
31731
</td>
<td style="text-align:right;">
15.18
</td>
<td style="text-align:right;">
60.73
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
8828
</td>
<td style="text-align:right;">
4.22
</td>
<td style="text-align:right;">
16.90
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
103
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.20
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
886
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
1.70
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
8874
</td>
<td style="text-align:right;">
4.25
</td>
<td style="text-align:right;">
16.99
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
42383
</td>
<td style="text-align:right;">
20.28
</td>
<td style="text-align:right;">
81.12
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
42917
</td>
<td style="text-align:right;">
20.54
</td>
<td style="text-align:right;">
82.14
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
8492
</td>
<td style="text-align:right;">
4.06
</td>
<td style="text-align:right;">
16.25
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
745
</td>
<td style="text-align:right;">
0.36
</td>
<td style="text-align:right;">
1.43
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
93
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.18
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
8787
</td>
<td style="text-align:right;">
4.20
</td>
<td style="text-align:right;">
16.82
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
32734
</td>
<td style="text-align:right;">
15.66
</td>
<td style="text-align:right;">
62.65
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
10158
</td>
<td style="text-align:right;">
4.86
</td>
<td style="text-align:right;">
19.44
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
568
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
1.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
495
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
0.95
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
10381
</td>
<td style="text-align:right;">
4.97
</td>
<td style="text-align:right;">
19.87
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
33013
</td>
<td style="text-align:right;">
15.80
</td>
<td style="text-align:right;">
63.19
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
8358
</td>
<td style="text-align:right;">
4.00
</td>
<td style="text-align:right;">
16.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
640
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
1.22
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
8331
</td>
<td style="text-align:right;">
3.99
</td>
<td style="text-align:right;">
15.95
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
43227
</td>
<td style="text-align:right;">
20.68
</td>
<td style="text-align:right;">
82.74
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="comparing-effects-on-the-age-classes" class="section level4 tabset">
<h4>Comparing effects on the age classes</h4>
<div id="figure-1" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>focal_dat &lt;- rbind(
  as.data.frame(mat_list[[3]]) %&gt;% 
    mutate(pp = &quot;A. Female fitness&quot;), 
  as.data.frame(mat_list[[4]]) %&gt;% 
    mutate(pp = &quot;B. Male fitness&quot;)) 

focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nlate-life fitness&quot;) + 
  xlab(&quot;Association with early-life fitness&quot;) + 
  ylab(&quot;Number of variants&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = &quot;Raleway&quot;, size = 12))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-14-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-14-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-1" class="section level5">
<h5>Table</h5>
<pre class="r"><code>focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of variants` = Freq) %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:left;">
Association with early-life fitness
</th>
<th style="text-align:left;">
Association with late-life fitness
</th>
<th style="text-align:right;">
Number of variants
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given effect on females)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
50560
</td>
<td style="text-align:right;">
24.19
</td>
<td style="text-align:right;">
96.77
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1687
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
3.23
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
1687
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
3.23
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
48453
</td>
<td style="text-align:right;">
23.18
</td>
<td style="text-align:right;">
92.74
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
2107
</td>
<td style="text-align:right;">
1.01
</td>
<td style="text-align:right;">
4.03
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
2107
</td>
<td style="text-align:right;">
1.01
</td>
<td style="text-align:right;">
4.03
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
48489
</td>
<td style="text-align:right;">
23.20
</td>
<td style="text-align:right;">
92.81
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1651
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
3.16
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1651
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
3.16
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
50595
</td>
<td style="text-align:right;">
24.21
</td>
<td style="text-align:right;">
96.84
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
49584
</td>
<td style="text-align:right;">
23.73
</td>
<td style="text-align:right;">
94.90
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
2661
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
5.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
2657
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
5.09
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
46371
</td>
<td style="text-align:right;">
22.19
</td>
<td style="text-align:right;">
88.75
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
3218
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
6.16
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.01
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
3214
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
6.15
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
46519
</td>
<td style="text-align:right;">
22.26
</td>
<td style="text-align:right;">
89.04
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
2508
</td>
<td style="text-align:right;">
1.20
</td>
<td style="text-align:right;">
4.80
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
2509
</td>
<td style="text-align:right;">
1.20
</td>
<td style="text-align:right;">
4.80
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
49736
</td>
<td style="text-align:right;">
23.80
</td>
<td style="text-align:right;">
95.20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="run-a-chi2-test" class="section level4">
<h4>Run a <span class="math inline">\(\chi^2\)</span> test</h4>
<p>The following <span class="math inline">\(\chi^2\)</span> test examines the null hypothesis that the proportion of candidate sexually antagonistic loci (i.e. those with a quartile 1 effect on fitness of sex <span class="math inline">\(i\)</span> and a quartile 4 effect on sex <span class="math inline">\(j\)</span>) is equal when fitness associations are calculated using the early- and late-life fitness data. This null hypothesis is rejected: the % of SA loci is 0.111% at early life and 0.067% at late life, which is a 1.6-fold difference.</p>
<pre class="r"><code>chisq_table &lt;- as.table(rbind(c(129+103, 208987 - (129+103)), c(93+48, 208987 - (93+48))))
dimnames(chisq_table) &lt;- list(life_stage = c(&quot;Early_life&quot;, &quot;Late_life&quot;),
                              antagonism_status = c(&quot;Sex_antag&quot;,&quot;Non_sex_antag&quot;))
chisq_table</code></pre>
<pre><code>            antagonism_status
life_stage   Sex_antag Non_sex_antag
  Early_life       232        208755
  Late_life        141        208846</code></pre>
<pre class="r"><code>chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 21.735, df = 1, p-value = 3.13e-06</code></pre>
</div>
</div>
<div id="frequencies-of-antagonistic-transcripts" class="section level3">
<h3>Frequencies of antagonistic transcripts</h3>
<div id="comparing-effects-on-the-sexes-1" class="section level4 tabset">
<h4>Comparing effects on the sexes</h4>
<div id="figure-2" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>TWAS_ED &lt;- readRDS(&quot;data/derived/TWAS/TWAS_ED.rds&quot;)

binned_transcripts &lt;- data.frame(
  FBID = read_csv(&quot;data/derived/TWAS/TWAS_results.csv&quot;)$FBID,
  as.data.frame(get_pm(TWAS_ED))) %&gt;% 
  as_tibble() %&gt;% rename_all(~ str_remove_all(.x, &quot;beta_&quot;)) %&gt;% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %&gt;% 
  mutate_at(vars(starts_with(&quot;quartile&quot;)), ~ {
    SNP_effects &lt;- .x
    SNP_effects[SNP_effects == 1] &lt;- &quot;Negative&quot;
    SNP_effects[SNP_effects == 2] &lt;- &quot;Weakly\nnegative&quot;
    SNP_effects[SNP_effects == 3] &lt;- &quot;Weakly\npositive&quot;
    SNP_effects[SNP_effects == 4] &lt;- &quot;Positive&quot;
    factor(SNP_effects, c(&quot;Negative&quot;, &quot;Weakly\nnegative&quot;, &quot;Weakly\npositive&quot;, &quot;Positive&quot;))})


mat_list &lt;- list(table(binned_transcripts$quartile_FE, binned_transcripts$quartile_ME),
                 table(binned_transcripts$quartile_FL, binned_transcripts$quartile_ML),
                 table(binned_transcripts$quartile_FE, binned_transcripts$quartile_FL),
                 table(binned_transcripts$quartile_ME, binned_transcripts$quartile_ML))  %&gt;% 
   map(~ as.data.frame(.x)) 
names(mat_list) &lt;- c(&quot;FE, ME&quot;, &quot;FL, ML&quot;, &quot;FE, FL&quot;, &quot;ME, ML&quot;)
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

focal_dat &lt;- rbind(
    as.data.frame(mat_list[[1]]) %&gt;% 
      mutate(pp = &quot;C. Early-life fitness&quot;), 
    as.data.frame(mat_list[[2]]) %&gt;% 
      mutate(pp = &quot;D. Late-life fitness&quot;)) 

# To count the total number of transcripts tested, 14286:
# focal_dat$Freq[focal_dat$pp == &quot;C. Early-life fitness&quot;] %&gt;% sum()

fig_4_bottom_row &lt;- focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nmale fitness&quot;) + 
  xlab(&quot;Association with female fitness&quot;) + 
  ylab(&quot;Number of transcripts&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        #strip.text = element_text(hjust = 0, family = &quot;Raleway&quot;, size = 12))
        strip.text = element_text(hjust = 0, size = 12))

fig_4_bottom_row</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-17-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-17-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-18-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-2" class="section level5">
<h5>Table</h5>
<pre class="r"><code>focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of transcripts` = Freq) %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Age class
</th>
<th style="text-align:left;">
Association with female fitness
</th>
<th style="text-align:left;">
Association with male fitness
</th>
<th style="text-align:right;">
Number of transcripts
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given effect on females)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
1578
</td>
<td style="text-align:right;">
11.05
</td>
<td style="text-align:right;">
44.18
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
752
</td>
<td style="text-align:right;">
5.26
</td>
<td style="text-align:right;">
21.05
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
551
</td>
<td style="text-align:right;">
3.86
</td>
<td style="text-align:right;">
15.43
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
691
</td>
<td style="text-align:right;">
4.84
</td>
<td style="text-align:right;">
19.34
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
773
</td>
<td style="text-align:right;">
5.41
</td>
<td style="text-align:right;">
21.64
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1168
</td>
<td style="text-align:right;">
8.18
</td>
<td style="text-align:right;">
32.70
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1043
</td>
<td style="text-align:right;">
7.30
</td>
<td style="text-align:right;">
29.20
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
588
</td>
<td style="text-align:right;">
4.12
</td>
<td style="text-align:right;">
16.46
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
532
</td>
<td style="text-align:right;">
3.72
</td>
<td style="text-align:right;">
14.90
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1058
</td>
<td style="text-align:right;">
7.41
</td>
<td style="text-align:right;">
29.63
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1201
</td>
<td style="text-align:right;">
8.41
</td>
<td style="text-align:right;">
33.63
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
780
</td>
<td style="text-align:right;">
5.46
</td>
<td style="text-align:right;">
21.84
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
689
</td>
<td style="text-align:right;">
4.82
</td>
<td style="text-align:right;">
19.29
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
594
</td>
<td style="text-align:right;">
4.16
</td>
<td style="text-align:right;">
16.63
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
776
</td>
<td style="text-align:right;">
5.43
</td>
<td style="text-align:right;">
21.73
</td>
</tr>
<tr>
<td style="text-align:left;">
C. Early-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1512
</td>
<td style="text-align:right;">
10.58
</td>
<td style="text-align:right;">
42.34
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
1603
</td>
<td style="text-align:right;">
11.22
</td>
<td style="text-align:right;">
44.88
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
758
</td>
<td style="text-align:right;">
5.31
</td>
<td style="text-align:right;">
21.22
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
534
</td>
<td style="text-align:right;">
3.74
</td>
<td style="text-align:right;">
14.95
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
677
</td>
<td style="text-align:right;">
4.74
</td>
<td style="text-align:right;">
18.95
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
765
</td>
<td style="text-align:right;">
5.35
</td>
<td style="text-align:right;">
21.42
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1175
</td>
<td style="text-align:right;">
8.22
</td>
<td style="text-align:right;">
32.89
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1053
</td>
<td style="text-align:right;">
7.37
</td>
<td style="text-align:right;">
29.48
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
579
</td>
<td style="text-align:right;">
4.05
</td>
<td style="text-align:right;">
16.21
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
540
</td>
<td style="text-align:right;">
3.78
</td>
<td style="text-align:right;">
15.12
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
1061
</td>
<td style="text-align:right;">
7.43
</td>
<td style="text-align:right;">
29.71
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
1206
</td>
<td style="text-align:right;">
8.44
</td>
<td style="text-align:right;">
33.77
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
764
</td>
<td style="text-align:right;">
5.35
</td>
<td style="text-align:right;">
21.39
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
664
</td>
<td style="text-align:right;">
4.65
</td>
<td style="text-align:right;">
18.59
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
578
</td>
<td style="text-align:right;">
4.05
</td>
<td style="text-align:right;">
16.19
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
778
</td>
<td style="text-align:right;">
5.45
</td>
<td style="text-align:right;">
21.79
</td>
</tr>
<tr>
<td style="text-align:left;">
D. Late-life fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1551
</td>
<td style="text-align:right;">
10.86
</td>
<td style="text-align:right;">
43.43
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="run-a-chi2-test-1" class="section level4">
<h4>Run a <span class="math inline">\(\chi^2\)</span> test</h4>
<p>The following <span class="math inline">\(\chi^2\)</span> test examines the null hypothesis that the proportion of candidate sexually antagonistic transcripts (i.e. those with a quartile 1 effect on fitness of sex <span class="math inline">\(i\)</span> and a quartile 4 effect on sex <span class="math inline">\(j\)</span>) is equal when fitness associations are calculated using the early- and late-life fitness data. This null hypothesis is not rejected: the % of SA transcripts is 9.66% at early life and 9.39% at late life, which is a 1.03-fold difference.</p>
<pre class="r"><code>chisq_table &lt;- as.table(rbind(c(691+689, 14286 - (691+689)), c(677 + 664, 14286 - (677 + 664))))
dimnames(chisq_table) &lt;- list(life_stage = c(&quot;Early_life&quot;, &quot;Late_life&quot;),
                              antagonism_status = c(&quot;Sex_antag&quot;,&quot;Non_sex_antag&quot;))
chisq_table</code></pre>
<pre><code>            antagonism_status
life_stage   Sex_antag Non_sex_antag
  Early_life      1380         12906
  Late_life       1341         12945</code></pre>
<pre class="r"><code>chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 0.58655, df = 1, p-value = 0.4438</code></pre>
</div>
<div id="comparing-effects-on-the-age-classes-1" class="section level4 tabset">
<h4>Comparing effects on the age classes</h4>
<div id="figure-3" class="section level5">
<h5>Figure</h5>
<pre class="r"><code>focal_dat &lt;- rbind(
  as.data.frame(mat_list[[3]]) %&gt;% 
    mutate(pp = &quot;A. Female fitness&quot;), 
  as.data.frame(mat_list[[4]]) %&gt;% 
    mutate(pp = &quot;B. Male fitness&quot;))

focal_dat %&gt;% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat=&quot;identity&quot;, colour = &quot;grey20&quot;) +
  scale_fill_manual(values = cols, name = &quot;Association with\nlate-life fitness&quot;) + 
  xlab(&quot;Association with early-life fitness&quot;) + 
  ylab(&quot;Number of transcripts&quot;) + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = &quot;Raleway&quot;, size = 12))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-21-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-21-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-3" class="section level5">
<h5>Table</h5>
<pre class="r"><code>focal_dat %&gt;% 
  split(.$pp) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  split(paste((.$Var1), .$pp)) %&gt;% 
  map_df(~ .x %&gt;% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %&gt;% 
  arrange(pp, Var1, Var2) %&gt;% 
  select(pp, everything()) %&gt;% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of transcripts` = Freq) %&gt;% 
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:left;">
Association with early-life fitness
</th>
<th style="text-align:left;">
Association with late-life fitness
</th>
<th style="text-align:right;">
Number of transcripts
</th>
<th style="text-align:right;">
Percentage (overall)
</th>
<th style="text-align:right;">
Percentage (given effect on females)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
3523
</td>
<td style="text-align:right;">
24.66
</td>
<td style="text-align:right;">
98.63
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
1.37
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
1.37
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
3464
</td>
<td style="text-align:right;">
24.25
</td>
<td style="text-align:right;">
96.98
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
1.65
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
1.65
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
3450
</td>
<td style="text-align:right;">
24.15
</td>
<td style="text-align:right;">
96.61
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.74
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.74
</td>
</tr>
<tr>
<td style="text-align:left;">
A. Female fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
3509
</td>
<td style="text-align:right;">
24.56
</td>
<td style="text-align:right;">
98.26
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
3550
</td>
<td style="text-align:right;">
24.85
</td>
<td style="text-align:right;">
99.38
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
3521
</td>
<td style="text-align:right;">
24.65
</td>
<td style="text-align:right;">
98.57
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.81
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.81
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
3520
</td>
<td style="text-align:right;">
24.64
</td>
<td style="text-align:right;">
98.57
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly negative
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Weakly positive
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.62
</td>
</tr>
<tr>
<td style="text-align:left;">
B. Male fitness
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
3549
</td>
<td style="text-align:right;">
24.84
</td>
<td style="text-align:right;">
99.38
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="plotting-and-modelling-the-evidence-for-antagonism" class="section level2">
<h2>Plotting and modelling the evidence for antagonism</h2>
<div id="calculate-the-evidence-ratios" class="section level3">
<h3>Calculate the evidence ratios</h3>
<div id="for-the-gwas-results" class="section level4">
<h4>For the GWAS results</h4>
<pre class="r"><code>get_antagonism_ratios &lt;- function(dat){
  dat %&gt;%
    
    # Convert the LFSR to the probability that effect size is positive
    mutate(pp_female_early = ifelse(pheno1 &gt; 0, LFSR1, 1 - LFSR1),
           pp_female_late  = ifelse(pheno2 &gt; 0, LFSR2, 1 - LFSR2),
           pp_male_early   = ifelse(pheno3 &gt; 0, LFSR3, 1 - LFSR3),
           pp_male_late    = ifelse(pheno4 &gt; 0, LFSR4, 1 - LFSR4)) %&gt;%
    
    # Calculate the probabilities that beta_i and beta_j have the same/opposite signs
    mutate(p_sex_concord_early = pp_female_early * pp_male_early + 
             (1 - pp_female_early) * (1 - pp_male_early),
           p_sex_antag_early = pp_female_early * (1 - pp_male_early) + 
             (1 - pp_female_early) * pp_male_early,
           p_sex_concord_late  = pp_female_late * pp_male_late + 
             (1 - pp_female_late) * (1 - pp_male_late),
           p_sex_antag_late = pp_female_late * (1 - pp_male_late) + 
             (1 - pp_female_late) * pp_male_late,
           p_age_concord_females = pp_female_early * pp_female_late + 
             (1 - pp_female_early) * (1 - pp_female_late),
           p_age_antag_females = pp_female_early * (1 - pp_female_late) + 
             (1 - pp_female_early) * pp_female_late,
           p_age_concord_males = pp_male_early * pp_male_late + (1 - pp_male_early) * (1 - pp_male_late),
           p_age_antag_males = pp_male_early * (1 - pp_male_late) + (1 - pp_male_early) * pp_male_late) %&gt;%
    
    # Find the ratios of some of these probabilities (i.e. &quot;evidence ratios&quot;)
    mutate(inter_sex_early = p_sex_concord_early / p_sex_antag_early,
           inter_sex_late = p_sex_concord_late / p_sex_antag_late,
           inter_age_females = p_age_concord_females / p_age_antag_females,
           inter_age_males = p_age_concord_males / p_age_antag_males) 
}

antagonism_evidence_ratios_gwas &lt;- all_snps %&gt;%
  rename(pheno1 = beta_female_early_mashr_ED, 
         pheno2 = beta_female_late_mashr_ED, 
         pheno3 = beta_male_early_mashr_ED, 
         pheno4 = beta_male_late_mashr_ED,
         LFSR1 = LFSR_female_early_mashr_ED, 
         LFSR2 = LFSR_female_late_mashr_ED, 
         LFSR3 = LFSR_male_early_mashr_ED, 
         LFSR4 = LFSR_male_late_mashr_ED) %&gt;% 
  select(SNP, SNP_clump, starts_with(&quot;pheno&quot;), LFSR1, LFSR2, LFSR3, LFSR4) %&gt;% 
  filter(!is.na(pheno1)) %&gt;% 
  collect(n=Inf) %&gt;% 
  distinct() %&gt;% 
  get_antagonism_ratios() %&gt;%
  select(SNP, SNP_clump, starts_with(&quot;inter&quot;)) </code></pre>
</div>
<div id="for-the-twas-results" class="section level4">
<h4>For the TWAS results</h4>
<pre class="r"><code>TWAS_ED &lt;- readRDS(&quot;data/derived/TWAS/TWAS_ED.rds&quot;)

twas &lt;- data.frame(
  FBID = read_csv(&quot;data/derived/TWAS/TWAS_results.csv&quot;)$FBID,
      as.data.frame(get_pm(TWAS_ED)),
      as.data.frame(get_lfsr(TWAS_ED)) %&gt;% 
        rename_all(~str_replace_all(., &quot;beta&quot;, &quot;LFSR&quot;))) %&gt;% 
      as_tibble() %&gt;%
  left_join(tbl(db, &quot;genes&quot;) %&gt;% select(FBID, chromosome) %&gt;% collect(), by = &quot;FBID&quot;) %&gt;%
  left_join(read_csv(&quot;data/derived/gene_expression_by_sex.csv&quot;), by = &quot;FBID&quot;) %&gt;%
  filter(chromosome %in% c(&quot;2L&quot;, &quot;2R&quot;, &quot;3L&quot;, &quot;3R&quot;, &quot;X&quot;)) %&gt;%
  rename(pheno1 = beta_FE, 
         pheno2 = beta_FL, 
         pheno3 = beta_ME, 
         pheno4 = beta_ML,
         LFSR1 = LFSR_FE, 
         LFSR2 = LFSR_FL, 
         LFSR3 = LFSR_ME, 
         LFSR4 = LFSR_ML)

antagonism_evidence_ratios_twas &lt;- twas %&gt;% 
  get_antagonism_ratios() %&gt;%
  select(FBID, chromosome, male_bias_in_expression, AveExpr, starts_with(&quot;inter&quot;)) %&gt;% 
  mutate(chromosome = relevel(factor(chromosome), ref = &quot;X&quot;))</code></pre>
</div>
</div>
<div id="plot-the-evidence-ratios" class="section level3">
<h3>Plot the evidence ratios</h3>
<pre class="r"><code>make_evidence_ratio_plot &lt;- function(ERs, ymax, ylab){
  # Argument needs to be a dataframe of loci or transcripts, each identified by a column called &quot;identifier&quot;
  # There should be a col called &quot;evidence_ratio&quot;, calculated from the LFSR from ED mashr. The &quot;trait&quot; col says which type of ER it is.
  
  ERs$trait[ERs$trait == &quot;inter_sex_early&quot;] &lt;- &quot;Inter-sex (early life)&quot;
  ERs$trait[ERs$trait == &quot;inter_sex_late&quot;] &lt;- &quot;Inter-sex (late life)&quot;
  ERs$trait[ERs$trait == &quot;inter_age_females&quot;] &lt;- &quot;Inter-age (females)&quot;
  ERs$trait[ERs$trait == &quot;inter_age_males&quot;] &lt;- &quot;Inter-age (males)&quot;
  
  antagonism_evidence_ratios &lt;- ERs %&gt;%
    mutate(trait = factor(trait, c(&quot;Inter-sex (early life)&quot;,
                                   &quot;Inter-sex (late life)&quot;,
                                   &quot;Inter-age (females)&quot;,
                                   &quot;Inter-age (males)&quot;)))
  
  antagonism_evidence_ratios %&gt;%
    ggplot(aes(log2(evidence_ratio))) + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio &lt; 1), 
                   bins = 500, fill = &quot;#FF635C&quot;) + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio &gt; 1), 
                   bins = 500, fill = &quot;#5B8AFD&quot;) +
    coord_cartesian(xlim = c(-10, 10), ylim = c(0, ymax)) + 
    scale_x_continuous(breaks = c(-10,  -6,  -2,   2,   6,  10), 
                       labels = c(paste(&quot;1/&quot;,2 ^ c(10,  6,  2), sep = &quot;&quot;), 2 ^ c(2,6,10))) + 
    facet_wrap(~ trait) + 
    xlab(&quot;Evidence ratio (log2 scale)&quot;) + ylab(ylab) + 
    theme_bw() + 
    theme(panel.border = element_rect(size = 0.8),
          text = element_text(family = &quot;Raleway&quot;, size = 12),
          axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
          strip.background = element_blank())
}

GWAS_ratios_plot &lt;- antagonism_evidence_ratios_gwas %&gt;% 
  select(SNP_clump,  starts_with(&quot;inter&quot;)) %&gt;%
  distinct() %&gt;% # Include SNPs in 100% LD a single time, such that these clumps are the unit of replication
  gather(trait, evidence_ratio, -SNP_clump) %&gt;% 
  rename(identifier = SNP_clump) %&gt;% 
  make_evidence_ratio_plot(ymax = 5000, ylab = &quot;Number of loci&quot;)

# For counting the number of transcripts with 50x more evidence for antagonism 
# antagonism_evidence_ratios_twas %&gt;% 
#   select(FBID,  starts_with(&quot;inter&quot;)) %&gt;%
#   gather(trait, evidence_ratio, -FBID) %&gt;% 
#   rename(identifier = FBID) %&gt;% filter(evidence_ratio &lt; 1/50) %&gt;% summarise(n())

TWAS_ratios_plot &lt;- antagonism_evidence_ratios_twas %&gt;% 
  select(FBID,  starts_with(&quot;inter&quot;)) %&gt;%
  gather(trait, evidence_ratio, -FBID) %&gt;% 
  rename(identifier = FBID) %&gt;% 
  make_evidence_ratio_plot(ymax = 500, ylab = &quot;Number of transcripts&quot;)

pp &lt;- plot_grid(GWAS_ratios_plot, TWAS_ratios_plot, 
                   labels = c(&#39;A&#39;, &#39;B&#39;), label_size = 12)
# ggdraw(add_sub(pp, &quot;Evidence ratio (log scale)&quot;, 
#                vpadding=grid::unit(0,&quot;lines&quot;), 
#                y=6.3, x=0.5, vjust=6))
ggsave(&quot;figures/antagonism_ratios.pdf&quot;, pp, width = 8.5, height = 4.9)

pp</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-evidence_ratio_plot-1">
Past versions of evidence_ratio_plot-1.png
</button>
</p>
<div id="fig-evidence_ratio_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/evidence_ratio_plot-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="model-the-evidence-ratios" class="section level3">
<h3>Model the evidence ratios</h3>
<pre class="r"><code># gwas models

# Annotate it with MAF, site, class, etc. 
dat &lt;- left_join(antagonism_evidence_ratios_gwas,
          collect(tbl(db, &quot;variants&quot;)), by = &quot;SNP&quot;) %&gt;% 
  distinct() %&gt;% mutate(site.class = str_to_title(site.class))

# Focus only on the commonest site classes:
dat &lt;- dat %&gt;% 
  filter(site.class %in% c(&quot;Intron&quot;, &quot;Intergenic&quot;, &quot;Downstream&quot;, &quot;Upstream&quot;,
                           &quot;Synonymous_coding&quot;, &quot;Non_synonymous_coding&quot;, &quot;Utr_3_prime&quot;,
                           &quot;Exon&quot;, &quot;Utr_5_prime&quot;)) 

# Remove chromosome 4 (too few sites)
dat &lt;- dat %&gt;% filter(chr != &quot;4&quot;) 

# If there are multiple site classes for a SNP, or multiple SNPs 
# in the same 100% LD clump, pick one SNP and/or 1 site class at random
set.seed(1)
dat &lt;- dat[sample(nrow(dat), nrow(dat)), ] %&gt;%
  split(.$SNP_clump) %&gt;%
  map_df(~ .x[1, ])

dat &lt;- dat %&gt;% arrange(site.class)
dat$site.class &lt;- relevel(factor(dat$site.class), ref = &quot;Synonymous_coding&quot;)
dat$chr &lt;- relevel(factor(dat$chr), ref = &quot;X&quot;)

n_loci &lt;- prettyNum(nrow(dat), big.mark = &quot;,&quot;, scientific = FALSE)


intersex_early_model_gwas &lt;- glm(antagonistic ~ chr + MAF + site.class, 
             data = dat %&gt;% 
               mutate(antagonistic = as.numeric(inter_sex_early &lt;= 
                        quantile(dat$inter_sex_early, probs=0.01))), 
    family = &quot;binomial&quot;)
car::Anova(intersex_early_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
           LR Chisq Df Pr(&gt;Chisq)    
chr           11.07  4    0.02584 *  
MAF          632.01  1    &lt; 2e-16 ***
site.class     4.48  7    0.72322    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_early_model_gwas &lt;- update(intersex_early_model_gwas,  ~ . -site.class)
car::Anova(intersex_early_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
    LR Chisq Df Pr(&gt;Chisq)    
chr    11.23  4    0.02411 *  
MAF   630.62  1    &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_early_model_gwas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ chr + MAF, family = &quot;binomial&quot;, 
    data = dat %&gt;% mutate(antagonistic = as.numeric(inter_sex_early &lt;= 
        quantile(dat$inter_sex_early, probs = 0.01))))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.2256  -0.1766  -0.1208  -0.0939   3.4183  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) -5.83559    0.08616 -67.730  &lt; 2e-16 ***
chr2L       -0.25998    0.08095  -3.212  0.00132 ** 
chr2R       -0.14129    0.07894  -1.790  0.07346 .  
chr3L       -0.14743    0.07801  -1.890  0.05877 .  
chr3R       -0.08119    0.08005  -1.014  0.31047    
MAF          4.37586    0.18468  23.695  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 18467  on 164800  degrees of freedom
Residual deviance: 17828  on 164795  degrees of freedom
AIC: 17840

Number of Fisher Scoring iterations: 8</code></pre>
<pre class="r"><code>intersex_late_model_gwas &lt;- glm(antagonistic ~ chr + MAF + site.class, 
             data = dat %&gt;% 
               mutate(antagonistic = inter_sex_late &lt;= 
                        quantile(dat$inter_sex_late, probs=0.01)), 
    family = &quot;binomial&quot;)
car::Anova(intersex_late_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
           LR Chisq Df Pr(&gt;Chisq)    
chr            7.85  4    0.09706 .  
MAF          537.35  1    &lt; 2e-16 ***
site.class     3.24  7    0.86166    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_late_model_gwas &lt;- update(intersex_late_model_gwas,  ~ . -site.class)
car::Anova(intersex_late_model_gwas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
    LR Chisq Df Pr(&gt;Chisq)    
chr     8.03  4    0.09041 .  
MAF   536.59  1    &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_late_model_gwas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ chr + MAF, family = &quot;binomial&quot;, 
    data = dat %&gt;% mutate(antagonistic = inter_sex_late &lt;= quantile(dat$inter_sex_late, 
        probs = 0.01)))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.2204  -0.1748  -0.1232  -0.0980   3.3596  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) -5.69624    0.08410 -67.735  &lt; 2e-16 ***
chr2L       -0.21675    0.07974  -2.718  0.00657 ** 
chr2R       -0.16673    0.07910  -2.108  0.03504 *  
chr3L       -0.15168    0.07778  -1.950  0.05118 .  
chr3R       -0.12758    0.08057  -1.583  0.11332    
MAF          4.00050    0.18112  22.087  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 18467  on 164800  degrees of freedom
Residual deviance: 17925  on 164795  degrees of freedom
AIC: 17937

Number of Fisher Scoring iterations: 8</code></pre>
<pre class="r"><code># twas models
n_transcripts &lt;- prettyNum(nrow(antagonism_evidence_ratios_twas), big.mark = &quot;,&quot;, scientific = FALSE)

twas_model_data &lt;- antagonism_evidence_ratios_twas %&gt;% 
  mutate(antagonistic = inter_sex_early &lt;= 
           quantile(antagonism_evidence_ratios_twas$inter_sex_early, probs=0.01),
         absolute_sex_bias_in_expression = abs(male_bias_in_expression),
         mean_expression_level = AveExpr)


intersex_early_twas &lt;- glm(antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level, 
             data = twas_model_data, 
    family = &quot;binomial&quot;)
car::Anova(intersex_early_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                                LR Chisq Df Pr(&gt;Chisq)   
chromosome                        3.5741  4   0.466695   
absolute_sex_bias_in_expression   0.0446  1   0.832701   
mean_expression_level             8.7170  1   0.003153 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_early_twas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ chromosome + absolute_sex_bias_in_expression + 
    mean_expression_level, family = &quot;binomial&quot;, data = twas_model_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.2018  -0.1541  -0.1377  -0.1236   3.2023  

Coefficients:
                                Estimate Std. Error z value Pr(&gt;|z|)   
(Intercept)                     -1.56221    1.11609  -1.400  0.16160   
chromosome2L                     0.35439    0.29528   1.200  0.23008   
chromosome2R                     0.15257    0.30175   0.506  0.61312   
chromosome3L                     0.32701    0.29486   1.109  0.26742   
chromosome3R                    -0.02928    0.30172  -0.097  0.92268   
absolute_sex_bias_in_expression -0.05345    0.25445  -0.210  0.83363   
mean_expression_level           -0.52326    0.17945  -2.916  0.00355 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1590.3  on 14190  degrees of freedom
Residual deviance: 1577.4  on 14184  degrees of freedom
AIC: 1591.4

Number of Fisher Scoring iterations: 7</code></pre>
<pre class="r"><code>intersex_late_model_twas &lt;- glm(antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level, 
             data = antagonism_evidence_ratios_twas %&gt;% 
  mutate(antagonistic = inter_sex_late &lt;= 
           quantile(antagonism_evidence_ratios_twas$inter_sex_late, probs=0.01),
         absolute_sex_bias_in_expression = abs(male_bias_in_expression),
         mean_expression_level = AveExpr), 
    family = &quot;binomial&quot;)
car::Anova(intersex_late_model_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                                LR Chisq Df Pr(&gt;Chisq)   
chromosome                        3.7808  4   0.436479   
absolute_sex_bias_in_expression   0.0003  1   0.986291   
mean_expression_level             8.3362  1   0.003886 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>intersex_late_model_twas &lt;- update(intersex_late_model_twas,  ~ . -absolute_sex_bias_in_expression -chromosome)
car::Anova(intersex_late_model_twas, type = 3)</code></pre>
<pre><code>Analysis of Deviance Table (Type III tests)

Response: antagonistic
                      LR Chisq Df Pr(&gt;Chisq)   
mean_expression_level   8.7545  1   0.003088 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>summary(intersex_late_model_twas)</code></pre>
<pre><code>
Call:
glm(formula = antagonistic ~ mean_expression_level, family = &quot;binomial&quot;, 
    data = antagonism_evidence_ratios_twas %&gt;% mutate(antagonistic = inter_sex_late &lt;= 
        quantile(antagonism_evidence_ratios_twas$inter_sex_late, 
            probs = 0.01), absolute_sex_bias_in_expression = abs(male_bias_in_expression), 
        mean_expression_level = AveExpr))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.1836  -0.1540  -0.1392  -0.1261   3.2298  

Coefficients:
                      Estimate Std. Error z value Pr(&gt;|z|)   
(Intercept)            -1.3931     1.0835  -1.286  0.19852   
mean_expression_level  -0.5255     0.1790  -2.935  0.00333 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1590.3  on 14190  degrees of freedom
Residual deviance: 1581.5  on 14189  degrees of freedom
AIC: 1585.5

Number of Fisher Scoring iterations: 7</code></pre>
<pre class="r"><code># plotting models
new &lt;- data.frame(MAF = seq(from = 0.05, to = 0.5, by = 0.01), chr = &quot;X&quot;)
inv_logit &lt;- function(x) exp(x)/(1+exp(x))
preds &lt;- data.frame(new, prop = predict(intersex_early_model_gwas, newdata = new, se.fit=T, type = &quot;link&quot;)) %&gt;% 
  mutate(y = prop.fit, lwr = prop.fit - 1.96*prop.se.fit, upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), lwr = inv_logit(lwr), upr = inv_logit(upr))
p1 &lt;- preds %&gt;% 
  ggplot(aes(MAF,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  scale_x_continuous(limits = c(0, 0.5)) + 
  scale_y_continuous(limits = c(0, max(preds$y))) + 
  theme_bw() +#+ theme(text = element_text(family = &quot;Raleway&quot;, size = 12)) 
  ylab(&quot;Probability of being in the top 1%\nsexually antagonistic loci (\u00B1 95% CIs)&quot;) +
  xlab(&quot;Minor allele frequency&quot;)

new2 &lt;- data.frame(chr = levels(dat$chr), MAF = 0.3)
inv_logit &lt;- function(x) exp(x)/(1+exp(x))
preds2 &lt;- data.frame(new2, prop = predict(intersex_early_model_gwas, newdata = new2, se.fit=T, type = &quot;link&quot;)) %&gt;% 
  mutate(y = prop.fit, lwr = prop.fit - 1.96*prop.se.fit, upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), lwr = inv_logit(lwr), upr = inv_logit(upr))
p2 &lt;- preds2 %&gt;% 
  ggplot(aes(chr,  y)) + 
  geom_point(size = 3) + 
  geom_errorbar(aes(ymin = lwr, ymax = upr), width = 0) + 
  theme_bw() +#+ theme(text = element_text(family = &quot;Raleway&quot;, size = 12)) 
  ylab(&quot; \n &quot;) +
  xlab(&quot;Chromosome arm&quot;)

new &lt;- data.frame(
  mean_expression_level = seq(from = min(twas_model_data$mean_expression_level), 
                                to = max(twas_model_data$mean_expression_level), by = 0.01), 
                  chromosome = &quot;X&quot;, 
  absolute_sex_bias_in_expression = 0,
                  mean_expression_level = mean(twas_model_data$absolute_sex_bias_in_expression))
inv_logit &lt;- function(x) exp(x)/(1+exp(x))
preds &lt;- data.frame(new, prop = predict(intersex_early_twas, newdata = new, se.fit=T, type = &quot;link&quot;)) %&gt;% 
  mutate(y = prop.fit, lwr = prop.fit - 1.96*prop.se.fit, upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), lwr = inv_logit(lwr), upr = inv_logit(upr))
p3 &lt;- preds %&gt;% 
  ggplot(aes(mean_expression_level,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  ylab(&quot;Probability of being in the top 1%\nsexually antagonistic transcripts (\u00B1 95% CIs)&quot;) +
  xlab(&quot;Average expression level&quot;) + 
  theme_bw() #+ theme(text = element_text(family = &quot;Raleway&quot;, size = 12)) 


blank &lt;- ggplot() + theme_void()


fig_6_models &lt;- grid.arrange(
grid.arrange(p1, p2, ncol = 2), grid.arrange(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))</code></pre>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-28-1.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-28-1.png" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/e112260eb7edb940edc701a4180d2cf98f9d1145/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-28-1.png" target="_blank">e112260</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/plot_models_variant_effects.Rmd/unnamed-chunk-28-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-2">
Past versions of unnamed-chunk-28-2.png
</button>
</p>
<div id="fig-unnamed-chunk-28-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/652106368c95ac4a04b2383d560d3c8073990f18/docs/figure/plot_models_variant_effects.Rmd/unnamed-chunk-28-2.png" target="_blank">6521063</a>
</td>
<td>
lukeholman
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(&quot;figures/fig6_models.pdf&quot;, fig_6_models, width = 6, height = 7)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] RColorBrewer_1.1-2 cowplot_1.0.0      kableExtra_1.3.4   mashr_0.2.38      
 [5] ashr_2.2-47        showtext_0.9-5     showtextdb_3.0     sysfonts_0.8.3    
 [9] Hmisc_4.4-0        Formula_1.2-3      survival_3.2-7     lattice_0.20-41   
[13] ggbeeswarm_0.6.0   qqman_0.1.4        gridExtra_2.3      forcats_0.5.1     
[17] stringr_1.4.0      dplyr_1.0.7        purrr_0.3.4        readr_2.0.0       
[21] tidyr_1.1.4        tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1   
[25] workflowr_1.7.0   

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1    rio_0.5.16          ellipsis_0.3.2     
  [4] rprojroot_1.3-2     htmlTable_1.13.3    base64enc_0.1-3    
  [7] fs_1.4.1            rstudioapi_0.13     hexbin_1.28.1      
 [10] farver_2.0.3        bit64_0.9-7         fansi_0.4.1        
 [13] mvtnorm_1.1-0       lubridate_1.7.10    xml2_1.3.2         
 [16] splines_4.0.3       knitr_1.32          jsonlite_1.7.3     
 [19] broom_0.7.11        cluster_2.1.0       dbplyr_2.1.1       
 [22] png_0.1-7           compiler_4.0.3      httr_1.4.2         
 [25] backports_1.1.7     assertthat_0.2.1    Matrix_1.2-18      
 [28] cli_3.1.0           later_1.0.0         acepack_1.4.1      
 [31] htmltools_0.5.0     tools_4.0.3         gtable_0.3.0       
 [34] glue_1.4.2          Rcpp_1.0.4.6        carData_3.0-4      
 [37] cellranger_1.1.0    jquerylib_0.1.1     vctrs_0.3.8        
 [40] nlme_3.1-149        svglite_1.2.3       xfun_0.26          
 [43] ps_1.3.3            openxlsx_4.1.5      rvest_1.0.2        
 [46] lifecycle_1.0.1     irlba_2.3.3         getPass_0.2-2      
 [49] MASS_7.3-53         scales_1.1.1        vroom_1.5.3        
 [52] hms_1.1.1           promises_1.1.0      parallel_4.0.3     
 [55] curl_4.3            yaml_2.2.1          memoise_1.1.0      
 [58] gdtools_0.2.2       rpart_4.1-15        RSQLite_2.2.0      
 [61] latticeExtra_0.6-29 calibrate_1.7.7     stringi_1.5.3      
 [64] SQUAREM_2020.2      rmeta_3.0           highr_0.8          
 [67] checkmate_2.0.0     zip_2.1.1           truncnorm_1.0-8    
 [70] systemfonts_0.2.2   rlang_0.4.12        pkgconfig_2.0.3    
 [73] evaluate_0.14       invgamma_1.1        labeling_0.3       
 [76] htmlwidgets_1.5.1   bit_1.1-15.2        processx_3.5.2     
 [79] tidyselect_1.1.0    plyr_1.8.6          magrittr_2.0.1     
 [82] R6_2.4.1            generics_0.0.2      DBI_1.1.0          
 [85] mgcv_1.8-33         pillar_1.6.4.9003   haven_2.3.1        
 [88] whisker_0.4         foreign_0.8-80      withr_2.4.3        
 [91] abind_1.4-5         mixsqp_0.3-43       nnet_7.3-14        
 [94] car_3.0-8           modelr_0.1.8        crayon_1.4.2       
 [97] utf8_1.1.4          tzdb_0.1.2          rmarkdown_2.11     
[100] jpeg_0.1-8.1        isoband_0.2.1       readxl_1.3.1       
[103] data.table_1.14.2   blob_1.2.1          callr_3.7.0        
[106] git2r_0.27.1        webshot_0.5.2       reprex_2.0.1       
[109] digest_0.6.25       httpuv_1.5.3.1      munsell_0.5.0      
[112] viridisLite_0.3.0   beeswarm_0.4.0      vipor_0.4.5        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
