<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Estimating genetic (co)variance and heritability from the GRM</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Estimating genetic (co)variance and heritability from the GRM</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree372da3e1f31ee1905466e161ba4d89c135c796d0targetblank372da3ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/372da3e1f31ee1905466e161ba4d89c135c796d0" target="_blank">372da3e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree372da3e1f31ee1905466e161ba4d89c135c796d0targetblank372da3ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/372da3e1f31ee1905466e161ba4d89c135c796d0" target="_blank">372da3e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    analysis/plot_models_variant_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    figures/fig1_inkscape.svg
    Ignored:    figures/figure1a.pdf
    Ignored:    figures/figure1b.pdf

Untracked files:
    Untracked:  analysis/quantitative_genetics.Rmd
    Untracked:  big_model.rds
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds
    Untracked:  figures/SNP_effect_ED.pdf
    Untracked:  figures/boyle_plot.pdf
    Untracked:  figures/fig1.pdf

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/GWAS_tables.Rmd
    Modified:   analysis/TWAS.Rmd
    Modified:   analysis/TWAS_tables.Rmd
    Modified:   analysis/get_predicted_line_means.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/perform_gwas.Rmd
    Modified:   analysis/plot_line_means.Rmd
    Modified:   analysis/plot_models_variant_effects.Rmd
    Deleted:    data/input/clough_2014_dsx_targets.csv
    Deleted:    figures/figure1.eps
    Deleted:    figures/figure2.eps
    Modified:   manuscript/GWAS_manuscript.Rmd
    Modified:   manuscript/GWAS_manuscript.pdf
    Modified:   manuscript/GWAS_manuscript.tex

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<!-- # knitr::purl(input = "analysis/quantitative_genetics.Rmd", output = "code/quant_gen_1.R") -->
<pre class="r"><code>library(tidyverse)
library(brms)
library(kableExtra)

brms_data &lt;- read_csv(&quot;data/derived/predicted_line_means.csv&quot;) %&gt;% 
  mutate(line = factor(str_remove_all(line, &quot;_&quot;))) %&gt;%
  filter(!is.na(male.fitness.early)) %&gt;%
  rename(femalefitnessearly = female.fitness.early,
         femalefitnesslate = female.fitness.late,
         malefitnessearly = male.fitness.early,
         malefitnesslate = male.fitness.late)</code></pre>
<div id="estimate-the-genomic-relatedness-matrix-between-lines" class="section level2">
<h2>Estimate the genomic relatedness matrix between lines</h2>
<p>Using the genotypes of the DGRP from the PLINK files provided by the Mackay lab, we calculate the genomic covariance matrix using <code>sommer::A.mat()</code>. This matrix is used to specify how the line random effects co-vary with each other in the model below.</p>
<pre class="r"><code>if(!file.exists(&quot;data/input/genomic_relatedness_matrix.rds&quot;)){
  
  library(bigsnpr)
  library(sommer)
  
  bigsnp &lt;- snp_readBed(&quot;data/derived/dgrp2_QC_all_lines.bed&quot;, backingfile = tempfile())
  bigsnp_attached &lt;- snp_attach(bigsnp)
  G &lt;- A.mat(bigsnp_attached$genotypes[1:nrow(bigsnp_attached$genotypes), 
                                       1:ncol(bigsnp_attached$genotypes)] - 1) 
  colnames(G) &lt;- bigsnp_attached$fam[,1]
  rownames(G) &lt;- bigsnp_attached$fam[,1]
  focal_lines &lt;- brms_data %&gt;% pull(line)
  G &lt;- G[rownames(G) %in% focal_lines, 
         colnames(G) %in% focal_lines]
  saveRDS(G, &quot;data/input/genomic_relatedness_matrix.rds&quot;)
} else G &lt;- readRDS(&quot;data/input/genomic_relatedness_matrix.rds&quot;)</code></pre>
</div>
<div id="fit-multivariate-mixed-effects-model" class="section level2">
<h2>Fit multivariate mixed effects model</h2>
<p>The following (run on a supercomputer as it needs more than 32GB RAM) fits a multivariate mixed effects model using <code>brms</code>. The input data are the line means for our four phenotypes estimated in <code>get_predicted_line_means.Rmd</code>, as well as the genomic relatedness matrix (GRM) estimated above. The model assumes each trait follows a Gaussian distribution, and that the traits (co)vary across lines, with <code>line</code> modelled as a random effect. The line effects are modelled as correlated (see <code>?brms::gr</code>), with the line effect covariances constrained to follow those in the GRM. The estimated standard deviation of the line effects, and the correlations among them, are used below to estimate quantitative genetic parameters. To ensure the model converges without divergent transitions and other warnings, we used a large number of warmup iterations, a large value for the <code>adapt_delta</code> parameter (which affects the NUTS sampler used by Stan), and quite informative priors on the line effects, residual variances, and correlations.</p>
<pre class="r"><code>if(!file.exists(&quot;data/derived/posterior_brms_quant_gen.csv&quot;)){
  brms_quant_gen &lt;- brm(
    bf(femalefitnessearly  ~ 0 + (1 | p | gr(line, cov = G))) +
      bf(femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) +
      bf(malefitnessearly  ~ 0 + (1 | p | gr(line, cov = G))) +
      bf(malefitnesslate   ~ 0 + (1 | p | gr(line, cov = G))) +
      set_rescor(FALSE),
    data = brms_data, data2 = list(G = G), 
    iter = 90000, chains = 4, cores = 1, warmup = 80000,
    control = list(adapt_delta = 0.99999999999999, max_treedepth = 20), 
    
    prior = c(
      prior(normal(0.5, 0.1), &quot;sd&quot;, resp = &#39;femalefitnessearly&#39;, group = &quot;line&quot;),
      prior(normal(0.5, 0.1), &quot;sd&quot;, resp = &#39;femalefitnesslate&#39;, group = &quot;line&quot;),
      prior(normal(0.5, 0.1), &quot;sd&quot;, resp = &#39;malefitnessearly&#39;, group = &quot;line&quot;),
      prior(normal(0.5, 0.1), &quot;sd&quot;, resp = &#39;malefitnesslate&#39;, group = &quot;line&quot;),
      prior(normal(0.5, 0.1), &quot;sigma&quot;, resp = &#39;femalefitnessearly&#39;),
      prior(normal(0.5, 0.1), &quot;sigma&quot;, resp = &#39;femalefitnesslate&#39;),
      prior(normal(0.5, 0.1), &quot;sigma&quot;, resp = &#39;malefitnessearly&#39;),
      prior(normal(0.5, 0.1), &quot;sigma&quot;, resp = &#39;malefitnesslate&#39;),
      prior(lkj(0.5), class = &quot;cor&quot;, group = &quot;line&quot;))
  )
  writeLines(capture.output(summary(brms_quant_gen)), 
             con = &quot;data/derived/model_summary_brms_quant_gen.txt&quot;)
  write_csv(posterior_samples(brms_quant_gen)[,1:14], 
            &quot;data/derived/posterior_brms_quant_gen.csv&quot;)
}</code></pre>
<div id="inspect-the-output-of-summary-for-the-brms-model" class="section level3">
<h3>Inspect the output of <code>summary()</code> for the <code>brms</code> model</h3>
<pre class="r"><code>cat(readLines(&quot;data/derived/model_summary_brms_quant_gen.txt&quot;), sep = &#39;\n&#39;)</code></pre>
<pre style="max-height: 300px;"><code> Family: MV(gaussian, gaussian, gaussian, gaussian) 
  Links: mu = identity; sigma = identity
         mu = identity; sigma = identity
         mu = identity; sigma = identity
         mu = identity; sigma = identity 
Formula: femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G)) 
         femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G)) 
         malefitnessearly ~ 0 + (1 | p | gr(line, cov = G)) 
         malefitnesslate ~ 0 + (1 | p | gr(line, cov = G)) 
   Data: brms_data (Number of observations: 124) 
Samples: 4 chains, each with iter = 90000; warmup = 80000; thin = 1;
         total post-warmup samples = 40000

Group-Level Effects: 
~line (Number of levels: 124) 
                                                              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(femalefitnessearly_Intercept)                                  0.60      0.05     0.51     0.70 1.00     9027    19097
sd(femalefitnesslate_Intercept)                                   0.61      0.05     0.52     0.69 1.00     8733    21410
sd(malefitnessearly_Intercept)                                    0.63      0.04     0.55     0.71 1.00    15179    25638
sd(malefitnesslate_Intercept)                                     0.63      0.04     0.55     0.71 1.00    14331    23334
cor(femalefitnessearly_Intercept,femalefitnesslate_Intercept)     0.93      0.04     0.83     0.99 1.00     3185     9324
cor(femalefitnessearly_Intercept,malefitnessearly_Intercept)      0.23      0.10     0.03     0.40 1.00     9300    18465
cor(femalefitnesslate_Intercept,malefitnessearly_Intercept)       0.34      0.09     0.15     0.51 1.00     6525    16078
cor(femalefitnessearly_Intercept,malefitnesslate_Intercept)       0.17      0.10    -0.02     0.36 1.00     9988    17651
cor(femalefitnesslate_Intercept,malefitnesslate_Intercept)        0.33      0.09     0.14     0.50 1.00    10495    20928
cor(malefitnessearly_Intercept,malefitnesslate_Intercept)         0.97      0.02     0.92     1.00 1.00     9907    17866

Family Specific Parameters: 
                         Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma_femalefitnessearly     0.46      0.07     0.31     0.59 1.00     5133     6593
sigma_femalefitnesslate      0.43      0.07     0.28     0.57 1.00     3324     5961
sigma_malefitnessearly       0.36      0.06     0.22     0.47 1.00     7707     7136
sigma_malefitnesslate        0.37      0.06     0.24     0.48 1.00     9739     9486

Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
</div>
<div id="table-of-quantitative-genetic-estimates" class="section level3">
<h3>Table of quantitative genetic estimates</h3>
<p>The table shows the posterior median estimate, the associated error, and the 95% credible intervals for each heritability, genetic correlation, and additive genetic (co)variance, as well as differences between two pairs of genetic correlations that are of interest.</p>
<ul>
<li>Additive genetic covariances for trait <span class="math inline">\(i\)</span> (<span class="math inline">\(V_{Ai}\)</span>) were estimated as the square of the line-level standard deviation estimated by the model.</li>
<li>Heritability for trait <span class="math inline">\(i\)</span> was estimated as <span class="math inline">\(V_{Ai} / (V_{Ai} + V_{Ei})\)</span>, where <span class="math inline">\(V_{Ei}\)</span> the square of the residual standard deviation (estimated by the model) for the <span class="math inline">\(i\)</span>th trait.</li>
<li>Additive genetic correlations for trait pair <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> – <span class="math inline">\(cor_{Ai,j}\)</span> – were estimated as the correlation in line effects for that trait pair.</li>
<li>Additive genetic covariances were estimated as <span class="math inline">\(cor_{Ai,j} = cor_{Ai,j}V_{A1}V_{A2}\)</span></li>
<li>We also computed the posterior difference between two pairs of genetic correlations, to test whether the inter-sex correlation changes between age classes, and whether the inter-age correlation differs between sexes (the 95% CIs of the posterior estimate of the difference include zero, so there is no strong evidence for this)</li>
</ul>
<pre class="r"><code>p &lt;- read_csv(&quot;data/derived/posterior_brms_quant_gen.csv&quot;) %&gt;% as.data.frame()
p &lt;- tibble(
  VA_FE = p[, &quot;sd_line__femalefitnessearly_Intercept&quot;]^2,
  VA_FL = p[, &quot;sd_line__femalefitnesslate_Intercept&quot;]^2,
  VA_ME = p[, &quot;sd_line__malefitnessearly_Intercept&quot;]^2,
  VA_ML = p[, &quot;sd_line__malefitnesslate_Intercept&quot;]^2,
  h2_FE = VA_FE / (VA_FE + p[, &quot;sigma_femalefitnessearly&quot;]^2),
  h2_FL = VA_FL /  (VA_FL + p[, &quot;sigma_femalefitnesslate&quot;]^2),
  h2_ME = VA_ME /  (VA_ME + p[, &quot;sigma_malefitnessearly&quot;]^2),
  h2_ML = VA_ML /  (VA_ML + p[, &quot;sigma_malefitnesslate&quot;]^2),
  corr_FE_FL = p[,&quot;cor_line__femalefitnessearly_Intercept__femalefitnesslate_Intercept&quot;],
  corr_FE_ME = p[,&quot;cor_line__femalefitnessearly_Intercept__malefitnessearly_Intercept&quot;],
  corr_FL_ME = p[,&quot;cor_line__femalefitnesslate_Intercept__malefitnessearly_Intercept&quot;],
  corr_FE_ML = p[,&quot;cor_line__femalefitnessearly_Intercept__malefitnesslate_Intercept&quot;],
  corr_FL_ML = p[,&quot;cor_line__femalefitnesslate_Intercept__malefitnesslate_Intercept&quot;],
  corr_ME_ML = p[,&quot;cor_line__malefitnessearly_Intercept__malefitnesslate_Intercept&quot;],
  cov_FE_FL = corr_FE_FL * sqrt(VA_FE) * sqrt(VA_FL),
  cov_FE_ME = corr_FE_ME * sqrt(VA_FE) * sqrt(VA_ME),
  cov_FL_ME = corr_FL_ME * sqrt(VA_FL) * sqrt(VA_ME),
  cov_FE_ML = corr_FE_ML * sqrt(VA_FE) * sqrt(VA_ML),
  cov_FL_ML = corr_FL_ML * sqrt(VA_FL) * sqrt(VA_ML),
  cov_ME_ML = corr_ME_ML * sqrt(VA_ME) * sqrt(VA_ML),
  corr_FE_ME_minus_corr_FL_ML = corr_FE_ME - corr_FL_ML,
  corr_FE_FL_minus_corr_ME_ML = corr_FE_FL - corr_ME_ML) %&gt;% 
  gather(Parameter, value) %&gt;% 
  mutate(Parameter = factor(Parameter, unique(Parameter)))

post_summary &lt;- p %&gt;%
  group_by(Parameter) %&gt;%
  summarise(value = list(posterior_summary(value))) %&gt;%
  mutate(Estimate = map_dbl(value, ~ .x[[1]]),
         `Est. Error` = map_dbl(value, ~ .x[[2]]),
         `Lower 95% CI` = map_dbl(value, ~ .x[[3]]),
         `Upper 95% CI` = map_dbl(value, ~ .x[[4]])) %&gt;%
  dplyr::select(-value) %&gt;%
  mutate(sp = str_split(Parameter, &quot;_&quot;), 
         Type = map_chr(sp, ~ .x[[1]]),
         Parameter = map_chr(sp, ~ paste0(.x[-1], collapse = &quot; - &quot;)),
         Type = replace(Type, str_detect(Parameter, &quot;minus&quot;), 
                        &quot;Comparing corrs&quot;),
         Type = factor(Type, c(&quot;h2&quot;, &quot;corr&quot;, &quot;VA&quot;, &quot;cov&quot;, 
                               &quot;Comparing corrs&quot;)),
         Parameter = replace(
           Parameter, 
           Parameter == &quot;FE - ME - minus - corr - FL - ML&quot;, 
           &quot;corr(FE,ME) - corr(FL,ML)&quot;),
         Parameter = replace(
           Parameter, 
           Parameter == &quot;FE - FL - minus - corr - ME - ML&quot;, 
           &quot;corr(FE,FL) - corr(ME,ML)&quot;)
  ) %&gt;% arrange(Type)

post_summary %&gt;%
  dplyr::select(-Type, -sp) %&gt;%
  kable(digits = 3) %&gt;%
  kable_styling(full_width = FALSE) %&gt;%
  pack_rows(&quot;Heritabilities&quot;, 1, 4) %&gt;%
  pack_rows(&quot;Genetic correlations&quot;, 5, 10) %&gt;%
  pack_rows(&quot;Additive genetic variances&quot;, 11, 14) %&gt;%
  pack_rows(&quot;Additive genetic covariances&quot;, 15, 20) %&gt;%
  pack_rows(&quot;Comparisons of genetic correlations&quot;, 21, 22)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Est. Error
</th>
<th style="text-align:right;">
Lower 95% CI
</th>
<th style="text-align:right;">
Upper 95% CI
</th>
</tr>
</thead>
<tbody>
<tr grouplength="4">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>Heritabilities</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE
</td>
<td style="text-align:right;">
0.629
</td>
<td style="text-align:right;">
0.094
</td>
<td style="text-align:right;">
0.453
</td>
<td style="text-align:right;">
0.819
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FL
</td>
<td style="text-align:right;">
0.660
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
0.485
</td>
<td style="text-align:right;">
0.841
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
ME
</td>
<td style="text-align:right;">
0.756
</td>
<td style="text-align:right;">
0.075
</td>
<td style="text-align:right;">
0.606
</td>
<td style="text-align:right;">
0.900
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
ML
</td>
<td style="text-align:right;">
0.740
</td>
<td style="text-align:right;">
0.073
</td>
<td style="text-align:right;">
0.596
</td>
<td style="text-align:right;">
0.882
</td>
</tr>
<tr grouplength="6">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>Genetic correlations</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE - FL
</td>
<td style="text-align:right;">
0.928
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
0.827
</td>
<td style="text-align:right;">
0.990
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE - ME
</td>
<td style="text-align:right;">
0.226
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:right;">
0.405
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FL - ME
</td>
<td style="text-align:right;">
0.337
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
0.150
</td>
<td style="text-align:right;">
0.513
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE - ML
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:right;">
0.098
</td>
<td style="text-align:right;">
-0.023
</td>
<td style="text-align:right;">
0.363
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FL - ML
</td>
<td style="text-align:right;">
0.329
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
0.144
</td>
<td style="text-align:right;">
0.502
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
ME - ML
</td>
<td style="text-align:right;">
0.970
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
0.917
</td>
<td style="text-align:right;">
0.998
</td>
</tr>
<tr grouplength="4">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>Additive genetic variances</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE
</td>
<td style="text-align:right;">
0.368
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:right;">
0.489
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FL
</td>
<td style="text-align:right;">
0.369
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:right;">
0.269
</td>
<td style="text-align:right;">
0.483
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
ME
</td>
<td style="text-align:right;">
0.401
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.307
</td>
<td style="text-align:right;">
0.510
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
ML
</td>
<td style="text-align:right;">
0.398
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
0.303
</td>
<td style="text-align:right;">
0.509
</td>
</tr>
<tr grouplength="6">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>Additive genetic covariances</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE - FL
</td>
<td style="text-align:right;">
0.341
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
0.263
</td>
<td style="text-align:right;">
0.431
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE - ME
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
0.163
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FL - ME
</td>
<td style="text-align:right;">
0.129
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
0.057
</td>
<td style="text-align:right;">
0.207
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FE - ML
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
-0.008
</td>
<td style="text-align:right;">
0.142
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
FL - ML
</td>
<td style="text-align:right;">
0.126
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
0.203
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
ME - ML
</td>
<td style="text-align:right;">
0.387
</td>
<td style="text-align:right;">
0.044
</td>
<td style="text-align:right;">
0.307
</td>
<td style="text-align:right;">
0.481
</td>
</tr>
<tr grouplength="2">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>Comparisons of genetic correlations</strong>
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
corr(FE,ME) - corr(FL,ML)
</td>
<td style="text-align:right;">
-0.103
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
-0.285
</td>
<td style="text-align:right;">
0.075
</td>
</tr>
<tr>
<td style="text-align:left; padding-left: 2em;" indentlevel="1">
corr(FE,FL) - corr(ME,ML)
</td>
<td style="text-align:right;">
-0.042
</td>
<td style="text-align:right;">
0.049
</td>
<td style="text-align:right;">
-0.150
</td>
<td style="text-align:right;">
0.042
</td>
</tr>
</tbody>
</table>
<!-- ## Estimate genetic (co)variances -->
<!-- The following code fit (sometimes multivariate) mixed models using `brms`, to estimate the amount of variance explained by the DGRP line, using a covariance structure that accounts for the similarity between lines captured by the genomic relatedness matrix (GRM). -->
<!-- Initial testing revealed that  -->
<!-- The response variables are the predicted line means for the four phenotypes, as estimated in the previous section. We fit line as a random effect, specifying a covariance structure among the line effects that is equal to the genomic relatedness matrix (GRM), which we estimate using the SNP genotypes of the full sample of 205 DGRP lines. We use a somewhat informative prior to help the model converge; the prior is that the line and residual variances are not large (which must be true, given that the response variables have overall variance of 1), and that only 10% of the phenotype variance is explained by line (the posterior estimates suggest much higher line variance, indicating strong signal in the data). -->
<!-- The table below indicates that much of the variance in fitness is due to additive genetic effects, and that there is a positive geneti -->
<!-- <!-- # conside this paper method too: https://onlinelibrary.wiley.com/doi/epdf/10.1111/j.1558-5646.2012.01673.x -->
<p>–&gt;</p>
<!-- ## Models to find heritability -->
<!-- ```{r} -->
<!-- if(!file.exists("data/derived/posterior_brms_quant_gen_FE.csv")){  -->
<!--   brms_quant_gen_FE <- brm( # female early -->
<!--     femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G)), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 15000, chains = 4, cores = 1, warmup = 5000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->
<!--     prior = prior(normal(0.5, 0.15), "sd", group = "line") -->
<!--   ) -->
<!--   writeLines(capture.output(summary(brms_quant_gen_FE)), con = "data/derived/model_summary_brms_quant_gen_FE.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_FE)[,1:2], "data/derived/posterior_brms_quant_gen_FE.csv") -->
<!-- } -->
<!-- if(!file.exists("data/derived/posterior_brms_quant_gen_FL.csv")){ -->
<!--   brms_quant_gen_FL <- brm( # female late -->
<!--     femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G)),    -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 80000, chains = 4, cores = 1, warmup = 70000, -->
<!--     control = list(adapt_delta = 0.9999999999999999, max_treedepth = 20), -->
<!--     prior = prior(normal(0.5, 0.15), "sd", group = "line")  -->
<!--   ) -->
<!--   writeLines(capture.output(summary(brms_quant_gen_FL)), con = "data/derived/model_summary_brms_quant_gen_FL.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_FL)[,1:2], "data/derived/posterior_brms_quant_gen_FL.csv") -->
<!-- } -->
<!-- if(!file.exists("data/derived/posterior_brms_quant_gen_ME.csv")){ -->
<!--   brms_quant_gen_ME <- brm( # male early -->
<!--     femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G)), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 15000, chains = 4, cores = 1, warmup = 5000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->
<!--     prior = prior(normal(0.5, 0.15), "sd", group = "line") -->
<!--   ) -->
<!--   writeLines(capture.output(summary(brms_quant_gen_ME)), con = "data/derived/model_summary_brms_quant_gen_ME.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_ME)[,1:2], "data/derived/posterior_brms_quant_gen_ME.csv") -->
<!-- } -->
<!-- if(!file.exists("data/derived/posterior_brms_quant_gen_FE.csv")){ -->
<!--   brms_quant_gen_ML <- brm( # male late -->
<!--     femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G)), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 15000, chains = 4, cores = 1, warmup = 5000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->
<!--     prior = prior(normal(0.5, 0.15), "sd", group = "line") -->
<!--   ) -->
<!--   writeLines(capture.output(summary(brms_quant_gen_ML)), con = "data/derived/model_summary_brms_quant_gen_ML.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_ML)[,1:2], "data/derived/posterior_brms_quant_gen_ML.csv") -->
<!-- } -->
<!-- ``` -->
<!-- ## Models to find genetic correlations -->
<!-- ```{r} -->
<!-- switch <- 4 -->
<!-- # Run quant gen model using brms -->
<!-- if(!file.exists("data/derived/brms_quant_gen_FE_ME.rds") & switch == 1){ -->
<!--   # Female early - male early model -->
<!--   brms_quant_gen_FE_ME <- brm( -->
<!--     bf(femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(malefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(FALSE), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 60000, chains = 4, cores = 1, warmup = 40000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->
<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'malefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnessearly'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'malefitnessearly'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->
<!--   # saveRDS(brms_quant_gen_FE_ME, "data/derived/brms_quant_gen_FE_ME.rds") -->
<!--   writeLines(capture.output(summary(brms_quant_gen_FE_ME)), con = "data/derived/model_summary_brms_quant_gen_FE_ME.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_FE_ME)[,1:5], "data/derived/posterior_brms_quant_gen_FE_ME.csv") -->
<!-- } -->
<!-- if(!file.exists("data/derived/brms_quant_gen_FL_ML.rds") & switch == 2){ -->
<!--   # Female late - male late model -->
<!--   brms_quant_gen_FL_ML <- brm( -->
<!--     bf(femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(malefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(FALSE), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 60000, chains = 4, cores = 1, warmup = 40000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->
<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'malefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnesslate'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'malefitnesslate'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->
<!--   # saveRDS(brms_quant_gen_FL_ML, "data/derived/brms_quant_gen_FL_ML.rds") -->
<!--   writeLines(capture.output(summary(brms_quant_gen_FL_ML)), con = "data/derived/model_summary_brms_quant_gen_FL_ML.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_FL_ML)[,1:5], "data/derived/posterior_brms_quant_gen_FL_ML.csv") -->
<!-- } -->
<!-- if(!file.exists("data/derived/brms_quant_gen_FE_FL.rds") & switch == 3){    -->
<!--   # Female early - female late model -->
<!--   brms_quant_gen_FE_FL <- brm(  -->
<!--     bf(femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(FALSE), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 50000, chains = 4, cores = 1, warmup = 40000, -->
<!--     control = list(adapt_delta = 0.999999999999, max_treedepth = 20), -->
<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnessearly'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnesslate'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->
<!--   # saveRDS(brms_quant_gen_FE_FL, "data/derived/brms_quant_gen_FE_FL.rds") -->
<!--   writeLines(capture.output(summary(brms_quant_gen_FE_FL)), con = "data/derived/model_summary_brms_quant_gen_FE_FL.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_FE_FL)[,1:5], "data/derived/posterior_brms_quant_gen_FE_FL.csv") -->
<!-- } -->
<!-- if(!file.exists("data/derived/brms_quant_gen_ME_ML.rds") & switch == 4){   -->
<!--   # Male early - male late model -->
<!--   brms_quant_gen_ME_ML <- brm( -->
<!--     bf(malefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(malefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(FALSE), -->
<!--     data = brms_data, data2 = list(G = G),  -->
<!--     iter = 60000, chains = 4, cores = 1, warmup = 40000, -->
<!--     control = list(adapt_delta = 0.99999999999999, max_treedepth = 20),  -->
<!--     prior = c( -->
<!--       prior(normal(0.5, 0.1), "sd", resp = 'malefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.1), "sd", resp = 'malefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.1), "sigma", resp = 'malefitnessearly'), -->
<!--       prior(normal(0.5, 0.1), "sigma", resp = 'malefitnesslate'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->
<!--   # saveRDS(brms_quant_gen_ME_ML, "data/derived/brms_quant_gen_ME_ML.rds") -->
<!--   writeLines(capture.output(summary(brms_quant_gen_ME_ML)), con = "data/derived/model_summary_brms_quant_gen_ME_ML.txt") -->
<!--   write_csv(posterior_samples(brms_quant_gen_ME_ML)[,1:5], "data/derived/posterior_brms_quant_gen_ME_ML.csv") -->
<!-- } -->
<!-- ``` -->
<!-- ## Inspect the results -->
<!-- ```{r} -->
<!-- # brms_quant_gen_FE_ME <- readRDS("data/derived/brms_quant_gen_FE_ME.rds") -->
<!-- # brms_quant_gen_FL_ML <- readRDS("data/derived/brms_quant_gen_FL_ML.rds") -->
<!-- # brms_quant_gen_FE_FL <- readRDS("data/derived/brms_quant_gen_FE_FL.rds") -->
<!-- # brms_quant_gen_ME_ML <- readRDS("data/derived/brms_quant_gen_ME_ML.rds") -->
<!-- p_FE <- read_csv("data/derived/posterior_brms_quant_gen_FE.csv") -->
<!-- p_FL <- read_csv("data/derived/posterior_brms_quant_gen_FL.csv") -->
<!-- p_ME <- read_csv("data/derived/posterior_brms_quant_gen_ME.csv") -->
<!-- p_ML <- read_csv("data/derived/posterior_brms_quant_gen_ML.csv") -->
<!-- p_FE_ME <- read_csv("data/derived/posterior_brms_quant_gen_FE_ME.csv") -->
<!-- p_FL_ML <- read_csv("data/derived/posterior_brms_quant_gen_FL_ML.csv") -->
<!-- p_FE_FL <- read_csv("data/derived/posterior_brms_quant_gen_FE_FL.csv") -->
<!-- p_ME_ML <- read_csv("data/derived/posterior_brms_quant_gen_ME_ML.csv") -->
<!-- p <- list( -->
<!--   VA_FE = p_FE[, "sd_line__Intercept"]^2, -->
<!--   VA_FL = p_FL[, "sd_line__Intercept"]^2, -->
<!--   VA_ME = p_ME[, "sd_line__Intercept"]^2, -->
<!--   VA_ML = p_ML[, "sd_line__Intercept"]^2, -->
<!--   h2_FE = p_FE[, "sd_line__Intercept"]^2 / (p_FE[, "sd_line__Intercept"]^2 + p_FE[, "sigma"]^2), -->
<!--   h2_FL = p_FL[, "sd_line__Intercept"]^2 /  (p_FL[, "sd_line__Intercept"]^2 + p_FL[, "sigma"]^2), -->
<!--   h2_ME = p_ME[, "sd_line__Intercept"]^2 /  (p_ME[, "sd_line__Intercept"]^2 + p_ME[, "sigma"]^2), -->
<!--   h2_ML = p_ML[, "sd_line__Intercept"]^2 /  (p_ML[, "sd_line__Intercept"]^2 + p_ML[, "sigma"]^2), -->
<!--   corr_FE_FL = p_FE_FL[,"cor_line__femalefitnessearly_Intercept__femalefitnesslate_Intercept"], -->
<!--   corr_FE_ME = p_FE_ME[,"cor_line__femalefitnessearly_Intercept__malefitnessearly_Intercept"], -->
<!--   # corr_FL_ME = p[,"cor_line__femalefitnesslate_Intercept__malefitnessearly_Intercept"], -->
<!--   # corr_FE_ML = p[,"cor_line__femalefitnessearly_Intercept__malefitnesslate_Intercept"], -->
<!--   corr_FL_ML = p_FL_ML[,"cor_line__femalefitnesslate_Intercept__malefitnesslate_Intercept"], -->
<!--   corr_ME_ML = p_ME_ML[,"cor_line__malefitnessearly_Intercept__malefitnesslate_Intercept"], -->
<!--   # cov_FE_FL = corr_FE_FL * VA_FE^2 * VA_FL^2, -->
<!--   # cov_FE_ME = corr_FE_ME * VA_FE^2 * VA_ME^2, -->
<!--   # cov_FL_ME = corr_FL_ME * VA_FL^2 * VA_ME^2, -->
<!--   # cov_FE_ML = corr_FE_ML * VA_FE^2 * VA_ML^2, -->
<!--   # cov_FL_ML = corr_FL_ML * VA_FL^2 * VA_ML^2, -->
<!--   # cov_ME_ML = corr_ME_ML * VA_ME^2 * VA_ML^2, -->
<!--   corr_FE_ME_minus_corr_FL_ML = p_FE_ME[,"cor_line__femalefitnessearly_Intercept__malefitnessearly_Intercept"] - p_FL_ML[,"cor_line__femalefitnesslate_Intercept__malefitnesslate_Intercept"], -->
<!--   corr_FE_FL_minus_corr_ME_ML = p_FE_FL[,"cor_line__femalefitnessearly_Intercept__femalefitnesslate_Intercept"] - p_ME_ML[,"cor_line__malefitnessearly_Intercept__malefitnesslate_Intercept"] -->
<!-- ) #%>% gather(Parameter, value) -->
<!-- post_summary <- p %>% -->
<!--   lapply(function(x){posterior_summary(x[,1])}) %>% -->
<!--   enframe() %>%  -->
<!--   mutate(Estimate = map_dbl(value, ~ .x[[1]]), -->
<!--          `Est. Error` = map_dbl(value, ~ .x[[2]]), -->
<!--          `Lower 95% CI` = map_dbl(value, ~ .x[[3]]), -->
<!--          `Upper 95% CI` = map_dbl(value, ~ .x[[4]])) %>% -->
<!--   select(-value) -->
<!-- post_summary %>% -->
<!--   kable(digits = 3) %>% -->
<!--   kable_styling(full_width = FALSE) -->
<!--   # mutate(sp = str_split(Parameter, "_"), -->
<!--   #        Type = factor(map_chr(sp, ~ .x[[1]]), c("h2", "corr", "VA", "cov")), -->
<!--   #        Parameter = map_chr(sp, ~ paste0(.x[-1], collapse = " - "))) %>% -->
<!--   # arrange(Type) %>% -->
<!--   # dplyr::select(-sp, -Type) %>% -->
<!--   # kable(digits = 3) %>% -->
<!--   # kable_styling(full_width = FALSE) #%>% -->
<!--   # pack_rows("Heritabilities", 1, 4) %>% -->
<!--   # pack_rows("Genetic correlations", 5, 10) %>% -->
<!--   # pack_rows("Additive genetic variances", 11, 14) %>% -->
<!--   # pack_rows("Additive genetic covariances", 15, 20) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- dat <- brms_data %>% -->
<!--   select(-block) %>% -->
<!--   gather(trait, value, -line) -->
<!-- brms_quant_gen_v2 <- brm( -->
<!--   bf(value ~ 0 + (trait | p | gr(line, cov = G)),  -->
<!--      sigma ~ (trait | q | line)), -->
<!--   data = dat, data2 = list(G = G),  -->
<!--   iter = 60000, chains = 4, cores = 1, warmup = 40000, -->
<!--   control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->
<!--   prior = prior(normal(0.5, 0.3), "sd", group = "line") -->
<!-- ) -->
<!-- saveRDS(brms_quant_gen_v2, "data/derived/brms_quant_gen_v2.rds") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- xx <- readRDS("data/derived/brms_quant_gen.rds") -->
<!-- new <- dat %>% -->
<!--   select(line, trait) %>% -->
<!--   distinct() -->
<!-- preds <- data.frame(new, fitted(xx, new)) %>% -->
<!--   select(trait, line, Estimate) %>% -->
<!--   spread(trait, Estimate) -->
<!-- ggplot(preds, aes(Estimate)) + -->
<!--   geom_histogram() + -->
<!--   facet_wrap(~ trait) -->
<!-- ggplot(preds, aes(femalefitnesslate, malefitnesslate)) + -->
<!--   geom_point() -->
<!-- p <- posterior_samples(xx) -->
<!-- p <- tibble( -->
<!--   VA_FE = (p[, "sd_line__Intercept"])^2, -->
<!--   VA_FL = (p[, "sd_line__traitfemalefitnesslate"])^2, -->
<!--   VA_ME = (p[, "sd_line__traitmalefitnessearly"])^2, -->
<!--   VA_ML = (p[, "sd_line__traitmalefitnesslate"])^2, -->
<!--   corr_FE_FL = p[,"cor_line__Intercept__traitfemalefitnesslate"], -->
<!--   corr_FE_ME = p[,"cor_line__Intercept__traitmalefitnessearly"], -->
<!--   corr_FL_ME = p[,"cor_line__traitfemalefitnesslate__traitmalefitnessearly"], -->
<!--   corr_FE_ML = p[,"cor_line__Intercept__traitmalefitnesslate"], -->
<!--   corr_FL_ML = p[,"cor_line__traitfemalefitnesslate__traitmalefitnesslate"], -->
<!--   corr_ME_ML = p[,"cor_line__traitmalefitnessearly__traitmalefitnesslate"], -->
<!--   cov_FE_FL = corr_FE_FL * VA_FE^2 * VA_FL^2, -->
<!--   cov_FE_ME = corr_FE_ME * VA_FE^2 * VA_ME^2, -->
<!--   cov_FL_ME = corr_FL_ME * VA_FL^2 * VA_ME^2, -->
<!--   cov_FE_ML = corr_FE_ML * VA_FE^2 * VA_ML^2, -->
<!--   cov_FL_ML = corr_FL_ML * VA_FL^2 * VA_ML^2, -->
<!--   cov_ME_ML = corr_ME_ML * VA_ME^2 * VA_ML^2, -->
<!--   corr_FE_ME_minus_corr_FL_ML = corr_FE_ME - corr_FL_ML, -->
<!--   corr_FE_FL_minus_corr_ME_ML = corr_FE_FL - corr_ME_ML -->
<!-- ) %>% gather(Parameter, value) -->
<!-- library(kableExtra) -->
<!-- p %>% -->
<!--   group_by(Parameter) %>% -->
<!--   summarise(summ = list(posterior_summary(value)), -->
<!--             Estimate = map_dbl(summ, ~ .x[[1]]), -->
<!--             `Est. Error` = map_dbl(summ, ~ .x[[2]]), -->
<!--             `Lower 95% CI` = map_dbl(summ, ~ .x[[3]]), -->
<!--             `Upper 95% CI` = map_dbl(summ, ~ .x[[4]]), -->
<!--             .groups = "drop") %>% -->
<!--   dplyr::select(-summ) %>% -->
<!--   mutate(sp = str_split(Parameter, "_"), -->
<!--          Type = factor(map_chr(sp, ~ .x[[1]]), c("h2", "corr", "VA", "cov")), -->
<!--          Parameter = map_chr(sp, ~ paste0(.x[-1], collapse = " - "))) %>% -->
<!--   arrange(Type) %>% -->
<!--   dplyr::select(-sp, -Type) %>% -->
<!--   kable(digits = 3) %>% -->
<!--   kable_styling(full_width = FALSE) %>% -->
<!--   pack_rows("Genetic correlations", 1, 8) %>% -->
<!--   pack_rows("Additive genetic variances", 9, 12) %>% -->
<!--   pack_rows("Additive genetic covariances", 13, 18) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.1.0 brms_2.14.4      Rcpp_1.0.4.6     forcats_0.5.0    stringr_1.4.0    dplyr_1.0.0     
 [7] purrr_0.3.4      readr_1.3.1      tidyr_1.1.0      tibble_3.0.1     ggplot2_3.3.2    tidyverse_1.3.0 

loaded via a namespace (and not attached):
  [1] readxl_1.3.1         backports_1.1.7      workflowr_1.6.2      plyr_1.8.6           igraph_1.2.5        
  [6] splines_4.0.3        crosstalk_1.1.0.1    TH.data_1.0-10       rstantools_2.1.1     inline_0.3.15       
 [11] digest_0.6.25        htmltools_0.5.0      rsconnect_0.8.16     fansi_0.4.1          magrittr_2.0.1      
 [16] modelr_0.1.8         RcppParallel_5.0.1   matrixStats_0.56.0   xts_0.12-0           sandwich_2.5-1      
 [21] prettyunits_1.1.1    colorspace_1.4-1     blob_1.2.1           rvest_0.3.5          haven_2.3.1         
 [26] xfun_0.19            callr_3.4.3          crayon_1.3.4         jsonlite_1.7.0       lme4_1.1-23         
 [31] survival_3.2-7       zoo_1.8-8            glue_1.4.2           gtable_0.3.0         emmeans_1.4.7       
 [36] webshot_0.5.2        V8_3.4.0             pkgbuild_1.0.8       rstan_2.21.2         abind_1.4-5         
 [41] scales_1.1.1         mvtnorm_1.1-0        DBI_1.1.0            miniUI_0.1.1.1       viridisLite_0.3.0   
 [46] xtable_1.8-4         stats4_4.0.3         StanHeaders_2.21.0-3 DT_0.13              htmlwidgets_1.5.1   
 [51] httr_1.4.1           threejs_0.3.3        ellipsis_0.3.1       pkgconfig_2.0.3      loo_2.3.1           
 [56] dbplyr_1.4.4         tidyselect_1.1.0     rlang_0.4.6          reshape2_1.4.4       later_1.0.0         
 [61] munsell_0.5.0        cellranger_1.1.0     tools_4.0.3          cli_2.0.2            generics_0.0.2      
 [66] broom_0.5.6          ggridges_0.5.2       evaluate_0.14        fastmap_1.0.1        yaml_2.2.1          
 [71] processx_3.4.2       knitr_1.30           fs_1.4.1             nlme_3.1-149         mime_0.9            
 [76] projpred_2.0.2       xml2_1.3.2           compiler_4.0.3       bayesplot_1.7.2      shinythemes_1.1.2   
 [81] rstudioapi_0.11      gamm4_0.2-6          curl_4.3             reprex_0.3.0         statmod_1.4.34      
 [86] stringi_1.5.3        highr_0.8            ps_1.3.3             Brobdingnag_1.2-6    lattice_0.20-41     
 [91] Matrix_1.2-18        nloptr_1.2.2.1       markdown_1.1         shinyjs_1.1          vctrs_0.3.0         
 [96] pillar_1.4.4         lifecycle_0.2.0      bridgesampling_1.0-0 estimability_1.3     httpuv_1.5.3.1      
[101] R6_2.4.1             promises_1.1.0       gridExtra_2.3        codetools_0.2-16     boot_1.3-25         
[106] colourpicker_1.0     MASS_7.3-53          gtools_3.8.2         assertthat_0.2.1     rprojroot_1.3-2     
[111] withr_2.2.0          shinystan_2.5.0      multcomp_1.4-13      mgcv_1.8-33          parallel_4.0.3      
[116] hms_0.5.3            grid_4.0.3           coda_0.19-3          minqa_1.2.4          rmarkdown_2.5       
[121] git2r_0.27.1         shiny_1.4.0.2        lubridate_1.7.8      base64enc_0.1-3      dygraphs_1.1.1.6    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
