<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Estimating genetic (co)variance and heritability</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Estimating genetic (co)variance and
heritability</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree3a5750d27bd2a271c583d40eae71f0ac91f48159targetblank3a5750da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/fitnessGWAS/tree/3a5750d27bd2a271c583d40eae71f0ac91f48159" target="_blank">3a5750d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree3a5750d27bd2a271c583d40eae71f0ac91f48159targetblank3a5750da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/fitnessGWAS/tree/3a5750d27bd2a271c583d40eae71f0ac91f48159" target="_blank">3a5750d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    old_analyses/.DS_Store

Untracked files:
    Untracked:  old_analyses/Data for old analyses/
    Untracked:  old_analyses/eQTL_analysis.Rmd
    Untracked:  old_analyses/fitness_data.csv
    Untracked:  old_analyses/gcta_quant_genetics_OLD.Rmd
    Untracked:  old_analyses/quantitative_genetics_OLD_brms.Rmd

Unstaged changes:
    Modified:   figures/fig3_boyle_plot.pdf
    Modified:   figures/fig4_mutation_load.pdf
    Modified:   figures/fig5_quartiles_plot.pdf
    Modified:   figures/fig6_antagonism_ratios.pdf
    Modified:   figures/fig7_models.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/quant_genetics.Rmd</code>) and
HTML (<code>docs/quant_genetics.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/358afb47ecdf8a5f5f2f16cb2d94fc30a90d5aff/analysis/quant_genetics.Rmd" target="_blank">358afb4</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
Tidying up and small edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/b3604fe70872246830cea295df76d6daaa28c02a/analysis/quant_genetics.Rmd" target="_blank">b3604fe</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-15
</td>
<td>
Tweaks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/ea08653bfa0cdbe3e8bfb6c43cbd93407a5350b6/docs/quant_genetics.html" target="_blank">ea08653</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/d16c98dea21fa40538307acfbf7fd5bf4d5cee89/analysis/quant_genetics.Rmd" target="_blank">d16c98d</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-13
</td>
<td>
wflow_publish("analysis/quant_genetics.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7fc071b186f5f9fbca000e6bd0f33fb63c23c5a4/docs/quant_genetics.html" target="_blank">7fc071b</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8613e36860859c03cae8e109c3bac731215e8621/analysis/quant_genetics.Rmd" target="_blank">8613e36</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-13
</td>
<td>
wflow_publish("analysis/quant_genetics.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(kableExtra)
library(rptR)
options(readr.show_col_types = FALSE)

# Load the &#39;raw&#39; data, giving the fitness of each replicate vial
fitness_data &lt;- bind_rows(
  read_csv(&quot;data/input/female_fitness_CLEANED.csv&quot;),
  read_csv(&quot;data/input/male_fitness_CLEANED.csv&quot;)) %&gt;% 
  mutate(male_fitness_early = early.male.focal / (early.male.focal + early.male.rival),
         male_fitness_late = late.male.focal / (late.male.focal + late.male.rival)) %&gt;% 
  rename_all(~ str_replace_all(.x, &quot;[.]&quot;, &quot;_&quot;))


# Load the estimated line mean fitness values calculated by brms
line_means &lt;- read_csv(&quot;data/derived/predicted_line_means.csv&quot;) %&gt;% 
  mutate(line = factor(str_remove_all(line, &quot;_&quot;))) %&gt;%
  filter(!is.na(male.fitness.early)) %&gt;%
  mutate(`Female fitness early` = female.fitness.early,
         `Female fitness late` = female.fitness.late,
         `Male fitness early` = male.fitness.early,
         `Male fitness late` = male.fitness.late) %&gt;%
  rename(female_fitness_early = female.fitness.early,
         female_fitness_late = female.fitness.late,
         male_fitness_early = male.fitness.early,
         male_fitness_late = male.fitness.late)

line_means %&gt;% 
  select(line, contains(&quot; &quot;)) %&gt;% gather(Trait, Fitness, -line) %&gt;% 
  ggplot(aes(line, ))</code></pre>
<p><img src="figure/quant_genetics.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="proportion-of-variance-in-fitness-explained-by-dgrp-line"
class="section level2 tabset">
<h2 class="tabset">Proportion of variance in fitness explained by DGRP
line</h2>
<p>Here, we calculate the proportion of variance in each fitness trait
that is explained by DGRP line. The proportion of explained variance
(also called “repeatability”) for DGRP line is approximately equal to
the broad sense heritability, since in our study flies from the same
line have identical genotypes.</p>
<p>To calculate repeatability, we use the approach recommended by
Nakagawa &amp; Schielzeth (2010, <em>Biological Reviews</em> 85:
935-956), which is implemented in the function <code>rpt</code> from the
package <code>rptR</code>. In short, in the code below, <code>rpt</code>
fits either a Poisson GLMM (for the two female fitness traits) or a
Binomial GLMM (for the two male fitness traits) using the
<code>lme4</code> package, and estimates the proportion of the total
variance that is explained for each random effect in the model. For all
four fitness traits, the GLMM we fitted had the model formula
<code>Y ~ (1 | line) + (1 | block)</code>, i.e. we fit a random
intercept for DGRP line and experimental block, and no fixed effects.
Using a model with the block effect included is more conservative and
more accurate, since not all lines were equally represented in every
block; therefore, we would artificially inflate the line-level
repeatability it we ignored variation in fitness between blocks.</p>
<p>When calculating repeatability from models with a link function, one
can calculate either the “Latent scale approximation” or the “Original
scale approximation” (see the <a
href="https://rdrr.io/cran/rptR/f/vignettes/rptR.Rmd">‘rptR’
vignette</a>). We focus on the former, though the repeatability
estimates are very similar, since in most of our analyses we focused on
the estimated fitness of each DGRP line on the latent scale.</p>
<p>For simplicity, we present just the results of interest in the first
table (which is the one discussed in the paper), and present all the
results in a second table that can be viewed by clicking the tab. The
“Simple table” shows the line-level repeatability, <span
class="math inline">\(R\)</span>, for each trait (the Latent scale
approximation), along with the associated SE and CIs from parametric
bootstrapping. The “Full table” also shows the Original scale
approximation estimates and the block effects.</p>
<pre class="r"><code>female_early_repeatability &lt;- rpt(female_fitness_early ~ (1 | line) + (1 | block),  
                                  grname = c(&quot;line&quot;, &quot;block&quot;),  
                                  data = fitness_data %&gt;% filter(!is.na(female_fitness_early)), 
                                  datatype = &quot;Poisson&quot;, nboot = 1000, npermut = 0)

female_late_repeatability &lt;- rpt(female_fitness_late ~ (1 | line) + (1 | block),  
                                  grname = c(&quot;line&quot;, &quot;block&quot;),  
                                  data = fitness_data %&gt;% filter(!is.na(female_fitness_late)), 
                                  datatype = &quot;Poisson&quot;, nboot = 1000, npermut = 0)

male_early_repeatability &lt;- rpt(cbind(early_male_focal, early_male_rival) ~ (1 | line) + (1 | block),  
                                grname = c(&quot;line&quot;, &quot;block&quot;),  
                                data = fitness_data %&gt;% 
                                  filter(!is.na(early_male_focal) &amp; !is.na(early_male_rival)) %&gt;% 
                                  filter(early_male_focal + early_male_rival &gt; 0), 
                                datatype = &quot;Proportion&quot;, nboot = 1000, npermut = 0)

male_late_repeatability &lt;- rpt(cbind(late_male_focal, late_male_rival) ~ (1 | line) + (1 | block),  
                                grname = c(&quot;line&quot;, &quot;block&quot;),  
                                data = fitness_data %&gt;% 
                                  filter(!is.na(late_male_focal) &amp; !is.na(late_male_rival)) %&gt;% 
                                  filter(late_male_focal + late_male_rival &gt; 0), 
                                datatype = &quot;Proportion&quot;, nboot = 1000, npermut = 0)

process_rpt_object &lt;- function(rpt, fitness_trait){
  
  link &lt;- cbind(data.frame(R = as.numeric(rpt$R[&quot;R_link&quot;, ]),
                   SE = as.numeric(rpt$se[&quot;se_link&quot;, ])),
        rpt$CI_emp$CI_link) %&gt;% mutate(Scale = &quot;Latent scale approximation&quot;,
                                       Predictor = c(&quot;line&quot;, &quot;block&quot;),
                                       Trait = fitness_trait)
  
  original &lt;- cbind(data.frame(R = as.numeric(rpt$R[&quot;R_org&quot;, ]),
                               SE = as.numeric(rpt$se[&quot;se_org&quot;, ])),
                    rpt$CI_emp$CI_org) %&gt;% mutate(Scale = &quot;Original scale approximation&quot;,
                                       Predictor = c(&quot;line&quot;, &quot;block&quot;),
                                       Trait = fitness_trait)
  
  bind_rows(link, original) %&gt;% 
    select(Trait, Scale, Predictor, everything()) %&gt;% 
    as_tibble() %&gt;% 
    rename(`Lower 95% CI` = `2.5%`, `Upper 95% CI` = `97.5%`)
}

repeatability_table &lt;- bind_rows(process_rpt_object(female_early_repeatability, &quot;Female fitness early&quot;),
          process_rpt_object(female_late_repeatability, &quot;Female fitness late&quot;),
          process_rpt_object(male_early_repeatability, &quot;Male fitness early&quot;),
          process_rpt_object(male_late_repeatability, &quot;Male fitness late&quot;)) 

simple_repeatability_table &lt;- repeatability_table %&gt;% 
  filter(Scale == &quot;Latent scale approximation&quot;, Predictor == &quot;line&quot;) %&gt;% 
  select(-Scale, -Predictor) 

saveRDS(simple_repeatability_table, &quot;data/derived/simple_repeatability_table.rds&quot;)</code></pre>
<div id="simple-table" class="section level3">
<h3>Simple table</h3>
<pre class="r"><code>simple_repeatability_table %&gt;% 
  kable(digits = 2) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Trait
</th>
<th style="text-align:right;">
R
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
Lower 95% CI
</th>
<th style="text-align:right;">
Upper 95% CI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
0.59
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
0.59
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.08
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:right;">
0.17
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
</tbody>
</table>
</div>
<div id="full-table" class="section level3">
<h3>Full table</h3>
<pre class="r"><code>repeatability_table %&gt;% 
  kable(digits = 2) %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Trait
</th>
<th style="text-align:left;">
Scale
</th>
<th style="text-align:left;">
Predictor
</th>
<th style="text-align:right;">
R
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
Lower 95% CI
</th>
<th style="text-align:right;">
Upper 95% CI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
0.59
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.11
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
0.50
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.16
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
0.59
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.35
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.09
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.01
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.08
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.09
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.16
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.08
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.09
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.16
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.17
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:left;">
Latent scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.13
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
line
</td>
<td style="text-align:right;">
0.14
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.10
</td>
<td style="text-align:right;">
0.19
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:left;">
Original scale approximation
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="correlations-in-the-line-means" class="section level2">
<h2>Correlations in the line means</h2>
<p>The table shows the Pearson correlation coefficients (<span
class="math inline">\(r\)</span>) between the DGRP line means (estimated
from the two <code>brms</code> models in the script
<code>get_predicted_line_means.Rmd</code>) for each pair of fitness
traits, across our sample of DGRP lines. These correlations approximate
the genetic correlations between each pair of fitness traits.</p>
<pre class="r"><code>tidy_correlation_test &lt;- function(x, y){
  
  focal_dat &lt;- line_means %&gt;% 
    select(!!x, !!y)
  
  focal_dat &lt;- focal_dat[complete.cases(focal_dat), ]
  
  x_dat &lt;- focal_dat %&gt;% pull(!! x)
  y_dat &lt;- focal_dat %&gt;% pull(!! y)

  se_r &lt;- function(corr){
    sqrt((1-corr$estimate^2) / (corr$parameter))
  }
    
  corr &lt;- cor.test(x_dat, y_dat)
   
  tibble(
    `Variable 1` = x, 
    `Variable 2` = y,
    `Pearson correlation` = as.numeric(corr$estimate),
    SE = se_r(corr),
    `Lower 95% CI` = corr$conf.int[1],
    `Upper 95% CI` = corr$conf.int[2],
    `p value` = corr$p.value
  ) %&gt;% mutate(
    `p value` = round(`p value`, 4),
    `p value` = replace(`p value`, `p value` &lt; 0.0001, &quot;&lt; 0.0001&quot;))
}

firstup &lt;- function(x) {
  substr(x, 1, 1) &lt;- toupper(substr(x, 1, 1))
  x
}

line_mean_corrs &lt;- bind_rows(
  tidy_correlation_test(&quot;female_fitness_early&quot;, &quot;female_fitness_late&quot;),
  tidy_correlation_test(&quot;female_fitness_early&quot;, &quot;male_fitness_early&quot;),
  tidy_correlation_test(&quot;female_fitness_early&quot;, &quot;male_fitness_late&quot;),
  tidy_correlation_test(&quot;female_fitness_late&quot;, &quot;male_fitness_early&quot;),
  tidy_correlation_test(&quot;female_fitness_late&quot;, &quot;male_fitness_late&quot;),
  tidy_correlation_test(&quot;male_fitness_early&quot;, &quot;male_fitness_late&quot;),
) %&gt;% mutate(`Variable 1` = str_replace_all(`Variable 1`, &quot;_&quot;, &quot; &quot;),
             `Variable 2` = str_replace_all(`Variable 2`, &quot;_&quot;, &quot; &quot;),
             `Variable 1` = firstup(`Variable 1`),
             `Variable 2` = firstup(`Variable 2`))

saveRDS(line_mean_corrs, &quot;data/derived/line_mean_corrs.rds&quot;)

line_mean_corrs %&gt;% kable(digits = 3) %&gt;% kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Variable 1
</th>
<th style="text-align:left;">
Variable 2
</th>
<th style="text-align:right;">
Pearson correlation
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
Lower 95% CI
</th>
<th style="text-align:right;">
Upper 95% CI
</th>
<th style="text-align:left;">
p value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:right;">
0.766
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.682
</td>
<td style="text-align:right;">
0.830
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:right;">
0.228
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.389
</td>
<td style="text-align:left;">
0.0108
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness early
</td>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:right;">
0.172
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.004
</td>
<td style="text-align:right;">
0.338
</td>
<td style="text-align:left;">
0.056
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:right;">
0.317
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.149
</td>
<td style="text-align:right;">
0.467
</td>
<td style="text-align:left;">
3e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
Female fitness late
</td>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:right;">
0.317
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.149
</td>
<td style="text-align:right;">
0.467
</td>
<td style="text-align:left;">
3e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
Male fitness early
</td>
<td style="text-align:left;">
Male fitness late
</td>
<td style="text-align:right;">
0.864
</td>
<td style="text-align:right;">
0.046
</td>
<td style="text-align:right;">
0.812
</td>
<td style="text-align:right;">
0.903
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
</tbody>
</table>
</div>
<div
id="testing-for-differences-in-genetic-correlations-between-sexes-and-ages"
class="section level2">
<h2>Testing for differences in genetic correlations between sexes and
ages</h2>
<p>We here use bootstrapping to test whether A) the inter-sex genetic
correlation for fitness changes between the two age classes, and B) the
genetic correlation between fitness in early- and late-life differs
between males and females.</p>
<div id="difference-in-inter-sex-correlation-between-age-classes"
class="section level3">
<h3>Difference in inter-sex correlation between age classes</h3>
<pre class="r"><code>set.seed(1)

female_w_early &lt;- line_means$female_fitness_early
male_w_early &lt;- line_means$male_fitness_early
female_w_late &lt;- line_means$female_fitness_late
male_w_late &lt;- line_means$male_fitness_late
cor1_actual &lt;- cor(female_w_early, male_w_early)
cor2_actual &lt;- cor(female_w_late, male_w_late)

n_bootstraps &lt;- 10000
n_lines &lt;- length(female_w_early)
cor1 &lt;- cor2 &lt;- rep(0, n_bootstraps)
for(i in 1:n_bootstraps){
  samp &lt;- sample(n_lines, n_lines, replace = TRUE)  
  cor1[i] &lt;- cor(female_w_early[samp], male_w_early[samp])
  cor2[i] &lt;- cor(female_w_late[samp], male_w_late[samp])
}

data.frame(diff = cor2 - cor1) %&gt;% 
  summarise(`Inter-sex correlation (early life)` = cor1_actual,
            `Inter-sex correlation (late life)` = cor2_actual,
            `Difference in estimated correlations` = cor2_actual - cor1_actual,
            `Difference, lower 95% CI (bootstrapped)` = 
              as.numeric(quantile(diff, probs = c(0.025))),
            `Difference, upper 95% CI (bootstrapped)` = 
              as.numeric(quantile(diff, probs = c(0.975)))) %&gt;%
  gather(Parameter, `Estimated value`) %&gt;% 
  kable(digits = 3) %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:right;">
Estimated value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Inter-sex correlation (early life)
</td>
<td style="text-align:right;">
0.228
</td>
</tr>
<tr>
<td style="text-align:left;">
Inter-sex correlation (late life)
</td>
<td style="text-align:right;">
0.317
</td>
</tr>
<tr>
<td style="text-align:left;">
Difference in estimated correlations
</td>
<td style="text-align:right;">
0.089
</td>
</tr>
<tr>
<td style="text-align:left;">
Difference, lower 95% CI (bootstrapped)
</td>
<td style="text-align:right;">
-0.057
</td>
</tr>
<tr>
<td style="text-align:left;">
Difference, upper 95% CI (bootstrapped)
</td>
<td style="text-align:right;">
0.232
</td>
</tr>
</tbody>
</table>
</div>
<div id="difference-in-inter-age-correlation-between-sexes"
class="section level3">
<h3>Difference in inter-age correlation between sexes</h3>
<pre class="r"><code>rm(cor1); rm(cor2)
cor1_actual &lt;- cor(female_w_early, female_w_late)
cor2_actual &lt;- cor(male_w_early, male_w_late)

n_bootstraps &lt;- 10000
n_lines &lt;- length(female_w_early)
cor1 &lt;- cor2 &lt;- rep(0, n_bootstraps)
for(i in 1:n_bootstraps){
  samp &lt;- sample(n_lines, n_lines, replace = TRUE)  
  cor1[i] &lt;- cor(female_w_early[samp], female_w_late[samp])
  cor2[i] &lt;- cor(male_w_early[samp], male_w_late[samp])
}


data.frame(diff = cor2 - cor1) %&gt;% 
  summarise(`Inter-age correlation (females)` = cor1_actual,
            `Inter-age correlation (males)` = cor2_actual,
            `Difference in estimated correlations` = cor2_actual - cor1_actual,
            `Difference, lower 95% CI (bootstrapped)` = 
              as.numeric(quantile(diff, probs = c(0.025))),
            `Difference, upper 95% CI (bootstrapped)` = 
              as.numeric(quantile(diff, probs = c(0.975)))) %&gt;% 
  gather(Parameter, `Estimated value`) %&gt;% 
  kable(digits = 3) %&gt;% kable_styling(full_width = F)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:right;">
Estimated value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Inter-age correlation (females)
</td>
<td style="text-align:right;">
0.766
</td>
</tr>
<tr>
<td style="text-align:left;">
Inter-age correlation (males)
</td>
<td style="text-align:right;">
0.864
</td>
</tr>
<tr>
<td style="text-align:left;">
Difference in estimated correlations
</td>
<td style="text-align:right;">
0.098
</td>
</tr>
<tr>
<td style="text-align:left;">
Difference, lower 95% CI (bootstrapped)
</td>
<td style="text-align:right;">
0.008
</td>
</tr>
<tr>
<td style="text-align:left;">
Difference, upper 95% CI (bootstrapped)
</td>
<td style="text-align:right;">
0.215
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>rm(cor1); rm(cor2)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rptR_0.9.22       kableExtra_1.3.4  lubridate_1.9.2   forcats_1.0.0    
 [5] stringr_1.5.0     dplyr_1.1.0       purrr_1.0.1       readr_2.1.4      
 [9] tidyr_1.3.0       tibble_3.1.8      ggplot2_3.4.1     tidyverse_2.0.0  
[13] workflowr_1.7.0.4

loaded via a namespace (and not attached):
 [1] httr_1.4.5        sass_0.4.5        bit64_4.0.5       vroom_1.6.1      
 [5] jsonlite_1.8.4    viridisLite_0.4.1 splines_4.2.2     bslib_0.4.2      
 [9] getPass_0.2-2     highr_0.10        yaml_2.3.7        pillar_1.8.1     
[13] lattice_0.20-45   glue_1.6.2        digest_0.6.31     promises_1.2.0.1 
[17] rvest_1.0.3       minqa_1.2.5       colorspace_2.1-0  htmltools_0.5.4  
[21] httpuv_1.6.9      Matrix_1.5-1      pkgconfig_2.0.3   scales_1.2.1     
[25] webshot_0.5.4     processx_3.8.0    svglite_2.1.1     whisker_0.4.1    
[29] later_1.3.0       tzdb_0.3.0        lme4_1.1-31       timechange_0.2.0 
[33] git2r_0.31.0      generics_0.1.3    ellipsis_0.3.2    cachem_1.0.7     
[37] withr_2.5.0       pbapply_1.7-0     cli_3.6.0         magrittr_2.0.3   
[41] crayon_1.5.2      evaluate_0.20     ps_1.7.2          fs_1.6.1         
[45] fansi_1.0.4       nlme_3.1-160      MASS_7.3-58.1     xml2_1.3.3       
[49] tools_4.2.2       hms_1.1.2         lifecycle_1.0.3   munsell_0.5.0    
[53] callr_3.7.3       compiler_4.2.2    jquerylib_0.1.4   systemfonts_1.0.4
[57] rlang_1.0.6       grid_4.2.2        nloptr_2.0.3      rstudioapi_0.14  
[61] rmarkdown_2.20    boot_1.3-28       gtable_0.3.1      R6_2.5.1         
[65] knitr_1.42        fastmap_1.1.1     bit_4.0.5         utf8_1.2.2       
[69] rprojroot_2.0.3   stringi_1.7.12    parallel_4.2.2    Rcpp_1.0.10      
[73] vctrs_0.5.2       tidyselect_1.2.0  xfun_0.37        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
