<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Transcriptome-wide association study (TWAS)</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Transcriptome-wide association study
(TWAS)</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree5204ef735e83bc0b12278d9a8cdc57205cbfd2cctargetblank5204ef7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lukeholman/fitnessGWAS/tree/5204ef735e83bc0b12278d9a8cdc57205cbfd2cc" target="_blank">5204ef7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStree5204ef735e83bc0b12278d9a8cdc57205cbfd2cctargetblank5204ef7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lukeholman/fitnessGWAS/tree/5204ef735e83bc0b12278d9a8cdc57205cbfd2cc" target="_blank">5204ef7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    old_analyses/.DS_Store

Untracked files:
    Untracked:  old_analyses/Data for old analyses/
    Untracked:  old_analyses/eQTL_analysis.Rmd
    Untracked:  old_analyses/fitness_data.csv
    Untracked:  old_analyses/gcta_quant_genetics_OLD.Rmd
    Untracked:  old_analyses/quantitative_genetics_OLD_brms.Rmd

Unstaged changes:
    Modified:   figures/fig3_boyle_plot.pdf
    Modified:   figures/fig4_mutation_load.pdf
    Modified:   figures/fig5_quartiles_plot.pdf
    Modified:   figures/fig6_antagonism_ratios.pdf
    Modified:   figures/fig7_models.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/TWAS.Rmd</code>) and HTML
(<code>docs/TWAS.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/5204ef735e83bc0b12278d9a8cdc57205cbfd2cc/analysis/TWAS.Rmd" target="_blank">5204ef7</a>
</td>
<td>
lukeholman
</td>
<td>
2023-03-21
</td>
<td>
wflow_publish("analysis/TWAS.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/7449a9097af63aa27121cdc1dc12b529a29695ca/docs/TWAS.html" target="_blank">7449a90</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/01226ab272891d92a8444e19cac034c1b079d263/analysis/TWAS.Rmd" target="_blank">01226ab</a>
</td>
<td>
lukeholman
</td>
<td>
2021-10-01
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/8d14298b4d59c2198b6f9c62d415e6adc9b7e658/docs/TWAS.html" target="_blank">8d14298</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/TWAS.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3124aadbe5529f0699cccfa26d211656fa275127/docs/TWAS.html" target="_blank">3124aad</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/6704dc8c17aeed88ef381f57849d09a6a4f40ec4/analysis/TWAS.Rmd" target="_blank">6704dc8</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-05
</td>
<td>
wflow_publish("analysis/TWAS.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/c18c92523fef9ce3a10d717544b9cb3f9e8a0818/docs/TWAS.html" target="_blank">c18c925</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/382f3a321944b25e83ca9516df8973032eae79f6/analysis/TWAS.Rmd" target="_blank">382f3a3</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-05
</td>
<td>
wflow_publish("analysis/TWAS.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3855d3359649c74f955042725d07276742cb649a/analysis/TWAS.Rmd" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3855d3359649c74f955042725d07276742cb649a/docs/TWAS.html" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/8d54ea56db332d3ac2d391c232936a4dc8106cc1/analysis/TWAS.Rmd" target="_blank">8d54ea5</a>
</td>
<td>
Luke Holman
</td>
<td>
2018-12-23
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(edgeR) # BiocManager::install(&quot;edgeR&quot;)
library(tidyverse) 
library(glue)
library(future)
library(future.apply)
library(parallel)
library(kableExtra)
library(DT)
library(ashr)
library(mashr)

options(stringsAsFactors = FALSE)

# Connect to the database of annotations
db &lt;- DBI::dbConnect(RSQLite::SQLite(), &quot;data/derived/annotations.sqlite3&quot;)

# Helper to run shell commands
run_command &lt;- function(shell_command, wd = getwd(), path = &quot;&quot;){
  cat(system(glue(&quot;cd &quot;, wd, path, &quot;\n&quot;,shell_command), intern = TRUE), sep = &#39;\n&#39;)
}

kable_table &lt;- function(df) { # cool tables
  kable(df, &quot;html&quot;) %&gt;%
  kable_styling() %&gt;%
  scroll_box(height = &quot;300px&quot;)
}

my_data_table &lt;- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;TWAS_enrichment&#39;)),
       columnDefs = list(list(targets = c(8,10), visible = FALSE)),
      pageLength = 50
    )
  )
}

# Helper to load Huang et al.&#39;s data
load_expression_data &lt;- function(sex, both_sex_chromosomes = TRUE){
  
  # Note: Huang et al&#39;s data contains weird stuff like supposedly Y-linked genes that have
  # higher/equal expression in *females* in all lines, presumably microarray issues/errors.
  # To be conservative, we restrict our analyses to genes that are known to be on a
  # chromosomes that is present in both sexes
  if(both_sex_chromosomes){
    genes_allowed &lt;- tbl(db, &quot;genes&quot;) %&gt;%
      filter(chromosome %in% c(&quot;2L&quot;, &quot;2R&quot;, &quot;3L&quot;, &quot;3R&quot;, &quot;4&quot;, &quot;X&quot;)) %&gt;%
      pull(FBID)
  } else {
    genes_allowed &lt;- tbl(db, &quot;genes&quot;) %&gt;%
      pull(FBID)
  }
  
  if(sex != &quot;both&quot;){
    expression &lt;- glue(&quot;data/input/huang_transcriptome/dgrp.array.exp.{sex}.txt&quot;) %&gt;% 
      read_delim(delim = &quot; &quot;) %&gt;%
      filter(gene %in% genes_allowed)
    sample_names &lt;- names(expression)[names(expression) != &quot;gene&quot;] %&gt;% str_remove(&quot;:[12]&quot;) 
    gene_names &lt;- expression$gene
    expression &lt;- expression %&gt;% select(-gene) %&gt;% as.matrix() %&gt;% t() 
    rownames(expression) &lt;- sample_names # rows are samples, columns are genes
    colnames(expression) &lt;- gene_names 
    return(expression %&gt;% as.data.frame() %&gt;% 
             tibble::rownames_to_column(&quot;line&quot;) %&gt;% 
             as_tibble() %&gt;%
             mutate(line = str_remove_all(line, &quot;[.]1&quot;)))
  }
  
  females &lt;- read_delim(&quot;data/input/huang_transcriptome/dgrp.array.exp.female.txt&quot;, delim = &quot; &quot;) %&gt;%
    filter(gene %in% genes_allowed)
  names(females)[-1] &lt;- paste(&quot;F_&quot;, names(females)[-1],sep=&quot;&quot;) #%&gt;% str_remove(&quot;:[12]&quot;) 
  females &lt;- females %&gt;% 
    left_join(read_delim(&quot;data/input/huang_transcriptome/dgrp.array.exp.male.txt&quot;, delim = &quot; &quot;), by = &quot;gene&quot;)
  sample_names &lt;- names(females)[names(females) != &quot;gene&quot;] %&gt;% str_remove(&quot;:[12]&quot;) 
  gene_names &lt;- females$gene
  sex &lt;- ifelse(str_detect(sample_names, &quot;F_&quot;), &quot;female&quot;, &quot;male&quot;)
  line &lt;- str_remove_all(sample_names, &quot;F_&quot;)
  females &lt;- females %&gt;% select(-gene) %&gt;% t()
  colnames(females) &lt;- gene_names
  list(
    sampleIDs = tibble(sex, line),
    expression = females
  )
}

# Table S5: heritability of the expression level of each transcript (as measured in males, or females)
# huang_heritability &lt;- read_csv(&quot;data/input/huang_2015_tableS5_transcript_heritability.csv&quot;)

# Load the predicted line means from present study, as calculated in get_predicted_line_means.Rmd
predicted_line_means &lt;- read_csv(&quot;data/derived/predicted_line_means.csv&quot;)</code></pre>
<!-- # Load the supplementary data files from Huang et al. 2015 PNAS -->
<!-- # Table S2: results of statistical tests for sex, line and sex-by-line effects on expression of each transcript -->
<!-- # huang_expression <- read_csv("data/input/huang_2015_tableS2_gene_expression.csv") -->
<!-- # # Table S11+S12: statistically significant eQTLs, and the transcripts they affect (for each sex) -->
<!-- # huang_eQTL_females <- read_csv("data/input/huang_2015_tableS11_eQTL_females.csv") %>% -->
<!-- #   left_join(tbl(db, "genes") %>% select(FBID, gene_name) %>% collect(), by = "FBID") %>% -->
<!-- #   rename(Affected_FBID = FBID, Affected_gene = gene_name) -->
<!-- #  -->
<!-- # huang_eQTL_males <- read_csv("data/input/huang_2015_tableS12_eQTL_males.csv") %>% -->
<!-- #   left_join(tbl(db, "genes") %>% select(FBID, gene_name) %>% collect(), by = "FBID") %>% -->
<!-- #   rename(Affected_FBID = FBID, Affected_gene = gene_name) -->
<!-- ``` -->
<div
id="calculate-sex-bias-in-expression-and-average-expression-for-each-transcript"
class="section level2">
<h2>Calculate sex bias in expression and average expression, for each
transcript</h2>
<p>The following takes the 368 female samples and the 369 male samples,
and finds the log fold difference in expression between sexes, and the
average expression across both sexes, using the <code>edgeR</code>
package.</p>
<pre class="r"><code>if(!file.exists(&quot;data/derived/gene_expression_by_sex.csv&quot;)){
expression_data_both_sexes &lt;- load_expression_data(&quot;both&quot;) %&gt;% unname()

voom_gene_data &lt;- calcNormFactors(DGEList(t(expression_data_both_sexes[[2]])))
mm &lt;- model.matrix(~ sex, data = expression_data_both_sexes[[1]])
colnames(mm) &lt;- gsub(&quot;sex&quot;, &quot;&quot;, colnames(mm))

sex_bias_in_expression &lt;- voom_gene_data %&gt;% 
  voom(mm, plot = FALSE) %&gt;% 
  lmFit(mm) %&gt;% 
  eBayes() %&gt;%
  topTable(n = Inf) %&gt;% 
  rownames_to_column(&quot;FBID&quot;) %&gt;%
  select(FBID, logFC, AveExpr) %&gt;% 
  rename(male_bias_in_expression = logFC) %&gt;%
  as_tibble() %&gt;% arrange(male_bias_in_expression)

write_csv(sex_bias_in_expression, &quot;data/derived/gene_expression_by_sex.csv&quot;)

} else {
  sex_bias_in_expression &lt;- read_csv(&quot;data/derived/gene_expression_by_sex.csv&quot;)
}</code></pre>
</div>
<div
id="using-linear-models-to-find-transcripts-whose-expression-correlates-with-fitness-across-lines-twas"
class="section level2">
<h2>Using linear models to find transcripts whose expression correlates
with fitness across lines (“TWAS”)</h2>
<p>This analysis uses the expression data from Huang et al. (2015),
which was downloaded from the <a
href="http://dgrp2.gnets.ncsu.edu/data.html">DGRP website</a>.</p>
<p>Here, we perform a large number of simple linear regressions, and
obtain the slope (beta or <span class="math inline">\(\beta\)</span>),
and the associated standard error, from a regression of transcript <span
class="math inline">\(i\)</span> on fitness trait <span
class="math inline">\(j\)</span>. The number of regressions run was
57,148, i.e. 4 fitness traits <span
class="math inline">\(\times\)</span> 14,287 transcripts. This approach
is often called a ‘TWAS’, i.e. transcriptome-wide association study.</p>
<pre class="r"><code>transcript_selection_analysis &lt;- function(expression_data, phenotypes){
  
  if(&quot;block&quot; %in% names(phenotypes)) phenotypes &lt;- phenotypes %&gt;% select(-block)
  
  expression_data &lt;- expression_data %&gt;% 
    filter(line %in% phenotypes$line)
  
  # Find line mean expression for each gene (average across the c. 2 replicate samples per line) 
  chunk_cols &lt;- split(2:ncol(expression_data), 
                      ceiling(seq_along(2:ncol(expression_data)) / 500))
  
  mean_expression_data &lt;- mclapply(1:2, function(i){  
    expression_data[, c(1, chunk_cols[[i]])] %&gt;% 
      group_by(line) %&gt;% 
      summarise_all(mean) %&gt;% 
      ungroup()
  }) %&gt;% bind_rows()

  # Scale each transcript&#39;s expression level so that mean is 0, 
  # and the variance is 1, across all the lines measured by Huang et al.
  for(i in 2:ncol(mean_expression_data)) mean_expression_data[,i] &lt;- as.numeric(scale(mean_expression_data[,i]))
  
  # Join the microarray data with the phenotypes (i.e. our fitness data), and 
  # keep only the lines where we have both sets of measurements
  expression_data &lt;- phenotypes %&gt;% left_join(expression_data, by = &quot;line&quot;)
  expression_data &lt;- expression_data[complete.cases(expression_data), ] %&gt;% select(-line)
  
  print(&quot;Data ready for analysis. Starting TWAS...&quot;)

  # Create chunks of transcript names, which will be used to facilitate parallel processing
  transcripts &lt;- names(expression_data)[-c(1:4)]
  transcripts &lt;- split(transcripts, ceiling(seq_along(transcripts) / 100))
  
  # Define a function to run 4 linear models, and get the beta and SE 
  # for regressions of expression level on the 4 fitness traits
  do_one_transcript &lt;- function(transcript){
    expression_level &lt;- expression_data %&gt;% pull(!!transcript)
    FE &lt;- summary(lm(female.fitness.early ~ expression_level, data = expression_data))$coefficients
    FL &lt;- summary(lm(female.fitness.late ~ expression_level, data = expression_data))$coefficients
    ME &lt;- summary(lm(male.fitness.early ~ expression_level, data = expression_data))$coefficients
    ML &lt;- summary(lm(male.fitness.late ~ expression_level, data = expression_data))$coefficients
    
    c(FE[2,1], FL[2,1], ME[2,1], ML[2,1], # effect size
      FE[2,2], FL[2,2], ME[2,2], ML[2,2], # SE
      FE[2,4], FL[2,4], ME[2,4], ML[2,4]) # p-value
  }
  
  # Runs do_one_transcript() on all the transcripts listed in the vector &#39;transcripts&#39;
  do_chunk_of_transcripts &lt;- function(transcripts){
    output &lt;- data.frame(transcripts, lapply(transcripts, do_one_transcript) %&gt;% do.call(&quot;rbind&quot;, .))
    names(output) &lt;- c(&quot;gene&quot;, &quot;beta_FE&quot;, &quot;beta_FL&quot;, &quot;beta_ME&quot;, &quot;beta_ML&quot;, 
                       &quot;SE_FE&quot;, &quot;SE_FL&quot;, &quot;SE_ME&quot;, &quot;SE_ML&quot;,
                       &quot;pval_FE&quot;, &quot;pval_FL&quot;, &quot;pval_ME&quot;, &quot;pval_ML&quot;)
    output
  }
  
  # Run it all, in parallel
  transcripts %&gt;% 
    mclapply(do_chunk_of_transcripts) %&gt;% 
    do.call(&quot;rbind&quot;, .) %&gt;% as_tibble() %&gt;% mutate(gene = as.character(gene))
}

if(!file.exists(&quot;data/derived/TWAS/TWAS_result_males.csv&quot;)){

  TWAS_result_females &lt;- load_expression_data(&quot;female&quot;) %&gt;% 
    transcript_selection_analysis(predicted_line_means) 
  TWAS_result_females %&gt;% write_csv(&quot;data/derived/TWAS/TWAS_result_females.csv&quot;)
  
  TWAS_result_males &lt;- load_expression_data(&quot;male&quot;) %&gt;% 
    transcript_selection_analysis(predicted_line_means)
  TWAS_result_males %&gt;% write_csv(&quot;data/derived/TWAS/TWAS_result_males.csv&quot;)
  
} else {
  TWAS_result_females &lt;- read_csv(&quot;data/derived/TWAS/TWAS_result_females.csv&quot;)
  TWAS_result_males &lt;- read_csv(&quot;data/derived/TWAS/TWAS_result_males.csv&quot;)
}</code></pre>
<div id="run-mashr-to-adjust-the-twas-results" class="section level3">
<h3>Run <code>mashr</code> to adjust the TWAS results</h3>
<p>This section re-uses the custom function <code>run_mashr()</code>.
See the earlier script (where <code>mashr</code> was applied to the GWAS
data) for the function definition. As the GWAS data, we use
<code>mashr</code>’s data-driven mode to derive adjusted (shrinked)
estimates of beta (i.e. the slope of the regression of transcript
abundance on the phenotype of interest), which are sensitive to the
covariance structure in the data (which is estimated from the data). As
well as producing these adjusted estimates, we derive the local false
sign rate for each combination of transcript and fitness trait (used
later to calculate evidence ratios).</p>
<pre class="r"><code>if(!file.exists(&quot;data/derived/TWAS/TWAS_ED.rds&quot;)){
  
  input_data &lt;- data.frame(TWAS_result_females[,1:3], 
                           TWAS_result_males[,4:5], 
                           TWAS_result_females[,6:7], 
                           TWAS_result_males[,8:9])
  
  TWAS_ED &lt;- input_data %&gt;% 
    run_mashr(mashr_mode = &quot;ED&quot;, 
              ED_p_cutoff = 0.4) 
  
  saveRDS(TWAS_ED, &quot;data/derived/TWAS/TWAS_ED.rds&quot;)
  
} else {
  TWAS_ED &lt;- readRDS(&quot;data/derived/TWAS/TWAS_ED.rds&quot;)
}</code></pre>
<!-- ### Find mixture probabilities for each transcript -->
<!-- This uses the canonical analysis' classifications. Each transcript gets a posterior probability that it belongs to the $i$'th mixture component -- only the mixture components that are not very rare are included. These are: `equal_effects` (i.e. the transcript is predicted to affect all 4 phenotypes equally), female-specific and male-specific (i.e. an effect on females/males only, which is concordant across age categories), sexually antagonistic (again, regardless of age), and null (note: the prior for `mashr` was that null transcripts are $10\times$ more common than any other type).  -->
<!-- ```{r get_mixture_assignments} -->
<!-- # Get the mixture weights, as advised by mashr authors here: https://github.com/stephenslab/mashr/issues/68 -->
<!-- posterior_weights_cov <- TWAS_canonical$posterior_weights  -->
<!-- colnames(posterior_weights_cov) <- sapply( -->
<!--   str_split(colnames(posterior_weights_cov), '\\.'),  -->
<!--   function(x) { -->
<!--     if(length(x) == 1) return(x) -->
<!--     else if(length(x) == 2) return(x[1]) -->
<!--     else if(length(x) == 3) return(paste(x[1], x[2], sep = ".")) -->
<!--   }) -->
<!-- posterior_weights_cov <- t(rowsum(t(posterior_weights_cov),  -->
<!--                                   colnames(posterior_weights_cov))) -->
<!-- # Make a neat dataframe -->
<!-- TWAS_mixture_assignment_probabilities <- data.frame( -->
<!--   FBID = TWAS_mashr_results$FBID, -->
<!--   posterior_weights_cov, -->
<!--   stringsAsFactors = FALSE -->
<!-- ) %>% as_tibble() %>% -->
<!--   rename(P_equal_effects = equal_effects, -->
<!--          P_female_specific = Female_specific_1, -->
<!--          P_male_specific = Male_specific_1, -->
<!--          P_null = null, -->
<!--          P_sex_antag = Sex_antag_0.25) -->
<!-- # TWAS_mixture_assignment_probabilities <- TWAS_mixture_assignment_probabilities %>% -->
<!-- #   left_join(huang_heritability, by = "FBID") -->
<!-- saveRDS(TWAS_mixture_assignment_probabilities,  -->
<!--         "data/derived/TWAS/TWAS_mixture_assignment_probabilities.rds") -->
<!-- ``` -->
</div>
</div>
<div id="tidy-and-save-the-results-of-twas" class="section level2">
<h2>Tidy and save the results of TWAS</h2>
<pre class="r"><code>TWAS_results &lt;- data.frame(
  FBID = TWAS_result_females$gene,
  as.data.frame(get_pm(TWAS_ED)),
  as.data.frame(get_lfsr(TWAS_ED))) %&gt;% 
  as_tibble()

names(TWAS_results)[6:9] &lt;- paste(
  &quot;LFSR&quot;, c(&quot;FE&quot;, &quot;FL&quot;, &quot;ME&quot;, &quot;ML&quot;), sep = &quot;_&quot;)

TWAS_results &lt;- TWAS_results  %&gt;%
  left_join(tbl(db, &quot;genes&quot;) %&gt;% 
              select(FBID, gene_name, chromosome) %&gt;% 
              collect(), by = &quot;FBID&quot;) %&gt;% 
  left_join(sex_bias_in_expression, by = &quot;FBID&quot;) %&gt;%
  mutate(across(where(is.numeric), ~ round(.x, 2))) %&gt;%
  as_tibble() %&gt;%
  distinct()

TWAS_results %&gt;%
  write_csv(&quot;data/derived/TWAS/TWAS_results.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] mashr_0.2.69        ashr_2.2-54         DT_0.27            
 [4] kableExtra_1.3.4    future.apply_1.10.0 future_1.30.0      
 [7] glue_1.6.2          lubridate_1.9.2     forcats_1.0.0      
[10] stringr_1.5.0       dplyr_1.1.0         purrr_1.0.1        
[13] readr_2.1.4         tidyr_1.3.0         tibble_3.1.8       
[16] ggplot2_3.4.1       tidyverse_2.0.0     edgeR_3.40.2       
[19] limma_3.54.2        workflowr_1.7.0.4  

loaded via a namespace (and not attached):
 [1] fs_1.6.1          bit64_4.0.5       webshot_0.5.4     httr_1.4.5       
 [5] rprojroot_2.0.3   tools_4.2.2       bslib_0.4.2       utf8_1.2.2       
 [9] R6_2.5.1          irlba_2.3.5.1     DBI_1.1.3         colorspace_2.1-0 
[13] rmeta_3.0         withr_2.5.0       tidyselect_1.2.0  processx_3.8.0   
[17] bit_4.0.5         compiler_4.2.2    git2r_0.31.0      cli_3.6.0        
[21] rvest_1.0.3       xml2_1.3.3        sass_0.4.5        scales_1.2.1     
[25] SQUAREM_2021.1    mvtnorm_1.1-3     callr_3.7.3       mixsqp_0.3-48    
[29] systemfonts_1.0.4 digest_0.6.31     rmarkdown_2.20    svglite_2.1.1    
[33] pkgconfig_2.0.3   htmltools_0.5.4   parallelly_1.34.0 dbplyr_2.3.0     
[37] fastmap_1.1.1     invgamma_1.1      htmlwidgets_1.6.1 rlang_1.0.6      
[41] rstudioapi_0.14   RSQLite_2.3.0     jquerylib_0.1.4   generics_0.1.3   
[45] jsonlite_1.8.4    vroom_1.6.1       magrittr_2.0.3    Matrix_1.5-1     
[49] Rcpp_1.0.10       munsell_0.5.0     fansi_1.0.4       abind_1.4-5      
[53] lifecycle_1.0.3   stringi_1.7.12    whisker_0.4.1     yaml_2.3.7       
[57] plyr_1.8.8        blob_1.2.3        grid_4.2.2        listenv_0.9.0    
[61] promises_1.2.0.1  crayon_1.5.2      lattice_0.20-45   hms_1.1.2        
[65] locfit_1.5-9.7    knitr_1.42        ps_1.7.2          pillar_1.8.1     
[69] codetools_0.2-18  evaluate_0.20     getPass_0.2-2     vctrs_0.5.2      
[73] tzdb_0.3.0        httpuv_1.6.9      gtable_0.3.1      assertthat_0.2.1 
[77] cachem_1.0.7      xfun_0.37         later_1.3.0       viridisLite_0.4.1
[81] truncnorm_1.0-8   memoise_2.0.1     timechange_0.2.0  globals_0.16.2   
[85] ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
