<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tables summarising the TWAS results</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fitnessGWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/fitnessGWAS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tables summarising the TWAS results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fitnessGWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180914code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180914)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180914code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180914)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreeaf15dd63a476b2499f2be7f3002df3ec72c28522targetblankaf15dd6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/fitnessGWAS/tree/af15dd63a476b2499f2be7f3002df3ec72c28522" target="_blank">af15dd6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanfitnessGWAStreeaf15dd63a476b2499f2be7f3002df3ec72c28522targetblankaf15dd6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/fitnessGWAS/tree/af15dd63a476b2499f2be7f3002df3ec72c28522" target="_blank">af15dd6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth
    Ignored:    .pversion
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/correlations_SNP_effects_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/Drosophila_GWAS.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/derived/
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/.pversion
    Ignored:    data/input/dgrp.fb557.annot.txt
    Ignored:    data/input/dgrp2.bed
    Ignored:    data/input/dgrp2.bim
    Ignored:    data/input/dgrp2.fam
    Ignored:    data/input/huang_transcriptome/
    Ignored:    figures/.DS_Store
    Ignored:    figures/fig1_inkscape.svg
    Ignored:    figures/figure1a.pdf
    Ignored:    figures/figure1b.pdf

Untracked files:
    Untracked:  big_model.rds
    Untracked:  code/gcta64
    Untracked:  code/plink
    Untracked:  code/quant_gen_1.R
    Untracked:  data/input/genomic_relatedness_matrix.rds
    Untracked:  figures/SNP_effect_ED.pdf
    Untracked:  figures/boyle_plot.pdf
    Untracked:  figures/fig1.pdf
    Untracked:  manuscript/GWAS_manuscript.log
    Untracked:  old_analyses/

Unstaged changes:
    Modified:   .gitignore
    Deleted:    Rplots.pdf
    Deleted:    analysis/GO_KEGG_enrichment.Rmd
    Deleted:    analysis/plotting_results.Rmd
    Deleted:    code/GO_and_KEGG_gsea.R
    Deleted:    code/make_sites_files_for_ARGweaver.R
    Deleted:    code/plotting_results.Rmd
    Deleted:    code/run_argweaver.R
    Modified:   code/run_mashr.R
    Deleted:    data/input/clough_2014_dsx_targets.csv
    Modified:   figures/GWAS_stats_figure.pdf
    Modified:   figures/TWAS_stats_figure.pdf
    Modified:   figures/antagonism_ratios.pdf
    Modified:   figures/composite_mixture_figure.pdf
    Modified:   figures/eff_size_histos.pdf
    Deleted:    figures/figure1.eps
    Deleted:    figures/figure2.eps
    Modified:   manuscript/GWAS_manuscript.Rmd
    Modified:   manuscript/GWAS_manuscript.pdf
    Modified:   manuscript/GWAS_manuscript.tex

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TWAS_tables.Rmd</code>) and HTML (<code>docs/TWAS_tables.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/af15dd63a476b2499f2be7f3002df3ec72c28522/analysis/TWAS_tables.Rmd" target="_blank">af15dd6</a>
</td>
<td>
lukeholman
</td>
<td>
2021-09-26
</td>
<td>
Commit Sept 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/372da3e1f31ee1905466e161ba4d89c135c796d0/docs/TWAS_tables.html" target="_blank">372da3e</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/ac8ee226735121184d893eabcccace22a9a4ac8f/analysis/TWAS_tables.Rmd" target="_blank">ac8ee22</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-05
</td>
<td>
wflow_publish(“analysis/TWAS_tables.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/fitnessGWAS/blob/3855d3359649c74f955042725d07276742cb649a/analysis/TWAS_tables.Rmd" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/fitnessGWAS/3855d3359649c74f955042725d07276742cb649a/docs/TWAS_tables.html" target="_blank">3855d33</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-04
</td>
<td>
big fist commit 2021
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(DT)
library(kableExtra)

kable_table &lt;- function(df) { # cool tables
  kable(df, &quot;html&quot;) %&gt;%
  kable_styling(full_width = FALSE) %&gt;%
  scroll_box(height = &quot;300px&quot;)
}

my_data_table &lt;- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;TWAS_sig_genes&#39;)),
      pageLength = 50
    )
  )
}

sig_transcripts_table &lt;- read_csv(&quot;data/derived/TWAS/TWAS_results.csv&quot;) %&gt;%
   filter(LFSR_FE &lt; 0.01 | LFSR_FL &lt; 0.01 | LFSR_ME &lt; 0.01 | LFSR_ML &lt; 0.01) 

tallies &lt;- sig_transcripts_table %&gt;%
  select(starts_with(&quot;sig_&quot;)) %&gt;%
  summarise_all(~ sum(!is.na(.x))) %&gt;%
  gather(type, n_transcripts) %&gt;%
  arrange(-n_transcripts)

rnd&lt;- function(x) format(round(x, 2), nsmall = 2)


big_table &lt;- sig_transcripts_table %&gt;% 
  select(FBID, gene_name, chromosome, AveExpr, male_bias_in_expression, everything()) %&gt;%
  select(-contains(&quot;heritability&quot;)) %&gt;%
  mutate(beta_FE = paste(beta_FE, &quot; (&quot;, rnd(LFSR_FE), &quot;)&quot;, sep = &quot;&quot;),
         beta_FL = paste(beta_FL, &quot; (&quot;, rnd(LFSR_FL), &quot;)&quot;, sep = &quot;&quot;),
         beta_ME = paste(beta_ME, &quot; (&quot;, rnd(LFSR_ME), &quot;)&quot;, sep = &quot;&quot;),
         beta_ML = paste(beta_ML, &quot; (&quot;, rnd(LFSR_ML), &quot;)&quot;, sep = &quot;&quot;)) %&gt;%
  select(-contains(&quot;LFSR&quot;)) %&gt;%
  rename(`Gene name` = gene_name,
         Chromosome = chromosome,
         `Female early` = beta_FE,
         `Female late` = beta_FL,
         `Male early` = beta_ME,
         `Male late` = beta_ML,
         `Male bias in expression (logFC)` = male_bias_in_expression,
         `Average expression level` = AveExpr) %&gt;%
  arrange(-P_sex_antag)</code></pre>
<div id="table-showing-the-significant-transcripts" class="section level2">
<h2>Table showing the significant transcripts</h2>
<p>The table shows the 99 transcripts that were associated with one or more of the four phenotypes, with a local false sign rate &lt; 0.01 (LFSR; computed by <code>mashr</code>). Columns 5-8 show the regression coefficients (adjusted with <code>mashr</code> in data-driven mode) that relate the line mean transcript abundance to the line mean fitnesses, while columns 9-12 give the local false sign rates that for these coefficients from <code>mashr</code>. The <code>sig_</code> (signficant) columns provide a binary summary of whether the transcript was significantly pleiotropic between ages and sex classes (see below). Finally, the last five columns give the mixture assignment probabilities from <code>mashr</code> (this time, run in canonical mode).</p>
<p>To make the ‘significance’ columns, we defined significantly antagonistic transcripts as those where the relationship with fitness is significantly positive for one sex (or age class) and significantly negative for the other, with LFSR &lt; 0.01. Similarly, we define significantly concordant transcripts as those where the relationship with fitness is significantly positive for one sex or age class and also significantly positive for the other (LFSR &lt; 0.01). This is quite conservative, because a transcript needs to have a LFSR &lt; 0.01 for two tests, giving two chances for a ‘false negative’. One can sort the table by these columns, for example to see a list of transcripts that showed a significantly antagonistic relationship with fitness in the early-life fitness assays, sort by the <code>sig_SA_early</code> column.</p>
<p>Abbreviations: AC = age concordant, SC = sexually concordant, SA = sexually antagonistic.</p>
<pre class="r"><code>big_table %&gt;% my_data_table()</code></pre>
<div id="htmlwidget-d0f3eb3d5ad01f715d25" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d0f3eb3d5ad01f715d25">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["FBgn0263092","FBgn0000022","FBgn0020645","FBgn0034099","FBgn0261615","FBgn0013275","FBgn0035372","FBgn0265422","FBgn0030645","FBgn0031756","FBgn0265349","FBgn0053109","FBgn0028700","FBgn0052440","FBgn0086072","FBgn0034335","FBgn0050100","FBgn0015765","FBgn0083035","FBgn0083039","FBgn0028955","FBgn0039472","FBgn0264720","FBgn0041579","FBgn0039040","FBgn0039203","FBgn0051133","FBgn0263331","FBgn0010357","FBgn0085732","FBgn0031940","FBgn0040801","FBgn0041581","FBgn0002564","FBgn0029800","FBgn0031729","FBgn0032205","FBgn0052071","FBgn0014454","FBgn0063493","FBgn0031547","FBgn0032593","FBgn0034578","FBgn0046793","FBgn0262961","FBgn0031774","FBgn0040989","FBgn0263320","FBgn0263449","FBgn0039437","FBgn0050457","FBgn0082953","FBgn0265064","FBgn0036157","FBgn0011361","FBgn0031942","FBgn0262985","FBgn0015527","FBgn0031360","FBgn0031378","FBgn0262022","FBgn0003189","FBgn0031824","FBgn0039474","FBgn0260653","FBgn0037368","FBgn0261932","FBgn0262838","FBgn0033426","FBgn0036290","FBgn0040070","FBgn0051054","FBgn0025739","FBgn0044812","FBgn0035736","FBgn0082960","FBgn0004414","FBgn0044809","FBgn0045800","FBgn0033598","FBgn0038236","FBgn0031397","FBgn0034325","FBgn0250820","FBgn0259937","FBgn0058006","FBgn0033807","FBgn0004240","FBgn0015521","FBgn0030603","FBgn0052523","FBgn0032376","FBgn0028396","FBgn0053263","FBgn0085250","FBgn0030501","FBgn0032763","FBgn0037836","FBgn0260746"],["pseudo","achaete","Lcp65Aa","uncharacterized protein","uncharacterized protein","Heat-shock-protein-70Aa","uncharacterized protein","long non-coding RNA:CR44334","ALG14, UDP-N-acetylglucosaminyltransferase subunit","uncharacterized protein","Seminal fluid protein 33A4","uncharacterized protein","Replication factor C 38kD subunit","uncharacterized protein","ncRNA","Glutathione S transferase E1","uncharacterized protein","p38a MAP kinase","ncRNA","ncRNA","uncharacterized protein","uncharacterized protein","long non-coding RNA:CR43988","Attacin-C","uncharacterized protein","uncharacterized protein","Seryl-tRNA synthetase-like insect mitochondrial protein","long non-coding RNA:CR43412","betaTrypsin","pseudo","uncharacterized protein","uncharacterized protein","Attacin-B","Larval serum protein 1 gamma","lin-52","uncharacterized protein","uncharacterized protein","uncharacterized protein","Adult cuticle protein 1","Glutathione S transferase E7","Scavenger receptor class C, type IV","Transient receptor potential cation channel gamma","uncharacterized protein","uncharacterized protein","uncharacterized protein","uncharacterized protein","uncharacterized protein","uncharacterized protein","antisense RNA:CR43472","TweedleL","uncharacterized protein","ncRNA","long non-coding RNA:CR44175","uncharacterized protein","NADH dehydrogenase (ubiquinone) acyl carrier protein","uncharacterized protein","uncharacterized protein","penguin","uncharacterized protein","uncharacterized protein","uncharacterized protein","rudimentary","uncharacterized protein","uncharacterized protein","serpentine","uncharacterized protein","uncharacterized protein","uncharacterized protein","uncharacterized protein","uncharacterized protein","thioredoxin-2","pseudo","partner of numb","Turandot C","Cuticular protein 65Eb","ncRNA","male-specific opa containing gene","Turandot Z","U snoRNA host gene 1","Cuticular protein 47Eb","Cyp313a1","uncharacterized protein","uncharacterized protein","medial glomeruli","Nucleolar protein at 60B","uncharacterized protein","aquaporin","Diptericin A","Ribosomal protein S21","uncharacterized protein","uncharacterized protein","Tetraspanin 33B","Turandot A","uncharacterized protein","uncharacterized protein","BthD selenoprotein","uncharacterized protein","uncharacterized protein","Ectoderm-expressed 3"],["2R","X","3L","2R","X","3R","3L","3R","X","2L","2L","3R","2L","3L","3L","2R","2R","3R","2L","3R","2R","3R","3L","2R","3R","3R","3R","2L","2R","2R","2L","3L","2R","3L","X","2L","2L","3L","2L","2R","2L","2L","2R","3L","2R","2L","2L","2L","3R","3R","2R","3L","3R","3L","3L","2L","3L","X","2L","2L","3L","X","2L","3R","3L","3R","3R","2R","2R","3L","2L","3R","X","3R","3L","3R","3L","3R","2R","2R","3R","2L","2R","3R","2R","2L","2R","2R","2L","X","X","2L","3R","3L","2R","X","2L","3R","3R"],[5.43,5.52,5.53,6.05,6.15,5.23,6.42,5.26,6.21,6.04,6.41,6.47,5.45,6.19,6.29,6.73,6.59,6.31,5.78,6.44,6.64,5.78,5.78,6.46,6.16,5.83,5.81,5.46,7.16,6.15,6.61,6.7,6.35,5.77,5.87,6.07,6.04,5.73,6.58,5.81,5.41,5.85,5.85,6.09,5.54,6.09,6.14,6.2,5.38,6.23,5.74,5.76,5.71,5.8,6.87,6.84,6.18,6.37,6.67,6.23,5.81,6.55,5.84,6.25,6.37,5.99,5.55,5.49,6.5,5.99,6.87,6.02,6.17,6,6.55,6.79,6.33,5.33,5.98,6.07,6.61,5.94,5.94,6.63,6.75,6.25,6.62,6.51,7.13,6.69,6.1,5.99,6.25,5.34,5.62,6.17,6.19,5.4,6.1],[-0.05,-0.13,0.41,0.83,-1.69,-0,-0.19,-0.06,-0,-0.35,0.78,-0.05,-0.37,1.51,-0.25,-0.1,-0.26,-0.19,-0.26,-0.53,-0.02,0.05,0.08,0.02,0.57,0.35,-0.22,0.23,0.04,-0.36,0.16,0.01,0.04,-0.02,-0.32,0.46,-0.24,0,0.32,0.13,0.02,0.05,-0.23,1.2,0.67,0.37,0.97,0.13,0.24,-0.04,-0.16,-0.15,0.04,-0.01,-0.01,0.1,0.89,-0.59,-0.06,-0.03,0.17,-0.52,-0.28,0.37,0.05,-0.86,0.2,0.08,0.09,-0.19,-0.13,-0.24,-0.6,-0,0.07,-0.14,1.84,-0.07,-0.88,-0.38,0.55,0.19,0.27,-0,-0.76,-0.2,0.23,0.07,-0.15,-0.18,0.08,0.66,0.03,-0.03,-0.05,-0.83,-0.22,0.06,0.04],["-0.25 (0.01)","-0.31 (0.00)","-0.3 (0.00)","-0.27 (0.00)","0.15 (0.01)","-0.11 (0.00)","-0.17 (0.05)","-0.13 (0.06)","-0.34 (0.00)","-0.14 (0.04)","0.13 (0.00)","0.06 (0.05)","-0.46 (0.00)","-0.09 (0.15)","0.08 (0.00)","-0.06 (0.14)","0.05 (0.29)","-0.07 (0.20)","-0.04 (0.02)","-0.03 (0.10)","0.23 (0.00)","0.01 (0.34)","-0.07 (0.21)","-0.04 (0.13)","-0.29 (0.00)","-0.01 (0.43)","-0.06 (0.25)","-0.01 (0.44)","-0.1 (0.00)","-0.03 (0.21)","0.27 (0.00)","0.14 (0.00)","-0.04 (0.14)","0 (0.49)","-0.1 (0.03)","-0.04 (0.26)","0.01 (0.47)","0.21 (0.00)","0.16 (0.00)","-0.16 (0.00)","0.03 (0.34)","-0.02 (0.43)","0.01 (0.42)","0.36 (0.00)","-0.05 (0.23)","-0.41 (0.00)","-0.06 (0.23)","0.01 (0.46)","0.06 (0.23)","0.05 (0.01)","0.02 (0.31)","0.02 (0.20)","0.05 (0.01)","-0.12 (0.03)","-0.35 (0.00)","0.13 (0.00)","-0.05 (0.17)","-0.44 (0.00)","-0.11 (0.11)","-0.09 (0.17)","-0.09 (0.03)","-0.54 (0.00)","0.13 (0.03)","-0.25 (0.00)","0.4 (0.00)","-0.08 (0.23)","0.15 (0.02)","0.01 (0.45)","-0.12 (0.14)","-0.1 (0.00)","-0.17 (0.03)","0.06 (0.00)","-0.37 (0.00)","-0.11 (0.00)","0.07 (0.27)","-0.08 (0.01)","-0.14 (0.01)","-0.21 (0.00)","0.11 (0.00)","0.24 (0.00)","0.11 (0.00)","-0.21 (0.04)","-0.12 (0.00)","-0.21 (0.02)","-0.34 (0.00)","0.37 (0.00)","0.22 (0.01)","-0.1 (0.00)","-0.26 (0.00)","-0.42 (0.00)","-0.21 (0.00)","0.25 (0.01)","-0.1 (0.00)","0.26 (0.00)","0.24 (0.00)","-0.25 (0.00)","0.32 (0.00)","0.29 (0.01)","-0.34 (0.00)"],["-0.24 (0.01)","-0.3 (0.00)","-0.3 (0.00)","-0.26 (0.00)","0.14 (0.01)","-0.11 (0.01)","-0.16 (0.06)","-0.13 (0.07)","-0.34 (0.00)","-0.13 (0.05)","0.13 (0.00)","0.05 (0.06)","-0.46 (0.00)","-0.08 (0.17)","0.08 (0.00)","-0.06 (0.17)","0.05 (0.32)","-0.06 (0.24)","-0.04 (0.01)","-0.03 (0.09)","0.23 (0.00)","0.01 (0.39)","-0.06 (0.25)","-0.04 (0.10)","-0.3 (0.00)","-0.01 (0.47)","-0.05 (0.30)","0 (0.48)","-0.11 (0.00)","-0.03 (0.16)","0.27 (0.00)","0.14 (0.00)","-0.05 (0.12)","0.01 (0.45)","-0.11 (0.02)","-0.04 (0.23)","0 (0.49)","0.22 (0.00)","0.16 (0.00)","-0.16 (0.00)","0.04 (0.30)","-0.01 (0.48)","0.02 (0.38)","0.37 (0.00)","-0.06 (0.20)","-0.41 (0.00)","-0.07 (0.20)","0.02 (0.41)","0.07 (0.21)","0.05 (0.01)","0.03 (0.27)","0.03 (0.14)","0.05 (0.01)","-0.13 (0.02)","-0.35 (0.00)","0.13 (0.00)","-0.06 (0.13)","-0.44 (0.00)","-0.12 (0.09)","-0.1 (0.14)","-0.09 (0.02)","-0.54 (0.00)","0.14 (0.03)","-0.25 (0.00)","0.41 (0.00)","-0.09 (0.19)","0.15 (0.02)","0 (0.48)","-0.13 (0.11)","-0.11 (0.00)","-0.18 (0.02)","0.07 (0.00)","-0.38 (0.00)","-0.11 (0.00)","0.08 (0.22)","-0.08 (0.00)","-0.15 (0.01)","-0.22 (0.00)","0.11 (0.00)","0.24 (0.00)","0.11 (0.00)","-0.22 (0.03)","-0.12 (0.00)","-0.23 (0.02)","-0.35 (0.00)","0.38 (0.00)","0.22 (0.00)","-0.1 (0.00)","-0.27 (0.00)","-0.43 (0.00)","-0.22 (0.00)","0.26 (0.01)","-0.1 (0.00)","0.27 (0.00)","0.25 (0.00)","-0.27 (0.00)","0.33 (0.00)","0.31 (0.01)","-0.36 (0.00)"],["0.26 (0.00)","0.21 (0.01)","0.16 (0.03)","0.16 (0.03)","-0.2 (0.00)","0.12 (0.00)","0.37 (0.00)","0.23 (0.00)","0.13 (0.08)","0.35 (0.00)","-0.08 (0.05)","-0.12 (0.00)","0.14 (0.14)","0.25 (0.00)","0.02 (0.09)","0.21 (0.00)","-0.23 (0.00)","0.32 (0.00)","-0.08 (0.00)","-0.07 (0.00)","-0.03 (0.29)","-0.13 (0.00)","0.35 (0.00)","-0.12 (0.00)","0.01 (0.43)","0.28 (0.00)","0.4 (0.00)","0.15 (0.00)","-0.19 (0.00)","-0.13 (0.00)","0.1 (0.08)","0.01 (0.44)","-0.15 (0.00)","0.28 (0.00)","-0.25 (0.00)","-0.16 (0.00)","-0.26 (0.00)","0.07 (0.12)","0.03 (0.24)","-0.03 (0.22)","0.23 (0.00)","0.36 (0.00)","0.18 (0.00)","0.11 (0.12)","-0.19 (0.00)","-0.1 (0.17)","-0.23 (0.00)","0.28 (0.00)","0.22 (0.00)","0.09 (0.00)","0.18 (0.00)","0.15 (0.00)","0.07 (0.00)","-0.26 (0.00)","-0.13 (0.12)","0.06 (0.09)","-0.26 (0.00)","-0.09 (0.24)","-0.32 (0.00)","-0.34 (0.00)","-0.17 (0.00)","0 (0.48)","0.28 (0.00)","-0.15 (0.00)","0.18 (0.04)","-0.41 (0.00)","0.28 (0.00)","-0.33 (0.00)","-0.45 (0.00)","-0.07 (0.03)","-0.34 (0.00)","0.1 (0.00)","-0.18 (0.09)","-0.18 (0.00)","0.5 (0.00)","-0.11 (0.00)","-0.23 (0.00)","-0.16 (0.02)","0.08 (0.01)","0.18 (0.03)","0.13 (0.01)","-0.37 (0.00)","-0.12 (0.01)","-0.39 (0.00)","-0.23 (0.00)","0.24 (0.01)","0.22 (0.01)","-0.13 (0.00)","-0.24 (0.01)","-0.25 (0.03)","-0.23 (0.00)","0.39 (0.00)","-0.12 (0.00)","0.29 (0.00)","0.29 (0.00)","-0.35 (0.00)","0.32 (0.01)","0.42 (0.00)","-0.53 (0.00)"],["0.26 (0.00)","0.22 (0.01)","0.17 (0.02)","0.17 (0.03)","-0.2 (0.00)","0.12 (0.00)","0.37 (0.00)","0.23 (0.00)","0.14 (0.07)","0.35 (0.00)","-0.08 (0.04)","-0.12 (0.00)","0.14 (0.12)","0.25 (0.00)","0.02 (0.10)","0.21 (0.00)","-0.23 (0.00)","0.32 (0.00)","-0.08 (0.00)","-0.07 (0.00)","-0.04 (0.27)","-0.13 (0.00)","0.35 (0.00)","-0.11 (0.00)","0.02 (0.40)","0.28 (0.00)","0.4 (0.00)","0.14 (0.00)","-0.19 (0.00)","-0.13 (0.00)","0.1 (0.09)","0 (0.46)","-0.15 (0.00)","0.28 (0.00)","-0.25 (0.00)","-0.16 (0.00)","-0.26 (0.00)","0.07 (0.13)","0.03 (0.26)","-0.03 (0.24)","0.23 (0.00)","0.36 (0.00)","0.18 (0.00)","0.1 (0.13)","-0.19 (0.00)","-0.09 (0.18)","-0.23 (0.00)","0.28 (0.00)","0.22 (0.00)","0.09 (0.00)","0.18 (0.00)","0.15 (0.00)","0.07 (0.00)","-0.26 (0.00)","-0.12 (0.13)","0.06 (0.10)","-0.25 (0.00)","-0.09 (0.26)","-0.32 (0.00)","-0.34 (0.00)","-0.17 (0.00)","0.01 (0.49)","0.28 (0.00)","-0.14 (0.00)","0.17 (0.04)","-0.41 (0.00)","0.28 (0.00)","-0.33 (0.00)","-0.44 (0.00)","-0.07 (0.03)","-0.34 (0.00)","0.09 (0.00)","-0.17 (0.09)","-0.18 (0.00)","0.5 (0.00)","-0.11 (0.00)","-0.22 (0.00)","-0.15 (0.02)","0.08 (0.01)","0.17 (0.03)","0.13 (0.01)","-0.37 (0.00)","-0.12 (0.01)","-0.39 (0.00)","-0.22 (0.00)","0.23 (0.01)","0.21 (0.01)","-0.13 (0.00)","-0.23 (0.01)","-0.24 (0.03)","-0.23 (0.00)","0.39 (0.00)","-0.11 (0.00)","0.29 (0.00)","0.28 (0.00)","-0.34 (0.00)","0.31 (0.01)","0.41 (0.00)","-0.52 (0.00)"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"*",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"*",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"*",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"*",null,null,null,null,null,null,null,"*",null,"*",null,"*","*",null,null,null,null,null,"*",null,"*",null,"*","*",null,null,"*",null,"*","*","*","*","*",null,"*"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"*","*",null,null,null,"*",null,null,null,null,"*","*",null,null,"*",null,"*","*","*",null,null,"*",null,null,null,null,"*","*","*",null,null,null,null,null,null,"*","*",null,"*","*","*","*","*",null,null,"*",null,"*",null,"*","*","*","*",null,"*","*",null,"*","*","*",null,null,null,"*","*","*","*",null,null,"*","*","*","*",null,"*",null,null,"*",null,null,"*"],["*","*","*","*",null,"*",null,null,"*",null,null,null,"*",null,null,null,null,null,null,null,null,null,null,null,"*",null,null,null,"*",null,null,null,null,null,null,null,null,null,null,"*",null,null,null,null,null,"*",null,null,null,null,null,null,null,null,"*",null,null,"*",null,null,null,"*",null,"*",null,null,null,null,null,"*",null,null,"*","*",null,"*","*","*",null,null,null,null,"*",null,"*",null,null,"*","*","*","*",null,"*",null,null,"*",null,null,"*"],[null,null,null,null,"*",null,null,null,null,null,null,"*",null,null,null,null,"*",null,"*","*",null,"*",null,"*",null,null,null,null,"*","*",null,null,"*",null,"*","*","*",null,null,null,null,null,null,null,"*",null,"*",null,null,null,null,null,null,"*",null,null,"*",null,"*","*","*",null,null,"*",null,"*",null,"*","*",null,"*",null,null,"*",null,"*","*",null,null,null,null,"*","*","*","*",null,null,"*",null,null,"*",null,"*",null,null,"*",null,null,"*"],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.1,0,0.02,0,0.39,0.34,0,0.01,0,0.26,0.01,0.01,0,0.01,0.58,0.04,0.29,0.04,0.13,0.03,0.37,0.2,0.02,0.24,0.12,0.11,0.08,0.01,0.03,0.18,0.21,0.1,0.16,0.03,0.19,0.6,0.04,0.01,0.61,0.47,0.34,0.45,0.03,0.12,0.22,0.1,0.6,0.01,0.52,0.71,0.49,0.06,0.6,0,0.12,0.65,0.58,0.73,0.55,0.76,0.02,0.75,0.75,0.74,0.75,0.74,0.78,0.67,0.8,0.69,0.82,0.78,0.82,0.86,0.83,0.74,0.86,0.82,0.88,0.88,0.88,0.9,0.89,0.88,0.93],[0.01,0.04,0.1,0.08,0.01,0.01,0,0,0.22,0,0.12,0,0.38,0,0.26,0,0,0,0,0,0.52,0,0,0,0.58,0,0,0,0,0,0.32,0.53,0,0,0,0,0,0.38,0.49,0.5,0,0,0,0.47,0,0.57,0,0,0,0,0,0,0,0,0.34,0.21,0,0.57,0,0,0,0.71,0,0.01,0.24,0,0,0,0,0.07,0,0,0.22,0,0,0,0.01,0.05,0.03,0.07,0.02,0,0.01,0,0.01,0.06,0.02,0,0.02,0.11,0,0,0,0,0,0,0.01,0,0],[0.02,0,0,0,0.04,0.01,0.2,0.18,0,0.23,0,0.13,0,0.36,0,0.4,0.42,0.5,0.06,0.12,0,0.48,0.56,0.28,0,0.59,0.6,0.53,0.03,0.53,0,0,0.46,0.59,0.25,0.4,0.6,0,0,0,0.56,0.63,0.59,0,0.42,0,0.5,0.63,0.45,0.03,0.6,0.64,0.02,0.21,0,0,0.66,0,0.47,0.6,0.09,0,0.19,0,0,0.68,0.14,0.74,0.63,0,0.17,0.01,0,0,0.77,0.01,0.03,0,0,0,0,0.15,0,0.14,0.01,0,0.01,0,0,0,0,0.06,0,0.01,0.01,0.01,0.01,0.04,0.02],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.96,0.95,0.89,0.89,0.89,0.8,0.79,0.78,0.77,0.76,0.72,0.71,0.62,0.61,0.56,0.55,0.54,0.49,0.49,0.48,0.45,0.44,0.44,0.41,0.4,0.4,0.4,0.4,0.39,0.39,0.38,0.38,0.38,0.37,0.37,0.37,0.37,0.37,0.36,0.36,0.35,0.35,0.35,0.35,0.35,0.34,0.34,0.34,0.34,0.33,0.33,0.33,0.33,0.32,0.31,0.31,0.31,0.3,0.3,0.3,0.29,0.28,0.28,0.28,0.27,0.26,0.26,0.26,0.25,0.25,0.25,0.24,0.23,0.23,0.22,0.22,0.21,0.2,0.2,0.19,0.19,0.17,0.17,0.17,0.17,0.16,0.15,0.14,0.14,0.14,0.13,0.12,0.11,0.11,0.11,0.09,0.09,0.08,0.05],[0.01,0.01,0.01,0.02,0.06,0.18,0.01,0.03,0.01,0.01,0.16,0.16,0,0.02,0.07,0.04,0.02,0.01,0.06,0.06,0.02,0.07,0,0.05,0.01,0,0,0.05,0,0.04,0,0.04,0.03,0.01,0,0.03,0.01,0.01,0.03,0.02,0.01,0,0.02,0,0.02,0,0.01,0.01,0.01,0.04,0.02,0.03,0.04,0,0,0.02,0,0,0,0,0.01,0,0,0,0,0,0,0,0,0.02,0,0.02,0,0,0,0.02,0,0,0.01,0,0.01,0,0.01,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FBID<\/th>\n      <th>Gene name<\/th>\n      <th>Chromosome<\/th>\n      <th>Average expression level<\/th>\n      <th>Male bias in expression (logFC)<\/th>\n      <th>Female early<\/th>\n      <th>Female late<\/th>\n      <th>Male early<\/th>\n      <th>Male late<\/th>\n      <th>sig_SA_early<\/th>\n      <th>sig_SA_late<\/th>\n      <th>sig_AA_females<\/th>\n      <th>sig_AA_males<\/th>\n      <th>sig_SC_early<\/th>\n      <th>sig_SC_late<\/th>\n      <th>sig_AC_females<\/th>\n      <th>sig_AC_males<\/th>\n      <th>P_equal_effects<\/th>\n      <th>P_female_specific<\/th>\n      <th>P_male_specific<\/th>\n      <th>P_null<\/th>\n      <th>P_sex_antag<\/th>\n      <th>Sex_antag_0.5<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"TWAS_sig_genes"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":[3,4,17,18,19,20,21,22]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="count-the-numbers-of-significant-genes" class="section level3">
<h3>Count the numbers of significant genes</h3>
<p>Abbreviations: AC = age concordant, SC = sexually concordant, SA = sexually antagonistic. We defined significantly antagonistic transcripts as those where the relationship with fitness is significantly positive for one sex (or age class) and significantly negative for the other, with LFSR &lt; 0.01. Similarly, we define significantly concordant transcripts as those where the relationship with fitness is significantly positive for one sex or age class and also significantly positive for the other (LFSR &lt; 0.01). This is quite conservative, because a transcript needs to have a LFSR &lt; 0.01 for two tests, giving two chances for a ‘false negative’.</p>
<pre class="r"><code>tallies %&gt;%
  mutate(x = str_remove_all(type, &quot;sig_&quot;),
         `Number statistically significant` = n_transcripts) %&gt;%
  mutate(x = str_replace_all(x, &quot;AC&quot;, &quot;Age concordant&quot;),
         x = str_replace_all(x, &quot;AA&quot;, &quot;Age antagonistic&quot;),
         x = str_replace_all(x, &quot;SC&quot;, &quot;Sexually concordant&quot;),
         x = str_replace_all(x, &quot;SA&quot;, &quot;Sexually antagonistic&quot;),
         x = str_replace_all(x, &quot;_females&quot;, &quot; (in females)&quot;),
         x = str_replace_all(x, &quot;_males&quot;, &quot; (in males)&quot;),
         x = str_replace_all(x, &quot;_early&quot;, &quot; (in early life)&quot;),
         x = str_replace_all(x, &quot;_late&quot;, &quot; (in late life)&quot;)) %&gt;%
  rename(`Type of transcript` = x) %&gt;%
  select(`Type of transcript`, `Number statistically significant`) %&gt;%
  kable_table()</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; ">
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Type of transcript
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Number statistically significant
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Sexually concordant (in late life)
</td>
<td style="text-align:right;">
42
</td>
</tr>
<tr>
<td style="text-align:left;">
Age concordant (in males)
</td>
<td style="text-align:right;">
37
</td>
</tr>
<tr>
<td style="text-align:left;">
Age concordant (in females)
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Sexually concordant (in early life)
</td>
<td style="text-align:right;">
17
</td>
</tr>
<tr>
<td style="text-align:left;">
Sexually antagonistic (in late life)
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Sexually antagonistic (in early life)
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age antagonistic (in females)
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Age antagonistic (in males)
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.3.4 DT_0.13          forcats_0.5.0    stringr_1.4.0   
 [5] dplyr_1.0.0      purrr_0.3.4      readr_2.0.0      tidyr_1.1.0     
 [9] tibble_3.0.1     ggplot2_3.3.2    tidyverse_1.3.0  workflowr_1.6.2 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      svglite_1.2.3     lubridate_1.7.10  lattice_0.20-41  
 [5] assertthat_0.2.1  rprojroot_1.3-2   digest_0.6.25     R6_2.4.1         
 [9] cellranger_1.1.0  backports_1.1.7   reprex_0.3.0      evaluate_0.14    
[13] highr_0.8         httr_1.4.1        pillar_1.4.4      gdtools_0.2.2    
[17] rlang_0.4.6       readxl_1.3.1      rstudioapi_0.11   whisker_0.4      
[21] blob_1.2.1        rmarkdown_2.5     webshot_0.5.2     htmlwidgets_1.5.1
[25] bit_1.1-15.2      munsell_0.5.0     broom_0.5.6       compiler_4.0.3   
[29] httpuv_1.5.3.1    modelr_0.1.8      xfun_0.22         systemfonts_0.2.2
[33] pkgconfig_2.0.3   htmltools_0.5.0   tidyselect_1.1.0  viridisLite_0.3.0
[37] fansi_0.4.1       crayon_1.3.4      tzdb_0.1.2        dbplyr_1.4.4     
[41] withr_2.2.0       later_1.0.0       grid_4.0.3        nlme_3.1-149     
[45] jsonlite_1.7.0    gtable_0.3.0      lifecycle_0.2.0   DBI_1.1.0        
[49] git2r_0.27.1      magrittr_2.0.1    scales_1.1.1      vroom_1.5.3      
[53] cli_2.0.2         stringi_1.5.3     fs_1.4.1          promises_1.1.0   
[57] xml2_1.3.2        ellipsis_0.3.1    generics_0.0.2    vctrs_0.3.0      
[61] tools_4.0.3       bit64_0.9-7       glue_1.4.2        crosstalk_1.1.0.1
[65] hms_0.5.3         parallel_4.0.3    yaml_2.2.1        colorspace_1.4-1 
[69] rvest_0.3.5       knitr_1.32        haven_2.3.1      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
