---
title: "Estimating the mean phenotype for each DGRP line"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r, include=FALSE}
library(knitrhooks) # for scrollable R output. remotes::install_github("nathaneastwood/knitrhooks")
output_max_height()
knitr::opts_chunk$set(output_max_height = "300px")
# setwd("~/Rprojects/fitnessGWAS/")
```

```{r message=FALSE, warning=FALSE, results="hide"}
library(tidyverse)
library(brms)
library(mice)
library(ICC)
library(future)
library(future.apply)
library(kableExtra)
```

## Estimate the line means for relative fitness
Most studies of the DGRP calculate "line means" simply by averaging the phenotypes of several individuals per line. Instead, we used Bayesian generalised linear mixed model (GLMMs) to estimate the true line means for each of the 4 fitness traits, implemented in the `brms` package.

One notable advantage of using a model instead of a simple average is that a model allows one to statistically remove confounding factors such as block effects, which would otherwise bias the line means. Note that we measured line 352 in every block in our experiments, in order to improve precision when estimating block effects. Additionally, we must properly account for our experimental design in order to avoid pseudoreplication. Our design involved repeatedly measuring the same individuals to determine early- and late-life fitness, which we incorporate into the model via the random effect `vial`.

For the progeny count data used to measure early and late female fitness, we used a multivariate model with two components, one for early- and one for late-life fitness, which had the following formulae (in `brms`-like pseudocode):

`Progeny count (early) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)`

`Progeny count (late) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)`

Here, the model terms with the form `(1 | x | y)` represent random intercepts for line, block, and vial ID (i.e. `y`), where the letter between the pipe operators (`x`) indicates which random intercepts are assumed to be correlated with which other random intercepts. Thus, we assume that the effects of line, block, and vial ID on early-life fitness are potentially correlated with the effects of these three random variables on late-life fitness, and estimate this correlation from the data and prior.

The progeny count data were assumed to follow a Poisson distribution (with the standard log link function).

Similarly, for our measure of male fitness, proportion of offspring sired, we used a multivariate model consisting of a pair of binomial GLMMs (with logit link), with the following formula:

`Proportion of progeny sired (early) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)`

`Proportion of progeny sired (late) = Intercept + (1 | p | line) + (1 | q | block) + (1 | r | vial)`

Where the response variable was a Bernoulli outcome describing the paternity of each offspring observed (either a male from the focal DGRP line, or from one of the competitor males). 

For both models, we mostly used the `brms` default (weak) priors, except that we chose more informative, 'skeptical' priors for the random intercept terms, of the form `cauchy(0, 0.1)`. This prior helps the model to converge by making very large estimates implausible, constraining the parameter space. 


### Define functions for prediction
```{r}
# function to load the raw data on male and female fitness, and tidy it for analysis
load_and_tidy_fitness_data <- function(){
  female_fitness <- read_csv("data/input/female_fitness.csv") %>% 
    arrange(line) %>%
    split(paste(.$block, .$line)) %>%
    map(~ .x %>% mutate(replicate = 1:n())) %>%
    bind_rows() %>%
    mutate(vial = 1:n(),
           vial = paste("F", line, block, replicate, sep = "_"),
           line = paste("line_", line, sep = "")) %>% 
    dplyr::select(-num.F.late1, -num.F.late2, -num.laying.F.early, -replicate) %>%
    mutate(female.fitness.early = replace(female.fitness.early, is.na(female.fitness.early), 0),
           female.fitness.late = replace(female.fitness.late, is.na(female.fitness.late), 0)) 
  
  male_fitness <- read_csv("data/input/male_fitness.csv") %>% 
    arrange(line) %>%
    split(paste(.$block, .$line)) %>%
    map(~ .x %>% mutate(replicate = 1:n())) %>%
    bind_rows() %>%
    mutate(vial = paste("M", line, block, replicate, sep = "_"),
           line = paste("line_", line, sep = "")) %>%
    dplyr::select(-num.DGRP.males, -replicate)
  
  # Save these tidy files, for data archiving
  write.csv(female_fitness, file = "data/input/female_fitness_CLEANED.csv", row.names = FALSE)
  write.csv(male_fitness, file = "data/input/male_fitness_CLEANED.csv", row.names = FALSE)
  
  list(female_fitness = female_fitness, male_fitness = male_fitness)
}


# This function calculates the traditional line means (i.e. simple averages) for each line
get_line_means <- function(){
  left_join(
    load_and_tidy_fitness_data()[[1]] %>% 
      select(line, female.fitness.early, female.fitness.late) %>% 
      group_by(line) %>%
      summarise(female.fitness.early = mean(female.fitness.early), 
                female.fitness.late = mean(female.fitness.late)),
    load_and_tidy_fitness_data()[[2]] %>% 
      group_by(line) %>%
      summarise(early.male.rival = sum(early.male.rival),
                early.male.focal = sum(early.male.focal),
                late.male.rival = sum(late.male.rival),
                late.male.focal = sum(late.male.focal),
                male.fitness.early = early.male.focal / (early.male.focal + early.male.rival),
                male.fitness.late = late.male.focal / (late.male.focal + late.male.rival)) %>%
      select(line, male.fitness.early, male.fitness.late),
    by = "line") %>% as.data.frame()
}

# A function to calculate each line's expected fitness, using brms models to adjust for block + vial random effects
# For the male late life assay, we also adjust for the number of competitor males that died 
# (this assumes that their deaths happened randomly with respect to which line we were measuring)
get_predicted_line_means <- function(overwrite = FALSE){
  
  # If it's already calculated, just load up the file. Otherwise, make the file
  out.file <- "data/derived/predicted_line_means.csv"
  if(file.exists(out.file) & !overwrite) return(read_csv(out.file))
  
  ####### Load the individual-level data for females
  female_fitness <- load_and_tidy_fitness_data()[[1]] 
  
  ####### Load the individual-level data for males, and impute missing values
  
  # For one vial, there is no early-life measurement because the females did not produce offspring. 
  # Therefore we impute that datapoint and make 5 imputed datasets using the package 'mice'
  # See this link for explanation: https://cran.r-project.org/web/packages/brms/vignettes/brms_missings.html
  male_fitness <- load_and_tidy_fitness_data()[[2]]
  male_fitness <- male_fitness %>% 
    mutate(early.male.rival = replace(early.male.rival, early.male.rival + early.male.focal == 0, NA),
           early.male.focal = replace(early.male.focal, is.na(early.male.rival), NA)) 
  male_fitness_imputed <- mice(male_fitness, m = 5, print = FALSE, seed = 1)
  male_fitness_imputed <- lapply(1:5, function(i) { # for each imputed dataset, 
    complete(male_fitness_imputed, i) %>% 
      mutate(total1 = early.male.rival + early.male.focal, # tally up the focal+rival offspring for binomial model
             total2 = late.male.rival + late.male.focal,
             num.GFP.males = as.numeric(scale(num.GFP.males))) %>% # scale this covariate
      # For late-life assays where the 5 focal males all died before reaching age 14, 
      # we need to assign them zero fitness. We did this by assuming that the rivals sired 100/100 offspring.
      # This scenario happened in 47 vials (out of 810)
      mutate(late.male.focal = replace(late.male.focal, total2 == 0, 100),
             total2 = replace(total2, total2 == 0, 100))
  })

  ####### Run model on the female data 
  
  # Define the priors for the models
  prior_females <- 
    c(set_prior("cauchy(0, 0.1)", class = "sd", resp = 'femalefitnessearly', group = "line"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'femalefitnessearly', group = "block"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'femalefitnessearly', group = "vial"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'femalefitnesslate', group = "line"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'femalefitnesslate', group = "block"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'femalefitnesslate', group = "vial"))
  
  prior_males <- 
    c(set_prior("cauchy(0, 0.1)", class = "sd", resp = 'earlymalefocal', group = "line"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'earlymalefocal', group = "block"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'earlymalefocal', group = "vial"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'latemalefocal', group = "line"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'latemalefocal', group = "block"),
      set_prior("cauchy(0, 0.1)", class = "sd", resp = 'latemalefocal', group = "vial"))
      
   
  female1 <- bf(female.fitness.early ~ (1 | p | line) + (1 | q | block) + (1 | r | vial))
  female2 <- bf(female.fitness.late ~  (1 | p | line) + (1 | q | block) + (1 | r | vial))
  female_model <- brm(female1 + female2, 
                      family = "poisson", data = female_fitness,
                      prior = prior_females,
                      iter = 5000, chains = 4, cores = 4,
                      seed = 1, control = list(adapt_delta = 0.999, max_treedepth = 15)) 
  writeLines(capture.output(summary(female_model)), con = "data/derived/model_summary_females.txt")
  
  # saveRDS(female_model, "data/derived/female_fitness_prediction_model.rds")
  
  
  ### Run model on male data 
  male1 <- bf(early.male.focal | trials(total1) ~ (1 | p | line) + (1 | q | block) + (1 | r | vial))
  male2 <- bf(late.male.focal  | trials(total2) ~ (1 | p | line) + (1 | q | block) + (1 | r | vial))
  
  # Run 5 models, on each of the 5 imputed datasets, and combine the results
  male_model <- brm_multiple(male1 + male2,
                             family = "binomial", 
                             prior = prior_males,
                             data = male_fitness_imputed,
                             iter = 5000, chains = 1, cores = 4,
                             seed = 1, control = list(adapt_delta = 0.999, max_treedepth = 15)) 
  writeLines(capture.output(summary(male_model)), con = "data/derived/model_summary_males.txt")
  # saveRDS(male_model, "data/derived/male_fitness_prediction_model.rds") 

  ####### Define new data for prediction
  new.data.female <- female_fitness %>% 
    select(line) %>% distinct() %>% arrange(line)
  
  new.data.male <- male_fitness %>% 
    select(line) %>% distinct() %>% arrange(line) %>% 
    mutate(total1 = 100, # changing this number does not affect the line means, as expected
           total2 = 100, 
           num.GFP.males = mean(male_fitness$num.GFP.males, na.rm = TRUE))
  
  # Get predicted lines means for early and late life *female* fitness on linear predictor scale (i.e. fairly normal)
  female_line_preds <- data.frame(
    new.data.female, 
    fitted(female_model, 
           newdata = new.data.female, 
           re_formula = "~ (1 | line)", 
           scale = "linear")) %>%
    select(line, Estimate.femalefitnessearly, Estimate.femalefitnesslate) %>% 
    rename(female.fitness.early = Estimate.femalefitnessearly,
           female.fitness.late = Estimate.femalefitnesslate) %>%
    mutate(female.fitness.early = as.numeric(scale(female.fitness.early)), # scale the line means
           female.fitness.late = as.numeric(scale(female.fitness.late)))
  
  # Get predicted lines means for early and late life *male* fitness on linear predictor scale (i.e. fairly normal)
  male_line_preds <- data.frame(
    new.data.male, 
    fitted(male_model, 
           newdata = new.data.male, 
           re_formula = "~ (1 | line)", 
           scale = "linear")) %>%
    select(line, Estimate.earlymalefocal, Estimate.latemalefocal) %>% 
    rename(male.fitness.early = Estimate.earlymalefocal,
           male.fitness.late = Estimate.latemalefocal) %>%
    mutate(male.fitness.early = as.numeric(scale(male.fitness.early)), # scale the line means
           male.fitness.late = as.numeric(scale(male.fitness.late))) 

  ## Define a function to get the ICC for line, for each posterior sample of predicted line means
  # get_ICC_line <- function(predictions){
  #   cols <- str_detect(names(predictions), "X")
  #   first_posterior_column <- which(cols)[1] 
  #   n_draws <- sum(cols)
  #   lapply(0:(n_draws - 1), function(i){
  #     ICCbare(predictions$line, predictions[, i + first_posterior_column])
  #   }) %>% unlist()
  # }
  
  # Calculate ICC for each sample, separately for old and young females
  # For females, this is done on the estimated line mean progeny produced (i.e. response scale)
  # predicted <- fitted(female_model, summary = FALSE)
  # Line_ICC_female_early <- get_ICC_line(data.frame(female_fitness, t(predicted[, , 1])))
  # Line_ICC_female_late <- get_ICC_line(data.frame(female_fitness, t(predicted[, , 2])))
  
  # Do the same for males
  # For males, this is done on the estimated proportion progeny sired (i.e. response scale)
  # new_data <- male_fitness %>% 
  #   select(block, line, vial) %>% 
  #   mutate(total1 = 100, 
  #          total2 = 100, 
  #          num.GFP.males = male_fitness %>% .$num.GFP.males %>% mean(na.rm=TRUE))
  # predicted <- fitted(male_model, summary = FALSE, newdata = new_data)
  # Line_ICC_male_early <- get_ICC_line(data.frame(new_data, t(predicted[, , 1])))
  # Line_ICC_male_late <- get_ICC_line(data.frame(new_data, t(predicted[, , 2])))
  # 
  # ICC <- list(Line_ICC_female_early, 
  #             Line_ICC_female_late,
  #             Line_ICC_male_early,
  #             Line_ICC_male_late) %>% lapply(posterior_summary) %>% 
  #   do.call("rbind", .) %>% as_tibble() %>%
  #   mutate(Fitness_trait = c("Female early-life fitness", "Female late-life fitness",
  #                            "Male early-life fitness", "Male late-life fitness"))
  # write.csv(ICC, file = "data/derived/ICC.csv", row.names = TRUE)
  

  # Record which block(s) each line was measured in. Generally one block, except for ref line (all blocks)
  line_blocks <- rbind(female_fitness %>% select(block, line), 
                       male_fitness   %>% select(block, line)) %>% 
    mutate(block = replace(block, line == "352", "reference_line")) %>% 
    distinct() %>% 
    group_by(line) %>% summarise(block = paste0(block, collapse = ", ")) 
  
  predicted_line_means <- female_line_preds %>%
    left_join(male_line_preds, by = "line") %>% 
    left_join(line_blocks, by = "line") %>%
    arrange(line) 
  
  write.csv(predicted_line_means, out.file, row.names = FALSE)
}
```

### Predict line means and save the results
The object `line_means` is the raw line means: unscaled, arithmetic means and in the original units.

The object `predicted_line_means` is the estimated line means from the Bayesian models. These means are on the scale of the linear predictor (i.e. Poisson for females, binomial for males) rather than on the scale of the original response. They have been scaled to have a mean of 0 and a variance of 1. 

```{r message=FALSE, warning=FALSE}
line_means <- get_line_means()
predicted_line_means <- get_predicted_line_means(overwrite = FALSE) 
```


### Summary of model used to predict **female** line means:
```{r comment=''}
cat(readLines('data/derived/model_summary_females.txt'), sep = '\n')
```

### Summary of model used to predict **male** line means:
```{r comment=''}
cat(readLines('data/derived/model_summary_males.txt'), sep = '\n')
```






<!-- ## Estimate genetic (co)variance from the genomic relatedness matrix -->

<!-- The following fits a multivariate mixed model using `brms`, to estimate the amount of variance explained by the DGRP line, accounting for the genomic relatedness matrix. The response variables are the predicted line means for the four phenotypes, as estimated in the previous section. We fit line as a random effect, specifying a covariance structure among the line effects that is equal to the genomic relatedness matrix (GRM), which we estimate using the SNP genotypes of the full sample of 205 DGRP lines. We use a somewhat informative prior to help the model converge; the prior is that the line and residual variances are not large (which must be true, given that the response variables have overall variance of 1), and that only 10% of the phenotype variance is explained by line (the posterior estimates suggest much higher line variance, indicating strong signal in the data). -->

<!-- The table below indicates that much of the variance in fitness is due to additive genetic effects, and that there is a positive geneti -->

<!-- ```{r eval=FALSE} -->
<!-- # Run quant gen model using brms -->
<!-- if(!file.exists("data/derived/brms_quant_gen.rds")){ -->

<!--   library(bigsnpr) -->
<!--   library(MCMCglmm) -->
<!--   library(sommer) -->

<!--   # Get the data -->
<!--   mcmc_dat <- predicted_line_means %>%  -->
<!--     mutate(line = factor(str_remove_all(line, "_"))) %>% -->
<!--     filter(!is.na(male.fitness.early)) %>% -->
<!--     rename(femalefitnessearly = female.fitness.early, -->
<!--            femalefitnesslate = female.fitness.late, -->
<!--            malefitnessearly = male.fitness.early, -->
<!--            malefitnesslate = male.fitness.late) -->

<!--   # Make genomic relatedness matrix -->
<!--   bigsnp <- snp_readBed("data/derived/dgrp2_QC_all_lines.bed", backingfile = tempfile()) -->
<!--   bigsnp_attached <- snp_attach(bigsnp) -->
<!--   G <- A.mat(bigsnp_attached$genotypes[1:nrow(bigsnp_attached$genotypes),  -->
<!--                                        1:ncol(bigsnp_attached$genotypes)] - 1)  -->
<!--   colnames(G) <- bigsnp_attached$fam[,1] -->
<!--   rownames(G) <- bigsnp_attached$fam[,1] -->
<!--   focal_lines <- mcmc_dat %>% pull(line) -->
<!--   G <- G[rownames(G) %in% focal_lines,  -->
<!--          colnames(G) %in% focal_lines] -->
<!--   G <- cov2cor(G) -->

<!--   # Female early - male early model -->
<!--   brms_quant_gen_FE_ME <- brm( -->
<!--     bf(femalefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(malefitnessearly ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(TRUE), -->
<!--     data = mcmc_dat, data2 = list(G = G),  -->
<!--     iter = 30000, chains = 4, cores = 1, warmup = 20000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->

<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'malefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnessearly'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'malefitnessearly'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->

<!--   saveRDS(brms_quant_gen_FE_ME, "data/derived/brms_quant_gen_FE_ME.rds") -->

<!--   # Female late - male late model -->
<!--   brms_quant_gen_FL_ML <- brm( -->
<!--     bf(femalefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(malefitnesslate ~ 0 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(TRUE), -->
<!--     data = mcmc_dat, data2 = list(G = G),  -->
<!--     iter = 30000, chains = 4, cores = 1, warmup = 20000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->

<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'malefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnesslate'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'malefitnesslate'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->

<!--   saveRDS(brms_quant_gen_FL_ML, "data/derived/brms_quant_gen_FL_ML.rds") -->



<!--   # Female early - female late model -->
<!--   brms_quant_gen_FE_FL <- brm( # failed badly, try with rescor and intercepts too? -->
<!--     bf(femalefitnessearly ~ 1 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(femalefitnesslate ~ 1 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(TRUE), -->
<!--     data = mcmc_dat, data2 = list(G = G),  -->
<!--     iter = 30000, chains = 4, cores = 1, warmup = 20000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->

<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'femalefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnessearly'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'femalefitnesslate'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->

<!--   saveRDS(brms_quant_gen_FE_FL, "data/derived/brms_quant_gen_FE_FL.rds") -->

<!--   # Male early - male late model -->
<!--   brms_quant_gen_ME_ML <- brm( -->
<!--     bf(malefitnessearly ~ 1 + (1 | p | gr(line, cov = G))) + -->
<!--       bf(malefitnesslate ~ 1 + (1 | p | gr(line, cov = G))) + -->
<!--       set_rescor(TRUE), -->
<!--     data = mcmc_dat, data2 = list(G = G),  -->
<!--     iter = 30000, chains = 4, cores = 1, warmup = 20000, -->
<!--     control = list(adapt_delta = 0.9999999999, max_treedepth = 20), -->

<!--     prior = c( -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'malefitnessearly', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sd", resp = 'malefitnesslate', group = "line"), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'malefitnessearly'), -->
<!--       prior(normal(0.5, 0.15), "sigma", resp = 'malefitnesslate'), -->
<!--       prior(lkj(0.5), class = "cor", group = "line")) -->
<!--     ) -->

<!--   saveRDS(brms_quant_gen_ME_ML, "data/derived/brms_quant_gen_ME_ML.rds") -->


<!-- } else { -->
<!--   brms_quant_gen_FE_ME <- readRDS("data/derived/brms_quant_gen_FE_ME.rds") -->
<!--   brms_quant_gen_FL_ML <- readRDS("data/derived/brms_quant_gen_FL_ML.rds") -->
<!--   brms_quant_gen_FE_FL <- readRDS("data/derived/brms_quant_gen_FE_FL.rds") -->
<!--   brms_quant_gen_ME_ML <- readRDS("data/derived/brms_quant_gen_ME_ML.rds") -->
<!-- } -->

<!-- p <- posterior_samples(brms_quant_gen_FE_ME) -->
<!-- p2 <- posterior_samples(brms_quant_gen_FE_FL) -->
<!-- p <- tibble( -->
<!--   VA_FE = (p[, "sd_line__femalefitnessearly_Intercept"])^2, -->
<!--   VA_FL = (p[, "sd_line__femalefitnesslate_Intercept"])^2, -->
<!--   VA_ME = (p[, "sd_line__malefitnessearly_Intercept"])^2, -->
<!--   VA_ML = (p[, "sd_line__malefitnesslate_Intercept"])^2, -->
<!--   h2_FE = VA_FE / (VA_FE + (p[, "sigma_femalefitnessearly"])^2), -->
<!--   h2_FL = VA_FL / (VA_FL + (p[, "sigma_femalefitnesslate"])^2), -->
<!--   h2_ME = VA_ME / (VA_ME + (p[, "sigma_malefitnessearly"])^2), -->
<!--   h2_ML = VA_ML / (VA_ML + (p[, "sigma_malefitnesslate"])^2), -->
<!--   corr_FE_FL = p[,"cor_line__femalefitnessearly_Intercept__femalefitnesslate_Intercept"], -->
<!--   corr_FE_ME = p[,"cor_line__femalefitnessearly_Intercept__malefitnessearly_Intercept"], -->
<!--   corr_FL_ME = p[,"cor_line__femalefitnesslate_Intercept__malefitnessearly_Intercept"], -->
<!--   corr_FE_ML = p[,"cor_line__femalefitnessearly_Intercept__malefitnesslate_Intercept"], -->
<!--   corr_FL_ML = p[,"cor_line__femalefitnesslate_Intercept__malefitnesslate_Intercept"], -->
<!--   corr_ME_ML = p[,"cor_line__malefitnessearly_Intercept__malefitnesslate_Intercept"], -->
<!--   cov_FE_FL = corr_FE_FL * VA_FE^2 * VA_FL^2, -->
<!--   cov_FE_ME = corr_FE_ME * VA_FE^2 * VA_ME^2, -->
<!--   cov_FL_ME = corr_FL_ME * VA_FL^2 * VA_ME^2, -->
<!--   cov_FE_ML = corr_FE_ML * VA_FE^2 * VA_ML^2, -->
<!--   cov_FL_ML = corr_FL_ML * VA_FL^2 * VA_ML^2, -->
<!--   cov_ME_ML = corr_ME_ML * VA_ME^2 * VA_ML^2, -->
<!--   corr_FE_ME_minus_corr_FL_ML = corr_FE_ME - corr_FL_ML, -->
<!--   corr_FE_FL_minus_corr_ME_ML = corr_FE_FL - corr_ME_ML -->
<!-- ) %>% gather(Parameter, value) -->

<!-- p %>% -->
<!--   group_by(Parameter) %>% -->
<!--   summarise(summ = list(posterior_summary(value)), -->
<!--             Estimate = map_dbl(summ, ~ .x[[1]]), -->
<!--             `Est. Error` = map_dbl(summ, ~ .x[[2]]), -->
<!--             `Lower 95% CI` = map_dbl(summ, ~ .x[[3]]), -->
<!--             `Upper 95% CI` = map_dbl(summ, ~ .x[[4]]), -->
<!--             .groups = "drop") %>% -->
<!--   dplyr::select(-summ) %>% -->
<!--   mutate(sp = str_split(Parameter, "_"), -->
<!--          Type = factor(map_chr(sp, ~ .x[[1]]), c("h2", "corr", "VA", "cov")), -->
<!--          Parameter = map_chr(sp, ~ paste0(.x[-1], collapse = " - "))) %>% -->
<!--   arrange(Type) %>% -->
<!--   dplyr::select(-sp, -Type) %>% -->
<!--   kable(digits = 3) %>% -->
<!--   kable_styling(full_width = FALSE) %>% -->
<!--   pack_rows("Heritabilities", 1, 4) %>% -->
<!--   pack_rows("Genetic correlations", 5, 10) %>% -->
<!--   pack_rows("Additive genetic variances", 11, 14) %>% -->
<!--   pack_rows("Additive genetic covariances", 15, 20) -->
<!-- ``` -->
