---
title: "Tables summarising the GWAS results"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

## Load and clean the GWAS results
```{r libraries, message=FALSE, warning=FALSE, results='hide'}
library(tidyverse)
library(kableExtra)
library(DT)

kable_table <- function(df) {
  kable(df, "html") %>%
  kable_styling() %>%
  scroll_box(height = "300px")
}

my_data_table <- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c("Scroller",  "Buttons"),
    options = list(
      dom = 'Bfrtip',
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list('pageLength', 'colvis', 'csv', list(
          extend = 'pdf',
          pageSize = 'A4',
          orientation = 'landscape',
          filename = 'GWAS_sig_loci')),
      pageLength = 50
    )
  )
}

db <- DBI::dbConnect(RSQLite::SQLite(), "data/derived/annotations.sqlite3")


# Load the SNP-clumped GWAS results, created with GEMMA and clumped with PLINK
xx <- bind_rows(
  read_tsv("data/derived/female_early_SNP_clumps.txt"),
  read_tsv("data/derived/female_late_SNP_clumps.txt"),
  read_tsv("data/derived/male_early_SNP_clumps.txt"),
  read_tsv("data/derived/male_late_SNP_clumps.txt")) %>%
  distinct(index_SNP, .keep_all = TRUE) %>%
  mutate(FBIDs = str_remove_all(FBIDs, ", NA"),
         FBIDs = replace(FBIDs, is.na(FBIDs), "")) %>%
  mutate(genes = str_remove_all(genes, ", NA"),
         genes = replace(genes, is.na(genes), "")) %>%
  select(index_SNP, other_snps, FBIDs, genes) 

# Because we clumped separately for the 4 traits, some of the "index SNPs" appear inside the ranges
# tagged by other index SNPs, so there is some redundancy. The following ugly code sorts this out 
# by putting index SNPs that appear inside the range of another index SNP into the 'other SNPs' 
# column for that index SNP
ranges <- xx$other_snps %>% 
  str_extract_all("[_][:digit:]+") %>% 
  map(~ as.numeric(str_remove_all(.x, "_"))) %>% 
    map(~ c(min(.x), max(.x)))

ranges <- map_df(1:nrow(xx), 
                 ~ data.frame(index = xx$index_SNP[.x], 
                              min =ranges[[.x]][1], 
                              max = ranges[[.x]][2]))%>%
  filter(!is.na(min)) %>% 
  mutate(chr = substr(index, 1,2))

to_modify <- lapply(1:nrow(xx), function(i){
  chr <- substr(xx$index_SNP[i],1,2) # get chr and pos
  pos <- as.numeric(str_remove_all(str_extract_all(xx$index_SNP[i], "_[:digit:]+"), "_"))
  foc_ranges <- ranges[ranges$chr == chr & 
                         ranges$min <= pos & 
                         ranges$max >= pos &
                         ranges$index != xx$index_SNP[i], ]
  if(nrow(foc_ranges) > 0) {
    return(data.frame(index_SNP = foc_ranges$index, 
                      to_add = xx$index_SNP[i]))
  }
  return(NULL)
}) %>% bind_rows()

xx <- xx %>% 
  filter(!(index_SNP %in% to_modify$to_add)) %>% 
  left_join(to_modify) %>% 
  mutate(to_add = replace(to_add, is.na(to_add), "NA"))

xx <- xx %>% 
  mutate(other_snps = str_split(str_c(xx$other_snps, to_add, sep = ", "), ", "),
              other_snps = map_chr(other_snps, 
                                   ~ paste0(.x[.x != "NA"], collapse = ", "))) %>%
  select(-to_add) %>%
  mutate(other_snps = str_replace_all(other_snps, "NA", ""))

# Now get the RAW effect sizes for the index SNPs, 
# and their mixture assignment probs.
univariate_lmm_results <- 
  tbl(db, "univariate_lmm_results") %>%
  filter(SNP %in% !! xx$index_SNP) %>%
  select(-contains("canonical"), 
         -contains("ED"), -contains("P_"), -contains("SE")) %>% 
  inner_join(tbl(db, "variants"), 
             by = "SNP") %>%
  left_join(
    tbl(db, "genes") %>% 
      select(FBID, gene_name), by = "FBID") %>%
  collect(n=Inf) %>%
  mutate(beta_female_early_raw = round(beta_female_early_raw, 2),
         beta_female_late_raw = round(beta_female_late_raw, 2),
         beta_male_early_raw = round(beta_male_early_raw, 2),
         beta_male_late_raw = round(beta_male_late_raw, 2)) %>% 
  rename_all(~ gsub("beta_", "", .x)) %>%
  rename_all(~ gsub("_raw", "", .x)) %>%
  select(SNP, MAF, site.class, starts_with("female"), 
          starts_with("male"), starts_with("P_"), starts_with("pvalue")) %>%
  distinct()

GWAS_table <- xx %>%
  left_join(univariate_lmm_results, by = c("index_SNP" = "SNP")) %>%
  filter(!is.na(female_early)) %>% 
  arrange(pvalue_female_early + pvalue_female_late + pvalue_male_early + pvalue_male_late) %>% 
  rename(`Index variant` = index_SNP,
         `MAF of index` = MAF,
         `Site class of index` = site.class,
         `Other variants` = other_snps,
         `Genes` = genes, 
         `Female early effect` = female_early,
         `Female late effect` = female_late,
         `Male early effect` = male_early,
         `Male late effect` = male_late,
         `Female early pval` = pvalue_female_early,
         `Female late pval` = pvalue_female_late,
         `Male early pval` = pvalue_male_early,
         `Male late pval` = pvalue_male_late) 

write_csv(GWAS_table, "data/derived/GWAS_significant_snps.csv")
```

## Table of significant groups of linked variants

This table shows loci (i.e. groups of linked variants) that passed the arbitrary statistical significance threshold of $p < 10^{-5}$ for one or more of the four phenotypes (in a linear mixed model GWAS implemented in GEMMA). The significant loci (and groups of loci surrounding the significant one) were identified using the SNP clumping method of PLINK; each group of loci contains at least one variant that met the significance threshold, plus other loci that are within 250kb of the most significant locus, plus in linkage disequilibrium with it with r-squared > 0.5 (see the [PLINK documentation](http://zzz.bwh.harvard.edu/plink/clump.shtml)). The third and fourth columns identify the gene(s) whose exons and introns overlap the focal locus. Columns 6-9 show the estimated effect size of the index variant on the four phenotypes (corrected by `mashr` using the data-driven method), while the final columns show the mixture proportions estimated by `mashr` using the canonical method.  

```{r}
GWAS_table %>%
  my_data_table()
```


