---
title: "Tables summarising the GWAS results"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r libraries, message=FALSE, warning=FALSE, results='hide'}
library(tidyverse)
library(kableExtra)
library(DT)

sci_notation <- function(x) formatC(x, format = "e", digits = 1)

kable_table <- function(df) {
  kable(df, "html") %>%
  kable_styling() %>%
  scroll_box(height = "300px")
}

my_data_table <- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c("Scroller",  "Buttons"),
    options = list(
      dom = 'Bfrtip',
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list('pageLength', 'colvis', 'csv', list(
          extend = 'pdf',
          pageSize = 'A4',
          orientation = 'landscape',
          filename = 'GWAS_sig_loci')),
      pageLength = 50
    )
  ) %>%
    formatStyle(
        columns = c("Female early effect", "Male early effect", "Female late effect", "Male late effect"), 
        color = styleInterval(cuts = 0, values = c("tomato", "steelblue")),
        fontWeight = "bold")
}

db <- DBI::dbConnect(RSQLite::SQLite(), "data/derived/annotations.sqlite3")

p_cutoff <- 1e-05

univariate_lmm_results <- 
  tbl(db, "univariate_lmm_results") %>%
  select(SNP, SNP_clump, contains("raw"), -contains("SE")) %>% 
  filter(pvalue_female_early_raw < p_cutoff | pvalue_female_late_raw < p_cutoff | 
           pvalue_male_early_raw < p_cutoff | pvalue_male_late_raw < p_cutoff) %>% 
  inner_join(tbl(db, "variants"), by = "SNP") %>%
  left_join(
    tbl(db, "genes") %>% 
      select(FBID, gene_name), by = "FBID") %>%
  collect(n=Inf) %>%
  mutate_at(vars(contains("beta")), ~ format(round(.x, 2), nsmall = 2)) %>% 
  rename_all(~ gsub("beta_", "", .x)) %>%
  rename_all(~ gsub("_raw", "", .x)) %>%
  select(SNP, SNP_clump, MAF, FBID, gene_name, site.class, starts_with("female"), 
          starts_with("male"), starts_with("P_"), starts_with("pvalue")) %>%
  distinct()

GWAS_table <- univariate_lmm_results %>% 
  split(.$SNP_clump) %>% # Only keep one SNP from each clump
  map_df(~ mutate(.x, sum_p = pvalue_female_early + pvalue_female_late + pvalue_male_early + pvalue_male_late) %>% 
        arrange(sum_p) %>% head(1)) %>% 
  select(-sum_p) %>% 
  mutate_at(vars(contains("pval")), sci_notation) %>% 
  mutate(SNP_clump = replace(SNP_clump, SNP == SNP_clump, "")) %>% 
  rename(`Variant` = SNP,
         `Variant clump` = SNP_clump,
         `Site class` = site.class,
         `Gene` = gene_name, 
         `Female early effect` = female_early,
         `Female late effect` = female_late,
         `Male early effect` = male_early,
         `Male late effect` = male_late,
         `Female early pval` = pvalue_female_early,
         `Female late pval` = pvalue_female_late,
         `Male early pval` = pvalue_male_early,
         `Male late pval` = pvalue_male_late) 

write_csv(GWAS_table, "data/derived/GWAS_significant_snps.csv")
```

## Table showing significant loci from the GWAS

This table shows variants (either individual SNPs or indels, or clumps of variants in complete linkage disequilibrium) that passed the statistical significance threshold of $p < 10^{-5}$ for at least one of the four phenotypes, in a linear mixed model GWAS implemented in GEMMA. Column 3 gives the minor allele frequency, columns 4-5 identify genes that overlap the variant, and column 6 shows the site class. Columns 7-10 show the estimated effect size of the variant on the four phenotypes (where positive values mean that the minor allele is associated with higher fitness, and negative values that the minor allele is associated with lower fitness), and the final columns show the corresponding $p$-values.

```{r}
GWAS_table %>% my_data_table()
```

### Table showing the numbers of loci related to fitness
The table tallies the numbers of loci showing a particular significant relationship with fitness (rows) for various different $p$-value thresholds (columns). For example, 'Female early only' counts the number of loci whose genotype significantly correlated with mean female early life fitness across lines. 'Age concordant, males' counts loci whose genotype correlated with early- _and_ late-life fitness in males, in the same direction, while 'Age antagonistic, males' counts loci showing significant, opposite relationships with male early- and male late-life fitness. Similarly, 'Sex concordant, early' counts loci showing a concordant relationship with early life fitness in both males and females, and 'Sex antagonistic, late' counts those showing opposing relationships with late-life fitness in males and females. All categories are mutually exclusive, such that loci are only counted towards the most specific category that applies to them. Note that this method has low power to detect loci that correlate with two or more fitness metrics, because there are two or more opportunities to make a 'false negative' error, and the power is low for any given locus (we therefore use a range of $p$-value thresholds, including some permissive ones, to illustrate general patterns of genetic covariance).

```{r make_tally_table}
make_tally <- function(p_cutoff){
  GWAS_results <- tbl(db, "univariate_lmm_results") %>%
    select(SNP_clump, contains("raw"), -contains("SE")) %>% 
    mutate(sum_p = pvalue_female_early_raw + pvalue_female_late_raw + pvalue_male_early_raw + pvalue_male_late_raw) %>% 
    arrange(sum_p) %>% 
    rename(beta_FE = beta_female_early_raw, 
           beta_FL = beta_female_late_raw,
           beta_ME = beta_male_early_raw, 
           beta_ML = beta_male_late_raw,
           pval_FE = pvalue_female_early_raw, 
           pval_FL = pvalue_female_late_raw,
           pval_ME = pvalue_male_early_raw, 
           pval_ML = pvalue_male_late_raw) %>% 
    distinct() %>% # Analyse clumps of SNPs in <100% LD with one another
    collect(n = Inf) %>% 
    select(-sum_p) %>% 
    mutate(
      type = 
        factor(case_when(
          
          (sign(beta_FE) != sign(beta_ME) & pval_FE < p_cutoff & pval_ME < p_cutoff) & 
            (sign(beta_FL) != sign(beta_ML) & pval_FL < p_cutoff & pval_ML < p_cutoff) ~ "Sex antagonistic, both ages",
          
          (sign(beta_FE) == sign(beta_ME) & pval_FE < p_cutoff & pval_ME < p_cutoff) & 
            (sign(beta_FL) == sign(beta_ML) & pval_FL < p_cutoff & pval_ML < p_cutoff) ~ "Sex concordant, both ages",
          
          (sign(beta_FE) != sign(beta_FL) & pval_FE < p_cutoff & pval_FL < p_cutoff) & 
            (sign(beta_ME) != sign(beta_ML) & pval_ME < p_cutoff & pval_ML < p_cutoff) ~ "Age antagonistic, both sexes",
          
          (sign(beta_FE) == sign(beta_FL) & pval_FE < p_cutoff & pval_FL < p_cutoff) &
            (sign(beta_ME) == sign(beta_ML) & pval_ME < p_cutoff & pval_ML < p_cutoff) ~ "Age concordant, both sexes",
          
          sign(beta_FE) != sign(beta_ME) & pval_FE < p_cutoff & pval_ME < p_cutoff ~ "Sex antagonistic, early",
          sign(beta_FL) != sign(beta_ML) & pval_FL < p_cutoff & pval_ML < p_cutoff ~ "Sex antagonistic, late",
          
          sign(beta_FE) == sign(beta_ME) & pval_FE < p_cutoff & pval_ME < p_cutoff ~ "Sex concordant, early",
          sign(beta_FL) == sign(beta_ML) & pval_FL < p_cutoff & pval_ML < p_cutoff ~ "Sex concordant, late",
          
          sign(beta_FE) != sign(beta_FL) & pval_FE < p_cutoff & pval_FL < p_cutoff ~ "Age antagonistic, females",
          sign(beta_ME) != sign(beta_ML) & pval_ME < p_cutoff & pval_ML < p_cutoff ~ "Age antagonistic, males",
          
          sign(beta_FE) == sign(beta_FL) & pval_FE < p_cutoff & pval_FL < p_cutoff ~ "Age concordant, females",
          sign(beta_ME) == sign(beta_ML) & pval_ME < p_cutoff & pval_ML < p_cutoff ~ "Age concordant, males",
          
          pval_FE < p_cutoff ~ "Female early only",
          pval_FL < p_cutoff ~ "Female late only",
          pval_ME < p_cutoff ~ "Male early only",
          pval_ML < p_cutoff ~ "Male late only",
          TRUE ~ "Uncorrelated with fitness"
        ), levels = c("Sex antagonistic, both ages", "Sex concordant, both ages", 
                      "Age antagonistic, both sexes", "Age concordant, both sexes",
                      "Sex antagonistic, early", "Sex concordant, early", "Sex antagonistic, late", "Sex concordant, late",
                      "Age antagonistic, females", "Age antagonistic, males", "Age concordant, females", "Age concordant, males",
                      "Female early only", "Male early only", "Female late only", "Male late only", "Uncorrelated with fitness")))
  
  tally <- table(GWAS_results$type) %>% sort(decreasing = T) %>% enframe()
  names(tally)[2] <- paste("p <", p_cutoff)
  tally
}

GWAS_tally_table <- make_tally(0.01) %>% 
  left_join(make_tally(0.001), by = "name") %>% 
  left_join(make_tally(0.0001), by = "name") %>% 
  left_join(make_tally(0.00001), by = "name") %>% 
  left_join(make_tally(0.000001), by = "name") %>% 
  left_join(make_tally(0.0000001), by = "name") %>% 
  rename(`Relationship to fitness` = name)

saveRDS(GWAS_tally_table, "data/derived/GWAS_tally_table.rds")

GWAS_tally_table %>% kable_table()
```


