---
title: "The genetic basis of fitness in _Drosophila_; A genome-wide association study"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

This website relates to the forthcoming paper 'Pleiotropic fitness effects across sexes and age classes in the _Drosophila_ genome and transcriptome'. 

<!-- by Wong and Holman (for contact info, see [lukeholman.org](https://www.lukeholman.org/)). -->

Click the headings below to see code, results, plots, tables and figures from this study. 

### [1. Setting up a database to hold variant/gene annotations and GWAS results](make_annotation_database.html)

This script creates a SQLite3 database holding two tables: one with annotations for each SNP/indel variant (annotations created by the Mackay lab), and one with annotations for each gene (from annotation hub). In step 4 below, we also add the GWAS results to this database, allowing memory-efficient handling of the results. 

### [2. Estimating line mean fitness using Bayesian models](get_predicted_line_means.html)

This script uses Bayesian mixed models implemented in the package `brms` to estimate the line means for our four fitness traits, while imputing missing values and adjusting for block effects.

### [3. Tables of the raw data and estimated line means](raw_data_and_line_means.html)

To facilitate data re-use, we here provide tables showing the raw data (i.e. the measurements of male and female fitness that were collected on each individual replicate vial), as well as the estimated line means that were calculated in [2. Estimating line mean fitness using Bayesian models](get_predicted_line_means.html).

### [4. Calculating quantitative genetic parameters](quant_genetics.html)

We first present a table showing the proportion of variance in fitness explained by 'DGRP line', which approximates heritability. We then estimate the correlations among line means in the 4 fitness traits, which approximates genetic correlations.

### [5. Running the GWAS](perform_gwas.html)

This script first performs quality control and imputation on the dataset of SNPs and indels for the DGRP (e.g filtering by MAF). Second, it runs mixed model association tests on our four phenotypes using the software [GEMMA](http://www.xzlab.org/software/GEMMAmanual.pdf). Third, it groups SNPs/indels that are in complete linkage disequilibrium in our sample of DGRP lines. Fourth, it uses PLINK to identify a subset of SNPs that are in approximate LD for downstream analyses.

### [6. Applying adaptive shrinkage to the GWAS results](gwas_adaptive_shrinkage.html)

This script uses the R package `mashr` to perform multivariate adaptive shrinkage on the results of the GWAS, for an LD-pruned subset of loci. This produces corrected estimates of each SNP's effect size, and allows estimation of the frequencies of different types of loci (e.g. sexually- or age-antagonistic loci).

### [7. Running the TWAS and applying mashr to the TWAS results](TWAS.html)

This script uses the transcriptomic data on the DGRP from Huang et al. 2015 _PNAS_ to run a 'transcriptome-wide association study' (TWAS). In this script, we

- Calculate the average expression and sex bias in expression for each transcript in Huang et al.
- Perform 'TWAS', i.e. searching for transcripts whose expression correlates with our fitness measures across DGRP lines
- Adjust the results of the TWAS linear models using `mashr`.


## Plots, tables, and analyses of the results

### [8. Plots showing line mean fitness](plot_line_means.html)

This script plots the estimated line means for each of the four fitness metrics, i.e. Figure 1 in the paper.

### [9. Tables of GWAS results](GWAS_tables.html)

This script presents a searchable HTML table showing a list of significant SNPs and indels from the GWAS, with annotations, effect sizes, and $p$-values for each. 

### [10. Tables of TWAS results](TWAS_tables.html)

This script presents a searchable HTML table showing a list of significant transcripts from the TWAS, with annotations, effect sizes, and $p$-values for each. 

### [11. Plots and statistical analyses](plot_models_variant_effects.html)

Here, we present various plots and statistical analyses of the GWAS and TWAS results, specifically:

- Hex bin plots showing the correlations in effect sizes from GWAS across the 4 fitness traits
- A statistical test showing that minor alleles tend to be associated with lower fitness
- A plot inspired by Boyle et al. 2017 ("An expanded view of complex traits: from polygenic to omnigenic", _Cell_) illustrating that fitness is highly polygenic ('omnigenic') and pleiotropy between male and female fitness is common. 
- An analysis of mutation load, testing whether the number of putatively harmful alleles present in each DGRP line correlates with fitness.
- Bar plots showing an estimate of the frequencies of sexually antagonistic and sexually concordant SNPs/indels from GWAS, and transcripts from TWAS (plus the same for age concordant vs age antagonistic variants/transcripts)
- Statistics indicating that the frequency of sexually antagonistic loci declines with age
- Plots of the evidence ratio for each locus and transcript, where the ER compares evidence for the hypotheses that 1) the locus is concordant between sexes or ages, and B) the locus is antagonistic between sexes or ages. This plot avoids having to create a binary distinction between antagonistic and concordant loci, when in reality there is a continuum of evidence.
- Statistical models of the evidence ratios, showing _inter alia_ that candidate sexually antagonistic alleles tend to be more common and to be enriched on the X chromosome.
- GO enrichment of the top 1% most sexually antagonistic transcripts, implicating GO terms relating to mitochondria among other things.  
