---
title: "Plots and models of variant effect sizes"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r include=FALSE}
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
```


```{r message=FALSE, warning=FALSE, results="hide"}
library(tidyverse)
library(gridExtra)
library(qqman)
library(ggbeeswarm)
library(Hmisc)
library(showtext) # For fancy Google font in figures
library(mashr)
library(kableExtra)
library(cowplot)
library(grid)
library(RColorBrewer)
font_add_google(name = "Raleway", family = "Raleway", regular.wt = 400, bold.wt = 700) # Install font from Google Fonts
showtext_auto()

db <- DBI::dbConnect(RSQLite::SQLite(), 
                     "data/derived/annotations.sqlite3")

# Results for all 1,613,615 SNPs, even those that are in 100% LD with others (these are grouped up by the SNP_clump column)
all_snps <- tbl(db, "univariate_lmm_results")

# All SNPs and SNP groups that are in <100% LD with one another (n = 1,207,357)
SNP_clumps <- all_snps %>% select(-SNP) %>% distinct() %>% collect(n = Inf)

# Subsetting variable to get the approx-LD subset of SNPs
LD_subset <- !is.na(SNP_clumps$LFSR_female_early_mashr_ED)
```

## Load and clean the variant effect sizes
Modify column names, reorder columns, filter based on the p-value, etc.
```{r load_data, message=FALSE}
# Univariate analysis, mashr-adjusted, data-driven approach
univariate_lmm_results <- tbl(db, "univariate_lmm_results") %>% 
  select(-contains("canonical"), 
         -contains("raw")) %>% 
  inner_join(tbl(db, "variants") %>% 
               select(SNP, FBID, site.class, distance.to.gene, MAF), 
             by = "SNP") %>%
  left_join(
    tbl(db, "genes") %>% 
      select(FBID, gene_name), by = "FBID") %>%
  collect(n = Inf) %>%
  rename_all(~ gsub("beta_", "", .x)) %>%
  rename_all(~ gsub("_mashr_ED", "", .x))

univariate_lmm_results <- univariate_lmm_results %>%
  mutate(site.class = gsub("5_", "5-", site.class),
         site.class = gsub("3_", "3-", site.class),
         site.class = gsub("NON_", "NON-", site.class),
         site.class = gsub("_", " ", site.class),
         site.class = capitalize(tolower(site.class)),
         site.class = gsub("Utr", "UTR", site.class)
  )
```


## Q-Q plots {.tabset}

Here are some quantile-quantile plots, which are commonly used to check GWAS results, and to test the hypothesis that there are more SNPs than expected showing large effects on the trait of interest. There is an excess of loci with effects on female fitness, but not much of a visible excess for males. These plots use the raw $p$-values results from GEMMA.

### Female early-life fitness

```{r qqplot1}
univariate_lmm_pvals <- SNP_clumps %>%
  select(contains("pvalue")) %>% filter(LD_subset) %>% as.data.frame()
qqman::qq(univariate_lmm_pvals$pvalue_female_early_raw)
```

### Female late-life fitness

```{r qqplot2}
qqman::qq(univariate_lmm_pvals$pvalue_female_late_raw)
```

### Male early-life fitness

```{r qqplot3}
qqman::qq(univariate_lmm_pvals$pvalue_male_early_raw)
```

### Male late-life fitness

```{r qqplot4}
qqman::qq(univariate_lmm_pvals$pvalue_male_late_raw)
```

## Hex bin plots and correlations in variant effect sizes

### Effect sizes adjusted using 'data-driven' adaptive shrinkage in `mashr` {.tabset}

#### Plot

```{r hex1, message=FALSE, warning=FALSE, fig.showtext=TRUE, fig.height = 5, fig.width = 9}
hex_plot <- function(x, y, xlab, ylab, title,
                     xlim = c(-0.5, 0.5),
                     ylim = c(-0.3, 0.3)
                     ){
  dat <- SNP_clumps %>%
    mutate(facet = title) %>% 
    filter(LD_subset) # ensure only the LD SNPs from mashr are plotted
  ggplot(dat, aes_string(x, y)) + 
    geom_abline(linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 3) +
    geom_hline(yintercept = 0, linetype = 3) +
    stat_binhex(bins = 50)  +
    geom_density_2d(colour = "white", alpha = 0.6) + 
    scale_fill_viridis_c() + 
    coord_cartesian(xlim = xlim, ylim = ylim) + 
    facet_wrap(~ facet) +
    theme_bw() + xlab(xlab) + ylab(ylab) +
    theme(legend.position = "none",
          strip.background = element_blank(),
          strip.text = element_text(hjust=0))   + 
    theme(text = element_text(family = "Raleway", size = 12))

}

p1 <- hex_plot("beta_female_early_mashr_ED", 
               "beta_male_early_mashr_ED", 
               "Effect on female fitness", 
               NULL,
               "A. Early life fitness")

p2 <- hex_plot("beta_female_late_mashr_ED", 
               "beta_male_late_mashr_ED", 
               "Effect on female fitness", 
               NULL,
               "B. Late life fitness")

grid.arrange(p1, p2, ncol = 2, left = "Effect on male fitness")
```

**Figure 2**: Effect sizes of `r nrow(SNP_clumps)` loci (i.e. groups of one or more polymorphic sites in complete linkage disequilibrium) on male and female fitness, plotted separately for the early-life and late-life estimates. The effect sizes estimated using GEMMA have been corrected using `mashr`, using the data-driven method to apply shrinkage (Figure SX shows the raw estimates). The data have been binned into hexagons, with the colour and contour lines indicating the number of loci. The diagonal line represents $y=x$. Positive effect sizes indicate that the minor allele is associated with higher fitness.

```{r echo=FALSE, fig.showtext=TRUE, results='hide'}
fig2_top <- grid.arrange(p1, p2, ncol = 2, left = "Effect on male fitness")

# ggsave("figures/SNP_effect_ED.pdf",
#        grid.arrange(p1, p2, ncol = 2, left = "Effect on male fitness"),
#   height = 4, width = 8)
```


#### Pearson correlation

```{r}
SNP_clumps %>%
  select(contains("mashr_ED")) %>%
  select(contains("beta")) %>%
  rename_all(~ paste(ifelse(str_detect(.x, "female"), "Female", "Male"), 
                     ifelse(str_detect(.x, "early"), "early", "late"))) %>%
  cor(use = "pairwise.complete.obs") %>% 
  kable(digits = 3) %>% kable_styling(full_width = FALSE)
```


<!-- ### Effect sizes adjusted using 'canonical' adaptive shrinkage {.tabset} -->

<!-- #### Plot -->

<!-- The unusual distribution here reflects the fact that this analysis imposed constraints on the effects of each locus. For example, loci assigned to the 'Female-specific' category with high probability necessarily have an effect size close to zero in males. The figure should be interpreted cautiously - the main purpose of the canonical analysis is to inference the mixture proportions, and to assign mixture probabilities to each locus, not to accurately estimate the true effect size of each variant.  -->

<!-- ```{r hex2, message=FALSE, warning=FALSE, fig.showtext=TRUE, fig.height = 5, fig.width = 9} -->
<!-- p1 <- hex_plot("beta_female_early_mashr_canonical",  -->
<!--            "beta_male_early_mashr_canonical",  -->
<!--            "Effect on female fitness",  -->
<!--            NULL, -->
<!--            "A. Early life fitness") -->

<!-- p2 <- hex_plot("beta_female_late_mashr_canonical",  -->
<!--            "beta_male_late_mashr_canonical",  -->
<!--            "Effect on female fitness",  -->
<!--            NULL, -->
<!--            "B. Late life fitness") -->

<!-- grid.arrange(p1, p2, ncol = 2, left = "Effect on male fitness") -->
<!-- ``` -->



<!-- #### Pearson correlation -->

<!-- ```{r} -->
<!-- SNP_clumps %>% -->
<!--   select(contains("mashr_canonical")) %>% -->
<!--   select(contains("beta")) %>% -->
<!--   rename_all(~ paste(ifelse(str_detect(.x, "female"), "Female", "Male"),  -->
<!--                      ifelse(str_detect(.x, "early"), "early", "late"))) %>% -->
<!--   cor(use = "pairwise.complete.obs") %>%  -->
<!--   kable(digits = 3) %>% kable_styling(full_width = FALSE) -->
<!-- ``` -->

### Unadjusted effect sizes {.tabset}

#### Plot

Uses the variant effect sizes output by GEMMA, without applying any shrinkage (i.e. this is the data that was adjusted using `mashr`). 

```{r hex3, message=FALSE, warning=FALSE, fig.showtext=TRUE, fig.height = 5, fig.width = 9}
p1 <- hex_plot("beta_female_early_raw", 
           "beta_male_early_raw", 
           "Effect on female fitness", 
           NULL,
           "A. Early life fitness",
           xlim = c(-1, 1),
           ylim = c(-1, 1))

p2 <- hex_plot("beta_female_late_raw", 
           "beta_male_late_raw", 
           "Effect on female fitness", 
           NULL,
           "B. Late life fitness",
           xlim = c(-1, 1),
           ylim = c(-1, 1))

grid.arrange(p1, p2, ncol = 2, left = "Effect on male fitness")
```

#### Pearson correlation

```{r}
SNP_clumps %>%
  select(contains("raw")) %>%
  select(contains("beta")) %>%
  rename_all(~ paste(ifelse(str_detect(.x, "female"), "Female", "Male"), 
                     ifelse(str_detect(.x, "early"), "early", "late"))) %>%
  cor(use = "pairwise.complete.obs") %>% 
  kable(digits = 3) %>% kable_styling(full_width = FALSE)
```

## Average effect sizes are negative {.tabset}
Each of the following four tests is an intercept-only linear model, weighted by the inverse of the standard error for the focal variant's effect size (so, loci where the effect effect size was measured with more precision are weighted more heavily). The tests are run on the LD-pruned subset of SNPs, minimising pseudoreplication. A non-zero intercept term indicates that major (or minor) alleles tend to have consistently positive or negative associations with the focal fitness trait.

These results indicate that the minor allele tends to _reduce_ fitness, relative to the major allele. It's a weak effect (note the small value in the Estimate column), this may reflect the large uncertainty with which the effect sizes are measured, in addition to a true biological result that most loci have little or no relationship with the fitness traits we measured. 

### Female early-life
```{r}
summary(lm(beta_female_early_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_female_early_raw))
```

### Female late-life
```{r}
summary(lm(beta_female_late_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_female_late_raw))
```

### Male early-life
```{r}
summary(lm(beta_male_early_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_male_early_raw))
```

### Male late-life
```{r}
summary(lm(beta_male_late_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_male_late_raw))
```

## Manhattan plot

**Figure legend**: Manhattan plot showing the p-values (-log_{10} transformed) for each variant, from mixed model GWAS of female (top) and male (bottom) early-life fitness testing the null hypothesis that the two alleles are associated with equal fitness values.  

```{r fig.height=12}
manhattan_data <- tbl(db, "univariate_lmm_results") %>% 
  select(SNP, starts_with("P_"),
         pvalue_female_early_raw, pvalue_male_early_raw) %>%
  distinct() %>%
  collect(n=Inf) %>%
  mutate(position = str_split(SNP, "_"),
         chr = map_chr(position, ~ .x[1]),
         position = as.numeric(map_chr(position, ~ .x[2]))) %>% 
         # P_pleiotropy = 1 - P_null - P_female_specific - P_male_specific,
         # SA_not_concord = P_sex_antag > P_equal_effects,
         # top_SA = P_sex_antag >= quantile(P_sex_antag, probs=0.9999)) %>% 
  filter(chr != "4") 

max_pos <- manhattan_data %>%
  group_by(chr) %>%
  summarise(max_pos = max(position), .groups = "drop") %>%
  as.data.frame()
max_pos$max_pos <- c(0, cumsum(max_pos$max_pos[1:4]))

manhattan_data <- manhattan_data %>%
    left_join(max_pos, by = "chr") %>%
    mutate(position = position + max_pos) 


p1 <- manhattan_data %>%
  ggplot(aes(position, -1 * log10(pvalue_female_early_raw), group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  # geom_point(data = manhattan_data %>% filter(top_SA), 
  #            aes(y = 8), size = 3, pch = 6) +
  scale_colour_brewer(palette = "Paired", name = "Chromosome") + 
  scale_y_continuous(limits = c(0,8)) +
  ylab(expression(paste("Effect on female early-life fitness (-", Log[10], " p)"))) + 
  xlab("") +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        # text = element_text(family = "Raleway", size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

p2 <- manhattan_data %>%
  ggplot(aes(position, -1 * log10(pvalue_male_early_raw), group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  scale_colour_brewer(palette = "Paired", name = "Chromosome") + 
  ylab(expression(paste("Effect on male early-life fitness (-", Log[10], " p)"))) + 
  xlab("Position") +
  scale_y_reverse(limits = c(8,0)) +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        # text = element_text(family = "Raleway", size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

grid_arrange_shared_legend <- function(..., ncol = length(list(...)), nrow = 1) {

  plots <- list(...)
  # position <- match.arg(position)
  g <- ggplotGrob(plots[[1]] + theme(legend.position = "right"))$grobs
  legend <- g[[which(sapply(g, function(x) x$name) == "guide-box")]]
  lheight <- sum(legend$height)
  lwidth <- sum(legend$width)
  gl <- lapply(plots, function(x) x + theme(legend.position="none"))
  gl <- c(gl, ncol = ncol, nrow = nrow)
  
  combined <- arrangeGrob(arrangeGrob(gl[[1]], gl[[2]], nrow = 2),
                          legend,
                          ncol = 2,
                          widths = unit.c(unit(1, "npc") - lwidth, lwidth))
  
  grid.newpage()
  grid.draw(combined)
  invisible(combined) # return gtable invisibly
}

grid_arrange_shared_legend(p1, p2)
```


```{r make_fig2, echo=FALSE, results='hide', message=F, warning=F, fig.showtext=TRUE}
# Make composite figure 2, with SNP effects and Manhattan plot

try(png("figures/fig2_SNPs_manhattan_plot.png", height = 8, width = 8, units = "in", res = 300))
try(grid.arrange(fig2_top, grid_arrange_shared_legend(p1, p2)))
try(dev.off())
```



## Investigating pleiotropy and polygenicity of fitness

Inspired by Boyle et al. 2017 (_Cell_, specifically the analysis in their Figure 1C), we sorted all the variants in order of effect size on female fitness, placed the variants in bins of 1000, and then calculated the average effect size for each bin for both male and female fitness. This analysis was performed on the raw SNP effect sizes from GEMMA, pruned to a set of SNPs in approximate LD with one another. 

If there is pleiotropy between male and female fitness, we would predict a correlation between the effect sizes for male and female fitness; on the contrary if there were no pleiotropy, we would see no correlation. 

Moreover, if fitness is highly polygenic ('omnigenic'; Boyle et al. 2017), we predict a tight, straight line relationship, because each bin would contain some variants with small but genuine associations with fitness, and these effects would replicate in the other fitness trait. On the contrary if genetic variance in fitness stems from just a few genes with larger effects, the relationship between male and female fitness would be flat in the center and sloped at the extremes.

Figure 4 shows that there was a very tight correlation between the average effects of the variants in each bin on male and female fitness. The data therefore suggest that variants associated with male fitness tend to also affect female fitness (in the same direction), and that a very large number of loci have small, concordant effects on fitness in both sexes. 

Interestingly there appears to be a curve in Figure 4A to the left of the x-axis, indicating that variants where the minor allele is strongly, negatively associated with female fitness are (on average) less negatively associated with male fitness than expected based on predictions from variants with weaker effects on female fitness. One possible explanation is that alleles that are highly highly detrimental to both sexes are usually purged by selection (or at least kept below the 5% MAF threshold that we used), whereas female-harming alleles that are neutral or beneficial in males are purged less often, resulting a greater proportion of female-limited or sexually antagonistic alleles towards the left of the x-axis.

```{r}
p1 <- SNP_clumps %>%
  filter(LD_subset) %>% # The figure looks the same whether or not the data are thinned to the LD set of SNPs
  arrange(beta_female_early_raw) %>%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         age = "A. Early life") %>%
  group_by(bin, age) %>%
  summarise(females = mean(beta_female_early_raw), males = mean(beta_male_early_raw)) %>%
  ggplot(aes(females, males)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  geom_point() +
  stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 0.6) +
  facet_wrap(~ age) +
  scale_y_continuous(limits = c(-0.1, 0.2)) +
  xlab(NULL) + ylab(NULL) + 
  theme_bw() +
  theme(strip.background = element_blank(),
        strip.text = element_text(hjust=0))   + 
  theme(text = element_text(family = "Raleway", size = 12))

p2 <- SNP_clumps %>%
  filter(LD_subset) %>% 
  arrange(beta_female_late_raw) %>%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         age = "B. Late life") %>%
  group_by(bin, age) %>%
  summarise(females = mean(beta_female_late_raw), males = mean(beta_male_late_raw)) %>%
  ggplot(aes(females, males)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  geom_point() +
  stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 0.6) +
  facet_wrap(~ age) +
  scale_y_continuous(limits = c(-0.1, 0.2)) +
  xlab(NULL) + ylab(NULL) + 
  theme_bw() +
  theme(strip.background = element_blank(),
        strip.text = element_text(hjust=0))   + 
  theme(text = element_text(family = "Raleway", size = 12))

grid.arrange(p1, p2, nrow = 1, bottom = "Mean effect size on female fitness", 
  left = "Mean effect size on male fitness")
```

```{r save_boyle_plot, echo=FALSE, fig.showtext=TRUE}
# ggsave("figures/boyle_plot.pdf", 
#        grid.arrange(p1, p2, nrow = 1, 
#                     bottom = "Mean effect size on female fitness", 
#                     left = "Mean effect size on male fitness"),
#        height = 3, width = 6)

try(png("figures/boyle_plot.png", height = 3, width = 6, units = "in", res = 300))
try(grid.arrange(p1, p2, nrow = 1, 
                    bottom = "Mean effect size on female fitness", 
                    left = "Mean effect size on male fitness"))
try(dev.off())


```





<!-- ## Effect of the minor allele -->

<!-- ```{r echo=FALSE} -->
<!-- n_loci <- tbl(db, "univariate_lmm_results") %>% pull(P_sex_antag) %>% length() -->
<!-- fraction_n <- floor(n_loci*0.001) -->
<!-- n_loci <- prettyNum(n_loci, big.mark = ",", scientific = FALSE) -->
<!-- fraction_n <- prettyNum(fraction_n, big.mark = ",", scientific = FALSE) -->
<!-- ``` -->


<!-- Evolutionary theory makes testable predictions about the effects of the major and minor alleles on fitness-related phenotypes. For example, the minor allele should generally be associated with the lower-fitness phenotype due to selection. Furthermore, it is interesting to ask whether the female-beneficial allele or the male-beneficial allele tends to be the major allele at sexually-antagonistic loci, as this reveals something about which sex is under stronger selection in the wild (provided that our lab-based fitness assays are somewhat correlated with natural selection). Finally, we might expect there to be a difference in effect sizes between loci where the minor allele is quite rare, compared to loci where the minor allele is quite common, for a few reasons. Selection should keep MAF low at loci with a strong (sex-averaged) fitness effect rare.  -->

<!-- To address these questions graphically, we here plot the effects on male and female early-life fitness of the minor allele, either for A) all loci in the LD-pruned set, B) the top 1% most sexually-antagonistic loci in this set, and C) the top 1% most sexually concordant loci in the set. For the latter two cases, we used the evidence ratios calculated above to rank loci from most to least sexually antagonistic (or concordant). The data are further subdivided into loci with lower MAF (between 0.05 and 0.1) and higher MAF (above 0.1). The plot shows the `mashr`-adjusted effect sizes, and considers the effect on early-life fitness in males and females (the corresponding plot for late-life fitness is very similar). -->

<!-- Considering all loci, most alleles have an estimated effect size close to zero, and there is higher variance for male fitness -->






<!-- ```{r} -->
<!-- histo_data <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_early < quantile(inter_sex_early, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_early > quantile(inter_sex_early, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_early_mashr_ED, beta_male_early_mashr_ED, class, MAF) -->

<!-- all_loci <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_early_mashr_ED, beta_male_early_mashr_ED, class, MAF) -->

<!-- bind_rows(histo_data, all_loci) %>%  -->
<!--   filter(!is.na(MAF)) %>%  -->
<!--   arrange(MAF, class) %>%  -->
<!--   mutate(facet = paste(class, MAF, sep = "\n"), -->
<!--          facet = factor(facet, unique(facet))) %>%  -->
<!-- ggplot(aes(beta_female_early_mashr_ED)) +  -->
<!--   geom_vline(xintercept = 0, linetype = 2) + -->
<!--   geom_histogram(bins=30, fill="tomato", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(aes(beta_male_early_mashr_ED), bins=30, fill="steelblue", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(bins=30, fill=NA, colour="black") +  -->
<!--   geom_histogram(aes(beta_male_early_mashr_ED), bins=30,fill=NA,colour="black") +  -->
<!--   facet_wrap(~ facet, scales = "free_y") + -->
<!--   xlab("Effect of the minor allele on early-life fitness\n(females:red, males:blue)") +  -->
<!--   theme_bw() + -->
<!--   theme(legend.position = "none", -->
<!--         strip.background = element_blank(),  -->
<!--         strip.text = element_text(hjust = 0), -->
<!--         text = element_text(family = "Raleway", size = 12)) + -->
<!--   ylab("Number of loci") -->
<!-- ``` -->

<!-- ```{r} -->
<!-- histo_data <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_late < quantile(inter_sex_late, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_late > quantile(inter_sex_late, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_late_mashr_ED, beta_male_late_mashr_ED, class, MAF) -->

<!-- all_loci <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female_late_mashr_ED, beta_male_late_mashr_ED, class, MAF) -->

<!-- bind_rows(histo_data, all_loci) %>%  -->
<!--   filter(!is.na(MAF)) %>%  -->
<!--   arrange(MAF, class) %>%  -->
<!--   mutate(facet = paste(class, MAF, sep = "\n"), -->
<!--          facet = factor(facet, unique(facet))) %>%  -->
<!--   ggplot(aes(beta_female_late_mashr_ED)) +  -->
<!--   geom_vline(xintercept = 0, linetype = 2) + -->
<!--   geom_histogram(bins=30, fill="tomato", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(aes(beta_male_late_mashr_ED), bins=30, fill="steelblue", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(bins=30, fill=NA, colour="black") +  -->
<!--   geom_histogram(aes(beta_male_late_mashr_ED), bins=30,fill=NA,colour="black") +  -->
<!--   facet_wrap(~ facet, scales = "free_y") + -->
<!--   xlab("Effect of the minor allele on late-life fitness\n(females:red, males:blue)") +  -->
<!--   theme_bw() + -->
<!--   theme(legend.position = "none", -->
<!--         strip.background = element_blank(),  -->
<!--         strip.text = element_text(hjust = 0), -->
<!--         text = element_text(family = "Raleway", size = 12)) + -->
<!--   ylab("Number of loci") -->
<!-- ``` -->

<!-- ```{r} -->
<!-- histo_data_early <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_early < quantile(inter_sex_early, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_early > quantile(inter_sex_early, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_early_mashr_ED, beta_male = beta_male_early_mashr_ED, class, MAF) -->

<!-- all_loci_early <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_early_mashr_ED, beta_male_early_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_early_mashr_ED, beta_male = beta_male_early_mashr_ED, class, MAF) -->

<!-- histo_data_late <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(class =  -->
<!--            case_when( -->
<!--              inter_sex_late < quantile(inter_sex_late, probs = 0.01)  ~ "Top 1% sexually antagonistic loci", -->
<!--              inter_sex_late > quantile(inter_sex_late, probs = 0.99)  ~ "Top 1% sexually concordant loci", -->
<!--            )) %>% -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5")) %>% -->
<!--   filter(!is.na(class)) %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_late_mashr_ED, beta_male = beta_male_late_mashr_ED, class, MAF) -->

<!-- all_loci_late <- left_join( -->
<!--   antagonism_evidence_ratios_gwas, -->
<!--   collect(tbl(db, "variants") %>% select(SNP, MAF)), by = "SNP") %>%  # Add the MAF for each SNP -->
<!--   select(-SNP) %>% distinct() %>% # Include SNPs in 100% LD a single time, such that SNP clumps are the unit of replication  -->
<!--   mutate(MAF = ifelse(MAF < 0.1, "0.05 < MAF < 0.1", "0.1 < MAF < 0.5"),  -->
<!--          class = "All loci") %>% -->
<!--   left_join(SNP_clumps %>% select(SNP_clump, beta_female_late_mashr_ED, beta_male_late_mashr_ED), by = "SNP_clump") %>%  -->
<!--   select(beta_female = beta_female_late_mashr_ED, beta_male = beta_male_late_mashr_ED, class, MAF) -->

<!-- bind_rows( -->
<!--   bind_rows(histo_data_early, all_loci_early) %>%  -->
<!--     mutate(stage = "early") %>%  -->
<!--     mutate(facet = paste(class, "early", sep = "\n")), -->

<!--   bind_rows(histo_data_late, all_loci_late) %>%  -->
<!--     mutate(stage = "late") %>%  -->
<!--     mutate(facet = paste(class, "late", sep = "\n")))%>%  -->
<!--   arrange(stage, class) %>%  -->
<!--   mutate(facet = factor(facet, unique(facet))) %>%  -->
<!--   # filter(MAF == "0.05 < MAF < 0.1") %>%  -->
<!--   ggplot(aes(beta_female)) +  -->
<!--   geom_vline(xintercept = 0, linetype = 2) + -->
<!--   geom_histogram(bins=30, fill="tomato", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(aes(beta_male), bins=30, fill="steelblue", colour=NA, alpha = 0.5) +  -->
<!--   geom_histogram(bins=30, fill=NA, colour="black") +  -->
<!--   geom_histogram(aes(beta_male), bins=30,fill=NA,colour="black") +  -->
<!--   facet_wrap(~ facet, scales = "free_y") + -->
<!--   xlab("Effect of the minor allele on fitness\n(females:red, males:blue)") +  -->
<!--   theme_bw() + -->
<!--   theme(legend.position = "none", -->
<!--         strip.background = element_blank(),  -->
<!--         strip.text = element_text(hjust = 0), -->
<!--         text = element_text(family = "Raleway", size = 12)) + -->
<!--   ylab("Number of loci") -->


<!-- ``` -->




## Frequencies of antagonistic loci and transcripts

The following figures show the proportions of loci/transcripts whose effect on female fitness (or early-life fitness) is similar or different to their effect on male fitness (or late-life fitness). The figures were made by first placing variant or transcript effect sizes (for female or early-life fitness) into quartiles, which we label as negative effects, weak negative effects, weak positive effects, and positive effects (taking advantage of the fact that the median effect size is very close to zero). We then do the same for male or late-life fitness, and plot the number of transcripts showing each combination of quartiles, giving an indication of how many loci/transcripts have aligned or opposing effects on the different fitness metrics.

The following plots were made after processing the effect sizes with `mashr`, which should reduce the number of false signals. Because there is a positive correlation between the sexes and age classes, the shrinkage applied by `mashr` reduces the number of loci/transcripts that appear to show antagonism between ages and sexes (this should control the number of 'false positive' antagonistic loci/transcripts).



### Frequencies of antagonistic loci 

#### Comparing effects on the sexes {.tabset}

Rather few loci are in the 1st quartile in their effect on male fitness and the 4th quartile for their effect on female fitness, or _vice versa_, suggesting that variants with strongly sex-opposite effects on fitness are rare

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 8}
SNP_effects <- SNP_clumps %>%
  filter(LD_subset) %>% 
  select(contains("beta")) %>% 
  select(contains("ED")) # this can be changed to "raw" to see the non-mashr effect sizes (i.e. statistical noise)
names(SNP_effects)[grepl("female_early", names(SNP_effects))] <- "FE"
names(SNP_effects)[grepl("female_late", names(SNP_effects))] <- "FL"
names(SNP_effects)[grepl("male_early", names(SNP_effects))] <- "ME"
names(SNP_effects)[grepl("male_late", names(SNP_effects))] <- "ML"

temp <- SNP_effects %>% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %>% 
  mutate_at(vars(starts_with("quartile")), ~ {
    SNP_effects <- .x
    SNP_effects[SNP_effects == 1] <- "Negative"
    SNP_effects[SNP_effects == 2] <- "Weakly\nnegative"
    SNP_effects[SNP_effects == 3] <- "Weakly\npositive"
    SNP_effects[SNP_effects == 4] <- "Positive"
    factor(SNP_effects, c("Negative", "Weakly\nnegative", "Weakly\npositive", "Positive"))})

mat_list <- list(table(temp$quartile_FE, temp$quartile_ME),
                 table(temp$quartile_FL, temp$quartile_ML),
                 table(temp$quartile_FE, temp$quartile_FL),
                 table(temp$quartile_ME, temp$quartile_ML))  %>% 
   map(~ as.data.frame(.x)) 
names(mat_list) <- c("FE, ME", "FL, ML", "FE, FL", "ME, ML")
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

cols <- c(brewer.pal(9, "Reds")[c(6,3)], brewer.pal(9, "Blues")[c(3,6)])

focal_dat <- rbind(
    as.data.frame(mat_list[[1]]) %>% 
      mutate(pp = "A. Early-life fitness"), 
    as.data.frame(mat_list[[2]]) %>% 
      mutate(pp = "B. Late-life fitness")) 

fig_4_top_row <- focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nmale fitness") + 
  xlab("Association with female fitness") + 
  ylab("Number of variants") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        #strip.text = element_text(hjust = 0, family = "Raleway", size = 12))
        strip.text = element_text(hjust = 0, size = 12))

# To count the total number of transcripts tested:
# focal_dat$Freq[focal_dat$pp == "C. Early-life fitness"] %>% sum()

fig_4_top_row
```

##### Table

```{r}
focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of variants` = Freq) %>% 
  kable() %>% kable_styling(full_width = FALSE)
```


#### Comparing effects on the age classes {.tabset}

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 8}
focal_dat <- rbind(
  as.data.frame(mat_list[[3]]) %>% 
    mutate(pp = "A. Female fitness"), 
  as.data.frame(mat_list[[4]]) %>% 
    mutate(pp = "B. Male fitness")) 

focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nlate-life fitness") + 
  xlab("Association with early-life fitness") + 
  ylab("Number of variants") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = "Raleway", size = 12))
```

##### Table

```{r}
focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of variants` = Freq) %>% 
  kable() %>% kable_styling(full_width = FALSE)
```


#### Run a $\chi^2$ test

The following $\chi^2$ test examines the null hypothesis that the proportion of candidate sexually antagonistic loci (i.e. those with a quartile 1 effect on fitness of sex $i$ and a quartile 4 effect on sex $j$) is equal when fitness associations are calculated using the early- and late-life fitness data. This null hypothesis is rejected: the % of SA loci is `r round(100 *(129+103) / 208987, 3)`% at early life and `r round(100 *(93+48) / 208987, 3)`% at late life, which is a `r round(0.1110117 / 0.06746831, 1)`-fold difference.

```{r}
chisq_table <- as.table(rbind(c(129+103, 208987 - (129+103)), c(93+48, 208987 - (93+48))))
dimnames(chisq_table) <- list(life_stage = c("Early_life", "Late_life"),
                              antagonism_status = c("Sex_antag","Non_sex_antag"))
chisq_table
chisq.test(chisq_table)
```


### Frequencies of antagonistic transcripts

#### Comparing effects on the sexes {.tabset}

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 8}
TWAS_ED <- readRDS("data/derived/TWAS/TWAS_ED.rds")

binned_transcripts <- data.frame(
  FBID = read_csv("data/derived/TWAS/TWAS_results.csv")$FBID,
  as.data.frame(get_pm(TWAS_ED))) %>% 
  as_tibble() %>% rename_all(~ str_remove_all(.x, "beta_")) %>% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %>% 
  mutate_at(vars(starts_with("quartile")), ~ {
    SNP_effects <- .x
    SNP_effects[SNP_effects == 1] <- "Negative"
    SNP_effects[SNP_effects == 2] <- "Weakly\nnegative"
    SNP_effects[SNP_effects == 3] <- "Weakly\npositive"
    SNP_effects[SNP_effects == 4] <- "Positive"
    factor(SNP_effects, c("Negative", "Weakly\nnegative", "Weakly\npositive", "Positive"))})


mat_list <- list(table(binned_transcripts$quartile_FE, binned_transcripts$quartile_ME),
                 table(binned_transcripts$quartile_FL, binned_transcripts$quartile_ML),
                 table(binned_transcripts$quartile_FE, binned_transcripts$quartile_FL),
                 table(binned_transcripts$quartile_ME, binned_transcripts$quartile_ML))  %>% 
   map(~ as.data.frame(.x)) 
names(mat_list) <- c("FE, ME", "FL, ML", "FE, FL", "ME, ML")
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

focal_dat <- rbind(
    as.data.frame(mat_list[[1]]) %>% 
      mutate(pp = "C. Early-life fitness"), 
    as.data.frame(mat_list[[2]]) %>% 
      mutate(pp = "D. Late-life fitness")) 

# To count the total number of transcripts tested, 14286:
# focal_dat$Freq[focal_dat$pp == "C. Early-life fitness"] %>% sum()

fig_4_bottom_row <- focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nmale fitness") + 
  xlab("Association with female fitness") + 
  ylab("Number of transcripts") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        #strip.text = element_text(hjust = 0, family = "Raleway", size = 12))
        strip.text = element_text(hjust = 0, size = 12))

fig_4_bottom_row
```


```{r fig.showtext=T, echo = FALSE, results='hide'}
# Code to make the composite Figure 4, the stacked bar chart
grid_arrange_shared_legend <- function(..., ncol = 1, nrow = 2, position = c("bottom", "right")) {

  plots <- list(...)
  position <- match.arg(position)
  g <- ggplotGrob(plots[[1]] + theme(legend.position = position))$grobs
  legend <- g[[which(sapply(g, function(x) x$name) == "guide-box")]]
  lheight <- sum(legend$height)
  lwidth <- sum(legend$width)
  gl <- lapply(plots, function(x) x + theme(legend.position="none"))
  gl <- c(gl, ncol = ncol, nrow = nrow)

  combined <- switch(position,
                     "bottom" = arrangeGrob(do.call(arrangeGrob, gl),
                                            legend,
                                            ncol = 1,
                                            heights = unit.c(unit(1, "npc") - lheight, lheight)),
                     "right" = arrangeGrob(do.call(arrangeGrob, gl),
                                           legend,
                                           ncol = 2,
                                           widths = unit.c(unit(1, "npc") - lwidth, lwidth)))

  grid.newpage()
  grid.draw(combined)

  # return gtable invisibly
  invisible(combined)

}


ggsave("figures/fig4_quartiles_plot.png", 
       grid_arrange_shared_legend(fig_4_top_row + 
                                    theme(axis.text.x = element_blank(),
                                          axis.ticks.x = element_blank()) +
                                    xlab(" "), 
                                  fig_4_bottom_row, position = "right"), 
       height = 6, width = 7)
```


##### Table

```{r}
focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of transcripts` = Freq) %>% 
  kable() %>% kable_styling(full_width = FALSE)
```

#### Run a $\chi^2$ test

The following $\chi^2$ test examines the null hypothesis that the proportion of candidate sexually antagonistic transcripts (i.e. those with a quartile 1 effect on fitness of sex $i$ and a quartile 4 effect on sex $j$) is equal when fitness associations are calculated using the early- and late-life fitness data. This null hypothesis is not rejected: the % of SA transcripts is `r round(100 *(691+689) / 14286, 2)`% at early life and `r round(100 *(677 + 664) / 14286, 2)`% at late life, which is a `r round(9.659807 / 9.386812, 2)`-fold difference.

```{r}
chisq_table <- as.table(rbind(c(691+689, 14286 - (691+689)), c(677 + 664, 14286 - (677 + 664))))
dimnames(chisq_table) <- list(life_stage = c("Early_life", "Late_life"),
                              antagonism_status = c("Sex_antag","Non_sex_antag"))
chisq_table
chisq.test(chisq_table)
```


#### Comparing effects on the age classes {.tabset}

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 11}
focal_dat <- rbind(
  as.data.frame(mat_list[[3]]) %>% 
    mutate(pp = "A. Female fitness"), 
  as.data.frame(mat_list[[4]]) %>% 
    mutate(pp = "B. Male fitness"))

focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nlate-life fitness") + 
  xlab("Association with early-life fitness") + 
  ylab("Number of transcripts") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = "Raleway", size = 12))
```

##### Table

```{r}
focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given effect on females)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of transcripts` = Freq) %>% 
  kable() %>% kable_styling(full_width = FALSE)
```


## Plotting and modelling the evidence for antagonism

### Calculate the evidence ratios

#### For the GWAS results
```{r}
get_antagonism_ratios <- function(dat){
  dat %>%
    
    # Convert the LFSR to the probability that effect size is positive
    mutate(pp_female_early = ifelse(pheno1 > 0, LFSR1, 1 - LFSR1),
           pp_female_late  = ifelse(pheno2 > 0, LFSR2, 1 - LFSR2),
           pp_male_early   = ifelse(pheno3 > 0, LFSR3, 1 - LFSR3),
           pp_male_late    = ifelse(pheno4 > 0, LFSR4, 1 - LFSR4)) %>%
    
    # Calculate the probabilities that beta_i and beta_j have the same/opposite signs
    mutate(p_sex_concord_early = pp_female_early * pp_male_early + 
             (1 - pp_female_early) * (1 - pp_male_early),
           p_sex_antag_early = pp_female_early * (1 - pp_male_early) + 
             (1 - pp_female_early) * pp_male_early,
           p_sex_concord_late  = pp_female_late * pp_male_late + 
             (1 - pp_female_late) * (1 - pp_male_late),
           p_sex_antag_late = pp_female_late * (1 - pp_male_late) + 
             (1 - pp_female_late) * pp_male_late,
           p_age_concord_females = pp_female_early * pp_female_late + 
             (1 - pp_female_early) * (1 - pp_female_late),
           p_age_antag_females = pp_female_early * (1 - pp_female_late) + 
             (1 - pp_female_early) * pp_female_late,
           p_age_concord_males = pp_male_early * pp_male_late + (1 - pp_male_early) * (1 - pp_male_late),
           p_age_antag_males = pp_male_early * (1 - pp_male_late) + (1 - pp_male_early) * pp_male_late) %>%
    
    # Find the ratios of some of these probabilities (i.e. "evidence ratios")
    mutate(inter_sex_early = p_sex_concord_early / p_sex_antag_early,
           inter_sex_late = p_sex_concord_late / p_sex_antag_late,
           inter_age_females = p_age_concord_females / p_age_antag_females,
           inter_age_males = p_age_concord_males / p_age_antag_males) 
}

antagonism_evidence_ratios_gwas <- all_snps %>%
  rename(pheno1 = beta_female_early_mashr_ED, 
         pheno2 = beta_female_late_mashr_ED, 
         pheno3 = beta_male_early_mashr_ED, 
         pheno4 = beta_male_late_mashr_ED,
         LFSR1 = LFSR_female_early_mashr_ED, 
         LFSR2 = LFSR_female_late_mashr_ED, 
         LFSR3 = LFSR_male_early_mashr_ED, 
         LFSR4 = LFSR_male_late_mashr_ED) %>% 
  select(SNP, SNP_clump, starts_with("pheno"), LFSR1, LFSR2, LFSR3, LFSR4) %>% 
  filter(!is.na(pheno1)) %>% 
  collect(n=Inf) %>% 
  distinct() %>% 
  get_antagonism_ratios() %>%
  select(SNP, SNP_clump, starts_with("inter")) 
```

#### For the TWAS results
```{r}
TWAS_ED <- readRDS("data/derived/TWAS/TWAS_ED.rds")

twas <- data.frame(
  FBID = read_csv("data/derived/TWAS/TWAS_results.csv")$FBID,
      as.data.frame(get_pm(TWAS_ED)),
      as.data.frame(get_lfsr(TWAS_ED)) %>% 
        rename_all(~str_replace_all(., "beta", "LFSR"))) %>% 
      as_tibble() %>%
  left_join(tbl(db, "genes") %>% select(FBID, chromosome) %>% collect(), by = "FBID") %>%
  left_join(read_csv("data/derived/gene_expression_by_sex.csv"), by = "FBID") %>%
  filter(chromosome %in% c("2L", "2R", "3L", "3R", "X")) %>%
  rename(pheno1 = beta_FE, 
         pheno2 = beta_FL, 
         pheno3 = beta_ME, 
         pheno4 = beta_ML,
         LFSR1 = LFSR_FE, 
         LFSR2 = LFSR_FL, 
         LFSR3 = LFSR_ME, 
         LFSR4 = LFSR_ML)

antagonism_evidence_ratios_twas <- twas %>% 
  get_antagonism_ratios() %>%
  select(FBID, chromosome, male_bias_in_expression, AveExpr, starts_with("inter")) %>% 
  mutate(chromosome = relevel(factor(chromosome), ref = "X"))
```

### Plot the evidence ratios

```{r evidence_ratio_plot, fig.showtext=TRUE, message=FALSE}
make_evidence_ratio_plot <- function(ERs, ymax, ylab){
  # Argument needs to be a dataframe of loci or transcripts, each identified by a column called "identifier"
  # There should be a col called "evidence_ratio", calculated from the LFSR from ED mashr. The "trait" col says which type of ER it is.
  
  ERs$trait[ERs$trait == "inter_sex_early"] <- "Inter-sex (early life)"
  ERs$trait[ERs$trait == "inter_sex_late"] <- "Inter-sex (late life)"
  ERs$trait[ERs$trait == "inter_age_females"] <- "Inter-age (females)"
  ERs$trait[ERs$trait == "inter_age_males"] <- "Inter-age (males)"
  
  antagonism_evidence_ratios <- ERs %>%
    mutate(trait = factor(trait, c("Inter-sex (early life)",
                                   "Inter-sex (late life)",
                                   "Inter-age (females)",
                                   "Inter-age (males)")))
  
  antagonism_evidence_ratios %>%
    ggplot(aes(log2(evidence_ratio))) + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio < 1), 
                   bins = 500, fill = "#FF635C") + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio > 1), 
                   bins = 500, fill = "#5B8AFD") +
    coord_cartesian(xlim = c(-10, 10), ylim = c(0, ymax)) + 
    scale_x_continuous(breaks = c(-10,  -6,  -2,   2,   6,  10), 
                       labels = c(paste("1/",2 ^ c(10,  6,  2), sep = ""), 2 ^ c(2,6,10))) + 
    facet_wrap(~ trait) + 
    xlab("Evidence ratio (log2 scale)") + ylab(ylab) + 
    theme_bw() + 
    theme(panel.border = element_rect(size = 0.8),
          text = element_text(family = "Raleway", size = 12),
          axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
          strip.background = element_blank())
}

GWAS_ratios_plot <- antagonism_evidence_ratios_gwas %>% 
  select(SNP_clump,  starts_with("inter")) %>%
  distinct() %>% # Include SNPs in 100% LD a single time, such that these clumps are the unit of replication
  gather(trait, evidence_ratio, -SNP_clump) %>% 
  rename(identifier = SNP_clump) %>% 
  make_evidence_ratio_plot(ymax = 5000, ylab = "Number of loci")

# For counting the number of transcripts with 50x more evidence for antagonism 
# antagonism_evidence_ratios_twas %>% 
#   select(FBID,  starts_with("inter")) %>%
#   gather(trait, evidence_ratio, -FBID) %>% 
#   rename(identifier = FBID) %>% filter(evidence_ratio < 1/50) %>% summarise(n())

TWAS_ratios_plot <- antagonism_evidence_ratios_twas %>% 
  select(FBID,  starts_with("inter")) %>%
  gather(trait, evidence_ratio, -FBID) %>% 
  rename(identifier = FBID) %>% 
  make_evidence_ratio_plot(ymax = 500, ylab = "Number of transcripts")

pp <- plot_grid(GWAS_ratios_plot, TWAS_ratios_plot, 
                   labels = c('A', 'B'), label_size = 12)
# ggdraw(add_sub(pp, "Evidence ratio (log scale)", 
#                vpadding=grid::unit(0,"lines"), 
#                y=6.3, x=0.5, vjust=6))
ggsave("figures/antagonism_ratios.pdf", pp, width = 8.5, height = 4.9)

pp
```


### Model the evidence ratios

```{r}
# gwas models

# Annotate it with MAF, site, class, etc. 
dat <- left_join(antagonism_evidence_ratios_gwas,
          collect(tbl(db, "variants")), by = "SNP") %>% 
  distinct() %>% mutate(site.class = str_to_title(site.class))

# Focus only on the commonest site classes:
dat <- dat %>% 
  filter(site.class %in% c("Intron", "Intergenic", "Downstream", "Upstream",
                           "Synonymous_coding", "Non_synonymous_coding", "Utr_3_prime",
                           "Exon", "Utr_5_prime")) 

# Remove chromosome 4 (too few sites)
dat <- dat %>% filter(chr != "4") 

# If there are multiple site classes for a SNP, or multiple SNPs 
# in the same 100% LD clump, pick one SNP and/or 1 site class at random
set.seed(1)
dat <- dat[sample(nrow(dat), nrow(dat)), ] %>%
  split(.$SNP_clump) %>%
  map_df(~ .x[1, ])

dat <- dat %>% arrange(site.class)
dat$site.class <- relevel(factor(dat$site.class), ref = "Synonymous_coding")
dat$chr <- relevel(factor(dat$chr), ref = "X")

n_loci <- prettyNum(nrow(dat), big.mark = ",", scientific = FALSE)


intersex_early_model_gwas <- glm(antagonistic ~ chr + MAF + site.class, 
             data = dat %>% 
               mutate(antagonistic = as.numeric(inter_sex_early <= 
                        quantile(dat$inter_sex_early, probs=0.01))), 
    family = "binomial")
car::Anova(intersex_early_model_gwas, type = 3)
intersex_early_model_gwas <- update(intersex_early_model_gwas,  ~ . -site.class)
car::Anova(intersex_early_model_gwas, type = 3)

summary(intersex_early_model_gwas)


intersex_late_model_gwas <- glm(antagonistic ~ chr + MAF + site.class, 
             data = dat %>% 
               mutate(antagonistic = inter_sex_late <= 
                        quantile(dat$inter_sex_late, probs=0.01)), 
    family = "binomial")
car::Anova(intersex_late_model_gwas, type = 3)
intersex_late_model_gwas <- update(intersex_late_model_gwas,  ~ . -site.class)
car::Anova(intersex_late_model_gwas, type = 3)

summary(intersex_late_model_gwas)
```


```{r}
# twas models
n_transcripts <- prettyNum(nrow(antagonism_evidence_ratios_twas), big.mark = ",", scientific = FALSE)

twas_model_data <- antagonism_evidence_ratios_twas %>% 
  mutate(antagonistic = inter_sex_early <= 
           quantile(antagonism_evidence_ratios_twas$inter_sex_early, probs=0.01),
         absolute_sex_bias_in_expression = abs(male_bias_in_expression),
         mean_expression_level = AveExpr)


intersex_early_twas <- glm(antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level, 
             data = twas_model_data, 
    family = "binomial")
car::Anova(intersex_early_twas, type = 3)
summary(intersex_early_twas)
```


```{r}
intersex_late_model_twas <- glm(antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level, 
             data = antagonism_evidence_ratios_twas %>% 
  mutate(antagonistic = inter_sex_late <= 
           quantile(antagonism_evidence_ratios_twas$inter_sex_late, probs=0.01),
         absolute_sex_bias_in_expression = abs(male_bias_in_expression),
         mean_expression_level = AveExpr), 
    family = "binomial")
car::Anova(intersex_late_model_twas, type = 3)
intersex_late_model_twas <- update(intersex_late_model_twas,  ~ . -absolute_sex_bias_in_expression -chromosome)
car::Anova(intersex_late_model_twas, type = 3)
summary(intersex_late_model_twas)
```



```{r} 
# plotting models
new <- data.frame(MAF = seq(from = 0.05, to = 0.5, by = 0.01), chr = "X")
inv_logit <- function(x) exp(x)/(1+exp(x))
preds <- data.frame(new, prop = predict(intersex_early_model_gwas, newdata = new, se.fit=T, type = "link")) %>% 
  mutate(y = prop.fit, lwr = prop.fit - 1.96*prop.se.fit, upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), lwr = inv_logit(lwr), upr = inv_logit(upr))
p1 <- preds %>% 
  ggplot(aes(MAF,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  scale_x_continuous(limits = c(0, 0.5)) + 
  scale_y_continuous(limits = c(0, max(preds$y))) + 
  theme_bw() +#+ theme(text = element_text(family = "Raleway", size = 12)) 
  ylab("Probability of being in the top 1%\nsexually antagonistic loci (\u00B1 95% CIs)") +
  xlab("Minor allele frequency")

new2 <- data.frame(chr = levels(dat$chr), MAF = 0.3)
inv_logit <- function(x) exp(x)/(1+exp(x))
preds2 <- data.frame(new2, prop = predict(intersex_early_model_gwas, newdata = new2, se.fit=T, type = "link")) %>% 
  mutate(y = prop.fit, lwr = prop.fit - 1.96*prop.se.fit, upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), lwr = inv_logit(lwr), upr = inv_logit(upr))
p2 <- preds2 %>% 
  ggplot(aes(chr,  y)) + 
  geom_point(size = 3) + 
  geom_errorbar(aes(ymin = lwr, ymax = upr), width = 0) + 
  theme_bw() +#+ theme(text = element_text(family = "Raleway", size = 12)) 
  ylab(" \n ") +
  xlab("Chromosome arm")

new <- data.frame(
  mean_expression_level = seq(from = min(twas_model_data$mean_expression_level), 
                                to = max(twas_model_data$mean_expression_level), by = 0.01), 
                  chromosome = "X", 
  absolute_sex_bias_in_expression = 0,
                  mean_expression_level = mean(twas_model_data$absolute_sex_bias_in_expression))
inv_logit <- function(x) exp(x)/(1+exp(x))
preds <- data.frame(new, prop = predict(intersex_early_twas, newdata = new, se.fit=T, type = "link")) %>% 
  mutate(y = prop.fit, lwr = prop.fit - 1.96*prop.se.fit, upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), lwr = inv_logit(lwr), upr = inv_logit(upr))
p3 <- preds %>% 
  ggplot(aes(mean_expression_level,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  ylab("Probability of being in the top 1%\nsexually antagonistic transcripts (\u00B1 95% CIs)") +
  xlab("Average expression level") + 
  theme_bw() #+ theme(text = element_text(family = "Raleway", size = 12)) 


blank <- ggplot() + theme_void()


fig_6_models <- grid.arrange(
grid.arrange(p1, p2, ncol = 2), grid.arrange(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))
ggsave("figures/fig6_models.pdf", fig_6_models, width = 6, height = 7)
```

