---
title: "Plots and models of variant effect sizes"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r, include=FALSE}
library(knitrhooks) # for scrollable R output. remotes::install_github("nathaneastwood/knitrhooks")
output_max_height()
knitr::opts_chunk$set(output_max_height = "300px")
```

```{r include=FALSE}
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
```


```{r message=FALSE, warning=FALSE, results="hide"}
library(tidyverse)
library(gridExtra)
# library(qqman)
library(ggbeeswarm)
library(Hmisc)
library(showtext) # For fancy Google font in figures
library(mashr)
library(kableExtra)
library(cowplot)
library(grid)
library(RColorBrewer)
library(car)
library(brms)
library(tidybayes)
library(bigsnpr)
library(glue)
library(patchwork)
library(ggpubr)
library(staplr) 

# Note: you may need to re-download plink to get this to run on non-Mac systems
# I used bigsnpr::download_plink()
plink <- paste(getwd(), "code/plink", sep = "/")

font_add_google(name = "Raleway", family = "Raleway", regular.wt = 400, bold.wt = 700) # Install font from Google Fonts
showtext_auto()

db <- DBI::dbConnect(RSQLite::SQLite(), 
                     "data/derived/annotations.sqlite3")

# Results for all 1,613,615 SNPs, even those that are in 100% LD with others (these are grouped up by the SNP_clump column)
all_snps <- tbl(db, "univariate_lmm_results")

# All SNPs and SNP groups that are in <100% LD with one another (n = 1,207,357)
SNP_clumps <- all_snps %>% select(-SNP) %>% distinct() %>% collect(n = Inf)

# Subsetting variable to get the approx-LD subset of SNPs
LD_subset <- !is.na(SNP_clumps$LFSR_female_early_mashr_ED)

# Load the predicted line means, as calculated by get_predicted_line_means
predicted_line_means <- read_csv("data/derived/predicted_line_means.csv")

# Load and clean the effect sizes from GWAS and join with SNP annotations
univariate_lmm_results <- tbl(db, "univariate_lmm_results") %>% 
  select(-contains("canonical"), 
         -contains("raw")) %>% 
  inner_join(tbl(db, "variants") %>% 
               select(SNP, FBID, site.class, distance.to.gene, MAF), 
             by = "SNP") %>%
  left_join(
    tbl(db, "genes") %>% 
      select(FBID, gene_name), by = "FBID") %>%
  collect(n = Inf) %>%
  rename_all(~ gsub("beta_", "", .x)) %>%
  rename_all(~ gsub("_mashr_ED", "", .x))

univariate_lmm_results <- univariate_lmm_results %>%
  mutate(site.class = gsub("5_", "5-", site.class),
         site.class = gsub("3_", "3-", site.class),
         site.class = gsub("NON_", "NON-", site.class),
         site.class = gsub("_", " ", site.class),
         site.class = capitalize(tolower(site.class)),
         site.class = gsub("Utr", "UTR", site.class)
  )
```


<!-- ```{r} -->
<!-- # testing ruzicka comparison -->

<!-- xx <- read_tsv("~/Downloads/Assoc_Data.txt") -->

<!-- xx %>% arrange(-MAF_DGRP) %>% head() %>% as.data.frame() -->

<!-- ruz_data <- xx %>% select(ruzicka_SNP_name = Predictor, chr = Chromosome, position = Basepair, ruzicka_MAF = MAF, ruzicka_effect_size = Effect, ruzicka_wald_stat = Wald_Stat, ruzicka_wald_p = Wald_P) -->

<!-- merged <- all_snps %>% select(SNP, SNP_clump, contains("_mashr_ED")) %>% -->
<!--   filter(!is.na(beta_female_early_mashr_ED)) %>% -->
<!--   # collect(n=1000) %>% -->
<!--   collect(n=Inf) %>% -->
<!--   distinct() %>% -->
<!--   mutate(position = str_split(SNP, "_"), -->
<!--          chr = map_chr(position, ~ .x[1]), -->
<!--          position = as.numeric(map_chr(position, ~ .x[2]))) %>% -->
<!--   left_join(ruz_data, by = c("position", "chr")) -->

<!-- merged <- merged[complete.cases(merged), ] -->

<!-- merged <- left_join(merged, -->
<!--           antagonism_evidence_ratios_gwas %>% arrange(inter_sex_early) %>% mutate(inter_sex_early = log2(inter_sex_early))) -->

<!-- ggplot(merged, aes(x = ruzicka_wald_p, y = inter_sex_early)) + -->
<!--   geom_point() -->

<!-- ruzicka_SA <- xx %>% arrange(-MAF_DGRP) %>% filter(fdr<0.3) %>% select(chr = Chromosome, position = Basepair) -->

<!-- ruz_hits <- merged %>% filter(paste(chr, position) %in% paste(ruzicka_SA$chr, ruzicka_SA$position)) %>% pull(inter_sex_early) -->
<!-- nonruz_hits <- merged %>% filter(!(paste(chr, position) %in% paste(ruzicka_SA$chr, ruzicka_SA$position))) %>% pull(inter_sex_early) -->
<!-- wilcox.test(nonruz_hits, ruz_hits) -->
<!-- median(nonruz_hits) -->
<!-- median(ruz_hits) -->

<!-- hist(-1 * (merged %>% filter(inter_sex_early <= quantile(inter_sex_early, 0.99)) %>% pull(ruzicka_effect_size) %>% log10)) -->

<!-- bind_rows(merged %>% filter(inter_sex_early <= quantile(inter_sex_early, 0.001)) %>% mutate(type ="holmanSA"), -->
<!--           merged %>% filter(inter_sex_early >= quantile(inter_sex_early, 0.999)) %>% mutate(type ="holmanSC")) %>% -->
<!--   ggplot(aes(type, ruzicka_effect_size)) + -->
<!--   geom_boxplot(outlier.shape = NA) + geom_beeswarm(alpha=0.3, aes(size = -1*log10(ruzicka_wald_p))) -->

<!-- merged %>% filter(inter_sex_early <= quantile(inter_sex_early, 0.01) & -->
<!--                     ruzicka_wald_stat < 0 & -->
<!--                     ruzicka_wald_p < quantile(ruzicka_wald_p, 0.01)) %>% -->
<!--   pull(SNP_clump) -->
<!-- tbl(db, "variants") %>% filter(SNP=="3L_4306295_SNP") -->
<!-- tbl(db, "variants") %>% filter(SNP=="3L_4346362_SNP") -->

<!-- merged %>% filter(SNP=="3L_4346362_SNP") %>% as.data.frame() -->


<!-- xx<- select(merged, SNP, ruzicka_MAF) %>% left_join(tbl(db, "variants") %>% select(SNP, MAF) %>%  filter(SNP %in% !!merged$SNP) %>% collect(n=Inf)) -->
<!-- cor(xx$ruzicka_MAF, xx$MAF, use = "pairwise.complete.obs") -->
<!-- ``` -->


<!-- ## Q-Q plots {.tabset} -->

<!-- Here are some quantile-quantile plots, which are commonly used to check GWAS results, and to test the hypothesis that there are more SNPs than expected showing large effects on the trait of interest. There is a modest excess of loci with effects on female fitness, but not much of a visible excess for males. These plots use the raw $p$-values results from GEMMA. -->

<!-- ### Female early-life fitness -->

<!-- ```{r qqplot1} -->
<!-- univariate_lmm_pvals <- SNP_clumps %>% -->
<!--   select(contains("pvalue")) %>% filter(LD_subset) %>% as.data.frame() -->
<!-- qqman::qq(univariate_lmm_pvals$pvalue_female_early_raw) -->
<!-- ``` -->

<!-- ### Female late-life fitness -->

<!-- ```{r qqplot2} -->
<!-- qqman::qq(univariate_lmm_pvals$pvalue_female_late_raw) -->
<!-- ``` -->

<!-- ### Male early-life fitness -->

<!-- ```{r qqplot3} -->
<!-- qqman::qq(univariate_lmm_pvals$pvalue_male_early_raw) -->
<!-- ``` -->

<!-- ### Male late-life fitness -->

<!-- ```{r qqplot4} -->
<!-- qqman::qq(univariate_lmm_pvals$pvalue_male_late_raw) -->
<!-- ``` -->

## Hex bin plots and correlations in variant effect sizes

### Effect sizes adjusted using 'data-driven' adaptive shrinkage in `mashr` {.tabset}

#### Plot

```{r hex1, message=FALSE, warning=FALSE, fig.showtext=TRUE, fig.height = 5, fig.width = 9}
snp_effect_fig <- bind_rows(
  SNP_clumps %>%
    filter(LD_subset) %>%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_early_mashr_ED, male = beta_male_early_mashr_ED) %>% 
    mutate(age_class = "A. Early-life fitness"),
  SNP_clumps %>%
    filter(LD_subset) %>%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_late_mashr_ED, male = beta_male_late_mashr_ED) %>% 
    mutate(age_class = "B. Late-life fitness")) %>% 
  ggplot(aes(female, male)) + 
  geom_vline(xintercept = 0, linetype = 3) +
  geom_hline(yintercept = 0, linetype = 3) +
  stat_binhex(bins = 50)  +
  geom_density_2d(colour = "white", alpha = 0.6) + 
  scale_fill_viridis_c() + 
  facet_wrap(~ age_class) +
  theme_bw() + xlab("Effect on female fitness") + ylab("Effect on male fitness") +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text = element_text(hjust=0)) + 
  theme(text = element_text(family = "Raleway", size = 12))

fig2_top <- snp_effect_fig

snp_effect_fig
```

**Top part of Figure 2**: `mashr`-adjusted effect sizes of `r nrow(SNP_clumps)` loci (i.e. groups of one or more polymorphic sites in complete linkage disequilibrium) on male and female early- and late-life fitness. The data have been binned into hexagons, with the colour and contour lines indicating the number of loci. The diagonal line represents $y=x$. Positive effect sizes indicate that the minor allele is associated with higher fitness, while negative effects indicate that the major allele is associated with higher fitness.


#### Pearson correlation matrix

```{r}
SNP_clumps %>%
  select(contains("mashr_ED")) %>%
  select(contains("beta")) %>%
  rename_all(~ paste(ifelse(str_detect(.x, "female"), "Female", "Male"), 
                     ifelse(str_detect(.x, "early"), "early", "late"))) %>%
  cor(use = "pairwise.complete.obs") %>% 
  kable(digits = 3) %>% kable_styling(full_width = FALSE)
```


### Unadjusted effect sizes {.tabset}

#### Plot

Uses the variant effect sizes output by GEMMA, without applying any shrinkage (i.e. this is the input data that was adjusted using `mashr`). 

```{r hex3, message=FALSE, warning=FALSE, fig.showtext=TRUE, fig.height = 5, fig.width = 9}
bind_rows(
  SNP_clumps %>%
    filter(LD_subset) %>%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_early_raw, male = beta_male_early_raw) %>% 
    mutate(age_class = "A. Early-life fitness"),
  SNP_clumps %>%
    filter(LD_subset) %>%  # ensure only the LD SNPs from mashr are plotted
    select(female = beta_female_late_raw, male = beta_male_late_raw) %>% 
    mutate(age_class = "B. Late-life fitness")) %>% 
  ggplot(aes(female, male)) + 
  geom_vline(xintercept = 0, linetype = 3) +
  geom_hline(yintercept = 0, linetype = 3) +
  stat_binhex(bins = 50)  +
  geom_density_2d(colour = "white", alpha = 0.6) + 
  scale_fill_viridis_c() + 
  facet_wrap(~ age_class) +
  theme_bw() + xlab("Effect on female fitness") + ylab("Effect on male fitness") +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text = element_text(hjust=0)) + 
  theme(text = element_text(family = "Raleway", size = 12))
```

#### Pearson correlation matrix

```{r}
SNP_clumps %>%
  select(contains("raw")) %>%
  select(contains("beta")) %>%
  rename_all(~ paste(ifelse(str_detect(.x, "female"), "Female", "Male"), 
                     ifelse(str_detect(.x, "early"), "early", "late"))) %>%
  cor(use = "pairwise.complete.obs") %>% 
  kable(digits = 3) %>% kable_styling(full_width = FALSE)
```

## Average effect sizes are negative {.tabset}
Each of the following four tests is an intercept-only linear model, weighted by the inverse of the standard error for the focal variant's effect size (so, loci where the effect effect size was measured with more precision are weighted more heavily). The tests are run on the LD-pruned subset of SNPs, minimising pseudoreplication caused by non-independence in the data. An intercept term less than zero indicates that on average, the minor allele tends to be associated with lower values of the focal fitness trait (compared to the major allele).

These results indicate that the minor allele tends to _reduce_ fitness, relative to the major allele. It's a weak effect (note the small value in the Estimate column), this may reflect the large uncertainty with which the effect sizes are measured, in addition to a true biological result that most loci have little or no relationship with the fitness traits we measured. 

### Female early-life
```{r}
mod1 <- summary(lm(beta_female_early_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_female_early_raw))
mod1
```

### Female late-life
```{r}
mod2 <- summary(lm(beta_female_late_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_female_late_raw))
mod2
```

### Male early-life
```{r}
mod3 <- summary(lm(beta_male_early_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_male_early_raw))
mod3
```

### Male late-life
```{r}
mod4 <- summary(lm(beta_male_late_raw ~ 1, 
           data = SNP_clumps %>% filter(LD_subset), 
           weights = 1 / SE_male_late_raw))
mod4

medians <- SNP_clumps %>% filter(LD_subset) %>% 
  summarise(beta_female_early_raw = mean(beta_female_early_raw),
            beta_female_late_raw = mean(beta_female_late_raw),
            beta_male_early_raw = mean(beta_male_early_raw),
            beta_male_late_raw = mean(beta_male_late_raw)) %>% 
  unlist() %>% unname()

rbind(mod1$coefficients,
      mod2$coefficients,
      mod3$coefficients,
      mod4$coefficients) %>% as_tibble() %>% 
  mutate(`Fitness component` = 
           c("Female fitness early", "Female fitness late", 
             "Male fitness early", "Male fitness late"), 
         .before = names(.)[1]) %>% 
  mutate(Estimate = paste(format(round(Estimate, 5), nsmall = 5), 
                          " (", format(round(`Std. Error`, 5), nsmall = 5), ")", sep = "")) %>% 
  mutate(`Median variant effect` = format(round(medians, 5), nsmall = 5), .after = "Estimate") %>% 
  rename(p = `Pr(>|t|)`, `Mean (SE) variant effect` = Estimate) %>% 
  mutate(p = formatC(p, format = "e", digits = 2)) %>% 
  select(-`Std. Error`) %>% 
  saveRDS("data/derived/supp_table_of_variant_effect_means.rds")
```

## Manhattan plot

The code chunk below makes the Manhattan plot, and combines it with the plot above to make the composite Figure 2 shown in the paper. 

```{r manhattan, fig.showtext=TRUE, fig.height = 8, fig.width = 8}
manhattan_data <- tbl(db, "univariate_lmm_results") %>% 
  select(SNP, pvalue_female_early_raw, pvalue_male_early_raw) %>%
  distinct() %>%
  collect(n=Inf) %>%
  mutate(position = str_split(SNP, "_"),
         chr = map_chr(position, ~ .x[1]),
         position = as.numeric(map_chr(position, ~ .x[2]))) %>% 
  filter(chr != "4") 

max_pos <- manhattan_data %>%
  group_by(chr) %>%
  summarise(max_pos = max(position), .groups = "drop") %>%
  as.data.frame()
max_pos$max_pos <- c(0, cumsum(max_pos$max_pos[1:4]))

manhattan_data <- manhattan_data %>%
    left_join(max_pos, by = "chr") %>%
    mutate(position = position + max_pos) 

manhat_cols <- c("#1B96C6", "#456990", "#EEB868", "#EF767A", "#49DCB1")

x_labels <- manhattan_data %>% 
  group_by(chr) %>% 
  summarise(position = min(position) + (max(position) - min(position))/2)


p1 <- manhattan_data %>%
  ggplot(aes(position, -1 * log10(pvalue_female_early_raw), 
             group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  scale_colour_manual(values = manhat_cols) +
  scale_y_continuous(limits = c(0,8)) +
  scale_x_continuous(breaks = x_labels$position, labels = x_labels$chr) +
  ylab("") +  xlab("") +
  theme_bw() + 
  theme(legend.position = "none",
        text = element_text(family = "Raleway", size = 12),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())


p2 <- manhattan_data %>%
  ggplot(aes(position, -1 * log10(pvalue_male_early_raw), 
             group = chr, colour = chr, stroke = 0.2)) + 
  geom_point(size = 0.5) +
  scale_colour_manual(values = manhat_cols) +
  xlab("Genomic position") + ylab("") +
  scale_y_reverse(limits = c(8,0)) +
  theme_bw() + 
  theme(axis.text.x = element_blank(),
        legend.position = "none",
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        text = element_text(family = "Raleway", size = 12),
        panel.border = element_blank(),
        axis.ticks.x = element_blank())

left_label <- text_grob(expression(Effect~on~`early-life`~fitness~(-Log[10]~p)), 
                        size = 12, family = "Raleway", rot = 90)

combined_manhattan <- arrangeGrob(
  arrangeGrob(p1, p2, nrow = 2, 
              left = left_label))

png("figures/fig2_SNPs_manhattan_plot.png", 
    height = 8, width = 8, res = 300, units = "in")
grid.arrange(fig2_top, combined_manhattan)
invisible(dev.off())

grid.arrange(fig2_top, combined_manhattan)
```

***Figure 2***: Panels A and B show the mashr-adjusted effect sizes of 1,207,357 polymorphic loci on male and female early- and late-life fitness. The data have been binned into hexagons, with the colour and contour lines indicating the number of loci in each bin. Positive effect sizes indicate that the minor allele is associated with higher fitness and the major allele with lower fitness. Panel C shows a pair of Manhattan plots, showing the chromosomal position and $-Log_{10}$ $p$-value (from linear mixed model GWAS using GEMMA) for each locus's effect on female (top) and male (bottom) early-life fitness.

## Investigating pleiotropy and polygenicity of fitness

Inspired by Boyle et al. 2017 (_Cell_, specifically the analysis in their Figure 1C), we sorted all the variants in order of effect size on female fitness, placed the variants in bins of 1000, and then calculated the average effect size for each bin for both male and female fitness. This analysis was performed on the raw SNP effect sizes from GEMMA, pruned to a set of `r sum(LD_subset)` SNPs in approximate LD with one another. 

If there is pleiotropy between male and female fitness, we predict a correlation between the effect sizes for male and female fitness; on the contrary if there were no pleiotropy, we would see no correlation. Moreover, if fitness is highly polygenic ('omnigenic'; Boyle et al. 2017), we predict a tight, straight line relationship, because each bin would contain some variants with small but genuine associations with fitness, and these effects would replicate in the other fitness trait. On the contrary, if genetic variance in fitness stems from just a few genes with larger effects (with most loci having no true effect on fitness), the relationship between male and female fitness would be flat in the center and sloped at the extremes.

Figure 3 shows that there was a very tight correlation between the average effects of the variants in each bin on male and female fitness. The data therefore suggest that variants associated with male fitness tend to also affect female fitness (in the same direction), and that a very large number of loci have small, concordant effects on fitness in both sexes. 

Interestingly there appears to be a curve in Figure 3A to the left of the x-axis, indicating that variants where the minor allele is strongly, negatively associated with female fitness are (on average) less negatively associated with male fitness than expected based on predictions from variants with weaker effects on female fitness. One possible explanation is that alleles that are highly highly detrimental to both sexes are usually purged by selection (or at least kept below the 5% MAF threshold that we used), whereas female-harming alleles that are neutral or beneficial in males are purged less often, resulting a greater proportion of female-limited or sexually antagonistic alleles towards the left of the x-axis.

```{r boyle_plot, message=FALSE}
p1_data <- SNP_clumps %>%
  filter(LD_subset) %>% # The figure looks the same whether or not the data are thinned to the LD set of SNPs
  arrange(beta_female_early_raw) %>%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         facet = "A. Early-life fitness") %>%
  group_by(bin, facet) %>%
  summarise(females = mean(beta_female_early_raw), males = mean(beta_male_early_raw)) 

p2_data <- SNP_clumps %>%
  filter(LD_subset) %>% 
  arrange(beta_female_late_raw) %>%
  mutate(bin = c(rep(1:floor(n()/1000), each = 1000), 
                 rep(floor(n()/1000) + 1, each = n() %% 1000)),
         facet = "B. Late-life fitness") %>%
  group_by(bin, facet) %>%
  summarise(females = mean(beta_female_late_raw), males = mean(beta_male_late_raw))

boyle_plot <- bind_rows(p1_data, p2_data) %>% 
  ggplot(aes(females, males)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  geom_point(alpha = 0.8) +
  stat_smooth(method = "lm", formula = y ~ x + I(x^2), size = 0.6) +
  facet_wrap(~ facet) +
  xlab("Mean effect size on female fitness") + ylab("Mean effect size on male fitness") + 
  theme_bw() +
  theme(strip.background = element_blank(),
        strip.text = element_text(hjust=0)) + 
  theme(text = element_text(family = "Raleway", size = 12))

boyle_plot
```

***Figure 3***: Estimated mean effect size for groups of 1,000 variants, on male and female early-life (panel A) and late-life (panel B) fitness. The variant groups were created by sorting variants by their estimated effect size on female fitness, then dividing the sorted list into groups of 1,000. This analysis was performed on a pruned set of 208,987 variants in approximate linkage disequilibrium with one another. The observed positive relationships imply that large numbers of loci have small effects on the fitness of both sexes (see main text). The fit lines are from a quadratic linear regression, and the shaded area shows the standard error. 

```{r save_boyle_plot, echo=FALSE, fig.showtext=TRUE, message = FALSE}
pdf("figures/fig3_boyle_plot.pdf", height = 3.5, width = 6)
boyle_plot
invisible(dev.off())
```

## Correlations between mutation load and fitness

### Count all the mutations in each DGRP line

The following code counts the number of mutations per DGRP line which A) have a minor allele frequency of 0.05 or lower, and B) were classified by SnpEff as having an effect that we designated to be a "major effect", defined as an insertion, deletion, or nonsynonymous substitution inside a coding sequence (as well as gains and losses of start codons). We reasoned that such alleles are probably mostly deleterious. The following mutation types were classified as major effect mutations (see `SnpEff`'s classification scheme [here](https://pcingola.github.io/SnpEff/se_outputsummary/)):

- EXON_DELETED
- NON_SYNONYMOUS_CODING
- FRAME_SHIFT
- CODON_CHANGE
- CODON_INSERTION
- CODON_CHANGE_PLUS_CODON_INSERTION
- CODON_DELETION
- CODON_CHANGE_PLUS_CODON_DELETION
- STOP_GAINED
- STOP_LOST
- RARE_AMINO_ACID
- START_LOST
- START_GAINED


```{r count_mutations, message=FALSE, results='hide'}
# Load up the DGRP genotype data using the bigsnpr package
if(!file.exists("snp_backing_file.bk")) {
  rds <- snp_readBed("data/input/dgrp2.bed", backingfile = "snp_backing_file")
}
bed_file <- snp_attach("snp_backing_file.rds")
annot <- read.table("data/input/dgrp.fb557.annot.txt", 
                    header = FALSE, stringsAsFactors = FALSE)

# Use PLINK to count allele freqs in the whole DGRP
plink <- paste(getwd(), "code/plink", sep = "/")
run_command <- function(shell_command, wd = getwd(), path = ""){
  cat(system(glue("cd ", wd, path, "\n",shell_command), intern = TRUE), sep = '\n')
}
run_command("{plink} --bfile dgrp2 --freq", path = "/data/input")

# Get a list of variants that constitute a 'major mutation' (types as shown in code below)

major_effect_types <- c(
  "EXON_DELETED", "NON_SYNONYMOUS_CODING",
  "FRAME_SHIFT", "CODON_CHANGE",
  "CODON_INSERTION",
  "CODON_CHANGE_PLUS_CODON_INSERTION",
  "CODON_DELETION",
  "CODON_CHANGE_PLUS_CODON_DELETION",
  "STOP_GAINED",
  "STOP_LOST",
  "RARE_AMINO_ACID",
  "START_LOST",
  "START_GAINED"
)

all_major_mutations <- annot$V1[unique(c(
  which(str_detect(annot$V3, major_effect_types[1])),
  which(str_detect(annot$V3, major_effect_types[2])),
  which(str_detect(annot$V3, major_effect_types[3])),
  which(str_detect(annot$V3, major_effect_types[4])),
  which(str_detect(annot$V3, major_effect_types[5])),
  which(str_detect(annot$V3, major_effect_types[6])),
  which(str_detect(annot$V3, major_effect_types[7])),
  which(str_detect(annot$V3, major_effect_types[8])),
  which(str_detect(annot$V3, major_effect_types[9])),
  which(str_detect(annot$V3, major_effect_types[10])),
  which(str_detect(annot$V3, major_effect_types[11])),
  which(str_detect(annot$V3, major_effect_types[12])),
  which(str_detect(annot$V3, major_effect_types[13]))
))]

# Get a list of variants with 0 > MAF < 0.05
rare_alleles <- read.table("data/input/plink.frq", header = T) %>% 
  filter(MAF < 0.05 & MAF > 0) %>% pull(SNP)

# Get the indexes of variants that are major mutations and also have MAF < 0.05
indexes <- intersect(
  which(bed_file$map$marker.ID %in% all_major_mutations), 
  which(bed_file$map$marker.ID %in% rare_alleles)
)

# Function to count the number of mutations in all 205 DGRP lines
count_mutations <- function(indexes){
  tibble(
    line = bed_file$fam$family.ID,
    mutation_load = sapply(
      1:205, function(i) sum(bed_file$genotypes[i, ][indexes] == 2, na.rm = T))
  )
}

mutation_load <- count_mutations(indexes)

# Join the mutation counts with our line mean fitness data
joined <- left_join(
  predicted_line_means %>% 
    select(-block) %>% 
    gather(fitness_trait, fitness_value, -line),
  mutation_load, by = join_by(line)) %>% 
  mutate(fitness_trait = str_to_sentence(str_replace_all(fitness_trait, "[.]", " ")))
```

### Run Bayesian multivariate models

The following Bayesian multivariate model estiamtes the relationship between mutation count and each of the four fitness traits, across DGRP lines. The model has the multivariate formula `(FE, FL, ME, ML) ~ mutation_count`, assumes Gaussian errors, and estimates the residual correlation between the four fitness traits (thereby accounting for the correlation between fitness traits when estimating the relationship with mutation count). The model uses `brms` default priors. 
```{r message=F, warning=F, results='hide'}
brms_data_muts <- joined %>% 
  mutate(fitness_trait = str_replace_all(fitness_trait, " ", "_")) %>% 
  spread(fitness_trait, fitness_value) %>% 
  mutate(mutation_count = mutation_load / 100) %>% select(-mutation_load)

mut_model <- brm(bf(
  mvbind(Female_fitness_early, 
         Female_fitness_late, 
         Male_fitness_early, 
         Male_fitness_late) ~ 
    mutation_count, sigma ~ 0) + set_rescor(TRUE), 
  data = brms_data_muts,
  warmup = 2000, iter = 8000, chains = 4, cores = 4) 
```

#### Inspect model summary

```{r comment=''}
summary(mut_model)
```


#### Model checks (posterior predictive plots) 

Here, we plot the distribution of the model's posterior predictions, for each of the four response variables, for 50 samples from the posterior. If the model's assumptions and prior are suitable, there should be a good match between the posterior predictions (shown in blue; one line per sample from the posterior) and the original data (shown in black). See `?pp_check`.

```{r}
do_ppcheck <- function(rr, title){
    pp_check(mut_model, resp = rr, ndraws = 50) + 
    scale_y_continuous(limits = c(0, 0.55)) +
    labs(subtitle=title) + 
    theme_bw() + 
    theme(legend.position = "none") 
}

grid.arrange(
  do_ppcheck("Femalefitnessearly", "Female fitness early"),
  do_ppcheck("Femalefitnesslate", "Female fitness late"),
  do_ppcheck("Malefitnessearly", "Male fitness early"),
  do_ppcheck("Malefitnesslate", "Male fitness late"))
```


### Estimated effect size of mutation load on fitness {.tabset}

```{r fig.showtext=TRUE}
mut_plot_data <- conditional_effects(mut_model, plot = F)

get_regression_line_data <- function(plot_data){
  regression_line_rare_muts <- bind_rows(
    plot_data[[1]] %>% 
      select(mutation_count, estimate__, lower__, upper__) %>% 
      mutate(trait = "A. Female early-life fitness"),
    plot_data[[2]] %>% 
      select(mutation_count, estimate__, lower__, upper__) %>% 
      mutate(trait = "B. Female late-life fitness"),
    plot_data[[3]] %>% 
      select(mutation_count, estimate__, lower__, upper__) %>% 
      mutate(trait = "C. Male early-life fitness"),
    plot_data[[4]] %>% 
      select(mutation_count, estimate__, lower__, upper__) %>% 
      mutate(trait = "D. Male late-life fitness")) %>% 
    mutate(mut_type = "Load of mutations with MAF < 0.05") %>% 
    as_tibble()
}

get_points_data <- function(brms_data){
  brms_data %>% select(-line) %>% 
  rename(`A. Female early-life fitness` = Female_fitness_early,
         `B. Female late-life fitness` = Female_fitness_late,
         `C. Male early-life fitness` = Male_fitness_early,
         `D. Male late-life fitness` = Male_fitness_late) %>% 
  gather(trait, estimate__, -mutation_count)
}


mut_load_correlation_plot <- get_regression_line_data(mut_plot_data) %>% 
  ggplot(aes(x = mutation_count, y = estimate__)) + 
  geom_ribbon(aes(ymin = lower__, ymax = upper__, linetype = NA, fill = trait)) +
  geom_line(lwd = 0.5) +
  geom_point(data = get_points_data(brms_data_muts), alpha = 0.7, size = 0.7) +
  labs(x = "Mutation load (100s)", 
       y = "Line mean fitness") + 
  facet_wrap(~trait, nrow = 2) +
  scale_fill_manual(values = c("pink", "deeppink2", "lightblue", "steelblue")) + 
  theme_bw() + 
  theme(text = element_text(family = "Raleway", size = 11)) +
  theme(plot.title = element_text(hjust = 0), 
        strip.text = element_text(hjust = 0),
        strip.background = element_blank(),
        legend.position = "none")
```

#### Plot

***Figure 4***: Panels A-D show the relationship across DGRP lines between mutation load and line mean fitness; the regression lines are from a Bayesian multivariate model that accounts for the covariance in line mean fitness. Panel E shows the posterior estimates of the four regression slopes (i.e. the effect size of 100 mutations on fitness, where fitness is measured in standard units on the scale of the linear predictor), with the black bars summarising the median and 66% and 95% credible intervals. Panel F shows the posterior estimates of the differences in this effect size between pairs of fitness traits. 

```{r fig.showtext=TRUE, fig.width=9.55, fig.height=4.7, message=F}
posterior_muts <- as_draws_df(mut_model, variable = "^b_", regex = T) %>% 
  as_tibble() %>% select(contains("mutation"))

make_plot1 <- function(posterior, title){
  tibble(
    `Female\nearly-life\n(FE)` = unlist(posterior[,1]), 
    `Female\nlate-life\n(FL)` = unlist(posterior[,2]),  
    `Male\nearly-life\n(ME)` = unlist(posterior[,3]),  
    `Male\nlate-life\n(ML)` = unlist(posterior[,4])) %>% 
    gather() %>% 
    mutate(key = factor(key, rev(unique(key))),
           x = title) %>% 
    ggplot(aes(value, key)) + 
    geom_vline(xintercept = 0, linetype = 2) +
    stat_halfeye(aes(fill = key)) + 
    facet_wrap(~ x) +
    labs(x = "Change per 100 mutations", y = NULL) +
    scale_fill_manual(values = rev(c("pink", "deeppink2", "lightblue", "steelblue"))) + 
    theme_bw() + 
    theme(text = element_text(family = "Raleway", size = 11)) +
    theme(plot.title = element_text(hjust = 0), 
          strip.text = element_text(hjust = 0),
          strip.background = element_blank(),
          panel.background = element_blank(),
          legend.position = "none")
}

make_plot2 <- function(posterior, title, col){
  tibble(
    `ME − FE` = unlist(posterior[,3] - posterior[,1]), 
    `ML − FE` = unlist(posterior[,4] - posterior[,1]), 
    `ME − FL` = unlist(posterior[,3] - posterior[,2]), 
    `ML − FL` = unlist(posterior[,4] - posterior[,2]), 
    `FE − FL` = unlist(posterior[,1] - posterior[,2]), 
    `ME − ML` = unlist(posterior[,3] - posterior[,4]) ) %>% 
    gather() %>% 
    mutate(key = factor(key, rev(c("FE − FL", "ME − ML",
                                   "ME − FE", "ML − FE",
                                   "ME − FL", "ML − FL"))),
           x = title) %>% 
    ggplot(aes(value, key)) + 
    geom_vline(xintercept = 0, linetype = 2) +
    stat_halfeye(fill = col) + labs(x = "Effect size difference", y = NULL) +
    facet_wrap(~ x) +
    
    theme_bw() + 
    theme(text = element_text(family = "Raleway", size = 11)) +
    theme(plot.title = element_text(hjust = 0), 
          strip.text = element_text(hjust = 0),
          panel.background = element_blank(),
          strip.background = element_blank())
}

pp1 <- make_plot1(posterior_muts, "E. Effect sizes (posterior)")
pp2 <- make_plot2(posterior_muts, "F. Differences in effect size", col = "#F5CA7B")


mut_load_correlation_plot + pp1 + pp2 + 
  plot_layout(ncol = 3, widths = c(1, 0.5,0.5))

pdf("figures/fig4_mutation_load.pdf", height = 4.7, width = 9.55)
mut_load_correlation_plot + pp1 + pp2 + 
  plot_layout(ncol = 3, widths = c(1, 0.5,0.5))
invisible(dev.off())

# Clean up files:
unlink(list("snp_backing_file.rds", 
            "snp_backing_file.bk", 
            "data/input/plink.log",
            "data/input/plink.frq"))
```


#### Table

The table shows the medians and 95% CIs of the posterior effect size estimates shown in panels E and F of the figure. 

```{r}
make_posterior_table <- function(posterior){
  as_tibble(rbind(
    posterior_summary(posterior[,1]),
    posterior_summary(posterior[,2]),
    posterior_summary(posterior[,3]),
    posterior_summary(posterior[,4]),
    posterior_summary(posterior[,1] - posterior[,2]), 
    posterior_summary(posterior[,3] - posterior[,4]), 
    posterior_summary(posterior[,3] - posterior[,1]), 
    posterior_summary(posterior[,4] - posterior[,1]), 
    posterior_summary(posterior[,3] - posterior[,2]), 
    posterior_summary(posterior[,4] - posterior[,2]))) %>% 
    mutate(`Effect size or effect size difference` = c(
      "Female early-life (FE)", "Female late-life (FL)",
      "Male late-life (ME)", "Male late-life (ML)",
      "FE - FL", "ME - ML",
      "ME - FE", "ML - FE", 
      "ME - FL", "ML - FL"), .before = 1) %>% 
    rename("Error" = `Est.Error`,
           `Lower 95% CI` = Q2.5,
           `Upper 95% CI` = Q97.5,)
  
}

saveRDS(make_posterior_table(posterior_muts), 
        "data/derived/supp_table_mutation_load_effects.rds")

make_posterior_table(posterior_muts) %>% 
  kable(digits = 3) %>% 
  kable_styling(full_width = F)
```


## Frequencies of antagonistic loci and transcripts

The following figures show the proportions of loci/transcripts whose effect on female fitness (or early-life fitness) is similar or different to their effect on male fitness (or late-life fitness). The figures were made by first placing variant or transcript effect sizes (for female or early-life fitness) into quartiles, which we label as negative effects, weak negative effects, weak positive effects, and positive effects (taking advantage of the fact that the median effect size is very close to zero). We then do the same for male or late-life fitness, and plot the number of transcripts showing each combination of quartiles, giving an indication of how many loci/transcripts have aligned or opposing effects on the different fitness metrics.

The following plots were made after processing the effect sizes with `mashr`, which should reduce the number of false signals. Because there is a positive correlation between the sexes and age classes, the shrinkage applied by `mashr` reduces the number of loci/transcripts that appear to show antagonism between ages and sexes (this should control the number of 'false positive' antagonistic loci/transcripts).



### Frequencies of antagonistic loci 

#### Comparing effects on the sexes {.tabset}

Rather few loci are in the 1st quartile in their effect on male fitness and the 4th quartile for their effect on female fitness, or _vice versa_, suggesting that variants with strongly sex-opposite effects on fitness are rare

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 8}
SNP_effects <- SNP_clumps %>%
  filter(LD_subset) %>% 
  select(contains("beta")) %>% 
  select(contains("ED")) # this can be changed to "raw" to see the non-mashr effect sizes (i.e. statistical noise)
names(SNP_effects)[grepl("female_early", names(SNP_effects))] <- "FE"
names(SNP_effects)[grepl("female_late", names(SNP_effects))] <- "FL"
names(SNP_effects)[grepl("male_early", names(SNP_effects))] <- "ME"
names(SNP_effects)[grepl("male_late", names(SNP_effects))] <- "ML"

temp <- SNP_effects %>% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %>% 
  mutate_at(vars(starts_with("quartile")), ~ {
    SNP_effects <- .x
    SNP_effects[SNP_effects == 1] <- "Negative"
    SNP_effects[SNP_effects == 2] <- "Weakly\nnegative"
    SNP_effects[SNP_effects == 3] <- "Weakly\npositive"
    SNP_effects[SNP_effects == 4] <- "Positive"
    factor(SNP_effects, c("Negative", "Weakly\nnegative", "Weakly\npositive", "Positive"))})

mat_list <- list(table(temp$quartile_FE, temp$quartile_ME),
                 table(temp$quartile_FL, temp$quartile_ML),
                 table(temp$quartile_FE, temp$quartile_FL),
                 table(temp$quartile_ME, temp$quartile_ML))  %>% 
   map(~ as.data.frame(.x)) 
names(mat_list) <- c("FE, ME", "FL, ML", "FE, FL", "ME, ML")
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

cols <- c(brewer.pal(9, "Reds")[c(6,3)], brewer.pal(9, "Blues")[c(3,6)])

focal_dat <- rbind(
    as.data.frame(mat_list[[1]]) %>% 
      mutate(pp = "A. Early-life fitness"), 
    as.data.frame(mat_list[[2]]) %>% 
      mutate(pp = "B. Late-life fitness")) 

labs <- c("Negative", "Weakly negative", "Weakly positive", "Positive")

fig_5_top_row <- focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nmale fitness",
                    labels = labs) + 
  xlab("Association with female fitness") + 
  ylab("Number of variants") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        text = element_text(family = "Raleway", size = 12),
        strip.text = element_text(hjust = 0))

# To count the total number of transcripts tested:
# focal_dat$Freq[focal_dat$pp == "C. Early-life fitness"] %>% sum()

fig_5_top_row
```

##### Table

```{r}
supp_tabl_gwas_intersex <- focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given association with female fitness)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of variants` = Freq) 

saveRDS(supp_tabl_gwas_intersex, "data/derived/supp_tabl_gwas_intersex.rds")

supp_tabl_gwas_intersex %>% 
  kable() %>% kable_styling(full_width = FALSE)
```


#### Comparing effects on the age classes {.tabset}

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 8}
focal_dat <- rbind(
  as.data.frame(mat_list[[3]]) %>% 
    mutate(pp = "A. Female fitness"), 
  as.data.frame(mat_list[[4]]) %>% 
    mutate(pp = "B. Male fitness")) 

focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nlate-life fitness",
                    labels = labs) + 
  xlab("Association with early-life fitness") + 
  ylab("Number of variants") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = "Raleway", size = 12))
```

##### Table

```{r}
supp_tabl_gwas_interage <- focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given association with early-life fitness)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of variants` = Freq)

saveRDS(supp_tabl_gwas_interage, "data/derived/supp_tabl_gwas_interage.rds")

supp_tabl_gwas_interage %>% 
  kable() %>% kable_styling(full_width = FALSE)
```


#### Run a $\chi^2$ test

The following $\chi^2$ test examines the null hypothesis that the proportion of candidate sexually antagonistic loci (i.e. those with a quartile 1 effect on fitness of sex $i$ and a quartile 4 effect on sex $j$) is equal when fitness associations are calculated using the early- and late-life fitness data. This null hypothesis is rejected: the % of SA loci is `r round(100 *(129+103) / 208987, 3)`% at early life and `r round(100 *(93+48) / 208987, 3)`% at late life, which is a `r round(0.1110117 / 0.06746831, 1)`-fold difference.

```{r}
chisq_table <- as.table(rbind(c(129+103, 208987 - (129+103)), c(93+48, 208987 - (93+48))))
dimnames(chisq_table) <- list(life_stage = c("Early_life", "Late_life"),
                              antagonism_status = c("Sex_antag","Non_sex_antag"))
chisq_table
chisq.test(chisq_table)
```


### Frequencies of antagonistic transcripts

#### Comparing effects on the sexes {.tabset}

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 8}
TWAS_ED <- readRDS("data/derived/TWAS/TWAS_ED.rds")

binned_transcripts <- data.frame(
  FBID = read_csv("data/derived/TWAS/TWAS_results.csv")$FBID,
  as.data.frame(get_pm(TWAS_ED))) %>% 
  as_tibble() %>% rename_all(~ str_remove_all(.x, "beta_")) %>% 
  mutate(quartile_FE = ntile(FE, 4),
         quartile_ME = ntile(ME, 4),
         quartile_FL = ntile(FL, 4),
         quartile_ML = ntile(ML, 4)) %>% 
  mutate_at(vars(starts_with("quartile")), ~ {
    SNP_effects <- .x
    SNP_effects[SNP_effects == 1] <- "Negative"
    SNP_effects[SNP_effects == 2] <- "Weakly\nnegative"
    SNP_effects[SNP_effects == 3] <- "Weakly\npositive"
    SNP_effects[SNP_effects == 4] <- "Positive"
    factor(SNP_effects, c("Negative", "Weakly\nnegative", "Weakly\npositive", "Positive"))})


mat_list <- list(table(binned_transcripts$quartile_FE, binned_transcripts$quartile_ME),
                 table(binned_transcripts$quartile_FL, binned_transcripts$quartile_ML),
                 table(binned_transcripts$quartile_FE, binned_transcripts$quartile_FL),
                 table(binned_transcripts$quartile_ME, binned_transcripts$quartile_ML))  %>% 
   map(~ as.data.frame(.x)) 
names(mat_list) <- c("FE, ME", "FL, ML", "FE, FL", "ME, ML")
# var1 is rows (first one, e.g. FE), 2 is cols (second one, e.g. ME)

focal_dat <- rbind(
    as.data.frame(mat_list[[1]]) %>% 
      mutate(pp = "C. Early-life fitness"), 
    as.data.frame(mat_list[[2]]) %>% 
      mutate(pp = "D. Late-life fitness")) 

# To count the total number of transcripts tested, 14286:
# focal_dat$Freq[focal_dat$pp == "C. Early-life fitness"] %>% sum()

fig_5_bottom_row <- focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nmale fitness",
                    labels = labs) + 
  xlab("Association with female fitness") + 
  ylab("Number of transcripts") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        text = element_text(family = "Raleway", size = 12),
        strip.text = element_text(hjust = 0))
        # strip.text = element_text(hjust = 0, size = 12))

fig_5_bottom_row
```


```{r fig.showtext=T, echo = FALSE, results='hide'}
# Code to make the composite Figure 4, the stacked bar chart
grid_arrange_shared_legend <- function(..., ncol = 1, nrow = 2, position = c("bottom", "right")) {

  plots <- list(...)
  position <- match.arg(position)
  g <- ggplotGrob(plots[[1]] + theme(legend.position = position))$grobs
  legend <- g[[which(sapply(g, function(x) x$name) == "guide-box")]]
  lheight <- sum(legend$height)
  lwidth <- sum(legend$width)
  gl <- lapply(plots, function(x) x + theme(legend.position="none"))
  gl <- c(gl, ncol = ncol, nrow = nrow)

  combined <- switch(position,
                     "bottom" = arrangeGrob(do.call(arrangeGrob, gl),
                                            legend,
                                            ncol = 1,
                                            heights = unit.c(unit(1, "npc") - lheight, lheight)),
                     "right" = arrangeGrob(do.call(arrangeGrob, gl),
                                           legend,
                                           ncol = 2,
                                           widths = unit.c(unit(1, "npc") - lwidth, lwidth)))
  
  grid.newpage()
  grid.draw(combined)

  # return gtable invisibly
  invisible(combined)

}

pdf("figures/fig5_quartiles_plot.pdf", height = 6, width = 7)
grid_arrange_shared_legend(fig_5_top_row + 
                             theme(axis.text.x = element_blank(),
                                   axis.ticks.x = element_blank()) +
                             xlab(" "), 
                           fig_5_bottom_row, position = "right")
dev.off()
remove_pages(1, "figures/fig5_quartiles_plot.pdf", "figures/fig5_quartiles_plot.pdf")
```


##### Table

```{r}
supp_tabl_twas_intersex <- focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given association with female fitness)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Age class` = pp, `Association with female fitness` = Var1, 
         `Association with male fitness` = Var2, 
         `Number of transcripts` = Freq)

saveRDS(supp_tabl_twas_intersex, "data/derived/supp_tabl_twas_intersex.rds")

supp_tabl_twas_intersex %>% 
  kable() %>% kable_styling(full_width = FALSE)
```

#### Run a $\chi^2$ test

The following $\chi^2$ test examines the null hypothesis that the proportion of candidate sexually antagonistic transcripts (i.e. those with a quartile 1 effect on fitness of sex $i$ and a quartile 4 effect on sex $j$) is equal when fitness associations are calculated using the early- and late-life fitness data. This null hypothesis is not rejected: the % of SA transcripts is `r round(100 *(691+689) / 14286, 2)`% at early life and `r round(100 *(677 + 664) / 14286, 2)`% at late life, which is a `r round(9.659807 / 9.386812, 2)`-fold difference.

```{r}
chisq_table <- as.table(rbind(c(691+689, 14286 - (691+689)), c(677 + 664, 14286 - (677 + 664))))
dimnames(chisq_table) <- list(life_stage = c("Early_life", "Late_life"),
                              antagonism_status = c("Sex_antag","Non_sex_antag"))
chisq_table
chisq.test(chisq_table)
```


#### Comparing effects on the age classes {.tabset}

##### Figure

```{r fig.showtext=TRUE, fig.height=5, fig.width = 11}
focal_dat <- rbind(
  as.data.frame(mat_list[[3]]) %>% 
    mutate(pp = "A. Female fitness"), 
  as.data.frame(mat_list[[4]]) %>% 
    mutate(pp = "B. Male fitness"))

focal_dat %>% 
  ggplot(aes(Var1, Freq, fill = Var2)) + 
  facet_wrap(~ pp) + 
  geom_bar(stat="identity", colour = "grey20") +
  scale_fill_manual(values = cols, name = "Association with\nlate-life fitness",
                    labels = labs) + 
  xlab("Association with early-life fitness") + 
  ylab("Number of transcripts") + 
  theme_bw() + 
  theme(panel.grid.major.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(hjust = 0, family = "Raleway", size = 12))
```

##### Table

```{r}
supp_tabl_twas_interage <- focal_dat %>% 
  split(.$pp) %>% 
  map_df(~ .x %>% mutate(`Percentage (overall)` = round(100 * Freq / sum(Freq), 2))) %>% 
  split(paste((.$Var1), .$pp)) %>% 
  map_df(~ .x %>% mutate(`Percentage (given association with early-life fitness)` = round(100 * Freq / sum(Freq), 2))) %>% 
  arrange(pp, Var1, Var2) %>% 
  select(pp, everything()) %>% 
  rename(`Sex` = pp, `Association with early-life fitness` = Var1, 
         `Association with late-life fitness` = Var2, 
         `Number of transcripts` = Freq)

saveRDS(supp_tabl_twas_interage, "data/derived/supp_tabl_twas_interage.rds")

supp_tabl_twas_interage %>% 
  kable() %>% kable_styling(full_width = FALSE)
```


## Plotting and modelling the evidence for antagonism

### Calculate the evidence ratios

#### For the GWAS results
```{r}
get_antagonism_ratios <- function(dat){
  dat %>%
    
    # Convert the LFSR to the probability that the effect size is positive
    mutate(pp_female_early = ifelse(pheno1 > 0, LFSR1, 1 - LFSR1),
           pp_female_late  = ifelse(pheno2 > 0, LFSR2, 1 - LFSR2),
           pp_male_early   = ifelse(pheno3 > 0, LFSR3, 1 - LFSR3),
           pp_male_late    = ifelse(pheno4 > 0, LFSR4, 1 - LFSR4)) %>%
    
    # Calculate the probabilities that beta_i and beta_j have the same/opposite signs
    mutate(p_sex_concord_early = pp_female_early * pp_male_early + 
             (1 - pp_female_early) * (1 - pp_male_early),
           p_sex_antag_early = pp_female_early * (1 - pp_male_early) + 
             (1 - pp_female_early) * pp_male_early,
           p_sex_concord_late  = pp_female_late * pp_male_late + 
             (1 - pp_female_late) * (1 - pp_male_late),
           p_sex_antag_late = pp_female_late * (1 - pp_male_late) + 
             (1 - pp_female_late) * pp_male_late,
           p_age_concord_females = pp_female_early * pp_female_late + 
             (1 - pp_female_early) * (1 - pp_female_late),
           p_age_antag_females = pp_female_early * (1 - pp_female_late) + 
             (1 - pp_female_early) * pp_female_late,
           p_age_concord_males = pp_male_early * pp_male_late + (1 - pp_male_early) * (1 - pp_male_late),
           p_age_antag_males = pp_male_early * (1 - pp_male_late) + (1 - pp_male_early) * pp_male_late) %>%
    
    # Find the ratios of some of these two probabilities (i.e. the "evidence ratios")
    mutate(inter_sex_early = p_sex_concord_early / p_sex_antag_early,
           inter_sex_late = p_sex_concord_late / p_sex_antag_late,
           inter_age_females = p_age_concord_females / p_age_antag_females,
           inter_age_males = p_age_concord_males / p_age_antag_males) 
}

antagonism_evidence_ratios_gwas <- all_snps %>%
  rename(pheno1 = beta_female_early_mashr_ED, 
         pheno2 = beta_female_late_mashr_ED, 
         pheno3 = beta_male_early_mashr_ED, 
         pheno4 = beta_male_late_mashr_ED,
         LFSR1 = LFSR_female_early_mashr_ED, 
         LFSR2 = LFSR_female_late_mashr_ED, 
         LFSR3 = LFSR_male_early_mashr_ED, 
         LFSR4 = LFSR_male_late_mashr_ED) %>% 
  select(SNP, SNP_clump, starts_with("pheno"), LFSR1, LFSR2, LFSR3, LFSR4) %>% 
  filter(!is.na(pheno1)) %>% 
  collect(n=Inf) %>% 
  distinct() %>% 
  get_antagonism_ratios() %>%
  select(SNP, SNP_clump, starts_with("inter")) 
```

#### For the TWAS results
```{r get_TWAS_ERs}
TWAS_ED <- readRDS("data/derived/TWAS/TWAS_ED.rds")

twas <- data.frame(
  FBID = read_csv("data/derived/TWAS/TWAS_results.csv")$FBID,
      as.data.frame(get_pm(TWAS_ED)),
      as.data.frame(get_lfsr(TWAS_ED)) %>% 
        rename_all(~str_replace_all(., "beta", "LFSR"))) %>% 
      as_tibble() %>%
  left_join(tbl(db, "genes") %>% select(FBID, chromosome) %>% collect(), by = "FBID") %>%
  left_join(read_csv("data/derived/gene_expression_by_sex.csv"), by = "FBID") %>%
  filter(chromosome %in% c("2L", "2R", "3L", "3R", "X")) %>%
  rename(pheno1 = beta_FE, 
         pheno2 = beta_FL, 
         pheno3 = beta_ME, 
         pheno4 = beta_ML,
         LFSR1 = LFSR_FE, 
         LFSR2 = LFSR_FL, 
         LFSR3 = LFSR_ME, 
         LFSR4 = LFSR_ML)

antagonism_evidence_ratios_twas <- twas %>% 
  get_antagonism_ratios() %>%
  select(FBID, chromosome, male_bias_in_expression, AveExpr, starts_with("inter")) %>% 
  mutate(chromosome = relevel(factor(chromosome), ref = "X"))
```

### Plot the evidence ratios

```{r evidence_ratio_plot, fig.showtext=TRUE, message=FALSE}
make_evidence_ratio_plot <- function(ERs, ymax, ylab){
  # Argument needs to be a dataframe of loci or transcripts, each identified by a column called "identifier"
  # There should be a col called "evidence_ratio", calculated from the LFSR from ED mashr. The "trait" col says which type of ER it is.
  
  ERs$trait[ERs$trait == "inter_sex_early"] <- "Inter-sex (early life)"
  ERs$trait[ERs$trait == "inter_sex_late"] <- "Inter-sex (late life)"
  ERs$trait[ERs$trait == "inter_age_females"] <- "Inter-age (females)"
  ERs$trait[ERs$trait == "inter_age_males"] <- "Inter-age (males)"
  
  antagonism_evidence_ratios <- ERs %>%
    mutate(trait = factor(trait, c("Inter-sex (early life)",
                                   "Inter-sex (late life)",
                                   "Inter-age (females)",
                                   "Inter-age (males)")))
  
  antagonism_evidence_ratios %>%
    ggplot(aes(log2(evidence_ratio))) + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio < 1), 
                   bins = 500, fill = "#FF635C") + 
    geom_histogram(data=subset(antagonism_evidence_ratios, evidence_ratio > 1), 
                   bins = 500, fill = "#5B8AFD") +
    coord_cartesian(xlim = c(-10, 10), ylim = c(0, ymax)) + 
    scale_x_continuous(breaks = c(-10,  -6,  -2,   2,   6,  10), 
                       labels = c(paste("1/",2 ^ c(10,  6,  2), sep = ""), 2 ^ c(2,6,10))) + 
    facet_wrap(~ trait) + 
    xlab("Evidence ratio (log2 scale)") + ylab(ylab) + 
    theme_bw() + 
    theme(panel.border = element_rect(size = 0.8),
          text = element_text(family = "Raleway", size = 12),
          axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
          strip.background = element_blank())
}

GWAS_ratios_plot <- antagonism_evidence_ratios_gwas %>% 
  select(SNP_clump,  starts_with("inter")) %>%
  distinct() %>% # Include SNPs in 100% LD a single time, such that these clumps are the unit of replication
  gather(trait, evidence_ratio, -SNP_clump) %>% 
  rename(identifier = SNP_clump) %>% 
  make_evidence_ratio_plot(ymax = 5000, ylab = "Number of loci")

# For counting the number of transcripts with 50x more evidence for antagonism 
# antagonism_evidence_ratios_twas %>% 
#   select(FBID,  starts_with("inter")) %>%
#   gather(trait, evidence_ratio, -FBID) %>% 
#   rename(identifier = FBID) %>% filter(evidence_ratio < 1/50) %>% summarise(n())

TWAS_ratios_plot <- antagonism_evidence_ratios_twas %>% 
  select(FBID,  starts_with("inter")) %>%
  gather(trait, evidence_ratio, -FBID) %>% 
  rename(identifier = FBID) %>% 
  make_evidence_ratio_plot(ymax = 500, ylab = "Number of transcripts")

pp <- plot_grid(GWAS_ratios_plot, TWAS_ratios_plot, 
                   labels = c('A', 'B'), label_size = 12)

ggsave("figures/fig6_antagonism_ratios.pdf", pp, width = 8.5, height = 4.9)

pp
```

***Figure 6***: Distribution of evidence ratios across loci (panel A) and transcripts (panel B), illustrating the strength of evidence for a  concordant relationship with fitness (evidence ratio > 1, blue), or an antagonistic relationship (<1, red). The top row of both panels considers the evidence for concordance/antagonism between the sexes (within each age class), while the bottom row shows concordance/antagonism between age classes (within each sex). The figure illustrates that many loci and transcripts (i.e. those with large evidence ratios) show strong evidence for concordant effects across sexes and age classes. By contrast, there are are relatively few candidate sexually antagonistic loci (i.e. those with evidence ratios well below 1), somewhat more sexually antagonistic transcripts, and essentially zero age antagonistic loci or transcripts.


### Model the evidence ratios

#### Inter-sex, early-life model (GWAS)

The initial full model contains the predictors `MAF`, `chromosome`, and `major_effect` (which records whether or not the focal variant causes an insertion, deletion, or nonsynonymous change in a coding sequence). `major_effect` was not significant, so the model was refitted without it.

```{r model_evidence_ratios}
# Annotate data with MAF, site, class, etc. 
dat <- left_join(antagonism_evidence_ratios_gwas,
          collect(tbl(db, "variants")), by = "SNP", multiple = "all") %>% 
  distinct() %>% rename(chromosome = chr)

# Remove chromosome 4 (too few sites)
dat <- dat %>% filter(chromosome != "4") 
dat$chromosome <- relevel(factor(dat$chromosome), ref = "X")

# Classify mutations as being major effect or not
dat <- dat %>% 
  mutate(major_effect = "No",
         major_effect = replace(major_effect, site.class %in% major_effect_types, "Yes"))

# For mutations with multiple site classes, record whether any site class is major or not,
# ensuring that there is one row in 'dat' for each SNP
dat <- dat %>% 
  group_by(SNP_clump) %>% 
  summarise(inter_sex_early = inter_sex_early[1],
            inter_sex_late = inter_sex_late[1],
            MAF = MAF[1],
            chromosome = chromosome[1],
            major_effect = ifelse(any(major_effect == "Yes"), "Yes", "No"))

n_loci <- prettyNum(nrow(dat), big.mark = ",", scientific = FALSE)

intersex_early_model_gwas <- glm(antagonistic ~ chromosome + MAF + major_effect, 
             data = dat %>% 
               mutate(antagonistic = as.numeric(inter_sex_early <= 
                        quantile(dat$inter_sex_early, probs = 0.01))), 
    family = "binomial")
car::Anova(intersex_early_model_gwas, type = 3)

summary(intersex_early_model_gwas)
```



#### Inter-sex, late-life model (GWAS)

The initial full model contains the predictors `MAF`, `chromosome`, and `major_effect` (which records whether or not the focal variant causes an insertion, deletion, or nonsynonymous change in a coding sequence). `major_effect` was not significant, so the model was refitted without it.


```{r ER_model1}
intersex_late_model_gwas <- glm(antagonistic ~ chromosome + MAF + major_effect, 
             data = dat %>% 
               mutate(antagonistic = inter_sex_late <= 
                        quantile(dat$inter_sex_late, probs = 0.01)), 
    family = "binomial")
car::Anova(intersex_late_model_gwas, type = 3)

summary(intersex_late_model_gwas)
```

#### Inter-sex, early-life model (TWAS)

The full model fitted here contains the predictors `chromosome`, `absolute_sex_bias_in_expression`, and `mean_expression_level`. Only the latter is significant, with highly-expressed transcripts being less likely to be sexually antagonistic.

```{r ER_model2}
n_transcripts <- prettyNum(nrow(antagonism_evidence_ratios_twas), 
                           big.mark = ",", scientific = FALSE)

twas_model_data <- antagonism_evidence_ratios_twas %>% 
  mutate(antagonistic = inter_sex_early <= 
           quantile(antagonism_evidence_ratios_twas$inter_sex_early, probs=0.01),
         absolute_sex_bias_in_expression = abs(male_bias_in_expression),
         mean_expression_level = AveExpr)


intersex_early_twas <- glm(
  antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level, 
  data = twas_model_data, 
  family = "binomial")
car::Anova(intersex_early_twas, type = 3)
summary(intersex_early_twas)
```

#### Inter-sex, late-life model (TWAS)

The full model fitted here contains the predictors `chromosome`, `absolute_sex_bias_in_expression`, and `mean_expression_level`. Only the latter is significant, with highly-expressed transcripts being less likely to be sexually antagonistic.

```{r ER_model3}
intersex_late_model_twas <- glm(
  antagonistic ~ chromosome + absolute_sex_bias_in_expression + mean_expression_level,
  data = antagonism_evidence_ratios_twas %>% 
    mutate(antagonistic = inter_sex_late <= 
             quantile(antagonism_evidence_ratios_twas$inter_sex_late, probs=0.01),
           absolute_sex_bias_in_expression = abs(male_bias_in_expression),
           mean_expression_level = AveExpr), 
  family = "binomial")

car::Anova(intersex_late_model_twas, type = 3)
summary(intersex_late_model_twas)
```

#### Plotting key results of these models

```{r ER_figure, fig.showtext=T, fig.height=7, fig.width=7} 
inv_logit <- function(x) exp(x)/(1+exp(x))

# new <- data.frame(MAF = seq(from = 0.05, to = 0.5, by = 0.01), chromosome = "X", major_effect = "No")
# preds <- data.frame(new, 
#                     prop = predict(intersex_early_model_gwas, 
#                                    newdata = new, 
#                                    se.fit=T, type = "link")) %>% 
#   mutate(y = prop.fit, 
#          lwr = prop.fit - 1.96*prop.se.fit, 
#          upr = prop.fit + 1.96*prop.se.fit,
#          y = inv_logit(y), 
#          lwr = inv_logit(lwr), 
#          upr = inv_logit(upr))

# p1 <- preds %>% 
#   mutate(panel = "A.") %>% 
#   ggplot(aes(MAF,  y)) + geom_line() + 
#   geom_hline(yintercept = .01, linetype = 3) + 
#   geom_line(aes(y = upr), linetype = 2) + 
#   geom_line(aes(y = lwr), linetype = 2) + 
#   scale_x_continuous(limits = c(0, 0.5)) + 
#   scale_y_continuous(limits = c(0, max(preds$y))) + 
#   facet_wrap( ~ panel) + 
#   theme_bw() + 
#   theme(text = element_text(family = "Raleway", size = 12)) +
#   theme(strip.background = element_blank(), 
#         strip.text = element_text(hjust = 0)) +
#   ylab("Probability of being in the top 1%\nsexually antagonistic loci (\u00B1 95% CIs)") +
#   xlab("Minor allele frequency")

new2 <- data.frame(chromosome = levels(dat$chromosome), MAF = 0.3, major_effect = "No")
inv_logit <- function(x) exp(x)/(1+exp(x))
preds2 <- data.frame(new2, prop = predict(intersex_early_model_gwas, newdata = new2, se.fit=T, type = "link")) %>% 
  mutate(y = prop.fit, 
         lwr = prop.fit - 1.96*prop.se.fit, 
         upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), 
         lwr = inv_logit(lwr), 
         upr = inv_logit(upr))

p2 <- preds2 %>% 
  mutate(panel = "A.") %>%
  ggplot(aes(chromosome,  y)) + 
  geom_point(size = 3) + 
  geom_errorbar(aes(ymin = lwr, ymax = upr), width = 0) + 
  facet_wrap( ~ panel) + 
  theme_bw() +
  theme(text = element_text(family = "Raleway", size = 12)) +
  theme(strip.background = element_blank(), strip.text = element_text(hjust = 0)) +
  ylab("Probability of being in the top 1%\nsexually antagonistic loci (\u00B1 95% CIs)") +
  xlab("Chromosome arm")

new <- data.frame(
  mean_expression_level = seq(
    from = min(twas_model_data$mean_expression_level), 
    to = max(twas_model_data$mean_expression_level), by = 0.01), 
  chromosome = "X", 
  absolute_sex_bias_in_expression = 0,
  mean_expression_level = mean(twas_model_data$absolute_sex_bias_in_expression))

preds <- data.frame(new, 
                    prop = predict(intersex_early_twas, 
                                   newdata = new, 
                                   se.fit=T, type = "link")) %>% 
  mutate(y = prop.fit, 
         lwr = prop.fit - 1.96*prop.se.fit, 
         upr = prop.fit + 1.96*prop.se.fit,
         y = inv_logit(y), 
         lwr = inv_logit(lwr), 
         upr = inv_logit(upr))

p3 <- preds %>% 
  mutate(panel = "B.") %>%
  ggplot(aes(mean_expression_level,  y)) + geom_line() + 
  geom_hline(yintercept = .01, linetype = 3) + 
  geom_line(aes(y = upr), linetype = 2) + 
  geom_line(aes(y = lwr), linetype = 2) + 
  ylab("Probability of being in the top 1%\nsexually antagonistic transcripts (\u00B1 95% CIs)") +
  xlab("Average expression level") + 
  facet_wrap( ~ panel) + 
  theme_bw() + 
  theme(text = element_text(family = "Raleway", size = 12)) +
  theme(strip.background = element_blank(), strip.text = element_text(hjust = 0)) 

grid.arrange(p2, p3, ncol = 2)

pdf("figures/fig7_models.pdf", height = 3.9, width = 7)
grid.arrange(p2, p3, ncol = 2)
invisible(dev.off())
# 
# grid.arrange(arrangeGrob(p1, p2, ncol = 2), 
#              arrangeGrob(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))
# 
# 
# blank <- ggplot() + theme_void()
# 
# grid.arrange(arrangeGrob(p1, p2, ncol = 2), 
#              arrangeGrob(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))
# 
# 
# pdf("figures/fig7_models.pdf", height = 7, width = 7)
# grid.arrange(arrangeGrob(p1, p2, ncol = 2), 
#              arrangeGrob(blank, p3, blank, nrow = 1, widths = c(0.2,0.6,0.2)))
# invisible(dev.off())
```

***Figure 7***: Panels A and B show the predicted effects of minor allele frequency and chromosome, respectively, on the probability that a locus is among the top 1% candidate sexually antagonistic loci, as ranked by their evidence ratio. Panel C shows the predicted effect of average gene expression level on the probability that a transcript is among the top 1% candidate sexually antagonistic transcript. The dashed lines and error bars show 95% confidence intervals, estimated as 1.96SE, and predictions are from binomial GLMs (one for panels A-B, another for Panel C).


### GO enrichment of transcripts with extreme evidence ratios {.tabset}

The following code performs a GO enrichment test using the Wilcoxon method, which tests for enrichment of GO terms at the top and the bottom of a gene list that has been ordered by some variable. In this case, the ordering variable is the evidence ratio for early-life sexual antagonism/concordance that is plotted in the top left of panel B in the previous figure. Transcripts with a more negative evidence ratio are more likely to be sexually antagonistic, while those with a more positive evidence ratio are more likely to be sexually concordant. The two tables below (accessible by clicking the tabs) illustrate the GO: Biological Process terms that are most associated with the most sexually antagonistic and sexually concordant transcripts, according to the Wilcoxon method. The `GOfuncR::go_enrich()` function uses permutation to calculate the 'family wise error rate' (FWER) to correct for multiple testing, and we define significantly enriched GO terms as those with FWER < 0.05. See the `GOfuncR` [vignette](https://bioconductor.org/packages/devel/bioc/vignettes/GOfuncR/inst/doc/GOfuncR.html) for more information on this method.


```{r GO_enrich}
library(GOfuncR) # BiocManager::install("GOfuncR")
library(org.Dm.eg.db) 
set.seed(12345)

# Get a list of Drosophila genes that have GO terms (can't analyse genes that do not)
genes_with_GO <- select(org.Dm.eg.db, 
                        keys = keys(org.Dm.eg.db), 
                        columns = c("SYMBOL","GO")) %>% 
  filter(GO != "<NA>") %>% dplyr::pull(SYMBOL)

# This code gets the gene symbols for each transcript and sets up the data in the format required by GOfuncR
# for the Wilcoxon test, which works on a gene list that is ordered by a variable 
# (here, the variable is the evidence ratio for early-life sexual antagonism/concordance)
wilcoxon_input <- antagonism_evidence_ratios_twas %>% 
  dplyr::select(FLYBASE = FBID, gene_score = inter_sex_early) %>% 
  left_join(select(org.Dm.eg.db, 
                   keys = keys(org.Dm.eg.db), 
                   columns = c("SYMBOL","FLYBASE")), by = "FLYBASE") %>% 
  mutate(gene_score = log2(gene_score)) %>% 
  dplyr::select(SYMBOL, gene_score) %>% 
  dplyr::filter(SYMBOL %in% genes_with_GO) %>% 
  dplyr::arrange(gene_score) %>% as.data.frame()

# Run the GO enrichment test
GO_results_wilcoxon <- go_enrich(wilcoxon_input, orgDb='org.Dm.eg.db', test='wilcoxon', 
                                 silent = T, domains = "biological_process",
                                 n_randsets = 1000)


most_antagonistic_genes_GO <- GO_results_wilcoxon$results %>% # most antagonistic
  arrange(FWER_low_rank) %>% 
  filter(raw_p_low_rank < 0.001) %>% 
  dplyr::select(node_id, node_name, raw_p_low_rank, FWER_low_rank) %>% 
  dplyr::rename(GO = node_id, Meaning = node_name, `log10 p` = raw_p_low_rank, FWER = FWER_low_rank) %>% 
  mutate(`log10 p` = format(round(-1 * log10(`log10 p`), 1), nsmall = 1))


most_concordant_genes_GO <- GO_results_wilcoxon$results %>% # most concordant
  filter(FWER_high_rank < 0.05) %>% 
  dplyr::select(node_id, node_name, raw_p_high_rank, FWER_high_rank)  %>% 
  dplyr::rename(GO = node_id, Meaning = node_name, `log10 p` = raw_p_high_rank, FWER = FWER_high_rank) %>% 
  mutate(`log10 p` = format(round(-1 * log10(`log10 p`), 1), nsmall = 1))

saveRDS(most_concordant_genes_GO, "data/derived/supp_table_GO_terms.rds")
```


#### GO enrichment among most concordant transcripts

***Table***: The table shows all the GO terms with a family wise error rate (FWER) threshold of FWER < 0.05, when testing for GO enrichment among genes whose transcripts had a high evidence ratio (indicating greater likelihood of being sexually concordant, as measured in the early-life fitness assay). GO terms related to gene expression and translation were enriched, which suggests that there is strong, sexually concordant selection on the expression levels of genes involved in gene expression and translation.


```{r}
most_concordant_genes_GO %>% 
  kable() %>% kable_styling(full_width = F)
```

#### GO enrichment among most anatagonistic transcripts

***Table***: The table shows all the GO terms with a Wilcoxon test p-value below 0.001, when testing for GO enrichment among genes whose transcripts had a low evidence ratio (indicating greater likelihood of being sexually antagonistic, as measured in the early-life fitness assay). However, no GO terms were significantly enriched among the most sexually antagonistic genes using family wise error rate (FWER) threshold of FWER < 0.05, and so these enrichment results may represent false positives due to multiple testing. 

```{r}
most_antagonistic_genes_GO %>% 
  kable() %>% kable_styling(full_width = F)
```
