---
title: "Transcriptome-wide association study (TWAS)"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r libraries, message=FALSE, warning=FALSE, results='hide'}
library(edgeR) # BiocManager::install("edgeR")
library(tidyverse) 
library(glue)
library(future)
library(future.apply)
library(parallel)
library(kableExtra)
library(DT)
library(ashr)
library(mashr)

options(stringsAsFactors = FALSE)

# Connect to the database of annotations
db <- DBI::dbConnect(RSQLite::SQLite(), "data/derived/annotations.sqlite3")

# Helper to run shell commands
run_command <- function(shell_command, wd = getwd(), path = ""){
  cat(system(glue("cd ", wd, path, "\n",shell_command), intern = TRUE), sep = '\n')
}

kable_table <- function(df) { # cool tables
  kable(df, "html") %>%
  kable_styling() %>%
  scroll_box(height = "300px")
}

my_data_table <- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c("Scroller",  "Buttons"),
    options = list(
      dom = 'Bfrtip',
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list('pageLength', 'colvis', 'csv', list(
          extend = 'pdf',
          pageSize = 'A4',
          orientation = 'landscape',
          filename = 'TWAS_enrichment')),
       columnDefs = list(list(targets = c(8,10), visible = FALSE)),
      pageLength = 50
    )
  )
}

# Helper to load Huang et al.'s data
load_expression_data <- function(sex, both_sex_chromosomes = TRUE){
  
  # Note: Huang et al's data contains weird stuff like supposedly Y-linked genes that have
  # higher/equal expression in *females* in all lines, presumably microarray issues/errors.
  # To be conservative, we restrict our analyses to genes that are known to be on a
  # chromosomes that is present in both sexes
  if(both_sex_chromosomes){
    genes_allowed <- tbl(db, "genes") %>%
      filter(chromosome %in% c("2L", "2R", "3L", "3R", "4", "X")) %>%
      pull(FBID)
  } else {
    genes_allowed <- tbl(db, "genes") %>%
      pull(FBID)
  }
  
  if(sex != "both"){
    expression <- glue("data/input/huang_transcriptome/dgrp.array.exp.{sex}.txt") %>% 
      read_delim(delim = " ") %>%
      filter(gene %in% genes_allowed)
    sample_names <- names(expression)[names(expression) != "gene"] %>% str_remove(":[12]") 
    gene_names <- expression$gene
    expression <- expression %>% select(-gene) %>% as.matrix() %>% t() 
    rownames(expression) <- sample_names # rows are samples, columns are genes
    colnames(expression) <- gene_names 
    return(expression %>% as.data.frame() %>% 
             tibble::rownames_to_column("line") %>% 
             as_tibble() %>%
             mutate(line = str_remove_all(line, "[.]1")))
  }
  
  females <- read_delim("data/input/huang_transcriptome/dgrp.array.exp.female.txt", delim = " ") %>%
    filter(gene %in% genes_allowed)
  names(females)[-1] <- paste("F_", names(females)[-1],sep="") #%>% str_remove(":[12]") 
  females <- females %>% 
    left_join(read_delim("data/input/huang_transcriptome/dgrp.array.exp.male.txt", delim = " "), by = "gene")
  sample_names <- names(females)[names(females) != "gene"] %>% str_remove(":[12]") 
  gene_names <- females$gene
  sex <- ifelse(str_detect(sample_names, "F_"), "female", "male")
  line <- str_remove_all(sample_names, "F_")
  females <- females %>% select(-gene) %>% t()
  colnames(females) <- gene_names
  list(
    sampleIDs = tibble(sex, line),
    expression = females
  )
}

# Table S5: heritability of the expression level of each transcript (as measured in males, or females)
# huang_heritability <- read_csv("data/input/huang_2015_tableS5_transcript_heritability.csv")

# Load the predicted line means from present study, as calculated in get_predicted_line_means.Rmd
predicted_line_means <- read_csv("data/derived/predicted_line_means.csv")
```

<!-- # Load the supplementary data files from Huang et al. 2015 PNAS -->
<!-- # Table S2: results of statistical tests for sex, line and sex-by-line effects on expression of each transcript -->
<!-- # huang_expression <- read_csv("data/input/huang_2015_tableS2_gene_expression.csv") -->

<!-- # # Table S11+S12: statistically significant eQTLs, and the transcripts they affect (for each sex) -->
<!-- # huang_eQTL_females <- read_csv("data/input/huang_2015_tableS11_eQTL_females.csv") %>% -->
<!-- #   left_join(tbl(db, "genes") %>% select(FBID, gene_name) %>% collect(), by = "FBID") %>% -->
<!-- #   rename(Affected_FBID = FBID, Affected_gene = gene_name) -->
<!-- #  -->
<!-- # huang_eQTL_males <- read_csv("data/input/huang_2015_tableS12_eQTL_males.csv") %>% -->
<!-- #   left_join(tbl(db, "genes") %>% select(FBID, gene_name) %>% collect(), by = "FBID") %>% -->
<!-- #   rename(Affected_FBID = FBID, Affected_gene = gene_name) -->
<!-- ``` -->

## Calculate sex bias in expression and average expression, for each transcript

The following takes the 368 female samples and the 369 male samples, and finds the log fold difference in expression between sexes, and the average expression across both sexes, using the `edgeR` package.
```{r results='hide', message=FALSE, warning=FALSE}

if(!file.exists("data/derived/gene_expression_by_sex.csv")){
expression_data_both_sexes <- load_expression_data("both") %>% unname()

voom_gene_data <- calcNormFactors(DGEList(t(expression_data_both_sexes[[2]])))
mm <- model.matrix(~ sex, data = expression_data_both_sexes[[1]])
colnames(mm) <- gsub("sex", "", colnames(mm))

sex_bias_in_expression <- voom_gene_data %>% 
  voom(mm, plot = FALSE) %>% 
  lmFit(mm) %>% 
  eBayes() %>%
  topTable(n = Inf) %>% 
  rownames_to_column("FBID") %>%
  select(FBID, logFC, AveExpr) %>% 
  rename(male_bias_in_expression = logFC) %>%
  as_tibble() %>% arrange(male_bias_in_expression)

write_csv(sex_bias_in_expression, "data/derived/gene_expression_by_sex.csv")

} else {
  sex_bias_in_expression <- read_csv("data/derived/gene_expression_by_sex.csv")
}
```


## Using linear models to find transcripts whose expression correlates with fitness across lines ("TWAS")

This analysis uses the expression data from Huang et al. (2015), which was downloaded from the [DGRP website](http://dgrp2.gnets.ncsu.edu/data.html). 

```{r echo=FALSE, message=FALSE, warning=FALSE}
n_transcripts <- ncol(load_expression_data("male", both_sex_chromosomes = TRUE))
n_tests <- n_transcripts * 4

n_transcripts <- prettyNum(n_transcripts, big.mark = ",", scientific = FALSE)
n_tests <- prettyNum(n_tests, big.mark = ",", scientific = FALSE)
```

Here, we perform a large number of simple linear regressions, and obtain the slope (beta or $\beta$), and the associated standard error, from a regression of transcript $i$ on fitness trait $j$. The number of regressions run was `r n_tests`, i.e. 4 fitness traits $\times$ `r n_transcripts` transcripts. This approach is often called a 'TWAS', i.e. transcriptome-wide association study. 

```{r message=FALSE}
transcript_selection_analysis <- function(expression_data, phenotypes){
  
  if("block" %in% names(phenotypes)) phenotypes <- phenotypes %>% select(-block)
  
  expression_data <- expression_data %>% 
    filter(line %in% phenotypes$line)
  
  # Find line mean expression for each gene (average across the c. 2 replicate samples per line) 
  chunk_cols <- split(2:ncol(expression_data), 
                      ceiling(seq_along(2:ncol(expression_data)) / 500))
  
  mean_expression_data <- mclapply(1:2, function(i){  
    expression_data[, c(1, chunk_cols[[i]])] %>% 
      group_by(line) %>% 
      summarise_all(mean) %>% 
      ungroup()
  }) %>% bind_rows()

  # Scale each transcript's expression level so that mean is 0, 
  # and the variance is 1, across all the lines measured by Huang et al.
  for(i in 2:ncol(mean_expression_data)) mean_expression_data[,i] <- as.numeric(scale(mean_expression_data[,i]))
  
  # Join the microarray data with the phenotypes (i.e. our fitness data), and 
  # keep only the lines where we have both sets of measurements
  expression_data <- phenotypes %>% left_join(expression_data, by = "line")
  expression_data <- expression_data[complete.cases(expression_data), ] %>% select(-line)
  
  print("Data ready for analysis. Starting TWAS...")

  # Create chunks of transcript names, which will be used to facilitate parallel processing
  transcripts <- names(expression_data)[-c(1:4)]
  transcripts <- split(transcripts, ceiling(seq_along(transcripts) / 100))
  
  # Define a function to run 4 linear models, and get the beta and SE 
  # for regressions of expression level on the 4 fitness traits
  do_one_transcript <- function(transcript){
    expression_level <- expression_data %>% pull(!!transcript)
    FE <- summary(lm(female.fitness.early ~ expression_level, data = expression_data))$coefficients
    FL <- summary(lm(female.fitness.late ~ expression_level, data = expression_data))$coefficients
    ME <- summary(lm(male.fitness.early ~ expression_level, data = expression_data))$coefficients
    ML <- summary(lm(male.fitness.late ~ expression_level, data = expression_data))$coefficients
    
    c(FE[2,1], FL[2,1], ME[2,1], ML[2,1], # effect size
      FE[2,2], FL[2,2], ME[2,2], ML[2,2], # SE
      FE[2,4], FL[2,4], ME[2,4], ML[2,4]) # p-value
  }
  
  # Runs do_one_transcript() on all the transcripts listed in the vector 'transcripts'
  do_chunk_of_transcripts <- function(transcripts){
    output <- data.frame(transcripts, lapply(transcripts, do_one_transcript) %>% do.call("rbind", .))
    names(output) <- c("gene", "beta_FE", "beta_FL", "beta_ME", "beta_ML", 
                       "SE_FE", "SE_FL", "SE_ME", "SE_ML",
                       "pval_FE", "pval_FL", "pval_ME", "pval_ML")
    output
  }
  
  # Run it all, in parallel
  transcripts %>% 
    mclapply(do_chunk_of_transcripts) %>% 
    do.call("rbind", .) %>% as_tibble() %>% mutate(gene = as.character(gene))
}

if(!file.exists("data/derived/TWAS/TWAS_result_males.csv")){

  TWAS_result_females <- load_expression_data("female") %>% 
    transcript_selection_analysis(predicted_line_means) 
  TWAS_result_females %>% write_csv("data/derived/TWAS/TWAS_result_females.csv")
  
  TWAS_result_males <- load_expression_data("male") %>% 
    transcript_selection_analysis(predicted_line_means)
  TWAS_result_males %>% write_csv("data/derived/TWAS/TWAS_result_males.csv")
  
} else {
  TWAS_result_females <- read_csv("data/derived/TWAS/TWAS_result_females.csv")
  TWAS_result_males <- read_csv("data/derived/TWAS/TWAS_result_males.csv")
}
```

### Run `mashr` to adjust the TWAS results 

```{r, include=FALSE}
run_mashr <- function(beta_and_se, mashr_mode, ED_p_cutoff = NULL){
  
  mashr_setup <- function(beta_and_se){
    betas <- beta_and_se %>% select(starts_with("beta")) %>% as.matrix()
    SEs <- beta_and_se %>% select(starts_with("SE")) %>% as.matrix()
    rownames(betas) <- beta_and_se$SNP
    rownames(SEs) <- beta_and_se$SNP
    mash_set_data(betas, SEs)
  }
  
  mash_data <- mashr_setup(beta_and_se)
  
  # Setting mashr_mode == "ED" makes mashr choose the covariance matrices for us, using the
  # software Extreme Deconvolution. This software "reconstructs the error-deconvolved or 'underlying' 
  # distribution function common to all samples, even when the individual data points are samples from different distributions"
  # Following the mashr vignette, we initialise the algorithm in ED using the principal components of the strongest effects in the dataset
  # Reference for ED: https://arxiv.org/abs/0905.2979
  if(mashr_mode == "ED"){
    # Find the strongest effects in the data
    m.1by1 <- mash_1by1(mash_data) 
    strong <- get_significant_results(m.1by1, thresh = 1)   
    # Obtain data-driven covariance matrices by running Extreme Deconvolution
    U.pca <- cov_pca(mash_data, npc = 4, subset = strong)
    U <- cov_ed(mash_data, U.pca, subset = strong)
  }
  
  # Otherwise, we define the covariance matrices ourselves (a long list of a priori interesting matrices are checked)
  if(mashr_mode == "canonical"){
    make_SA_matrix <- function(r) matrix(c(1,1,r,r,1,1,r,r,r,r,1,1,r,r,1,1), ncol=4)
    make_age_antag_matrix <- function(r) matrix(c(1,r,1,r,r,1,r,1,1,r,1,r,r,1,r,1), ncol=4)
    make_sex_specific <- function(mat, sex){
      if(sex == "F") {mat[, 3:4] <- 0;  mat[3:4, ] <- 0}
      if(sex == "M") {mat[, 1:2] <- 0; mat[1:2, ] <- 0}
      mat
    }
    make_age_specific <- function(mat, age){
      if(age == "early") {mat[, c(2,4)] <- 0; mat[c(2,4), ] <- 0}
      if(age == "late")  {mat[, c(1,3)] <- 0; mat[c(1,3), ] <- 0}
      mat
    }
    
    add_matrix <- function(mat, mat_list, name){
      mat_list[[length(mat_list) + 1]] <- mat
      names(mat_list)[length(mat_list)] <- name
      mat_list
    }
    id_matrix <- matrix(1, ncol=4, nrow=4)
    
    # Get the mashr default canonical covariance matrices: this includes the ones 
    # called "null", "uniform", and "same sign" in the list that precedes this code chunk
    U <- cov_canonical(mash_data)
    
    # And now our custom covariance matrices: 
    
    # Identical across ages, but sex-antagonistic
    U <- make_SA_matrix(-0.25) %>% add_matrix(U, "Sex_antag_0.25")   
    U <- make_SA_matrix(-0.5) %>% add_matrix(U, "Sex_antag_0.5")
    U <- make_SA_matrix(-0.75) %>% add_matrix(U, "Sex_antag_0.75")   
    U <- make_SA_matrix(-1) %>% add_matrix(U, "Sex_antag_1.0")       
    
    # Identical across sexes, but age-antagonistic
    U <- make_age_antag_matrix(-0.25) %>% add_matrix(U, "Age_antag_0.25")
    U <- make_age_antag_matrix(-0.5) %>% add_matrix(U, "Age_antag_0.5")
    U <- make_age_antag_matrix(-0.75) %>% add_matrix(U, "Age_antag_0.75")
    U <- make_age_antag_matrix(-1) %>% add_matrix(U, "Age_antag_1.0")
    
    # Sex-specific, identical effect in young and old
    U <- id_matrix %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_1")
    U <- id_matrix %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_1")
    
    # Age-specific, identical effect in males and females
    U <- id_matrix %>% make_age_specific("early") %>% add_matrix(U, "Early_life_specific_1")
    U <- id_matrix %>% make_age_specific("late")  %>% add_matrix(U, "Late_life_specific_1")
    
    # Positively correlated but variable effect across ages, and also sex-specific
    U <- make_age_antag_matrix(0.25) %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_0.25")
    U <- make_age_antag_matrix(0.5) %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_0.5")
    U <- make_age_antag_matrix(0.75) %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_0.75")
    U <- make_age_antag_matrix(0.25) %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_0.25")
    U <- make_age_antag_matrix(0.5) %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_0.5")
    U <- make_age_antag_matrix(0.75) %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_0.75")
    
    # Nwegatively correlated across ages, and also sex-specific
    U <- make_age_antag_matrix(-0.25) %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_age_antag_0.25")
    U <- make_age_antag_matrix(-0.5) %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_age_antag_0.5")
    U <- make_age_antag_matrix(-0.75) %>% make_sex_specific("F") %>% add_matrix(U, "Female_specific_age_antag_0.75")
    U <- make_age_antag_matrix(-0.25) %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_age_antag_0.25")
    U <- make_age_antag_matrix(-0.5) %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_age_antag_0.5")
    U <- make_age_antag_matrix(-0.75) %>% make_sex_specific("M") %>% add_matrix(U, "Male_specific_age_antag_0.75")
    
    # Positively correlated but variable effect across sexes, and also age-specific
    U <- make_SA_matrix(0.25) %>% make_age_specific("early") %>% add_matrix(U, "Early_life_specific_0.25")
    U <- make_SA_matrix(0.5) %>% make_age_specific("early") %>% add_matrix(U, "Early_life_specific_0.5")
    U <- make_SA_matrix(0.75) %>% make_age_specific("early") %>% add_matrix(U, "Early_life_specific_0.75")
    U <- make_SA_matrix(0.25) %>% make_age_specific("late") %>% add_matrix(U, "Late_life_specific_0.25")
    U <- make_SA_matrix(0.5) %>% make_age_specific("late") %>% add_matrix(U, "Late_life_specific_0.5")
    U <- make_SA_matrix(0.75) %>% make_age_specific("late") %>% add_matrix(U, "Late_life_specific_0.75")
    
    # Negatively correlated but variable effect across sexes, and also age-specific
    U <- make_SA_matrix(-0.25) %>% make_age_specific("early") %>% add_matrix(U, "Early_life_antag_0.25")
    U <- make_SA_matrix(-0.5) %>% make_age_specific("early") %>% add_matrix(U, "Early_life_antag_0.5")
    U <- make_SA_matrix(-0.75) %>% make_age_specific("early") %>% add_matrix(U, "Early_life_antag_0.75")
    U <- make_SA_matrix(-0.25) %>% make_age_specific("late") %>% add_matrix(U, "Late_life_antag_0.25")
    U <- make_SA_matrix(-0.5) %>% make_age_specific("late") %>% add_matrix(U, "Late_life_antag_0.5")
    U <- make_SA_matrix(-0.75) %>% make_age_specific("late") %>% add_matrix(U, "Late_life_antag_0.75")
  }
  
  return(mash(data = mash_data, Ulist = U)) # Run mashr
}
```

This section re-uses the custom function `run_mashr()`. See the earlier script (where `mashr` was applied to the GWAS data) for the function definition. As the GWAS data, we use `mashr`'s data-driven mode to derive adjusted (shrinked) estimates of beta (i.e. the slope of the regression of transcript abundance on the phenotype of interest), which are sensitive to the covariance structure in the data (which is estimated from the data). As well as producing these adjusted estimates, we derive the local false sign rate for each combination of transcript and fitness trait (used later to calculate evidence ratios).

```{r message=FALSE}
if(!file.exists("data/derived/TWAS/TWAS_ED.rds")){
  
  input_data <- data.frame(TWAS_result_females[,1:3], 
                           TWAS_result_males[,4:5], 
                           TWAS_result_females[,6:7], 
                           TWAS_result_males[,8:9])
  
  TWAS_ED <- input_data %>% 
    run_mashr(mashr_mode = "ED", 
              ED_p_cutoff = 0.4) 
  
  saveRDS(TWAS_ED, "data/derived/TWAS/TWAS_ED.rds")
  
} else {
  TWAS_ED <- readRDS("data/derived/TWAS/TWAS_ED.rds")
}
```


<!-- ### Find mixture probabilities for each transcript -->
<!-- This uses the canonical analysis' classifications. Each transcript gets a posterior probability that it belongs to the $i$'th mixture component -- only the mixture components that are not very rare are included. These are: `equal_effects` (i.e. the transcript is predicted to affect all 4 phenotypes equally), female-specific and male-specific (i.e. an effect on females/males only, which is concordant across age categories), sexually antagonistic (again, regardless of age), and null (note: the prior for `mashr` was that null transcripts are $10\times$ more common than any other type).  -->

<!-- ```{r get_mixture_assignments} -->
<!-- # Get the mixture weights, as advised by mashr authors here: https://github.com/stephenslab/mashr/issues/68 -->
<!-- posterior_weights_cov <- TWAS_canonical$posterior_weights  -->
<!-- colnames(posterior_weights_cov) <- sapply( -->
<!--   str_split(colnames(posterior_weights_cov), '\\.'),  -->
<!--   function(x) { -->
<!--     if(length(x) == 1) return(x) -->
<!--     else if(length(x) == 2) return(x[1]) -->
<!--     else if(length(x) == 3) return(paste(x[1], x[2], sep = ".")) -->
<!--   }) -->
<!-- posterior_weights_cov <- t(rowsum(t(posterior_weights_cov),  -->
<!--                                   colnames(posterior_weights_cov))) -->

<!-- # Make a neat dataframe -->
<!-- TWAS_mixture_assignment_probabilities <- data.frame( -->
<!--   FBID = TWAS_mashr_results$FBID, -->
<!--   posterior_weights_cov, -->
<!--   stringsAsFactors = FALSE -->
<!-- ) %>% as_tibble() %>% -->
<!--   rename(P_equal_effects = equal_effects, -->
<!--          P_female_specific = Female_specific_1, -->
<!--          P_male_specific = Male_specific_1, -->
<!--          P_null = null, -->
<!--          P_sex_antag = Sex_antag_0.25) -->

<!-- # TWAS_mixture_assignment_probabilities <- TWAS_mixture_assignment_probabilities %>% -->
<!-- #   left_join(huang_heritability, by = "FBID") -->

<!-- saveRDS(TWAS_mixture_assignment_probabilities,  -->
<!--         "data/derived/TWAS/TWAS_mixture_assignment_probabilities.rds") -->
<!-- ``` -->


## Tidy and save the results of TWAS

```{r}
TWAS_results <- data.frame(
  FBID = TWAS_result_females$gene,
  as.data.frame(get_pm(TWAS_ED)),
  as.data.frame(get_lfsr(TWAS_ED))) %>% 
  as_tibble()

names(TWAS_results)[6:9] <- paste(
  "LFSR", c("FE", "FL", "ME", "ML"), sep = "_")

TWAS_results <- TWAS_results  %>%
  left_join(tbl(db, "genes") %>% 
              select(FBID, gene_name, chromosome) %>% 
              collect(), by = "FBID") %>% 
  left_join(sex_bias_in_expression, by = "FBID") %>%
  mutate(across(where(is.numeric), ~ round(.x, 2))) %>%
  as_tibble() %>%
  distinct()

TWAS_results %>%
  write_csv("data/derived/TWAS/TWAS_results.csv")
```

