---
title: "Tables summarising the TWAS results"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r libraries, message=FALSE, warning=FALSE, results='hide'}
library(tidyverse)
library(DT)
library(kableExtra)

kable_table <- function(df) { # cool tables
  kable(df, "html") %>%
  kable_styling(full_width = FALSE) %>%
  scroll_box(height = "300px")
}

my_data_table <- function(df){ # Make html tables:
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c("Scroller",  "Buttons"),
    options = list(
      dom = 'Bfrtip',
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list('pageLength', 'colvis', 'csv', list(
          extend = 'pdf',
          pageSize = 'A4',
          orientation = 'landscape',
          filename = 'TWAS_sig_genes')),
      pageLength = 50
    )
  )
}

sig_transcripts_table <- read_csv("data/derived/TWAS/TWAS_results.csv") %>%
   filter(LFSR_FE < 0.01 | LFSR_FL < 0.01 | LFSR_ME < 0.01 | LFSR_ML < 0.01) 

tallies <- sig_transcripts_table %>%
  select(starts_with("sig_")) %>%
  summarise_all(~ sum(!is.na(.x))) %>%
  gather(type, n_transcripts) %>%
  arrange(-n_transcripts)

rnd<- function(x) format(round(x, 2), nsmall = 2)


big_table <- sig_transcripts_table %>% 
  select(FBID, gene_name, chromosome, AveExpr, male_bias_in_expression, everything()) %>%
  select(-contains("heritability")) %>%
  mutate(beta_FE = paste(beta_FE, " (", rnd(LFSR_FE), ")", sep = ""),
         beta_FL = paste(beta_FL, " (", rnd(LFSR_FL), ")", sep = ""),
         beta_ME = paste(beta_ME, " (", rnd(LFSR_ME), ")", sep = ""),
         beta_ML = paste(beta_ML, " (", rnd(LFSR_ML), ")", sep = "")) %>%
  select(-contains("LFSR")) %>%
  rename(`Gene name` = gene_name,
         Chromosome = chromosome,
         `Female early` = beta_FE,
         `Female late` = beta_FL,
         `Male early` = beta_ME,
         `Male late` = beta_ML,
         `Male bias in expression (logFC)` = male_bias_in_expression,
         `Average expression level` = AveExpr) %>%
  arrange(-P_sex_antag)
```



## Table showing the significant transcripts

The table shows the `r nrow(big_table)` transcripts that were associated with one or more of the four phenotypes, with a local false sign rate < 0.01 (LFSR; computed by `mashr`). Columns 5-8 show the  regression coefficients (adjusted with `mashr` in data-driven mode) that relate the line mean transcript abundance to the line mean fitnesses, while columns 9-12 give the local false sign rates that for these coefficients from `mashr`. The `sig_` (signficant) columns provide a binary summary of whether the transcript was significantly pleiotropic between ages and sex classes (see below). Finally, the last five columns give the mixture assignment probabilities from `mashr` (this time, run in canonical mode). 

To make the 'significance' columns, we defined significantly antagonistic transcripts as those where the relationship with fitness is significantly positive for one sex (or age class) and significantly negative for the other, with LFSR < 0.01. Similarly, we define significantly concordant transcripts as those where the relationship with fitness is significantly positive for one sex or age class and also significantly positive for the other (LFSR < 0.01). This is quite conservative, because a transcript needs to have a LFSR < 0.01 for two tests, giving two chances for a 'false negative'. One can sort the table by these columns, for example to see a list of transcripts that showed a significantly antagonistic relationship with fitness in the early-life fitness assays, sort by the `sig_SA_early` column.

Abbreviations: AC = age concordant, SC = sexually concordant, SA = sexually antagonistic. 

```{r}
big_table %>% my_data_table()
```


### Count the numbers of significant genes
Abbreviations: AC = age concordant, SC = sexually concordant, SA = sexually antagonistic. We defined significantly antagonistic transcripts as those where the relationship with fitness is significantly positive for one sex (or age class) and significantly negative for the other, with LFSR < 0.01. Similarly, we define significantly concordant transcripts as those where the relationship with fitness is significantly positive for one sex or age class and also significantly positive for the other (LFSR < 0.01). This is quite conservative, because a transcript needs to have a LFSR < 0.01 for two tests, giving two chances for a 'false negative'.
```{r}
tallies %>%
  mutate(x = str_remove_all(type, "sig_"),
         `Number statistically significant` = n_transcripts) %>%
  mutate(x = str_replace_all(x, "AC", "Age concordant"),
         x = str_replace_all(x, "AA", "Age antagonistic"),
         x = str_replace_all(x, "SC", "Sexually concordant"),
         x = str_replace_all(x, "SA", "Sexually antagonistic"),
         x = str_replace_all(x, "_females", " (in females)"),
         x = str_replace_all(x, "_males", " (in males)"),
         x = str_replace_all(x, "_early", " (in early life)"),
         x = str_replace_all(x, "_late", " (in late life)")) %>%
  rename(`Type of transcript` = x) %>%
  select(`Type of transcript`, `Number statistically significant`) %>%
  kable_table()
```
